{
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Data collection of PubMed Central full text articles"
      ],
      "metadata": {
        "id": "0dEbrcihk0zD"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 1. Introduction\n",
        "\n",
        "Europe PubMed Central (Europe PMC) is an open-access repository of biomedical research that contains ~45 million abstracts and ~10 million full text articles, including research articles, preprints, micropublications, books, reviews, and protocols.\n",
        "\n",
        " Data will be collected from [Europe PMC](https://europepmc.org/) for the PubMed Central index of full text scientific papers via the [Articles RESTful API](https://europepmc.org/RestfulWebService). The  query used for the project relates to COVID-19 and drug repurposing from 2019-2022."
      ],
      "metadata": {
        "id": "ByCbTf63m-GQ"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 2. Install/import libraries"
      ],
      "metadata": {
        "id": "-KN0ZQjInJsA"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install diskcache"
      ],
      "metadata": {
        "id": "cgI6DR-GP9XL"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2EtQf2j7ucoS"
      },
      "outputs": [],
      "source": [
        "import requests\n",
        "import urllib\n",
        "import urllib.parse as urlparse\n",
        "import json\n",
        "import pandas as pd\n",
        "import time\n",
        "import pickle\n",
        "import re\n",
        "import concurrent\n",
        "\n",
        "\n",
        "from collections import Counter\n",
        "from diskcache import Cache\n",
        "from multiprocessing.pool import ThreadPool\n",
        "from bs4 import BeautifulSoup, NavigableString, Tag\n",
        "from itertools import chain"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 3. Download metadata\n",
        "\n",
        "Construct a function to download metadata using the RESTful web service [search module](https://europepmc.org/RestfulWebService#!/Europe32PMC32Articles32RESTful32API/search) to search the publications database for the following fields: 'pmcid', 'published', 'revised', 'title', 'journal', 'authors', 'doi', 'pdf_url', and convert the dictionary to a DataFrame.\n",
        "\n",
        "The PubMed Central Identifier (pmcid) has been chosen as this is only returned if the full text is available in Europe PMC as opposed to the PubMed Identifier (pmid) which links to abstracts.\n",
        "\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "2Wjxkp3NwdWa"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Adapted from https://github.com/carrlucy/HSL_OA/blob/main/streamlit_app.py\n",
        "\n",
        "def get_search_results(query):\n",
        "    dct = {}\n",
        "    for col in ['pmcid','published','revised','title','journal','authors','doi', 'pdf_url']:\n",
        "        dct[col] = []\n",
        "\n",
        "    cr_mrk= '' #current cursor mark\n",
        "    nxt_mrk = '*' #next cursor mark\n",
        "    while cr_mrk != nxt_mrk:\n",
        "        url = 'https://www.ebi.ac.uk/europepmc/webservices/rest/search?'\n",
        "        query = query\n",
        "        params = {'query':query, 'resultType':'core','synonym':'TRUE','cursorMark':nxt_mrk,'pageSize':'1000','format':'json'}\n",
        "        response = requests.get(url,params)\n",
        "        rjson = response.json()\n",
        "        cr_mrk = urlparse.unquote(rjson['request']['cursorMark'])\n",
        "        nxt_mrk = urlparse.unquote(rjson['nextCursorMark'])\n",
        "        for rslt in rjson['resultList']['result']:\n",
        "            dct['pmcid'].append(rslt['pmcid']) if 'pmcid' in rslt.keys() else dct['pmcid'].append(0)\n",
        "            dct['published'].append(rslt['firstPublicationDate']) if 'firstPublicationDate' in rslt.keys() else dct['published'].append(0)\n",
        "            dct['revised'].append(rslt['dateOfRevision']) if 'dateOfRevision' in rslt.keys() else dct['revised'].append(0)\n",
        "            dct['title'].append(rslt['title']) if 'title' in rslt.keys() else dct['title'].append(0)\n",
        "            dct['journal'].append(rslt['journalInfo']['journal']['title']) if 'journalInfo' in rslt.keys() else dct['journal'].append(0)\n",
        "            dct['authors'].append(rslt['authorString']) if 'authorString' in rslt.keys() else dct['authors'].append(0)\n",
        "            dct['doi'].append(rslt['doi']) if 'doi' in rslt.keys() else dct['doi'].append(0)\n",
        "            dct['pdf_url'].append(f\"https://europepmc.org/articles/{rslt['pmcid']}?pdf=render\") if 'pmcid' in rslt.keys() else dct['pdf_url'].append(0)\n",
        "\n",
        "    df=pd.DataFrame.from_dict(dct, orient='columns')\n",
        "    return df"
      ],
      "metadata": {
        "id": "BN07JbF3ewhB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 3.1 Construct search query\n",
        "\n",
        "The direct search syntax allows for boolean operators AND, OR and NOT, as well as phrases inside double quotes, which can be used along with field searches to construct more complex combinations.\n",
        "\n",
        "Field names IN_EPMC, HAS_FT and OPEN_ACCESS are used to return full text Open Access publications in Europe PMC. The search can also be limited by PUB_TYPE. Here Journal Article has been chosen - the [most common and general publication type](https://www.nlm.nih.gov/bsd/indexing/training/PUB_040.html) used for original full text research, review or other reports published in a journal. Research Article has also been selected. See the [Advanced search](https://europepmc.org/advancesearch) page for a full list of publication types.\n",
        "\n",
        "The [Search syntax reference](https://europepmc.org/searchsyntax) provides more information on query syntax and search field combinations with examples.\n",
        "\n",
        "Refer also the [Web Service Reference](https://europepmc.org/docs/EBI_Europe_PMC_Web_Service_Reference.pdf) for method parameters and output elements.\n",
        "\n"
      ],
      "metadata": {
        "id": "xbeGMZ0NO9td"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "query = '(covid OR coronavirus OR sars-cov-2) AND (\"drug discovery\" OR \"drug repurposing\" OR \"drug repositioning\")\\\n",
        " AND (IN_EPMC:Y) AND (HAS_FT:Y) AND (OPEN_ACCESS:Y) AND (PUB_TYPE:\"Journal Article\" OR PUB_TYPE:\"Research-article\")\\\n",
        "  NOT (PUB_TYPE:\"Abstract\" OR PUB_TYPE:\"News\" OR PUB_TYPE:\"Editorial\" OR PUB_TYPE:\"Letter\")\\\n",
        "   AND (FIRST_PDATE:[2019-01-01 TO 2022-12-31])'\n",
        "search_results = get_search_results(query)\n",
        "search_results"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "3RWmnQXiO85e",
        "outputId": "2a14085b-2503-42c3-de92-ecae1c3a7fde"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "0      PMC9549161  2022-09-26  2022-10-14   \n",
              "1      PMC9539342  2022-09-22  2022-11-12   \n",
              "2      PMC9357751  2022-12-01  2022-12-05   \n",
              "3      PMC9346052  2022-08-03  2022-09-05   \n",
              "4      PMC9775208  2022-12-15  2022-12-25   \n",
              "...           ...         ...         ...   \n",
              "11411  PMC6328940  2019-01-01  2020-03-09   \n",
              "11412  PMC6598402  2019-06-21  2020-09-28   \n",
              "11413  PMC6481739  2019-02-05  2020-09-28   \n",
              "11414  PMC6624471  2019-07-05  2020-09-28   \n",
              "11415  PMC6409730  2019-02-06  2020-09-30   \n",
              "\n",
              "                                                   title  \\\n",
              "0           Drug repositioning: A bibliometric analysis.   \n",
              "1      A review on computer-aided chemogenomics and d...   \n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "3      Scope of repurposed drugs against the potentia...   \n",
              "4      Drug Repurposing Using Gene Co-Expression and ...   \n",
              "...                                                  ...   \n",
              "11411  β-RA reduces DMQ/CoQ ratio and rescues the enc...   \n",
              "11412  Alzheimer Disease Pathogenesis: Insights From ...   \n",
              "11413  Modeling cardiac complexity: Advancements in m...   \n",
              "11414  Tissue Response to Neural Implants: The Use of...   \n",
              "11415  24th International Symposium on Infections in ...   \n",
              "\n",
              "                                                 journal  \\\n",
              "0                              Frontiers in pharmacology   \n",
              "1                         Chemical biology & drug design   \n",
              "2      Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                                   Structural chemistry   \n",
              "4                                                Biology   \n",
              "...                                                  ...   \n",
              "11411                            EMBO molecular medicine   \n",
              "11412                          Frontiers in neuroscience   \n",
              "11413                                 APL bioengineering   \n",
              "11414                          Frontiers in neuroscience   \n",
              "11415              Medical sciences (Basel, Switzerland)   \n",
              "\n",
              "                                                 authors  \\\n",
              "0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                      0   \n",
              "3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                    Mailem RC, Tayo LL.   \n",
              "...                                                  ...   \n",
              "11411  Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...   \n",
              "11412                                Chen XQ, Mobley WC.   \n",
              "11413  Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...   \n",
              "11414       Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.   \n",
              "11415  Artigas A, Carlet J, Garnacho J, Niederman M, ...   \n",
              "\n",
              "                              doi  \\\n",
              "0       10.3389/fphar.2022.974849   \n",
              "1              10.1111/cbdd.14136   \n",
              "2                               0   \n",
              "3      10.1007/s11224-022-02020-z   \n",
              "4         10.3390/biology11121827   \n",
              "...                           ...   \n",
              "11411     10.15252/emmm.201809466   \n",
              "11412    10.3389/fnins.2019.00659   \n",
              "11413           10.1063/1.5055873   \n",
              "11414    10.3389/fnins.2019.00689   \n",
              "11415       10.3390/medsci7020023   \n",
              "\n",
              "                                                 pdf_url  \n",
              "0      https://europepmc.org/articles/PMC9549161?pdf=...  \n",
              "1      https://europepmc.org/articles/PMC9539342?pdf=...  \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...  \n",
              "3      https://europepmc.org/articles/PMC9346052?pdf=...  \n",
              "4      https://europepmc.org/articles/PMC9775208?pdf=...  \n",
              "...                                                  ...  \n",
              "11411  https://europepmc.org/articles/PMC6328940?pdf=...  \n",
              "11412  https://europepmc.org/articles/PMC6598402?pdf=...  \n",
              "11413  https://europepmc.org/articles/PMC6481739?pdf=...  \n",
              "11414  https://europepmc.org/articles/PMC6624471?pdf=...  \n",
              "11415  https://europepmc.org/articles/PMC6409730?pdf=...  \n",
              "\n",
              "[11416 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-f8d3fa8e-e9f2-4dec-8363-988766770a4a\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11411</th>\n",
              "      <td>PMC6328940</td>\n",
              "      <td>2019-01-01</td>\n",
              "      <td>2020-03-09</td>\n",
              "      <td>β-RA reduces DMQ/CoQ ratio and rescues the enc...</td>\n",
              "      <td>EMBO molecular medicine</td>\n",
              "      <td>Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...</td>\n",
              "      <td>10.15252/emmm.201809466</td>\n",
              "      <td>https://europepmc.org/articles/PMC6328940?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11412</th>\n",
              "      <td>PMC6598402</td>\n",
              "      <td>2019-06-21</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Alzheimer Disease Pathogenesis: Insights From ...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Chen XQ, Mobley WC.</td>\n",
              "      <td>10.3389/fnins.2019.00659</td>\n",
              "      <td>https://europepmc.org/articles/PMC6598402?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11413</th>\n",
              "      <td>PMC6481739</td>\n",
              "      <td>2019-02-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Modeling cardiac complexity: Advancements in m...</td>\n",
              "      <td>APL bioengineering</td>\n",
              "      <td>Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...</td>\n",
              "      <td>10.1063/1.5055873</td>\n",
              "      <td>https://europepmc.org/articles/PMC6481739?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11414</th>\n",
              "      <td>PMC6624471</td>\n",
              "      <td>2019-07-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Tissue Response to Neural Implants: The Use of...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.</td>\n",
              "      <td>10.3389/fnins.2019.00689</td>\n",
              "      <td>https://europepmc.org/articles/PMC6624471?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11415</th>\n",
              "      <td>PMC6409730</td>\n",
              "      <td>2019-02-06</td>\n",
              "      <td>2020-09-30</td>\n",
              "      <td>24th International Symposium on Infections in ...</td>\n",
              "      <td>Medical sciences (Basel, Switzerland)</td>\n",
              "      <td>Artigas A, Carlet J, Garnacho J, Niederman M, ...</td>\n",
              "      <td>10.3390/medsci7020023</td>\n",
              "      <td>https://europepmc.org/articles/PMC6409730?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11416 rows × 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-f8d3fa8e-e9f2-4dec-8363-988766770a4a')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-f8d3fa8e-e9f2-4dec-8363-988766770a4a button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-f8d3fa8e-e9f2-4dec-8363-988766770a4a');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 5
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Time to complete was 2m 18s for 11,416 rows of metadata."
      ],
      "metadata": {
        "id": "4a9LALR0B_0j"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_europepmc_df_json_ft_urls_all.pickle', 'wb') as f:\n",
        "  pickle.dump(search_results, f)"
      ],
      "metadata": {
        "id": "y4VCjS5zRo2P"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 4. Data cleaning\n",
        "\n",
        "Basic data cleaning to identify and remove missing values, duplicates, unwanted categories etc."
      ],
      "metadata": {
        "id": "ym8-j22TChmQ"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "PESjO792IcOw"
      },
      "source": [
        "### 4.1 Rows with missing authors\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "search_results.loc[search_results['authors']==0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "hxRyIbZwI-nP",
        "outputId": "a8d276f7-d5c9-4b0e-999b-6ec322c2db14"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "2      PMC9357751  2022-12-01  2022-12-05   \n",
              "287    PMC9803264  2022-12-30  2023-01-06   \n",
              "334    PMC9428111  2022-08-31  2022-09-30   \n",
              "361    PMC9548488  2022-11-01  2022-11-25   \n",
              "362    PMC9798891  2022-08-26  2023-01-05   \n",
              "397    PMC9711896  2022-12-01  2022-12-05   \n",
              "438    PMC9762046  2022-12-19  2023-01-04   \n",
              "590    PMC9812491  2022-01-01  2023-01-06   \n",
              "917    PMC9452103  2022-08-17  2022-09-12   \n",
              "986    PMC9273052  2022-07-11  2022-07-18   \n",
              "1417   PMC9615638  2022-09-28  2022-11-08   \n",
              "1753   PMC8891120  2022-03-03  2022-09-16   \n",
              "1775   PMC9364697  2022-07-06  2022-10-06   \n",
              "1993   PMC9727096  2022-01-01  2022-12-15   \n",
              "3778   PMC9116973  2022-01-01  2022-08-06   \n",
              "4522   PMC9466301  2022-10-01  2022-10-25   \n",
              "4556   PMC8683579  2021-06-13  2022-12-19   \n",
              "4563   PMC8158979  2021-05-01  2022-02-10   \n",
              "4979   PMC8831917  2021-07-15  2022-10-04   \n",
              "5155   PMC8547798  2021-01-01  2023-01-04   \n",
              "5385   PMC7836883  2021-01-12  2022-02-10   \n",
              "5705   PMC7713544  2020-12-01  2020-12-24   \n",
              "5880   PMC9066232  2022-06-01  2022-06-08   \n",
              "6088   PMC7280571  2020-05-22  2020-06-15   \n",
              "6140   PMC9098571  2022-05-13  2022-06-22   \n",
              "6467   PMC8592013  2021-11-11  2022-04-01   \n",
              "6541   PMC9253443  2022-06-16  2022-07-13   \n",
              "6543   PMC8135843  2021-05-03  2021-05-24   \n",
              "6666   PMC8653278  2021-10-08  2022-02-12   \n",
              "6697   PMC7280709  2020-05-22  2020-06-15   \n",
              "6757   PMC8882381  2022-01-01  2022-03-02   \n",
              "6972   PMC7280669  2020-05-22  2020-06-15   \n",
              "7021   PMC7538618  2020-01-01  2020-10-17   \n",
              "7040   PMC7247787  2020-05-25  2020-09-28   \n",
              "7154   PMC8679246  2021-12-17  2022-04-07   \n",
              "7183   PMC9756308  2022-12-01  2022-12-22   \n",
              "7810   PMC9812568  2022-01-01  2023-01-06   \n",
              "7897   PMC9128831  2021-04-26  2022-06-20   \n",
              "7955   PMC8656214  2021-12-09  2022-09-01   \n",
              "8336   PMC7370905  2020-07-20  2020-09-28   \n",
              "8637   PMC9730458  2022-11-17  2022-12-15   \n",
              "8689   PMC8875429  2022-01-26  2022-03-02   \n",
              "8691   PMC7162068  2020-03-01  2020-04-21   \n",
              "8797   PMC9748101  2022-01-01  2022-12-23   \n",
              "8838   PMC8004045  2021-03-01  2021-04-01   \n",
              "8909   PMC8445810  2021-09-12  2021-10-21   \n",
              "8983   PMC7185951  2020-04-27  2020-09-28   \n",
              "9372   PMC8004658  2021-03-01  2021-04-01   \n",
              "9537   PMC7474829  2020-01-01  2020-12-28   \n",
              "9812   PMC7162151  2020-02-28  2020-04-21   \n",
              "10167  PMC7020986  2020-03-01  2021-02-02   \n",
              "10415  PMC7492056  2020-09-15  2020-09-28   \n",
              "11155  PMC8015682  2021-03-26  2021-04-15   \n",
              "11195  PMC7702993  2019-04-08  2020-12-16   \n",
              "\n",
              "                                                   title  \\\n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "287    DFT investigations and molecular docking as po...   \n",
              "334    Evaluation of flavonoids as potential inhibito...   \n",
              "361    Language models for the prediction of SARS-CoV...   \n",
              "362    Identification and semisynthesis of (−)-anisom...   \n",
              "397    Potential of vibrational spectroscopy coupled ...   \n",
              "438    COVID-19 infection and metabolic comorbidities...   \n",
              "590    Drug repositioning based on heterogeneous netw...   \n",
              "917    SARS-CoV-2 immunity and vaccine strategies in ...   \n",
              "986                                         Cover Story.   \n",
              "1417   Atti 55° Congresso Nazionale SItI Padova, 28 s...   \n",
              "1753   CEBIT screening for inhibitors of the interact...   \n",
              "1775   3<sup>a</sup> edizione Giornate della ricerca ...   \n",
              "1993   Discovery of the key active compounds in Citri...   \n",
              "3778   Emerging technologies for the management of CO...   \n",
              "4522   A tissue-engineered artificial human thymus fr...   \n",
              "4556   Bioeconomy during the COVID-19 and perspective...   \n",
              "4563   Increased complement activation is a distincti...   \n",
              "4979   Artificial Intelligence and COVID-19: A System...   \n",
              "5155   In silico screening of antiviral compounds fro...   \n",
              "5385   Drug targets, mechanisms of drug action, and t...   \n",
              "5705   Scientific Collaboration and Communication in ...   \n",
              "5880   Can employers mandate their employees receive ...   \n",
              "6088                                               News.   \n",
              "6140   COVID vision: An integrated face mask detector...   \n",
              "6467                            First person - Li Wang.    \n",
              "6541   ADDRESSING VACCINE INEQUITY DURING THE COVID-1...   \n",
              "6543   Compulsory licence and Crown use provisions in...   \n",
              "6666   SUDs increase risk for breakthrough COVID‐19 i...   \n",
              "6697                   SCI's corporate partners respond.   \n",
              "6757   When speed matters: a discussion on the benefi...   \n",
              "6972                                               News.   \n",
              "7021   Bibliometric and Visualization Analysis of Hum...   \n",
              "7040                                     Full Issue PDF.   \n",
              "7154   Program Abstracts from The GSA 2021 Annual Sci...   \n",
              "7183   Identification of Sex-Specific Genetic Variant...   \n",
              "7810   Exploring the effects of drug, disease, and pr...   \n",
              "7897   GlobeChain: An Interoperable Blockchain for Gl...   \n",
              "7955        Face mask recognition system using CNN model   \n",
              "8336   Announcements of the Phytochemical Society of ...   \n",
              "8637   Synthesis of A-9758, an Inverse Agonist of Ret...   \n",
              "8689   Between Science and Big Business: Tapping Mary...   \n",
              "8691                                       Health scare.   \n",
              "8797   Nucleobase-modified nucleosides and nucleotide...   \n",
              "8838   Genetics of Host Protection against Helicobact...   \n",
              "8909   Do Uncommon Plant Phenolic Compounds Have Unco...   \n",
              "8983                                     FULL ISSUE PDF.   \n",
              "9372   Molecular Modeling of Histamine Receptors—Rece...   \n",
              "9537              Applications of PLA in modern medicine   \n",
              "9812                                               News.   \n",
              "10167                                      Introduction.   \n",
              "10415                                    Full Issue PDF.   \n",
              "11155  2020 Annual World Congress of the Pulmonary Va...   \n",
              "11195                      ACMI conference posters 2019.   \n",
              "\n",
              "                                                 journal authors  \\\n",
              "2      Journal of Generic Medicines : Duplicate, mark...       0   \n",
              "287                       Journal of molecular structure       0   \n",
              "334               Journal of the Indian Chemical Society       0   \n",
              "361    The international journal of high performance ...       0   \n",
              "362                              National science review       0   \n",
              "397         Computer methods and programs in biomedicine       0   \n",
              "438                         Human Nutrition & Metabolism       0   \n",
              "590                            Frontiers in pharmacology       0   \n",
              "917                               Oxford open immunology       0   \n",
              "986                         Acta pharmaceutica Sinica. B       0   \n",
              "1417          Journal of preventive medicine and hygiene       0   \n",
              "1753                                Fundamental Research       0   \n",
              "1775          Journal of preventive medicine and hygiene       0   \n",
              "1993                           Frontiers in pharmacology       0   \n",
              "3778                Sustainable Operations and Computers       0   \n",
              "4522                                      Nature methods       0   \n",
              "4556                              EFB Bioeconomy Journal       0   \n",
              "4563                                  Science immunology       0   \n",
              "4979   Journal of King Saud University - Computer and...       0   \n",
              "5155   Current Research in Green and Sustainable Chem...       0   \n",
              "5385                             Chemical Physics Impact       0   \n",
              "5705                                      Cell stem cell       0   \n",
              "5880                             Alternative Law Journal       0   \n",
              "6088                                Chemistry & industry       0   \n",
              "6140   International Journal of Cognitive Computing i...       0   \n",
              "6467                         Disease models & mechanisms       0   \n",
              "6541                           The Cambridge law journal       0   \n",
              "6543     Journal of Intellectual Property Law & Practice       0   \n",
              "6666                      Alcoholism & Drug Abuse Weekly       0   \n",
              "6697                                Chemistry & industry       0   \n",
              "6757                  Journal of law and the biosciences       0   \n",
              "6972                                Chemistry & industry       0   \n",
              "7021    Frontiers in cellular and infection microbiology       0   \n",
              "7040                JACC. Basic to translational science       0   \n",
              "7154                                 Innovation in aging       0   \n",
              "7183                                 Neurology: Genetics       0   \n",
              "7810                           Frontiers in pharmacology       0   \n",
              "7897                  IEEE Consumer Electronics Magazine       0   \n",
              "7955                            Neuroscience Informatics       0   \n",
              "8336                              Phytochemistry letters       0   \n",
              "8637                                           ACS omega       0   \n",
              "8689                                 ACS central science       0   \n",
              "8691                                Chemistry & industry       0   \n",
              "8797                              Frontiers in chemistry       0   \n",
              "8838         International journal of molecular sciences       0   \n",
              "8909             Journal of Bioresources and Bioproducts       0   \n",
              "8983                JACC. Basic to translational science       0   \n",
              "9372                      Molecules (Basel, Switzerland)       0   \n",
              "9537                             Engineered Regeneration       0   \n",
              "9812                                Chemistry & industry       0   \n",
              "10167  Protein science : a publication of the Protein...       0   \n",
              "10415                               JACC. CardioOncology       0   \n",
              "11155                              Pulmonary circulation       0   \n",
              "11195                                Access microbiology       0   \n",
              "\n",
              "                                      doi  \\\n",
              "2                                       0   \n",
              "287                                     0   \n",
              "334                                     0   \n",
              "361                                     0   \n",
              "362                                     0   \n",
              "397                                     0   \n",
              "438                                     0   \n",
              "590                                     0   \n",
              "917                                     0   \n",
              "986         10.1016/s2211-3835(22)00283-0   \n",
              "1417   10.15167/2421-4248/jpmh2022.63.2s1   \n",
              "1753                                    0   \n",
              "1775   10.15167/2421-4248/jpmh2022.63.1s1   \n",
              "1993                                    0   \n",
              "3778                                    0   \n",
              "4522           10.1038/s41592-022-01584-2   \n",
              "4556                                    0   \n",
              "4563           10.1126/sciimmunol.abh2259   \n",
              "4979                                    0   \n",
              "5155                                    0   \n",
              "5385                                    0   \n",
              "5705           10.1016/j.stem.2020.11.007   \n",
              "5880                                    0   \n",
              "6088                 10.1002/cind.3_845.x   \n",
              "6140                                    0   \n",
              "6467                   10.1242/dmm.049322   \n",
              "6541                                    0   \n",
              "6543                                    0   \n",
              "6666                                    0   \n",
              "6697                10.1002/cind.14_845.x   \n",
              "6757                                    0   \n",
              "6972                 10.1002/cind.5_845.x   \n",
              "7021                                    0   \n",
              "7040        10.1016/s2452-302x(20)30205-9   \n",
              "7154               10.1093/geroni/igab046   \n",
              "7183                                    0   \n",
              "7810                                    0   \n",
              "7897                                    0   \n",
              "7955                                    0   \n",
              "8336        10.1016/s1874-3900(20)30551-6   \n",
              "8637                                    0   \n",
              "8689                                    0   \n",
              "8691                 10.1002/cind.843_2.x   \n",
              "8797                                    0   \n",
              "8838                                    0   \n",
              "8909                                    0   \n",
              "8983        10.1016/s2452-302x(20)30151-0   \n",
              "9372                                    0   \n",
              "9537                                    0   \n",
              "9812                 10.1002/cind.842_3.x   \n",
              "10167                    10.1002/pro.3838   \n",
              "10415       10.1016/s2666-0873(20)30180-0   \n",
              "11155            10.1177/2045894020976949   \n",
              "11195                                   0   \n",
              "\n",
              "                                                 pdf_url  \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...  \n",
              "287    https://europepmc.org/articles/PMC9803264?pdf=...  \n",
              "334    https://europepmc.org/articles/PMC9428111?pdf=...  \n",
              "361    https://europepmc.org/articles/PMC9548488?pdf=...  \n",
              "362    https://europepmc.org/articles/PMC9798891?pdf=...  \n",
              "397    https://europepmc.org/articles/PMC9711896?pdf=...  \n",
              "438    https://europepmc.org/articles/PMC9762046?pdf=...  \n",
              "590    https://europepmc.org/articles/PMC9812491?pdf=...  \n",
              "917    https://europepmc.org/articles/PMC9452103?pdf=...  \n",
              "986    https://europepmc.org/articles/PMC9273052?pdf=...  \n",
              "1417   https://europepmc.org/articles/PMC9615638?pdf=...  \n",
              "1753   https://europepmc.org/articles/PMC8891120?pdf=...  \n",
              "1775   https://europepmc.org/articles/PMC9364697?pdf=...  \n",
              "1993   https://europepmc.org/articles/PMC9727096?pdf=...  \n",
              "3778   https://europepmc.org/articles/PMC9116973?pdf=...  \n",
              "4522   https://europepmc.org/articles/PMC9466301?pdf=...  \n",
              "4556   https://europepmc.org/articles/PMC8683579?pdf=...  \n",
              "4563   https://europepmc.org/articles/PMC8158979?pdf=...  \n",
              "4979   https://europepmc.org/articles/PMC8831917?pdf=...  \n",
              "5155   https://europepmc.org/articles/PMC8547798?pdf=...  \n",
              "5385   https://europepmc.org/articles/PMC7836883?pdf=...  \n",
              "5705   https://europepmc.org/articles/PMC7713544?pdf=...  \n",
              "5880   https://europepmc.org/articles/PMC9066232?pdf=...  \n",
              "6088   https://europepmc.org/articles/PMC7280571?pdf=...  \n",
              "6140   https://europepmc.org/articles/PMC9098571?pdf=...  \n",
              "6467   https://europepmc.org/articles/PMC8592013?pdf=...  \n",
              "6541   https://europepmc.org/articles/PMC9253443?pdf=...  \n",
              "6543   https://europepmc.org/articles/PMC8135843?pdf=...  \n",
              "6666   https://europepmc.org/articles/PMC8653278?pdf=...  \n",
              "6697   https://europepmc.org/articles/PMC7280709?pdf=...  \n",
              "6757   https://europepmc.org/articles/PMC8882381?pdf=...  \n",
              "6972   https://europepmc.org/articles/PMC7280669?pdf=...  \n",
              "7021   https://europepmc.org/articles/PMC7538618?pdf=...  \n",
              "7040   https://europepmc.org/articles/PMC7247787?pdf=...  \n",
              "7154   https://europepmc.org/articles/PMC8679246?pdf=...  \n",
              "7183   https://europepmc.org/articles/PMC9756308?pdf=...  \n",
              "7810   https://europepmc.org/articles/PMC9812568?pdf=...  \n",
              "7897   https://europepmc.org/articles/PMC9128831?pdf=...  \n",
              "7955   https://europepmc.org/articles/PMC8656214?pdf=...  \n",
              "8336   https://europepmc.org/articles/PMC7370905?pdf=...  \n",
              "8637   https://europepmc.org/articles/PMC9730458?pdf=...  \n",
              "8689   https://europepmc.org/articles/PMC8875429?pdf=...  \n",
              "8691   https://europepmc.org/articles/PMC7162068?pdf=...  \n",
              "8797   https://europepmc.org/articles/PMC9748101?pdf=...  \n",
              "8838   https://europepmc.org/articles/PMC8004045?pdf=...  \n",
              "8909   https://europepmc.org/articles/PMC8445810?pdf=...  \n",
              "8983   https://europepmc.org/articles/PMC7185951?pdf=...  \n",
              "9372   https://europepmc.org/articles/PMC8004658?pdf=...  \n",
              "9537   https://europepmc.org/articles/PMC7474829?pdf=...  \n",
              "9812   https://europepmc.org/articles/PMC7162151?pdf=...  \n",
              "10167  https://europepmc.org/articles/PMC7020986?pdf=...  \n",
              "10415  https://europepmc.org/articles/PMC7492056?pdf=...  \n",
              "11155  https://europepmc.org/articles/PMC8015682?pdf=...  \n",
              "11195  https://europepmc.org/articles/PMC7702993?pdf=...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-8551f5e2-59c0-4f2b-97d8-f289f4d52236\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>287</th>\n",
              "      <td>PMC9803264</td>\n",
              "      <td>2022-12-30</td>\n",
              "      <td>2023-01-06</td>\n",
              "      <td>DFT investigations and molecular docking as po...</td>\n",
              "      <td>Journal of molecular structure</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9803264?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>334</th>\n",
              "      <td>PMC9428111</td>\n",
              "      <td>2022-08-31</td>\n",
              "      <td>2022-09-30</td>\n",
              "      <td>Evaluation of flavonoids as potential inhibito...</td>\n",
              "      <td>Journal of the Indian Chemical Society</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9428111?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>361</th>\n",
              "      <td>PMC9548488</td>\n",
              "      <td>2022-11-01</td>\n",
              "      <td>2022-11-25</td>\n",
              "      <td>Language models for the prediction of SARS-CoV...</td>\n",
              "      <td>The international journal of high performance ...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9548488?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>362</th>\n",
              "      <td>PMC9798891</td>\n",
              "      <td>2022-08-26</td>\n",
              "      <td>2023-01-05</td>\n",
              "      <td>Identification and semisynthesis of (−)-anisom...</td>\n",
              "      <td>National science review</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9798891?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>397</th>\n",
              "      <td>PMC9711896</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Potential of vibrational spectroscopy coupled ...</td>\n",
              "      <td>Computer methods and programs in biomedicine</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9711896?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>438</th>\n",
              "      <td>PMC9762046</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>2023-01-04</td>\n",
              "      <td>COVID-19 infection and metabolic comorbidities...</td>\n",
              "      <td>Human Nutrition &amp; Metabolism</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9762046?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>590</th>\n",
              "      <td>PMC9812491</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2023-01-06</td>\n",
              "      <td>Drug repositioning based on heterogeneous netw...</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9812491?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>917</th>\n",
              "      <td>PMC9452103</td>\n",
              "      <td>2022-08-17</td>\n",
              "      <td>2022-09-12</td>\n",
              "      <td>SARS-CoV-2 immunity and vaccine strategies in ...</td>\n",
              "      <td>Oxford open immunology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9452103?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>986</th>\n",
              "      <td>PMC9273052</td>\n",
              "      <td>2022-07-11</td>\n",
              "      <td>2022-07-18</td>\n",
              "      <td>Cover Story.</td>\n",
              "      <td>Acta pharmaceutica Sinica. B</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2211-3835(22)00283-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9273052?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1417</th>\n",
              "      <td>PMC9615638</td>\n",
              "      <td>2022-09-28</td>\n",
              "      <td>2022-11-08</td>\n",
              "      <td>Atti 55° Congresso Nazionale SItI Padova, 28 s...</td>\n",
              "      <td>Journal of preventive medicine and hygiene</td>\n",
              "      <td>0</td>\n",
              "      <td>10.15167/2421-4248/jpmh2022.63.2s1</td>\n",
              "      <td>https://europepmc.org/articles/PMC9615638?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1753</th>\n",
              "      <td>PMC8891120</td>\n",
              "      <td>2022-03-03</td>\n",
              "      <td>2022-09-16</td>\n",
              "      <td>CEBIT screening for inhibitors of the interact...</td>\n",
              "      <td>Fundamental Research</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8891120?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1775</th>\n",
              "      <td>PMC9364697</td>\n",
              "      <td>2022-07-06</td>\n",
              "      <td>2022-10-06</td>\n",
              "      <td>3&lt;sup&gt;a&lt;/sup&gt; edizione Giornate della ricerca ...</td>\n",
              "      <td>Journal of preventive medicine and hygiene</td>\n",
              "      <td>0</td>\n",
              "      <td>10.15167/2421-4248/jpmh2022.63.1s1</td>\n",
              "      <td>https://europepmc.org/articles/PMC9364697?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1993</th>\n",
              "      <td>PMC9727096</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>Discovery of the key active compounds in Citri...</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9727096?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3778</th>\n",
              "      <td>PMC9116973</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2022-08-06</td>\n",
              "      <td>Emerging technologies for the management of CO...</td>\n",
              "      <td>Sustainable Operations and Computers</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9116973?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4522</th>\n",
              "      <td>PMC9466301</td>\n",
              "      <td>2022-10-01</td>\n",
              "      <td>2022-10-25</td>\n",
              "      <td>A tissue-engineered artificial human thymus fr...</td>\n",
              "      <td>Nature methods</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1038/s41592-022-01584-2</td>\n",
              "      <td>https://europepmc.org/articles/PMC9466301?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4556</th>\n",
              "      <td>PMC8683579</td>\n",
              "      <td>2021-06-13</td>\n",
              "      <td>2022-12-19</td>\n",
              "      <td>Bioeconomy during the COVID-19 and perspective...</td>\n",
              "      <td>EFB Bioeconomy Journal</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8683579?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4563</th>\n",
              "      <td>PMC8158979</td>\n",
              "      <td>2021-05-01</td>\n",
              "      <td>2022-02-10</td>\n",
              "      <td>Increased complement activation is a distincti...</td>\n",
              "      <td>Science immunology</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1126/sciimmunol.abh2259</td>\n",
              "      <td>https://europepmc.org/articles/PMC8158979?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4979</th>\n",
              "      <td>PMC8831917</td>\n",
              "      <td>2021-07-15</td>\n",
              "      <td>2022-10-04</td>\n",
              "      <td>Artificial Intelligence and COVID-19: A System...</td>\n",
              "      <td>Journal of King Saud University - Computer and...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8831917?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5155</th>\n",
              "      <td>PMC8547798</td>\n",
              "      <td>2021-01-01</td>\n",
              "      <td>2023-01-04</td>\n",
              "      <td>In silico screening of antiviral compounds fro...</td>\n",
              "      <td>Current Research in Green and Sustainable Chem...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8547798?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5385</th>\n",
              "      <td>PMC7836883</td>\n",
              "      <td>2021-01-12</td>\n",
              "      <td>2022-02-10</td>\n",
              "      <td>Drug targets, mechanisms of drug action, and t...</td>\n",
              "      <td>Chemical Physics Impact</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7836883?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5705</th>\n",
              "      <td>PMC7713544</td>\n",
              "      <td>2020-12-01</td>\n",
              "      <td>2020-12-24</td>\n",
              "      <td>Scientific Collaboration and Communication in ...</td>\n",
              "      <td>Cell stem cell</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/j.stem.2020.11.007</td>\n",
              "      <td>https://europepmc.org/articles/PMC7713544?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5880</th>\n",
              "      <td>PMC9066232</td>\n",
              "      <td>2022-06-01</td>\n",
              "      <td>2022-06-08</td>\n",
              "      <td>Can employers mandate their employees receive ...</td>\n",
              "      <td>Alternative Law Journal</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9066232?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6088</th>\n",
              "      <td>PMC7280571</td>\n",
              "      <td>2020-05-22</td>\n",
              "      <td>2020-06-15</td>\n",
              "      <td>News.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.3_845.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7280571?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6140</th>\n",
              "      <td>PMC9098571</td>\n",
              "      <td>2022-05-13</td>\n",
              "      <td>2022-06-22</td>\n",
              "      <td>COVID vision: An integrated face mask detector...</td>\n",
              "      <td>International Journal of Cognitive Computing i...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9098571?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6467</th>\n",
              "      <td>PMC8592013</td>\n",
              "      <td>2021-11-11</td>\n",
              "      <td>2022-04-01</td>\n",
              "      <td>First person - Li Wang.</td>\n",
              "      <td>Disease models &amp; mechanisms</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1242/dmm.049322</td>\n",
              "      <td>https://europepmc.org/articles/PMC8592013?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6541</th>\n",
              "      <td>PMC9253443</td>\n",
              "      <td>2022-06-16</td>\n",
              "      <td>2022-07-13</td>\n",
              "      <td>ADDRESSING VACCINE INEQUITY DURING THE COVID-1...</td>\n",
              "      <td>The Cambridge law journal</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9253443?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6543</th>\n",
              "      <td>PMC8135843</td>\n",
              "      <td>2021-05-03</td>\n",
              "      <td>2021-05-24</td>\n",
              "      <td>Compulsory licence and Crown use provisions in...</td>\n",
              "      <td>Journal of Intellectual Property Law &amp; Practice</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8135843?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6666</th>\n",
              "      <td>PMC8653278</td>\n",
              "      <td>2021-10-08</td>\n",
              "      <td>2022-02-12</td>\n",
              "      <td>SUDs increase risk for breakthrough COVID‐19 i...</td>\n",
              "      <td>Alcoholism &amp; Drug Abuse Weekly</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8653278?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6697</th>\n",
              "      <td>PMC7280709</td>\n",
              "      <td>2020-05-22</td>\n",
              "      <td>2020-06-15</td>\n",
              "      <td>SCI's corporate partners respond.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.14_845.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7280709?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6757</th>\n",
              "      <td>PMC8882381</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2022-03-02</td>\n",
              "      <td>When speed matters: a discussion on the benefi...</td>\n",
              "      <td>Journal of law and the biosciences</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8882381?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6972</th>\n",
              "      <td>PMC7280669</td>\n",
              "      <td>2020-05-22</td>\n",
              "      <td>2020-06-15</td>\n",
              "      <td>News.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.5_845.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7280669?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7021</th>\n",
              "      <td>PMC7538618</td>\n",
              "      <td>2020-01-01</td>\n",
              "      <td>2020-10-17</td>\n",
              "      <td>Bibliometric and Visualization Analysis of Hum...</td>\n",
              "      <td>Frontiers in cellular and infection microbiology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7538618?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7040</th>\n",
              "      <td>PMC7247787</td>\n",
              "      <td>2020-05-25</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Full Issue PDF.</td>\n",
              "      <td>JACC. Basic to translational science</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2452-302x(20)30205-9</td>\n",
              "      <td>https://europepmc.org/articles/PMC7247787?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7154</th>\n",
              "      <td>PMC8679246</td>\n",
              "      <td>2021-12-17</td>\n",
              "      <td>2022-04-07</td>\n",
              "      <td>Program Abstracts from The GSA 2021 Annual Sci...</td>\n",
              "      <td>Innovation in aging</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1093/geroni/igab046</td>\n",
              "      <td>https://europepmc.org/articles/PMC8679246?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7183</th>\n",
              "      <td>PMC9756308</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-22</td>\n",
              "      <td>Identification of Sex-Specific Genetic Variant...</td>\n",
              "      <td>Neurology: Genetics</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9756308?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7810</th>\n",
              "      <td>PMC9812568</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2023-01-06</td>\n",
              "      <td>Exploring the effects of drug, disease, and pr...</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9812568?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7897</th>\n",
              "      <td>PMC9128831</td>\n",
              "      <td>2021-04-26</td>\n",
              "      <td>2022-06-20</td>\n",
              "      <td>GlobeChain: An Interoperable Blockchain for Gl...</td>\n",
              "      <td>IEEE Consumer Electronics Magazine</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9128831?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7955</th>\n",
              "      <td>PMC8656214</td>\n",
              "      <td>2021-12-09</td>\n",
              "      <td>2022-09-01</td>\n",
              "      <td>Face mask recognition system using CNN model</td>\n",
              "      <td>Neuroscience Informatics</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8656214?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8336</th>\n",
              "      <td>PMC7370905</td>\n",
              "      <td>2020-07-20</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Announcements of the Phytochemical Society of ...</td>\n",
              "      <td>Phytochemistry letters</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s1874-3900(20)30551-6</td>\n",
              "      <td>https://europepmc.org/articles/PMC7370905?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8637</th>\n",
              "      <td>PMC9730458</td>\n",
              "      <td>2022-11-17</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>Synthesis of A-9758, an Inverse Agonist of Ret...</td>\n",
              "      <td>ACS omega</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9730458?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8689</th>\n",
              "      <td>PMC8875429</td>\n",
              "      <td>2022-01-26</td>\n",
              "      <td>2022-03-02</td>\n",
              "      <td>Between Science and Big Business: Tapping Mary...</td>\n",
              "      <td>ACS central science</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8875429?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8691</th>\n",
              "      <td>PMC7162068</td>\n",
              "      <td>2020-03-01</td>\n",
              "      <td>2020-04-21</td>\n",
              "      <td>Health scare.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.843_2.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7162068?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8797</th>\n",
              "      <td>PMC9748101</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2022-12-23</td>\n",
              "      <td>Nucleobase-modified nucleosides and nucleotide...</td>\n",
              "      <td>Frontiers in chemistry</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9748101?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8838</th>\n",
              "      <td>PMC8004045</td>\n",
              "      <td>2021-03-01</td>\n",
              "      <td>2021-04-01</td>\n",
              "      <td>Genetics of Host Protection against Helicobact...</td>\n",
              "      <td>International journal of molecular sciences</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8004045?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8909</th>\n",
              "      <td>PMC8445810</td>\n",
              "      <td>2021-09-12</td>\n",
              "      <td>2021-10-21</td>\n",
              "      <td>Do Uncommon Plant Phenolic Compounds Have Unco...</td>\n",
              "      <td>Journal of Bioresources and Bioproducts</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8445810?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8983</th>\n",
              "      <td>PMC7185951</td>\n",
              "      <td>2020-04-27</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>FULL ISSUE PDF.</td>\n",
              "      <td>JACC. Basic to translational science</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2452-302x(20)30151-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7185951?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9372</th>\n",
              "      <td>PMC8004658</td>\n",
              "      <td>2021-03-01</td>\n",
              "      <td>2021-04-01</td>\n",
              "      <td>Molecular Modeling of Histamine Receptors—Rece...</td>\n",
              "      <td>Molecules (Basel, Switzerland)</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8004658?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9537</th>\n",
              "      <td>PMC7474829</td>\n",
              "      <td>2020-01-01</td>\n",
              "      <td>2020-12-28</td>\n",
              "      <td>Applications of PLA in modern medicine</td>\n",
              "      <td>Engineered Regeneration</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7474829?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9812</th>\n",
              "      <td>PMC7162151</td>\n",
              "      <td>2020-02-28</td>\n",
              "      <td>2020-04-21</td>\n",
              "      <td>News.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.842_3.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7162151?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10167</th>\n",
              "      <td>PMC7020986</td>\n",
              "      <td>2020-03-01</td>\n",
              "      <td>2021-02-02</td>\n",
              "      <td>Introduction.</td>\n",
              "      <td>Protein science : a publication of the Protein...</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/pro.3838</td>\n",
              "      <td>https://europepmc.org/articles/PMC7020986?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10415</th>\n",
              "      <td>PMC7492056</td>\n",
              "      <td>2020-09-15</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Full Issue PDF.</td>\n",
              "      <td>JACC. CardioOncology</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2666-0873(20)30180-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7492056?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11155</th>\n",
              "      <td>PMC8015682</td>\n",
              "      <td>2021-03-26</td>\n",
              "      <td>2021-04-15</td>\n",
              "      <td>2020 Annual World Congress of the Pulmonary Va...</td>\n",
              "      <td>Pulmonary circulation</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1177/2045894020976949</td>\n",
              "      <td>https://europepmc.org/articles/PMC8015682?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11195</th>\n",
              "      <td>PMC7702993</td>\n",
              "      <td>2019-04-08</td>\n",
              "      <td>2020-12-16</td>\n",
              "      <td>ACMI conference posters 2019.</td>\n",
              "      <td>Access microbiology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7702993?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8551f5e2-59c0-4f2b-97d8-f289f4d52236')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-8551f5e2-59c0-4f2b-97d8-f289f4d52236 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-8551f5e2-59c0-4f2b-97d8-f289f4d52236');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 7
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(search_results.loc[search_results['authors']==0])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MNMWdCRbJgIe",
        "outputId": "052de43f-711a-4f3e-aceb-88e0182a50db"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "54"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "This is not necessarily a problem so we will keep these rows and still extract the full text for them."
      ],
      "metadata": {
        "id": "dU7tLulFDHLq"
      }
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "EDpcGcitpECi"
      },
      "source": [
        "### 4.2 Remove unwanted categories\n",
        "\n",
        "To ensure that only full text articles remain, we will remove other categories (Abstracts, Research Abstract, Annual Meeting, Congress, Conference)."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "str_remove = search_results[(search_results['title'].str.contains('Abstract', case=False)) |\n",
        "                            (search_results['title'].str.contains('Abstracts', case=False)) |\n",
        "                            (search_results['title'].str.contains('Annual Meeting', case=False)) |\n",
        "                            (search_results['title'].str.contains('Congress', case=False)) |\n",
        "                            (search_results['title'].str.contains('Conference', case=False))]\n",
        "str_remove"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "XfiqyHj_BRnp",
        "outputId": "8a55a1d7-df5d-4960-9b1c-bd65101261b8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "1417   PMC9615638  2022-09-28  2022-11-08   \n",
              "1712   PMC9623364  2022-01-01  2022-11-04   \n",
              "1763   PMC8011435  2021-03-01  2022-09-29   \n",
              "2995   PMC8041998  2021-04-12  2021-04-29   \n",
              "5581   PMC7418285  2020-08-11  2020-12-18   \n",
              "6137   PMC7767910  2020-12-28  2021-01-06   \n",
              "6256   PMC8265285  2021-07-08  2022-03-17   \n",
              "7109   PMC9169230  2022-04-08  2022-07-16   \n",
              "7154   PMC8679246  2021-12-17  2022-04-07   \n",
              "7213   PMC7376524  2020-07-23  2021-01-27   \n",
              "8489   PMC7770499  2020-12-29  2021-05-14   \n",
              "8517   PMC7105345  2019-07-11  2021-01-10   \n",
              "8622   PMC8440011  2021-08-01  2022-05-31   \n",
              "9512   PMC8514745  2021-01-01  2021-10-16   \n",
              "10273  PMC6965503  2020-01-14  2020-09-28   \n",
              "11070  PMC7645912  2020-01-01  2020-11-08   \n",
              "11155  PMC8015682  2021-03-26  2021-04-15   \n",
              "11195  PMC7702993  2019-04-08  2020-12-16   \n",
              "\n",
              "                                                   title  \\\n",
              "1417   Atti 55° Congresso Nazionale SItI Padova, 28 s...   \n",
              "1712   Meeting report: 34th international conference ...   \n",
              "1763   Drugmonizome and Drugmonizome-ML: integration ...   \n",
              "2995   COVID-19 information retrieval with deep-learn...   \n",
              "5581   Meeting report of the 49th annual meeting of t...   \n",
              "6137   Accelerating bioinformatics research with Inte...   \n",
              "6256   Lewy Body Dementia Association's Industry Advi...   \n",
              "7109   The Cure VCP Scientific Conference 2021: Molec...   \n",
              "7154   Program Abstracts from The GSA 2021 Annual Sci...   \n",
              "7213   Reflections on the upsurge of virtual cancer c...   \n",
              "8489   The International Conference on Intelligent Bi...   \n",
              "8517   Meeting report: 32nd International Conference ...   \n",
              "8622   Cancer Clinical Trials in Africa-An Untapped O...   \n",
              "9512                 2021 ASN Virtual Meeting Abstracts.   \n",
              "10273  A report of the 18th congress of the Japanese ...   \n",
              "11070                               ABSTRACTS ADDRM 2020   \n",
              "11155  2020 Annual World Congress of the Pulmonary Va...   \n",
              "11195                      ACMI conference posters 2019.   \n",
              "\n",
              "                                                 journal  \\\n",
              "1417          Journal of preventive medicine and hygiene   \n",
              "1712                  Antiviral chemistry & chemotherapy   \n",
              "1763   Database : the journal of biological databases...   \n",
              "2995                                NPJ digital medicine   \n",
              "5581   Inflammation research : official journal of th...   \n",
              "6137                                  BMC bioinformatics   \n",
              "6256                      Alzheimer's research & therapy   \n",
              "7109                             Neurobiology of disease   \n",
              "7154                                 Innovation in aging   \n",
              "7213                           British journal of cancer   \n",
              "8489                                        BMC genomics   \n",
              "8517                                  Antiviral research   \n",
              "8622                                 JCO global oncology   \n",
              "9512                                           ASN neuro   \n",
              "10273                               Regenerative therapy   \n",
              "11070           Nederlands Tijdschrift voor Diabetologie   \n",
              "11155                              Pulmonary circulation   \n",
              "11195                                Access microbiology   \n",
              "\n",
              "                                                 authors  \\\n",
              "1417                                                   0   \n",
              "1712   Brancale A, Carter K, Delang L, Deval J, Duran...   \n",
              "1763   Kropiwnicki E, Evangelista JE, Stein DJ, Clark...   \n",
              "2995   Esteva A, Kale A, Paulus R, Hashimoto K, Yin W...   \n",
              "5581                                     Kay L, Obara I.   \n",
              "6137        Guo Y, Shen L, Shi X, Wang K, Dai Y, Zhao Z.   \n",
              "6256   Goldman JG, Boeve BF, Armstrong MJ, Galasko DR...   \n",
              "7109   Johnson MA, Klickstein JA, Khanna R, Gou Y, Cu...   \n",
              "7154                                                   0   \n",
              "7213                                           Speirs V.   \n",
              "8489             Zhang W, Zhao Z, Wang K, Shen L, Shi X.   \n",
              "8517   Tramontano E, Tarbet B, Spengler JR, Seley-Rad...   \n",
              "8622   Ibraheem A, Pillai C, Okoye I, Smith JJ, Reidy...   \n",
              "9512                 American Society of Neurochemistry.   \n",
              "10273                                 Itaba N, Shiota G.   \n",
              "11070                                           Serné E.   \n",
              "11155                                                  0   \n",
              "11195                                                  0   \n",
              "\n",
              "                                      doi  \\\n",
              "1417   10.15167/2421-4248/jpmh2022.63.2s1   \n",
              "1712            10.1177/20402066221130853   \n",
              "1763             10.1093/database/baab017   \n",
              "2995           10.1038/s41746-021-00437-0   \n",
              "5581           10.1007/s00011-020-01390-6   \n",
              "6137           10.1186/s12859-020-03890-y   \n",
              "6256           10.1186/s13195-021-00868-7   \n",
              "7109            10.1016/j.nbd.2022.105722   \n",
              "7154               10.1093/geroni/igab046   \n",
              "7213            10.1038/s41416-020-1000-x   \n",
              "8489           10.1186/s12864-020-07256-9   \n",
              "8517      10.1016/j.antiviral.2019.104550   \n",
              "8622                  10.1200/go.21.00096   \n",
              "9512            10.1177/17590914211039028   \n",
              "10273          10.1016/j.reth.2019.11.002   \n",
              "11070                                   0   \n",
              "11155            10.1177/2045894020976949   \n",
              "11195                                   0   \n",
              "\n",
              "                                                 pdf_url  \n",
              "1417   https://europepmc.org/articles/PMC9615638?pdf=...  \n",
              "1712   https://europepmc.org/articles/PMC9623364?pdf=...  \n",
              "1763   https://europepmc.org/articles/PMC8011435?pdf=...  \n",
              "2995   https://europepmc.org/articles/PMC8041998?pdf=...  \n",
              "5581   https://europepmc.org/articles/PMC7418285?pdf=...  \n",
              "6137   https://europepmc.org/articles/PMC7767910?pdf=...  \n",
              "6256   https://europepmc.org/articles/PMC8265285?pdf=...  \n",
              "7109   https://europepmc.org/articles/PMC9169230?pdf=...  \n",
              "7154   https://europepmc.org/articles/PMC8679246?pdf=...  \n",
              "7213   https://europepmc.org/articles/PMC7376524?pdf=...  \n",
              "8489   https://europepmc.org/articles/PMC7770499?pdf=...  \n",
              "8517   https://europepmc.org/articles/PMC7105345?pdf=...  \n",
              "8622   https://europepmc.org/articles/PMC8440011?pdf=...  \n",
              "9512   https://europepmc.org/articles/PMC8514745?pdf=...  \n",
              "10273  https://europepmc.org/articles/PMC6965503?pdf=...  \n",
              "11070  https://europepmc.org/articles/PMC7645912?pdf=...  \n",
              "11155  https://europepmc.org/articles/PMC8015682?pdf=...  \n",
              "11195  https://europepmc.org/articles/PMC7702993?pdf=...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-880b7b7e-5967-474a-93aa-d3aee48a61a6\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1417</th>\n",
              "      <td>PMC9615638</td>\n",
              "      <td>2022-09-28</td>\n",
              "      <td>2022-11-08</td>\n",
              "      <td>Atti 55° Congresso Nazionale SItI Padova, 28 s...</td>\n",
              "      <td>Journal of preventive medicine and hygiene</td>\n",
              "      <td>0</td>\n",
              "      <td>10.15167/2421-4248/jpmh2022.63.2s1</td>\n",
              "      <td>https://europepmc.org/articles/PMC9615638?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1712</th>\n",
              "      <td>PMC9623364</td>\n",
              "      <td>2022-01-01</td>\n",
              "      <td>2022-11-04</td>\n",
              "      <td>Meeting report: 34th international conference ...</td>\n",
              "      <td>Antiviral chemistry &amp; chemotherapy</td>\n",
              "      <td>Brancale A, Carter K, Delang L, Deval J, Duran...</td>\n",
              "      <td>10.1177/20402066221130853</td>\n",
              "      <td>https://europepmc.org/articles/PMC9623364?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1763</th>\n",
              "      <td>PMC8011435</td>\n",
              "      <td>2021-03-01</td>\n",
              "      <td>2022-09-29</td>\n",
              "      <td>Drugmonizome and Drugmonizome-ML: integration ...</td>\n",
              "      <td>Database : the journal of biological databases...</td>\n",
              "      <td>Kropiwnicki E, Evangelista JE, Stein DJ, Clark...</td>\n",
              "      <td>10.1093/database/baab017</td>\n",
              "      <td>https://europepmc.org/articles/PMC8011435?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2995</th>\n",
              "      <td>PMC8041998</td>\n",
              "      <td>2021-04-12</td>\n",
              "      <td>2021-04-29</td>\n",
              "      <td>COVID-19 information retrieval with deep-learn...</td>\n",
              "      <td>NPJ digital medicine</td>\n",
              "      <td>Esteva A, Kale A, Paulus R, Hashimoto K, Yin W...</td>\n",
              "      <td>10.1038/s41746-021-00437-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC8041998?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5581</th>\n",
              "      <td>PMC7418285</td>\n",
              "      <td>2020-08-11</td>\n",
              "      <td>2020-12-18</td>\n",
              "      <td>Meeting report of the 49th annual meeting of t...</td>\n",
              "      <td>Inflammation research : official journal of th...</td>\n",
              "      <td>Kay L, Obara I.</td>\n",
              "      <td>10.1007/s00011-020-01390-6</td>\n",
              "      <td>https://europepmc.org/articles/PMC7418285?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6137</th>\n",
              "      <td>PMC7767910</td>\n",
              "      <td>2020-12-28</td>\n",
              "      <td>2021-01-06</td>\n",
              "      <td>Accelerating bioinformatics research with Inte...</td>\n",
              "      <td>BMC bioinformatics</td>\n",
              "      <td>Guo Y, Shen L, Shi X, Wang K, Dai Y, Zhao Z.</td>\n",
              "      <td>10.1186/s12859-020-03890-y</td>\n",
              "      <td>https://europepmc.org/articles/PMC7767910?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6256</th>\n",
              "      <td>PMC8265285</td>\n",
              "      <td>2021-07-08</td>\n",
              "      <td>2022-03-17</td>\n",
              "      <td>Lewy Body Dementia Association's Industry Advi...</td>\n",
              "      <td>Alzheimer's research &amp; therapy</td>\n",
              "      <td>Goldman JG, Boeve BF, Armstrong MJ, Galasko DR...</td>\n",
              "      <td>10.1186/s13195-021-00868-7</td>\n",
              "      <td>https://europepmc.org/articles/PMC8265285?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7109</th>\n",
              "      <td>PMC9169230</td>\n",
              "      <td>2022-04-08</td>\n",
              "      <td>2022-07-16</td>\n",
              "      <td>The Cure VCP Scientific Conference 2021: Molec...</td>\n",
              "      <td>Neurobiology of disease</td>\n",
              "      <td>Johnson MA, Klickstein JA, Khanna R, Gou Y, Cu...</td>\n",
              "      <td>10.1016/j.nbd.2022.105722</td>\n",
              "      <td>https://europepmc.org/articles/PMC9169230?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7154</th>\n",
              "      <td>PMC8679246</td>\n",
              "      <td>2021-12-17</td>\n",
              "      <td>2022-04-07</td>\n",
              "      <td>Program Abstracts from The GSA 2021 Annual Sci...</td>\n",
              "      <td>Innovation in aging</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1093/geroni/igab046</td>\n",
              "      <td>https://europepmc.org/articles/PMC8679246?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7213</th>\n",
              "      <td>PMC7376524</td>\n",
              "      <td>2020-07-23</td>\n",
              "      <td>2021-01-27</td>\n",
              "      <td>Reflections on the upsurge of virtual cancer c...</td>\n",
              "      <td>British journal of cancer</td>\n",
              "      <td>Speirs V.</td>\n",
              "      <td>10.1038/s41416-020-1000-x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7376524?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8489</th>\n",
              "      <td>PMC7770499</td>\n",
              "      <td>2020-12-29</td>\n",
              "      <td>2021-05-14</td>\n",
              "      <td>The International Conference on Intelligent Bi...</td>\n",
              "      <td>BMC genomics</td>\n",
              "      <td>Zhang W, Zhao Z, Wang K, Shen L, Shi X.</td>\n",
              "      <td>10.1186/s12864-020-07256-9</td>\n",
              "      <td>https://europepmc.org/articles/PMC7770499?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8517</th>\n",
              "      <td>PMC7105345</td>\n",
              "      <td>2019-07-11</td>\n",
              "      <td>2021-01-10</td>\n",
              "      <td>Meeting report: 32nd International Conference ...</td>\n",
              "      <td>Antiviral research</td>\n",
              "      <td>Tramontano E, Tarbet B, Spengler JR, Seley-Rad...</td>\n",
              "      <td>10.1016/j.antiviral.2019.104550</td>\n",
              "      <td>https://europepmc.org/articles/PMC7105345?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8622</th>\n",
              "      <td>PMC8440011</td>\n",
              "      <td>2021-08-01</td>\n",
              "      <td>2022-05-31</td>\n",
              "      <td>Cancer Clinical Trials in Africa-An Untapped O...</td>\n",
              "      <td>JCO global oncology</td>\n",
              "      <td>Ibraheem A, Pillai C, Okoye I, Smith JJ, Reidy...</td>\n",
              "      <td>10.1200/go.21.00096</td>\n",
              "      <td>https://europepmc.org/articles/PMC8440011?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9512</th>\n",
              "      <td>PMC8514745</td>\n",
              "      <td>2021-01-01</td>\n",
              "      <td>2021-10-16</td>\n",
              "      <td>2021 ASN Virtual Meeting Abstracts.</td>\n",
              "      <td>ASN neuro</td>\n",
              "      <td>American Society of Neurochemistry.</td>\n",
              "      <td>10.1177/17590914211039028</td>\n",
              "      <td>https://europepmc.org/articles/PMC8514745?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10273</th>\n",
              "      <td>PMC6965503</td>\n",
              "      <td>2020-01-14</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>A report of the 18th congress of the Japanese ...</td>\n",
              "      <td>Regenerative therapy</td>\n",
              "      <td>Itaba N, Shiota G.</td>\n",
              "      <td>10.1016/j.reth.2019.11.002</td>\n",
              "      <td>https://europepmc.org/articles/PMC6965503?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11070</th>\n",
              "      <td>PMC7645912</td>\n",
              "      <td>2020-01-01</td>\n",
              "      <td>2020-11-08</td>\n",
              "      <td>ABSTRACTS ADDRM 2020</td>\n",
              "      <td>Nederlands Tijdschrift voor Diabetologie</td>\n",
              "      <td>Serné E.</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7645912?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11155</th>\n",
              "      <td>PMC8015682</td>\n",
              "      <td>2021-03-26</td>\n",
              "      <td>2021-04-15</td>\n",
              "      <td>2020 Annual World Congress of the Pulmonary Va...</td>\n",
              "      <td>Pulmonary circulation</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1177/2045894020976949</td>\n",
              "      <td>https://europepmc.org/articles/PMC8015682?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11195</th>\n",
              "      <td>PMC7702993</td>\n",
              "      <td>2019-04-08</td>\n",
              "      <td>2020-12-16</td>\n",
              "      <td>ACMI conference posters 2019.</td>\n",
              "      <td>Access microbiology</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7702993?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-880b7b7e-5967-474a-93aa-d3aee48a61a6')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-880b7b7e-5967-474a-93aa-d3aee48a61a6 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-880b7b7e-5967-474a-93aa-d3aee48a61a6');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(str_remove)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XgIspVjhBf9D",
        "outputId": "eb2504e2-367b-4e3c-cdbe-f6311f86a80f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "18"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "k0sgS_JdXObs"
      },
      "source": [
        "Make copy of DataFrame, drop the 18 rows and reset index."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_new = pd.concat([search_results, str_remove]).drop_duplicates(keep=False).reset_index(drop=True)\n",
        "search_results_new"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "wZWBpKhBBqnF",
        "outputId": "ddd6a0f3-5e06-4275-badd-ddf6d90c27a0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "0      PMC9549161  2022-09-26  2022-10-14   \n",
              "1      PMC9539342  2022-09-22  2022-11-12   \n",
              "2      PMC9357751  2022-12-01  2022-12-05   \n",
              "3      PMC9346052  2022-08-03  2022-09-05   \n",
              "4      PMC9775208  2022-12-15  2022-12-25   \n",
              "...           ...         ...         ...   \n",
              "11393  PMC6328940  2019-01-01  2020-03-09   \n",
              "11394  PMC6598402  2019-06-21  2020-09-28   \n",
              "11395  PMC6481739  2019-02-05  2020-09-28   \n",
              "11396  PMC6624471  2019-07-05  2020-09-28   \n",
              "11397  PMC6409730  2019-02-06  2020-09-30   \n",
              "\n",
              "                                                   title  \\\n",
              "0           Drug repositioning: A bibliometric analysis.   \n",
              "1      A review on computer-aided chemogenomics and d...   \n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "3      Scope of repurposed drugs against the potentia...   \n",
              "4      Drug Repurposing Using Gene Co-Expression and ...   \n",
              "...                                                  ...   \n",
              "11393  β-RA reduces DMQ/CoQ ratio and rescues the enc...   \n",
              "11394  Alzheimer Disease Pathogenesis: Insights From ...   \n",
              "11395  Modeling cardiac complexity: Advancements in m...   \n",
              "11396  Tissue Response to Neural Implants: The Use of...   \n",
              "11397  24th International Symposium on Infections in ...   \n",
              "\n",
              "                                                 journal  \\\n",
              "0                              Frontiers in pharmacology   \n",
              "1                         Chemical biology & drug design   \n",
              "2      Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                                   Structural chemistry   \n",
              "4                                                Biology   \n",
              "...                                                  ...   \n",
              "11393                            EMBO molecular medicine   \n",
              "11394                          Frontiers in neuroscience   \n",
              "11395                                 APL bioengineering   \n",
              "11396                          Frontiers in neuroscience   \n",
              "11397              Medical sciences (Basel, Switzerland)   \n",
              "\n",
              "                                                 authors  \\\n",
              "0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                      0   \n",
              "3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                    Mailem RC, Tayo LL.   \n",
              "...                                                  ...   \n",
              "11393  Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...   \n",
              "11394                                Chen XQ, Mobley WC.   \n",
              "11395  Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...   \n",
              "11396       Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.   \n",
              "11397  Artigas A, Carlet J, Garnacho J, Niederman M, ...   \n",
              "\n",
              "                              doi  \\\n",
              "0       10.3389/fphar.2022.974849   \n",
              "1              10.1111/cbdd.14136   \n",
              "2                               0   \n",
              "3      10.1007/s11224-022-02020-z   \n",
              "4         10.3390/biology11121827   \n",
              "...                           ...   \n",
              "11393     10.15252/emmm.201809466   \n",
              "11394    10.3389/fnins.2019.00659   \n",
              "11395           10.1063/1.5055873   \n",
              "11396    10.3389/fnins.2019.00689   \n",
              "11397       10.3390/medsci7020023   \n",
              "\n",
              "                                                 pdf_url  \n",
              "0      https://europepmc.org/articles/PMC9549161?pdf=...  \n",
              "1      https://europepmc.org/articles/PMC9539342?pdf=...  \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...  \n",
              "3      https://europepmc.org/articles/PMC9346052?pdf=...  \n",
              "4      https://europepmc.org/articles/PMC9775208?pdf=...  \n",
              "...                                                  ...  \n",
              "11393  https://europepmc.org/articles/PMC6328940?pdf=...  \n",
              "11394  https://europepmc.org/articles/PMC6598402?pdf=...  \n",
              "11395  https://europepmc.org/articles/PMC6481739?pdf=...  \n",
              "11396  https://europepmc.org/articles/PMC6624471?pdf=...  \n",
              "11397  https://europepmc.org/articles/PMC6409730?pdf=...  \n",
              "\n",
              "[11398 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-4ee49c60-4a81-460a-8d82-c2d5e1190eb7\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11393</th>\n",
              "      <td>PMC6328940</td>\n",
              "      <td>2019-01-01</td>\n",
              "      <td>2020-03-09</td>\n",
              "      <td>β-RA reduces DMQ/CoQ ratio and rescues the enc...</td>\n",
              "      <td>EMBO molecular medicine</td>\n",
              "      <td>Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...</td>\n",
              "      <td>10.15252/emmm.201809466</td>\n",
              "      <td>https://europepmc.org/articles/PMC6328940?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11394</th>\n",
              "      <td>PMC6598402</td>\n",
              "      <td>2019-06-21</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Alzheimer Disease Pathogenesis: Insights From ...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Chen XQ, Mobley WC.</td>\n",
              "      <td>10.3389/fnins.2019.00659</td>\n",
              "      <td>https://europepmc.org/articles/PMC6598402?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11395</th>\n",
              "      <td>PMC6481739</td>\n",
              "      <td>2019-02-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Modeling cardiac complexity: Advancements in m...</td>\n",
              "      <td>APL bioengineering</td>\n",
              "      <td>Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...</td>\n",
              "      <td>10.1063/1.5055873</td>\n",
              "      <td>https://europepmc.org/articles/PMC6481739?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11396</th>\n",
              "      <td>PMC6624471</td>\n",
              "      <td>2019-07-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Tissue Response to Neural Implants: The Use of...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.</td>\n",
              "      <td>10.3389/fnins.2019.00689</td>\n",
              "      <td>https://europepmc.org/articles/PMC6624471?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11397</th>\n",
              "      <td>PMC6409730</td>\n",
              "      <td>2019-02-06</td>\n",
              "      <td>2020-09-30</td>\n",
              "      <td>24th International Symposium on Infections in ...</td>\n",
              "      <td>Medical sciences (Basel, Switzerland)</td>\n",
              "      <td>Artigas A, Carlet J, Garnacho J, Niederman M, ...</td>\n",
              "      <td>10.3390/medsci7020023</td>\n",
              "      <td>https://europepmc.org/articles/PMC6409730?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11398 rows × 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-4ee49c60-4a81-460a-8d82-c2d5e1190eb7')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-4ee49c60-4a81-460a-8d82-c2d5e1190eb7 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-4ee49c60-4a81-460a-8d82-c2d5e1190eb7');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(search_results_new)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "kGVnn-mfBzb2",
        "outputId": "aaa79288-a337-4414-8962-55c027be0b89"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "11398"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Validate by checking that no titles contain 'Abstract'."
      ],
      "metadata": {
        "id": "sSNnV2edAhNj"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "contain_abstract = search_results_new[search_results_new['title'].str.contains('Abstract')]\n",
        "len(contain_abstract)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v6nXzKtrCA5F",
        "outputId": "6bad9478-5b7c-4405-bf3c-65f191bf8e1c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "0"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 4.3 Check for missing values\n",
        "\n",
        "Concise summary of DataFrame to see if there are any columns with missing values."
      ],
      "metadata": {
        "id": "9zmbSSbCqJ3X"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_new.info()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "f6MTA2CVDEbT",
        "outputId": "dec5b735-8d96-43f9-dcd6-7d7549eeb93f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 11398 entries, 0 to 11397\n",
            "Data columns (total 8 columns):\n",
            " #   Column     Non-Null Count  Dtype \n",
            "---  ------     --------------  ----- \n",
            " 0   pmcid      11398 non-null  object\n",
            " 1   published  11398 non-null  object\n",
            " 2   revised    11398 non-null  object\n",
            " 3   title      11398 non-null  object\n",
            " 4   journal    11398 non-null  object\n",
            " 5   authors    11398 non-null  object\n",
            " 6   doi        11398 non-null  object\n",
            " 7   pdf_url    11398 non-null  object\n",
            "dtypes: object(8)\n",
            "memory usage: 712.5+ KB\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 4.4 Remove duplicates\n",
        "\n",
        "Check value counts for duplicate pmcids."
      ],
      "metadata": {
        "id": "2edO1OzlH9b3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_new['pmcid'].value_counts(ascending=True)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "czP2sIShCuzW",
        "outputId": "df1fe66a-71fb-48b0-abd8-425640681cac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "PMC9549161    1\n",
              "PMC7892713    1\n",
              "PMC8576417    1\n",
              "PMC8100288    1\n",
              "PMC9032529    1\n",
              "             ..\n",
              "PMC7705431    1\n",
              "PMC8216129    1\n",
              "PMC7454275    1\n",
              "PMC6409730    1\n",
              "PMC7365084    2\n",
              "Name: pmcid, Length: 11397, dtype: int64"
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "There is one duplicate for PMC7365084"
      ],
      "metadata": {
        "id": "E8zf58XWp0GO"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "dup_pmcid = search_results_new.loc[search_results_new['pmcid']=='PMC7365084']\n",
        "dup_pmcid"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 350
        },
        "id": "Wa486pVkbGcL",
        "outputId": "f82305ca-043a-44fb-c9a6-45a5e7bb555b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "           pmcid   published     revised  \\\n",
              "7504  PMC7365084  2020-08-01  2021-08-31   \n",
              "7530  PMC7365084  2020-07-16  2020-07-20   \n",
              "\n",
              "                                                  title  \\\n",
              "7504  Targeting Two-Pore Channels: Current Progress ...   \n",
              "7530  Targeting Two-Pore Channels: Current Progress ...   \n",
              "\n",
              "                                 journal  \\\n",
              "7504  Trends in pharmacological sciences   \n",
              "7530  Trends in pharmacological sciences   \n",
              "\n",
              "                                                authors  \\\n",
              "7504  Jin X, Zhang Y, Alharbi A, Hanbashi A, Alhosha...   \n",
              "7530  Jin X, Zhang Y, Alharbi A, Hanbashi A, Alhosha...   \n",
              "\n",
              "                             doi  \\\n",
              "7504  10.1016/j.tips.2020.06.002   \n",
              "7530                           0   \n",
              "\n",
              "                                                pdf_url  \n",
              "7504  https://europepmc.org/articles/PMC7365084?pdf=...  \n",
              "7530  https://europepmc.org/articles/PMC7365084?pdf=...  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-3bdf23cb-feae-4640-833f-b076137b46bc\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>7504</th>\n",
              "      <td>PMC7365084</td>\n",
              "      <td>2020-08-01</td>\n",
              "      <td>2021-08-31</td>\n",
              "      <td>Targeting Two-Pore Channels: Current Progress ...</td>\n",
              "      <td>Trends in pharmacological sciences</td>\n",
              "      <td>Jin X, Zhang Y, Alharbi A, Hanbashi A, Alhosha...</td>\n",
              "      <td>10.1016/j.tips.2020.06.002</td>\n",
              "      <td>https://europepmc.org/articles/PMC7365084?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7530</th>\n",
              "      <td>PMC7365084</td>\n",
              "      <td>2020-07-16</td>\n",
              "      <td>2020-07-20</td>\n",
              "      <td>Targeting Two-Pore Channels: Current Progress ...</td>\n",
              "      <td>Trends in pharmacological sciences</td>\n",
              "      <td>Jin X, Zhang Y, Alharbi A, Hanbashi A, Alhosha...</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7365084?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-3bdf23cb-feae-4640-833f-b076137b46bc')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-3bdf23cb-feae-4640-833f-b076137b46bc button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-3bdf23cb-feae-4640-833f-b076137b46bc');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 20
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Drop duplicate pmcid keeping the row with the DOI and most recent revised date."
      ],
      "metadata": {
        "id": "e708AQmrKUU-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_new = search_results_new.drop(7530).reset_index(drop=True)\n",
        "search_results_new"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "pFJtiplTD83b",
        "outputId": "5b34166c-3ead-445d-a3cb-3246c20852c7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "0      PMC9549161  2022-09-26  2022-10-14   \n",
              "1      PMC9539342  2022-09-22  2022-11-12   \n",
              "2      PMC9357751  2022-12-01  2022-12-05   \n",
              "3      PMC9346052  2022-08-03  2022-09-05   \n",
              "4      PMC9775208  2022-12-15  2022-12-25   \n",
              "...           ...         ...         ...   \n",
              "11392  PMC6328940  2019-01-01  2020-03-09   \n",
              "11393  PMC6598402  2019-06-21  2020-09-28   \n",
              "11394  PMC6481739  2019-02-05  2020-09-28   \n",
              "11395  PMC6624471  2019-07-05  2020-09-28   \n",
              "11396  PMC6409730  2019-02-06  2020-09-30   \n",
              "\n",
              "                                                   title  \\\n",
              "0           Drug repositioning: A bibliometric analysis.   \n",
              "1      A review on computer-aided chemogenomics and d...   \n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "3      Scope of repurposed drugs against the potentia...   \n",
              "4      Drug Repurposing Using Gene Co-Expression and ...   \n",
              "...                                                  ...   \n",
              "11392  β-RA reduces DMQ/CoQ ratio and rescues the enc...   \n",
              "11393  Alzheimer Disease Pathogenesis: Insights From ...   \n",
              "11394  Modeling cardiac complexity: Advancements in m...   \n",
              "11395  Tissue Response to Neural Implants: The Use of...   \n",
              "11396  24th International Symposium on Infections in ...   \n",
              "\n",
              "                                                 journal  \\\n",
              "0                              Frontiers in pharmacology   \n",
              "1                         Chemical biology & drug design   \n",
              "2      Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                                   Structural chemistry   \n",
              "4                                                Biology   \n",
              "...                                                  ...   \n",
              "11392                            EMBO molecular medicine   \n",
              "11393                          Frontiers in neuroscience   \n",
              "11394                                 APL bioengineering   \n",
              "11395                          Frontiers in neuroscience   \n",
              "11396              Medical sciences (Basel, Switzerland)   \n",
              "\n",
              "                                                 authors  \\\n",
              "0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                      0   \n",
              "3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                    Mailem RC, Tayo LL.   \n",
              "...                                                  ...   \n",
              "11392  Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...   \n",
              "11393                                Chen XQ, Mobley WC.   \n",
              "11394  Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...   \n",
              "11395       Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.   \n",
              "11396  Artigas A, Carlet J, Garnacho J, Niederman M, ...   \n",
              "\n",
              "                              doi  \\\n",
              "0       10.3389/fphar.2022.974849   \n",
              "1              10.1111/cbdd.14136   \n",
              "2                               0   \n",
              "3      10.1007/s11224-022-02020-z   \n",
              "4         10.3390/biology11121827   \n",
              "...                           ...   \n",
              "11392     10.15252/emmm.201809466   \n",
              "11393    10.3389/fnins.2019.00659   \n",
              "11394           10.1063/1.5055873   \n",
              "11395    10.3389/fnins.2019.00689   \n",
              "11396       10.3390/medsci7020023   \n",
              "\n",
              "                                                 pdf_url  \n",
              "0      https://europepmc.org/articles/PMC9549161?pdf=...  \n",
              "1      https://europepmc.org/articles/PMC9539342?pdf=...  \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...  \n",
              "3      https://europepmc.org/articles/PMC9346052?pdf=...  \n",
              "4      https://europepmc.org/articles/PMC9775208?pdf=...  \n",
              "...                                                  ...  \n",
              "11392  https://europepmc.org/articles/PMC6328940?pdf=...  \n",
              "11393  https://europepmc.org/articles/PMC6598402?pdf=...  \n",
              "11394  https://europepmc.org/articles/PMC6481739?pdf=...  \n",
              "11395  https://europepmc.org/articles/PMC6624471?pdf=...  \n",
              "11396  https://europepmc.org/articles/PMC6409730?pdf=...  \n",
              "\n",
              "[11397 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-8961cbae-3a6c-410a-ace6-1155907f4df5\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11392</th>\n",
              "      <td>PMC6328940</td>\n",
              "      <td>2019-01-01</td>\n",
              "      <td>2020-03-09</td>\n",
              "      <td>β-RA reduces DMQ/CoQ ratio and rescues the enc...</td>\n",
              "      <td>EMBO molecular medicine</td>\n",
              "      <td>Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...</td>\n",
              "      <td>10.15252/emmm.201809466</td>\n",
              "      <td>https://europepmc.org/articles/PMC6328940?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11393</th>\n",
              "      <td>PMC6598402</td>\n",
              "      <td>2019-06-21</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Alzheimer Disease Pathogenesis: Insights From ...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Chen XQ, Mobley WC.</td>\n",
              "      <td>10.3389/fnins.2019.00659</td>\n",
              "      <td>https://europepmc.org/articles/PMC6598402?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11394</th>\n",
              "      <td>PMC6481739</td>\n",
              "      <td>2019-02-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Modeling cardiac complexity: Advancements in m...</td>\n",
              "      <td>APL bioengineering</td>\n",
              "      <td>Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...</td>\n",
              "      <td>10.1063/1.5055873</td>\n",
              "      <td>https://europepmc.org/articles/PMC6481739?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11395</th>\n",
              "      <td>PMC6624471</td>\n",
              "      <td>2019-07-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Tissue Response to Neural Implants: The Use of...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.</td>\n",
              "      <td>10.3389/fnins.2019.00689</td>\n",
              "      <td>https://europepmc.org/articles/PMC6624471?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11396</th>\n",
              "      <td>PMC6409730</td>\n",
              "      <td>2019-02-06</td>\n",
              "      <td>2020-09-30</td>\n",
              "      <td>24th International Symposium on Infections in ...</td>\n",
              "      <td>Medical sciences (Basel, Switzerland)</td>\n",
              "      <td>Artigas A, Carlet J, Garnacho J, Niederman M, ...</td>\n",
              "      <td>10.3390/medsci7020023</td>\n",
              "      <td>https://europepmc.org/articles/PMC6409730?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11397 rows × 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8961cbae-3a6c-410a-ace6-1155907f4df5')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-8961cbae-3a6c-410a-ace6-1155907f4df5 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-8961cbae-3a6c-410a-ace6-1155907f4df5');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 25
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check that the duplicate has gone."
      ],
      "metadata": {
        "id": "YFs2HZvBD376"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_new['pmcid'].value_counts(ascending=True)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UVBZ4mIbEmqn",
        "outputId": "791a657e-b470-4679-c85b-1cddae8100f7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "PMC9549161    1\n",
              "PMC8875160    1\n",
              "PMC7892713    1\n",
              "PMC8576417    1\n",
              "PMC8100288    1\n",
              "             ..\n",
              "PMC7705431    1\n",
              "PMC8216129    1\n",
              "PMC9558053    1\n",
              "PMC8015232    1\n",
              "PMC6409730    1\n",
              "Name: pmcid, Length: 11397, dtype: int64"
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_europepmc_df_json_ft_urls_11397.pickle', 'wb') as f:\n",
        "  pickle.dump(search_results_new, f)"
      ],
      "metadata": {
        "id": "G396LuiN-Q-s"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "qQg20SQzXQbT"
      },
      "source": [
        "## 5. Download full text as XML"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 5.1 DiskCache\n",
        "\n",
        "[DiskCache](https://github.com/grantjenks/python-diskcache) can effectively cache expensive computations, improve performance and reduce computation time for heavy data processing, especially for functions with repeatable results over time."
      ],
      "metadata": {
        "id": "vbS7mP1Sqobi"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# Initialise the Cache object, specifying the path to the cache directory\n",
        "cache = Cache('/content/drive/MyDrive/cache')"
      ],
      "metadata": {
        "id": "NAtSeBcImQmT"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Define a function to download full article text for each pmcid using Europe PMC API endpoint for full text XML."
      ],
      "metadata": {
        "id": "13EcjCeHO_hm"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "s6wAEU66qKE1"
      },
      "outputs": [],
      "source": [
        "# Define function and use the cache.memoize decorator to cache its results\n",
        "@cache.memoize(expire=3600) # Results are cached for 1 hour\n",
        "def dl_article_xml(pmcid: str):\n",
        "\n",
        "    url = f'https://www.ebi.ac.uk/europepmc/webservices/rest/{pmcid}/fullTextXML'\n",
        "    response = requests.get(url, allow_redirects=True)\n",
        "    if response.status_code != 200:\n",
        "      raise Exception(response.text)\n",
        "    parsed_article = response.text\n",
        "\n",
        "    return parsed_article"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "###  5.2 Multithreading"
      ],
      "metadata": {
        "id": "3n5hgCdV_KMK"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Multithreading is used here since the code is I/O-bound rather than CPU-bound. Executing multiple threads concurrently speeds up the process which took 22m 47s to complete as opposed to ~6h 30m without concurrency and just iterating through the pmcid list using a for loop calling the dl_article_xml() function sequentially for each pmcid."
      ],
      "metadata": {
        "id": "6BsWEIJJQhiM"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with concurrent.futures.ThreadPoolExecutor(20) as executor:\n",
        "     futures = [executor.submit(dl_article_xml, pmcid) for pmcid in search_results_new.pmcid]\n",
        "     concurrent.futures.wait(futures)"
      ],
      "metadata": {
        "id": "xiw5GPJz_OKo"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create a dictionary of futures which are proxies for results that do not yet exist but will in the future."
      ],
      "metadata": {
        "id": "5co2XSZGpCG2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "futures_map = dict(zip(search_results_new.pmcid, futures))"
      ],
      "metadata": {
        "id": "R0ZWAKK2_OKp"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(futures_map)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Z1cTC4HS4N9e",
        "outputId": "bd15e60f-9c3c-47b8-c067-b77fd04387bb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "11397"
            ]
          },
          "metadata": {},
          "execution_count": 29
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create dictionary of exceptions with pmcid as key and exception error message as value."
      ],
      "metadata": {
        "id": "PvJS76R4pIEH"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "exceptions = {pmcid: f.exception() for pmcid, f in futures_map.items() if f.exception() is not None}\n",
        "exceptions"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "INI2Rz0K4R2G",
        "outputId": "77548845-4bd3-4ee2-cd2e-ace427645568"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'PMC8018918': Exception(''),\n",
              " 'PMC7640961': Exception(''),\n",
              " 'PMC8018905': Exception(''),\n",
              " 'PMC7098069': Exception(''),\n",
              " 'PMC7382535': Exception(''),\n",
              " 'PMC7936759': Exception(''),\n",
              " 'PMC8014535': Exception(''),\n",
              " 'PMC7383733': Exception(''),\n",
              " 'PMC7558230': Exception(''),\n",
              " 'PMC8115429': Exception(''),\n",
              " 'PMC7497212': Exception(''),\n",
              " 'PMC7321661': Exception(''),\n",
              " 'PMC8459260': Exception('')}"
            ]
          },
          "metadata": {},
          "execution_count": 30
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(exceptions)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "0f8446a7-0bab-4ac4-e749-bb0604b2b973",
        "id": "tI_CSrk4_OKq"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "13"
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "13 exceptions were found which will be handled separately."
      ],
      "metadata": {
        "id": "LXCQnAIKRkIv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_europepmc_df_json_ft_urls_13_exceptions.pickle', 'wb') as f:\n",
        "  pickle.dump(exceptions, f)"
      ],
      "metadata": {
        "id": "kZ5Q1QbEexyZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Initiate a cache for the full text XML."
      ],
      "metadata": {
        "id": "uAwDfR6fOkQ5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "cache = Cache('/content/drive/MyDrive/cache')"
      ],
      "metadata": {
        "id": "l8t6h5d5Oqdf"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Amend the function to handle the exceptions and run again to download the full text."
      ],
      "metadata": {
        "id": "U8k4L50YS_5f"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "@cache.memoize(expire=3600)\n",
        "def dl_article_xml(pmcid: str):\n",
        "\n",
        "    url = f'https://www.ebi.ac.uk/europepmc/webservices/rest/{pmcid}/fullTextXML'\n",
        "    response = requests.get(url, allow_redirects=True)\n",
        "    if response.status_code != 200:\n",
        "      if response.text == '':\n",
        "        return '0'\n",
        "      else:\n",
        "          raise Exception(response.text)\n",
        "    parsed_article = response.text\n",
        "\n",
        "\n",
        "    return parsed_article"
      ],
      "metadata": {
        "id": "YWHAGkJRfXV2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "with ThreadPool(20) as pool:\n",
        "  dl_results = pool.map(dl_article_xml, search_results_new.pmcid)"
      ],
      "metadata": {
        "id": "VJTFl5wsfXV3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(dl_results)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "711db359-ccb6-492c-f5e3-df1116976558",
        "id": "99UielFsfXV3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "11397"
            ]
          },
          "metadata": {},
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_europepmc_ft_xml.pickle', 'wb') as f:\n",
        "  pickle.dump(dl_results, f)"
      ],
      "metadata": {
        "id": "QUZtN-wNmBRr"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 6. Clean and return XML as Beautiful Soup object\n",
        "\n",
        "Function to remove DOCTYPE declaration and return the cleaned, parsed text as a [Beautiful Soup](https://www.crummy.com/software/BeautifulSoup/) object. This object represents the document as a nested data structure so that we can navigate and extract text using XML tags.\n",
        "\n",
        "The simplest way to navigate the parse tree is to find a tag by name so we will use the find() method to return the XML for the `<body>` tag content."
      ],
      "metadata": {
        "id": "nKRh3jicUQ5v"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "1Tq4GQAPrj5y"
      },
      "outputs": [],
      "source": [
        "def soupify(text):\n",
        "    cleaned_text = re.sub('<!DOCTYPE.*(\\[[\\s\\S]*?\\])?>', '', text)\n",
        "    return  BeautifulSoup(cleaned_text, 'lxml-xml').find(\"body\")"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Results were downloaded individually in batches of 2500 and took ~3m each to complete for file sizes of ~9-10 GB."
      ],
      "metadata": {
        "id": "GJ7P8u8QP-Iu"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_1 = list(map(soupify, dl_results[0:2500]))"
      ],
      "metadata": {
        "id": "qVLG5mrCrj5y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_2 = list(map(soupify, dl_results[2500:5000]))"
      ],
      "metadata": {
        "id": "yhgoimDSrj5y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_3 = list(map(soupify, dl_results[5000:7500]))"
      ],
      "metadata": {
        "id": "gs9uk1W3rj5y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_4 = list(map(soupify, dl_results[7500:10000]))"
      ],
      "metadata": {
        "id": "PlDqEu0lrj5z"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_5 = list(map(soupify, dl_results[10000:11397]))"
      ],
      "metadata": {
        "id": "n8CIW4kwrj5z"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 6.1 Find all tags including descendants\n",
        "\n",
        "The function below uses the find_all() method to return all of the direct child and descendant tags of the `<body>` tag for the first 2500 articles."
      ],
      "metadata": {
        "id": "DE6lOMyWT_UF"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_all_tags(article):\n",
        "\n",
        "    tags_list = []\n",
        "\n",
        "    try:\n",
        "        for tag in article.find_all(True):\n",
        "            if tag.name != None:\n",
        "              tags_list.append(tag.name)\n",
        "    except Exception as e:\n",
        "      print(e)\n",
        "\n",
        "    return sorted(list(set(tags_list)))"
      ],
      "metadata": {
        "id": "KLuhMqvOACao"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "all_tags = list(map(find_all_tags, soup_results_1))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Bb9Rw2ZvXVFZ",
        "outputId": "d37d313d-42d6-4719-9f18-8f429d032056"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n",
            "'NoneType' object has no attribute 'find_all'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The find_all() method scans the entire document looking for tags and returns a list. Some attribute errors were returned as there are exceptions for some of the articles."
      ],
      "metadata": {
        "id": "gnGMaQBhVEKY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "len(all_tags)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "O8DdQLiTAVEC",
        "outputId": "ef1890fc-8ee6-4992-f7f1-32c1f04ba944"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Print out a list of tags."
      ],
      "metadata": {
        "id": "xBAdwaPuW2a3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "all_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FZbJpzA-AZGN",
        "outputId": "58bedaff-b7d1-4023-ceb0-fdbd0a8e040c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['italic', 'list', 'list-item', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula', 'italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['attrib',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mmultiscripts',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mprescripts',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['disp-formula', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'statement',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munderover',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'label', 'list', 'list-item', 'p', 'sec', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mpadded',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'munder',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'disp-formula-group',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['label', 'p', 'sec', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-quote',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'disp-quote',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'list', 'list-item', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['copyright-statement',\n",
              "  'copyright-year',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'long-desc',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sup',\n",
              "  'xref'],\n",
              " ['bold', 'ext-link', 'italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'notes',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'caption', 'fig', 'graphic', 'italic', 'label', 'p', 'xref'],\n",
              " ['bold',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula', 'label', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msubsup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'monospace', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['array',\n",
              "  'bold',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['ext-link', 'italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['italic', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'named-content',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fig-group',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sc', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'glyph-data',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'disp-quote',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['ext-link', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'notes',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'term',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table-wrap',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'term',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text', 'italic', 'p', 'sec', 'styled-content', 'title', 'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-quote', 'italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'inline-formula',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'notes',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'underline', 'uri', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mpadded',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'named-content',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['array',\n",
              "  'bold',\n",
              "  'disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'list', 'list-item', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'monospace',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'named-content',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mpadded',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'x',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'term',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'statement',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold', 'funding-source', 'label', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'named-content',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'monospace', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'label', 'list', 'list-item', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'disp-formula-group',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['disp-formula', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['list', 'list-item', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sc', 'sec', 'sub', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msubsup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " [],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'copyright-holder',\n",
              "  'copyright-statement',\n",
              "  'copyright-year',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'free_to_read',\n",
              "  'graphic',\n",
              "  'inline-supplementary-material',\n",
              "  'label',\n",
              "  'license',\n",
              "  'license-p',\n",
              "  'license_ref',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['attrib',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'glyph-data',\n",
              "  'glyph-ref',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-quote',\n",
              "  'ext-link',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sc', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'disp-formula-group',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'x',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'strike',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fn',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sec', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mn',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['label', 'list', 'list-item', 'p', 'sec', 'title', 'xref'],\n",
              " ['italic', 'p', 'sc', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'statement',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['disp-formula', 'label', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['disp-formula', 'label', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-quote',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sc', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'notes',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['array',\n",
              "  'bold',\n",
              "  'disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'named-content',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['ext-link', 'list', 'list-item', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mo',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'term',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msubsup',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'def',\n",
              "  'def-item',\n",
              "  'def-list',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'term',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munderover',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'monospace',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'glyph-data',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'private-char',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'munder',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['caption', 'fig', 'graphic', 'italic', 'label', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['label', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sc', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fn',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mover',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sup', 'xref'],\n",
              " ['bold',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msubsup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'x',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mmultiscripts',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mprescripts',\n",
              "  'none',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " [],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'x',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['ext-link', 'italic', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'x',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sup', 'title', 'uri', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'list', 'list-item', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'italic',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'attrib',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'styled-content',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['disp-formula',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msub',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mi',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munder',\n",
              "  'munderover',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'msqrt',\n",
              "  'mstyle',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtext',\n",
              "  'mtr',\n",
              "  'munderover',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alt-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'funding-source',\n",
              "  'graphic',\n",
              "  'institution',\n",
              "  'institution-id',\n",
              "  'institution-wrap',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'uri', 'xref'],\n",
              " ['italic', 'list', 'list-item', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msubsup',\n",
              "  'msup',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'inline-formula',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " [],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'boxed-text',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'object-id',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alt-text',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sc',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'hr',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['alternatives',\n",
              "  'bold',\n",
              "  'caption',\n",
              "  'disp-formula',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mfenced',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'mtext',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'tex-math',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'inline-graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'media',\n",
              "  'mfenced',\n",
              "  'mfrac',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'mspace',\n",
              "  'msub',\n",
              "  'msup',\n",
              "  'mtable',\n",
              "  'mtd',\n",
              "  'mtr',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'copyright-holder',\n",
              "  'disp-formula',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'fn',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'math',\n",
              "  'mi',\n",
              "  'mn',\n",
              "  'mo',\n",
              "  'mrow',\n",
              "  'p',\n",
              "  'permissions',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'table-wrap-foot',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'break',\n",
              "  'caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['italic',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'uri',\n",
              "  'xref'],\n",
              " ['bold', 'italic', 'p', 'sec', 'sub', 'sup', 'title', 'xref'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['bold',\n",
              "  'boxed-text',\n",
              "  'caption',\n",
              "  'col',\n",
              "  'colgroup',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'list',\n",
              "  'list-item',\n",
              "  'media',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table',\n",
              "  'table-wrap',\n",
              "  'tbody',\n",
              "  'td',\n",
              "  'th',\n",
              "  'thead',\n",
              "  'title',\n",
              "  'tr',\n",
              "  'xref'],\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 16
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in all_tags nested list and return the sum of all elements."
      ],
      "metadata": {
        "id": "hWV2_R3cXIBO"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_all_tags = sum(all_tags, [])\n",
        "sum_all_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z6Z5wbH3AdzH",
        "outputId": "d59b61c1-89b4-41b9-a3db-b2b478cab7da"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['caption',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'caption',\n",
              " 'col',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'media',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'styled-content',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'caption',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'math',\n",
              " 'mi',\n",
              " 'mrow',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'title',\n",
              " 'uri',\n",
              " 'xref',\n",
              " 'italic',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'disp-formula',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'uri',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'hr',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'underline',\n",
              " 'xref',\n",
              " 'attrib',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'graphic',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'hr',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'math',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mover',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'mstyle',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtext',\n",
              " 'munder',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'disp-formula',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfenced',\n",
              " 'mi',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'msub',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'title',\n",
              " 'uri',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-graphic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alternatives',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'math',\n",
              " 'media',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mover',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtext',\n",
              " 'munder',\n",
              " 'munderover',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'tex-math',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mmultiscripts',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mprescripts',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msqrt',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'media',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sc',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'col',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'math',\n",
              " 'media',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtable',\n",
              " 'mtd',\n",
              " 'mtext',\n",
              " 'mtr',\n",
              " 'munderover',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'styled-content',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'boxed-text',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'col',\n",
              " 'colgroup',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'boxed-text',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'tex-math',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'disp-formula',\n",
              " 'italic',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mstyle',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'title',\n",
              " 'uri',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'mstyle',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtable',\n",
              " 'mtd',\n",
              " 'mtext',\n",
              " 'mtr',\n",
              " 'munder',\n",
              " 'munderover',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alternatives',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msub',\n",
              " 'msup',\n",
              " 'mtext',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'tex-math',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'disp-formula',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'hr',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'msqrt',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtext',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'hr',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtable',\n",
              " 'mtd',\n",
              " 'mtr',\n",
              " 'munder',\n",
              " 'munderover',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'underline',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mover',\n",
              " 'mrow',\n",
              " 'mstyle',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtable',\n",
              " 'mtd',\n",
              " 'mtext',\n",
              " 'mtr',\n",
              " 'munder',\n",
              " 'munderover',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'statement',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alternatives',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'col',\n",
              " 'colgroup',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'math',\n",
              " 'media',\n",
              " 'mfenced',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msqrt',\n",
              " 'mstyle',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtext',\n",
              " 'munderover',\n",
              " 'object-id',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'list-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'uri',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'caption',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'msqrt',\n",
              " 'mtable',\n",
              " 'mtd',\n",
              " 'mtext',\n",
              " 'mtr',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'boxed-text',\n",
              " 'caption',\n",
              " 'col',\n",
              " 'copyright-holder',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'graphic',\n",
              " 'inline-graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'permissions',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'funding-source',\n",
              " 'graphic',\n",
              " 'hr',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'msub',\n",
              " 'msup',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'alt-text',\n",
              " 'bold',\n",
              " 'break',\n",
              " 'caption',\n",
              " 'disp-formula',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'math',\n",
              " 'mfrac',\n",
              " 'mi',\n",
              " 'mn',\n",
              " 'mo',\n",
              " 'mrow',\n",
              " 'mspace',\n",
              " 'msqrt',\n",
              " 'mstyle',\n",
              " 'msub',\n",
              " 'msubsup',\n",
              " 'msup',\n",
              " 'mtable',\n",
              " 'mtd',\n",
              " 'mtext',\n",
              " 'mtr',\n",
              " 'munder',\n",
              " 'munderover',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " 'bold',\n",
              " 'caption',\n",
              " 'fig',\n",
              " 'fn',\n",
              " 'graphic',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'tbody',\n",
              " 'td',\n",
              " 'th',\n",
              " 'thead',\n",
              " 'title',\n",
              " 'tr',\n",
              " 'xref',\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Counter class with additional most_common() method to return a list of the *n* most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "2aYa4m5SXwVC"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_all_unique_tags = Counter(sum_all_tags).most_common()\n",
        "num_all_unique_tags = num_all_unique_tags[::]\n",
        "\n",
        "for tag, count in num_all_unique_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "20MANi_GAjp2",
        "outputId": "628cd873-3262-4959-fe0b-ab0ad377fa4c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "p 2461\n",
            "xref 2456\n",
            "title 2446\n",
            "sec 2438\n",
            "italic 2261\n",
            "label 2060\n",
            "caption 1978\n",
            "graphic 1902\n",
            "fig 1897\n",
            "sup 1889\n",
            "sub 1628\n",
            "table-wrap 1567\n",
            "td 1551\n",
            "tr 1551\n",
            "tbody 1550\n",
            "table 1545\n",
            "thead 1511\n",
            "bold 1497\n",
            "th 1483\n",
            "ext-link 1182\n",
            "table-wrap-foot 797\n",
            "fn 582\n",
            "supplementary-material 580\n",
            "media 574\n",
            "disp-formula 537\n",
            "math 507\n",
            "mo 493\n",
            "mrow 492\n",
            "mi 480\n",
            "list 429\n",
            "list-item 429\n",
            "alt-text 411\n",
            "col 408\n",
            "msub 398\n",
            "mn 364\n",
            "break 356\n",
            "inline-formula 308\n",
            "inline-graphic 297\n",
            "uri 281\n",
            "mfrac 280\n",
            "alternatives 262\n",
            "colgroup 259\n",
            "hr 250\n",
            "msup 244\n",
            "sc 222\n",
            "mfenced 205\n",
            "mspace 201\n",
            "mtext 198\n",
            "tex-math 190\n",
            "msubsup 155\n",
            "mtable 120\n",
            "mtd 120\n",
            "mtr 120\n",
            "boxed-text 119\n",
            "underline 118\n",
            "object-id 106\n",
            "munder 89\n",
            "funding-source 89\n",
            "munderover 88\n",
            "permissions 83\n",
            "copyright-holder 80\n",
            "mover 72\n",
            "mstyle 70\n",
            "msqrt 59\n",
            "institution 59\n",
            "institution-id 59\n",
            "institution-wrap 59\n",
            "def 58\n",
            "def-item 58\n",
            "def-list 58\n",
            "term 58\n",
            "styled-content 53\n",
            "glyph-data 24\n",
            "private-char 24\n",
            "named-content 21\n",
            "x 19\n",
            "glyph-ref 17\n",
            "disp-quote 13\n",
            "attrib 10\n",
            "notes 9\n",
            "monospace 9\n",
            "inline-supplementary-material 9\n",
            "statement 8\n",
            "copyright-statement 8\n",
            "copyright-year 8\n",
            "license 8\n",
            "license-p 8\n",
            "free_to_read 7\n",
            "license_ref 7\n",
            "array 6\n",
            "mpadded 4\n",
            "disp-formula-group 4\n",
            "fig-group 3\n",
            "preformat 3\n",
            "mmultiscripts 2\n",
            "mprescripts 2\n",
            "strike 2\n",
            "long-desc 1\n",
            "none 1\n",
            "tfoot 1\n",
            "email 1\n",
            "table-wrap-group 1\n",
            "related-article 1\n",
            "chem-struct 1\n",
            "chem-struct-wrap 1\n",
            "roman 1\n",
            "mroot 1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3UWLRSq4xCUk"
      },
      "source": [
        "#### 6.1.1 Example: section titles\n",
        "\n",
        "Function to find section titles to view the structure of an article using the find_all() method to return all `<title`> tags for each article."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "qpbieRlLxPk1"
      },
      "outputs": [],
      "source": [
        "def find_section_titles(article):\n",
        "    \"\"\"\n",
        "    Return list of sections from an article\n",
        "    \"\"\"\n",
        "\n",
        "    section_titles_list = []\n",
        "    section_titles = article.find_all(\"title\")\n",
        "    for section_title in section_titles:\n",
        "        title = section_title.text\n",
        "        section_titles_list.append(\n",
        "            {\n",
        "                \"section_title\": title,\n",
        "            }\n",
        "        )\n",
        "    return section_titles_list"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View section titles for article 0"
      ],
      "metadata": {
        "id": "t0m2I4GRS7rr"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "section_titles_article_0 = find_section_titles(soup_results_1[0])\n",
        "section_titles_article_0"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Yhx4pnLBW95b",
        "outputId": "df7e3ff2-37db-417f-9ffd-23acf1ab4e08"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'section_title': '1 Introduction'},\n",
              " {'section_title': '2 Methodology and data processing'},\n",
              " {'section_title': '2.1 Data collection'},\n",
              " {'section_title': '2.2 Data import and deduplication'},\n",
              " {'section_title': '2.3 Data splitting or merging'},\n",
              " {'section_title': '2.4 Data analysis and visualization'},\n",
              " {'section_title': '3 Results'},\n",
              " {'section_title': '3.1 Number and type of publications'},\n",
              " {'section_title': '3.2 Countries and number of publications'},\n",
              " {'section_title': '3.3 National/regional cooperation'},\n",
              " {'section_title': '3.4 Contributions of leading bodies'},\n",
              " {'section_title': '3.5 Contribution of leading research areas'},\n",
              " {'section_title': '3.6 Contribution of major journals'},\n",
              " {'section_title': '3.7 Contribution of the lead author'},\n",
              " {'section_title': '3.8 Research hotspots and trends'},\n",
              " {'section_title': '3.8.1 Author keyword analysis'},\n",
              " {'section_title': '3.8.2 Analysis of hot research topics'},\n",
              " {'section_title': '3.8.3 Analysis of the most cited studies'},\n",
              " {'section_title': '4 Discussion'},\n",
              " {'section_title': '5 Conclusion'},\n",
              " {'section_title': '6 Limitations'}]"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View section titles for article 1"
      ],
      "metadata": {
        "id": "9VPCu64gTBmY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "section_titles_article_1 = find_section_titles(soup_results_1[1])\n",
        "section_titles_article_1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nyzFN2KAXKpQ",
        "outputId": "4567d6c8-38f9-4451-9482-a6104ad173da"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'section_title': 'INTRODUCTION'},\n",
              " {'section_title': 'COMPUTATIONAL DRUG DISCOVERY APPROACHES FOR COVID‐19'},\n",
              " {'section_title': 'Ligand‐based drug design for COVID‐19'},\n",
              " {'section_title': 'Structure‐based drug design and molecular docking'},\n",
              " {'section_title': 'Chemogenomic approaches for COVID‐19 drug discovery'},\n",
              " {'section_title': 'Target fishing Chemogenomics approaches for COVID‐19 drug discovery'},\n",
              " {'section_title': 'Drug repurposing application of Chemogenomics for COVID‐19 drug discovery'},\n",
              " {'section_title': 'Predicting the bio‐profile of drugs via Chemogenomics for COVID‐19 drug discovery'},\n",
              " {'section_title': 'DRUG REPOSITIONING'},\n",
              " {'section_title': 'Drug repositioning for COVID‐19 drug discovery'},\n",
              " {'section_title': 'CONCLUSION AND FUTURE PROSPECTS'},\n",
              " {'section_title': 'FUNDING INFORMATION'},\n",
              " {'section_title': 'Supporting information'}]"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 6.2 Find direct child tags of `<body>` tag\n",
        "\n",
        "Function to find direct child tags of `<body>` tag using the `.name` attribute and append to a list."
      ],
      "metadata": {
        "id": "3O85qiU0doDQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_body_child_tags(article):\n",
        "\n",
        "    body_child_list = []\n",
        "\n",
        "    try:\n",
        "        for tag in article:\n",
        "            if tag.name != None:\n",
        "                  body_child_list.append(tag.name)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "    return sorted(list(set(body_child_list)))"
      ],
      "metadata": {
        "id": "GJgy_EdoA5wG"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "body_child_tags = list(map(find_body_child_tags, soup_results_1))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "I0zdO6yNA9Zi",
        "outputId": "5cf5b36b-051c-41e6-8f3a-234c543fd34c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Printing out the tags reveals some empty lists for the rows giving 'NoneType' object exceptions.\n",
        "\n"
      ],
      "metadata": {
        "id": "1mYV9MtfeBBZ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "body_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "okgFBGgjBH56",
        "outputId": "4050c449-044c-4ebc-f37c-7cf9939820c7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['graphic', 'p'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['def-list', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['boxed-text', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['def-list', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'table-wrap'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['def-list', 'sec'],\n",
              " ['sec'],\n",
              " ['boxed-text', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec', 'table-wrap'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'table-wrap'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec', 'table-wrap'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['def-list', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['boxed-text', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['p'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['p', 'sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['fig', 'sec'],\n",
              " [],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ['sec'],\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 21
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in body_child_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "BNh7eMTke2o8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_body_child_tags = sum(body_child_tags, [])\n",
        "sum_body_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "FUzoXjdPBLDL",
        "outputId": "58f3103c-eb78-4580-ebc0-976b9f41eda9"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'graphic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'def-list',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'boxed-text',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'def-list',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'table-wrap',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'def-list',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'boxed-text',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'table-wrap',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'def-list',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'boxed-text',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " 'sec',\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again use the Counter class with most_common() method to return a list of the *n* most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "3XaW866XfauP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_body_child_tags = Counter(sum_body_child_tags).most_common()\n",
        "num_unique_body_child_tags = num_unique_body_child_tags[::]\n",
        "\n",
        "for tag, count in num_unique_body_child_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "B0IHSWrSXoYx",
        "outputId": "7dfb7542-e1ce-4c0c-af1c-9fcd73cfa82f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "sec 2438\n",
            "p 149\n",
            "fig 40\n",
            "def-list 24\n",
            "table-wrap 8\n",
            "boxed-text 5\n",
            "disp-quote 3\n",
            "graphic 2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 6.3 Find all direct child tags\n",
        "\n",
        "You can iterate over a tag's direct children using the `.children` generator."
      ],
      "metadata": {
        "id": "ewULpg-agEG_"
      }
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "T862qM-BSBw3"
      },
      "outputs": [],
      "source": [
        "def find_child_tags(article: str):\n",
        "\n",
        "    child_tags_list = []\n",
        "\n",
        "\n",
        "    try:\n",
        "        for tag in article:\n",
        "            for child in tag.children:\n",
        "                if child.name != None:\n",
        "                  child_tags_list.append(child.name)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "\n",
        "\n",
        "    return sorted(list(set(child_tags_list)))\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "child_tags = list(map(find_child_tags, soup_results_1))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "EaWmKBu3-_Du",
        "outputId": "02e8a969-576c-4e49-d0c7-368239d62682"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NavigableString' object has no attribute 'children'\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'Comment' object has no attribute 'children'\n",
            "'NavigableString' object has no attribute 'children'\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n",
            "'NoneType' object is not iterable\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again the empty list 'NoneType' objects, but three additional attribute errors this time.\n",
        "\n",
        "Two instances of 'NavigableString' object has no attribute 'children', possibly caused by newlines and spaces in markup between nodes which Beautiful Soup turns into NavigableStrings. A string does not have children because it cannot contain anything in the way a tag may contain a string or another tag.\n",
        "\n",
        "And one instance of 'Comment' object which represents a comment in the XML document.\n",
        "According to the [Beautiful Soup documentation](https://www.crummy.com/software/BeautifulSoup/bs4/doc/#bs4.Comment) this is a special type of NavigableString that adds something extra to the string on output, displaying with special formatting. Again, because the content of the comment is a string it cannot have child nodes."
      ],
      "metadata": {
        "id": "0XugvP3JghS2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sZ1w8sRWdHrS",
        "outputId": "f443c4b3-d77a-4f9a-b17d-805927fb448d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['bold', 'fig', 'italic', 'list', 'table-wrap', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'list', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'italic', 'table-wrap', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['sec', 'sup', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['disp-quote', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['ext-link', 'fig', 'italic', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'italic', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['italic', 'p', 'sec', 'title', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " [],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['fig', 'p', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " [],\n",
              " ['def-item',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['caption', 'list', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table-wrap',\n",
              "  'title',\n",
              "  'underline',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'sub', 'sup', 'title'],\n",
              " ['def-item', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " [],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['caption', 'graphic', 'italic', 'label', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['caption', 'graphic', 'label', 'p', 'sub', 'sup', 'title', 'xref'],\n",
              " ['p', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['fig', 'label', 'p', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['list', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'sup', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'table-wrap',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['list', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['caption', 'graphic', 'label', 'list', 'p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title', 'xref'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " [],\n",
              " ['caption',\n",
              "  'fig',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'table-wrap',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['boxed-text', 'ext-link', 'p', 'sec', 'title', 'xref'],\n",
              " ['fig', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['caption', 'graphic', 'label', 'p', 'title', 'xref'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['fig', 'p', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sub',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['caption', 'graphic', 'italic', 'label', 'table', 'xref'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['def-item', 'fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['disp-quote', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text',\n",
              "  'fig',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap-foot',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'ext-link',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'table',\n",
              "  'table-wrap-foot',\n",
              "  'xref'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text',\n",
              "  'fig',\n",
              "  'list',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table-wrap',\n",
              "  'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'title'],\n",
              " ['caption', 'label', 'p', 'table', 'title', 'xref'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['boxed-text',\n",
              "  'fig',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table-wrap',\n",
              "  'title'],\n",
              " ['caption', 'fig', 'graphic', 'label', 'p', 'sec', 'sup', 'title', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sup', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['boxed-text', 'fig', 'list', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['disp-quote', 'fig', 'p', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-item', 'label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['def-list', 'label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " [],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'list', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['italic', 'p', 'sec', 'supplementary-material', 'title', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'graphic', 'label', 'list', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'sub', 'supplementary-material', 'title', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['bold', 'italic', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'sup', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text', 'fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['caption',\n",
              "  'ext-link',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sub',\n",
              "  'sup',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " [],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['bold',\n",
              "  'caption',\n",
              "  'graphic',\n",
              "  'italic',\n",
              "  'label',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'sup',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['italic', 'p', 'title', 'xref'],\n",
              " ['boxed-text',\n",
              "  'ext-link',\n",
              "  'fig',\n",
              "  'italic',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'title',\n",
              "  'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['boxed-text',\n",
              "  'fig',\n",
              "  'p',\n",
              "  'sec',\n",
              "  'supplementary-material',\n",
              "  'table-wrap',\n",
              "  'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['label', 'p', 'title'],\n",
              " ['fig', 'p', 'sec', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'sup', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'supplementary-material', 'title'],\n",
              " ['italic', 'p', 'title', 'xref'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['graphic', 'label', 'p', 'sec', 'title'],\n",
              " [],\n",
              " ['p', 'sec', 'title'],\n",
              " ['boxed-text', 'p', 'sec', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'label', 'p', 'sec', 'supplementary-material', 'table-wrap', 'title'],\n",
              " ['label', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['fig', 'p', 'sec', 'title'],\n",
              " ['list', 'p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'title'],\n",
              " ['p', 'sec', 'supplementary-material', 'title'],\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(child_tags)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zS_DsI8AdLpS",
        "outputId": "52284a8b-c361-4930-df68-1ba64d9127e4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 27
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in child_tags nested list and return the sum of all elements."
      ],
      "metadata": {
        "id": "RIuo9TeKiO8l"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_child_tags = sum(child_tags, [])\n",
        "sum_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aP7Uh0lLbi-U",
        "outputId": "40cfadab-68f5-481f-81f2-e4417442f6c6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'bold',\n",
              " 'fig',\n",
              " 'italic',\n",
              " 'list',\n",
              " 'table-wrap',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'list',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'italic',\n",
              " 'table-wrap',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'def-list',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'title',\n",
              " 'disp-quote',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'italic',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'italic',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'def-list',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'def-list',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'boxed-text',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'def-item',\n",
              " 'italic',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'caption',\n",
              " 'list',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'table-wrap',\n",
              " 'title',\n",
              " 'underline',\n",
              " 'xref',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'title',\n",
              " 'def-item',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'supplementary-material',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'def-list',\n",
              " 'label',\n",
              " 'p',\n",
              " 'sec',\n",
              " 'title',\n",
              " 'fig',\n",
              " 'p',\n",
              " 'sec',\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again, counts from most to least common."
      ],
      "metadata": {
        "id": "vO50CBCKidNU"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_child_tags = Counter(sum_child_tags).most_common()\n",
        "num_unique_child_tags = num_unique_child_tags[::]\n",
        "\n",
        "for tag, count in num_unique_child_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "W1S-POaobuzv",
        "outputId": "3247c5a6-54bf-446e-839c-625ccbcf4b4a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "title 2434\n",
            "p 2425\n",
            "sec 2210\n",
            "label 715\n",
            "fig 449\n",
            "supplementary-material 286\n",
            "table-wrap 220\n",
            "xref 109\n",
            "italic 81\n",
            "boxed-text 80\n",
            "sup 64\n",
            "caption 42\n",
            "graphic 40\n",
            "sub 37\n",
            "list 31\n",
            "def-list 27\n",
            "def-item 24\n",
            "ext-link 21\n",
            "bold 19\n",
            "table 7\n",
            "table-wrap-foot 6\n",
            "uri 5\n",
            "disp-quote 4\n",
            "sc 3\n",
            "disp-formula 2\n",
            "underline 1\n",
            "inline-formula 1\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_europepmc_child_tags_2500.pickle', 'wb') as f:\n",
        "  pickle.dump(child_tags, f)"
      ],
      "metadata": {
        "id": "d5XbE_V6dODK"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 6.4 Create single sorted list of child tags\n",
        "\n",
        "From itertools module import chain() function to group nested list into a single sorted iterable."
      ],
      "metadata": {
        "id": "jJlZHHcoPOol"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sorted(list(set(chain(*child_tags))))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9Fkj1sQgZvce",
        "outputId": "939bfb94-eeab-4c00-97e0-4eb51efe8dac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['bold',\n",
              " 'boxed-text',\n",
              " 'caption',\n",
              " 'def-item',\n",
              " 'def-list',\n",
              " 'disp-formula',\n",
              " 'disp-quote',\n",
              " 'ext-link',\n",
              " 'fig',\n",
              " 'graphic',\n",
              " 'inline-formula',\n",
              " 'italic',\n",
              " 'label',\n",
              " 'list',\n",
              " 'p',\n",
              " 'sc',\n",
              " 'sec',\n",
              " 'sub',\n",
              " 'sup',\n",
              " 'supplementary-material',\n",
              " 'table',\n",
              " 'table-wrap',\n",
              " 'table-wrap-foot',\n",
              " 'title',\n",
              " 'underline',\n",
              " 'uri',\n",
              " 'xref']"
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(sorted(list(set(chain(*child_tags)))))"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5S4kOzmaNsG",
        "outputId": "921fa0af-2689-43b7-d615-4e2c346782fd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "27"
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vNP947nwWHqk"
      },
      "source": [
        "## 7. Remove unwanted tags and content\n",
        "\n",
        "Functions to remove unwanted tags and their contents using BeautifulSoup's decompose() method."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "bHge3e9RWSP4"
      },
      "outputs": [],
      "source": [
        "def remove_section_titles(article):\n",
        "\n",
        "    for title in article(\"title\"):\n",
        "        title.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_figures(article):\n",
        "\n",
        "    for fig in article(\"fig\"):\n",
        "        fig.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_def_lists(article):\n",
        "\n",
        "    for def_list in article(\"def_list\"):\n",
        "        def_list.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_tables(article):\n",
        "\n",
        "    for table_wrap in article(\"table-wrap\"):\n",
        "        table_wrap.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_formulas(article):\n",
        "\n",
        "    for inline_formula in article(\"inline-formula\"):\n",
        "        inline_formula.decompose()\n",
        "    for disp_formula in article(\"disp-formula\"):\n",
        "        disp_formula.decompose()\n",
        "    for disp_formula_group in article(\"disp-formula-group\"):\n",
        "        disp_formula_group.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_ext_links(article):\n",
        "\n",
        "    for ext_link in article(\"ext-link\"):\n",
        "        ext_link.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_labels(article):\n",
        "\n",
        "    for label in article(\"label\"):\n",
        "        label.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_refs(article):\n",
        "\n",
        "    for xref in article(\"xref\"):\n",
        "        xref.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_supp_material(article):\n",
        "\n",
        "    for supp_material in article(\"supplementary-material\"):\n",
        "        supp_material.decompose()\n",
        "\n",
        "    return article\n"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Function to call the functions above on each article and return new article list."
      ],
      "metadata": {
        "id": "zwXYABLRyWBv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def remove_tags(article):\n",
        "\n",
        "    new_article_list = []\n",
        "\n",
        "    try:\n",
        "       if article != None:\n",
        "          new_article = remove_section_titles(article)\n",
        "          new_article = remove_figures(new_article)\n",
        "          new_article = remove_def_lists(new_article)\n",
        "          new_article = remove_tables(new_article)\n",
        "          new_article = remove_formulas(new_article)\n",
        "          new_article = remove_ext_links(new_article)\n",
        "          new_article = remove_labels(new_article)\n",
        "          new_article = remove_refs(new_article)\n",
        "          new_article = remove_supp_material(new_article)\n",
        "          new_article_list.append(new_article)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "    return new_article_list"
      ],
      "metadata": {
        "id": "itZOkcn7C4US"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.1 First batch of 2500 remove tags\n",
        "\n",
        "Map the remove_tags() function on to first batch of 2500 articles to remove unwanted tags and content."
      ],
      "metadata": {
        "id": "VwAm9DUDC0pq"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_removed_tags_2500 = list(map(remove_tags, soup_results_1))"
      ],
      "metadata": {
        "id": "rMA2QIJsEspS"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_removed_tags_2500)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5SIA2LWnE89C",
        "outputId": "dc0d1586-e321-46cf-bc1a-b08ff666581c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 36
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View list of first 10 articles with unwanted tags and contents removed."
      ],
      "metadata": {
        "id": "NhBaZ8VOoXt4"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_removed_tags_2500[0:10]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0ukufmQKYpJQ",
        "outputId": "111c824d-b47f-4345-84f3-1ecb524feef3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[[<body><sec id=\"s1\"><p>Sir James Black, a winner of the 1988 Nobel Prize, clearly recognized well before the 21st century that drug repurposing strategies would occupy an important place in the future of new drug discovery (). In 2004, Ted T. Ashburn et al. () summarized previous research and developed a general approach to drug development using drug repurposing, retrospectively looking for new indications for approved drugs and molecules that are waiting for approval for new pathways of action and targets. These molecules are usually safe in clinical trials but do not show sufficient efficacy for the treatment of the disease originally targeted (). The definition of the term “drug repurposing” has been endorsed by scholars () and used by them (; ). It should be pointed out that the synonyms of “drug repurposing” often used by academics also include drug repositioning (), drug rediscovery (), drug redirecting (), drug retasking (), and therapeutic switching (; ). After the research study by Ashburn et al., Allarakhia et al. expanded the starting materials for drug repositioning to include products that were discontinued for commercial reasons, expired patents, and candidates for laboratory testing (). In the discovery process of a completely new drug, the difficulty usually lies in its safety and efficacy, which are the main potential causes of failure of most drugs in the approval () or clinical development stage (). Using existing knowledge about a drug or known target (), the time, risk, and cost of developing a drug using drug repositioning are reduced (), thereby greatly increasing the efficiency and economics of drug development, providing a better risk–reward trade-off, and making it easier to win the favor of venture capital firms ().</p><p>Since the 1990s, the repositioning of sildenafil for male erectile dysfunction () and pulmonary hypertension (), the development of a new efficacy of bupropion for smoking cessation (), new applications of thalidomide for multiple myeloma (; ), and chronic graft-versus-host disease () have generated intense interest from pharmaceutical companies and academics (). These classic success stories rely on three traditional approaches: 1) molecular biology approaches (), 2) <italic>in vivo</italic> and <italic>ex vivo</italic> experimental approaches (; ), and 3) expert knowledge-based approaches (). Due to the unknown, complex, and information-fragmented nature of drug candidates and potential new mechanisms of action (), this activity is dependent on multiple factors, and success is often fortuitous (). At the beginning of the 21<sup>st</sup> century, cheminformatics (; ), bioinformatics (; ), systems biology (; ), genomics (; ), polypharmacology (; ), precision medicine (; ), and other disciplines, combined with artificial intelligence (), have developed rapidly. These rapidly growing disciplines have promoted the generation of systematic () computer methods to make the drug repositioning process cheaper and shorter (; ). Computational drug repositioning is classified as “disease-centric” or “target/gene-centric” or “drug-centric” depending on the source of discovery (). This process relies on public biochemical databases such as DrugBank (; ), ChEMBL (), Cmap (), PDB (), OMIM (), etc., to provide the appropriate information. In fact, to make the computational drug repurposing process, including the molecular docking and virtual screening steps, more convenient, database tools specifically developed for drug repurposing, such as EK-DRD (), DREIMT (), DrugSig (), RepoDB (), Promiscuous 2.0 (), etc., have been reported in the last few years. In addition, it has been found in the literature that only 10% of the research results have been carried out in the “drug-centric” pathway, which holds great prospects for future development (). With the help of database tools, it is now possible to perform computational screening of even a staggering number of hundreds of millions of compounds (). Computer methods to carry out this screening include machine learning (), network modeling (; ), text mining, and semantic reasoning (; ; ), among others. The ultimate objective of repositioning is to transfer one or two of the most relevant results to clinical applications. Therefore, validation is quite important () and requires consideration of multiple factors, such as price, toxicity levels, bioavailability, and differences between validated and computational models (; ). Current validation methods include experimental validation (), electronic health records to aid validation (), cross-validation (; ), gold standard dataset evaluation (), literature citation validation (), and expert consultation ().</p><p>Today, drug repositioning is increasingly prominent in the development of drugs for a variety of neurological diseases (; ), cancer (; ), rare diseases (; ), and infectious diseases (; ). An increasing number of pharmaceutical companies are also establishing relevant R&amp;D programs () or funding support (). To translate relevant research results efficiently and smoothly, national departments within the United Kingdom, the United States, and the Netherlands have (; ) launched initiatives or programs to build partnerships between pharmaceutical companies and academia and to further explore scientific and commercial opportunities (). It is certain that drug repositioning currently presents several dilemmas, such as intellectual property challenges (), data platforms, and analytical techniques that need to be improved (), that financial support remains important for technology development and clinical trials (), and that some scientists deny the practical utility of the approach ().</p><p>There have been systematic analyses of terminology in the drug repurposing literature (), text mining of drug–disease combinations (), and the progression of a particular drug (), but no studies have yet provided a broad overview of publications on the topic of drug repurposing research. When independent researchers or collectives (including pharmaceutical companies, academia, and government departments) seek drug repurposing partnership partners and seek to obtain a concise overview of comprehensive current research hotspots, the lack of relevant intelligence analysis to aid decision-making often makes the process convoluted and time-consuming (). The bibliometric approach can solve the aforementioned problems relatively fairly, but at present, scholars have only studied the bibliometrics of aspirin, a drug repurposing (); there has not been a panoramic study of drug repurposing, and therefore, this study is necessary. Bibliometrics is a useful tool combining multiple parameters for the quantitative analysis of scholarly publications and is currently used to assess research hotspots and trends in a wide range of disciplines and industries, such as management (; ), sociology (; ), economics (), medicine (; ), environmental engineering (; ), and agronomy (; ). Therefore, this study uses bibliometric methods () to quantitatively assess the following elements of drug repositioning publications: 1) major contributors: countries, research institutions, and authors; 2) modes of collaboration: intercountry collaborations; 3) the most productive journals; 4) the most frequently used disciplinary knowledge; and 5) research trends, judged by analyzing author keywords, Essential Science Indicators (ESI) high citations, and hot research studies.</p></sec><sec id=\"s2\"><sec id=\"s2-1\"><p>We use the Web of Science™ core database, an authoritative academic information data service platform produced by Clarivate (version <sup>©</sup> 2021 Clarivate.). Due to its rigorous selection of journals, the Web of Science (WOS) Core Collection Database is now internationally recognized as a database for evaluating the scientific output or disciplinary development of scholars and institutions. Among the subdatabases, SCI-E mainly includes global journals in basic science research, covering basic pharmacological and medical research related to the theme of this study, “drug repositioning,” while SSCI includes social science, covering ethical, nursing, psychological, and other social science research related to this study.</p><p>The data were obtained on 25 October 2021 through the WOS Core Collection Database Citation Indexes SCI-E and SSCI, using the formula “drug repurposing” OR “drug repositioning” OR “drug rediscovery” OR “therapeutic switching” OR “drug redirecting” OR “drug rediscovery” OR “drug retasking” search query, searching in the “subject” field and defining the document type as “Article” and “Review”. The publication time parameters were initially limited to publications related to “drug repositioning” published between 1990 and 2020. A total of 3,009 documents were obtained, of which only 31 were published in two decades from 1990 to 2009. Of these 31 documents, except for one document that is still frequently used by scholars as a retrospective source for drug repurposing definitions in these years (), the remaining 30 were cited by other authors during the period of 2010–2020 as shown in . The overall level of interest in these studies shows a fluctuating downward trend as opposed to the rising citation fervor for drug repurposing, entering a stage of decline even under the less-demanding evaluation criterion of a 5-year maturation window (). As the literature ages, its content becomes stale and obsolete in the perspective of intelligence sources, and the value of the metrics for judging current research trends is low. Therefore, we further narrowed the study to 2,978 publications published from 2010 to 2020.</p></sec><sec id=\"s2-2\"><p>The complete records of all retrieved documents are downloaded and imported for processing into Derwent Data Analyzer (DDA) version V10, a data cleaning, multiperspective data mining, and visualization software from Clarivate that improves data analysis efficiency and reduces labor costs. After importing all records of WOS documents into DDA, they are classified and measured according to a list of fields such as keyword, country/region, institution, author, research field, journal, etc. For each item in the list fieldset, DDA has a built-in data cleaning tool for automatic data deduplication.</p></sec><sec id=\"s2-3\"><p>After the machine has removed duplicates, the items in the set of fields still need to be manually verified for splitting or merging. It is to be noted that the regions of certain countries are presented separately, while they are usually considered as a single country internationally. Therefore, we need to perform merging, such as combining Wales, Scotland, England, and Northern Ireland into the United Kingdom column and combining Hong Kong and Macau regions into the China column.</p><p>To address the possible problem of different authors with the same name, the following two main verification steps were performed: 1) returning to the WOS database to search for publications under that author’s name under the original search formula conditions and 2) for authors whose publications provide disputed information (this also includes three cases: first, two or more authors with the same name but not the same person; second, two or more authors with the same abbreviated name, but the full names were found to be different after a search; and third, similar signatures being different variants of the same author’s name), in addition to searching the ORCID-related information of the authors concerned for judgment, we checked different institutional websites as well as encyclopedias to look for changes in the study and work history of authors with the same or similar names from 2010 to 2020 to determine whether they were the same person. Based on the verification, we then split or merged the results.</p></sec><sec id=\"s2-4\"><p>After data cleaning and matrix analysis by DDA, various types of cluster plots and bubble plots can be obtained to reveal the useful information behind the data. The bibliometric fields of publication volumes, countries, international collaborations, institutions, research areas, journals, authors, highly cited research studies, and author keywords were analyzed in this study. It should be noted that because some studies were published online ahead of time and the study publication date was a year or two behind, for statistical purposes, the year of publication of such research studies was included as the year of online publication. (e.g., a study shown in the reference as published in 2022 may have been published online in 2020).</p></sec></sec><sec id=\"s3\"><sec id=\"s3-1\"><p>Of the 2,978 papers obtained using the search criteria mentioned previously, the main ones were research studies (2248; 75.49%) and reviews (730; 24.51%). Furthermore, individual publications are not only classified by journals in the single category of research studies or reviews but also belong to other categories. These publications were also related to proceeding studies (68; 2.28%), early access (24; 0.81%), book chapters (7; 0.24%), data studies (2; 0.07%), and retracted publications (2; 0.07%). The vast majority of research studies and reviews were published in English (2967; 99.631%), with the remainder in Japanese (3; 0.101%), Chinese (2; 0.067%), Czech (1; 0.034%), French (1; 0.034%), German (1; 0.034%), Hungarian (1; 0.034%), Korean (1; 0.034%), and Portuguese (1; 0.034%). Ninety were from SSCI, and the remaining 2888 were from SCI-E. Further, 1,996 were from Open Access. An annual analysis of published research studies is shown in . The number of publications for every year expanded from 17 in 2010 to 970 in 2020. Annual publications on the subject have increased by more than 64 times. The number of annual publications has been increasing at a relatively high rate since 2015, while in 2020, there was a spike in the number of publications and annual citations, probably due to the COVID-19 pandemic, a global public health emergency that prompted special attention from scientists. Among the four countries with the highest number of publications (the United States, China, the United Kingdom, and India), the United States has maintained a high growth volume since 2010, while China was the fastest in terms of average annual growth rate in the last three years. In 2020, the number of publications in India surged and surpassed the production of the United Kingdom.</p></sec><sec id=\"s3-2\"><p>With respect to the 2978 publications related to drug repositioning research, 89 countries contributed to the field of drug repositioning research. The number of publications and citations from the 20 most productive countries/regions is shown in . There are nine countries/regions in Europe, five in the Americas, five in Asia, and one in Oceania.</p><p>The four most productive countries/regions are, in order, the United States, China, the United Kingdom, and India. The United States is the absolute leader in this field, with 918 research studies on drug repositioning published since 2010, which is already more than the next highest number of publications in China and the United Kingdom combined. This is followed by India (247), Italy (232), Germany (171), South Korea (161), and Japan (146). Other productive countries include Brazil (125), France (116), Canada (111), Spain (109), Australia (73), the Netherlands (73), and Switzerland (59). In terms of publication impact, the United States led the Total citations (TC) rankings with 27,355, twice as many as that of China (11,147), which ranked second. We also included the average citations per publication (ACPP) in the comparison, which is calculated by dividing the TC by the TP (total papers) value and is a relative number that may better reflect the individual or collective level of attention than the individual TC and TP values. Canada ranked first in ACPP at 41.81, closely followed by the United Kingdom (30.85) and Germany (30.49). In addition, the h-index was originally proposed as a simple quantification that a researcher had at least h publications cited h times, reflecting to a certain extent the research results of the researcher as an individual (). Later, the word “researcher” in the definition began to be replaced by collective words such as “academic group or institution (),\" “journal (),\" and “country (),\" becoming an indicator of the level of collective research to some extent. Undoubtedly, the h-index of the United States ranks first in this field with 74 times. Taking all parameters into account, we find that publications in the United Kingdom, the United States, and Canada perform better on average. While the number of publications in China and India is significant, they have received low levels of attention.</p></sec><sec id=\"s3-3\"><p>It should be noted that DDA analysis software is nationally identified based on the location of each researcher’s institution address provided in the publication. If a publication is coauthored by institutions from more than two countries, the publication is defined as the result of an international collaboration. Whether there is some affiliation between the various institutions of the research group that produces the multicountry collaboration is not taken into account. As shown in , among the publications of the top 20 countries and regions, the proportion of international collaborations is quite high in European countries, especially in Sweden (86.21%) and Belgium (81.25%). Asian and Latin American countries are generally underrepresented. In addition, the United States, the most active country in publishing and the country with the most collaborations—with 59 countries or regions—still has over 50% of the studies published overall.</p><p>\n",
              "   depicts the academic collaboration network for the top 20 countries and regions in terms of productivity. Using DDA software, the network was mapped using a co-occurrence matrix. The size of the circles is proportional to the extent of each country’s contribution, the lines between the circles represent the collaboration between countries/regions, and the thickness of the connecting lines indicates the frequency of collaboration (). The results show that the United States cooperates most frequently with China and the United Kingdom and has the closest cooperation with them. In addition, Mexico, Belgium, Argentina, Taiwan, Japan, and Korea have slightly sparser cooperation networks among the 20 most productive countries/regions, while the remaining countries have more extensive cooperation networks among themselves.</p></sec><sec id=\"s3-4\"><p>A total of 3,530 institutes were involved in drug repositioning research. The top 20 productive institutes are shown in . Eight of the top 20 institutions are from the United States, again indicating the dominance of the United States in drug repositioning research; three are from the United Kingdom; two are from China; and Brazil, France, Mexico, Canada, the Netherlands, Austria, and Sweden each have one. The Chinese Academy of Science ranked first in terms of the number of research studies, followed by Case Western Reserve University and the NIH. The Icahn School of Medicine at Mt Sinai ranked first in ACPP at 77.32. The NIH had the highest h-index value of 22. The Icahn School of Medicine at Mt Sinai was the best performer in ACPP at 77.32, followed by the University of California, San Francisco (67.82) and Johns Hopkins University (65.68), both of which are US-based research institutions. Compared with US research institutions, Chinese research institutions are at the back of the pack in terms of ACPP, and their impact needs to be improved.</p><p>The collaboration network between the 15 largest institutions in 2010–2020 is shown in . The collaboration network provides a more visual view of the collaboration with different institutions and thus helps in the search for more beneficial collaborations. Next to the name of each institution is its total number of publications. At the intersections of these institutions, yellow dots indicate collaborations with the other top 10 research institutions. It should be noted that the number of yellow dots can indicate the output of cooperation and the strength of interagency cooperation. The nodal data with no crossover points represent the number of publications produced by the institute, either by its independent work or in collaboration with research institutions outside the top 15 (). From , we see that the University of Cambridge established the largest collaborative network, followed by the large network established by four institutions, the NIH, the Icahn School of Medicine at Mt Sinai, Karolinska Institute, and King’s College London. In terms of the number of copublications with established institutions, the Chinese Academy of Science and Shanghai Jiao Tong University copublished as many as six, followed by the University of Cambridge and King’s College London and the NIH and the Icahn School of Medicine at Mt Sinai. Analyzing the aforementioned three pairs of institutional combinations, King’s College London has two publications that are the product of collaboration between the three research institutions. The University of Sao Paulo and Aix-Marseille University are relatively independent in this research area. Combining the ranking of multiple parameters, we found that the NIH and Icahn Sch Med Mt Sinai in the United States are the most vocal institutions in terms of academic research result perspective on the topic.</p></sec><sec id=\"s3-5\"><p>An analysis of research areas gives a good indication of the scope of application of the research topic, with an unrestrained number of 74 areas covered, with the top 20 areas of research in terms of publication volume shown in . Briefly, “pharmacology and pharmacy” took first place with 962 articles, followed by “biochemistry and molecular biology”, and for ACPP, the top three were science and technology-other topics (36.1)\", “mathematics (32.79)\", and “cell biology (29.65)\".</p><p>\n",
              "   shows a bubble graph of the top 20 drug repositioning research areas. The bubble plot shows three dimensions of the data, namely, research area, year of publication, and the number of publications. The horizontal change in bubble size illustrates the growing trend of research areas over time, the vertical size of the bubble shows the most popular research areas in that year, and the number in the bubble indicates the frequency of the topic in the research area and the number of publications in that year (). The number of research results in each relevant field is increasing year by year. Biophysics increased from five in 2019 to 77 in 2020, a more than 15-fold increase, suggesting that drug repositioning may have made a breakthrough or become widely used in this field. The field of virology was in a downturn from 2010 to 2014, with only one publication, with a gradual increase in relevant studies after 2015.</p></sec><sec id=\"s3-6\"><p>For scholars studying drug repositioning-related topics, knowing which journals publish relevant research is important in deciding which journals to read or submit their research studies to. A total of 2,988 publications related to drug repositioning research were published in 845 journals during the period of 2010–2020. The top 25 journals in terms of a total number of studies published are shown in  Sci Rep topped the list with 75 studies published, followed by <italic>PLoS One</italic> (73; 2.52%), <italic>J. Biomol. Struct. Dyn</italic> (67; 2.45%), <italic>Bioinformatics</italic> (53; 2.25%), and <italic>BMC bioinformatics</italic> (50 articles; 1.78%). The rest of the journals had a share of less than 1.5%. In terms of total citations (TC), at present, studies in <italic>Drug Discov.</italic> have been cited a total of 2,119 times over the past 10 years, followed in rank by PLoS One (1800) and Bioinformatics (1,677). For ACPP, <italic>Drug Discov. Today</italic> still holds first place with a high frequency of 50.45 times, followed by <italic>PLoS Comput</italic> (33.14 times). The impact factor (IF) of a journal is calculated by dividing the total number of citations of all publications in the journal in the previous two years by the number of publications (). Thus,  shows that the ACPP of drug repurposing publications included in most journals is much higher than that of IF, which roughly verifies that the number of scholars interested in drug repurposing is relatively high. In terms of the impact factor (IF) of specific journals, except for <italic>Oncotarget</italic> and B<italic>MC Syst. Biol.</italic>, which have not been included in SCI since 2018 and 2020, <italic>Brief. Bioinform.</italic> has the highest value of 11.622, followed by <italic>Drug Discov. Today</italic> (7.851), <italic>Bioinformatics</italic> (6.937), <italic>Cancers</italic> (6.639), <italic>Eur. J. Med. Chem</italic> (6.514), and <italic>Expert. Opin. Drug Discov.</italic> (6.098). The bubble chart shows that <italic>J. Biomol. Struct. Dyn.</italic> featured 64 publications in 2020, compared to a combined total of only four publications in the previous ten years; the Oncotarget journal inclusion in this category peaked in 2016–2017 ().</p></sec><sec id=\"s3-7\"><p>For scholars interested in the topic of drug repositioning, it is useful to know how other researchers are working on the issue to facilitate communication and collaboration between scholars. A total of 15,620 authors contributed to studies within our measurement consideration, and  shows the top 20 prolific authors by a number of publications. Of these 20 highly productive authors, seven were from the United States, three were from Argentina, and two were from Germany, indicating a relatively high concentration of drug repositioning research in certain countries. In addition, the NIH (United States), Case Western Reserve University (United States), Tech University Dresden (Germany), and the National University of La Plata (Argentina) each have two of these academics.</p><p>Cheng, FX leads the list with 25 research studies, followed by Talevi, A (23) and Mucke, HAM (22). For the list of corresponding authors, the top three remain, in order, Mucke, HAM (22), Cheng, FX (17), and Talevi, A (17). In terms of ACPP ranking, Butte, AJ was ranked first with 154.33 points, followed by Cheng, FX (100.56), Tang, Y (95), and Dudley, JT (82). Cheng, FX still has the highest h-index at 21, followed by Zheng, W (17), Talevi, A (12), Xu, R (11), and Schroeder, M (11). The h-index has two drawbacks when researchers of the same topic are compared with each other (). One is that the scholar’s h-index does not decrease over time but only grows or stays the same, and it is not possible to obtain information on whether the scholar is still in an academic career. In this study, we narrow the study to the most recent publications from 2010 to 2020, taking into account the timeliness of the h-index response information. Second, older scholars usually enter academia earlier and have an advantage in their h-indexes in comparison with those of younger scholars. Therefore, this phenomenon must be targeted for analysis or illustration. Thus, by combining the authors’ educational experiences and employment relationship changes that were recorded in the WOS database and ORCID business cards, we inferred that more than half of the scholars in the top 20 in terms of the number of publications received their Ph.D. before 2008, and two scholars, Mucke, HAM and Zheng, Wei, are older. In contrast, Cheng, FX, a scholar from Case Western Reserve Univ, completed his Ph.D. without a gap in 2013 and may have a longer academic career in the future; therefore, Cheng, FX’s h-index in the field of drug repositioning is likely to grow more in the future and Cheng, FX is likely to have more academic influence.</p></sec><sec id=\"s3-8\"><p>To reveal the focus of drug repositioning research and research trends, the author keywords and the highly cited and hot research topics of the ESI for each of the 2978 publications were analyzed, which were also derived from the core database of the WOS database (SCI-E/SSCI) (). Highly cited studies were defined as studies in the top 1% of the citations for all studies in the same ESI discipline within the 10-year range of inclusion of ESI inclusion (). A hot research topic of the ESI refers to a study published in two years with a citation frequency within one of the corresponding disciplines in the world in the last two months ().</p><sec id=\"s3-8-1\"><p>Author keywords tend to provide more information and have thus become a widespread focus (; ). The data of 6,083 author keywords in the search results were merged to make keywords with the same meaning represented by a single unified word. In the end, 5,616 author keywords were obtained. It should be specified that some publications without author keywords were excluded from the statistical analysis. Of these author keywords, 4,296 were used only once, representing 76.50% of the total. A total of 1,216 (21.65%) appeared 2–10 times, 79 (1.41%) appeared 10–20 times, 37 (0.66%) appeared 21–50 times, and the remaining eight (0.14%) were used between 51 and 1,500 times. All keywords cumulatively appear a total of 12,400 times, while the top 30 most used author keywords appear 2,967 times alone, or approximately 23.93%, as shown in . The comparison of keywords in recent years allows for tracking the frontiers of research and predicting hotspots and trends in drug repositioning research. The bubble plots show the three dimensions of the data, namely, the year of publication, the author’s keywords, and the number of corresponding publications. The horizontal change in the size of the bubble illustrates the increasing trend of author keywords over time, the vertical size of the bubble shows the most popular keywords in that year, and the numbers in the bubble indicate the frequency of author keywords and the number of publications.</p><p>The top 30 keywords include five diseases: “COVID-19/SARS-CoV-2/Coronavirus” (239), “Cancer” (75), “Alzheimer’s disease” (48), “Breast cancer (36)\", and “Chagas disease” (27). Drug names appear four times, “Antiviral drugs” (56), “Metformin” (35), “Statins” (28), and “Antibiotics” (25), which reveal the diseases and applications to which drugs were often repositioned during these 11 years. There were four subject categories, “Bioinformatics” (43), “Polypharmacology” (42), “Systems biology” (42), and “Precision medicine” (28) and eight occurrences of research methods, namely, “Virtual screening” (81), “Molecular docking” (64), “Machine learning” (54, eighth), “Clinical trials” (36), “High-throughput screening” (35), “Connectivity map” (28), and “Molecular dynamics simulation” (26).</p><p>In the context of the pandemic in 2020, there was a surge in research on the subject, with “COVID-19/SARS-CoV-2/Coronavirus” topping the list of keywords as soon as they appeared that year. “Virtual screening” is a research method that appeared seven times more frequently in 2020 than in the previous year. Since “Drug repurposing/repositioning” is a subject matter and a strategy for drug discovery/drug development, it would not make much sense to analyze these two keywords. Molecular docking is one of the core steps of virtual screening, and the COVID-19 pandemic generated many opportunities for the practice of drug repositioning. Therefore, high-quality studies of the keywords “COVID-19/SARS-CoV-2/Coronavirus”, “Virtual screening”, and “Molecular docking” were surveyed, as shown in the bubble chart, in the past two years, reflecting the relevant research trend in recent years. Wang, F et al. developed a new free reverse docking server based on a consensus algorithm (combining several docking algorithm strategies) to address the original shortcomings of computational molecular docking in drug repositioning, such as a low success rate, cumbersome operational steps, and reliance on code writing (). M Lapillo et al. extensively evaluated the performance assessment of docking-based target fishing methods and developed a consensus docking-based target fishing tactic (). In a virtual screening process, Gervasoni, S. conducted a literature search for molecular binding sites for SARS-CoV-2-associated protein targets while combining pocket and docking searches to propose a new pocket mapping strategy that identifies binding cavities with significantly better performance than pocket detection alone (). Xie, L et al. screened antitoxic drugs based on the multitarget structure of the pathway center and stated that this inhibition of multiple targets in one pathway would be more effective than targeting a single protein, and the chance of drug resistance was smaller, which could be applied to other pathways (). Li, Z et al. reported a virtual screening method based on accelerated free energy perturbation absolute binding free energy (FEP-ABFE) prediction and stated that the virtual screening method based on the prediction of FEP-ABFE will play a role in many other drug repositioning studies (). After a series of drug repurposing computational screens and various validation activities by several scientists, it was agreed that raltegravir (; ), clonidine (; ), chloroquine and hydroxychloroquine () have therapeutic effects in the treatment of novel coronavirus.</p><p>In addition, from the studies on the keyword “Machine learning” over the 10-year period shown in the bubble chart, it was found that the classical machine learning algorithms of support vector machines (; ; ), regularized least squares (; ), logistic regression (; ; ), and random forests (; ) have been widely used in inferring drug–target and drug–disease interactions.</p></sec><sec id=\"s3-8-2\"><p>While the level of influence of a study is reflected by a combination of many aspects, the number of citations remains an important indicator (). Based on the definition of highly cited and hot ESI papers in  of this study, a total of 108 highly cited studies were obtained, of which 11 were hot research topics. Hot research topics are shown in . It should be noted that the first-ranked author is used here as a representative, and the corresponding institution is shown. This rule is followed in  of this study. All hot research topics were published in 2020, and with the exception of an article describing the damage caused by nonsteroidal anti-inflammatory drugs (NSAIDs) to multiple organs and new information on drug repurposing (), the remaining studies focused on drug repositioning therapeutic target studies in novel coronavirus pneumonia (; ), screening drug studies (; ; ; ; ), reviews of clinical trials (; ), and reports of other coronavirus therapeutic agents and vaccine studies (). From the perspective of cooperation, most of them were completed by a country’s independent agency. In terms of the countries and regions studied, four studies involved US scholars, five studies involved Asian scholars, and one contribution was from an African scholar. In addition, “A SARS-CoV-2 protein interaction map reveals targets for drug repurposing ()”, published in <italic>Nature</italic> by Gordon, DE with a total of 125 scholars from the United States, the United Kingdom, and France was the most cited publication with 952 citations.</p></sec><sec id=\"s3-8-3\"><p>Eleven hot research topics were removed from the 108 highly cited ESI studies, and the top 20 most cited studies were selected from the remaining highly cited studies for analysis. In terms of year of publication, the study by Dudley, JT et al. published in <italic>NUCLEIC ACIDS RESEARCH</italic> in February 2011 was the earliest of these studies (). Five highly cited studies were published in 2013, and three studies were published as recently as 2020. Two studies were published in <italic>Nature</italic>, and one each was published in <italic>Nat. Rev. Drug Discov.</italic> and <italic>Nat. Med. subj. of Nature E</italic>; <italic>J. Med. Chem. L</italic> was next with two studies. There were 12 studies with the first author or coauthor from the United States, representing more than half of those in , followed by China (4), Canada (2), England (2), Germany (2), Japan (2), and Switzerland (2) in order of contribution of two or more studies. Nine studies were based on collaborations between different institutions in multiple countries. One of them, entitled “Alcohol-abuse drug disulfiram targets cancer via p97 segregase adapter NPL4”, published in <italic>Nature</italic> in 2017 by Skrott, Z et al. is a collaboration between scholars from six countries: Czech Republic, the United States, Denmark, Sweden, Switzerland, and China (). In TC, “DrugBank 5.0: a major update to the DrugBank database for 2018” () by Canadian University of Alberta scientists Wishart, DS et al. ranked first (1820 total citations). The most cited publication on an annual basis was “Network-based drug repurposing for novel coronavirus 2019-nCoV/SARS-CoV-2”, published in 2020, which was authored by Zhou, YD et al. and was the highest annual average cited publication with 609 citations (). The scientists Cheng, FX and Dudley, JT, contributed to two of these 20 publications and are important influencers in the field.</p><p>The three studies published in 2020 focus on novel coronavirus-related drug rediscovery activities () and the use of deep learning techniques (; ).  and Pushpakom, S et al. (2019) provided systematic reviews of the methods and challenges of drug repositioning at that time (; ). Initially, Sirota, M et al. (2011) explored the role of integrating genome-wide computational approaches for predicting reusable drugs (), while from 2013 onward, Peters, JU et al., Medina-Franco et al., JL et al., Reddy, AS et al., and Anighoro, A et al. generally recognized the importance of combining multiple points of pharmacological knowledge for drug repositioning studies (; ; ; ). In the face of a worldwide health emergency caused by the Zika virus epidemic,  used drug repositioning to identify lead compounds for drug development (). Of course, techniques related to the mining of repositionable drugs through experimental high-throughput screening, a traditional experimental approach, are not without progress; for example,  () explored the challenges of transferring 3D cell culture technology to the use of high-throughput screening (HTS) ().</p></sec></sec></sec><sec id=\"s4\"><p>In 1995, Mchugh et al. investigated the immunomodulatory action mechanism of thalidomide in humans, which was the first relevant publication on drug repositioning (). The publication time can be divided into three phases: the growth period of 1995–2009, the steady growth period of 2010–2018, and the rapid rise from 2019 and beyond. The 2978 publications studied between 2010 and 2020 were completed by 15,338 authors from 3,530 research institutions in 89 countries, and at the time of this study’s completion, the WOS database had surpassed more than 1,400 publications in 2021 under the same search restrictions for the topic, with more than 31,000 citations for the year, supporting further evidence that the topic is still gaining momentum worldwide.</p><p>The publication countries/regions are divided into three types: first, countries with a traditionally developed medical level, mainly developed countries in Western Europe, North America, and Oceania; second, countries with a developed pharmaceutical manufacturing industry, such as India and Japan in Asia; and third, developing countries with some research potential, such as China, Brazil, Argentina, and Mexico. In terms of national cooperation, Western European countries have shown a high degree of cooperation, with the United States, China, and the United Kingdom cooperating more frequently. This may be because Western European countries have a tradition of cooperation in the field of research, and the United States, China, and the United Kingdom are the most powerful countries in terms of drug repositioning publications and therefore cooperate more with each other. The United States accounts for half of the 20 most productive institutions, which may explain why the United States still publishes more than 50% of its studies independently, despite having the largest international collaborative network base, because it already has the most active and high-quality producing institutions within the country for research institutions seeking collaboration. Furthermore, 19 of these 20 institutions are universities and research institutes, and one is a company, HM Pharma Consultancy, which was established in 2000 to focus on drug repositioning for the development of new drugs (). This evidence suggests that the topic of drug repositioning is not only widely studied in academia but also has a place in the industry.</p><p>The 2978 studies are spread across 74 research areas, but pharmacology and pharmacy and biochemistry and molecular biology account for a larger proportion of the total number of studies. It is quite notable that the majority of studies reported in biophysics did not rise significantly until 2020. The reasons for this may be the following: first, there was a breakthrough in basic research in this field in 2020 and second, due to the novel coronavirus, research in this direction has increased its application for the prevention and control of the pandemic.</p><p>In terms of journals, <italic>Sci Rep</italic> ranked first, followed by <italic>PLoS One</italic> and <italic>J. Biomol. Struct. Dyn</italic>. In terms of lead authors, three have the most productive and influential positions: Cheng, FX is the most prolific author, based on the number of papers and h-index; Mucke, HAM is the most frequent corresponding author; and Butte, AJ is the top author in terms of ACPP ranking. Even though Latin American countries do not have an advantage in terms of national cooperation or the total number of institutional funding units, Latin American scholars have overcome many obstacles and are actively at the forefront of scholarship, contributing significantly to the field.</p><p>Through the analysis of the authors’ keywords, cancer has been the main disease addressed by this method. Metformin has been found by many scientists to have a good inhibitory effect on various tumors, mainly in gynecology (; ; ; ), and it has become a specific drug that has been most frequently mentioned in drug reuse in recent years. In terms of “antiviral drugs”, scholars not only use drug repositioning to find antiviral drugs to treat diseases, such as Ebola (; ) and HIV (), that have plagued humans for a long time but also use this method to seek treatments for infectious diseases, such as Zika virus (; ) and novel coronaviruses that have threatened several countries and even the world. For these diseases, emergency research on drug repositioning has played an important role in reducing mortality, calming patient fears, and restoring economic production when no specific drugs or vaccines were initially available during the pandemic. The combination of precision medicine and drug repositioning studies, often used to seek treatments for rare diseases () and, in particular, genetically related diseases (), is expected to be fully developed in the future. In the past 2 years, “Virtual screening”, together with “Molecular docking” and “Machine learning”, has become the most cutting-edge and important research methods in related technology fields, constantly improving the accuracy of drug reuse and screening. Currently, to develop more efficient and accurate research, there are two trends in the use of drug repositioning. One is the combination of various methods, such as the use of text mining and network analysis, and the creation of statistical models for predicting semantic link association to assess the relationship between pharmacological target pairings (); text analysis combined with machine learning () to develop drugs for Parkinson’s disease; prediction of new DTIs using data from multiple databases (); and the obtained relocated anticancer drugs were verified by cross-validation, literature, and experimental verification (). Second, the most advanced algorithms are applied and improved, such as matrix decomposition (; ; ; ; ) and matrix completion (; ) and deep learning (; ; ; ; ; ).</p><p>In fact, some of the studies in the list of highly cited research topics on novel coronaviruses drug repurposing studies are currently approaching 3,000 citations on Google Scholar (). The percentage of highly cited studies and hot research topics related to novel coronaviruses is also a good indication that the method has made an indelible contribution to the study of novel coronaviruses and similar infectious diseases. Auxiliary technology for the experimental screening of traditional drugs is also developing (), which also promotes drug repositioning or other drug development processes. Furthermore, the high-quality results of Elfiky, AA, a scientist from Cairo University, Egypt (), suggest that relevant research in some economically underdeveloped countries may reach top levels worldwide due to the return or affiliation of some prominent scientists.</p></sec><sec id=\"s5\"><p>For this research, the literature on drug repositioning research published in the SCI-E and SSCI sections of WOS core journals from 2010 to 2020 was analyzed based on bibliometrics and DDA software. This area has been of interest to scientists since the end of the 20th century and entered a period of rapid growth in 2019, with the peak far from being reached. Using bibliometrics as a tool, the United States has become a world leader in terms of the number of submissions, number of high-quality studies, funding support, strength of research institutions, and number of top scholars, followed by China and the United Kingdom, where more research is being performed in this area. As a method of drug discovery, drug repurposing is closely related to the development of various biomedical disciplines, and computer-related disciplinary methods, such as mathematical computational biology and computer science, have taken an important place in the research of this field in the last decade. The authors’ keyword analysis suggests that research in the field of the novel coronavirus will remain valuable until the associated pandemic is completely contained. Virtual screening, molecular docking, machine learning, and other related technical fields still need long-term development to achieve efficient and accurate repositioning of drugs (). Precision medicine, combined with drug repositioning, is the most promising direction for the future. In conclusion, drug repositioning can help to treat more diseases, such as drug resistance, poor drug selectivity, and limited therapeutic options.</p><p>This study may help some scholars with an initial interest in drug repositioning-related research to gain a concise and rapid understanding of the current state of global research, as well as offer some relevant information to institutions or groups seeking collaboration.</p></sec><sec id=\"s5-1\"><p>It is worth noting that this study has some biases and limitations. First, there are still some issues with the publications included in the study based on subject terms: 1) some relevant publications that do not use the search formula in this study may have been excluded from this study and 2) there may also be a small number of articles whose use of some of the aforementioned search terms deviates significantly from the general understanding; yet, such publications are included in this study. Second, some extraneous factors distort the credibility of the bibliometric statistics. 1) When analyzing the keywords of publications, some publications are excluded from the statistical analysis because they do not list author keywords (e.g., ()). 2) Excessive self-citation by some authors () inflates the actual level of interest in the publication. 3) for a publication, when an author submits more than one institution’s address information, this publication is counted as research results by each institution. Finally, in future work, patents from the WOS database associated with the topic of drug repositioning will be analyzed to provide another perspective on the situation of the topic in terms of applications and technological innovations.</p></sec></body>],\n",
              " [<body id=\"cbdd14136-body-0001\"><sec id=\"cbdd14136-sec-0001\"><p>Tight and selective interaction between ligands and target proteins is of cardinal importance in drug discovery that various stages of identification and validation of targets and potential drug leads, as well as manifold preclinical and clinical trials, should be carried out until obtaining the final approval from the Food and Drug Administration (FDA) (Abel et al., ; Campillos et al., ; Olayan et al., ). Accurate prediction of drug–target interactions (DTIs) plays a vital role for in silico and modern drug discovery in developing new drug candidates (Olayan et al., ). Unfortunately, despite the promising advancements in genomics, proteomics, and systems biology, formidable scientific and regulatory obstacles such as high attrition rates, excessively time‐consuming, and high‐priced procedures have taken a heavy toll on proposing effective biologically active agents, and the success of drugs to pass clinical trials stage is about only 13% (Fotis et al., ; Liu et al., ; Murray &amp; Rees, ). Such attempts enable scientists to design energy harvesting materials and systems for the sake of higher efficiency and enlarging practical window (Massetti et al., ; Safaei et al., ).</p><p>Accordingly, computer‐aided drug discovery (CADD) has revolutionized the history of drug discovery, especially regarding its substantial advantages (Schneider &amp; Fechner, ). These favors are (a) giving more information about the beneficial interaction between the drug and target and the availability of three‐dimensional (3D) structure, (b) providing a cost‐efficient way to reduce failures for high‐throughput screening (HTS), (c) producing new ideas for rationally drug design, and (d) leading to rationally anticipate the targeted protein and candidate hits (Bajorath, ; Bleicher et al., ; Ou‐Yang et al., ; Sliwoski et al., ). These strategies are currently classified into three major categories: ligand‐based, structure‐based, and chemogenomic approaches (Imam &amp; Gilani, ), as summarized in Figure .</p><p>Most importantly, recent studies have illuminated the potential roles of CADD in designing new drug candidates for COVID‐19 treatment. Coronaviruses (CoVs) are cross‐species viruses and can spread expeditiously into new host species, which in turn cause epidemic diseases. The whole members of the CoV family have a similar genome sequence, and the severe acute respiratory syndrome CoV (SARS‐CoV) and Middle East respiratory syndrome CoV (MERS‐CoV) are considered the two prototypical well‐known CoVs (Aghamirza Moghim Aliabadi et al., ; Seidi et al., ; Yin &amp; Wunderink, ). The new CoV, identified as COVID‐19 and caused by SARS‐CoV‐2, has a very similar genome sequence to SARS‐CoV (C. Wang et al., ). The lack of effective drugs led to considerable unwholesomeness and death rate of SARS‐CoV‐2, especially in 2020 and 2021, when a novel coronavirus strain brought about a fatal respiratory illness. Since the outgrowth of the SARS‐CoV‐2, more than 528 million people have been infected with up to 6.3 million deaths, as reported by the WHO ().</p><p>Although effective vaccines have amazingly reduced the COVID‐19 cases recently, they will not change the world right away, and importantly, their permanent efficacy, short‐term and long‐term side effects, and long‐term protection against different variants of SARS‐CoV‐2 are still under debate. These issues are accompanied by mass vaccination campaigns (Wen et al., ). Furthermore, several studies have reported that vaccinated people can also be infected with novel coronavirus variants (Largent &amp; Miller, ). Therefore, the design and introduction of new drugs, either the current drugs or synthesized agents, for eradicating the SARS‐CoV‐2 is of extreme concern. To accelerate this process, repositioning broad‐spectrum antiviral drugs can be a potential strategy concerning the presence of the pharmacokinetic, pharmacodynamic, and toxicity data of these drugs (Serafin et al., ).</p><p>For instance, the efficacy of some antiviral drugs with previously known treatment of hepatitis, HIV, influenza, and Ebola viral diseases has been investigated. These agents, including adefovir, foscarnet, tenofovir, sofosbuvir, and uprifosbuvir, have phosphoramide and phosphate with various azole rings in their structures. By assessing their effectiveness against COVID‐19, remdesivir was the one that grabbed much attention and is believed to be the most successful drug for the therapy for combating COVID‐19 (Brown et al., ; Gholivand et al., ). Further potential macromolecular targets of SARS‐CoV‐2 are depicted in Figure . Therefore, finding the drugs that can target this enzyme may decrease the possibility of mutation‐mediated drug resistance and show significant antiviral activity (Pachetti et al., ).</p><p>Thus, it is likely that CADD might be able to provide us with effective small molecules targeting the main receptors of SARS‐CoV‐2 with fully therapeutical and preventional‐oriented outlooks (Muratov et al., ). The more recent update of FDA approval for COVID‐19 drug discovery shows that authorization of remdesivir (inhibits the RNA‐dependent RNA polymerase (RdRp) of the virus), molnupiravir (inhibition of viral reproduction through enhancing the vast mutations in the replication of viral RNA) and paxlovid (3c‐like protease inhibitor) for the treatment of mild‐to‐moderate COVID‐19 (). These drugs are used for various viral diseases such as hepatitis C, Ebola virus, influenza, etc. (Vangeel et al., ). Molnupiravir and remdesivir acted on the RdRp enzyme employed by the virus to increase the transcription and replication capability of the viral RNA genome. While some groups reported the promising results of remdesivir, others showed a lack of efficacy, and therefore, the WHO did not suggest using it (Mahase, ).</p><p>Our focus in this paper is to explore the role of CADD on COVID‐19; most specifically, chemogenomics and DR, with potential examples of COVID‐19 drug discovery. Unfortunately, with boundless energy and time to discover a successful small molecule by utilizing various databases to inhibit the main protease (Mpro) of SARS‐CoV‐2 via several virtual screening (VS) approaches, there are no effective drugs that have been approved. Only safety protocols such as wearing face masks, face shields, social distancing, and washing hands frequently are the best approaches to minimize the illness's danger and prevent transmission.</p><p>Basically, the process of bringing an FDA‐approved medicine to market for COVID‐19 drug discovery is extremely challenging, starting from the initial research of scientists for target identification and lead compound generation. After lead discovery and development of drug‐like candidates and investigation of in vitro and in vivo results, the process will go into three phases, i.e., preclinical and, if successful, into clinical studies, and ultimately a marketed medicine after FDA approval (Schaduangrat et al., ). It is where CADD is evolved as a valuable and powerful method to aid the challenging tasks of identifying small bioactive molecules and predicting targets.</p></sec><sec id=\"cbdd14136-sec-0002\"><p>In 1981, an influential article appeared in “Fortune magazine” titled “Next Industrial Revolution: Designing Drugs by Computer at Merck”. This article led to a realization of the importance of in silico studies in drug discovery and manifestation of CADD (Sliwoski et al., ). Since then, high‐throughput screening (HTS) has been increasingly used in the pharmaceutical sphere and academic institutions to the rapid discovery of hit and lead compounds. This potential and automated approach enables us to screen the entire compound library of either number of chemicals or biological compounds in a target‐based assay or even a cell‐based assay to obtain the desired biological response (Bittker &amp; Ross, ; Maghsoudi et al., ). It can spring to the mind that the stringent requirement of large resources of targets and ligands in HTS makes it expensive and time‐consuming, and we should overlook the use of the useless ligands having no probability of showing success. But, the CADD technologies are very powerful and able to address this issue easily by employing virtual high‐throughput screening (vHTS) that uses virtual compound libraries and allows experimentalists to focus on ligands which are more likely to have any activity of interest (Leelananda &amp; Lindert, ). In the first days of COVID‐19 outbreak, vHTS was employed and developed efficiently to investigate a group of approved drugs to identify the possible anti‐SARS‐CoV‐2 agents at the National Center for Advancing Translational Sciences (NCATS) of the National Institutes of Health (NIH), embracing a broad spectrum of the SARS‐CoV‐2 life cycle, such as viral entry, viral replication, in vitro infectivity, and live virus infectivity (Xu et al., ). Many possible drug candidates were selected for targeting various stages of the SARS‐CoV‐2 life cycle. For instance, cepharanthine, an anti‐inflammatory compound, effectively showed ability to rescue the cytopathic effects (CPEs) of SARS‐CoV‐2 (Rogosnitzky et al., ), or corilagin, a possible antifibrosis, revealed potential against the angiotensin‐converting enzyme (ACE2) receptor‐binding domain (RBD) (Yang et al., ). Therefore, it is urgent to build a high bio‐safety SARS‐CoV‐2 infection model with a high ability for HTS of drugs against SARS‐CoV‐2.</p><sec id=\"cbdd14136-sec-0003\"><p>This strategy just relies on ligand information—one of the broadly known and widely used ligand‐based approaches in QSAR modeling. The advantages accruing from the QSAR methods in drug discovery and development are noteworthy. This computational technique attempts to construct mathematical models of compounds' physical and chemical properties associated with their chemical structure. QSAR studies are extensively employed to predict pharmacokinetic properties such as ADME and toxicity by building a training set of measured properties of known compounds and making a mathematical model of the encoded chemical structure, and also is useful in sensory applications as well (Alam &amp; Khan, ; Farshchi et al., ; Kholafazad‐Kordasht et al., ; Nosrati et al., ; Seidi et al., ).</p><p>Generating molecular descriptors from compound structures is also significant in assessing compounds' different physical, chemical, and topological features. After computing molecular descriptors, machine learning techniques play indispensable roles in the quality of QSAR predictions (Popelier &amp; Smith, ; Rad &amp; Maghsoudi, ; Sliwoski et al., ). Using machine learning and QSAR, Muratov et al. identified 32 drugs and their 73 selected binary combinations for testing against SARS‐CoV‐2. They found that the combinations of four of these drugs, including nitazoxanide, remdesivir, amodiaquine, and umifenovir, could synergize against SARS‐CoV‐2. Interestingly, the combination of nitazoxanide and remdesivir showed a great synergistic interaction, and the combination of remdesivir and hydroxychloroquine exhibited a powerful antagonist role against SARS‐CoV‐2 (Bobrowski et al., ).</p><p>The two proteins pp1a and pp1ab are quickly translated after entry into the host cells and further separated by two viral proteases, 3 C‐like proteases (3CLpro) and papain‐like protease (PL<sup>pro</sup>) (Amin et al., ). The latter is a critical virus enzyme responsible for producing a functional replicase complex and stimulating viral spread (Shin et al., ). Amin et al. used chemical‐informatics approaches to build a Monte Carlo optimization based on (a) QSAR data of various PL<sup>pro</sup> inhibitors, (b) VS to choose the best in‐house molecules among 67 with the high likelihood of targeting against PL<sup>pro</sup>, and (c) eventually verification of these inhibitors by using receptor–ligand interaction investigation. Out of 67 compounds, 56 were anticipated as active from a QSAR model. Afterward, these 56 similar compounds were screened through SwissADME, and eventually, 13 compounds were approved for the ADME criteria. Furthermore, molecular docking has proved the potentiality of these candidates against SARS‐CoV‐2 PL<sup>pro</sup> (Figure ; Amin et al., ).</p><p>In an interesting work, Tropsha et al. developed open‐source data QSAR models of 42 inhibitors from DrugBank by employing QSAR models, docking, and similarity searching and utilized these models for virtual screening of all drugs in the DrugBank database. The 42 virtual hits were finalized for availability and cost using in‐house ZINC Express software. The National Center for Advancing Translational Sciences (NCATS) analyzed 11 of total 42 hits which were potential compounds, and three of them, cenicriviroc (AC50 of 8.9 μM), proglumetacin (tested twice independently, with AC50 of 8.9 μM and 12.5 μM), and sufugolix (AC50 12.6 μM). While molecular docking was unsuccessful to accurately distinguish between experimentally active and inactive compounds, QSAR was able to identify three inhibitors of SARS‐CoV‐2 replication through Mpro inhibition. Overall, these results show that QSAR models proposed using SARS‐CoV‐2 Mpro data can be useful to introduce compounds active against SARS‐CoV‐2 (Alves et al., ).</p></sec><sec id=\"cbdd14136-sec-0004\"><p>This strategy relies more on target information. The structure‐based drug design, also called target‐based drug design, rests on the assumption that the bioactivity of a ligand comes from the physical interaction of the ligand with the corresponding target protein. The prerequisite of this approach is the 3D structure of the biological target, obtained from crystal structures, NMR data, or molecular modeling. Molecular docking is an established structure‐based approach frequently used to predict the binding affinity and analyze the interactive mode between a ligand and a target/receptor. In other words, docking enables us to identify novel drug candidates by predicting ligand–target interactions at a molecular level (Pinzi &amp; Rastelli, ; Sargazi et al., ).</p><p>There are several references in which the significant role and applications of docking in drug design and development have been discussed (De Vivo &amp; Cavalli, ; Keretsu et al., ; Kitchen et al., ; Meng et al., ). Furthermore, in the current spread of SARS‐CoV‐2, molecular docking is important in screening potential drugs (Keretsu et al., ; Lokhande et al., ; Marinho et al., ; Yu et al., ). The recent work of Khelfaoui et al. () could be a good example of using molecular docking combined with MD to find promising candidate drugs against COVID‐19. They selected a library of 18 drugs that three of which include ramipril, delapril, and lisinopril, bound to angiotensin‐converting enzyme 2 (ACE2) receptor and [SARS‐CoV‐2/ACE2] complex well. ACE2 is one of the receptors in the human body with which the S proteins from coronaviruses have a great binding affinity and serve as the entry point into cells for SARS‐CoV‐2. In fact, due to the long time required to discover specific antiviral agents or a vaccine, the best strategy would be to repurpose existing drugs to treat COVID‐19, which will be discussed in the DR section.</p><p>Using VS and molecular docking, Abdellatiif et al. identified some promising antimicrobial and chemotherapeutic agents having coumarin and pyridine derivatives in their structures. In order to identify the most potential drug–target interactions, a molecular operating environment (MOE) program was carried out against the human ACE2, SARS‐CoV‐main peptidase, serine protease hepsin, human coronavirus papain‐like proteases, and SARS‐Coronavirus NSP12. As shown in Figure , among compounds a, b, c, d, and e with binding score energy values of −6.08, −6.48, −6.78, −7.08, and −6.63, respectively, compound d had the binding score energy against the aforementioned targets. Thus, compound d needs to be additionally evaluated in clinical trials as a candidate for potential inhibitor of the SARS‐CoV‐2 (Abdellatiif et al., ).</p><p>Hosseini et al. evaluated a virtual screening using docking of 1615 FDA‐approved drugs to identify novel potential inhibitors for protease protein of COVID‐19. They showed that among all investigated FDA‐approved drugs, simeprevir, a Hepatitis C virus (HCV) NS3/4A protease inhibitor, manifested high affinity to protease binding pocket and placed effectively into the binding pocket more beneficially than the lopinavir‐ritonavir. This study suggested this compound as a potential inhibitor of COVID‐19 (Hosseini &amp; Amanlou, ). In another docking study, Motiwale et al. worked to identify potential inhibitors against Mpro of SARS‐CoV‐2 from previously reported SARS‐3CL protease inhibitors where among 61 previously reported potential inhibitors, 4‐benzoic acid and 4‐(4‐methoxyphenyl)‐6‐oxo‐2‐[(2‐phenylethyl)sulfanyl]‐1,6‐dihydropyrimidine‐5‐carbonitrile revealed a minimum and maximum binding energy, respectively (Motiwale et al., ).</p><p>Table  presents a selection of recent studies that researched the docking strategy of different pharmacological classes as treatments of SARS‐Co and SARS‐CoV‐2. Although many studies are carried out and many drugs are being tested to open a new road for effective treatment of COVID‐19, there are no fully beneficial antiviral drugs against the disease. Many of these studies are now in clinical trials to assess their safety and efficacy against infected patients.</p></sec><sec id=\"cbdd14136-sec-0005\"><p>Chemical genetics, chemical biology, and chemogenomics (chemical genomics) are some cutting‐edge applied fields in target and drug discovery (Rognan, ; Urán Landaburu et al., ). Among the presented strategies, chemogenomics is an emerging discipline that principally investigates all molecules that can interact with any biological target. In humans, the number of the gene is about 26,000, whereas the amount of various proteins is approximately more than 1 million (Andersson et al., ). However, due to the extensive number of available molecules and biological targets, chemogenomics is limited to screening congeneric compound libraries against a selected target family. This alternative proposes an unprecedented chance to seek drug‐like compounds with a noteworthy target or significant subtype (Jones &amp; Bunnage, ; Kubinyi, ).</p><p>Chemogenomics is different from the traditional pharmaceutical methods due to linking the widespread software library directly to the methodology of modern genomics. In order to achieve this purpose, chemogenomics, along with a contribution of several chemical disciplines, including synthetic chemistry, combinatorial chemistry, cheminformatics, bioinformatics, and genomics, screen synthetic compound libraries for their impact on biological targets (Bredel &amp; Jacoby, ; Quinlan &amp; Brennan, ). The ultimate goal of chemogenomics is the rapid screening as well as the rational development of target‐specific chemical ligands to decrease the cost and enhance the drug discovery period, especially in the case of COVID‐19 drug investigation (Jones &amp; Bunnage, ).</p><p>Basically, the main role of chemogenomics is unifying the discovery of ligands and targets by taking advantage of active compounds as probes to identify proteome functions. Similar to genetics, and if the operation of investigation arises from the target to phenotype, or vice‐versa, there are two concepts: (a) forward and (b) reverse. In ‘forward chemogenomics’, the ligand is discovered through their phenotypic condition that could affect the entire biological system instead of the base of their mechanism of suppression of a certain desired protein. In other words, the molecular basis of the phenotype interest is unknown and phenotypic screening is carried out in cells of organisms having a single‐cell or multicellular utilizing a group of ligands. (Figure ).</p><p>On the other hand, in “reverse chemogenomics,” the desired sequence of genes is cloned and explicit at the first step as target proteins; afterward, they are preselected in a high‐throughput screening (HTS) ‘target‐based’ way by a panel of compounds. Here, the assays are normally classified as (i) organismal assays, (i) cell‐free, and (ii) cell‐based organismal assays.</p><p>Cell‐free assays are generally straightforward but precise and extremely automated. They are comprehensive binding trials in which manifold compounds are examined concurrently for their binding affinity toward a variety of particular goals. Here, DTIs are clearly recognized in the absence of confounding variables. However, in cell‐based and organismal assays, chosen compounds are delivered right to cells or organisms in vitro to select hits within a relevant cellular context. However, the hits need further mechanistic characterization due to simultaneously interacting with multiple targets.</p><p>The former approach is usually employed in reverse chemogenomics, whether target information is at hand, while organismal assays and cell‐based are mainly employed in forwarding chemogenomics to test a wide variety of compound effects on whole biological systems (Figure ) (Bredel &amp; Jacoby, ).\n",
              "  <list id=\"cbdd14136-list-0001\" list-type=\"alpha-lower\"><list-item id=\"cbdd14136-li-0001\"><p>In forward chemogenomics, cells or organisms are distributed in multi‐well cell culture plates for incubation. Subsequently to the incubation step, an aliquot from donor plates is taken to the recipient and mock plates (mock treatment), in which the ligand–target binding assay is performed. The endpoint of most cell‐based HTS assays is a spectrophotometric measurement in a spectrophotometer equipped with suitable software that calculates the final data, including SAR analysis and in silico QC. Then, active compounds that obtain the intended changes in the phenotype are chosen to elucidate their targets at the molecular level. This strategy could be performed using affinity chromatography, microarray, phage display, or transcriptional or proteomic profiling.</p></list-item><list-item id=\"cbdd14136-li-0002\"><p>In reverse chemogenomics, the main focus is to elucidate the protein families and the gene. In this regard, the sequencing of the target gene demonstrates a specific level of homology that is transfected in a host cell. These family targets are sequentially gathered, purified, and subjected to trial design. Finally, similar to forwarding chemogenomics, the measurement data on the spectrophotometer are directed to a suitable data analysis system.</p></list-item></list>\n",
              "  </p><sec id=\"cbdd14136-sec-0006\"><p>The biological target is extracted from the existent information in the biologically annotated chemical database in the target fishing approach (Huang et al., ). However this procedure needs an abundant effort from a time, cost, and resource perspective to identify all plausible chemical target associations; thus, computational target prediction techniques are strongly being pursued with the contribution of advanced data‐mining algorithms (Anwar et al., ). This current mode of action or the computational target approach is classified into four classes: data‐mining, pharmacophore searching (analogy searching), analysis of bioactivity spectra, and molecular docking. In the past, these methods were applied to investigate new compounds of known targets, but recently, they have been applied to predict novel targets for the known compounds <italic toggle=\"yes\">via</italic> similar approaches (Campbell &amp; Marchant, ; Ekins et al., ).</p><p>Target‐based (or receptor‐based chemogenomics) refers to the discovered biologically active compounds with phenotypically readout information. Nevertheless, no existing target data; thus, by applying the chemogenomics strategy, the hypotheses can be generated of the target by comparing the new biological compounds to the same known target (Fourches et al., ; Weston et al., ). For example, the target receptors or proteins that are necessary for replication and survival of SARS‐CoV‐2 are viral proteases involving papain‐like protease (PL<sup>pro</sup>), the 3‐chymotrypsin‐like protease (3CL<sup>pro</sup>), and M<sup>pro</sup>, RNA‐dependent RNA polymerase (RdRp) complex, the spike (S) receptor binding glycoprotein, the nucleocapsid (N) protein, nonstructural protein (NSP), and the membrane (M) protein including transmembrane (TM) as well as small envelope E protein (Cavasotto et al., ).</p><p>As an example of this part, it can be referred to the work of Mishra and his coworkers that reported an inclusive computational approach for the identification of drug molecules that are multi‐targeted struggling with the SARS‐CoV‐2 proteins. These drugs are essentially included in virus replication inside the host body, the viral–host interaction, disease development, and transmission of Sars‐CoV‐2 infection. They screened 72 potential antiviral drugs approved by the FDA against the S glycoprotein, hACE2, 3CL<sup>pro</sup>, Cathepsin L, N protein, RdRp, and NSP6 proteins.</p><p>Furthermore, they investigated the molecular interaction by applying free energy landscape, simulation of Molecular Dynamics, and binding free energy estimation using MM‐PBSA. Their finding showed favorable results for seven drugs, among which catechin (flavan‐3‐ol) can effectively bind to 3CLpro, Cathepsin L, RBD of S protein, NSP‐6, and Nucleocapsid protein to be used as a potential multi‐targeted agent in combating SARS‐CoV‐2. However, due to the limited timeframe of combating this pandemic, it is not realistic to be assured of finding a definitive therapy for combating the pandemic. Hence, vigorous attempts are still continuing for the DR as a clinical treatment strategy to fight COVID‐19 (Mishra et al., ). In another research, Omotuyi et al. investigated molecular docking protocols for fast screening of the FDA database for high‐affinity interaction of the resolved 3D structure of SARS‐CoV‐2 receptor‐binding domain (RBD) in complex with its receptor hACE‐2 interface.</p><p>Ubrogepant, as one of the preferable candidate drugs, was studied utilizing the atomistic molecular dynamics simulation method. They reported that UBR breaks the interaction at the RBD/ACE‐2 interface residues of SARS‐CoV‐2 RBD/ACE‐2 complex that might result in protein function losses with direct inference on the oligomerization in RBD and function loss in ACE‐2. Therefore, leading to a binding and cellular receptor recognition unattainable. Therefore, UBR exhibits a novel therapeutic candidate in combating SARS‐CoV‐2 since it binds to proportionally high affinity with free RBD, ACE‐2 receptor, and SARS‐CoV‐2 RBD/ACE‐2 complex based on only binding affinity calculations (Omotuyi et al., ). The other studies in this area are demonstrated in Table .</p></sec><sec id=\"cbdd14136-sec-0007\"><p>A primary example of the beneficial application of chemogenomics is DR, which accelerates drug discovery by identifying novel clinical targets for the available approved drugs. This method possesses the distinctive merit of rapid clinical development without bearing the long‐lasting and expensive procedures included in drug discovery. Therefore, according to the advantages of these proposed drug discovery methods, many COVID research orientations have been focused on DR (Galindez et al., ; R. Liu et al., ).</p><p>Computational DR applies available drugs for new or existing diseases that do not have an absolute cure, while the available drugs were not originally designed for the novel disease treatment. Furthermore, utilizing the enormous volumes of available omics information in digital form in conjunction with in silico screening intrigued the researcher to use this appropriate ground to significantly increase the speed of shortlisting promising candidates in reply to spreading diseases like SARS‐CoV‐2. To be exemplified, Lucchetta, M. and M. Pellegrini reported a methodology for preclinical computational DR called DrugMerge, and it is based on combining several drug rankings gained via a set of Disease Active Subnetwork construction algorithms.</p><p>DrugMerge utilizes differential transcriptomic data from drug perturbation assays and cell lines/tissues of patients affected by the illnesses on the subject of a considerable gene co‐expression network. Application of DrugMerge to COVID‐19 shows that DrugMerge can mimic human expert judgment (Lucchetta &amp; Pellegrini, ). Rima Hajjo and Alexander Tropsha developed a systems biology workflow for drug and vaccine repurposing. They reported that the identification of Bacillus Calmette–Guérin (BCG) and small molecules of BGC mimics involving antiviral drugs (raltegravir, emetine, and lopinavir as high confidence hits) are capable of boosting immunity and preventing emerging new viruses. Their results approved that BGC impacted the generation and maturation of naïve T cells, resulting in increasing long‐lasting trained intrinsic immune replies to combat novel viruses, including SARS‐CoV‐2 in vitro as a protective measurement from the lethal consequences (Onozuka et al., ). Due to the lengthy development cycle of novel drugs, DR turns into an efficient approach to screening drugs of potential therapeutic use for SARS‐CoV‐2. Che et al., in another attempt, screened potential therapeutic drugs for COVID‐19 based on a knowledge‐graph‐based DR method. In their experiment, five drugs predicted by the models exhibited efficiency in clinical treatment.</p><p>The experimental outputs prove that the model can predict drug–disease interaction effectively for normal diseases and SARS‐CoV‐2 (Lv et al., ). Therefore, DR presents a constant source of novel knowledge in virus biology and molecules with previously undefined antiviral properties, which can be applied as molecular tools in unknown molecular mechanisms of virus replication and pathogenesis (Mercorelli et al., ). Further investigation of SARS‐CoV‐2 in related research is summarized in Table .</p></sec><sec id=\"cbdd14136-sec-0008\"><p>A fundamental role of chemogenomics is predicting the activity profile of the compounds for a group of targets that can be implemented by using ligand‐based or target‐based approaches. The ligand‐based philosophy is underlying that molecules sharing enough similarity to available biologically defined ligands have an increased probability of sharing the same biological profile. Therefore, the efforts are devoted to organizing chemical libraries with biological information, including in vitro affinity, targets, and toxicology (Harris &amp; Stevens, ). As a result, designing chemical libraries is of great interest for a certain individual target or target family, which is attempted <italic toggle=\"yes\">via</italic> utilizing machine‐learning methods.</p><p>Herein, concentration is on protein–ligand interaction via applying physics‐based protein‐ligand docking on machine learning. Computational chemogenomics or quantitative multiple structure–activity relationships (QMSPR) modeling consists of the compound–protein interaction space modeling of large datasets at once, usually for drug discoveries, where the available techniques chiefly incorporate escalating numbers of bioactive samples or concentrate on individual subfamilies of proteins and ligands (Cheng et al., ; Xue et al., ). However, the 3D information of protein targets is needed most of the time. Thus, the application of precise target data in chemogenomics is confined to the explicitly known X‐ray structures or structures that homology models can evaluate. As a result, many QMSAR is centered on ligand‐based studies or similarity principles (Gawriljuk et al., ).</p><p>In this context, compounds of phenotype‐based preselection possess merits in discovering target‐based drugs due to unclimbable and missing perceptions of the mechanism of action of the drugs. Therefore, a handful of methods can carry out new chemical compound screening which is phenotype‐based. In this regard, Thai‐Hoang Pham et al. extended a sound learning framework for a high‐throughput mechanism‐driven neural network‐based method for phenotype compound screening. They applied this framework for COVID‐19 DR. Their proposed novel data escalating technique extracts beneficial information from untrustful assays in the dataset of L1000. Furthermore, they showed the value of DeepCE by applying it to the DR of COVID‐19 and produced novel lead compounds that represented clinical evidence. Therefore, DeepCE can provide a potentially strong framework for rapid predictive modeling <italic toggle=\"yes\">via</italic> using noisy omics data and screening new chemicals to modulate an organized response to illnesses (Wang et al., ).</p><p>A computational approach for increasing the rate of development of drugs and comprehension of the mechanism of action of small molecules is the identification of drug–target interaction (DTIs). Lee and his coworkers represented an AI‐DTI, a new method for predicting activation and inhibition of DTIs, by combining the mol2vec and genetically perturbed transcriptomes. Their finding showed that this methodology was a success in discovering about half of the DTIs for drugs that are employed for treating COVID‐19 (Lee et al., ). A myriad of compounds represent potential drug candidates, but merely a few of them will be formulated, approved, and registered as a drug for individual health issues which investigation for that compound will be a long, intricate, and costly process, while sometimes there is an urgent need for developing a drug like COVID‐19 pandemic (Muratov et al., ). Table  demonstrates chemogenomics related to COVID‐19 research.</p><p>To conclude, chemogenomics is an emerging interdisciplinary field at the interface of chemistry, biology, and informatics which decreases the cost and time of drug discovery dramatically <italic toggle=\"yes\">via</italic> three main approaches, including target fishing, predicting the bio‐active profile of drugs, and DR. Herein, of much current importance is chemogenomics, is in COVID‐19 drug discoveries, which this review paper highlighted specifically.</p></sec></sec></sec><sec id=\"cbdd14136-sec-0009\"><p>Drug repositioning, also known as drug repurposing, drug redirecting, drug reprofiling, and therapeutic switching, is defined as identifying new indications and therapeutic targets for existing drug compounds (Lotfi Shahreza et al., ). DR has received significant attention caused of its crucial role in finding and developing new therapies for complex, orphan, and chronic diseases with unknown or less effective treatments (Shameer et al., ). Since developing a new drug in the traditional drug discovery process is time‐consuming and takes approximately between 10 to 15 years to enter the market, repurposing existing drug compounds is capable of overcoming these shortcomings, and this type of research has been growing drastically (Figure ) (Kumar et al., ). Therefore, DR plays a versatile role in developing new drugs because it can reduce the costs of the whole process and shorten the development pathways of approved or investigational drugs (Pushpakom et al., ). DR works with two main approaches. The first is that a drug can bind to various targets simultaneously, which is good for searching novel target sites of action for the known biological agents, and the second is that targets related to a disease are often associated with different biological processes of pathogenesis, which is important for choosing of a new indication for the known target (Badary, ; Kamel et al., ). These are indispensable in proposing new drug candidates for unprecedented diseases like COVID‐19.</p><p>It has emerged as an alternative way of accelerating the process of developing new treatments against COVID‐19 (Yu et al., ). Based on data brought from PubMed, the number of works of literature on DR has been growing dramatically, and this number has increased since the outbreak of SARS‐CoV‐2 (Galindez et al., ). It should be noted that polypharmacolgy and DR are inextricably interrelated with each other (Cheng, ). DR has become an influential tactic from both academia and pharmaceutical industries as an alternative strategy to traditional drug discovery that is related to when new therapeutic applications are pinpointed for existing drugs (Karakida et al., ; Turanli et al., ).</p><p>Compared to traditional de novo drug discovery and development, this approach provides a new and fast way for drug development and reduces risk, particularly for drugs that cannot pass phase I clinical trials (Figure ; Ashburn &amp; Thor, ). It is able to reprocess the information on known drugs such as pharmacokinetics and safety for humans and manufacturing pathways, resulting in reduced time and expenses in traditional de novo drug discovery (Sawada et al., ). There are a wide variety of examples of repositioned drugs. Sildenafil is considered the most prominent one, beginning with cardiovascular treatment followed by accidentally treating erectile dysfunction clinical trial and later in low dosages, treating pulmonary hypertension (Ghofrani et al., ; Prasad et al., ). Furthermore, as repurposed drugs pass the preliminary clinical trials (phase I or safety level), the time needed for drugs to reach the market can be reduced significantly. Hence, scientists can identify the unintended secondary targets through CADD.</p><p>The history of discovering the effect of an established drug on new indications dates back to World War II, since considerable improvement had been observed for cutaneous rashes and arthritis in soldiers who were under treatment with an antimalarial medicine, Chloroquine (Schlitzer, ). Chloroquine is a well‐known antimalarial drug that was synthesized in 1934 and later on demonstrated successful various therapeutic effects, including anti‐inflammatory/immunomodulating, anti‐infective, antithrombotic, and metabolic. Furthermore, antitumoral properties make it available for the treatment of a number of cancers (Plantone &amp; Koudriavtseva, ). In 1942, Marcel Jonbon used sulfonamide antibiotics for typhoid treatment, but he faced hypoglycemia as an unexpected side effect in some patients; therefore, sulfonamides presented the role of an anti‐diabetic drug (Quianzon &amp; Cheikh, ). Aspirin is one of the most famous DR examples with an original indication as an analgesic drug that for the first time repositioned as an antiplatelet aggregation drug in 1980 and a second indication still is commonly used for cardiovascular problems (Jourdan et al., ). Furthermore, several types of research have been conducted on the anticancer effect of aspirin such as colorectal cancer (Drew et al., ), lung, and breast cancer (Li, Hu, et al., ). The notable potential and extensive applications of DR in novel therapies are entirely apparent and accordingly, the number of acceptable research studies which have been done or are under processing are growing every day. There are various classifications for DR methods which typically can be categorized into computational approaches and experimental approaches (Park, ). Experimental approaches refer to testing drugs in evaluations based on available comprehensive clinical compound databases in which the possible indications of drugs can be discovered by target‐based or cell‐based analysis (Luo et al., ). There are some hurdles for experimental identifications because they are mostly expensive and time‐consuming, requiring a set of marketed drugs, technical equipment, and also screen tests. In contrast to experimental methods, computational drug repositioning methods which are mainly data‐driven utilizes databases and bioinformatics tools to find out interactions between drugs, targets, and diseases (Pillaiyar et al., ). Analyzing a large type of data (like chemical structures, gene expression, genotype or proteomic data, and electronic health records (EHRs)) provides the possibility of understanding novel and unexpected functional gene interactions and unclear drug response or mechanisms of disease (Pushpakom et al., ).</p><sec id=\"cbdd14136-sec-0010\"><p>Since there is no specific and registered therapy to cure the infection of this virus, finding effective therapeutics by applying existing drugs is a pressing exigency and established strategy (Galindez et al., ). Concerning the time‐consuming regulatory process of the newly approved drugs, the DR strategy strongly utilizes the identification of new treatment methods by presenting FDA‐approved drugs. In addition, many repurposed drugs with acceptable results are accomplished by computational methods (Wang &amp; Guan, ).</p><p>There are myriad reasons why we need to go through DR for different diseases like COVID‐19 if we intend to accelerate the process of developing a new drug. These reasons can be (i) availability of marketed drugs and their excipients and APIs for formulation and distribution, (ii) the likelihood of taking advantage of a synergistic effect of various drugs with each other, (iii) cheap and less time‐consuming process for drugs to be marketed and (IV) emergence of new therapeutic pathways for former drugs and new categories of medicine (Chung et al., ). Therefore, repositioned drugs have very important impacts on COVID‐19 drug discovery. Therefore, different categories of drugs are being repositioned in in vivo and in vitro studies to treat SARS‐CoV‐2 (Wu et al., ). These potential agents are namely Ebola virus (remdesivir), antimalarial (chloroquine), anthelmintic and antiprotozoal drugs (niclosamide), diarrhea (nitazoxanide), antiviral (lopinavir and ritonavir), antineoplastic (carfilzomib and bortezomib), and antibacterial (azithromycin) (Figure ).</p><p>In this regard, the WHO supported repositioning remdesivir, lopinavir/ritonavir, and chloroquine/hydroxychloroquine as a potential for COVID‐19, although most might be only supportive symptomatic (Won &amp; Lee, ). Nevertheless, they will likely be approved for COVID‐19 therapy if they pass clinical trials and are published in literature and guidelines from leading health organizations (Nain et al., ). In order to do this, many groups have been attempting enthusiastically to discover the first approved drug for COVID‐19 therapy (Erlanson, ; Low et al., ; Muratov et al., ). As a result, we identified several clinical trials (Table ), in which 18 studies were at clinical phases 2, 3, or 4 ().</p><p>Luo et al. investigated the free energy perturbation‐absolute binding free energy (FEP‐ABFE) method and bioassay validation to reposition 25 potential therapeutic agents to inhibit M<sup>pro</sup>. By using the viral proteinase crystal structure, M<sup>pro</sup> (PDB ID: 6 LU7), up to 2500 FDA‐approved drugs were first screened <italic toggle=\"yes\">via</italic> Glide's molecular docking program. They found that Cys145‐His41/Ser144‐His163 could be the nucleophilic agent and acid responsible for hydrolysis of the substrate proteins, and Gly143 and Gln166 can form a hydrogen bond with the “CO‐NH‐Cα‐CO‐NH‐Cα” structure of the backbone of the substrate protein. Of 25 possible agents, 15 showed strong inhibitory properties against M<sup>pro</sup>. Among these agents, additionally, dipyridamole passed its first‐round clinical trials with considerable potencies (Li, Hu, et al., ; Li, Li, et al., ).</p><p>The network‐based methods have also given satisfying results in investigating the DTIs and CoV–host interactions. Zho et al. introduced 16 possible anti‐HCoV drugs such as melatonin, mercaptopurine, toremifene, mesalazine, and sirolimus from more than 2000 FDA‐approved drugs by applying a combination of network‐based approaches and systems pharmacology‐based network medicine platform with the ability to quantify the interactions between the drug targets and virus–host interactome in the human protein–protein interactions network. (Zhou et al., ). The docking screening method was applied for a set of 2201 approved drugs to find inhibitory activities at SARS‐CoV‐2 main protease, and based on that, carfilzomib, eravacycline, valrubicin, lopinavir, and elbasvir have nominated as an inhibitor for SARS‐CoV‐2 main protease (Wang, ). Table  includes recent studies in terms of computational DR for coronavirus drug discovery and various diseases, respectively.</p><p>Various transcriptomic datasets in relation to SARS‐CoV‐2 were reported since the COVID‐19 outbreak. Basically, the idea is that for a specific disease signature composing of a class of up and downregulated genes, if there is available drug candidate, where those same classes of genes are alternatively downregulated and up‐regulated, respectively, this drug might be potential for that disease (Hoffmann et al., ; Le et al., ). Le et al. investigated existing computational drug repositioning with remarkably reversed differential gene expression in comparison with manifold input signatures for SARS‐CoV‐2 effects on human cells. They tried to predict the potential drugs toward differentially expressed gene sets from both cell line and organoid disease models and human samples. They found 102 compounds, and among them, 25 were selected in at least two of the signatures, and some of them have been already analyzed in clinical trials. They eventually tested 16 of their shortlisted candidates in live SARS‐CoV‐2 antiviral assays. If a drug could notably (FDR &lt; 0.05) reverse the disease signature, then the drug could be a potential inhibitor for the SARS‐CoV‐2. In the end, they found that 16 of their high predicted hits showed antiviral assay in live SARS‐CoV‐2. Four potential inhibitors were tested for inhibition of SARS‐CoV‐2 in a human lung cell line, Calu‐3, infected with SARS‐CoV‐2. Three drugs revealed in vitro antiviral efficacy—bacampicillin, clofazimine, and haloperidol with no toxicity effects (Le et al., ). Ge et al. used an integrative drug repositioning analysis through machine learning to integrate and mine large‐scale knowledge graph, literature together with transcriptome data to identify the potential compounds against SARS‐CoV‐2. They chose to work on CVL218, a poly‐ADP‐ribose polymerase 1 (PARP1) inhibitor. Their <italic toggle=\"yes\">in vitro</italic> data showed an efficient inhibitory effect against SARS‐CoV‐2 replication without a clear cytopathic effect. They further revealed that CVL218 can effectively bind to the nucleocapsid (N) protein of SARS‐CoV‐2 and could suppress the LPS‐induced production of many inflammatory cytokine, which are highly related to the prevention of immunopathology induced by SARS‐CoV‐2 infection (Ge et al., ). Another PARP1 inhibitor, PJ‐34 was previously studied for acute inflammatory including inflammatory diseases of the lung. It was observed that PJ‐34 can interact with the N‐terminal domain of the coronavirus nucleocapsid (N) protein and decrease the RNA binding ability to inhibit virus replication (Ge et al., ).</p><p>Beclabuvir is an antiviral drug for the treatment of hepatitis C virus (HCV) infection studied for inhibitory activity against SARS‐CoV‐2. It was reported that beclabuvir could be a potential compound for the treatment of SARS‐CoV‐2 through virtual screening and molecular docking in order to bind to RdRp of the SARS‐CoV‐2 (Talluri, ). Raltegravir belongs to a class of HIV integrase inhibitors used in the treatment of HIV‐1 infection. It obstructs the insertion of HIV‐1 DNA into the host cell genome. Raltegravir was selected as a highly effective drug against Mpro and has been repurposed for the potential treatment of COVID‐19 in many studies (Nabi et al., ).</p></sec></sec><sec id=\"cbdd14136-sec-0011\"><p>There has been considerable progress in the practical development and the rational application of computational drug discovery, giving rise to a paradigm shift in both industry and academics. CADD is a powerful approach for proposing new drug candidates, particularly when used in tandem with current chemogenomic methods. Even though CADD takes advantage of manifold dilemmas and approximations, this knowledge‐driven approach plays a significant role in the drug discovery process because of its ability to fast‐track drug discovery. Since CADD has provided several achievements, there would be a promising future to aid drug discovery of many more medicines in the future. Chemogenomics is a superior interdisciplinary field at the interface of chemistry, biology, and informatics, which dramatically decreases the cost and time of drug discovery.</p><p>This technique allows sequential genetics that is further interpreted into potential biological targets. Over the last years, there has been great attention to DR in pharmaceutical and biomedical companies, highlighting the paramount importance of repurposing. This interest has led to victorious launches in new, highly economical, and appealing indications compared to traditional drug discovery with high‐priced drug development. Over the last 3 years, the COVID‐19 pandemic has killed many people globally since there are no effective drugs against it. In this sense, we need to highlight that DR attempts the search for effective drugs in an urgent situation such as the COVID‐19. Utilizing DR is strongly recommended to continue investigating the repositioning of existing drugs against different diseases. However, how effectively CADD, including DR and chemogenomics, will produce feasible treatments across different disease areas remains to be seen. Therefore, it should not be surprising that success is limited at times.</p><p>Several basic computational problems still need to be resolved. To ameliorate drug design and development dilemmas, rapid developments in chemical and structural biology, bioinformatics, clinical experiments, and computational technology are essential. For example, high‐throughput screening (HTS) has been increasingly used in the pharmaceutical sphere and academic institutions to discover hit and lead compounds rapidly. This potential automated approach lets us screen the entire compound library of the number of chemicals or biological compounds in a target‐based assay or even a cell‐based assay to obtain the desired biological response. In order to use HTS against SARS‐CoV‐2, it is important to take advantage of a similar homology of virus in genomics, specifically, chemogenomics, as a versatile way of drug screening model that provides better bio‐safety.</p><p>Hitherto, CADD is the only way for screening drugs against COVID‐19 and is based on a single viral protein or small‐scale drug screening of formerly FDA antiviral drugs utilizing wild‐type SARS‐CoV‐2. Nevertheless, this approach is not well‐suited to HTS when focusing on the bio‐security requirements of SARS‐CoV‐2. Moreover, developing new drugs takes too much time without having a precise crystal structure of the viral target (Liu et al., ).</p><p>Thus, the best way to rapidly find a drug for the treatment of COVID‐19 would be the use of CADD methods in order to discover lead compounds for clinical use; for instance, we can build a program that takes advantage of structure‐based drug design, HTS, and virtual drug screening to identify new leads targeting the M<sup>pro</sup>. More recently, a library of 10,000 compounds involving approved drugs, drug candidates in clinical trials, and biologically active compounds to inhibit M<sup>pro</sup> was examined. As a result, six compounds were found with a great IC<sub>50</sub> against M<sup>pro</sup> and strong antiviral activity in cell‐based assays. Moreover, using CADD, the crystal structure of M<sup>pro</sup> in complex with Michael acceptor inhibitor known as N3 can be determined (Jin et al., ).</p><p>It is also possible to identify drug leads against SARS‐CoV‐2 by targeting the 3‐chymotrypsin‐like cysteine protease (3CL<sup>pro</sup>) enzyme performing a vital role in the life cycle of CoVs, including MERS‐CoV and SARS‐CoV (Ul Qamar et al., ). Scientists employed vHTS tools to predict the 3D structure of the SARS‐CoV‐2 3CL<sup>pro</sup> enzyme by constructing a 3D homology model and screening it against 32,297 potential antiviral phytochemicals/traditional Chinese medicinal compounds of a medicinal plant library. This effort led to the discovery of nine hits that need further optimization to be identified as anti‐SARS‐CoV‐2 leads. In addition, an interesting investigation based on an HTS assay employing a compound library of 2000 approved drugs and pharmacologically active compounds has also been carried out to find broad‐spectrum inhibitors against the replication of CoVs (Shen et al., ). From their HTS data, 56 hits were screened, and 36 compounds were validated in vitro using genetically engineered human CoV OC43 (HCoV‐OC43). Consequently, seven effective compounds were selected as broad‐spectrum inhibitors in which lycorine protected BALB/c mice against lethal human coronavirus OC43 (HCoV‐OC43), and emetine blocked MERS‐CoV entry following pseudo‐virus entry assays. These studies prove the efficacy of screening strategies in the rapid discovery of drug leads, resulting in new drugs for infectious diseases having no effective drugs or available vaccine.</p></sec><sec id=\"cbdd14136-sec-1010\"><p>No funds were received for this article.</p></sec><sec sec-type=\"supplementary-material\"/></body>],\n",
              " [<body><sec id=\"sec1-17411343221115819\" sec-type=\"intro\"><p>The severe acute respiratory syndrome coronavirus-2 (SARS-CoV-2)-caused Coronavirus\n",
              "  Disease 2019 (COVID-19) has resulted in a significant increase in morbidity and\n",
              "  mortality around the world.<sup/> Finding therapies for COVID-19 has taken a significant\n",
              "  amount of time and effort.<sup/> But besides advancements in technology and increased\n",
              "  understanding of human related illness, therapeutic developments have become much\n",
              "  slower than anticipated. Drug discovery can take decades and is complex and\n",
              "  expensive.<sup/> To bring a medicine to market, it ends up taking an average of\n",
              "  10 years and at least $1 billion.<sup/> The COVID-19 pandemic has\n",
              "  prompted the researchers and doctors to repurpose antiviral medications to combat\n",
              "  SARSCoV-2 infection.<sup/> Drug repurposing (DR), also known as drug rescuing, drug\n",
              "  redirection, drug repositioning, therapeutic switching, drug reprofiling, drug\n",
              "  recycling and drug re-tasking, is a method of recognizing novel therapeutic evidence\n",
              "  from Investigational/pro-drugs/old/already marketed/existing/FDA approved/failed\n",
              "  drugs, and applying the new advanced medicine to the management of diseases other\n",
              "  than the ones for which they were originally developed.<sup/> Regulatory authorities\n",
              "  throughout the world have established fast-track methods to speed up the research\n",
              "  and approval of COVID-19 therapeutics.<sup/> Some of the recommended\n",
              "  approaches include the use of antiviral medicines or immune function\n",
              "  modulators.<sup/> Many medicines have demonstrated potent activity against\n",
              "  COVID-19 in animal/preclinical investigations and have progressed to human clinical\n",
              "  trials, among which Molnupiravir being authorized for treatment of COVID-19 in the\n",
              "  United States, India, the United Kingdom, and other countries.</p></sec><sec id=\"sec2-17411343221115819\"><p>De novo identification and development of new molecular entities (NME) is a classic\n",
              "  strategy to drug discovery that comprises five phases: discovery and preclinical,\n",
              "  safety evaluation, clinical trials, FDA clearance, and FDA post-market safety\n",
              "  monitoring. Because of the specific characteristics of the medicine for a mechanism,\n",
              "  the approach is arduous, time-consuming, and costly, and it comes with a significant\n",
              "  risk of failure. Drug repositioning, on the other side, has only four stages:\n",
              "  compound recognition, compound procurement, development, and FDA post-market safety\n",
              "  monitoring.<sup>,</sup>\n",
              "  In comparison to the lengthy traditional research and development methods, DR is\n",
              "  inexpensive and faster method of bringing effective treatments to patients. Because\n",
              "  data about safety, effectiveness, and the proper delivery route is available, less\n",
              "  efforts are necessary to bring a repositioned medicine to\n",
              "  commercialization.<sup/> The most challenging component of product repositioning is\n",
              "  discovering novel drug responses. A number of ways have been developed to solve this\n",
              "  problem, such as computational approaches, biological experimental approaches, and\n",
              "  mixed approaches.<sup/> While the advantages of repurposing are obvious, the majority\n",
              "  of accomplishments to yet have been coincidental. Due to the lack of a comprehensive\n",
              "  physical drug database, poor quality drug descriptions, and inadequate readouts of\n",
              "  therapeutic activity out of which additional conditions might be expected,\n",
              "  systematic, large-scale repurposing initiatives have proved impossible. Latest\n",
              "  technological advancements have allowed for a significant improvement in the\n",
              "  capacity to measure pharmacological actions in their entirety.<sup/> Indeed, as\n",
              "  the number of repositioned pharmaceuticals mounted, it was suggested that a\n",
              "  comprehensive screening of all known drugs would reveal new suitable\n",
              "  drugs.<sup/></p><sec id=\"sec3-17411343221115819\"><p>Molnupiravir, commercialized as Lagevrio, is an orally bioavailable, directly\n",
              "  acting antiviral therapeutic medicine that was initially approved to treat\n",
              "  influenza.<sup/> It’s a precursor of β-d-N4-hydroxycytidine, a new\n",
              "  antiviral nucleoside analogue. SARSCoV-2, seasonal or pandemic influenza, and\n",
              "  MERS corona virus are all targets of this broad-spectrum antiviral\n",
              "  agent.<sup/> It was reported to be extremely effective in reducing\n",
              "  nasopharyngeal infection rate and had a positive safety and tolerability record\n",
              "  in COVID-19 patients who received short-course, 5-day treatment.<sup>–</sup> On 23\n",
              "  December 2021, the FDA approved the medicine under an EUA for the treatment of\n",
              "  mild-to-moderate COVID-19 in individuals who have a positive consequences of\n",
              "  SARS-CoV-2 virus test and are at high risk of developing chronic\n",
              "  COVID-19.<sup/></p></sec><sec id=\"sec4-17411343221115819\"><p>It is granted for patients who have received a positive test from significant\n",
              "  SARS-CoV-2 viral testing, or who are at danger of developing severe COVID-19,\n",
              "  which could result in hospitalization or death, or for whom other FDA-approved\n",
              "  COVID-19 treatment options are not available or efficacious.<sup/></p></sec><sec id=\"sec5-17411343221115819\"><p>It is not approved for use in individuals under the age of 18, for the\n",
              "  commencement of therapy in patients who require treatment due to COVID-19, for\n",
              "  use for more than 5 days, or as a pre-exposure or post-exposure prophylactic for\n",
              "  COVID-19 prevention.<sup/> Molnupiravir is not indicated for usage during\n",
              "  pregnancy. During therapy and for 4 days following the final dosage of the\n",
              "  medication, breastfeeding is not suggested.<sup/></p></sec><sec id=\"sec6-17411343221115819\"><p>For 5 days, four capsules of the medicine must be consumed per 12 h. Even if the\n",
              "  patient feels well, it is critical to finish the entire 5-days course of therapy\n",
              "  with the medicine.<sup/></p></sec></sec><sec id=\"sec7-17411343221115819\"><p>Molnupiravir acts by inhibiting RdRp of SARS-CoV-2 to induce RNA mutagenesis in two\n",
              "  steps. Molnupiravir is converted to EIDD-1931 in the body, which on phosphorylation\n",
              "  by host kinases provides the EIDD-1931-triphosphate. This triphosphate acts as an\n",
              "  alternate/competitive substrate for the RdRp enzyme of SARS-CoV-2. Therefore, RdRp\n",
              "  generates mutated RNA copies for SARS-CoV-2. This process causes the inhibition of\n",
              "  the normal functions of RdRp. Molnupiravir is a better electron donor than electron\n",
              "  acceptor, and hence this reducing property can contribute to the antiviral activity\n",
              "  as it affects the conditions required for viral infection.</p></sec><sec id=\"sec8-17411343221115819\"><p>Molnupiravir is an isopropyl prodrug of EIDD-1931, which has been shown to suppress\n",
              "  the reproduction of a wide range of viruses, including coronaviruses, with potential\n",
              "  safety profiles. In laboratory animals, EIDD-1931, on the other hand, had a poor\n",
              "  absorption. EIDD-1931 was quickly metabolised in nonhuman monkeys’ enterocytes after\n",
              "  oral administration. In the papers, there is information on EIDD-1931 uptake and\n",
              "  distribution in mice. EIDD-2801’s low bioavailability was addressed with the\n",
              "  development of Molnupiravir.</p><p>Molnupiravir was created with funding from the US government by scientists at Emory\n",
              "  University in Atlanta. Emory University, Ridgeback Biotherapeutics, Wayne and Wendy\n",
              "  Holman, and Merck have also agreed to collaborate on the development of Molnupiravir\n",
              "  as an oral therapy for non-hospitalized COVID-19 individuals. Molnupiravir was\n",
              "  developed to treat infections caused by the alpha virus. It was under pre-clinical\n",
              "  development for seasonal flu at the period of the pandemic’s outbreak. Following the\n",
              "  dissemination of COVID-19, the Molnupiravir development effort shifted to COVID-19\n",
              "  therapy.</p></sec><sec id=\"sec9-17411343221115819\"><p>In mice, ferrets, and nonhuman primates, it showed significant anti-influenza\n",
              "  efficacy and high oral absorption. It found a therapeutic window of &gt;1713\n",
              "  (antiviral effectiveness vs. cytotoxic effects) and proposed that Molnupiravir be\n",
              "  tested further in clinical trials for influenza therapy. It’s oral efficiency\n",
              "  against coronaviruses was also demonstrated in preclinical research in an animal\n",
              "  study. EIDD-1931 was also shown to be effective against a remdesivir-resistant\n",
              "  infection, suggesting that it may be potent against a larger variety of viruses than\n",
              "  remdesivir. SARS-CoV-2 propagation is ineffectively controlled by Remdesivir.\n",
              "  Molnupiravir, on the other hand, was shown to be efficient in ferrets in reducing\n",
              "  SARS-CoV-2 virus and halting dissemination. As a result, Molnupiravir has been\n",
              "  proposed as a protective measure to combat SARS-CoV-2 infection in the\n",
              "  community.<sup/></p></sec><sec id=\"sec10-17411343221115819\"><sec id=\"sec11-17411343221115819\"><p>Molnupiravir was well tolerated in the first double-blind, randomized-controlled\n",
              "  Phase 1 study (NCT04392219) on study participants, with dosage related\n",
              "  pharmacokinetics following medication. The prodrug Molnupiravir is quickly\n",
              "  degraded to its active component EIDD-1931 after oral ingestion, with an average\n",
              "  time of maximum detected concentration of 1–1.75 h. During the absorptive state,\n",
              "  there was no decline in absorption, but there was a drop in the absorption rate.\n",
              "  A single oral dose of up to 1600 mg and a range of 50–800 mg twice day for\n",
              "  5.5 days were shown to be harmless.</p><p>The Royal Liverpool and Broadgreen Clinical Research Facility used a Bayesian\n",
              "  technique to undertake an open label randomised controlled but small phase\n",
              "  Ib/IIa study called AGILE (NCT04746183). Elderly patients with SARS-CoV-2\n",
              "  illness diagnosed by RT-PCR within 5 days of appearance of symptoms were\n",
              "  randomised to standard of care (SOC) or 300, 600, and 800 mg Molnupiravir orally\n",
              "  twice for 5 days by oral route daily. Minimal detrimental reactions were\n",
              "  detected in all (4/4, 100%) patients who received 300 and 600 mg, 1/4 (25%)\n",
              "  patients having 800 mg, and 5/6 patients (83%) taking SOC. This study discovered\n",
              "  that the maximum dosage of 800 mg twice daily seemed to have a 0.9% chance of\n",
              "  causing 30% more toxic effects than the controls.</p></sec><sec id=\"sec12-17411343221115819\"><p>The safety and tolerance of COVID-19 in patients admitted with minimal to\n",
              "  moderate COVID-19 were analyzed in a double-blind, randomized-controlled,\n",
              "  multicentric phase 2a (MK-4482-006) (NCT04405570) study. After allocation, the\n",
              "  test group underwent twice daily oral dosages of 200 mg, 400 mg, and 800 mg\n",
              "  Molnupiravir for 5 days compared. Placebo. The study’s performance, wellbeing,\n",
              "  and tolerance were evaluated for 4 weeks after it commenced. When compared to\n",
              "  the placebo, the Molnupiravir 800 mg twice day group had a considerably shorter\n",
              "  time to elimination. Furthermore, as compared to placebo, the reduction in time\n",
              "  to viral RNA clearance was likewise larger and meaningful. On Day 3, viral\n",
              "  isolation was considerably lesser in subjects taking 800 mg twice daily versus\n",
              "  placebo. Other double-blind phase 2 experiment is now underway to assess the\n",
              "  safety of EIDD-2801 and its effect on SARS-CoV-2 infection shedding.</p></sec><sec id=\"sec13-17411343221115819\"><p>The independent data safety monitoring board recently halted the phase 3\n",
              "  double-blind, randomised trial (MOVe-OUT) that was designed to evaluate the\n",
              "  efficacy and safety of the drug in 1850 non-hospitalized participants with\n",
              "  COVID-19 (NCT04575597, MK-4482-002) due to the increased advantage in the active\n",
              "  therapy group compared to the placebo. A proven SARS-CoV-2 illness with samples\n",
              "  taken ≤5 days previous to the day of allocation and negative serum testing in\n",
              "  light of recent or earlier exposure were among the study’s eligibility criteria.\n",
              "  At day 29, the interim results (<italic toggle=\"yes\">n</italic> ¼ 775) of this phase 3\n",
              "  (NCT04575597) research revealed a 50% reduction in the probability of\n",
              "  hospitalization or mortality.</p><p>On day 29, 7.3% of Molnupiravir subjects (28/385) had been hospitalised or\n",
              "  deceased, compared to 14.1% of placebo individuals. On day 29, no deaths were\n",
              "  recorded in the Molnupiravir population, compared to eight casualties in the\n",
              "  placebo arm. Furthermore, the SARS-CoV-2 variant (gamma, delta, or mu), the\n",
              "  period of onset of manifestations, and the inherent risks had no influence on\n",
              "  the effectiveness of the drug. In both the Molnupiravir and control arms, the\n",
              "  prevalence of all side-effects and drug-related side effects was revealed to be\n",
              "  similar. However, compared to the placebo arm, fewer patients in the\n",
              "  Molnupiravir group stopped taking a medication.</p><p>A further phase 3 double-blind, randomised research (MOVe-IN, NCT04575584) that\n",
              "  has been set to investigate the effectiveness and safety of Molnupiravir in 304\n",
              "  hospitalised adult COVID-19 (MK-4482-001) subjects was halted after an initial\n",
              "  review of the data revealed that a clinical efficacy in hospitalised individuals\n",
              "  was improbable. Nonetheless, a huge (<italic toggle=\"yes\">n</italic> 14 1332) phase 3\n",
              "  multicentric, randomised, double-blind, placebo-controlled investigation\n",
              "  (NCT04939428) is actively exploring the safety and effectiveness of Molnupiravir\n",
              "  for the mitigation of COVID-19 in adults (MOVe-AHEAD) living with a human who\n",
              "  has COVID-19, with the assumption that Molnupiravir would be efficacious in\n",
              "  blocking laboratory-confirmed COVID-19 infestation through day 14 especially in\n",
              "  comparison to placebo. Meanwhile, many Indian pharmaceutical firms confirmed\n",
              "  preliminary results from phase 3 studies with Molnupiravir in mild/moderate\n",
              "  COVID-19 patient populations in India.<sup/></p></sec></sec><sec id=\"sec14-17411343221115819\"><p>Under agency from the Secretary of Department of Health and Human Services (HHS), the\n",
              "  Food and Drug Administration (FDA) may issue an Emergency Use Authorization (EUA)\n",
              "  approving the urgent use of an unauthorized medicine, unlicensed or uncleared\n",
              "  instrument, or unregistered biological component; or a non - authorized use of an\n",
              "  effective drug, authorized or approved machine, or registered biological\n",
              "  drug.<sup/> In other texts, when “no appropriate, authorized, and\n",
              "  accessible approaches” exist, an EUA can enable healthcare countermeasures (e.g.\n",
              "  prescription medications, vaccines) to be included during an announced outbreak “to\n",
              "  detect, treat, or avoid potential or life-threatening illnesses” associated with\n",
              "  biological as well as other entities.<sup/> An EUA differs from a\n",
              "  marketing authorization in that it is dependent on a higher standard of proof. In\n",
              "  order for an EUA to be granted, FDA must determine, based on existing data, that the\n",
              "  product is potentially effective for the intended purpose and that the product’s\n",
              "  known and prospective benefits exceed the product’s known and potential\n",
              "  dangers.<sup/></p><sec id=\"sec15-17411343221115819\"><p>The FDA grants EUAs, which represent the agency’s goal to safeguard public’s\n",
              "  health by guaranteeing the safety, effectiveness, and integrity of human and\n",
              "  veterinary pharmaceuticals, biologicals, medical instruments, the country’s food\n",
              "  system, cosmetics, and radiation-emitting goods. The FDA is in charge of the\n",
              "  following counterterrorism and evolving threats sectors:<list id=\"list1-17411343221115819\" list-type=\"simple\"><list-item><p>• Ensure that registered medical instruments are safe and secure;</p></list-item><list-item><p>• Developing emergency planning and response skills;</p></list-item><list-item><p>• Establishing a robust and effective food security policy;</p></list-item><list-item><p>• Accelerating the development and supply of healthcare\n",
              "  countermeasures;</p></list-item><list-item><p>• Assuring the agency’s assets are safe and secure</p></list-item></list></p></sec><sec id=\"sec16-17411343221115819\"><p>The procedure for obtaining an EUA consists of five phases:<list id=\"list2-17411343221115819\" list-type=\"simple\"><list-item><p>1. The identification of an urgent situation;</p></list-item><list-item><p>2. The announcement of an urgent situation;</p></list-item><list-item><p>3. Review of the request for EUA by the FDA;</p></list-item><list-item><p>4. Issuance of the EUA or denial of the request;</p></list-item><list-item><p>5. Termination of the EUA.<sup/></p></list-item></list></p><p>The FDA Commissioner must take numerous procedures before issuing an EUA. One of\n",
              "  the four criteria listed below must be met:<list id=\"list3-17411343221115819\" list-type=\"simple\"><list-item><p>1. The Secretary of Department of Defense (DoD) declares a combat\n",
              "  emergency or a serious possibility of a military urgent\n",
              "  situation.</p></list-item><list-item><p>2. The Secretary of the Department of Homeland Security (DHS)\n",
              "  declares a domestic emergency or a possibility for a domestic\n",
              "  emergency.</p></list-item><list-item><p>3. The Secretary of the Department of Health and Human Services (HHS)\n",
              "  announces a public health emergency or a serious risk of a public\n",
              "  health emergency.</p></list-item><list-item><p>4. The Secretary of Homeland Security issues a material security\n",
              "  decision.</p></list-item></list></p><p>The HHS Secretary can release a notice that situations prevail to support\n",
              "  awarding the EUA once one of the four decisions is made. This proclamation is\n",
              "  unique to EUAs and has nothing to do with other emergency announcements. EUA\n",
              "  will be approved if the conditions are satisfied.<sup/></p></sec><sec id=\"sec17-17411343221115819\"><p>Early involvement with FDA regarding possible EUA goods by corporate or\n",
              "  government sponsors can promote more thorough EUA proposals and improve FDA’s\n",
              "  capacity to examine and issue the EUA if needed. Even though an EUA may not be\n",
              "  given until the secretary has declared a state of emergency, FDA acknowledges\n",
              "  that the time available for the filing and evaluation of an EUA request may be\n",
              "  severely constrained in such situations. Discussions with the FDA regarding a\n",
              "  prospective EUA product might be part of the pre-EUA process. Such consultations\n",
              "  may take place before the HHS Secretary issues an EUA notification or submits an\n",
              "  official application for review of an EUA.</p></sec><sec id=\"sec18-17411343221115819\"><p>During whole EUA pathway, from pre-EUA filings until final judgment, including\n",
              "  the requirement for the EUA declaration, FDA normally works collaboratively with\n",
              "  HHS and other respective government entities, and any EUA discussions with\n",
              "  government affiliates. A sponsor requesting an EUA should make a formal request\n",
              "  in the format of an EUA filing, referencing any necessary pre-EUA filings\n",
              "  already examined by FDA and requesting that the EUA be issued through the same\n",
              "  procedure.</p></sec><sec id=\"sec19-17411343221115819\"><p>The Commissioner will sign a letter to the funder approving the emergency use,\n",
              "  that will include a summary of the licensed drug and its usage, any product\n",
              "  potential side effects, the factors for approval of the request, the context of\n",
              "  the approval, exemption of some specifications, and any restrictions on the\n",
              "  approved usages. The signed letter of approval and any supporting permitted\n",
              "  documents will make up an accepted EUA. If an EUA is approved, the FDA\n",
              "  commissioner must set essential criteria of use targeted at safeguarding patient\n",
              "  safety, to the degree practicable given the situation of the emergencies, and\n",
              "  may set additional restrictions for human safety. These restrictions, as well as\n",
              "  which are necessary, vary based upon whether EUA permits the need for an\n",
              "  unauthorized medicine or an FDA-authorized product used off-label.</p></sec><sec id=\"sec20-17411343221115819\"><p>These restrictions, as well as which are necessary, vary based upon whether EUA\n",
              "  permits the need for an unauthorized medicine or an FDA-authorized product used\n",
              "  off-label. Except if the EUA is repealed owing to the proclamation requirements,\n",
              "  it will remain in effect for the term specified in the EUA notification. The HHS\n",
              "  Secretary’s EUA announcement shall expire on the earliest of the HHS Secretary’s\n",
              "  judgment that the conditions that prompted the statement have expired or a\n",
              "  modification in the product’s authorized status so that the product’s approved\n",
              "  uses are no longer unauthorized. An unauthorized medication and its labelling,\n",
              "  as well as product description for an unlicensed use of a permitted drug, must\n",
              "  be discarded of upon withdrawal of an EUA or its discontinuation as just a\n",
              "  result of the cessation of the HHS EUA statement promoting it as shown in .<sup>,</sup></p></sec></sec><sec id=\"sec21-17411343221115819\"><p>A 505(b)(2) submission is a type of the US New Drug Application (NDA) that includes\n",
              "  comprehensive documents of efficacy and safety research, with part of the material\n",
              "  required for acceptance lacking a point of reference and relying on previously\n",
              "  authorized evidence.<sup/> Its goal is to boost drug research innovation without\n",
              "  necessitating duplicate research of already available data. Dosage form, strengths,\n",
              "  method of administration, dosing regimen, and API switch are examples of\n",
              "  modifications that support the filing of this application. It permits a\n",
              "  pharmaceutical company to use available data in its NDA by referencing it. This\n",
              "  saves money on license; the typical 505(b)(2) license costs $3-7 million,\n",
              "  significantly less than the predicted $1.3 billion it costs to bring the new\n",
              "  medication to market under 505(b)(2) and get FDA clearance in as short as\n",
              "  30 months<sup/> Furthermore, based on the scope of the modification to the\n",
              "  previously existing drug and the kind of clinical evidence included in the NDA, a\n",
              "  505(b)(2) candidate may be eligible for 3, 5, or 7 years of market dominance, rather\n",
              "  than the 180 days granted to generic medicines authorized under section\n",
              "  505(j).<sup/></p></sec><sec id=\"sec22-17411343221115819\"><sec id=\"sec23-17411343221115819\"><p>While a 505(b)(2) method provides a unique chance for quick approval, success is\n",
              "  dependent on discovering goods with established market difference, minimal\n",
              "  development threat, and huge revenue potential. Products with novel uses,\n",
              "  innovative combination drugs, and prodrugs of an established medicine are all\n",
              "  potential 505(b)(2) prospects.</p></sec><sec id=\"sec24-17411343221115819\"><p>Candidates should be assessed before to development since it is critical to\n",
              "  determine the marketing strategy of a product idea for sponsors and to avoid the\n",
              "  danger of expensive blunders. Some questions, such as scientific, medical,\n",
              "  regulatory, and economic viability, must be explored in order to produce\n",
              "  evidence that would validate a drug’s potential worth.</p></sec><sec id=\"sec25-17411343221115819\"><p>The pre-IND conference with the FDA kicks off the 505(b)(2) procedure, which then\n",
              "  proceeds forward to drug formulations (and, if essential, human trials) and\n",
              "  later submits the IND proposal. The goal of this conference is to get FDA\n",
              "  approval and advice on the studies, the chemistry, manufacturing, and controls\n",
              "  (CMC) strategy, and medical research programs in order to reduce the number of\n",
              "  new research needed. Certain development projects may undertake bridging\n",
              "  research that eliminate the necessity for animal or medical testing, or even\n",
              "  both, because the 505(b)(2) route enables the use of online records or the FDA’s\n",
              "  earlier findings in place of unique trial results. Phase I clinical testing\n",
              "  resources (typically used to establish therapeutic bioequivalence) must reflect\n",
              "  the industrial production process, along with packing. The three stability sets\n",
              "  that will be required for shelf-life assessments are manufactured in generally\n",
              "  at this point. As a result, even Phase I trials require a significant amount of\n",
              "  CMC research before they can begin.<sup/></p><p>Following the conclusion of all requisite studies, the applicant submits a\n",
              "  505(b)(2) application for drug commercialization as shown in . The FDA’s\n",
              "  evaluation team determines whether or not an NDA is complete after receiving it.\n",
              "  The review board has the right to decline to file the NDA if it is deficient. If\n",
              "  everything is in order, the review panel will have 6–10 months to decide whether\n",
              "  or not to license the medicine. When the FDA finds that a medicine is safe and\n",
              "  efficacious for such intended purpose, it must then collaborate with the\n",
              "  applicant to establish and modify prescription documentation. “Labeling” is the\n",
              "  term for this. However, before a medicine may be licensed for commercialization,\n",
              "  there are many unresolved difficulties and those should be resolved. FDA may ask\n",
              "  the manufacturer to respond to queries relying on existing information. In some\n",
              "  circumstances, the FDA needs additional research. The manufacturer can now make\n",
              "  a decision whether or not continue working on the project. There are pathways\n",
              "  for formal application if a manufacturer disagrees with an FDA\n",
              "  judgement.<sup/></p></sec></sec><sec id=\"sec26-17411343221115819\" sec-type=\"conclusion\"><p>The research community has rallied to discover and assess various medicines and\n",
              "  vaccinations as the COVID-19 outbreak spreads throughout the world, from academia\n",
              "  and government institutions to minor biotechnology startups and major pharmaceutical\n",
              "  enterprises. Repurposing preexisting medications authorized for other uses as\n",
              "  antiviral treatments for SARS-CoV-2 is one feasible method. However, there are a few\n",
              "  drawbacks to drug repositioning, such as the fact that the recommended dose for the\n",
              "  treatment of a novel disease is generally different from that needed for its\n",
              "  intended target disease, and if this is the case, the discovery group will have to\n",
              "  start from Phase I human development, effectively eliminating drug repositioning’s\n",
              "  upsides over de novo drug development. Although repurposed medications with\n",
              "  acceptable safety records might be beneficial, there are currently no widely\n",
              "  marketed COVID-19 preventive or therapy alternatives.</p></sec></body>],\n",
              " [<body><sec id=\"Sec1\"><p id=\"Par2\">The sudden outbreak of SARS-CoV-2 in 2019 took the world by storm and despite there being vaccines, numerous other alternative treatment approaches are also being researched. Coronavirus disease-2019 (COVID-19) is a communicable disease caused by severe acute respiratory syndrome coronavirus-2 (SARS Cov-2). This disease was first detected in Wuhan, China, and has expanded its reach globally, leading to myriad deaths. According to World Health Organisation, the total number of COVID-19 confirmed cases worldwide was found to be 53,22,01,219 and total number of deaths accounted to 63,05,358 as of 10th June, 2022. Among many regions, Europe holds the top position in highest number of confirmed COVID-19 cases (22,24,17,177) then followed by America (15,89,83,746), Western Pacific (6,17,35,224), South-East Asia (5,82,17,287), Eastern Mediterranean (2,18,07,376), and Africa (90,39,645) respectively []. These statistics reveal that even today, the disease is still infecting several people, many of whom die due to complications. This pandemic requires quick and easily accessible knowledge for hastening the development of treatments and diagnostic tests []. However, the development of specific drugs for this disease takes several years, paving the way for exploring alternative approaches. One such potent strategy is repurposing drugs, for alternative utilization of already existing and approved drugs. These drugs must have already undergone all safety and clinical trials, which would further speed up the drug discovery process. The COVID-19 pandemic has ensured research to explore various alternatives, drug repurposing being one of them, as part of the disease management strategy. Several drugs such as chloroquine, hydroxychloroquine, remdesivir, tocilizumab, kaletra, and favipiravir are currently being testing in clinal trials for repurposing against COVID-19 [].</p><sec id=\"Sec2\"><p id=\"Par3\">SARS-CoV 2 belongs to the Nidovirales family, under the beta coronaviruses genera along with other viruses such as Middle East Respiratory Syndrome Coronavirus (MERS-CoV), HCoV-HKU1, and the HCoV-OC43 (Human Coronavirus variants) []. The replication of SARS-CoV-2 occurs with the help of mRNAs. These viruses have the largest RNA genome among other viruses that have RNA as their genetic material. The RNA genome is approximately 30 kb in size and runs in the 5’ to 3’ direction, with single-stranded positive-sense RNA (+ssRNA). A cap at 5’ end and poly-A-tail at 3’ end at the end of the genomic RNA is a feature of the genetic material. Several open-reading frames in varying numbers are present in the genomic RNA and it encodes for 4 structural proteins, 16 non-structural proteins, and a few accessory proteins [].</p><p id=\"Par4\">The nucleocapsid in the virus protects the viral RNA genomes from external environmental conditions. At the relaxed position, the nucleocapsid appears to be in a helical shape, but within the virus, it appears spherical. The genome of SARS-CoV-2 codes for four main structural proteins when compared to five in other coronaviruses leaving out the hemagglutinin-esterase (HE) protein. Nucleocapsid (N), membrane (M), envelope (E), and spike (S) are the four important structural proteins expressed by the SARS-CoV-2 virus []. On the surface of the virion, three structural proteins (M, E, and S) are present, while the nucleocapsid protein resides in the core region bound to genomic RNA. The spike proteins of the virus help in the attachment, entry, pathogenesis, and tissue tropism, whereas, the infection is carried out by the envelope protein. Ribonucleoprotein is formed with nucleocapsid protein’s assistance and the membrane proteins aid in the overall viral architecture []. The SARS-CoV-2 virus enters with the help of spike proteins by binding to the human angiotensin-converting enzyme 2 (hACE2) through its receptor-binding domain (RBD) and human proteases that make them proteolytically active [].</p></sec><sec id=\"Sec3\"><p id=\"Par5\">Beta, gamma, delta, and omicron are the latest variants of concern for the SARS-CoV-2 virus. South Africa was first to report the beta variant of COVID-19, gamma was noted first in Brazil, the delta in India, and omicron in western Europe. Omicron which is the most recent variant of COVID-19 carried all the mutations that were present in the previous variants along with some new improved functions such as showing partial resistance to the vaccine and enhanced transmissibility compared to others []. The major coronavirus variants can be classified into two []: previously circulating variants of concern (VoC) and currently circulating variants of concern (Table ).</p><p id=\"Par6\">These five major variants of concerns are the current targets for research to find repurposed drugs for exploring treatment strategies. In addition to this, certain amino acids are conserved among the coronavirus viruses, while some have a unique mark in the amino acid indicating lineage mutations. The variants of concern belong to various lineages. B.1.1.7 has the mutations H69-V70del, P618H, and N501Y, all indicating alterations in the amino acids. Lineage B.1.351 has mutations K417N, N501Y, and E484K. Likewise, another variant of concern of lineage P.1 has mutations N501Y, K417T, and E484K. B.1.526 lineage is mutated at D614G, T95I, D253G, and S477N. B.1.427 has mutations at L452R and D614G. B.1.429 is mutated at sites D614G, S13I, W152C, and L452R. B.1.617 variant has variations P681R, E484Q, and L452R [] (Fig. ).</p><sec id=\"Sec4\"><p id=\"Par7\">Apart from the variants identified in the most common lineages, there are approximately 1,516 SARS-CoV-2 lineages to date [, ]. To identify the variants in all the existing lineages and those that exist not only in the spike protein of SARS-CoV-2, this study also performed a search in  and summarized the different types of variants (Tables , , , and ). Synonymous, non-synonymous, deletions, and non-coding variations were identified, and changes in the amino acid positions, and the reference and mutated codon as screened in all 1516 lineages of the coronavirus. As per the latest data as of 6th June 2022, 131 non-synonymous mutations have been noted, 51 synonymous variants, 11 deletions, and 11 non-coding variations. These mutations are in various regions of the virus, including the spike proteins, nucleocapsid proteins, different open reading frames, and membrane and envelope proteins.</p><p id=\"Par8\">Additionally, some mutations are characteristic of specific regions, while some others are present in all the regions in SARS-CoV-2. For instance, in Asia, three mutations in QLA46612 (isolated from South Korea) were observed within the same sequence []. This particular study stressed the fact that binding sites of the ligands and potential drugs change if the structure of the spike protein is damaged. Moreover, the study also tried to comprehend if more than one mutation affects the binding of the drug, thereby altering the potentially identified drug binding sites. It was later revealed that drug binding sites were greatly affected in structures with multiple variations []. Thus, the SARS-CoV-2 variants may have different binding sites making them identifying a suitable drug challenging. However, this arena will pave the way for prospective vaccine trials.</p></sec></sec><sec id=\"Sec5\"><p id=\"Par9\">It has been reported that traditional Chinese medical therapies have increased the clinical manifestation of the virus among patients, along with enhanced the severity of the disease []. Major concerns of the recent vaccines which were developed during the pandemic were their safety and efficacy, the emergence of new variants, challenges that occur due to vaccine distribution among a larger population, and many more. Recently, an mRNA-based vaccine (BNT162b2) was reported to have adverse side effects such as fatigue, headaches, and immense pain at the site of injection. Variants are concerns among the present therapies due to severe mutations that occur in the spike protein as well as the increased transmission rate when compared to its predecessor. Distribution of the vaccines to the greater population becomes highly challenging to the lack of storage facilities in different geographical locations as well as the stability of the vaccines which highly depends on the storage conditions such as temperature []. These limitations warrant the use of alternative approaches, with drug repurposing being the most effective strategy that is currently being researched.</p></sec></sec><sec id=\"Sec6\"><p id=\"Par10\">The concept of drug repurposing holds promises of swift clinical trials and a reduced cost than drug development de novo. Drug repurposing, also known as drug repositioning, is the process of using approved and existing drugs for treating diseases other than the intended. Drug repurposing is pragmatic and alluring, given the considerable time and cost requirements for typical drug development []. Additionally, a great number of potential drugs do not reach clinical trials, with lesser than 15% of the compounds receiving approval []. Some examples of approved repositioned drugs include aspirin, the cyclooxygenase inhibitor for coronary heart disease, antibiotic erythromycin for gastric motility, and sildenafil, a phosphodiesterase inhibitor for erectile dysfunction []. Other drugs that have disconcerting side effects also merit repurposing as corroborated by the successful implementation of thalidomide, an antiemetic against multiple myeloma [].</p><p id=\"Par11\">Currently, there is a huge interest in the repurposing of drugs to speed up the identification of drugs that can prevent or treat COVID-19. The chief importance of repositioning is to accelerate the traditional drug discovery process by the detection of novel, safe, and effective clinal-use drugs in humans. The primary rationale for this is the utilization of drugs that have the same molecular pathways but are involved in different diseases []. The mechanism of repurposing depends on the type of targets that the drugs are being used against. One of the most valuable drivers of repurposing has been the chance discovery of the pharmacological activity of new drug targets, implying a newer indication of drug use []. For instance, drugs lopinavir and ritonavir have been used for the treatment of Human Immunodeficiency Virus-1 and 2 (HIV) []. These drugs were also the first drugs to be repurposed for the treatment of COVID-19. The main reason for this is due to the resemblance of the SARS-CoV and MERS-CoV (beta coronavirus) genome sequences to SARS-CoV-2 [, ]. Thus, the drugs being utilized for the treatment of MERS-CoV and SARS-CoV-2 could be essentially used for COVID-19 treatment, implying that the mechanism of action of repositioned drugs against their targets should be similar to or resemble its family target members [, ]. Successful administration to patients with moderate COVID-19 symptoms has been previously observed, confirming the hypothesis of drug repurposing [, ]. From all this, it can be elucidated that the concept of drug repurposing relies on two major scientific bases: (1) the idea of pleiotropic drugs, (2) the discovery of drugs via human genome interpretation that certain diseases share the common targets.</p><p id=\"Par12\">It begins with the extraction of primary data on the drug, disease, and target. A drug-disease network is eventually constructed and its features of it are extracted to filter the data and pre-process it thoroughly. A computational-based evaluation is carried out via methods detailed in the following sections, including advanced approaches such as machine learning that entails the structural analysis of the drug candidates. The pharmacokinetic studies ensue, which are essential for obtaining an FDA approval for use by the public. This general workflow and roadmap to drug repurposing are detailed in Fig. .</p><sec id=\"Sec7\"><p id=\"Par13\">Drug repurposing is known to decrease the costs needed for the development of new drugs, with additional savings in pre-clinical phases I and II []. Aside from this, the scope of drug repurposing expands to the following, with instances in COVID-19 research, as discussed in the below sections:<list list-type=\"alpha-lower\"><list-item><p id=\"Par14\">Precision medicine: This area is an emerging strategy that studies the variability of individual genes, the lifestyle, and the environment for each person to decide upon the correct treatment approach []. Disorders with traits that are common are characterized as a single condition and are part of a spectrum of illnesses. Safer and more effective medications can be found if the drugs could be tailored to the variations in the genome of the individual, their proteome, transcriptome, or metabolome. This approach would help target the specific types of a general condition, with repositioning paving the way for precision medicine. Recent studies have implicated the use of computational techniques such as machine learning algorithms as a precision medicine tool for drug repurposing for the treatment of COVID-19 []. The decision tree algorithms used in the study identify patients with longer survival rates and better oxygen support when treatment was provided and suggest corticosteroids and remdesivir as effective COVID-19 repurposed drugs.</p></list-item><list-item><p id=\"Par15\">Rapid therapeutic options: With the advent of the COVID-19 pandemic and reduced therapeutic options for the disease, repurposing offers rapid therapeutic strategies to combat the disease. El Bairi et al. [] reviewed various anticancer drugs for the management of COVID-19. Likewise, another study has elucidated and suggested the use of interleukin inhibitors, Janus-kinase pathway inhibitors, interferons, vascular endothelial growth factor (VEGF) inhibitors, immune check-point inhibitors, Bruton tyrosine kinase inhibitors (BTK), and XPO-1 (a selective inhibitor of nuclear export) as some of the alternative drugs that could be repurposed for rapid COVID-19 treatment [].</p></list-item><list-item><p id=\"Par16\">Identification of multiple purposes for existing drugs: The latest applications of already approved drugs well beyond the scope of their existing medical aid in the identification of multiple purposes for such drugs in just four major steps–identification of compound, acquisition of the compound, development of the compound, and FDA approval after monitoring of its safety. This reduces the time taken for traditional drug discovery, lowers developmental cost as well as risk, and also reveals novel pathways and targets for already approved drugs, allowing them to have higher functionality [, ]. For instance, a recent study has repurposed antiviral protease inhibitors such as chloroquine, remdesivir, and lopinavir, which have already shown efficient inhibition of the SARS-CoV-2 virus in in vitro studies, using the extra-cellular vesicles, to make it more robust against the virus [].</p></list-item><list-item><p id=\"Par17\">Identification of direct-acting and host-targeting antivirals: Drug repurposing also has applications in the identification of existing antivirals that are either direct-acting or host-targeting. Studies have shown that anti-hepatitis C combinations act as potential direct-acting main protease inhibitors of SARS-CoV-2 []. Similarly, another study suggested p38 MAPK (mitogen-activated protein kinase), as a vital host target for inhibiting the replication of coronaviruses, after studies proved that Ralimetiniib mesylate was found to be active in both Calu-3 and HeLa-ACE2 cells [, ].</p></list-item></list></p><p id=\"Par18\">Thus, drug repositioning for COVID-19 research has now taken the front seat and has a huge scope for tackling the SARS-CoV-2 virus.</p></sec></sec><sec id=\"Sec8\"><p id=\"Par19\">Drugs are small molecules that affect the molecular pathways of important protein targets that are most likely responsible for causing a specific disease. These molecules have a lower molecular weight and are typically designed using rational drug design or structure-based drug design. With increasing incidences of emerging diseases such as COVID-19, for which immediate treatments were not available, the need for drugs to treat the disease became indispensable. With the SARS-CoV virus, mutating at a fast pace and affecting millions of people globally, there was a pressing need to address this issue by using all alternative approaches to discovering a drug [, ]. In this light, drug repurposing came into the picture. Comprehending detailed molecular mechanisms between the drugs and their targets is critical in the discovery of drugs. Even though several biochemical and biophysical methods can explicate the molecular interactions of the small molecules with their proteins, the most powerful tool is structural and computational biology which can aid in interpreting the mechanism of action of the interaction complexes []. Bearing this in mind, drug repurposing is the alternative approach meant to identify the most suitable drug candidates from among the existing libraries, to combat specific diseases.</p><sec id=\"Sec9\"><p id=\"Par20\">Currently, several approaches exist that pave the way for drug repurposing. Advancements in computational chemistry and chem-bioinformatics pave the way for exploring more accurate predictive repurposing.</p><sec id=\"Sec10\"><p id=\"Par21\">Regarding developments in multiplex assays, cell-based screening, mining of data, chem-informatics and bioinformatics approaches, and their databases, pharmaceutical industries have now shown augmented keenness in digging up molecules that had once failed due to several reasons []. Generally, there are three important steps before repositioning drugs: (a) drug candidate identification for the specific disease, (b) assessment of the selected drug candidate in pre-clinical models, (c) safety evaluation in phase II clinical trials [].</p><p id=\"Par22\">The approach of repositioning itself can be subdivided into experimental and computational approaches. Using these approaches in combination yields better outcomes than using them individually. Computational strategies are driven by a huge amount of data that entails the utilization of genetic expressions, the structure of the chemical compounds, and the proteomic drug candidate data []. Several computational medicinal approaches add value to the drug repurposing process.</p><sec id=\"FPar1\"><p id=\"Par23\">Structure-based approaches such as virtual screening help in the identification of small molecules from an existing library, that can be used for repurposing. Previous studies have most commonly employed molecular docking for virtual screening against targets of SARS-CoV-2, such as the main protease, with possible drugs for repurposing such as digitoxigenin, beta-eudesmol, crocin, rutin, ritonavir, emetine, hesperidin, and indinavir [, ]. Likewise, other virtual screening approaches used previously for COVID-19 drug repurposing include combinations of docking with molecular dynamic (MD) simulations and calculation of free energy using MM-GBSA studies [] and two-way docking with in silico ADMET studies []. The following strategies fall under structure-based approaches for drug repurposing.</p></sec></sec></sec><sec id=\"Sec11\"><p id=\"Par24\">Molecular docking is a chem-informatics strategy to predict the conformation of the docking site/binding site of a drug candidate to the protein target []. The prediction of the best conformation where the drug and protein form an interaction complex, that will stop the molecular pathway of the target, in turn affecting the occurrence of the disease is the idea of docking []. Novel interactions that will help in understanding the potential drug candidates for repurposing can be identified using docking studies. The advantage of this strategy is that it can be followed by other computational validation techniques such as molecular dynamics simulations and Quantitative Structure–Activity Relationship studies (QSAR), to understand the complex stability and thereby increase the drug repurposing opportunity. The authors’ previous studies have used this computational technique against various targets of SARS-CoV-2 [, ]. The disadvantage is that not all proteins of interest are available, due to a lack of macromolecular structure. Applying more than one computational medicine approach will provide further authenticity to the predicted results.</p></sec><sec id=\"Sec12\"><p id=\"Par25\">The static binding energies between the drugs and targets obtained via molecular docking studies are further simulated dynamically in real-time to comprehend the stability of the interaction complex in biological environments. Molecular dynamic simulations have proved to be one of the most reliant strategies in drug repurposing. Elfiky [, ] proved via MD simulations that Setrobuvir, YAK, and ID-184 had the best antiviral activity against RNRP (RNA-dependent RNA polymerase) of SARS-CoV-2, after screening out 31 compounds that were in clinical trials or had known antiviral activity. Similarly, MD simulations identified the stability of lopinavir complexed with SARS-CoV-2 main protease as the most stable in biological conditions [, ], along with other identified antivirals such as carfilzomib, valrubicin, elbasvir, and eravacycline.</p></sec><sec id=\"Sec13\"><p id=\"Par26\">Quantitative structure–activity relationship acts as a tool to identify the relationship between the structure and the activity of compounds. With regard to COVID-19, this approach combined with docking and simulations has brought immense benefits to the design and repurposing of drugs for SARS-CoV-2. A study has previously created a QSAR model founded on machine learning strategies utilizing several hundred inhibitor molecules of the SARS-CoV-2 main protease. This model was employed for virtual screening and docking, followed by shortlisting of the 20 best candidates for MD simulation validation []. Similarly, another recent study designed a binary-QSAR model for the virtual screening of several drugs that are FDA-approved and are in clinical studies against the main protease of SARS-CoV-2 []. This QSAR protocol resulted in cefuroxime pivoxetil as a potential drug repositioning molecule against SARS-CoV-2.</p><sec id=\"FPar2\"><p id=\"Par27\">Network-based approaches are commonly employed in repurposing due to the capability of amalgamating and integrating several sources of data [, ]. Advancements in high-throughput technology and computational approaches allow the modeling of biological systems by the network which facilitates diagnostic and pharmaceutical research that is structure-guided, with the possibility of identification of novel targets [, ]. Studies have stated the importance of drug-drug networks, drug-target networks, and drug-disease networks in drug repurposing []. The two-most common types of network-based strategies in computational biology include network-based propagation and network-based clustering approaches []. King et al. [] have previously studied PPI network-clustering technique for drug repurposing, while Vanunu et al. [] have used the disease-gene network propagation technique to accurately predict the disease-gene relationships.</p></sec><sec id=\"FPar3\"><p id=\"Par28\">Recently, the use of AI technology for drug repositioning has surfaced with several studies focusing on repurposing for COVID-19 research. A deep-learning-based approach to screen a huge number of molecules having allocated datasets to seek compounds possessing SARS-CoV-2-inhibiting activities was carried out previously by Ke et al. []. To validate the designed model and verify the antiviral activity of the compound that the model predicted, an in vitro cell line for feline coronavirus replication was also set up in the study. Likewise, Beck et al. [] utilized an already trained deep-learning prediction model for drug-target interactions known as Molecule Transformer-Drug Target Interaction (MT-DTI), to find antiviral drugs that are commercially available and which could perturb SARS-CoV-2 components including RNA-dependent RNA polymerase, proteinase, and helicase. Thus, AI-based techniques are providing the scientific communities a chance to fight COVID-19 in a much faster way, owing to the accessible web servers, tools, and models that reduce the time required for a drug to come into the market.</p><p id=\"Par29\">The scope and developments of computational medicinal chemistry toward COVID-19 treatment strategies are summarized in Fig. .</p></sec><sec id=\"Sec14\"><p id=\"Par30\">Several computational approaches for repurposing have been used over the years to enhance the efficiency of the predictive outcomes and decrease the drawbacks. Currently, advancements in the field of computer science have led to the use of machine learning methods, network-based techniques, and matrix decomposition methods, along with other algorithms such as tensor decomposition for repurposing []. This technique has been performing better than several basic algorithms such as Network-based Random Walk with Restart (NRWRH), SNScore, and Collective Matrix Factorization (CMF). This technique solves the difficulty of negative sample training in cases where deep learning methods are used such as multilayer perceptron, auto encoders, and deep belief networks []. The tensor decomposition technique involves the construction and decomposition of 3D tensors that enable the associations among protein targets, drugs, and diseases [].</p><p id=\"Par31\">Topological data analysis (TDA) is then taken into consideration to examine the drug properties, diseases, and disease targets to be able to recover the missing drug associations, diseases, and their targets, and to also predict the new repositioning candidate []. Moreover, drug repositioning can be performed on the drug candidates that arise from two clusters with robust connections, instead of assuming only the new-drug associations. To define these clusters, the diseases with similar molecular mechanisms are more inclined to appear within the same cluster. This approach, therefore, offers huge potential to explore predictive drug repurposing research.</p></sec></sec><sec id=\"Sec15\"><p id=\"Par32\">There are several applications of computational biology and chem-bioinformatic approaches for drug discovery in COVID-19 research. A recent study has developed a multi-model deep learning-based algorithm for drug repurposing to aid in COVID-19 research. This work utilized the multimodel-restricted Boltzmann machine approach (MM-RBM), which can associate data about multiple modalities, to amalgamate two data types, inclusive of the chemical structures of small drug molecules and genes expressed differentially []. The results from this study showed clusters that were employed to identify the drugs that were notably similar to the proposed and existing COVID-19 treatment medication. This work also demonstrated results in predicting drugs with minimum side effects and high efficiency.</p><p id=\"Par33\">Another recent study proposed a mechanism-driven neural network technique called DeepCE, that employs a graphical network to model the gene–gene associations and the chemical structure-gene associations to predict the gene expression profile []. This work also proposed a data augmentation approach that extricates only beneficial data from a set of variable datasets, to screen novel drug candidates for COVID-19. A recent application of signature mapping on COVID-19 research demonstrated the use of Connectivity Map databases [] for generating drug perturbation profiles []. A modified version of this study was implemented by Jia et al. [] where the input used was the expression profile data from healthy and infected individuals to generate a pathway-based drug repurposing framework. Co-clusters were identified before performing reverse signature matching [].</p><p id=\"Par34\">Another recent study designed and developed Deep Docking, which uses QSAR (Quantitative Structure–Activity Relationship) models to predict the docking scores of drug candidates that target the SARS-CoV-2 3CL protein. Deep Docking docked all the compounds and provided a list of potential drug hits, that could be used to repurpose for COVID-19 []. Several studies have implemented random-walk algorithms to identify drug repurposing candidates for COVID-19 [–]. The pathogenic mechanisms instigated by the spike protein using the information from all related coronaviruses were also explored and random walk algorithms were implemented on the molecular networks to recognize the most relevant COVID-19 drug targets []. Thus, these computational perspectives on drug repurposing provide a unique outlook on the alternative treatment options for the ongoing COVID-19 disease.</p></sec></sec><sec id=\"Sec16\"><p id=\"Par35\">Drugs such as lopinavir, ritonavir, boceprevir, carmofur, and doxycycline act on the 3-chymotrypsin-like proteins/SARS-CoV-2 main protease that play an important role in cleaving and releasing the 16 non-structural proteins []. Fostamatinib disodium and platycodin target the Nsp3 proteins which play a major role in the infection and pathogenesis. The RNA-dependent RNA polymerases (RdRp, Nsp 12) that are primarily involved in important activities such as transcription and replication of the viral RNA genome are targeted by drugs such as remdesivir, favipiravir, and ribavirin []. For the helicase (Nsp13) of SARS-CoV-2, a multifunctional protein that helps in the proliferation and replication of the virus, drugs such as vapreotide, atazanavir, daclatasvir, and bismuth potassium citrate commonly block this function of the target. Likewise, dinucleoside 123 and raltegravir affect the pathways of methyltransferases (guanine-N7-methyltransferase, 2’-O-methyltransferase), while gliclazide and memantine affect the SARS-CoV-2 E protein, which is the envelope protein of the virus. Arbidol and chloroquine target the spike protein that helps in the viral entry, along with ACE2 protein in the host, thereby affecting the spike protein pathway [] (Table ).\n",
              "  </p><p id=\"Par36\">The ACE receptors are blocked via losartan and captopril, transmembrane protease serine 2 (TMPRSS2) is blocked by camostat mesylate and with anyone, and cathepsin L is inhibited by hydroxychloroquine, which and teicoplanin. The viral neuraminidases and the DPP4 cell receptor pathways are hindered due to the action of oseltamivir and sitagliptin, while pemirolast, nitrofurantoin, isoniazid pyruvate, eridictoyl, hypericin, cepharanthine, and ergoloid block the human ACE2 and the S protein of SARS-CoV-2 []. The most important Janus kinase pathway is hindered by baricitinib, RdRp, helicase, exo-nuclease, and methyltransferase of SARS-CoV 2 by Efavirenz and Poly [ADP-ribose] polymerase 1 (PARP-1) by Mefuparib hydrochloride. Poly [ADP-ribose] polymerase 1 (PARP-1), glycogen synthase kinase 3 (GSK-3), and also biological pathways such as NF-kappa B signaling are all affected by the use of repurposed drugs toremifene, sirolimus, mercaptopurine, irbesartan, emodin, dactinomycin, and melatonin []. Other common drugs such as nitazoxanide, ivermectin, and azithromycin tend to symptoms of the virus, and play a role in inhibiting the role of SARS-CoV-2.</p><p id=\"Par37\">Despite these drugs being implicated as a potential treatment for COVID-19, ethical issues arise regarding the use of these drugs without proper clinical phase trials and FDA (Food and Drug Administration) approval. Therefore, a list of the FDA-approved drugs for the treatment of COVID-19 is provided in Table . Mahdian et al. [] and Yuce et al. [] state that Paritaprevir is important against 3-chymotrypsin-like proteins/SARS-CoV-2 Main protease and plays an important role in cleaving as well as releasing 16 non-structural proteins (Nsps). It also affects the ACE2-SARS-CoV-2-RBD complex. Likewise, ledipasvir, vancomycin, and sirolimus target the ACE2-SARS-CoV-2-RBD complex, a spike protein that helps in the viral entry along with the help of ACE2 protein in the host organism. Similarly, FDA-approved irinotecan, abemaciclib, estradiol benzoate, capmatinib, olaparib, lumacaftor, and pazopanib target CD147, a member of the immunoglobulin superfamily (IgSF) interacts with spike protein of SARS-CoV-2 virus [, ]. Other drugs such as nilotinib, rifabutin, dactinomycin, dihydroergotamine, bromocriptine, entrectinib, selinexor, elbasvir, quinupristin, rifapentine, rutin, digitoxin, diosmin, and antrafenine target 3-chymotrypsin-like proteins/SARS-CoV-2 Main protease that plays an important role in cleaving as well as releasing 16 non-structural proteins (Nsps).</p><sec id=\"Sec17\"><p id=\"Par38\">According to the latest updates from , phase 3 trials are completed for remdesivir, lopinavir/ritonavir, interferon beta-1A, hydroxychloroquine, and AZD7442 ( registration number: NCT04315948). Trials for Silmitasertib are active and in phase 2, while for Pioglitazone, phase 4 trials are completed ( registration number: NCT04473274). Phase 2 clinical trials are completed for dipyridamole 100 mg (NCT04391179), and phase 2 is completed for Leronlimab 700 mg (NCT04678830). L-ascorbic acid has phases 1 and 2 completed (NCT04357782), and phase 1 for aerolized hydroxychloroquine sulfate completed (NCT04461353). Similarly, drugs duvelisib (NCT04487886), ruconest (NCT04530136), and infliximab (NCT04425538) have phase 2 clinical trials completed. This data is provided in detail in Table  and the 2D structures of some of these drugs are shown in Fig. .\n",
              "  </p><p id=\"Par39\">When remdesivir was administered intravenously to the first patient diagnosed with SARS-CoV-2 in the USA, no adverse effects were noted with some clinical benefits in the patients []. Evidence also showed an accelerated recovery rate of 4 days when remdesivir was employed []. However, when randomized, double-blind, placebo-controlled trials were conducted by other researchers, it was found that remdesivir was not associated with significant improvement statistically [, ]. Additionally, due to some adverse effects early on, remdesivir was then discontinued, and WHO has now provided a conditional recommendation for the use of remdesivir against COVID-19 []. Likewise, real-world usage of lopinavir/ritonavir revealed no beneficial response against COVID-19 hospitalized patients in a randomized, controlled, trial []. Overall, the available data show an unfavourable response of patients to lopinavir/ritonavir, and hence the use of this drug has to be monitored and controlled vigilantly. Despite there being studies on hydroxychloroquine, an analogue of chloroquine, where improvements have been noted in patients suffering from COVID-19 in a study conducted in Wuhan, there still seems to be very limited data on the reliability of the drug []. Interferon-beta-1A has been used in hospitalized patients for treating COVID-19 at Shaid Behesti University of Medical Sciences, in combination with other drugs. However, despite it showing promising outcomes, more clinical data is required to corroborate the reliability of the drug []. Therefore, the clinical data presented provide evidence of the efficacy of the use of these drugs,however, more research is indispensable for the proper application and employability of these drugs for SARS-CoV-2 treatment.</p></sec></sec><sec id=\"Sec18\"><p id=\"Par40\">According to literature, coronavirus papain-like proteases (PLpro), main protease (3CLpro), helicase, and RNA-dependent RNA polymerase (RdRp) are the considered to be the potential molecular targets and are involved in important functional activities such as protein synthesis, RNA transcription, translation, replication, processing, and infection respectively. All these proteins come under non-structural proteins (nsps) for coronaviruses []. The below sections discuss the important protein targets from a structural perspective.</p><sec id=\"Sec19\"><p id=\"Par41\">The papain-like proteases have a “thumb-palm-fingers” structure architecture that houses domains and subdomains. The active site residues/binding pockets present in these domains are His272, Cys111, and Asp286 which form the catalytic triad of the same protease/protein. These papain-like proteases (PLpro) share their structure with the ubiquitin-specific proteases (USPs) which have two main domains, ubiquitin-like (Ubl) domain, an N-terminal domain along with the catalytic triad which forms thumb-palm-fingers architecture [].</p></sec><sec id=\"Sec20\"><p id=\"Par42\">RNA-dependent DNA polymerase is mainly involved in the replication and transcription of the virus. Since it plays an important role in viral infection and hence it is considered a potential molecular target of the SARS-CoV-2 virus, human telomerase reverse transcriptase (hTERT) was first thought to be RNA-dependent DNA polymerase but it is DNA polymerase that shares structural similarities with RdRp. Single-stranded RNA (ssRNA) encodes for RdRp proteins whose structure has a right-handed closed polymerase-like structure with the thumb, finger, and palm domains. At N-terminal, there is the presence of beta-hairpin nidovirus RdRp-associated nucleotidyltransferase (NiRAN) domain (residues ranging from D60 to R249) and C-terminal/polymerase domain (residues ranging from S367 to F920) which houses thumb-fingers-palm subdomains. There are seven motifs present in the RdRp protein of SARS-CoV 2 ranging from A to G, where five are said to be present in the most conserved domain, i.e., the thumb domain [–].</p></sec><sec id=\"Sec21\"><p id=\"Par43\">Non-structural protein 5 coding gene encodes the main protease (3CLpro) which is mainly involved in the proteolytic activity of viral polyproteins and replicates. They share structural similarities between the 3C proteases of picornaviruses in terms of core homology as well as substrate specificities. The size of the protein is approximately 30 kDa which has three domains rich in cysteine amino acids. The protease comprises chymotrypsin-like folds along with a catalytic triad and a domain that is inundated with cysteine [].</p></sec><sec id=\"Sec22\"><p id=\"Par44\">The nucleoprotein (N) of SARS CoV-2 binds to the genomic RNA (gRNA) to form the ribonucleoprotein (RNP) complex, a major virion core of the coronavirus. Nucleoprotein along with other proteins plays an important role replication and transcription of the virus. The N-terminal and the C-terminal domain of the nucleocapsid protein share similarities with CoVs N proteins. The N-terminal domain is a right-handed fist-shaped structure that has the topology of five beta strands sandwiched between two alpha-helices. The C-terminal domain is a homodimer with two protomers resembling a rectangular slab-like structure. Each protomer comprises two 3–10 helices, two beta-strands, and five alpha-helices respectively. The beta hairpin-like structure connects two protomers by inserting itself into their cavities [].</p></sec><sec id=\"Sec23\"><p id=\"Par45\">There are three domains in the enveloped protein of the SARS-CoV-2 virus which comprise of N-terminal domain, C-terminal domain, and transmembrane domain. The overall size of the E protein is about 8.4–12 kDa and forms the integral membrane protein of the virus comprising 76–109 amino acids. There are about 7–12 amino acids at the N-terminal which is short and makes it hydrophilic in nature. The transmembrane domain has about 25 amino acids, out of which two are non-polar and neutral making it hydrophobic in nature. The C-terminal domain forms the majority of the protein structure [].</p></sec><sec id=\"Sec24\"><p id=\"Par46\">The genome of the SARS-CoV-2 encodes for the ectodomain trimeric protein which projects out of mature virions and plays an important role in the initial attachment process and entry into the host organism. The spike glycoproteins in SARS-CoV-2 are strongly glycosylated and are trimeric in structure. The protomers present in the spike glycoprotein structure contain approximately 1260 amino acids. The first and foremost part of the spike glycoprotein is the surface subunit (S1) which comprises four domains (two SD1, SD2 subdomains, N-terminal domain, and C-terminal domain which is also the receptor-binding domain of the protein) with 672 amino acids in length. The transmembrane subunit S2 is made up of 588 amino acids and contains a transmembrane domain, cytoplasmic tail, and two heptad repeats [].</p></sec><sec id=\"Sec25\"><p id=\"Par47\">The non-structural proteins of the SARS-CoV-2 are encoded by the open reading frame 1a and ab (ORF1ab) which is present in the downstream region of the 5’ end of the ssRNA (single-stranded ribonucleic acid) in the SARS-CoV-2 virus. They are polyproteins pp1a and pp1ab. Nsp 1 to 11 is categorized under pp1 polyprotein, while Nsp 12 to 16 is classified under polyprotein pp1ab [].</p><p id=\"Par48\">These structural details are of utmost essential when the proteins interact with the drug candidates, to understand the precise molecular mechanism of action of the small molecule.</p></sec></sec><sec id=\"Sec26\"><p id=\"Par49\">Mutations in the spike received early attention and some of the residues in spike variations influenced the polyclonal antibody recognition, thereby causing immune escape. Studies have shown that among all the receptor-binding domains (RBD) residues for which mutations in the form of substitutions affected the recognition by certain convalescent serums, E484 was identified by deep mutation scanning as having importance, with changes to amino acids K, P, or Q []. Moreover, the mutation E484K is recognized as an escape mutation, that occurs when exposure to monoclonal antibodies C144 and C121 occurs []. Moreover, E484K was the only variable that was found to reduce the neutralization capacity of MAbs REGN10934 and REGN10989 []. It was also studied that mutations K444E, L452R, G446V, and F490S also escaped some of the convalescent sera, thereby making these variants of SARS-Cov-2 stronger and more robust for survival in the human body []. Similarly, the substitution N439K enhances the SARS-CoV-2 ACE2 receptor affinity []. This mutation is also identified to decrease the potential of neutralization of the plasma. However, other deep mutation scanning studies pose contradictory results, stating that N439K did not affect the impact of the polyclonal antibodies used []. This discrepancy could be because immune escape caused by this mutation is via ACE2 affinity than by direct antibody epitope identification, making it a variant that should compel more studies.</p><p id=\"Par50\">Studies have also identified that the n-terminal domain (NTD) mutations in spike also cause immune escape []. Studies have shown that mutations K150R, K150T, K150E, S151P, K150Q, and N148S were escape mutations []. Moreover, deletions in the NTD have been noted to be the cause of SARS-CoV-2 evolution repeatedly, by altering the antigenicity of the NTDs []. Another study noted that Δ140 mutant in the spike eventually acquired the E484K variation, thus directly resulting in escaping the antibody responses [].</p><p id=\"Par51\">Our previous studies show that mutations in the spike glycoprotein demonstrated enhanced stability in the protein, and the residues that were undergoing mutations were also part of the hydroxychloroquine drug-binding active pockets []. The mutations that occurred in other prioritized proteins of interest evidenced a decrease in the stability of the protein itself. These mutations were also part of the drug-binding active sites of remdesivir, favipiravir, and dexamethasone. The study also concluded that the presence of mutations in various regions reduced the effect of remdesivir on the targets, thereby allowing the potential drugs to not work on the targets of interest []. This shows the presence and effect the variants have on the drug-binding sites.</p><p id=\"Par52\">All these studies prove that there is distinct evidence as to the altering of the SARS-CoV-2 proteins’ structural antigenicity, causing them to evade the immune system and become more robust. As of now, evidence suggests the changes that mutations cause in the structure of spike proteins, thereby influencing the binding of small molecules. More research is required to assess the precise mechanism of how various structural and non-structural proteins are being affected by the variants. Amino acid deletions and substitutions in the spike affect the antibodies which may also lead to an increase in the global-viral resistance. A greater understanding of the viral genome sequence data and the impact of variants on the drug-binding sites will be essential to better obtain repurposed drugs.</p><sec id=\"Sec27\"><p id=\"Par53\">Currently, as we have seen from various evidences, there are several repurposed drugs reported for use against COVID-19, some already available on the market. The potential binding of several FDA-approved drugs has been studied previously; however, not much data is revealed on the precise molecular basis of these interactions against SARS-CoV-2 targets. In the previous section, evidence of how the variants alter the drug binding efficacy and how the virus escapes the immune system is provided. In our previous study, however, various targets of SARS-CoV-2 were analyzed at the molecular level to comprehend the changes it undergoes during the binding event of some of the important FDA-approved drugs for COVID-19. Favipiravir, lopinavir, chloroquine, hydroxychloroquine, lopinavir, ritonavir, and remdesivir were studied. These drugs have also been mentioned in other computational biological studies as well []. Though there are limited experimental data to highlight the spike glycoprotein of SARS-CoV-2 act as the target for repurposed drugs, however, several computational biology prediction revelated the scope of utilising spike glycoproteins as the putative molecular target for repurposed drugs. A recent study stated that drugs ritonavir and eriodictoyl bind to the active sites of spike protein of SARS-CoV-2 with potential binding energy and stability []. Both these drugs interacted with the binding energy of −7.6 kcal/mol, indicating stable binding. Likewise, another study showed that repurposed drugs such as pralatrexate, carumonam, aclerasteride, and granotapide interacted with spike protein of SARS-CoV-2 and acted as potential binders []. Additionally, another recent study has reported Remdesivir and ivermectin, potential repurposed drugs interacted with the spike protein of SARS-CoV-2 []. The study stated that ivermectin bound effectively to the spike protein of the virus, and inhibited the entry of SARS-CoV-2 into the host. Remdesivir was also suggested to be involved in stable interaction with the spike target, apart from the other targets. These studies demonstrate that repurposed drugs were suggested to be targeted the spike protein at various binding pockets and interacted with the spike glycoprotein. To understand further on the interaction modelling of some important repurposed drugs, molecular docking and molecular dynamic studies were performed in our previous study and the results showed ritonavir and lopinavir with better binding energy, followed by remdesivir against the prioritized targets [].</p><p id=\"Par54\">At a molecular level, Fig.  showed that chloroquine interacted with amino acids Leu6820, Leu6819, Met6818, Val7021, Tyr7020, Asp7018, and Asp6942 of the spike glycoprotein. Leu331, Tyr333, Arg277, Phe274, Val270, Trp301, Thr296, Ala264, Arg262, Phe314, Gln260, and Arg259 were the major amino acids involved when spike glycoprotein interacted with Lopinavir []. Likewise, Tyr421, Asn422, Pro491, Leu492, Gln493, Ser494, Tyr495, Gly496, Phe497, Gln498, Lys444, Asp442, Tyr351, Val350, Ser349, Ala348, Phe347, Arg346, Arg509, and Val401 were involved in complex favipiravir-nucleocapsid phosphoprotein. Lys6822, Glu6821, Leu6820, Val7021, Tyr7020, Asp6942, Asn6941, Glu6940, and Thr6938 were observed for complex remdesivir-nucleocapsid phosphoprotein. NSP16 interacted with hydroxychloroquine at residues Arg6817, Met6818, Leu6819, Leu6820, Tyr7020, Val7021, Asp7018, and Asp6942 and with ritonavir at residues Phe314, Gln260, Arg259, Arg277, Phe274, Tyr333, Trp330, Trp301, Thr282, Gly284, Phe286, Leu291, and Val270 [] (Fig. ).</p><p id=\"Par55\">Bearing in mind the changes that the drug binding sites undergo if variations occur as already evidenced in the previous sections, it may also be hinted that the residues at the interaction sites also change, rendering a different molecular basis in case of variants. More comprehensive studies are essential to further confirm this.</p></sec></sec><sec id=\"Sec28\"><p id=\"Par56\">The major repurposed drugs displayed in Table  dhow that research is ongoing and always improving for tackling the problem of COVID-19. Drugs such as lopinavir, ritonavir, remdesivir, ribavirin, and favipiravir are some repurposed drugs. Lopinavir/ritonavir was generally used against HIV [], remdesivir against major SARS viruses [], and favipiravir which was previously used as a general viral pro-drug []. With the advent of research techniques to identify more repurposed drugs, several other candidates have come up, evidencing the successful implementation of repositioning.</p><p id=\"Par57\">The potential of in silico studies has been documented for ages, especially in the field of drug discovery. In this case, repurposing beings with computational bioinformatics work, with much exploration into chem-informatics, chemo-genomics, and computer science aspects such as machine learning, deep learning, artificial intelligence, computational biology, structural biology, and genomics/proteomics. A thorough analysis of the chemical structures of drug candidates helps in the identification of its working mechanism against specific targets. Using computational biology and computer science algorithms will help in predicting the possible and potent drug candidates in silico, a work that might otherwise take years without computational studies. The best drug candidates working against the protein targets of interest causing a specific disease will then be tested experimentally in vitro to validate the predictions. Accurate laboratory confirmation narrows down the best working drug that could be taken forward for in vivo animal model studies, before clinical trials. For example, remdesivir a broad-spectrum antiviral came into the market to be repurposed as a COVID-19 drug after in vitro studies revealed that this drug inhibited the SARS-CoV and MERS-CoV in the human respiratory epithelial cells [] initially, followed by inhibiting the epithelial cells in SARS-CoV-2 [], the pathogenic strain of COVID-19 disease. Likewise, the in vivo study of remdesivir when infected with SARS-CoV-2 infected rhesus monkey demonstrated an improvement in the pulmonary lesions and decreased the viral load after 7 days of treatment with the drug. No coronaviral disease was observed after the successful use of this drug []. Thus, repurposed drugs follow the path of in silico to in vitro to the in vivo platform before they are successfully used.</p><sec id=\"Sec29\"><p id=\"Par58\">There are several FDA-approved drugs/drugs in the clinical phase, their original targets, and the repurposed targets (Table ). Remdesivir was originally meant for treating the Ebola virus against the Ebola RdRp target, and now with potential repurposing capabilities against the RdRp and NSP12 of SARS-Cov-2 [, ]. Lopinavir/ritonavir is a drug for the HIV-1 protease, that now evidences potential for the main protease of SARS-CoV-2 []. Silmitasertib has its original target as casein kinase 2, which inhibits the casein kinase 2 pathway in Sonic hedgehog (SHH). This drug is also known to inhibit the casein kinase 2 pathway in SARS-CoV-2 []. Likewise, dipyridamole 100 mg is an inhibitor of phosphodiesterase originally and works against the main protease of SARS-CoV-2 and stops replication []. Leronlimab 700 mg works against the C–C Chemokine receptor type 5 in HIV and now shows potential against the same target in SARS-CoV-2 [, ]. Ruconest targets the c1 esterase and inhibits angioedema, while it targets C1, C5a, and C3a esterases in SARS-CoV-2 []. Infliximab attacks the tumor necrosis factor in SARS-CoV-2 as well as in its original target in Rheumatoid arthritis, Crohn’s disease, and ankylosing spondylitis []. This data is also available at  [] and the same is elucidated in Table .\n",
              "  </p></sec></sec><sec id=\"Sec30\"><p id=\"Par59\">With all the important approaches and scope discussed for repurposing drugs, some important advantages can be classified into three main focuses:<list list-type=\"alpha-lower\"><list-item><p id=\"Par60\">Translational: drug repurposing directly benefits translational research by fostering major collaborations, essential for scientists to identify important repurposed drugs. Bridging this gap is one major advantage of drug repositioning research.</p></list-item><list-item><p id=\"Par61\">Target: drug repurposing research aids in understanding thoroughly the pathways associated with specific targets and how the same can be exploited to benefit patients who must be treated. The comprehension of the complete metabolic pathways for specific targets opens up avenues for remaining drug discovery research.</p></list-item><list-item><p id=\"Par62\">Disease: drug repurposing has huge merits in discovering drugs for new, emerging diseases when there are no other treatment approaches yet discovered. The emergence of SARS-CoV-2 has illuminated the purpose and benefits of repositioning drugs to provide emergency treatments.</p></list-item></list></p><p id=\"Par63\">Other advantages of repurposed drugs include:<list list-type=\"bullet\"><list-item><p id=\"Par64\">Lower cost of production</p></list-item><list-item><p id=\"Par65\">Lesser time for identifying useful, effective, and safe drugs for quick use</p></list-item><list-item><p id=\"Par66\">Research can be carried out using the existing, easily accessible bioinformatics and chem-informatics data and resources available for specific diseases, as was the case in COVID-19 cases</p></list-item></list></p><sec id=\"Sec31\"><p id=\"Par67\">\n",
              "  <list list-type=\"alpha-lower\"><list-item><p id=\"Par68\">Molecular docking: The emergence and prevalence of COVID-19 disease have paved the way for molecular docking studies. Previously, molecular docking has been used to determine the potential anti-virals for curbing the spread of the disease. Poly-ADP-ribose polymerase-1, CVL218, a potent inhibitor that binds to the N-terminal of the nucleocapsid protein of the SARS-CoV-2 was identified [].</p></list-item><list-item><p id=\"Par69\">Signature matching: Drug-drug comparisons and drug-disease comparisons can be carried out using the transcriptomic signature patterns. This technique has been previously used for repositioning, however, for lymphoid malignancies []. The use of signature matching for COVID-19 remains an unexplored avenue.</p></list-item><list-item><p id=\"Par70\">Pathway mapping: Identification of potential drugs using this technique has been previously carried out to analyze the network of drugs and diseases. Respiratory viral infections have been previously studied for repurposing candidates and a phosphodiesterase inhibitor was identified in the process [].</p></list-item><list-item><p id=\"Par71\">Genome-wide association studies: Studying the small variations (SNPs) of the genome of interest helps in identifying traits associated with specific diseases. Reports have stated the use of GWAS in combination with other computational approaches for target prediction in drug repositioning [, ].</p></list-item></list></p><p id=\"Par72\">Thus, despite there being several advantages of repurposing, there are some limitations as elucidated in the following sections.</p></sec></sec><sec id=\"Sec32\"><p id=\"Par73\">With so many repurposed drugs in the market today, and several more upcoming owing to the intensity and acceleration of COVID-19 research, there are more advantages than disadvantages to the use of such drugs. However, some limitations to the drug repositioning, include:<list list-type=\"bullet\"><list-item><p id=\"Par74\">Weak intellectual property rights protection for repurposed drugs due to reduced rate of return</p></list-item><list-item><p id=\"Par75\">The molecular docking technique used for identifying drugs for repurposing cannot be applied to all existing drugs due to the larger computational operations</p></list-item><list-item><p id=\"Par76\">Some network-based approaches used for drug repurposing do not always yield promising candidates</p></list-item><list-item><p id=\"Par77\">These approaches are data-dependent, due to which the results may be inaccurate due to the lower volume of existing COVID-19 data</p></list-item><list-item><p id=\"Par78\">Lack of strong, solid evidence for the mechanism of action of repurposed drugs on COVID-19 19</p></list-item><list-item><p id=\"Par79\">Experimental evidence is required for validating the predictive outcomes</p></list-item><list-item><p id=\"Par80\">Overall low success rates in new disease settings</p></list-item><list-item><p id=\"Par81\">Extensive data to be studied before repurposing</p></list-item><list-item><p id=\"Par82\">Dosing and safety issues, i.e., approved range of dosage and delivery capabilities are not very strong</p></list-item><list-item><p id=\"Par83\">Accurate, robust toxicological studies of repurposed drugs essential before delivery and testing</p></list-item><list-item><p id=\"Par84\">Rare to find unique and novel drug-protein interactions all within the restraints of the approved therapeutic window [].</p></list-item></list></p></sec><sec id=\"Sec33\"><p id=\"Par85\">With drug repurposing emerging as one of the most important research strategies for treating the occurrence of sudden, new diseases, there arrives the question of its efficacy when compared to existing viral vaccines or other small molecules, both considered standard treatment options. However, using high-throughput screening techniques and various other predictive approaches for identifying drugs for repositioning will help maintain the genomic integrity, thereby resolving the issue of efficacy. These approaches when amalgamated with advanced computational techniques such as machine learning will pave the way for producing better-repurposed drugs. In the wake of the emergence of highly mutated and resistant SARS-CoV-2 strains, the question of how already repurposed drugs work comes into the picture. But with improvements in technologies that can help repurposed drugs function better against such mutations, the future for drug repurposing looks bright.</p><p id=\"Par86\">Another perspective is the use of drug repurposing as a therapy for novel diseases for which specific drugs are not yet available, COVID-19 being one such important example. Repositioning played a major role in this case and is still in the research arena where different drugs are being explored using advanced computational techniques. Other advanced techniques are also being researched that will provide accurate predictive drug repositioning results that are more reliable and can be used for a variety of diseases. Although this review comprehensively lists all the current and latest drugs repurposed for SARS-CoV-2 and the existing computational approaches that in turn improve the process of repurposing, further advancements in the software and techniques such as using ensemble machine learning will improve the predictive process of repositioning.</p><p id=\"Par87\">A dimension that still requires improvement is the patent protection issues that still do not provide full protection to the repurposed drugs. If these avenues are improved, then the future of drug repurposing appears bright.</p></sec><sec id=\"Sec34\"><p id=\"Par88\">With high mortality rates, COVID-19 has become one of the most prevailing diseases globally. With the CDC and WHO declaring several recent variants of concern for SARS-CoV-2, there is an urgent need to explore alternative options of treatment when there are no specific drugs targeting the disease. One such alternative is drug repurposing, where the use of existing, FDA-approved drugs with similar functions and mechanisms of action against the same family of viruses can be employed. This concept relies heavily on robust preliminary in silico work using advanced and latest techniques in computational biology and chem-bioinformatics. The use of the machine and deep learning algorithms are also a way to obtain a better prediction of repurposed drugs. This review sheds light on the various existing drugs that have been repurposed for use against SARS-CoV-2 and also illuminates the important drug targets against which these drugs are used: spike glycoproteins, RDRP, main protease, papain-like protease, nucleoproteins, non-structural proteins, and enveloped proteins of SARS-CoV-2. The most important advantage of repositioning drugs against COVID-19 is the reduced cost and time of development and the drawback is the absence of safety/toxicity profiles via in vivo trials for the disease. Despite this, repurposing has found its niche with advancements in all related interdisciplinary fields of science and acts as a major research avenue in the current pandemic.\n",
              "  </p></sec></body>],\n",
              " [<body><sec id=\"sec1-biology-11-01827\" sec-type=\"intro\"><p>The 2019 novel coronavirus, now dubbed SARS-CoV-2, has led to a global pandemic as declared by the World Health Organization (WHO) on 11 March 2020. Many studies have described the complex immune response associated with viral infection, leading to the identification of several clinical and immunological features [,]. It shares its mechanism of viral entry with other viruses of the <italic toggle=\"yes\">Coronaviridae</italic> family as it is mediated by the spike (S) glycoprotein which binds to angiotensin converting enzyme 2 (ACE2) receptors that are localized in a variety of cell types, such as in the heart, liver, kidney, but most abundantly in the lungs and respiratory system (i.e., alveolar epithelial cells and capillary endothelial cells), leading to a wide range of various symptoms experienced by COVID-19 patients []. Despite this variance, it is of note that SARS-CoV-2 infections lead to systemic inflammatory response syndrome (SIRS) and acute respiratory distress syndrome (ARDS) [], two relevant conditions highly correlated with COVID-19 infections []. Specifically, COVID-19 infections are accompanied by an overaggressive inflammatory response corresponding to a significant release of pro-inflammatory cytokines known as a cytokine storm. Though the biological relevance in relation to COVID-19 of cytokine storms has been questioned [], studies have shown significant elevations of pro-inflammatory cytokines in COVID-19 infections as compared to other viral infections [,], highlighting that virulence may be more immune response-mediated rather than solely due to the viral titer. As the mechanisms of the cytokine storm have yet to be fully elucidated, there is a need to understand its genomic origin by comparing it known conditions that occur during COVID-19, such as SARS and ARDS.</p><p>With the advancement of high-throughput techniques, transcriptional profiling has proven to be an effective approach to elucidating biomarkers and drug targets, as well as identifying underlying processes and functions specific to certain diseases. In the case of COVID-19, studies have most often utilized differential gene analysis and limited research is available on clusters of highly correlated genes. Assessing the differences in gene expression between experimental and control conditions often assumes that expression levels change independently of each other, which is not the case in many biological systems in which complexity arises from gene-protein interactions as seen from []. Thus, a complementary approach is to study gene-gene relationships. Weighted gene co-expression network analysis (WGCNA) is a bioinformatics method for exploring the relationships between different gene sets (i.e., modules) by describing the correlation patterns between genes across multiple microarray datasets []. This method makes use of clustering techniques and computational analysis of module eigengenes in order to relate modules to one another. Literature shows that it has successfully been applied to construct gene co-expression networks in various diseases, most notably in cancer, to identify novel biomarkers and therapeutic targets []. In the case of COVID-19, where the variance in symptoms and the unclear mechanisms of the cytokine storm led to challenges in developing therapeutics to treat affected patients, the need arises to solve these challenges through relation of the COVID-19 induced cytokine storm to similar ailment (SIRS, ARDS, and sepsis), to connect their mechanisms and therefore elucidate potential novel drugs.</p><p>Adding on to this, new drug discovery is faced with the limitation of substantial costs, changing regulatory requirements, and above all, length of development and clinical testing, prior to getting government approval, making it a less desirable choice in the pharmaceutical industry. Drug repurposing (i.e., drug repositioning) is one good avenue where existing clinical drugs are applied to other diseases through relating their effects on gene expression by using bioinformatic prediction and machine learning. Compared to the development of novel drugs, drug repurposing could accelerate the translation of information regarding human diseases into therapeutic advances, all the while bypassing the lengthy and costly drug development process. Omics data, in particular, provides a useful way of relating genetics to pharmacology. The Connectivity Map (CMap) was developed as a database of over 1,000,000 expression profiles, genes, drugs, and disease states allowing connection of these variables by common gene expression signatures []. It has allowed for the repurposing of drugs based on gene expression values in neurogenerative diseases, cancer, and COVID-19 []. An integration of drug repurposing with systems biology is through module preservation analysis to identify module hub genes preserved across different diseases such as SIRS, sepsis, burns, and severe trauma []. These module hub genes serve as inputs into Connectivity Map analysis to identify repurposed drugs. </p><p>Herein, a bioinformatics and systems biology approach are adopted to study the similarities and differences between the whole blood transcriptome in ARDS, SIRS, sepsis, and COVID-19, to highlight possible reusable (i.e., repurposed) drugs. Microarray datasets (GSE147902, GSE66890, GSE74224, GSE152641) from ARDS, SIRS, sepsis, and COVID-19 whole blood transcriptomes were taken from the GEO database and analyzed using WGCNA through hierarchal clustering to identify significantly co-expressed modules or clusters of co-expressed genes across the aforementioned conditions. These ailments have been widely reported to be symptoms of COVID-19 infections, and often occur in conjunction with the so-called cytokine storm [,]. Although the transcriptomics of COVID-19 have been widely studied [,,], the specific mechanisms of cytokine storms have yet to be fully understood. The similarities between ARDS, SIRS, sepsis, and COVID-19 were analyzed at the module level using module preservation analysis and functional and pathway enrichment were used to understand their biological significance. Only the genetic similarities between the diseases were considered in this study, and temporal effects or variations were not accounted for. It is also important to note that the findings in this study are based on generated gene co-expression and PPI networks built from relating immune illnesses, and hence, offer insights into repurposed drugs to treat post-COVID-19 effects (i.e., cytokine storms). The hub genes within the highly preserved modules across the four phenotypes were determined and connectivity map analysis and database screening were used to identify possible candidate repurposed drugs.</p></sec><sec id=\"sec2-biology-11-01827\"><sec id=\"sec2dot1-biology-11-01827\"><sec><p>Microarray data was sourced from the NCBI GEO [<uri xlink:href=\"http://www.ncbi.nlm.nih.gov/geo\">http://www.ncbi.nlm.nih.gov/geo</uri> (accessed on 24 January 2022)] database, a public repository of high throughput genomics data. A total of four datasets were used and analyzed in this study.</p><p>ARDS microarray data was taken from GSE147902 which contains the blood transcriptome of 96 patients within 24 h of ARDS onset [], and from GSE66890, which contains the blood transcriptome of critically-ill patients with sepsis and sepsis-related ARDS []. SIRS and Sepsis microarray data were taken from GSE74224 which contains the blood transcriptome of 74 sepsis and 31 SIRS patients []. Finally, COVID-19 microarray data was taken from GSE177477 which contains the blood transcriptome of 29 COVID-19 patients and 18 healthy controls []. Probe identifiers of each dataset were converted to their corresponding gene symbols based on the respective probe lists of the platforms used obtained from the Affymetrix website. [<uri xlink:href=\"http://www.affymetrix.com/\">http://www.affymetrix.com/</uri> (accessed on 24 January 2022)].  presents the summary of the characteristics of the datasets used in this study.</p></sec></sec><sec id=\"sec2dot2-biology-11-01827\"><sec id=\"sec2dot2dot1-biology-11-01827\"><p>Raw data from the GEO datasets were pre-processed and normalized using the Robust Multichip Average algorithm in the “affy” R package within Bioconductor [<uri xlink:href=\"https://www.bioconductor.org/packages/release/bioc/html/affy.html\">https://www.bioconductor.org/packages/release/bioc/html/affy.html</uri> (accessed on 24 January 2022)]. Probes corresponding to controls were removed from the datasets. For the sepsis dataset (GSE66890), five of the datasets marked as excluded were removed from the analysis as per the source study []. The goodSampleGenes function of WGCNA was also utilized to filter out any genes and samples with too many missing values []. Preliminary clustering is also performed to check for any outliers. To minimize memory requirements, probe set level expression values were filtered using the coefficient of variation (CV) wherein only the highest CV probe for genes with multiple probes was retained. Additional filtering was done by only considering genes with average expression intensities of greater than 100; by removing data with lowest 20% percentile mean expression value shared by all samples; and by removing data with lowest 20% percentile variance across samples. Rows with empty gene symbols were removed, only the row with the largest mean expression value was kept when gene symbols are duplicated. Also, prior to network construction, expression values were log<sub>2</sub> normalized.</p></sec><sec id=\"sec2dot2dot2-biology-11-01827\"><p>In module preservation analysis, one dataset serves as the reference upon which other expression data is compared. In this study, the ARDS dataset with 96 samples was used as the reference and was therefore subjected to WGCNA. Co-expression network construction and consequent module identification was performed using the WGCNA R package according to literature []. Briefly, pre-processed data was input into R. Soft thresholding power was determined using the pickSoftThreshold function which analyzes the network topology to find the best power from a range of 1–20 based on the “scale-free topology criterion” of which many biological systems exhibit. Network type was set to signed, minimum module size was set to 30, and medium sensitivity (deepSplit) were set to 2. This results in an adjacency matrix.</p><p>The adjacency matrix was transformed into a network of Topological Overlaps (TO) which quantifies the co-expression relationships considering all pairs of genes in relation to all other genes in the network. Hierarchical clustering was then applied using 1–TO as the distance measure, resulting in a cluster dendrogram where the branches correspond to gene co-expression modules. This is known as the Dynamic Tree Cutting algorithm as described in []. The resulting groups of genes (i.e., modules) are representative of genes with consistently correlated expression profiles across the sample datasets.</p></sec><sec id=\"sec2dot2dot3-biology-11-01827\"><p>After construction of the weighted gene co-expression network of the ARDS dataset whose modules served as reference, the sepsis, SIRS, and COVID-19 datasets were projected onto the reference to analyze module preservation. This was also done using the WGCNA package through the modulePreservation function []. The parameters for module presentation were set as follows: the number of permutations (nPermutations) was set to 100 and the network type (networkType) was set to signed. The module qualities were also assessed and plotted in R.</p></sec></sec><sec id=\"sec2dot3-biology-11-01827\"><p>Genes within the identified modules of interest were extracted from the network and functional enrichment was performed using the Database for Annotation, Visualization, and Integrated Discovery (DAVID) webserver (accessed on 26 July 2022), which interprets and correlates the function of various genes based on previous literature []. Specifically, functional annotation was performed using the Gene Ontology (GO) database, a repository of the basic functions and characteristics of genes and their protein products, while the Kyoto Encyclopedia of Genes and Genomes (KEGG) database was used to infer information regarding the biological, genomic, chemical, and systemic functions of various genes [,]. Only those genes present with functional significance and those present within the databases were included for the analysis. Results with <italic toggle=\"yes\">p</italic> &lt; 0.05 were considered to be statistically significant when screening for GO terms and KEGG pathways.</p><sec><p>The constructed gene co-expression network was visualized using Cytoscape []. Hub genes within the identified modules were identified through the inter-connectivity value. Briefly, for each identified module, the module eigengene based connectivity (kME) was calculated using the “intramodular Connectivity” function from the WGCNA package []. Genes with the highest intra-modular connectivity values were considered to the hub genes. </p><p>Additionally, to account for non-protein coding genes, the genes of each preserved module were passed through the STRING database to construct a protein-protein interaction (PPI) network. The resulting PPI network was imported into Cytoscape and the hub genes were selected using cytoHubba. Degree, Maximum Neighborhood Component (MNC), Radiality Centrality, Stress Centrality, and Closeness Centrality were the algorithms used to evaluate the identified hub genes. Consistent genes were taken to be hub genes. The top three hub genes were taken for further downstream analysis. </p></sec></sec><sec id=\"sec2dot4-biology-11-01827\"><sec><p>To screen possible repurposed drug candidates that are associated with the preserved modules across ARDS, SIRS, sepsis, and COVID-19, connectivity map analysis was performed using the Connectivity Map online webserver []. Specifically, the top three genes were taken from each of the identified modules preserved in COVID-19. Each of the extracted genes were grouped into either up-regulated or down-regulated. The two groups of genes were submitted to Connectivity Map through the CLUE.io webserver. The significance threshold value was set to 0.01 and only the candidates with negative enrichment scores were considered. Connectivity scores range from +100 to −100 (from completely positive correlation to completely negative). The more negative the correlation, the greater the drug reverses the inputted upregulated genes and as such, means that the drug is able to lessen the expression of the hub genes used in this study. For simplicity, the top five drugs identified were recorded.</p></sec></sec></sec><sec id=\"sec3-biology-11-01827\" sec-type=\"results\"><sec id=\"sec3dot1-biology-11-01827\"><p>Prior to network construction, the datasets were normalized and filtered, where samples with excessive missing values were removed. Clustering was then performed and the results are shown in . No significant outliers were identified based on the generated dendrogram and as such the data was deemed fit for network construction.</p><p>The ARDS dataset was used for the construction of the reference gene co-expression network. The dataset had a total of 96 samples which were considered for construction. All datasets were gene expression profiles obtained from the whole blood samples of individuals within 24 h of ARDS onset. After filtering according to the criteria discussed in the methodology, the log10 normalized expression values of 19,253 genes were used to construct the network.</p><p>Gene co-expression modules were identified by WGCNA. Through checking the independence of the network on the scale, the softPower threshold of 18 was determined to be the most appropriate (, left), as also reflected by the maximum mean connectivity (, right). Setting the softPower threshold to this value led to an approximately scale-free network topology which is a common characteristic for biological networks (, right). The values used to compute for the linearity are presented as a histogram (, left).</p><p>Using the calculated softPower thresholds, a scale-free topology is obtained (see , right). This suggests that the genes under consideration in this study, at least statistically, are biologically relevant and are suitable for co-expression analysis. WGCNA was able to identify a total of 46 gene co-expression modules. These modules may have identified genes with significantly similar with significantly similar expression profiles. Co-expression similarities of the entire modules was quantified through calculation of their eigengenes and clustered based on their correlation, resulting in 13 merged gene co-expression modules from the initial 46. The identified gene modules before and after clustering and merging are shown in the cluster dendrogram in . The grey module represents unassigned genes which were not considered in subsequent downstream analysis.</p><p>Each identified module was assigned a color. The turquoise module had a total of 2803 genes; the magenta module had 410 genes; the light cyan module had 915 genes; the light green module has 265 genes; the light yellow module has 455 genes; the orange module has 123 genes; the sky blue module had 100 genes; the violet module had 94 genes; the dark olive green module had 91 genes; the yellow green module had 87 genes; the floral white module, which was the largest, had 6699 genes; the dark orange module had 120 genes; and finally, the brown module had 39 genes.</p></sec><sec id=\"sec3dot2-biology-11-01827\"><p>To understand the biological relevance of the identified modules, the genes were submitted to the DAVID and gProfiler webservers for functional GO enrichment. The RefSeq and Genbank IDs for each gene were submitted to gProfiler and converted to their Entrez IDs. Genes without a specific entrez ID were labeled with NONE and not considered for functional enrichment as they would not be found in any available database. For simplicity, only the top enriched GO terms were recorded for each module and presented in A–C, for biological process (BP), molecular function (MF), and cellular component (CC), respectively. More detailed information can be found in . The hub genes shown are based on the genes with the highest intra-connectivity score as computed by WGCNA.</p><p>All modules were significantly enriched with a specific BP, MF, and CC. Of relevance to the topic of this paper are the upregulated M9, M18, M28, M32, and M36, all of which were observed to be enriched in biological processes directly linked to the immune response (A–C). They were also observed to be enriched in cellular components related to immune cells, such as M9 being enriched in the T-cell receptor complex. Interestingly, the two largest modules, M44 and M1, were enriched in large-scale macromolecule metabolic processes (2295/5606) and chemical stimulus detection (399/1583) (A), cellular processes which have been observed to be affected by COVID-19. In terms of molecular function, M44 had 4893 genes involved in protein binding and binding. Small modules (those with sizes &lt; 50) were also found to be significantly enriched in a specific BP, MF, and CC. For example, the M46 was found to be enriched in genes involved in spermatogenesis (GO:0007283). However, only a small percentage of the genes in the M46 were found to be involved in the process. This is due to the small size of the original query. Further, modules were also observed to be involved in multiple immune-related molecular functions. M9 genes were observed to be enriched in peptide antigen binding; M28 genes were enriched in antigen binding; and M32 genes were involved in MHC Class II receptor activity. Collectively, the identified modules were significantly enriched in various BPs, MFs, and CCs.</p></sec><sec id=\"sec3dot3-biology-11-01827\"><p>To further assess the biological relevance of the identified modules, the modules were subjected to KEGG Pathway enrichment. The results are visualized in  by taking the most significant KEGG pathway enriched for each module. Detailed results are presented in .</p><p>Results show that not all modules were enriched in a specific KEGG pathway. Specifically, M28 and M46 were not significantly enriched in any KEGG pathway, potentially due to the low size of the modules. Other modules, however, were enriched in varying pathways. For example, M1 was enriched in olfactory transduction, similar to its observed enriched GO term; M9 genes played a role in Th1 and Th2 cell differentiation which is linked to its role in the adaptive immune response and the T-cell receptor complex; and M16 genes were involved in the chemokine signaling pathway; pathways linked to immune processes which suggest the genes’ involvement in ARDS, an inflammatory complication.</p><p>Other modules were also observed to be involved in KEGG pathways linked to the immune response. For example, M18, M33, and M36 were enriched in the NOD-like receptor signaling pathway (KEGG:04621), B-cell receptor signaling pathway (KEGG:hsa04662), and platelet activation (KEGG:04611), respectively. Further, some modules were involved in cell transport and differentiation, such as M44 and M332, which were involved in nucleocytoplasmic transport (KEGG:03013) and hematopoietic cell lineage (KEGG:04640), respectively.</p></sec><sec id=\"sec3dot4-biology-11-01827\"><p>To compare the similarity of gene expression at the module level between the four diseases included in this study (ARDS, SIRS, Sepsis, and COVID-19), the network constructed based on the ARDS dataset was used as reference for module preservation analysis where other datasets describing SIRS, sepsis, and COVID-19 whole blood transcriptome expressions were projected onto the ARDS dataset. The results are summarized in the graph, shown in , with each module labelled according to their assigned color. Additional statistical analysis revealed that preserved modules were all of high quality based on the summary quality statistic () suggesting that all the significantly preserved modules were of statistical significance.</p><p>Module preservation analysis revealed that the ARDS modules were most significantly preserved in the sepsis dataset (B), while only a few modules were strongly preserved in the SIRS and COVID-19 datasets. In SIRS, sky blue (M28) and dark orange (M45) showed no preservation, while orange (M25), violet (M32), dark olive green (M33), and yellow green (M36) showed moderate preservation, and turquoise (M1), magenta (M9), light cyan (M16), light green (M18), light yellow (M19), and floral white (M44) showed strong preservation (A). In sepsis, no modules showed no preservation. Turquoise (M1), sky blue (M28), violet (M32), and dark orange (M45) showed moderate preservation while magenta (M9), light cyan (M16), light green (M18), light yellow (M19), orange (M25), dark olive green (M33), yellow green (M36), and floral white (M44) showed strong preservation (B). Finally, in COVID-19, sky blue (M28) showed no preservation, orange (M25), dark orange (M45), violet (M32), and dark olive green (M33) showed moderate preservation, and turquoise (M1), magenta (M9), light cyan (M16), light green (M18), light yellow (M19), yellow green (M36), and floral white (M44) showed strong preservation (C). Collectively, the modules strongly preserved in COVID-19 were also strongly preserved in other modules and were therefore selected for further downstream analysis.</p></sec><sec id=\"sec3dot5-biology-11-01827\"><p>One of the limitations of WGCNA is that the network generated is undirected, that is, the network established only provides information on gene correlation and not on gene or protein interaction. Considering that most drugs have mechanism of actions that target proteins, the hub genes were selected by submitting the genes in each module preserved in the COVID-19 dataset to the STRING database for PPI network construction. The networks were imported into Cytoscape and cytoHubba was utilized to determine the top three hub genes by taking the three most significant genes as identified by the five algorithms, with the results being presented in . These top three hub genes were all found to be upregulated in their respective modules. The selected genes were selected for determining potential repurposed drugs.</p></sec><sec id=\"sec3dot6-biology-11-01827\"><p>To identify candidate drugs, the hub genes identified in the previous section were submitted for Connectivity Map analysis using the CLUE.io webserver and the top seven genes with the lowest connectivity score were selected as candidate drugs. The identified drugs and their respective functions are summarized in . Only those drugs with connectivity scores lower than −95.00 were statistically significant as per CLUE.io’s documentation. The negative connectivity scores of each drug suggest that the drugs, upon binding or exertion of their mechanism of action, reverse the gene expression of the query, or in other words, the perturbagens identified exhibit downregulation of the identified upregulated hub genes. Results showed that SB-201290 was the drug with the lowest connectivity score, while CAY-10470 had the highest score. It can be observed that all identified drugs served to inhibit several pathways involved in the immune response.</p></sec></sec><sec id=\"sec4-biology-11-01827\" sec-type=\"discussion\"><p>Correlation networks have been growing in popularity for the bioinformatics analysis of gene regulation. Weighted gene co-expression analysis (WGCNA) has been used to apply a systems biology network approach to describe gene correlation patterns across samples. Such an approach has been used to identify candidate biomarkers and therapeutic targets in cancer, mouse and yeast genetics, and the analysis of brain imaging data []. Further, cytokine storms have become recent issue with the rise of COVID-19 infections. Patients suffering from severe cases have been observed to present symptoms of a cytokine storm, described as the acute influx of pro-inflammatory cytokines. Despite the knowledge that cytokine storms are often accompanied by ARDS, SIRS, and sepsis, the relationship between these diseases has yet to be elucidated. Inherently, with the ARDS and SIRS in this present study, a quantitative network-based approach was employed to statistically determine the co-expressed genes across four related diseases: ARDS, SIRS and sepsis, and symptomatic COVID-19, using NCBI GEO datasets to investigate the genetic similarities between the diseases to highlight conserved and divergent pathways in cytokine storms and identify candidate repurposed drugs based on gene co-expression data. Though the study does not consider the temporal nature of the datasets, this module-based analysis is thought to give more consistent results despite the time-series nature of sepsis microarray samples as the function of module is more stable than that of a single gene.</p><p> presents the hub genes screened via intramodular connectivity along with their corresponding functions. Several of these genes have been linked to ARDS and inflammation and consequently, cytokine storms, as highlighted in the table. Of particular interest is TREM-1, which has been previously reported to accentuate lung injury during ARDS and has shown promise as an inflammatory biomarker and as therapeutic target for sepsis and septic shock [,]. It has been reported to function in regulating T-cell proliferation and activation of antigen presenting cells during the antiviral immune response by amplifying toll-like receptor (TLR)-initiated responses against microbial challenges []. Recently, the extracellular cold-inducible RNA-binding protein (eCIRP), which is the endogenous ligand of TREM-1, was found to induce acute lung injury during sepsis and has been shown to induce inflammation in alveolar cells through induction of TREM-1 and activation of cytokine production in macrophages [], further highlighting the protein’s role in stimulating the immune response. Considering its extensive role in regulating the expression of cytokines and that macrophages have been previously linked to symptoms reminiscent of COVID-19, it could be another potential therapeutic target for cytokine storms. These hub genes could be interesting biomarker and therapeutic target candidates for the discussed diseases and warrant further study.</p><p>Preserved gene co-expression modules were further identified via module preservation. Interestingly, two of the largest modules (M1 and M44) were significantly preserved in the COVID-19 dataset. M1 genes were found to be significantly enriched in olfactory transduction and may contain genes affected during COVID-19 infections leading to the loss of smell and taste. M44, on the other hand, was enriched in nucleocytoplasmic transport and its hub gene has been implicated in modulating cytokine production during sepsis. Several immune-related pathways were also significantly preserved in the COVID-19 dataset alluding to potential conserved mechanisms across the four diseases. M9 was enriched in Th1 and Th2 cell differentiation. T-helper (Th) cells regulate and produce IFN- and TNF-, which are macrophage activators and are responsible for cell-mediated immunity. These synergism of these two cytokines has been previously shown to be sufficient in inducing cell and tissue damage reminiscent of COVID-19 symptoms []. More recently, an imbalance in the expression of Th1/Th2 has been associated with disease severity and outcome. Under normal conditions Th1 cells can control the infection sufficiently and has even been linked with a favorable outcome []. However, in severe cases, imbalances such as in cytokine storms lead to Th2 cell priming that results in poor prognosis []. Thus, M9 is potentially a significant module related to COVID-19 disease severity brought about by cytokine storms.</p><p>Chemokine signaling pathways were also found to be significantly preserved in symptomatic COVID-19. This pathway is initiated by ligand-receptor binding in several chemokine receptors and is responsible for activation of several downstream signaling events such as the MAPK pathway, which has been implicated in cytokine storms []. Several of the chemokine receptors (e.g., CXCR3 and CXCR4) have also been shown to be upregulated during cytokine storm-like symptoms as upon binding of their natural ligands (e.g., CXCL12), inflammation is promoted and chemotaxis induces apoptosis potentially leading to significant tissue damage as seen in COVID-19 cases []. Considering also that cytokine storms begin with the accumulation of pro-inflammatory cells in sites of severe infection, chemokines are an important driver for inflammation as they serve as chemoattractants. Their overexpression could be a possible mechanism for observed lung tissue damage (i.e., ARDS) in COVID-19 cases. This further emphasizes the importance of chemokine signaling in the movement of immune cells during a cytokine storm and serves as additional support for chemokine receptors to be potential druggable targets. Another pathway conserved in symptomatic COVID-19 is the NOD-like receptor signaling pathway. NOD-like receptors (NLRs) are essential for detecting pathogen-specific molecules and upon sensing infections, activates signaling pathways that drive NF-B-dependent expression of pro-inflammatory cytokines.</p><p>Moving on, one of the main limitations of this study is that WGCNA generates an undirected network–that is, it simply presents gene correlation quantitatively based on statistical analysis and does not necessarily reflect any possible gene or protein interactions between genes. Hub genes identified using intramodular connectivity may not necessarily be correlated in terms of interaction, and merely serve as a basis of comparison herein. As the main goal of this paper is to be able to identify candidate repurposed drugs, to circumvent this limitation, and considering that drugs typically act on protein targets, this study submitted the identified gene co-expression modules to STRINGdb for PPI network construction to extract such interactions and to identify hub genes based on topological connectivity. Results suggested that the identified co-expressed genes were also significantly related in terms of biochemical interactions as the PPI networks generated were of high significance (<italic toggle=\"yes\">p</italic> &lt; 0.05). This approach allowed identification of several relevant hub genes as shown in . Some hub genes (e.g., STAT1, STAT3) were present in both methods of selection, suggesting at least partial similarity between methods. These genes, similar to the hubs identified by screening intramodular connectivity, have been closely linked with inflammation and potentially, cytokine storms []. Interestingly, the STRINGdb approach was able to identify several biologically relevant proteins such as TLR4 and IRF1 which have both been linked to cytokine storms with the former being responsible for induction of cytokine production and the latter having been shown to alleviate cytokine storm-like symptoms upon its downregulation in mice models []. The resulting PPI networks were also shown to be biologically significant statistically and be highly connected. Several of the identified pathways in the other method also remained conserved even after database screening indicating that the primary genes enriched in specific pathways remain present in the module networks. These results suggest that utilizing PPI network construction prior to CMap analysis is a good way to narrow down and specify results without loss of significant information especially when screening for possible repurposed drugs.</p><p>Another growing avenue for treating various diseases is the repurposing of already-established drugs otherwise known as drug repositioning. This study was able to utilize preserved gene modules in COVID-19 to identify such drugs. The identified drugs (see ) were all found to target immune-related pathways. The MAPK pathway has been a promising target for novel therapeutic approaches against COVID-19 as the loss of ACE2 activity brought about by viral infection leads to overexpression of p38, which leads to inflammation []. COX-inhibitors are common in treating inflammation and has been used for treating ARDS []. Antihistamines alleviate symptoms of allergies through inhibition of histamine receptors which have been theorized to induce TNF--stimulated accumulation of pro-inflammatory cytokines []. Further, a study that treated 22,560 COVID-19 patients with histamine antagonists (e.g., loratadine) resulted in attenuation of cytokine release leading to moderate to significant relief of COVID-19 symptoms, suggesting the potential of antihistamines in treating COVID-19 and cytokine storms in general []. Additionally, IKK<italic toggle=\"yes\">β</italic> is a protein responsible for transducing pro-inflammatory signals leading to its anti-inflammatory properties. Recently, a study has shown that nanosomal delivery of TPCA-1 to porcine chondrocytes can prevent TNF- stimulation thereby lowering inflammation []. Several of the drugs identified were also found to inhibit the NF-B pathway which is central to cytokine storms [,,]. Another paper has also shown that targeted delivery of TPCA-1 using platelet-derived extracellular vesicles is able to “calm” the cytokine storm in pneumonia mice models through inhibition of pro-inflammatory factors []. This suggests that TPCA-1 could be a good candidate drug for treatment of cytokine storms in COVID-19. Mepacrine is a known anti-malarial drug that has since been discovered to have inhibitory activity against NF-B and is now used as an NSAID to inhibit cytokine production. Previous studies have shown that it is effective in alleviating airway inflammation in murine models of asthma []. Such inflammation is also present in COVID-19 patients and thus points to its potential as a therapeutic. Recently, antimalarials have also shown some promising results for being repurposed to tackle COVID-19 cytokine storms and the resulting neuroinflammatory symptoms [].</p><p>The identified compounds were all found to have some significance with regards to attenuating or reducing inflammatory pathways that have been previously linked to cytokine storms []. Some of the identified drugs have also already been utilized to treat diseases unrelated to cytokine storms, thus indicating the drug repurposing is a promising avenue for drug discovery. Combining gene expression data and a network-based systems biology approach is a good pathway for transcriptome-driven drug discovery. The results of this study suggest the importance of considering gene expression when designing drugs or active compounds and may offer insights to the repositioning of drugs. Future studies can pursue validation of the results of this study to establish the viability of this approach as well as to identify potential effective therapeutics against cytokine storms.</p></sec><sec id=\"sec5-biology-11-01827\" sec-type=\"conclusions\"><p>Cytokine storms have been observed to lead to ARDS, sepsis, and SIRS, which may indicate specific shared genetic mechanisms. Herein, a network-based approach was applied to correlate the gene expression patterns between the aforementioned diseases by using WGCNA and module preservation analysis to identify co-expressed genes preserved across the four diseases. SARS-CoV-2 with the D614G mutation on the spike protein was the main strain analyzed in this study. A total of 13 stable gene co-expression modules were identified from the whole blood transcriptome of ARDS patients. Modules were more strongly preserved in the sepsis dataset, and less strongly preserved in the SIRS and COVID-19 datasets. Among these identified modules, M1, M9, M16, M18, M19, and M44 were strongly preserved in COVID-19 and across all four diseases. These modules were enriched in olfactory transduction, Th1 and Th2 cell differentiation, chemokine signaling, NOD-like receptor signaling, mitophagy-animal, and nucleocytoplasmic transport KEGG pathways. Genes within each module were shown to be significantly biologically-related based on PPI network construction. Identified hub genes and pathways align with a previous study and therefore reinforce their potential as therapeutic biomarkers and druggable targets. The seven identified candidate drugs also warrant further investigation as potential therapeutics against cytokine storms. Further experiments are recommended however to verify their effectiveness against cytokine storms.</p></sec></body>],\n",
              " [<body><sec id=\"sec1\"><p>Coronavirus disease 2019 (COVID-19), which was first identified in December 2019, and shortly after, declared a pandemic by World Health Organization (WHO) []. As of January 18, 2022, there have been more than 326 million confirmed cases and 5.54 million deaths globally []. Coronaviruses belong to the family of Coronaviridae, RNA viruses with crown-like spikes on the surface of the coronavirus particles. According to a meta-analysis of Macedo et al. [], the mortality rate of COVID-19 was 17.1% for patients admitted to hospitals, whereas WHO estimated a fatality rate of 6.73%, which was much lower than that calculated from published studies. Among the critical cases of COVID-19, the mortality rate reaches 40% [].</p><p>Substantial efforts have been made in the treatment of patients with COVID-19. The WHO recommendations in the treatment of COVID-19 are as follows []: molnupiravir (conditional), baricitinib (strong), ruxolitinib and tofacitinib (conditional), sotrovimab (conditional), casirivimab and imdevimab (conditional), IL-6 receptor blockers (tocilizumab and sarilumab) (strong), remdesivir (conditional), and systemic corticosteroids (strong). The WHO recommends not to use ivermectin, lopinavir/ritonavir, hydroxychloroquine, and convalescent plasma.</p><p>The pathogenesis of COVID-19 was explained by cytokine storm, reduction in ACE2 expression, and activation of complement pathway-induced microvascular injury and thrombosis []. The mechanisms of the recommended agents are focused on the mentioned pathogenesis to improve the clinical outcome of COVID-19, and antiviral therapies are missing. The anticoronaviral strategies include preventing the synthesis of viral RNA, inhibiting virus replication, blocking the virus binding to human cell receptors, or inhibiting the viruses' self-assembly process []. The SARS-CoV-2 contains at least four structural proteins: spike (S) protein, envelope (E) protein, membrane (M) protein, and nucleocapsid (N) protein, and 16 nonstructural proteins (NSPs). Among the translated NSPs, the main protease, also called chymotrypsin-like protease (3C-like protease), and the papain-like protease are two essential proteases for proteolytic processing of the coronavirus replicase polyprotein, therefore generating functional replication complex of the virus, whereas RNA-dependent RNA polymerase is the central enzyme for RNA synthesis. These three NSPs play crucial roles in coronavirus replication, making them attractive targets for anticoronaviral drug design [].</p><p>The S protein, a surface-located trimeric glycoprotein of coronaviruses, promotes the attachment of viruses to host cells through binding to angiotensin-converting enzyme 2 (ACE2) and virus-cell membrane fusion during viral infection. Thus, the S protein has been considered as a major target for the development of vaccines and drug [].</p><p>The development of a new therapeutic agent is a complex, lengthy, and expensive process, which can take 2–4 years of preclinical development and 3–6 years of clinical development and over 500 million dollar cost. There are three critical steps to develop a new drug including discovery and development, preclinical research, and clinical development [].</p><p>Drug discovery involves screening hits, medicinal chemistry, and optimization of hits to reduce potential drug side effects. For drug discovery, two different complementary approaches can be applied: classical pharmacology, also known as phenotypic drug discovery, which is the historical basis of drug discovery, and reverse pharmacology, also known as designated target-based drug discovery. Screening methods based on phenotypic drug discovery have been used to discover new natural products mainly from the terrestrial origin []. These two strategies have advantages and disadvantages and promote very different screening assays. The frequent re-discovery of the same compounds, the technical difficulties associated with the isolation of compounds from extracts, and the incompatibility of natural product extracts with high-throughput screening (HTS) campaigns were the disadvantages of phenotypic drug discovery. On the other hand, natural product structures have the characteristics of high chemical diversity, biochemical specificity, and other molecular properties that make them favorable as lead structures for drug discovery, which serve to differentiate them from libraries of synthetic and combinatorial compounds []. Overly simplified assays, acting of drug on more than one target, the multifactorial nature of diseases, and challenges to identify a single molecular target are some limitations of target-based drug discovery. Therefore, a comprehensive screening strategy will incorporate both targeted and phenotypic assays, with one format designated as the primary screen and the other as a secondary or follow-up assay. During the spread of COVID-19 outbreak, great efforts have been made in therapeutic drug discovery against the virus. Because COVID-19 is a new, acute, severe infectious disease, the anti-SARS-CoV-2 drug development strategies are to screen existing drugs to identify potentially effective drugs, to expand indications, and to develop a vaccine []. The safety of conventional drugs has been mostly verified; if effective, they can be quickly applied in clinical practice (repurposing of existing drugs). The recent rises of several high transmissible strains sounded alarms for currently used vaccines and drugs. Therefore, developing broad-spectrum antiviral drugs not only to combat COVID-19 but also to provide protective arsenals against future viral outbreaks is a requirement. Scientists continue the development of broad-spectrum antiviral drugs from natural or chemical sources, which have the potential advantages of broad-spectrum therapeutic effect and insensitivity to viral evasion. Given the urgency of the SARS-CoV-2 outbreak, here we discuss the discovery and development of new therapeutics for SARS-CoV-2 infection based on the strategies from preclinical drug discovery.</p></sec><sec id=\"sec2\"><p>We report this systematic review based on the Preferred Reporting Items for Systematic Reviews and Meta-Analysis (PRISMA) guidelines [].</p><sec id=\"sec2.1\"><p>Studies published in PubMed, Scopus, Web of Science, EMBASE, Google Scholar, and DrugBank were searched from December 2019 to March 2021 using the following search terms: “Coronavirus,” “Covid-19,” “SARS-CoV-2,” “Drug design,” “Drug development,” “In silico,” “In vitro,” and “In vivo” alone or in combination without language restrictions. The keywords were selected using expert opinions, mesh, and related article titles.</p><p>All articles with full text or in the absence of full text with abstract are included in the screening of this study. Studies were excluded if studies were comments, editorial, letters, review, and preprints.</p></sec><sec id=\"sec2.2\"><p>Two researchers independently extracted data from included studies using a predefined data extraction form. All disagreement was discussed and solved after rechecking the source data with a third investigator. The data extracted, including the last name of the first author, type of the study (in silico, in vitro, in vivo), and name of the agent (chemical compound, drug, herb, etc.), studied the mechanism and efficacy of the agent according to study design that specified according to the following definition.</p><p>Computer-aided drug design can be divided into three different categories. All are based on ligands and receptors, which are briefly described [] as follows:<list list-type=\"order\"><list-item><p><italic>Dock Receptor-Based Approach</italic> []. Once the three-dimensional structure of the ligand molecules and their receptor is known, the receptor-based method is a good candidate for identifying or optimizing drugs. Due to the presence of three-dimensional structures of compounds and receptors, the nature of the interaction between the ligand and the receptor and the type of structure that the ligand can have to interact with them in favorable conditions can be identified using this method. The compound is simulated on the active site of the dock (meaning anchoring) and on the interaction of the ligand with the receptor by molecular mechanics and molecular dynamics. In this method, due to the ligation of the ligand in the active position, the ligand changes in terms of conformity and changes its position in different conditions and shows interaction with the receptor in different types of situations. To determine the type of ligands that can be docked into the receptor site, the matching of the shape and the complementarity of the hydrophobic, hydrophilic, and charged parts must be considered. Various software packages such as AUTODOCK-Glide-LUDI and LigandFit are used to design the drug based on the structure of the receptor.</p></list-item><list-item><p><italic>Ligand-Based Approach</italic> []. This method is used where the three-dimensional structures of the receptor are unknown and instead the structure of the ligands is known, which is one of the common methods. In this method, by indirectly studying compounds that react with biomolecules, they seek to design compounds that are pharmacologically active. In ligand-based drug design methods, in the absence of biomolecule structure, by studying specific ligands, it seeks to identify the structural and physicochemical properties of the compounds so that the desired compound can be designed based on data extracted from the study of previous compounds. This method is a kind of drug design based on pharmacophore (pharmacophore refers to the part of the drug to which the effect of the drug depends on that part of the molecule), and by studying the quantitative relationship between structure and their activity, drugs can be designed by this method. It can be said that it is a method for designing the pharmacophores of drugs.</p></list-item><list-item><p><italic>Denovo Design-Based Approach</italic> []. This method is used when the structure of the ligand is unknown but the structure of the receptor is known. In this method, there is information about the structures of the receptor or quasi-receptors, but there is no structure of the main composition that can interact with the active site of the receptor. One of the functions of drug design based on this method is to suggest and present the main composition that is complementary to the active site. The basis of the method is that the database of existing 3D structures is used to find small molecules that can interact with the active site of the receptor in terms of size, geometry, and functional groups. Software packages such as GROW and LEGEND are used to design drugs by this method.</p></list-item></list></p><p>Drug design methods in the computer include quantitative structure-activity relationship (RASQ), docking, molecular dynamic simulation, and computational modeling. In these studies, the efficacy is evaluated based on the function of the drug or compound agent and the mechanism of action.  shows the methods of computer-aided drug design (CADD).</p><sec id=\"sec2.2.1\"><p>Quantitative Structure-Activity Relationship (QSAR). QSAR provides studies on the relationship between chemical structure and biological activity or other biological activities that are important in selecting or removing a compound before synthesis and testing. QSAR [] is especially important to predict the result, especially when it is not possible to experiment with a compound. Molecular descriptors, which are the most important components of QSAR, can be obtained experimentally or through mathematical formulas from various theories such as quantum mechanics, chemical graph theory, and study theories. QSAR seeks to establish a statistically significant relationship between structure and performance. It also explains the specific effect of a drug and can ultimately predict the effect of newly synthesized chemical compounds. QSAR model is also an equation that predicts a property through molecular descriptors and their coefficients. Evaluation of the effectiveness of new compounds that have been studied using this method can be reported as a percentage of enzyme inhibition if the modeling has been done and mentioned in the article.</p></sec><sec id=\"sec2.2.2\"><p>In this technique, to achieve a combination with a pharmacological effect and increase the pharmacological activity of the drug, different formulations of a drug interact with the receptor, and the structure that has the best interaction with the receptor and the lowest energy level is selected for laboratory steps []. In this way, possible structures that have a stronger interaction with the receiver can be isolated at this stage. The same issue is considered and reported as an effectiveness measure.</p></sec><sec id=\"sec2.2.3\"><p>In this technique, which is based on the simulation of drug-receptor interaction in the body, docking problems are solved and in fact play a complementary role in this technique []. Due to the time-consuming work with this technique, effective structures cannot be achieved directly through it, and the final stages of the study of drug-receptor interactions should be evaluated before starting laboratory work by having effective compounds from the previous stages. Molecular dynamic simulations produce information at the microscopic level (position and velocity of atoms). The conversion of these data into macroscopic values (pressure, energy, etc.) is done using statistical mechanics. In fact, molecular dynamics and statistical mechanics link microscopic concepts and macroscopically observable quantities. Molecular dynamic simulations are only able to predict the thermodynamic behavior and stability of the ligand binding mechanism at the active site of the target enzyme. This is reported as a criterion.</p></sec><sec id=\"sec2.2.4\"><p>Study of drug in cell culture medium: effectiveness in these studies means inhibition of the replication of COVID-19 by the compound or drug under study [].</p><p>The half-maximal inhibitory concentrations (IC<sub>50</sub>) are a measure of the effectiveness of a compound in inhibiting biological function [].</p><p>In vivo studies are those in which the effects of drugs are tested on whole living organisms or cells usually animals as opposed to a tissue organism or dead organism. In vivo testing is better studied for observing the overall effects of an experiment on living subjects [].</p></sec></sec></sec><sec id=\"sec3\"><p>In this review, we reported a significant number of articles with in silico, in vitro, and in vivo approaches for drug development of COVID-19. We retrieved a total of 2538 articles from the initial database search. After the removal of duplication and screening, 317 studies were selected for inclusion in this review.  shows the PRISMA diagram.</p><p>The analysis of article contents indicated that 266 studies performed in silico approaches against viral targets; 34 studies used in vitro approaches against SARS-CoV-2; and 15 studies used in vivo (animal) models.</p><sec id=\"sec3.1\"><p>From 267 studies used in silico approaches, 98 article studies repurposed approved drugs with a new mechanism of action and 91 studies evaluated natural products (e.g., herbal medicine) on COVID-19. The characteristics of these studies are summarized in Tables and . Also,  shows the characteristics of the remaining studies (<italic>N</italic> = 87).</p><p>In silico studies used the following component of novel coronavirus as targets: main protease (<italic>N</italic> = 154), spike glycoprotein (<italic>N</italic> = 62), nonstructural protein (<italic>N</italic> = 45), RNA-dependent RNA polymerase (<italic>N</italic> = 21), papain-like protease (<italic>N</italic> = 19).</p><p>About 260 drugs were repurposed by the computational methods for COVID-19 therapy such as about 120 drugs candidate against the main protease, 52 drugs against the spike glycoprotein, 14 drugs against RNA-dependent RNA polymerase, and 28 drugs against other nonstructural proteins.</p><p>Among the studied repurposed drugs, the best results (regarding efficacy) were observed with saquinavir (<italic>N</italic> = 9 study), ritonavir (<italic>N</italic> = 8 study), lopinavir (<italic>N</italic> = 6 study), remdesivir (<italic>N</italic> = 3 study), and amikacin, danoprevir, favipiravir, and telaprevir.</p><p>\n",
              "   shows target-based synthesis of data for COVID-19 drug repurposing. As presented, at least two studies show the efficacy of aprepitant, cobicistat, dipyridamole, and dihydroergotamine against the main protease and tegobuvir (<italic>N</italic> = 2) against spike protein. The following list of drugs had multitarget action: avapritinib, famotidine, bictegravir, ziprasidone, capmatinib, pexidartinib, amprenavir, zafirlukast, cilostazol, paromomycin, lopinavir, and remdesivir.</p><p>A total of 91 studies used in silico methods to evaluate the effects of natural products including herbal medicine against SARS-CoV-2. Among them, 54 studies used the main protease as main target, in which eucalyptus (<italic>N</italic> = 3), Tinospora cordifolia (<italic>N</italic> = 3), and flavonoids (e.g., hesperidin, rutin, and herbacetin) were the most studied and effective. Other studied targets were as follows: spike (<italic>N</italic> = 22) and multitarget (<italic>N</italic> = 20). From the plant metabolites, oleanolic acid, hesperidin, epigallocatechin gallate, jensenone, tinosponone, and anistone show promising results in computational methods against COVID-19. Aloe, green tea, eucalyptus, curcumin, and many Chinese, Indian, and African plants were also effective on COVID-19 in silico.</p><p>The derivatives of pyrazoles, oxadiazoles, phenyltriazolinones, triazoles, benzoylpinostrobin, benzoic acid, benzylidenechromanones, coumarin, and selenium show efficacy in computational methods and could be considered as lead molecules for drug design and synthesis against COVID-19. Melatonin, ebselen, phenyl furoxan, thimerosal, isatin, romidepsin, phenyl mercurin, pleconaril, and tyrosine kinase inhibitors (such as nilotinib and imatinib) are examples of other drugs that are effective in in silico methods.</p></sec><sec id=\"sec3.2\"><p>Screening of in vitro studies leads to finding 34 studies. Different compounds were evaluated and most studies focused on the inhibition of viral replication, which was assessed by the quantification of viral RNA by PCR and IC<sub>50</sub> value reported. Most of the studies use the Vero E6 cell line for the assessment of replication of SARS-CoV-2 (). In particular, main protease, papain-like protease, RNA-dependent RNA polymerase, NSP-14, NSP-15, spike protein, and TMPRSS2 were evaluated in vitro as targets. Also, three studies focused on in vitro inhibition of inflammatory markers such as interleukins and the effects of the cytopathic effect on infected cell.</p><p>Nitric oxide, ginkgolic acid, anacardic acid, troxerutin, bisindolylmaleimide derivatives, small molecules (GRL-172, and 5 h), baicalin and baicalein (phytochemicals), and bepridil were effective drugs in vitro against the main protease of COVID-19.</p><p>The following drugs could inhibit the spike/ACE2-mediated cell entry of COVID-19 in vitro: romidepsin, panobinostat, givinostat, sirtinol, saquinavir, lipopeptides, hydroxyzine, azelastine, heparin, and glycyrrhizic acid.</p><p>Alkenyl sulfonylurea derivatives [], an IMU-8381 inhibitor of human dihydroorotate dehydrogenase, pyrazole derivatives, and phillyrin, regulated the expression of inflammatory cytokines (e.g., IL, TNF-<italic>α</italic>, and NF-<italic>κ</italic>B) induced by SARS-CoV-2 markedly and could be considered as adjuvant treatment of COVID-19 severe disease.</p><p>Doxycycline, chlorpromazine, azithromycin, heparin, bepridil, tannic acid, and glycyrrhizic acid are well-known drugs that show both in silico and in vitro inhibitory effects against SARS-CoV-2 and should be considered for this purpose.</p><p>Also, in vitro studies show that lopinavir/ritonavir, sofosbuvir, and favipiravir have no antiviral effects against SARS-CoV-2 (huge gap between in vitro IC<sub>50</sub> and free plasma concentration).</p><p>We included 16 in vivo studies in our final analysis. Most of them use a combination of in vitro and in vivo methods for the evaluation of novel drugs.  shows the detail of these studies.</p><p>Human monoclonal antibodies were the most evaluated drugs (<italic>N</italic> = 4), promote the reduction in viral load (in vitro), and prevent infection in animal models of SARS-CoV-2.</p><p>A prodrug of hydroxycytidine (molnupiravir) improved pulmonary function and reduced viral titer in vivo and was introduced as a potential broad-spectrum antiviral agent against SARS-CoV-2.</p><p>Chloroquine and chlorpromazine did not inhibit viral replication in mouse lungs, but protected them against clinical disease.</p><p>Dalbavancin shows significant inhibitory ability in both in vitro and in vivo models of COVID-19. The drug binds directly to ACE2 and blocks its interaction with the SARS-CoV-2 spike protein.</p></sec></sec><sec id=\"sec4\"><p>Drug development is a multistep process, typically requiring more than five years to assure the safety and efficacy of the new compound. There are several strategies in antiviral drugs for coronaviruses including empirical testing of known antiviral drugs, large-scale phenotypic screening of compound libraries, and target-based drug discovery. To date, an increasing number of drugs have been shown to have anticoronavirus activities in vitro and in vivo, but only remdesivir and several neutralizing antibodies have been approved by the US FDA for treating COVID-19. However, remdesivir's clinical effects are controversial and new antiviral drugs are still urgently needed. Given the urgency of the SARS-CoV-2 outbreak, here we discuss the discovery and development of new therapeutics for SARS-CoV-2 infection, which have been conducted in basic research (in silico) and preclinical study (in vitro, in vivo). Our database search identified about 3000 studies, which means a global effort for drug development in the current COVID-19 pandemic. Although the chance of successful drug development is very low (less than about 10%) and till today, there are no approved drugs. Many potential candidates (at least 420 drugs) are in clinical trials.</p><p>As summarized in the results, many compounds are in the development process for COVID-19 disease. Some of these compounds are completely new and could serve as seeds (or leads) for developing antiviral drugs against COVID-19, but as we need therapeutics as soon as possible, half of the studies focused on drug repurposing (repositioning), which is a process of investigation of existing drugs for new therapeutic purposes. With the emergence of a growing COVID-19 pandemic, the drug repurposing process was being accelerated. Clinical trials using repurposed drugs may take less time and have a lower overall cost of manufacturing and could have a wide distribution of drugs. According to our results, 260 drugs repurposed by the computational methods for COVID-19, among them saquinavir, ritonavir, and lopinavir, showed the best efficacy in in silico environment. These drugs can be rapidly repurposed for clinic application for treating COVID-19 patients given their proven safety.</p><p>Many trials are performed using a combination of ritonavir-lopinavir. The results of a systematic review and meta-analysis showed that this drug combination has no more treatment effects than other therapeutic agents in COVID-19 patients and is currently not used anymore []. We could not find any clinical trial on saquinavir, which is the most studied drug in silico and show high potency against COVID-19. Saquinavir could be a suggestion for further clinical research.</p><p>Given that the development of synthetic chemicals for therapeutic use is a random process that might result in serendipitous discovery, many pharmaceutical companies are now focused on the development of plant-derived drugs. Natural products and their structural analogs have historically made a major contribution to pharmacotherapy, especially for cancer and infectious diseases. Nevertheless, natural products also present challenges for drug discovery, such as technical barriers to screening, isolation, characterization, and optimization. In recent years, several technological and scientific developments—including improved analytical tools, genome mining and engineering strategies, and microbial culturing advances—are addressing such challenges and opening up new opportunities. Consequently, interest in natural products as drug leads is being revitalized. Medicinal plants have attracted significant attention to treat infectious diseases. Complex molecular structures and a wide variety of natural compounds make medicinal plants an excellent biological resource for drug discovery. Our results show that various plants have potential antiviral activities and could use or be a basis for drug development against COVID-19.</p><p>Some of the studied plants are among what used by people on a daily basis, such as green tea, aloe, curcumin, and eucalyptus. A systematic review and meta-analysis of RCTs on herbal medicine in the management of COVID-19 show the significant effects of the combined therapy of herbal medicine in treating COVID-19 without any significant side effects [].</p><p>Our results retrieved 91 in silico studies of natural products including herbal medicine. The results of this in silico approach showed that some of these studied active ingredients have a high affinity for each of the four important viral proteins compared with the inhibitors previously reported for each of these proteins. They could possibly have an inhibitory effect on the SARS-CoV-2 and COVID-19. Most evaluated chemical compounds had inhibitory effects against one or two proteins of SARS-CoV-2. In addition to having a great affinity to attach to the viral proteins, these herbal compounds have antioxidant, vasoprotective, anticarcinogenic, and antiviral properties. Thus, they can be applied as extremely safe therapeutic natural compounds and clinical assessments might have notable outcomes for controlling COVID-19.</p><p>Due to the nature of preclinical (in vitro and in vivo) studies, the number of drug development studies in this area was less than in silico studies. Again, repurposed drugs were the most studied drugs in vitro (both herbal and synthetic). Some of them such as saquinavir, heparin, glycyrrhizic acid, and chlorpromazine show efficacy in both in vitro and in silico environments. Chlorpromazine is the single agent that was found to have efficacy in in silico, in vitro, and in vivo areas.</p><p>In terms of mechanism of action, different targets from the structural and nonstructural proteins of COVID-19 were evaluated. Most of the studies focused on the main protease, papain-like protease, and spike glycoprotein. Apart from the specific protein that leads to viral replication, SARS-CoV-2 causes a surge of pro-inflammatory cytokines and chemokines, which cause damage to lung tissue and deterioration of lung function. Therefore, the design of a drug with multitarget of action against different proteins of COVID-19 and also anti-inflammatory potential could be valuable.</p><p>Currently, there is no highly efficacious and specific treatment for SARS-CoV-2. Herein, we provided data on novel compounds in therapeutic drug discovery and development. Due to the nature of SARS-CoV-2 and the rises of several high transmissible strains, repurposing existing drugs has demonstrated power by bringing several drugs to approval for treating COVID-19 patients, such as remdesivir. Our results confirmed that a large number of repurposed agents are currently being explored for treating SARS-CoV-2 infection. However, these drugs still suffer from suboptimal therapeutic effect or known strong side effect. To accelerate drug discovery and development, especially during the current pandemic, natural products capture attention again. Our results showed that various natural bioactive compounds are being investigated in the preclinical step of drug development for COVID-19. In addition to having high affinity, these herb active ingredients have antioxidant, vasoprotective, anticarcinogenic, and antiviral properties. Therefore, they can be used as extremely safe therapeutic compounds in drug design studies to control COVID-19. However, the pharmacological effects and adverse reactions of some drugs under development are still unclear, and hence, well-designed high-quality studies are needed to further study the effectiveness and safety of these potential drugs to accelerate drug development targeting SARS-CoV-2 and thus promote progress towards ending the pandemic.</p></sec></body>],\n",
              " [<body><sec id=\"sec1\"><p id=\"p0035\">COVID-19 is an infectious disease caused by Coronaviruses (family: <italic>Coronaviridae</italic>), which has recently been identified as a cause of severe acute respiratory syndromes (SARS Cov-2). Coronaviruses are named as such because their surface is characterized by protrusions resembling crowns []. Corona was first reported in China (Wuhan), and COVID-2019 is unlikely to have been transmitted from animal to human in a seafood market [,]. The outbreak has been rapidly spread across the countries and territories due to its high human to human contagious nature. Until Oct 13, 2022, 628,556,214 people have been infected, and 6,566,411 have died (). Vaccines are without question the most efficient tool for combating viral diseases. However, there are inherent threats with vaccination, including non-complete viral deactivation or potential adverse effects in people; even after a vaccine has been produced, exact quality control is required to ensure the treatment's safety. As a result, developing vaccines is a lengthy process; a vaccine may not be commercially available for years following the emergence of a new disease. The current vaccines are not providing adequate protection against COVID-19 due to logistical difficulties in their distribution, diminishing immunity, and the possibility of transmission from asymptomatic infected patients [,]. The advent of vaccines may have broken the transmission chains in this viral pandemic, but there is currently concern about their effectiveness against mutant variants, reported adverse reactions, and possible long-term effects. We must, therefore, find safe, effective, preferably affordable, and most importantly, specific drugs rather than the non-specific drugs currently prescribed in order to save the lives of the infected [,].</p><p id=\"p0040\">A computer-assisted drug discovery strategy is the most common method of predicting potential compounds before they are synthesized and tested in vitro []. Drug repurposing is the most efficient method of rapidly identifying the unique clinical applications of currently licensed medications to treat COVID-19. Since the toxicity and pharmacokinetics of repurposed pharmaceuticals are already known, this method could minimize the time and expense required to develop a new medicine []. The repurposing of existing medications is made possible through molecular docking research against viral protein targets []. In terms of mathematics, molecular computation has several forms, some of which can be used in biology and protein research because of their ability to handle a wide range of particle systems. The interactions between these molecules are predicted using molecular docking and molecular dynamics modeling. In silico screening of different potential active chemicals can be accomplished cost- and time-efficient using these strategies [,]. Numerous scientific disciplines are using molecular modeling techniques, including drug discovery and design [, , ]. The Coronavirus genome is translated into two classes of proteins within the host cell: structural proteins such as spikes (S), envelopes (E), matrixes (M), and nucleocapsids (N), and nonstructural proteins such as 3-C proteases (3CLpro, NSP5) and RNA Dependent RNA Polymerases (RdRp, NSP12) [].</p><p id=\"p0045\">Among these proteins, RNA-dependent RNA polymerase (RdRp) has been demonstrated to contribute significantly to the replication of the viral genome (single-stranded RNA) and to the multiplication of the virus in various cells [,]. It is essential to inhibit RdRp in order to control the progression of infection in human cells and, as a consequence, the viral load in patients, as RdRp replicates viral genomes by synthesizing new RNA nucleotides. It has been established that RNA-dependent RNA polymerase (RdRp) is an essential enzyme that plays a significant role in the replication and translation of viral genomes; this makes it an excellent therapeutic target for COVID-19. Numerous clinical trials have shown that COVID-19 can be effectively treated with antiviral, antimalarial, and anti-HIV drugs. The FDA has approved Remdesivir (Veklury®) for treating COVID-19 in hospitalized patients aged 12 and older and weighing at least 40 kg. Several viral infections, including SARS-CoV-2, have been successfully treated with an analog of nucleotides (NA) known as Remdesivir. In the presence of NAs, a 5-triphosphate is formed in the cell, which replaces the RdRp substrate and competes with the endogenous nucleotides for incorporation into viral RNA. In a comprehensive two-tier screening approach, Yuan et al. demonstrated that FDA-approved medications cetilistat, abiraterone, diiodohydroxyquinoline, and bexarotene inhibited COVID-19 infection in vitro []. As part of this study, we investigated the interaction between four FDA-approved medications, bexarotene (anticancer retinoid), abiraterone (synthetic androstenedione steroid), diiodohydroxyquinoline (antiparasite), and cetilistat (antipancreatic lipase) with SARS-CoV-2 RdRp using docking simulation. In this study, remdesivir (an antiviral drug) was compared to the outcomes obtained with remdesivir. Dynamic simulations also clarify the dynamics of molecule behavior. In light of our findings, we may be able to develop a new method for combating COVID-19.</p></sec><sec id=\"sec2\"><sec id=\"sec2.1\"><sec id=\"sec2.1.1\"><p id=\"p0050\">The screening of the drugs against RdRp of SARS-CoV-2 was performed by AutoDock Vina [] with MGL tools 1.5.4 which makes more precise docking calculations and runs faster than AutoDock software []. The crystal structure of RdRp (PDB ID: ) was downloaded from the Protein data bank []. The receptor preparation was done by utilizing the AutoDock Tools (v.1.5.4). The addition of Kollman charged atoms to protein crystal structure and the combination of non-polar hydrogen atoms was done by MGL Tools (v.1.5.4). To perform molecular docking analysis the RdRp of SARS-CoV-2 was converted into PDBQT format (.pdbqt). The grid box was set to 100 × 90 × 122 with a grid spacing of 1.00 Å and center parameters of 150.05, 152.94, and 163.03.</p></sec><sec id=\"sec2.1.2\"><p id=\"p0055\">The 3D structure of bexarotene, diiodohydroxyquinoline, abiraterone, cetilistat, and remdesivir were received from PubChem as SDF files (.sdf) (. ) and converted to PDB format (.pdb) before perform docking utilizing MGL Tools (v.1.5.4). The BIOVIA Discovery Studio software was employed to visualize the docked results.</p></sec></sec><sec id=\"sec2.2\"><p id=\"p0060\">Because molecular docking results revealed that bexarotene has the best energy in interaction with RdRp of SARS-CoV-2, we investigated the molecular dynamic simulation of this drug with protein. We did the Classical MD simulations [,] with the CHARMM27 force field using ran GROMACS 2018.2 software [,]. SwissParam [] was used to generate the ligand topology. The free protein and protein-drug complex were solved using TIP3P water [] in the cubic box with periodic boundary situations in three directions. The solutes were placed in the box's center, with a minimum distance of 1.0 nm between their surfaces. Markedly, we added Na<sup>+</sup> ions to the system to neutralize the charge. The systems were balanced at 300 K and 1 bar, following energy minimization using the steepest descent method. The temperature was kept at 300 K using a modified Berendsen thermostat, and the pressure was kept at 1.0 bar using a Parrinello-Rahman barostat. Bond lengths were calculated using the LINCS algorithm, and long-range electrostatic forces were computed employing the particle-mesh Ewald scheme (PME) (grid spacing 0.16 nm) []. For short-range non-bonded interactions, cutoff ratios of 1.0 nm for Coulomb and van der Waals potentials were used. Finally, a 50 ns MD simulation with a time-step of 2 fs was running, with velocities generated randomly using a Maxwell distribution.</p></sec></sec><sec id=\"sec3\"><sec id=\"sec3.1\"><p id=\"p0065\">Because of its critical role in virus replication, RdRp is considered a key target for developing antiviral drugs [,,]. We recently reported the in-silico screening of bexarotene, diiodohydroxyquinoline, abiraterone, cetilistat, and remdesivir against different SARS-CoV-2 drug targets []. However, RdRp, an important drug target, was excluded from this study. The current study used the same strategy to screen the bexarotene, diiodohydroxyquinoline, abiraterone, cetilistat, and remdesivir for their potential to inhibit RdRp of SARS-CoV-2, thereby filling the gap. Molecular docking simulation is widely used in computational drug discovery and design. This technique was used to predict the binding of the selected drugs bexarotene, diiodohydroxyquinoline, abiraterone, cetilistat, and remdesivir with SARS-CoV-2 RdRp-RNA protein. Recently, researchers reported virtual screening inhibitors of SARS-CoV-2 RdRp [, , , ]. In our study, the FDA-approved RdRp inhibitor drug remdesivir was used as a comparator drug. Remdesivir is effective against many RNA viruses, including the newly discovered SARS-CoV-2 (COVID-19) and previous coronaviruses [,]. We hypothesized that the drugs mentioned above could disrupt SARS-CoV-2 RdRp.</p><p id=\"p0070\">Cetilistat, a pancreatic lipase inhibitor, can be prescribed for obesity treatment. This compound inhibits fat digestion and absorption []. Diiodohydroxyquinoline is a quinolone derivative used to treat amoebiasis as a luminal amebicide []. When bile is available, diiodohydroxyquinoline is unlikely to be absorbed into the circulation, whereas cetilistat is rapidly hydrolyzed into its metabolites [,]. Gastrointestinal complaints have been reported in 15–20% of COVID-19 patients, and some of these individuals have also shown infectious virus particles and identifiable viral RNA []. In the intestines of SARS-CoV-2-infected hamsters, high levels of viral nucleocapsid protein, inflammatory processes, and identifiable viral RNA were found []. Cetilistat and oral diiodohydroxyquinoline may be effective topical luminal antivirals in suppressing viral shedding in the gastrointestinal system in the same way feces may be a significant source of SARS-CoV-2 infection [,]. Nonchemotherapeutic antineoplastic medicines, including bexarotene and abiraterone acetate, have been shown to have minimal immunosuppressive properties []. Breast cancer, non-small cell lung cancer (NSCLC), and cutaneous T-cell lymphoma are effectively treated with bexarotene, a third-generation retinoid [, , , ]. Abiraterone acetate suppressed the activity of the enzyme CYP17A1, which is responsible for androgen synthesis when coupled with a corticosteroid. Androgen deprivation is the mechanism employed in this therapy to treat resistant prostate cancer [,]. As described by Yuan et al. [], members of the same medication class as bexarotene, such as tamibarotene and AM580, have demonstrated outstanding antiviral effectiveness against several viruses, including influenza, Zika, coronaviruses (SARS-CoV and MERS-CoV), adenoviruses, and enterovirus A71.</p><p id=\"p0075\">According to molecular docking studies, the binding energies of bexarotene, abiraterone, cetilistat, remdesivir, and diiodohydroxyquinoline to the RdRp were - 8.4, - 7.9, −7.6, −7.6 and - 5.5 kcal/mol, respectively, which demonstrate good binding potential towards the RdRp. The binding energy values of the five compounds mentioned with the SARS-CoV-2 RdRp protein are shown in \n",
              "  .</p><p id=\"p0080\">According to our findings, bexarotene was the most effective compound against RdRp, with the lowest binding energy. Also, bexarotene and abiraterone were discovered to have the lowest binding energies than remdesivir. In the meantime, the binding energies of cetilistat are equal to those of remdesivir. Moreover, the binding energy value for diiodohydroxyquinoline (−5.5 kcal/mol) suggests that the ligand-protein complex is stable. , , , , \n",
              "  represent the amino acid residues involved, bond lengths, and the type of interactions. , , , , \n",
              "  depict the RdRp amino acid residues that interacted with drugs. As evident, The SARS-CoV-2 RdRp protein had hydrophobic interactions with abiraterone and bexarotene. Meanwhile, cetilistat, diiodohydroxyquinoline, and remdesivir interacted with the RdRp of SARS-CoV-2 via hydrogen bonds and hydrophobic interactions. The chosen drugs interacted with various amino acid residues of the RdRp of the SARS-CoV-2 protein (, , , , ). It is evident from  that the bexarotene interacted with RdRp with LEU240, TYR129, LEU708, VAL128, ALA125, LEU207, TYR129, HIS133, TYR728, and TYR732 amino acids. These interactions can disrupt the biological activity of the SARS-CoV-2 protein RdRp, and, consequently, the viral replication process. Then, we used MD models to confirm the docking poses and analyze the dynamics of interactions between RdRp, RNA, and medicines.</p></sec><sec id=\"sec3.2\"><p id=\"p0085\">We conducted an MD simulation study to evaluate system dynamics and improve the accuracy of the docking computation. For further investigation, bexarotene was chosen as the compound with the highest interaction energy.</p><sec id=\"sec3.2.1\"><p id=\"p0090\">The system compactness can be determined based on the radius of gyration (Rg). The conformational stability of the complex structure during MD modeling is indicated by the graph's stability []. Improved structural integrity and folding treatment are indicated by a low Rg value []. During a 50ns MD running, the molecule's total extension can be quantified using the protein and ligand-protein complex's gyration radius (Rg) (\n",
              "  ). A constant average Rg of 3.276 nm is maintained by the ligand-protein complex during the whole 50 ns of the MD modeling. Similarly, 3.303 nm was the measured Rg value for the free 6NUR. This demonstrates how more ligand binding increases the protein's stability. The findings from the MD simulations provide unequivocal proof that bexarotene may construct a stable complex with 6NUR, which is an indication of its inhibitory effects on the RdRp receptor.</p></sec><sec id=\"sec3.2.2\"><p id=\"p0095\">During simulation, the protein's structural alterations and insights from RMSD analysis support the protein's stability and equilibrium. The higher stability of the protein-ligand complex is indicated by the lower RMSD value of the protein backbone []. \n",
              "  illustrates the RMSD graph of the backbone atoms in the 6NUR complex containing bexarotene and 6NUR. The RMSD, during the 50ns MD simulation, was determined for the converged 6NUR-Bex structure. According to the findings, after 100 ps, both structures reached a stable state. During the entire 50ns simulation, 6NUR-Bex had a mean of 0.197 A<sup>0</sup>, while 6NUR had a mean of 0.216 A<sup>0</sup>. The RMSD value is regarded as acceptable and appropriate if it is less than 1.5 Å. However, it is obviously disregarded if the RMSD number is greater than 3 Å. In addition to studying the conformational changes of the receptors, RMSD measurements are utilized to determine the stability of the receptors with and without ligands []. In the MD simulations, the fact that the RMSD value of 6NUR-Bex was significantly lower than that of the protein suggested that bexarotene contributes to the protein's stability.</p></sec><sec id=\"sec3.2.3\"><p id=\"p0100\">The flexibility of the entire protein concerning its typical structure was assessed using RMSF analysis. High RMSF values showed enhanced flexibility, but low RMSF values showed constricted motions []. Energy is released during ligand binding, and the amount of energy released has a direct association with the values of residual fluctuation (RMSF). \n",
              "  displays the RMSF plots generated for each individual residue that makes up the 6NUR complex with bexarotene and 6NUR. The RMSF values for 6NUR-Bex were exceptionally minimal (Average <sub>RMSF</sub> = 0.28 A<sup>0</sup>); thus, the ligand-protein complex displayed negligible movement, which demonstrates its stability. Residues inside the loop domain of 6NUR in 6NUR-Bex were found to fluctuate more than alpha-helix and beta-sheet domains during the simulation. This demonstrated that during the 50ns simulation, the protein maintained its stability []. The 6NUR residues 895 in the alpha-helix and 714 in the beta-sheet region, except for the loop area, showed considerable fluctuation up to 0.451 A<sup>0</sup> and 0.426 A<sup>0</sup>, respectively. The complex 6NUR-Bex appeared stable overall based on the protein RMSF values [].</p></sec></sec><sec id=\"sec3.3\"><p id=\"p0105\">The average free binding energy of the 6NUR-Bex was calculated using MmPbSaStat.py (\n",
              "  ). The information collected from g<sub>mmpbsa</sub> was used in the calculation of the average free binding energy as well as the standard deviation/error for each of the files. In the drug discovery process, one of the most important factors to consider is the durability of the ligand-receptor contact, which may be evaluated using the binding free energy.</p><p id=\"p0110\">The ligand and protein bind more tightly when the binding energy is lower []. To bind 6NUR to bexarotene, the van der Waals, SASA, and electrostatic energies, except for the polar solvation, were utilized. The polar solvation energy was the only energy that was not utilized. The involvement of van der Waals energy to the entire free binding energy was found to be significantly higher than electrostatic contribution energy. The bexarotene linked exclusively with the 6NUR protein, as shown by the binding free energies of 6NUR-Bex, which were found to be −89.635 ± 0.295 kJ/mol. \n",
              "  depict the binding energy versus time plots for 6LU7-Bex. Bexarotene may be a candidate for blocking the RdRp of the SARS-CoV-2 receptor, as shown by the results mentioned above.</p></sec></sec><sec id=\"sec4\"><p id=\"p0115\">Because they have already been evaluated in terms of toxicity and safety in humans for the treatment of various diseases, repurposing FDA-approved medications will be the best option for developing COVID-19 therapies at this time. This research computationally evaluated the inhibitory potential of the FDA-approved drugs bexarotene, diiodohydroxyquinoline, abiraterone, and cetilistat against RdRp of SARS-CoV-2. With further ex vivo and in vivo examinations, our finding suggests that the selected drugs could be the potential drug of choice for the treatment of COVID-19. Gromacs software was used in this study to simulate the interaction of bexarotene with the RdRp of SARS-CoV-2. Furthermore, the RdRp-Bex radius of gyration has values that indicate the systems’ stability. The binding free energy of 6NUR-Bex was determined to be −89.635 ± 0.295 kJ/mol, indicating that the bexarotene interacted with the 6NUR protein uniquely.</p></sec><sec sec-type=\"COI-statement\"><p id=\"p0120\">The authors declare that they have no known competing financial interests or personal relationships that could have appeared to influence the work reported in this paper.</p></sec></body>],\n",
              " [<body><sec id=\"sec0005\"><p id=\"p0025\">A novel coronavirus (CoV) first appeared by the end of 2019 in Wuhan, China and spread to over 70,000 individuals with 1800 death cases within the matter of first 50 days of the disease . Public health emergency of international health concern was declared by the World Health Organization (WHO) on January 30, and it scaled up to pandemic as of March 11, 2020 . The analytical methods like amino acid and nucleotide sequencing confirmed the relation of novel CoV to same class as severe acute respiratory syndrome coronavirus-1 (SARS-CoV-1) . This new version of coronavirus was named as severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) by the international committee on the taxonomy of viruses and similarly, the disease was named as Covid-19/SARS-CoV-2 . SARS-CoV-2 is a variant of coronavirus with enveloped single-stranded RNA which bears a corona like structure of spikes as shown in\n",
              "  (a). The main host protein for the attachment of spikes has identified as angiotensin converting enzyme 2 (ACE2) receptor. This variant of coronavirus evolved to become the most contagious and infectious among coronavirus family. It enters the lungs by inhaling the contaminated air with viral bodies and infects distal airway cells leading to further spread towards alveolar epithelial cells and causes the damage starting from infection to severe damage as shown in (a). As of May 25, 2021, total number of Covid-19 cases reported have surpassed the 30 million cases globally with 1.6 % in US only . Owing to its high mortality, morbidity, and contagiousness there is dire need for the effective treatments.</p><p id=\"p0030\">Whereas, drug development is complex, time taking, and expensive process. On average it takes around 12–15 years with $2–$3 billion of investment to take a de novo drug from concept to a market ready product . The total number of drugs which are getting the FDA approval are less than 10 % and this rate is declining with time . As a result, pharmaceuticals research suffers from decrease in productivity and faces a persistent gape between the need of treatments and available response , . Repurposing the already approved drugs is a quick and efficient to provide the new treatments by finding out novel uses of the drugs that have well known safety and efficacy profiles. Many drugs have already been repurposed based on variety of the methods and have been approved for new indication . While repurposing can be done at the any stage during drug discovery and development, but its actual potential can be seen in case of already approved drugs . Thus, recently studies have been shown to increasingly use the computational methods to predict the novel drug-target interactions for repurposing the drug or target. One of the methods widely used for drug repurposing is network-based approach. Network based approaches are in-silico approaches which are faster with lower-cost as compared to experimental approaches and serve as a robust strategy to identify the novel drug target interactions or repurposed compound among a pool of many.</p><p id=\"p0035\">Moreover, in the wake of current situation novel Covid-19 (severe acute respiratory syndrome coronavirus 2) has infected around 526.7 million people and have caused more than 6 million deaths globally . In an emergency situations like Covid-19 de novo drug development is not the feasible option since there is urgent need to find the treatments which can be immediately prescribed to prevent the mortality and morbidity. So, to face the crisis because of pandemic drug repurposing seems a tangible strategy for fighting against Covid-19 and other such viral diseases. Additionally, a lot of methods, applications, and databases have been proposed in literature motivating for information rich studies for furthering in unprecedented yet potential direction. Recently, multiple review articles have been published focusing on artificial intelligence (AI) algorithms  such as deep learning models , and network-based methods . Moreover, survey targeting the drug repurposing have also been performed ,  and these include comments and opinion focusing the drug design  and development process . Additionally, articles summarizing the covid-19 drug repurposing using machine learning  and deep learning ,  methods are published as well. To the best of our knowledge, until now there is no review study until now has published which summarizes the drug repurposing studies for Covid-19 with special emphasis on network-based drug repurposing approaches to Covid-19.</p><p id=\"p0040\">As new studies are reported widely almost every day, so it is not applicable to provide a wide and comprehensive overview of all the available repurposing studies. Thus, this review is mainly focused to bridge the gap by providing overview of machine learning and deep learning approaches to drug repurposing with focus on recent developments in network-based drug repurposing methods, tools, and databases used in Covid-19 as shown in (b). In this connection, first we present an overview to computational drug repurposing studies for Covid-19 along with drug and disease related databases followed by recent machine learning and deep learning-based drug repurposing studies. Next, the overview of different network methods and recent state of art studies categorized for each network type is explained. Moreover, recent network-based drug repurposing studies for Covid-19 are discussed and their limitations, benefits, and tools used are summarized. Finally, limitations and future recommendation for improved outcome of network-based drug repurposing in Covid-19 and in general are discussed followed by conclusion.</p></sec><sec id=\"sec0010\"><p id=\"p0045\">Generally, drug repurposing principals can be classified in to two major categories, drug-based or disease-based repurposing. As per the common and widely used hypothesis a drug can cure two diseases if there exist some similarity or the interdependence between two diseases. Similarly, a drug can be associated with various targets and targets owing to confounding nature of the drugs , . Additionally, when it comes to computational drug repurposing, a major challenge is to distinguish between the drug targets and additional gene products which are indirectly participating in the activity of the targets. Unfortunately, traditional methods are limited in their ability to recognize the molecular targets of drugs in a vast population of gene products followed using small datasets originated from different platforms and environmental settings. This might cause inaccuracies in the findings reported in some studies . Small datasets and networks may give the partial knowledge of the living systems.</p><p id=\"p0050\">The amount of biomedical and pharmaceutical data has seen the tremendous rise in the recent years so the improvement in the computational drug repurposing approaches is also visible. Computational approaches like data mining, machine and deep learning, and network analysis have been taken to depth to systematically repurpose the drugs to overcome the limitations faced by the traditional statistical approaches.</p><p id=\"p0055\">Newly adopted computational methods and approaches for demystifying the relationships among different types of biological networks have benefitted the drug repurposing to a greater length. Different biological entities such as genes, proteins, drugs, and diseases and interaction can uncover the potential therapeutic opportunities. The overview of the computational drug repurposing studies with emphasis on network-based approaches is given in , , .</p><sec id=\"sec0015\"><p id=\"p0060\">First attempt to drug repurposing started around two decades ago using simple similarity measure with single source of biological and biomedical data utilization and finally reached to machine learning ,  break through. Artificial and machine learning technology play a crucial role which is modernized with the advent of deep learning and neural network algorithms. Machine learning algorithm can learn from the data and mimic the cognitive human behavior . These days, biological scientists have integrated the machine learning in drug design and discovery processes for improved outcomes , , , .</p><p id=\"p0065\">The applications of the machine learning in drug discovery and development are numerous which include but are not limited to virtual screening (both primary and secondary), toxicity monitoring of the drug, drug efficacy prediction, dosage predictions, drug repurposing, and drug target interactions (DTIs) predictions . Some of the machine learning methods used for drug repurposing in Covid-19 include k-nearest neighbors , random forest , support vector machine , , and more. Thus, over the past decade AI based applications have helped the drug discovery efforts , . Moreover, with the invention of deep learning and its ability to extract the features from raw data automatically, has increased the impact of AI approaches as compared to other Insilico methods , . Various deep learning algorithms utilized in finding treatment against Covid-19 includes artificial neural networks (ANN), convolutional neural networks (CNN), and long short-term memory (LSTM).</p><p id=\"p0070\">Apart from this, to accelerate the process of drug development for Covid-19, machine learning based molecular docking frameworks were utilized on approved as well as new chemical entities. ML enabled molecular docking has been utilized a lot in virtual screening and repurposing , , , . ML enabled molecular docking methods help to find out the bioactive molecules against disease protein of interest.</p><sec id=\"sec0020\"><p id=\"p0075\">There are a lot of studies which have taken the drug repurposing task as binary classification or classification task. Drug-target interaction (DTI) prediction or drug-disease association prediction can help to reduce the volume of redundant compounds and thus serve as an important step in drug development. In our recent work, we developed a feature enriched drug repurposing framework named SperoPredictor  which consisted multiple state of art machine learning algorithms as shown in\n",
              "  (b). The algorithms were individually trained and later stacked together ((b)) followed by molecular docking-based validation ((c)). SperoPredictor training was performed on the feature enriched drug-disease association data of 1430 FDA approved drugs collected from various databases and then unified into binary strings as shown in (a). The training data contained four drug features and three disease features all unified in binary strings which enabled the reduction in training time and space required. The machine learning models were tested on the performance parameters such as accuracy, precision, f1 score, Mathew’s correlation coefficient (MCC), sensitivity, and specificity. The testing accuracies of the models were reported to be 99.3 % for Random Forest (RF) and 99.03 % for Tree Ensemble (TE). SperoPredictor was deployed to predict the potential drugs to be repurposed for Covid-19, NASH, and renal fibrosis. In case of Covid-19, 25 potential candidate drugs were predicted. Literature survey of peer reviewed journals, databases, and patents suggested 12/25 drugs (48 %) were already used for Covid-19. Additionally, molecular docking validation results suggested that four drugs (Balaglitazone, Cortivazol, Velusetrag, and 16 alpha Bromoepiandrosterone) can be used for preclinical validation and clinical trials prior to use in patients. The SperoPredictor results testify the usefulness of having such platforms that can be readily deployed in case of emergent situations like Covid-19. Moreover, the importance of the DTI prediction is highlighted in many studies such as in . These studies to predict the drug-disease associations or drug target interactions require various resources such as done in . Drug-drug interaction based on structural similarity and side effects and similarly disease-disease based on gene expression or disease phenotype have been used. Many online databases such as DrugBank , CTD , and ChEMBL  are providing the open-source drug-target data. The other databases for the various type of drug and disease related data are mentioned in the\n",
              "   and\n",
              "   along with other information. Whereas  provides information related Drug specific databases followed by  providing information on disease information contained databases. Additionally, drug-drug, disease-disease similarity measures were used here as classification features followed using logistic regression. These interactions are normally used in ML and DL studies as shown in\n",
              "  (a-b). Moreover, in the mentioned study classifier was trained to distinguish between the true and false drug-disease association leading to prediction of undiscovered drug-disease associations. Another study designed to predict the DTIs in Covid-19 is given in . The study avails the proteins and drugs provided in their structured format. The physical and chemical features were extracted from the protein amino-acid sequences of drug targets and from SMILES forms of the drug structures , , , . Then the developed model was compared with existing studies using k-fold cross validation ((b)). To deploy the model covid-19 was selected as a case study. Prediction model was deployed as shown in (c) on the proteins known to have been involved in the Covid-19 to predict the possible interactions or drug-target interactions. The model was further deployed to drugs active in Covid-19 and the drugs were predicted with 100 % accuracy. The interaction of the drugs was shown with ACE2 , , ,  self-membrane protein, a key player in Covid-19 enabling disease infection.</p></sec><sec id=\"sec0025\"><p id=\"p0080\">There are many studies which have applied the machine learning based collaborative filtering for predicting novel drug-disease associations or repurposed indications backed by trends found in the gene expressions of different samples , , , . One such study is present in , multiple drug similarity datasets were constructed for finding the therapeutic classes of FDA approved drugs. In this method any mismatches between known and unknow associations were considered as potential repurposed indication. Drug-drug similarity matrices were constructed based on gene expression, structural, and molecular target-based similarity. The prepared data was used to train the support vector machine (SVM) classifier followed by the collaborative filtering to predict novel drug-disease associations. Moreover, a unified computational framework was proposed  based upon integration of various heterogenous data sources to foresee the drug-drug, disease-disease, and drug-disease similarity (known) based data set for building drug-disease network. The problem set forth in the form of drug-disease network was optimally solved by using block coordinate descent (BCD) strategy. A causal inference-probabilistic matrix factorization (CI-PMF) approach was followed in  for classification of novel drug-disease associations. Many biomedical and biological sources were leveraged to build a causal network which contained multiple nodes such as drugs, genes, pathways, and diseases together. Probabilistic matrix factorization (PMF) model was constructed following the network creation to classify drug-disease associations in different groups followed by drug ranking leading to identification of novel drug-disease associations. Additionally, somehow similar studies on larger scale were carried out in , ,  to infer off-target drug interactions for undiscovered drug candidates’ identification. Machine learning plays a key role in drug repurposing against Covid-19 and pace of research in the field has skyrocketed since the emergence of Covid-19. However, there are still some issues to be addressed. For example, lack of standardized frameworks, because without standardization the results received from machine learning models cannot be translated to clinical research. Additionally, there is lack of replicability and reproducibility, despite these, benefits offered by ML methods are numerous within the field of drug repurposing.</p></sec></sec><sec id=\"sec0030\"><p id=\"p0085\">Robust and efficient computational methods making use deep learning are widely required for drug discovery against Covid-19 . The overall combined workflow of ML and DL is shown in  and few recent well-known studies along with further information are summarized in\n",
              "  . Compared to machine learning (ML), deep learning (DL) methods enable the automatic feature extraction and feature engineering. Convolutional neural networks and recursive neural networks have been widely used in bioinformatics studies with successful results such as in Covid-19 , , , . Since deep learning models are good at finding the intricate underlying patterns in data, thus they are suitable to large scale unstructured data such as electronics health records (EHR), whole proteome, and human genome leading to high chances and suitability to be applied in life sciences applications such as drug repurposing . In comparison to state of art machine learning methods, deep learning is distinguished by the flexibility of neural networks . Deep learning methods have also been widely used in drug repurposing against Covid-19. A few well-known studies are given in .</p><p id=\"p0090\">A recent work involving drug repurposing in Covid-19 using deep learning is given in  and are given in . Here, deep learning model named as Molecule Transformer-Drug Target Interaction (MT-DTI) was used for predicting novel drug target interactions (DTIs). It utilized chemical structure SMILES and amino acid sequences (as a chemical-protein pairs) as an input obtained from BindingDB and Drug Target Commons (DTC) databases. MT-DTI predicted several commercially available or in practice anti-viral drugs having potency against Covid-19 protein targets  confirmed by the pre-clinical validation assays of the Covid-19. MT-DTI model was deployed for the selection of compounds from 1400 FDA approved drugs having the strong affinity against the host cellular targets for viral diseases . Another deep learning-based drug repurposing study done in  used deep learning on a massive knowledge graph to identify and select promising anti-Covid-19 drug candidates. The knowledge graph consisted around 15 million edges with 39 types of relationship connecting the whole bunch of networks together. Deep learning was primarily used in this study to learn representation of graph entities leading to 41 repurposed drug candidates. These drugs include dexamethasone (an anti-inflammatory) agent having bioactivity against Covid-19 and Niclosamide  validated by proteomics of Covid-19 infected cell and gene expression.</p><p id=\"p0095\">Some studies have adopted hybrid strategy where deep learning is coupled with molecular simulation to further prioritize or performing the screening procedure for the predicted drug target interactions as shown in (b-c). One such study is done in  which focused on RNA-dependent RNA polymerase (RdRp) targeting candidate drug molecules from 1906 FDA approved drugs. Drug candidate such as pralatrexate  and azithromycin  predicted from this study were validated in in-vitro testing and found to inhibit Covid-19. First drug was active once the virus entered host cells while the later mentioned drug has efficacy in before and after stages.</p><p id=\"p0100\">Additionally, software tools (MolAICal) are also developed which combine the deep learning models along with classical algorithms to find the drugs which interact effectively with 3D pockets of proteins  (\n",
              "  (c)). Deep learning enabled framework, MolAICal, were trained using the approved drug fragment data available in PDBbind and drug like, bioactive molecules available in ZINC. To demonstrate the drug design features of the MolAICal, membrane protein glucagon receptor and Covid-19 main protease were modelled . Other tools include Deep Docking (DD)  platform developed using deep learning which avails the QSAR model good at performing virtual screening to find the potential drug candidate or lead like molecule from ZINC database. Using the proposed method in study  led to identification of 1000 top ranked potential drug compound supposed to have efficacy against Covid-19 Mpro.</p><p id=\"p0105\">Deep learning methods (such as (b)) has number of advantages in comparison to traditional methods such as ability of DL models to model and learn complex features automatically. DL has helped in finding the repurposed drugs against Covid-19 and in other pathogens however certain challenges still exist. First to mention is the black box nature of DL models, lack of interpretability, output is dependent on the input data, and large standardized and dataset of biochemical data is necessary to gain the better learning and performance outcome of the system. Although there is huge amount of data already available but mining and making the standard data ready for ML and DL applications is still challenging. Hopefully, the large dataset which will be developed in future will make it possible to ensure the standard characterization of the deep learning-based drug discovery and drug repurposing models , .</p></sec><sec id=\"sec0035\"><p id=\"p0110\">Networks are empirical and versatile data structures and now a days are widely used in drug repurposing studies as shown in . Myriad number of concepts in the field of biology are represented using networks and graphs . In these approaches data is collected and the networks are created ((a)) to infer the associations by using myriad number of statistical and computational approaches ((b)). Considerable research interest in network analysis on structural and functional level to has been developed to extract the relationships between entities in networks and their underlying biological properties . Thus, various promising insights comes from this emerging field of network medicine ,  where tools and concepts from network theory are applied. Network interaction concept is heavily used in biology since there exist the interactions between different biological phenomenon . Networks are composed of two components: nodes and edges. Nodes can be drugs, proteins, genes, and complexes etc. while the edges can be the interaction between them. Network edges can be directed , , , undirected  or weighted , . Mostly the quantitative information derived from high-throughput screening is used to construct the weighted networks where edge is represented by some numerical values. In directed networks, traversing is only done in the direction of the edge where in case of undirected networks traversing is possible in any direction unless it is connected.</p><p id=\"p0115\">Molecular networks are good at providing insights into contextual understanding of drug target and its corresponding mechanism of action . Network algorithms can easily achieve such targets by visualizing the various already added interactions in the network along with addition of newly found interactions with superimposition of additional properties . The networks can handle the heterogenous data collected from various data sources. Thus, analyzing the network has now become a diverse area to integrate multiple sources like publicly available datasets and high throughput data .</p><p id=\"p0120\">The perfection of these approaches is still partial since these methods suffer from lack of knowledge on molecular interactome often leading to false results , , . Moreover, the available interactomes provide the static results of the biological process in contrast to dynamic systems in real . Despite these limitations previous efforts have demonstrated the potential and efficacy of network-based approaches and has been proved many times in studies like drug-target interaction predictions. Thus, they can also be used as drug repurposing studies. Moreover, networks can be categorized based on the data source used to construct them such as molecular interaction networks, gene regulatory networks, metabolic networks, and protein-protein interaction networks etc. .</p><sec id=\"sec0040\"><p id=\"p0125\">In network biology, interactions between various biological concepts are presented using networks. Networks are made of nodes and edges, nodes can be various components such as genes, proteins, complex phenotypes such as diseases, and atoms etc. whereas edges can be interactions between proteins, genes, or complex phenotypes to show relationship or functional interaction and similarity between them. Moreover, multiple relationships ((a)) between two nodes can also be represented using edges at the same time , . Further, the nodes and edges can be directed , , undirected , and weighted  with qualitative and quantitative information to emphasize a proper concept . Systems biology contains various molecular interaction which are used to represent distinct molecular interactions depending upon the nature of interactions. Molecular networks are good at providing scenario based valuable insights into drug-targets interact and thus can be used to understand the mechanism of action of a drug .</p><p id=\"p0130\">Moreover, numerous studies have demonstrated the practical use of networks in drug repurposing . Network algorithms are developed to achieve tasks such as visualization of the graphs and interaction between them while adding and imposing the newly discovered relationships, additional properties, and connectivity over the primary component and their known interactions . Various type of data collected from heterogeneous data sources can be represented in one network using multiple node and edge arrangement. Due to the diversity and abundance of information contained in networks the analysis of the networks has become a versatile platform for integrating and analyzing multiple sources of high-throughput data for useful insights . One survey providing the detailed use of graph theoretical techniques is provided in .</p></sec><sec id=\"sec0045\"><p id=\"p0135\">Gene regulatory networks use the transcriptomic data which provides the information and insights into drug function by capturing the dynamic properties of the cell . Few of the most recent gene regulatory network-based studies are summarized in\n",
              "  . Gene expression-based drug repurposing studies hypothesize that drugs with similar gene expression signatures would target the same proteins . Moreover, Gene expressions change substantially when disease occur specifically the amount of RNA transcript for dysregulated genes changes considerably. Differential gene expression analysis can be performed to analyze the difference between genes expressions in disease and control samples . With the help of technologies now available such as microarrays, gene expression profiles rich source of disease data is easily available which can be used for multiple purposes. Differential gene expression profiles are used as input data in many drugs repurposing studies to prioritize the drug targets  because many drug targets work as transcription factors that are likely to regulate the expression of the genes.</p><p id=\"p0140\">There are many studies who used gene expression data for drug repurposing based on network-based approaches as shown in . Additionally, one such study which used neighborhood scoring, interconnectivity, and propagation random walks methods is presented in . Gene expression data used in this study was downloaded from Gene Expression Omnibus (GEO)  repository. The conducted study was confined to case studies such as scleroderma, different types of cancer and type 1 diabetes with results evaluated using area under the curve (AUC). Another network-based study using gene expression data used maximum flow method to repurpose the drugs for prostate cancer . The data developing network was downloaded from DrugBank, OMIM , KEGG , and PGDB . The evaluation criteria were set to mean, precision, and average position (AP). NFFinder  presented statistical analysis method to repurpose the drugs for neurofibromin and the data for this study was downloaded from GEO, CMAP , and DrugBank. Other studies include Bayesian networks , virtual gene technique Bayesian networks , Kolmogorov Smirnov enrichment , and functional linkage network  etc. developed for repurposing in different disease domains.</p><p id=\"p0145\">Despite being effective, gene regulatory network-based drug repurposing suffers from some limitations. First, it is difficult to define and set a robust gene signature because of the random nature of some gene expression data which ultimately results in biased extracted responsive networks . Second, the drug targeted genes and regulatory genes are not always found differentially expressed. So, the assumption that rests on the notion that mathematical optimum from the responsive network coincides with maximal biological relevance does not stand true. Third, proved from the previous studies, network topological characteristic points out that there is no significant relevance between potential target proteins and significant points in network . Fourth and final, the biological network and living organism’s response are complex to map thus estimated gene networks are often used to rely on . These limitations can be overcome by integration of information from multiple sources and finally a robust drug repurposing framework can be developed based on gene regulatory networks.</p></sec><sec id=\"sec0050\"><p id=\"p0150\">Metabolic networks are interconnected pathways depicting biochemical reactions taking place within living cells through involvement of metabolites with necessary compounds . In metabolic networks, metabolite and chemical compounds are represented as nodes. Whereas the edges represent the relation between the metabolites and compounds. Direct edges show the reactions done by one or more enzymes. In some other networks, nodes denote the compounds and metabolites whereas edges capture the reaction between nodes. Thus, a bipartite graph with directed edges is provided connected two distinct entities (nodes). Implying, if a metabolite is direct result of a reaction, then a directed edge from compound to metabolite using reaction notation denotes the relationship. In other words, if there is an edge from metabolite to reaction it implies that reaction is activated by metabolite. Once represented, metabolite networks are subject to topological feature analysis to capture different kind of relations and uncover repurposing opportunities. Additionally, positive change in compound concentration caused by specific enzymes may be the cause of disease and drugs are used to regulate and adjust the concentration of compounds cause for disease , .</p><p id=\"p0155\">Moreover, in metabolic networks flux-based analysis (FBA) methods are used to identify the drug targets. FBA and other similar methods aim at prediction of essential enzymes critical to survival and growth of pathogens. In practical, on metabolic networks FBA model was developed in  to predict the DTIs. The data used in the two-step study was retrieved from KEGG database followed by drug target prediction for human hyperuricemia. Another study done in  employed the greedy heuristic search approach to predict the DTIs for lunch cancer using metabolic networks. Whereas the methods describing anti-microbial DTIs identification were reviewed in  that used FBA modelling on metabolic networks. Advantages and disadvantages of methods are discussed clearly in.</p></sec><sec id=\"sec0055\"><p id=\"p0160\">Protein-protein interactions (PPIs) are required in almost every process in cell which emphasizes that the need for understanding is crucial for cell physiology in normal and disease states. It is essential in drug development because drugs have effect on PPIs. Protein interaction networks have proved useful in drug repurposing studies over the time (), since their interactions determine molecular and cellular mechanisms, which control healthy and disease states in organism . Therefore, with the help of such networks pathogenic mechanism of disease can also be understood. Some PPINs based repurposing studies are listed in . Moreover, confining this study to drug repurposing it can be stated that PPIs depicts the interactions between known druggable targets and other proteins. This interaction between the proteins can be direct or indirect . The central assumption used in drug repurposing when focusing on PPI networks is that proteins which are targeted by similar drugs are functionally related in networks or if the drugs are similar, they are supposed to treat same disease and vice versa. It is also believed that the topological analysis of the PPIN can provide useful insights in drug-target predictions as proteins function in form of interaction mostly .</p><p id=\"p0165\">Owing to the importance of PPIs, many repurposing studies have been successfully done. One such study using PPIN for drug repurposing is done in  which utilizes human PPIN obtained from UniHI and maps it with DTIs obtained from DrugBank and GeneCards databases. SVM, along with L1-regularized logistic regression and k-nearest neighbors’ methods are used for drug repurposing. Z-score and standard deviation are used as an evaluation criterion to judge the performance of developed models. Other study is presented in  predicting the drugs for metastasis in colon cancer. Here data was downloaded from GEO and set cover-based formulation of coordinate dysregulation in complex phenotypes was used followed by harmonic mean method as an evaluation of the predictions. Additionally, Parkinson’s disease was predicted by analyzing crosstalk in the network by betweenness centrality. The data here was downloaded from KEGG, OMIM, and iRef databases.</p><p id=\"p0170\">Despite progress and proving useful, PPIN based drug repurposing studies experiences some disadvantages such as, PPIs obtained as the part of experimental sources contains links which are not fully characterized in detail. Moreover, the data obtained are noisy and incomplete thus there are higher chances for network being ended up as biased extracted network.</p></sec><sec id=\"sec0060\"><p id=\"p0175\">DTI prediction is the basis of drug discovery and drug repurposing as shown in ((a)). Many of the studies show that the drugs have off-target effects and the process of drug repurposing become simplified if the targets are predicted accurately. Computational prediction of DTI makes discovery process fast, economical, and maximizes the chances of success whereas experimental validation of DTI is both time consuming and expensive. This emphasizes the necessity of development of computational potential methods to predict DTI accurately  and some of the well-known DTI based repurposing studies are listed in . Many network-based methods for predicting DTI are present. Bipartite or tripartite interaction models are created where nodes denote the drugs and targets, and edges show interaction between them in bipartite networks and in case of tripartite networks nodes represent drugs, targets, and disease and edges denote interaction or relationship between them.</p><p id=\"p0180\">Some of the notable network-based DTI prediction studies are described here in detail (). One such study is done in  which presents a probable soft logic method for DTI network analysis. The data was collected from the DrugBank, KEGG Drug, and DCDB databases ( and ). The evaluation of the network performance was judged using the AUC, AUPR, precision ((b)), and other bioinformatics tools ((c)). Additionally, recommendation algorithm was used in ,  where drug-based, target-based, and network-based similarity inferences were drawn to predict novel DTI for drug repurposing. The predicted drugs were pre-clinically validated in in-vitro binding assays. Another study used supervised bipartite graph learning approach  to predict enzyme interaction and GPCRs interaction network. For DTI prediction kernel regression-based method was used and method performance parameters used in this study were ROC and AUC etc. Other network-based DTI prediction studies for drug repurposing include bipartite local model (BLM) , neighbor-based BLM integrated inference , statistical analysis of topological measures , network consistency-based prediction method (NetCBP)  and more.</p><p id=\"p0185\">Network-based DTI prediction methods have widely been used because of their good performance and various methods as discussed in above paragraph. Instead, there are some limitations of these approaches such as in-ability or poor performance of the models in case of novel DTI prediction. Other limitations include, low amount of known drug-target interaction data available, poor dataset balance, selection of negative samples is difficult or not possible since there is either no or less negative DTI data available from experiments . Possible solutions for the said problems are suggested by many studies such as in  where they initiated the process by targeting, drug targets of highly similar drugs to new targets and vice versa for the drugs. Solving a problem leads another problem in this study, which is how to define the highly similar drugs or highly similar targets. Additionally, the problem of scarce data samples was solved by introducing the randomization technique for balancing data samples as in .</p></sec><sec id=\"sec0065\"><p id=\"p0190\">Similarity based drug-drug interaction shows the association between two drugs and talks about the aftermaths of the interaction. Drug-drug similarity is used in network-based drug repurposing to augment the network. Drug-drug interaction similarity can be constructed either as biological effect similarity, chemical similarity, side effect-based similarity and more. Specifically, structural similarity can be developed by different method, either by using 2D fingerprint similarity or 3D conformation of the structures. Having said that, let’s say, two drugs having similar expression patterns in cultured cells or lead to similar side effects then the drugs can be said to have interaction between them. Moreover, drug repurposing opportunities can be found using the chemical structures of the drugs. The hypothesis rests on the notion that, two drugs which are structurally similar are supposed to treat the same indication by targeting the similar protein targets and disease pathways. Whereas this hypothesis does not always hold true, however structural similarity have been used in successful drug repurposing studies  which motivates to deeply investigate these approaches in future.</p><p id=\"p0195\">Moreover, one drug-drug similarity-based study is given in  where two chemical compounds were compared based upon the 2D graphs. Atoms in this study were designated as nodes while bonds between atoms served as edges. This study was based on the basic hypothesis that two structurally similar drugs share the same bioactivity . Chemical similarity-based studies often extract multiple structural features and then compare the drugs one-on-one based on these features. Thus, if the hypothesis returns true, drug repurposing opportunities can thus be availed using chemical association, biological means, side effects, or by searching some known biological features such as known targets . Another similarity-based drug repurposing study using network approach is given in  which presents a method called SITAR and utilizes five similarities between drugs and targets followed by MANTRA  which studies the mechanism of action of the drug using network-based approach.</p><p id=\"p0200\">Uncertainty in drug-drug similarity and chemical properties of the known drug compounds leads to limitation of the drug-drug interaction-based drug repurposing studies. Additionally, there are some physiological effects which are hard to capture while using the structural features alone thus it leads to the use of multiple similarity measures to give a broader perspective like done in . The similarity-based strategy is typically employed by drug-drug interaction prediction with most similarity-based algorithms relying only on nearest similarities overlooking some of the important similarity measures.</p></sec></sec></sec><sec id=\"sec0070\"><p id=\"p0205\">Human coronaviruses (HCoV) include severe acute respiratory syndrome coronavirus (SARS-CoV) and novel 2019 coronavirus (Covid-19 also referred to as SARS-CoV-2). SARS-CoV is pathogenic coronavirus that is known to emerge from zoonotic reservoir leading to global dissemination of the virus . Recently, novel coronavirus (2019 nCoV) has caused global pandemic with high morbidity and mortality rate. Instead of highly accelerated efforts in the field of drug research, until now there is only drug approved for the treatment of Covid-19.</p><p id=\"p0210\">Moreover, de novo drug development is time taking, costly and risky process. It takes 12–15 years with $2~$3 billion of the investment to develop and new drug and bring it to market , , . An effective and timely possible approach to quickly response the drug needs in case of emergencies is known as drug repurposing. Drug repurposing is the process of finding the alternative uses of already approved drugs , , . Over the period, multiple techniques based on machine learning, deep learning, mining, and network or graph-based approaches have been widely used to repurpose the drugs for a Covid-19.</p><p id=\"p0215\">Herein, we provide the comprehensive review along with benefits, challenges, and future directions of network science-based drug repurposing studies specifically for the Covid-19. The studies are divided in to three categories based on network proximity, network diffusion, and artificial intelligence (\n",
              "  , , ).</p><sec id=\"sec0075\"><p id=\"p0220\">Network proximity-based drug repurposing studies use the proximity measures such as network centralities, node degree, and more for identifying and providing the key protein-protein, drug-drug, or drug-target associations to repurpose the drugs (a (i)). One of the recent studies done in  presented a powerful network-based drug methodologies for identification of potential repurposable drugs and their combination targeting Covid-19. Presented anti-viral and integrated repurposing methodology mainly quantified the interplay between HCoV-host proteins and drug targets in human protein-protein interactions (PPI). Use of network proximity analysis as shown in (b) (i) between HCoV-host targets and drug targets in human PPI led to prioritization of 16 anti-HCoV drugs potentially repurposable. Further validation  was performed by enrichment analysis (c (i)) of drug gene signatures (). Moreover, potential drug combinations were captured by complementary exposure pattern. Hypothesis of the study was that, for successful replication of viruses and infection host cellular factors are required, thus systematic study of human PPI (a (ii)) and virus hosts can present an effective method toward understanding the viral mechanism , . Despite successful results, this study suffered from limitations such as, lack of known host proteins on Covid-19, a low binding affinity value of 10 μM was used to specify the DTI. Whereas lower binding energy up to 1 μM or less could be a more effective cut off. Another network proximity-based drug repurposing study is given in  which used network medicine-based approach to repurpose drugs in Covid-19. This study enriched the PPI with transcriptomics and proteomics and then performed network proximity measurement where broach Covid-19 associated underlying pathogenesis of disease manifestation was explored. A further step was added, and network-based predictions were prioritized for treatment by combining predictions with propensity score (PS) matching 26,779 individuals from observational study. Here, the associations between 6 different diseases (consisting of autoimmune, malignant, cancer, cardiovascular, metabolic, neurological, pulmonary) and Covid-19 were created. Created network was analyzed followed by the single-cell RNA sequencing to uncover the underlying pathobiological and pathophysiological relationships between Covid-19 and its associated treatments. Moreover, like  in this study z-score was used to measure the closest proximity distance between pair of interaction proteins and all those with z &lt; −1.5 were considered closest ones. Although not significant, network proximity results in this study showed that rheumatoid arthritis has small network proximities (negative <italic>Z</italic> scores) across all 5 SARS-CoV-2 datasets. For checking the potential of repurposed drugs to reverse the expression at proteomics and transcriptomic level mutated by viruses Gene set enrichment analysis was performed to evaluate each drug. Next, the drugs were prioritized based on combination of factors such as strength of network based and bio-informatic based predictions, literature evidence of the prioritized drugs, meaningful evaluation by enough patient data and more. This type of muti-layered validation of the network predicted drugs find out the key repurposable drugs supposed to have potential to reverse the expression of Covid-19 expressed genes. In total 34 drugs (<italic>Z&lt;</italic> −1.5 and <italic>P&lt;</italic> 0.05, permutation test) (b (i)) were computationally repurposed with the SARS-CoV-2 datasets (SARS2-DEG, SARS2-DEP, HCoV-PPI, and SARS2-PPI) using the above-mentioned criteria. These drugs were significantly proximal to 2 or more SARS-CoV-2 host protein sets. We manually curated their reported antiviral profiles. Additionally, Knowledge of the complex interplays between SARS-CoV-2 targets and human diseases indicate possibilities of drug repurposing, as the drugs that target other diseases could potentially target SARS-CoV-2 through the shared functional PPI networks. Moreover, further information about the databases, toots, and techniques used in these studies are listed in . Some studies are combining different network approaches for improved results such as done in  which aims to captures the bigger picture of human molecular landscape of Covid-19 infection by exploiting the network medicine approach as strategy to repurpose the drugs for Covid-19. As a starting point, experimentally validated host proteins of the Covid-19 known to have viral interaction, tissue specific gene expression data, were explored by use of techniques such as connectivity significance, network diffusion, and network propagation. This process was followed to rank genes followed repurposing drugs targeting the ranked genes as a treatment for Covid-19. The prioritized collection of genes which served as key drug targets for repurposed anti-Covid-19 drugs responsible for causing pharmacological effect resulted from normalization and aggregation of different scoring methods. Ranking of genes focused on the contrasting the different phases of the virus infection and viral replication of the cycle. Further, in this study, drug repurposing was carried out by drug-gene interaction data obtained from DrugBank database. DrugBank was manually investigated to find out the drugs having direct and indirect interaction to virus host proteins. A set of criteria was made to find the restricted list of genes (18 genes) and related drugs. Criteria include top ranked genes in VarElect ranks of groups G124, G12345 etc. followed by main cellular location of target proteins, RNA replication, and translation, and finally the existence of drugs already approved as potential candidates for repositioning. Moreover, another study called SAveRUNNER  presents a novel network similarity-based drug repurposing framework. It first creates a bipartite drug-disease network between drug-targets and disease-causing genes, and it uses network based new similarity measure that prioritizes the associations sharing the same neighborhood. PPI data gathered from 15 commonly used databases was retrieved from . The data contained 217,160 PPIs between 15,970 unique proteins. Additionally, disease gene associations were obtained from Phenopedia  disease-centered view of genetic association studies by online Human Genome Epidemiology (HuGE) encyclopedia . Moreover, a total of 1875 FDA approved drugs were selected for analysis from DrugBank and Therapeutic Target Database (TTD)  databases. Whereas presented algorithm is six step procedure which takes the list of drug-targets and disease genes as input followed by the proximity and similarity computation which leads to proximal drug-disease associations. Clustering was performed on the drugs and quality score was computed to adjust the network similarity and finally normalization of similarity score was carried out using sigmoid function. This study utilized many useful steps from creating the network to outcome of repurposed drugs which ensure the robustness and usefulness of the approach. Additionally, another network-based repurposing approach for Covid-19 done in  was conducted by integrating GWAS and text mining the data. Here lung-specified gene subnetwork enriched with Covid-19 gene associated data obtained from GWAS was created to screen 220 FDA approved drugs obtained from DrugBank database. Their drug-perturbed gene expression profiles were obtained from LINCs and lung-targeted, drug-focused gene modules were created. This led to prioritization of the 13 FDA approved drugs as a potential treatment for Covid-19. Findings provide the timely and valuable insights to repurposed use options of the drugs to be further explored in Covid-19. Another very distinguished and interesting work in the field of network science and its applications to drug repurposing (Covid-19) was done in . Here in this study multiple layers or approaches normally applied individually on the networks were fused into a single algorithm. Deployed algorithms consist of artificial intelligence, network diffusion, and network proximity. Each of them was utilized to prioritize the 6340 drugs for expected efficacy in Covid-19. This led to a successful approach as a network medicine framework for identifying repurposing opportunities in Covid-19. In practice one cannot find a single algorithm that consistently performs across all the datasets which served as a motivation for this study. A fundamental hypothesis was created that multimodal approach that combines the predictions from all approaches, based on the consensus among all the approaches, combined approaches always exceed the individually best performing models. In result of this approach 918 drugs were prioritized based upon the predictions from all algorithms and were subsequently tested in wet lab experiment. As 77/918 showed the anti-viral effects in cultured non-human primate cells. As a step further all those 77 drugs were tested in human-derived cells and 76/77 drugs showed the efficacy against viral disease. 96 % of the drugs tested in the study were FDA approved which is beneficial as these drugs could be directly moved to clinical trials. A real predictive power of the approach was demonstrated by the positive outcomes of the second human screen with success rate of 62 % followed by identification of six drugs potentially easily repurposed for Covid-19. All in all, the approach presented here did not only provided an effective strategy and results in the repurposing of Covid-19 drugs but also provided an algorithmic toolset as well. The toolset could be readily deployed in future for diseases left untouched due to huge cost and time constraints associated with it. As only 918 of the 6340 drugs prioritized by CRank were screened, a selection driven by compound availability, many potentially efficacious FDA-approved drugs remain to be tested. Finally, it is also possible that some drugs that lacked activity in VeroE6 cells may nevertheless show efficacy in human cells, like loratadine. Additionally, despite the tremendous progress in the repurposing of the drugs, repurposing a combination of drugs with varying mechanism of actions remains an undefeated feat. Since drug combinations are widely used to fight many deadly diseases effectively. Severe and rapidly emerging replications of the Covid-19 made the world realize for the combination therapies that were expected to response more effectively. The ability to repurpose the drug combination was hampered by the availability of large range of possible pairs. To address this gap a study was done in  where an integrative network pharmacology inspired web-based tool named COVID-CDR was developed for in-silico repurposing of the drug combinations. Deployed web-based tool can visualize the interaction in the interactome on the cellular level involved in the mechanism of action of the drugs involved. COVID-CDR could prioritize Covid-19 repurposable drug combination for synergistic effect. In this approach, drugs with targets sharing the topological proximity with Covid-19 targets are used to prioritize the drug combinations based on drug-target separation. Previously it was hypothetical presented that different drug-targets lead to different footprints. Combination sharing the commentary exposure can be considered as putatively effective in this case. Whereas, successfully evaluated drug combinations because of COVID-CDR were evaluated using the network computation methods. Network proximity measures are calculated using the subnetworks of protein interactome that were active context of interest. Since this platform is not automated thus it can be further improved by automating the screening of drug-pairs created by multi-object optimizers. Finally, in the light of studies presented, it can be said that network proximity-based approaches have proved very useful for drug repurposing applications, possessing high flexibility and diversity. However, mere use of proximity measures to analyze the network may or may not present significant results but when combined with additional validation approaches such as GSEA, molecular docking, and pathway enrichment could present more putative yet diverse framework.</p></sec><sec id=\"sec0080\"><p id=\"p0225\">Network-based approaches play very important role in drug repurposing studies and few of the recent studies are listed in . Artificial Intelligence algorithms are widely applied on network data for variety of prediction tasks as shown in (b) (ii). These tasks include node prediction, link prediction, or graph prediction (b (ii)). These predictions can be applied to variety of applications and drug repurposing is one among them. In this section some of the recent studies, where AI on networks has been applied to repurpose the drugs, are presented. One among these studies is done in  in which a Covid-19 based knowledge graph was built. The knowledge graph consists of virus baits, host genes, pathways, and other information. To infer the representation of the candidate drugs, deep graph neural network approach based on biological interaction was used. Candidate drugs were later ranked based on the clinical trial history (c (iv)) to prioritize them followed by the validation using the genetic profiles and in vitro experimental efficacy (c (iii)). Use of deep neural networks and integration of the diverse interactions helped in quick identification of drugs repurposable for Covid-19. This study suffered from the limitations of missing the effective drugs possibly during the initial drug ranking. Since the initial group of 3635 drugs were taken as candidate. The validation of the drugs was carried out using the population based retrospective EHRs analysis and EHRs are incomplete and prone to errors in comparison to randomized experimental data. Other limitations include the inconsistency between the gene sets. Another Covid-19 based drug repurposing study using the Few-Shot learning is done in . This study introduces the multi-targeted drug repurposing strategy which is taken as most efficient approach to diseases such as Covid-19. Multi-targeted compound screening is hampered by the lack of good quality experimental evidence and information extraction from molecules. Recognizing this gap, MOLGNN, a deep learning model was proposed and presented which enabled the prediction of molecular property. Chemical molecule embedding was performed by using graph neural networks (GNNs). GNNs were used to model the ability of the drugs to target proteins and their interaction. Let’s say <italic>G</italic>\n",
              "   = <italic>(V, E)</italic> is a graph representing the nodes connected with edges. V represents the vertices or the nodes whereas E represents edges between the nodes. In connection with the present study, molecules are represented as graphs (G), nodes are items and bonds between them are the edges. Following this, an input data comprising molecules as graphs in the form of adjacency matrix <italic>A</italic>, was constructed for this study. This dataset contained more than 2 million unlabeled molecules obtained from ZINC database. The deployment of the trained models and downstream classification ((c)), Covid-19 related chemical molecules were used. The models trained with <italic>MolGNN</italic> performed well when applied to a small labeled fine-tuned dataset. This led to a potentially powerful method called few shots learning. This study provided a powerful tool for future research in drug repurposing specifically in case of poorly known diseases such as Covid-19. Instead of these benefits, there are certain limitations such as uncertainties in the datasets used and lack of validation of the predicted drugs in corresponding in-vitro and in-vivo models for efficacy testing in Covid-19. Moreover, multiple machine learning models are also used disease agnostic fashion for network datasets as done in  for repurposing targeted drugs in Covid-19. Used models are complementary covering Covid-19 and host factors. It starts with ranking of drugs matrix factorization on anti-viral first followed using graph kernels to rank drugs based on perturbation induced in subnetworks. Manually curated datasets containing drug-virus associations with known status obtained from  were used. Broad-spectrum analysis that started from in-vitro validation spanned over the animal models and clinical study. Furthermore, results suggest that the drugs ranked using anti-viral of the top predicted broad spectrum were found in clinical practice for Covid-19 and those from graph kernels were found experimentally validated. Additionally, online repurposing tools are also widely deployed recently and have proven useful in many cases. Similarly, here Covid-19 drug repurposing explorer abbreviated as CoREx was deployed as an interactive tool. This can be used to observe and understand the interaction between drugs and host Covid-19 targets in context of biological networks, clinical use of drugs, protein functioning, and cMAP. Despite the contribution and benefits of CoREx tool there some limitations such as the matrix decomposition approach can only be applied to the drugs for which association with viral disease is known. Main limitation of the study is the annotation of the drug-virus pairs based on the developmental stages and the related data is hard to find. Similarly, the other part of the study (Network medicine) has limitation that only those drugs for which targets are known can be used. Moreover, feature information is very important machine learning is involved. In this connection study emphasizing the informative features of network in machine learning based drug repurposing are conducted in . Human PPI, Covid-19 specific network, and DTIs were first created like most of the studies and later essential proteins were passed through a set of algorithms (Machine learning). ML step was followed by clustering which served as an important step in extraction of informative features. Enrichment analysis was then performed for these informative features to validate them in case of Covid-19. This approach helps identify key information followed by validation which minimizes the false positive or false negative rate of the final prediction. In addition, integrative approaches are used in drug repositioning and are found to perform more effective as compared to single or standalone approach. This follows the notion that, weaker when joined come together stronger. One such graph machine learning based drug repurposing study in Covid-19 is presented in  which avails the ML and statistical approaches to map and mine the knowledge graph, transcriptomic and literature data on the large scale to uncover the potential drugs for Covid-19. The computational approach in combination with pre-clinical validation indicated that a poly-ADP-ribose polymerase 1 (PARP1) inhibitor, CVL218, currently in Phase I clinical trial, may be repurposed to treat COVID-19. To observe all these predictions virus-centered knowledge graph was created and mined using the mining algorithm. In realization of the network seven different networks were considered consisting of DTI network, an interaction network of drug-target (virus), a human protein-virus protein, and more. The databases such as DrugBank, Uniprot, TTD, ChEMBL, PubChem, and IUPHAR_BPS , Binding DB , and GHDDI  were used. One other study that used DrugBank and other databases along with graph convolutional networks (GCN) is published as  but it consists of analysis of multiscale virus host interactome resulting in off-target prediction of the drugs. Coupled with cell-based experimental validation this study came up with several clinically efficient drugs as potential repurposing candidates found as a result GCN application to Covid-19 related data. Specifically, as a part of the mentioned study, capmatinib (MET inhibitor) was found to have potent and wide range antiviral activity in multiple coronaviruses. The antiviral activity was observed in MET-independent manner and additional roles of capmanitib for host cell proteins promoting the infection caused by human coronavirus thus it can inform further drug discovery studies. Thus, AI assisted network-based drug repurposing strategies present effective strategy along with many benefits such as time and cost efficiency and high reliability. Additionally, the use of GNN and GCN have opened the new doors of opportunities in network-based drug repurposing strategies and at the same time these approaches also suffer from limitation. Since like the other AI methods, GNN and GCN also need data and in most cases balanced data to be trained properly and biases can be avoided without compromising on the accuracy of the models. Additionally, there is no doubt huge amount of the data is being generated and is publicly available but there are chances for the data to contain uncertain information ultimately leading to improper and vague model training. Realizing these problems and tactfully handling them can unveil the unfathomable power of the GNN and GCN approaches being applied to network-based approaches to drug repurposing in Covid-19 and in general as well.</p></sec><sec id=\"sec0085\"><p id=\"p0230\">Diffusion is a process through which information related to genes, proteins, viruses, and other entities diffuse across the drug-target, drug-drug, disease-disease, or social networks , , , ,  as shown in (b) (iii). Some of the related studies are given in . A simple notion behind this concept is that behaviors such as information spread in environment, modeled by construction of simple graphs called <italic>Regular Graphs</italic> (GRID-based). There are multiple significant studies utilizing this approach to repurpose the drugs in Covid-19 and other diseases. Once recent Covid-19 based drug repurposing study is done in  where gene co-expression-based network for 1441 genes was constructed and analysis was performed. The analysis resulted in two significant gene co-expression network modules and then data related to miRNAs and transcription factors targeting co-expressed modules’ genes were gathered from reputable databases led to the establishment of two subnetworks consisted of transcription factors and miRNAs. Subsequently, the list of drugs targeting the obtained subnetworks was gained from DGIDb database and two networks were constructed again. Finally, because of proposed network analysis, five drugs were repurposed as candidate repurposable drugs for Covid-19 treatment. Additionally, around 10 miRNAs were found to be significant to treat Covid-19. This study using diffusion has the capacity to integrate and analyze the heterogeneous data to support the predictions. Where similar integrative approach is followed in  to perform validation of repurposed drugs using a comparatively different framework which contains KEGG enrichment, molecular docking, STITCH search tool, and virus-host-drug interactome. These techniques were used to investigate the interaction network linked with Covid-19 and chloroquine and hydroxychloroquine were utilized as reference drugs to further get into the network. As a result, 47 drug targets were found enriched in TLR pathway and were found overlapping with Covid-19. Another step of molecular docking and dynamics in investigation led to confirmation of binding affinity of TLR9 to the mentioned drugs. Additional insights from human-drug protein map identified and further confirmed the interaction of Covid-19 with TLR pathway will provide key information for Covid-19 itself and its treatment. Thus, ensemble or integrative approaches aimed at investigation are useful and results in very fruitful insights. Moreover, all the tools, databases, and evaluation matrices used in the studies for proximity, AI, and diffusion-based networks are given in \n",
              "  <italic>,</italic>\n",
              "  , and  respectively. Furthermore, combination of diffusion and matrix factorization methods is also used  to prioritize the Anti-Covid-19 drugs. The drug prioritization is based on integration of multiple informative features such as virus sequences, established viral-drug interactions, drug chemical structures, and the application of logistic matrix factorization coupled with diffusion kernel on the graph data. Molecular docking was used for prioritization or ranking of the compounds (c (ii)). Here, the gaussian models were first built followed by virus sequence and chemical structure similarity kernel are designed based on the biological features of identity matrix. Then the gaussian and similarity kernel are diffused and finally the logistic matrix factorization model with kernel diffusion is proposed to identify potential Covid-19 drugs. Finally, to validate the drugs molecular docking was performed to find the binding affinity of the drugs potentially repurposable with Covid-19 spike and ACE2 proteins. The algorithm is named as VDA-KLMF is compared with recent relevant studies (prediction and classical models) based on the 5-fold cross-validation. This cross-validation was performed on viruses, drugs, and VDAs and as a result best results (recalls, AUCs, AUPRs), significantly outperforming the other studies were obtained. Thus, it leads to conclusion that integration of various biological data features and suitable models, or kernels lead to development of tools that may assist in drug development for Covid-19.</p></sec></sec><sec id=\"sec0090\"><p id=\"p0235\">Drug repurposing has proved as an effective alternative to costly, time consuming, and risky process of de-novo drug development. Due to the sudden emergence of Covid-19 pandemic drug repurposing efforts helped greatly to make the treatments available to the patients faster. A large community of the researchers and scientists have been working on the different type of approaches to drug repurposing such as earlier defined: machine learning based, deep learning based, and network-based repurposing. These methods have made it possible for the drugs to enter the clinical trials to fight against Covid-19.</p><p id=\"p0240\">For example, a network-based drug repurposing study done in  predicted the 16 repurposable drugs with anti-viral evidence obtained from the literature and clinical trial data. Among these predicted drugs Sirolimus (originally approved as an immunosuppressant) is registered in phase-II clinical trials for Covid-19 . Another work done in  applied machine learning on networks and repurposed 14 potential repurposable drugs for Covid-19, of which Chloroquine is registered in Phase-II &amp; III  and Dexamethasone is registered in Phase-IV clinical trials  as mentioned in\n",
              "  . Similarly, a machine learning and network medicine drug repurposing work  retrieved a list of 23 FDA approved drugs as potential therapeutics for Covid-19. Many of these drugs are already in clinical trials for Covid-19.</p><p id=\"p0245\">Additionally, in our recent work of machine learning and molecular docking-based drug repurposing framework , 12 potential drugs were suggested as potential therapeutic for Covid-19. Out of 12 potential drugs, Ebselen was found under trial in Phase-II  whereas Artemesian and Nitrendipine are in Phase-IV clinical trials  as shown in . Similarly, multiple deep learning techniques have also been used to identify the repurposable drug candidates for Covid-19. One of the deep learning-based studies, is done in  where deep learning-based drug screening was performed followed by a study done in  where a resource based on deep learning is created to enable the fast repurposing of the drug candidates for Covid-19. Finally, in the light of the examples tabulated in , it can be concluded that the drug repurposing methods are quite helpful and quick to response in case Covid-19 and related emergent situations. Moreover, as the time is passing, current methods are being improved and advanced while newer methods are introduced. Thus, drug repurposing methods will continue to help in fighting against pandemics.</p></sec><sec id=\"sec0095\"><p id=\"p0250\">There is no doubt that network medicine approaches are gaining more attention and have successfully been deployed in several studies to repurpose the drugs for Covid-19 (, , ). Still there are some limitations which needs to be highlighted along with future recommendations to overcome them for improved results in the future.</p><p id=\"p0255\">The limitations in general include the inconsistency in subnetwork extraction in biological networks due to the parameter and thresholding adjustment. Modified genes and gene expression changes are not captured which limits the graphs to estimate the off-target effects of drugs on the genes. Additionally, genes used as pharmacological targets do not always show the observable signature changes which contrast with the commonly adopted notion of mathematical optimum of response networks. Similarly, inability of networks to comprehend pharmacokinetic modifications and noisy side effects data of drugs make it difficult for structural similarities alone to be utilized for drug repurposing.</p><p id=\"p0260\">Moreover, there are different limitations and challenges associated with different types of networks. Such as problem of functional links that are possibly not defined as they determined through various experimental background, is associated with PPINs. So, the constructed network will be a biased network because of limited and noisy data. Furthermore, there is no easy method to link a response network to an entire organism’s reactions. Following other networks, DTI networks suffer from the limitations such as various structural and chemical properties of known compounds are inaccurate limiting the drug repurposing studies based on drug-drug similarity to unlock the true potential. Furthermore, structural traits alone cannot predict many physiological outcomes. To address these constraints, a variety of similarity measures, such as the one employed, can be used.</p><p id=\"p0265\">To overcome these limitations, vast biological data including molecular, pathological, structural, and online health community data should be utilized to construction of networks with multiple interactions to capture the bigger picture. To construct them, heterogenous data from various sources must be utilized to improve the reliability and reach of repurposing methods. Since the biological systems of complex and cannot be reflected by one aspect only, thus, one type of feature or data present the singular aspect of biological system. Single data type-based systems may not be reliable and use of integration of heterogenous data to capture the multiple aspects of biological system may omit the major limitations. Finally, for the standardized evaluation of the network-based approaches to drug repurposing a reference body should be created.</p></sec><sec id=\"sec0100\"><p id=\"p0270\">The Coronavirus disease (Covid-19) is an infectious disease caused by SARS-CoV-2 virus. Started in November 2019 in the Wuhan-China, Covid-19 was soon declared as public health emergency of international health concern by the World Health Organization (WHO) on January 30, and it scaled up to pandemic as of March 11, 2020. Due to the severity of disease and its volatile speed drug research was accelerated to help the patients and hamper the growing health concerns. Since de-novo drug development is costly and time consuming and on average it takes around 12–15 years with $2~$3 billion of investment to take a de novo drug from concept to market ready product. Thus, drug repurposing emerged as an efficient and scalable approach to find the treatment for Covid-19. In this review, a comprehensive overview of drug repurposing studies for Covid-19 is provided. The review spans to machine learning, deep learning, and network-based approaches while specifically network-based approaches to drug repurposing in Covid-19 are focused. Network based approaches for Covid-19 drug repurposing are further divided in three categories: Network diffusion, network proximity, and AI on networks. The databases, tools, and other resources followed by in-silico, in-vitro, and in-vivo validation studies used are also summarized. Moreover, in simplest form machine learning studies take the different type of drug and disease related data, perform feature extraction, create predictive models, and then models are deployed for the prediction in Covid-19 and other diseases. Despite successful application in drug repurposing for Covid-19, machine learning approaches struggle to handle the diverse and sparse biological data but are easy to interpret which is essential for studies like drug repurposing. In addition, deep learning approaches have also been widely adopted for repurposing studies have gained attention. Since deep learning models are effective in terms of heterogenous and sparse data handling and have been deployed to predict repurposed candidate drugs for Covid-19. Despite being successful, deep learning approaches suffer from lack of interpretability of predictive model and lack of data quality. Additionally, network-based methods have also been widely and effectively deployed for Covid-19 repurposing. Networks as per the evidence and analysis from literature have proved best tools for discovering the treatments by linking and investigating protein-protein, molecular, phenotype, or gene-expression or other data types. Studies have demonstrated that the network models have the potential to generate amazing and timely results and their true potential can be uncovered by overcoming the limitations of data inconsistency, problem of functional links, inability to capture gene expression variation, gene regulation, and lack of structure data etc. Despite these limitations, network models offer many advantages such as the interpretation of the prediction results that can be validated and visualized by mathematical means. The biological network-based studies in Covid-19 covered multiple pharmacological aspects such as gene regulations, protein interactions, drug-target or drug-drug interactions, drug-side effect, and metabolic interactions etc. The integration of such diverse biological entities extracted from many different databases and experiments in network helps to mimic the complex biological systems good for drug repurposing specific in Covid-19 and in general. Finally, in the light of recent literature network-based drug repurposing methods offer more advantages than traditional method by simplifying and accelerating the process of drug repurposing instead of the challenges and limitations.</p></sec><sec id=\"sec0105\"><p id=\"p0275\"><bold>Faheem Ahmed:</bold> Conceptualization, Formal Analysis, original draft, Writing – review &amp; editing, and images. <bold>Afaque Manzoor Soomro:</bold> Conceptualization, Writing – review &amp; editing, <bold>Chethikkattuveli Salih Abdul Rahim:</bold> review &amp; editing, Anupama Samantasinghar: managing the revision and editing, Arun Asif: managing the revision and editing, <bold>In Suk Kang:</bold> review &amp; editing, <bold>Kyung Hyun Choi:</bold> Conceptualization, review &amp; editing, Resources, Supervision, Funding acquisition.</p></sec><sec sec-type=\"COI-statement\"><p id=\"p0280\">The authors declare that they have no known competing financial interests or personal relationships that could have appeared to influence the work reported in this paper.</p></sec></body>],\n",
              " [<body><sec id=\"sec1-molecules-27-07732\" sec-type=\"intro\"><p>Drug discovery is a time-consuming and costly process that involves a lengthy assessment of clinical trials and has a success rate of only 2%, with an average cost of USD 2–3 billion [,], and also fails to deliver at the time of pandemics []. On the other hand, drug repurposing has several advantages over drug discovery, and one is that it can be delivered in times of pandemics. At the beginning of February 2020, the World Health Organization (WHO) declared the novel coronavirus disease 2019 (COVID-19) a pandemic due to its rapid spread as well as its high mortality and morbidity rates []. COVID-19 first appeared in Wuhan city, China, in late 2019 and is still spreading all over the globe creating a disastrous effect on global health and the economy [,]. Due to its highly contagious nature, as of May 2022, the pandemic has spread to more than 200 countries and territories with over 5.5 million confirmed positive cases and mortality of over 350K across the globe [].</p><p>The etiological agent of COVID-19 is severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), and it has been reported to originate from the animal coronavirus []. SARS-CoV-2 belongs to the genus Betacoronavirus of the family <italic toggle=\"yes\">Coronaviridae</italic> which already has two other members, i.e., SARS coronavirus (SARS-CoV-1) and Middle East respiratory syndrome coronavirus (MERS-CoV) which caused pandemic onslaughts in 2002 and 2012, respectively, resulting in severe human losses both in terms of morbidity and mortality [].</p><p>Coronaviruses are positive-sense single-stranded RNA with a genome size of ~30 kb containing a 5′ cap-structure and 3′-polyadenylated tail []. Their genome encodes for ORF1a and ORF1ab polyprotein which cleaves into multiple subunits and performs a diverse range of functions, i.e., binding and invading the host cell, viral replication, and evading the host immune system []. The proteins of coronaviruses perform their function by aggregating multiple subunits into an heteromeric complex, i.e., replication-transcription complex and several other structural proteins that assemble into new virions [,]. One of these subunits is a non-structural protein 12 (NSP-12) which involves RNA-dependent RNA polymerase (RdRp) activity.</p><p>NSP-12 of SARS-CoV-2 is highly conserved by sharing 96% similarity with SARS-CoV-1 and 70% with MERS-CoV. The recently resolved structure has also revealed conserved structural architecture by comparison with non-structural protein 5 (NS5) of flaviviruses and polymerases of picornaviruses []. However, NSP-12 has a length of ~930 amino acids in contrast to the polymerases of Human Poliovirus (PV) and NS5B of Hepatitis C Virus (HCV) which usually comprises ~500 amino acids. NSP-12 is responsible for viral replication and transcription once the virus invades into the host cellular environment by aggregating with cofactors such as NSP-7 and NSP-8 along with some additional subunits to initiate viral replication []. Its tertiary structure comprises three highly conserved domains: N-terminal nucleotidyltransferase (also known as nidovirus RdRp-associated nucleotidyltransferase (NiRAN)), interface, and C-terminal RdRp domain []. The RdRp domain of NSP-12 is highly conserved and responsible for viral replication and has been reported in several families of RNA genome viruses, while the NiRAN domain is only conserved among the members of nidoviruses by possessing kinase-like fold and involves in attachment with NSP-7 and NSP-8 []. RdRp domain contains seven conserved motifs (motif A-G) that involve attachment with nucleotide triphosphate (NTP) and RNA primer during the viral replication [,]. Due to these conserved evolutionary features of the RdRp domain, it has been suggested to be an important therapeutic target [,,].</p><p>Currently, many studies including single-ligand docking [,] as well as screening of the compounds library [,] have been conducted to search for potential inhibitors against NSP-12 of SARS-CoV2. In our study, we employed the structural-based homology method to identify structurally-conserved proteins and then identified US Food and Drug Association (FDA)-approved inhibitors of the homologous structure. The selected FDA-approved inhibitors were Dasabuvir Sodium (Dasabuvir) [] and Sofosbuvir [] which were previously authorized for inhibiting the NS5B of HCV. Furthermore, the conserved active site of NSP-12 was characterized and the affinity of the inhibitors were evaluated using in silico assessments, i.e., molecular docking and simulations. </p></sec><sec id=\"sec2-molecules-27-07732\" sec-type=\"results\"><sec id=\"sec2dot1-molecules-27-07732\"><p>The DALI server was employed for the identification of homologous protein structures to NSP-12 of SARS-CoV-2 that were already submitted in the PDB repository. The submitted query resulted in a total of 1105 homologous structures with a default cutoff of Z-score ≥2 as shown in . For screening significant homologs, a cutoff Z-score value ≥19 was considered, resulting in 611 structures. These resulting homologous structures were involved in the RdRp activity and belonged to four viral families: <italic toggle=\"yes\">Coronaviridae</italic>, <italic toggle=\"yes\">Flaviviridae</italic>, <italic toggle=\"yes\">Picornaviridae</italic>, and <italic toggle=\"yes\">Caliciviridae</italic> and infected a diverse range of hosts. The representatives belonging to different viral genera were selected from the filtered homologous structures and their similarities were evaluated by using the similarity matrix and a cladogram by the neighbor joining method. The resulting matrix was visualized as a heatmap as shown in A. The heatmap of Z-score-based structural similarity revealed the close relationship of coronaviruses with the members of <italic toggle=\"yes\">Flaviviridae</italic> as well as <italic toggle=\"yes\">Caliciviridae</italic> and <italic toggle=\"yes\">Picornaviridae</italic>.</p><p>Based on structural similarity, the unrooted cladogram of homologous structures to NSP-12 was clustered into four viral families as shown in B. The clade of the coronaviruses, i.e., SARS-CoV-1 and SARS-CoV-2, shares a close relationship with the NS5B of HCV and also with the other representatives of <italic toggle=\"yes\">Flaviviridae</italic>, i.e., the Zika Virus (ZIKV), Dengue Virus (DENV), Bovine Viral Diarrhea Virus (BVDV), Japanese Encephalitis Virus (JEV), and Classical Swine Fever Virus (CSFV). The members of <italic toggle=\"yes\">Picornaviridae</italic> included PolV, Human Coxsackievirus (CoxV), Human Rhinovirus (HRV), Human Enterovirus (HEV), Human Enterovirus 71 (HEV71), Foot and Mouth Disease Virus (FMDV), and Encephalomyocarditis virus. The representatives of <italic toggle=\"yes\">Caliciviridae</italic> were the Sapporovirus (SapV), Rabbit Haemorrhagic Disease Virus (RHDV), Murine Norovirus (MNV), Norwalk Virus (NarV), and Human Norovirus (HNV). The clade of <italic toggle=\"yes\">Flaviviridae</italic> and coronaviruses shared significant relationships as compared to the members of <italic toggle=\"yes\">Caliciviridae</italic> and <italic toggle=\"yes\">Picornaviridae</italic>.</p></sec><sec id=\"sec2dot2-molecules-27-07732\"><p>The conserved motifs in NSP-12 were analyzed by multiple structures and sequence alignments implemented at the PROMALS3D server. The PROMLS3D server aligns multiple structures based on similarity at the amino acid sequence and secondary structure level. The identified conserved motifs in NSP-12 have been displayed in A. NSP-12 has seven highly conserved motifs (motif A-G) which are responsible for NTP catalysis, binding to the 3′ of RNA template, and viral replication by agglomerating with NSP7 and NSP8 []. </p><p>To display the conserved orientation of these motifs at the active site of NSP-12, multiple structures belonging to the family <italic toggle=\"yes\">Coronaviridae</italic>, <italic toggle=\"yes\">Flaviviridae</italic>, and <italic toggle=\"yes\">Picornaviridae</italic> were superposed (structurally aligned) by using the MatchMaker algorithm implemented in UCSF Chimera. The conserved motifs identified from multiple sequence and structural alignment have been shown in B. Motifs A–E are located in the palm domain while motifs F and G are present in the fingers domain of NSP-12. Motifs A and C have highly conserved aspartic acid residues which bind to the divalent ions (Mn<sup>2+</sup> or Mg<sup>2+</sup>) and are responsible for executing the catalysis of nucleotide-binding [,]. Motifs B, D, E, and G are involved in nucleotide recognition and coordination. Motif F is enriched with positively charged basic residues which interact with the triphosphate moieties of the NTP [,]. Motif G has been also reported to involve in adjusting the orientation of the primer and template []. </p></sec><sec id=\"sec2dot3-molecules-27-07732\"><p>Pharmacophore modeling is a very useful technique for the detection of ligand-binding or a druggable site in protein that could be targeted to either trigger or to block the biological activity []. Pharmacophore modeling was computed by using the CAVITY search module of the CavityPlus server with a selection of “No Ligand” mode and default parameters. The CAVITY search module evaluates the pharmacophore or druggable cavity based on surface energy and site geometry. The analysis resulted in the prediction of a total of 27 cavities in NSP-12 and their information has been shown in . The cavities were ranked based on factors, i.e., drug score, druggability, and predicted average PKD and the top-scored cavity as shown in C was selected for further analysis.</p><p>The top-ranked cavity has a higher drug score revealing significant druggability of the active site with a predicted maximum pKd score of 6.99. The predicted maximum pKd score ≥6 indicates a suitable ligand ability of a cavity binding site. There was a total of 154 amino acid residues constituting the predicted cavity and they were compared with the evolutionarily conserved residues of the RdRp domain of NSP-12 from multiple sequence alignments. The comparison has been plotted as a Venn diagram as shown in A. The highly conserved residues were K545 and R555 of motif F, D618, and D623 of motif A, S682, G683, T687, and N691 of motif B, D760, and D761 of motif C and W800 of motif D. These residues were further represented as a target in the RdRp domain of NSP-12 for molecular docking with the FDA-approved antivirals.</p></sec><sec id=\"sec2dot4-molecules-27-07732\"><p>Before molecular docking, the ligands, i.e., Dasabuvir, Sofosbuvir, Ribavirin, GTP, and UTP, were prepared using AutoDock Tools. The grid box was centered at the active site or druggable cavity and the residues were considered rigid for both receptors, i.e., NSP-12 of SARS-CoV-2 and NS5B of HCV. The molecular docking was performed by AutoDock Vina with the parameters that are suggested for a single-docking experiment and the results for each abovementioned ligand were stored. AutoDock Vina predicts the docking score in kcal/mol and the lower energy means higher binding affinity between the ligand and the receptor. The results score of ligand bindings with the active site of NSP-12 and NS5B has been compared and visualized as a bar chart as shown in B.</p><p>The docking score of NSP-12 with the ligands was in the range of −6.2 to −8.1 kcal/mol, while for NS5B, it was −6.5 to −9.1 kcal/mol, and both had an average difference of 0.84 kcal/mol. The average docking score of NSP-12 and NS5B for Sofosbuvir, Dasabuvir, Ribavirin, GTP, and GTP were −8.2, −8.6, −6.35, −8.25, and −7.6 kcal/mol, respectively. NSP-12 showed the lowest docking score (−8.1 kcal/mol) with Dasabuvir and higher (−6.2 kcal/mol) with Ribavirin. In general, the ligands showed a higher docking score for NS5B as compared to NSP-12. </p><p>To examine the interactions between ligand and the conserved residues present at the active site of NSP-12, LigPlot + v2.1, standalone software was used. The predicted interactions of Sofosbuvir and Dasabuvir with NSP 12 have been shown in A,B, respectively. The conserved residues involved in forming hydrogen bonds with Sofosbuvir were C622 of motif-A, W800 of motif-D, G811, and S814 of motif-E, while Dasabuvir interacted by the formation of hydrogen bonds with D761 of motif-C and W800 of motif-D. In addition, several conserved residues of motif-A, -C, and -E were also involved in hydrophobic interactions with Sofosbuvir and the residues of motif-A, -B, -C, -E, and -F with Dasabuvir. On the other hand, the native competitors such as GTP interacted with NSP-12 by forming hydrogen bonds with A554, R555, and T556 of motif-F, Y619, K621, and C622 of motif-A, while UTP interacted with S759, D760, and D761 of motif-C, W800 of motif-D, and G811 and S814 of motif-E. In the case of HCV as a receptor, Sofosbuvir binds with motif-B and -C, while Dasabuvir interacted with residues of motif-A, -B, -C, and motif-F, and their receptor–ligand interaction plots have been shown in C,D, respectively. </p></sec><sec id=\"sec2dot5-molecules-27-07732\"><p>By binding to a protein’s functional binding cleft, each small molecule has the potential to cause significant conformational changes in the protein’s structure. To investigate the structural stability of 6M71, 100 ns of molecular dynamics (MD) simulations were run with the protein in both its apo and ligand-bound forms. The stability of the systems was evaluated using root mean square deviation (RMSD), root mean square fluctuation (RMSF), radius of gyration, and Hydrogen bond analysis.</p><sec id=\"sec2dot5dot1-molecules-27-07732\"><p>The average RMSD for 6M71, 6M71-Dasabuvir, Ribavirin, and Sofosbuvir complexes was calculated as 0.23, 0.23, 0.27, and 0.27 nm, respectively. All three systems’ RMSD values, which quantify conformational changes over time, indicated that the simulation was stable for up to 100 ns. The backbone RMSDs plot analysis revealed that the complex was stable throughout MD simulation despite a small deviation from its primary conformation.  displays the RMSD plots of all complexes vs. simulated time. The resulting RMSD graph showed increasing trends with increasing RMSD values ranging from 0.10 to 0.33 nm between 0 and 60 ns in the case of 6M71/Ribavirin and 6M71/Sofosbuvir, suggesting the compounds were becoming accustomed to a new conformation inside the binding pocket; after that, the plateau continued and finally, settled at 0.29 nm, which does not exceed the 0.3 nm threshold. However, 6M71/Sofosbuvir has the same evolution to apo 6M71. Lower RMSD values for all complexes investigated shows that Dasabuvir inhibitor is stable inside 6M71 and gives a strong foundation for our research.</p></sec><sec id=\"sec2dot5dot2-molecules-27-07732\"><p>The root mean square fluctuation (RMSF) method was used to investigate the effect of ligand binding on the flexible structure of protein as well as the behavior of essential amino acids. Throughout the simulation, a lower RMSF value indicates more rigidity, whereas a higher RMSF value indicates more flexibility. In , RMSFs of the Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes are displayed. When compared to the apo form of CAII, it ise shown that the fluctuations of the ligand-bound complexes’ residues are relatively stable, particularly at the region where the residues are engaged in ligand binding. Furthermore, the average RMSF values of Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes were 0.10, 0.10, 0.11 and 0.11 nm, respectively. This finding implies that Dasabuvir, Ribavirin, and Sofosbuvir binding helped to maintain NSP-12 structurally stable.</p></sec><sec id=\"sec2dot5dot3-molecules-27-07732\"><p>We examined how binding of different ligands affected the protein’s overall compactness in its structure. For this goal, the radius of gyration (Rg) was calculated as a function of time. Any ligand with a high enough Rg value will be more likely to be flexible, making it unstable. On the other hand, lower Rg values point to confirmation of a dense and closely packed conformation. The average Rg values Apo NSP-12, NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes were found to be 2.99, 3.00, 2.97, 2.98 nm, respectively, suggesting that the binding of Dasabuvir, Ribavirin, and Sofosbuvir to NSP-12 packing did not cause significant change. Moreover, based on the graph (), it appears that the Rg of NSP-12/Dasabuvir reached a stable equilibrium during the 100 ns simulation than NSP-12/Ribavirin and NSP-12/Sofosbuvir complexes. </p></sec><sec id=\"sec2dot5dot4-molecules-27-07732\"><p>Further understanding of the examined chemicals’ molecular recognition, molecular interactions, and selectivity within the receptors can be attained using hydrogen bond analysis. For all MD trajectories of the complex, we calculated the number of hydrogen bonds created.  displays the number of hydrogen bonds and pairs within 0.35 nm generated between NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes during the MD simulation. It was noticed that the Dasabuvir binds to the active pocket of NSP-12 with an average of 3.46 hydrogen bonds and 1.13 pairs within 0.35 nm. Similarly, Ribavirin acting as a ligand interacted with NSP-12 as a target within the binding site with an average of 3.97 hydrogen bonds, although the average of pairs within 0.35 nm is 1.66. However, the average of both hydrogen bonds and pairs within 0.35 nm for the NSP-12/Sofosbuvir complex was 1. On the other hand, the H-bonding plot showed that Dasabuvir was able to maintain a more robust interaction with the binding pockets of NSP-12 throughout the simulation time than Ribavirin and Sofosbuvir.</p></sec></sec><sec id=\"sec2dot6-molecules-27-07732\"><p>Before MM/PBSA calculation, each complex between the receptors or inhibitors/ligands should reach the equilibrium state. From , it is clear that the temperature of each complex quickly reaches the target value (300 K) and remains stable over 100 ns.</p><p>After that, we utilized a python script MmPbSaStat.py to calculate average free binding energy of the selected complexes (), as provided in the g_mmpbsa package. This script calculates the average free binding energy and its standard deviation/error from the output files, which were obtained from g_mmpbsa.</p><p>The energy liberated during the process of bond formation, or alternatively, the interaction between a ligand and protein, is shown in the form of binding energy. The lesser the binding energy, the better the binding of the ligand and protein. The final binding energy is a cumulative sum of van der Wall, electrostatic, polar solvation, and SASA energy. Except for the polar solvation energy, all other forms of energy contributed favorably to the interaction between different molecules (Dasabuvir, Ribavirin, and Sofosbuvir) and NSP-12. The bioactive molecule Dasabuvir showed the least binding free energy (−42.151 KJ/mol) among all the selected molecules. A comparison of the binding free energies of all the complexes was made by plotting the binding energy versus time graphs. Further, we examined the contribution of each residue of NSP-12 in terms of binding free energy to the interaction with the selected. The contribution of each residue was calculated by decomposing the total binding free energy of the system into per residue contribution energy molecules using python script MmPbSaDecomp.py provided by MM-PBSA package. From the overall results, it was observed that the amino acid residue Trp 800 showed the lowest contribution binding energy for all the tree complexes, suggesting their significant role in binding; indeed, contribution energy analysis of amino acid residues Tyr619, Cys622, Trp800, Glu811, Ser814, and Asp761 revealed that these amino acids played an important role during interaction, which confirms the docking result.</p><p>Finally, the Sofosbuvir and Dasabuvir could be established as a repurposed drug against the COVID-19 main protease.</p></sec></sec><sec id=\"sec3-molecules-27-07732\" sec-type=\"discussion\"><p>Drug repurposing or repositioning is a term coined for the study of therapeutic uses of existing drugs that are either previously approved or currently under the investigational phase. Drug repurposing offers potential advantages over the discovery of novel drugs, and one is that it can be delivered quickly without lengthy assessments and time-consuming clinical trials. Due to this significance, drug repurposing offers an effective way to save lives and time during pandemics. Many drugs have been already suggested for the current pandemic caused by SARS-CoV-2, of which the most prominent are FDA-approved Chloroquine and its derivatives—an anti-malarial drug and Remdesivir. Remdesivir, a nucleoside analog, has been shown to significantly reduce the SARS-CoV-2 pathogenicity in in vitro study [] and is still under investigation []. However, there is still a need for the identification of more drugs that could be further investigated for efficiency in reducing viral pathogenicity by using in vitro studies. These drug searches can be improved by employing computational methods such as in silico screening and structure-based drug targeting methods. </p><p>In this study, we searched for homologs of NSP-12 of SARS-CoV-2 and evaluated its binding affinity with the FDA-approved antivirals, i.e., Sofosbuvir and Dasabuvir, which specifically target the evolutionarily conserved site of NS5B of HCV []. The initial search for a homologous protein identified a total of 1105 structures by employing the DALI server. Further screening of these structures based on the quality of structural alignment (Z-score ≥ 19) resulted in 611 structures that belonged to four viral families, i.e., <italic toggle=\"yes\">Coronaviridae</italic>, <italic toggle=\"yes\">Flaviviridae</italic>, <italic toggle=\"yes\">Picornaviridae</italic>, and <italic toggle=\"yes\">Caliciviridae</italic>, responsible for RdRp activity during viral replication. The NSP-12 of SARS-CoV-2 displayed significant homology to the NS5 of <italic toggle=\"yes\">Flaviviridae</italic> or NS5B of HCV as compared to the polymerases of <italic toggle=\"yes\">Picornaviridae</italic> and <italic toggle=\"yes\">Caliciviridae</italic>. Based on the pairwise alignment of amino acid sequences, NSP-12 shares 96% identity with SARS-CoV-1, 11% with HCV and HNV, and 10% with Poli. Despite sharing a lower identity based on sequence, NSP-12 has seven highly conserved motifs in the RdRp domain that play a key role in viral replication. These motifs have been previously reported to play essential roles in the viral replication of several viral families [,,]. Motifs A–C have highly conserved residues while motifs D-G have conserved structural orientation at the active site of the RdRp domain.</p><p>The active site of RdRp was also identified as a suitable site for drug targeting based on the highest druggability score evaluated by using the CavityPlus webserver. The conserved residues and residues predicted from CavityPlus that were participating in ligand binding showed significant overlap when compared. The overlap included the conserved residues of motifs A, B, C, D, and F of the RdRp domain of NSP-12. These conserved residues have been also described previously to involve interactions with nucleoside, RNA primer attachment, and the RdRp activity []. The active site of the RdRp domain of NSP-12 was further evaluated for binding affinity with FDA-approved antivirals by the molecular docking method. Molecular docking was performed using AutoDock Vina which revealed a significant binding affinity of NSP-12 with Sofosbuvir and Dasabuvir as compared to Ribavirin. Despite that, NS5B showed higher binding affinity (lower binding energy) as compared to NSP-12; however, the differences were still insignificant. Further, MD simulations were carried out to determine the time dependent structural stability and intermolecular interactions. </p><p>The profiling of NSP-12 and ligand interaction revealed firm binding due to the formation of hydrogen bonds and hydrophobic interactions. The participating conserved residues of motif-A, -B, -C, -D, -E, and -F of the RdRp domain were involved in these interactions with the ligands, i.e., Dasabuvir, Sofosbuvir, as well as with their native competitors, i.e., GTP and UTP. Similarly, in the case of HCV as a receptor, the residues of motif-A, -B, -C, and -F were involved in interactions with the above-mentioned ligands. In general, both Sofosbuvir and Dasabuvir showed higher binding to the active site of NS5B of HCV and NSP-12 of SARS-CoV-2. </p></sec><sec id=\"sec4-molecules-27-07732\"><sec id=\"sec4dot1-molecules-27-07732\"><p>We retrieved NSP-12 of SARS-CoV-2 with PDB ID: 6M71 at 2.9 Å resolution from the Protein Data Bank (PDB) repository that was recently submitted by Gao et al. []. For the identification of homologous structures in the PDB repository, 6M71 chain A was submitted to the DALI server (<uri xlink:href=\"http://ekhidna2.biocenter.helsinki.fi/dali/\">http://ekhidna2.biocenter.helsinki.fi/dali/</uri> accessed on 16 March 2020). The DALI server identifies and ranks homologous protein structures based on the quality of structural alignment in the form of a Z-score []. The Z-score is a quantitative expression for comparing the intramolecular similarity between two homologous protein structures and its value ≥2 is suggested to be significant [,]. In this study, we used a Z-score cutoff of ≥19. To obtain a similarity matrix and a cladogram based on structural similarity, the same server was employed by using an “all against all” analysis for a total of 53 representative homologs. The heatmap was visualized by using the Pheatmap v1.0.12 package in R and the cladogram in FigTree v1.4.4 software.</p></sec><sec id=\"sec4dot2-molecules-27-07732\"><p>The selected homologous proteins responsible for RdRp activity were analyzed for conserved sequences (motifs) and secondary structure by employing the PROMALS3D server []. The resulting alignments were exported into ESPript 3.0 server for visualization (<uri xlink:href=\"http://espript.ibcp.fr\">http://espript.ibcp.fr</uri> accessed on 5 April 2020). To further study the conserved residues at the active site of NSP-12, multiple structures were aligned by using the MatchMaker algorithm implemented in the University of California, San Francisco (UCSF) Chimera v.14 and visualized [].</p></sec><sec id=\"sec4dot3-molecules-27-07732\"><p>For pharmacophore modeling or assessing druggable cavities in 6M71-A, the CavityPlus server (<uri xlink:href=\"http://www.pkumdl.cn:8000/cavityplus/computation.php\">http://www.pkumdl.cn:8000/cavityplus/computation.php</uri> accessed on 12 April 2020) was employed by submitting the structure file. The CAVITY search module of the CavityPlus server was used to detect the ligand-binding cavity and druggability was predicted based on the energy of residues and site geometry with default parameters [,]. Those residues that were reported to be evolutionarily conserved in the RdRp domain as well as identified from druggable cavity search were visualized in the Venn diagram using Venny 2.1—an online server for plotting Venn diagram [].</p></sec><sec id=\"sec4dot4-molecules-27-07732\"><p>AutoDock Tools v1.4.5 was employed for the ligand and receptor preparation []. In general, we followed the “Single-docking experiment with AutoDock Vina” method by following the guidelines as suggested by Forli et al. for ligand-receptor docking []. Initially, the Canonical SMILES IDs of the ligand compounds, i.e., Sofosbuvir (CID: 45375808), Ribavirin (CID: 37542), and Dasabuvir (CID: 56640146) were retrieved from the PubChem library [] and then imported in Chimera v.14 for converting into PDB files. Subsequently, the structures were prepared by minimizing the free energy and addition of hydrogen atoms. On the other hand, the receptor structures were acquired from PDB and then processed in AutoDock Tools by removing water molecules, isolation of a single chain, addition of charges, merging non-polar hydrogens, and assigning appropriate atom types [].</p></sec><sec id=\"sec4dot5-molecules-27-07732\"><p>AutoDock Vina v1.1.2 was employed for molecular docking between ligand molecules and receptor []. A grid box of size 48 × 44 × 54 Å was selected and centered at the druggable cavity of 6M71-A oriented at 117.4, 115.3, 129.5 Å, while the NS5B of HCV (PDB ID: 1YUY-A) was considered as a control with the same size of the grid box used at the origin of 0.6, 80.0, and 64.2 Å. The parameters for molecular docking were used as described for running the single-docking experiment []. Furthermore, we chose the default scoring function implemented in AutoDock Vina []. The resulting score of molecular docking of NSP-12 of SARS-CoV-2 and NS5B of HCV were compared and plotted using a bar plot. For depicting protein–ligand interaction, LigPlot + v2.1 was used for characterization and visualization [].</p></sec><sec id=\"sec4dot6-molecules-27-07732\"><p>The MD simulation of the apo and ligand-bound complex of NSP-12 protein was done using the GROMACS v.2021 and GROMOS96 54a7 force field. The systems were solvated in a cubic box of TIP3P water model and neutralized with the addition of counter ions. The neutralized systems were minimized using the steepest descent algorithm with a maximum force of 1000 KJ mol<sup>−1</sup> nm<sup>−1</sup> to adjust the geometry. To obtain further convergence, the minimized systems were equilibrated for in an NVT ensemble followed by NPT ensemble. Finally, a 100ns production run was executed, with coordinates and energies being saved in the output trajectory file at 10ps intervals. The RMSD, RMSF, radius of gyration (Rg), and hydrogen bond contacts were used to determine the degree of convergence for all of the simulated systems.</p></sec><sec id=\"sec4dot7-molecules-27-07732\"><p>The molecular mechanic/Poisson–Boltzmann surface area (MM-PBSA) methods were used to calculate of binding free energy NSP-12/Dasabuvir, NSP-12/Ribavirin, and NSP-12/Sofosbuvir complexes. This calculation validates the thorough ligand–protein interaction investigation. The binding energy constitutes potential energy, polar and non-polar solvation energies, and they were calculated by the ‘g_mmpbsa’ packages using GROMACS. The binding energy calculations in this method were calculated using the following equation:</p><p>The ΔG binding represents the total binding energy of the complex, while the binding energy of the free receptor is the G receptor, and that of unbounded ligand is represented by the G ligand.</p></sec></sec><sec id=\"sec5-molecules-27-07732\" sec-type=\"conclusions\"><p>Structural homology-based drug repurposing is a significant approach to exploring the homologous proteins which have been already targeted by either approved or investigational drugs. In this study, we searched for structural homologs of NSP-12 of SARS-CoV-2 and identified their relationship with the members of three viral families, i.e., <italic toggle=\"yes\">Flaviviridae</italic>, <italic toggle=\"yes\">Picornaviridae</italic>, and <italic toggle=\"yes\">Caliciviridae</italic>. The RdRp domain of NSP-12 is highly conserved in terms of both sequence and architecture and assessed for binding affinity with FDA-approved antivirals, i.e., Dasabuvir and Sofosbuvir. These antivirals were initially approved for targeting the NS5B of HCV and serve as competitive inhibitors for binding to the active site that results in reducing viral pathogenesis. The results of this study revealed that the six conserved motifs (motif A-F) were involved in the formation of hydrogen bonds and hydrophobic interactions with the ligands. However, there is still a further need for in vitro analysis of these drugs to assess their efficiency before entering into clinical trials.</p></sec></body>],\n",
              " [<body><sec id=\"sec1\"><p id=\"p0050\">Coronavirus disease 2019 (COVID-19), caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2), is currently spreading around the world rapidly (). This coronavirus (CoV) has infected more than half a billion people and caused over 5 million deaths, implying a global health threat. Despite prevention strategies such as social distancing and vaccines (), the most common treatments for infected patients are supportive care and respiratory support because there are no effective antiviral drugs. As the development of a new drug takes years to a decade, drug repositioning can significantly accelerate the development cycle of therapies for COVID-19.</p><p id=\"p0055\">The key to drug repositioning of COVID-19 is to identify critical targets for the CoV replication cycle. Current candidates that have been tested in clinical trials can be divided into two groups according to their mechanism of action (MoA): virus-based and host-based treatment options. The former targets the virus components involved in the CoV replication cycle. For example, approved nucleoside analogues (favipiravir and ribavirin) and experimental nucleoside analogues (remdesivir and galidesivir) may have potential against SARS-CoV-2 (; ; ; ). Two oral antiviral treatments, Molnupiravir and Paxlovid, were recently authorized by the Food and Drug Adminstration (FDA). Molnupiravir has relatively low efficacy and may have potentially serious adverse effects, such as mutagenicity and congenital disabilities (). Paxlovid is an oral protease inhibitor that appears to be more effective than existing protease inhibitors with fewer safety concerns (). The host-based approaches are less straightforward than the virus-based approaches. They target the key host factors utilized by CoV for viral replication or stimulate innate antiviral responses in hosts. It was quickly discovered that SARS-CoV-2 uses the SARS-CoV receptor ACE2 for entry and leverages the serine protease TMPRSS2 for S protein initiation (). ACE2 regulates the renin-angiotensin system (RAS) in several diseases and expresses in the lungs and the kidney, small intestine, testis, and heart (; ). TMPRSS2 activates the virus-membrane fusion on the cell surface by cleaving the S protein. In addition to ACE2 and TMPRSS2, several candidate receptors can aid SARS-CoV-2 infection, such as the cell-surface proteins tyrosine-protein kinase receptor UFO (Axl), low-density lipoprotein receptor class A domain-containing protein 3 (LDLRAD3), and C-type lectin domain family 4 member G (CLEC4G) (; ). Moreover, members of pathways that mediate membrane fusion and lipid metabolism, such as PI3K type 3 (PIK3C3) and PI3K regulatory subunit 4 (PIK3R4) from phosphatidylinositol 3-kinase (PI3K) pathway and sterol regulatory element-binding protein 1 and 2 (SREBF1 and SREBF2), are also crucial for infection (). Several efforts have been made to identify drugs that inhibit SARS-CoV-2 by targeting host proteins. An <italic>in vitro</italic> study demonstrates that the inhibition of the protease activity of TMPRSS2 partially prevents the entry of SARS-CoV-2 into the lung epithelial cells (). <italic>In vitro</italic> studies suggest that both chloroquine and hydroxychloroquine may prevent the transport of SARS-CoV-2 from the early endosome to the endolysosome. In addition to treating malaria, chloroquine and hydroxychloroquine also have immunomodulatory effects. (). However, none of these options was confirmed to have an anti-SARS-CoV-2 effect, which motivates the discoveries of other uncovered but practical host factors.</p><p id=\"p0060\">Fortunately, more comprehensive approaches have been applied to identify host factors, such as affinity purification mass spectrometry (AP-MS) and CRISPR screening (; ). By systematically collecting and collating the results of these data, more host genes can be discovered. Thus, our goal is to find drug candidates that can target these host genes. Existing computational drug-repositioning methods can be classified into structure-based and network-based methods (). The network-based methodology combines the virus-host interactome and drug targets in the human protein-protein interaction (PPI) network as a systems pharmacology-based network medicine platform, which is vital and widely used in drug repositioning for COVID-19 (; ; ; ; ; , , ). These methods can be divided into three categories: clustering, propagation, and deep learning. They measure the similarity of sub-networks from the network, the proximity of drugs to the host network, or predicted interactions. Some even predict combinations of drugs (). Network pharmacology can choose drugs combined in lower doses but targeting the causal disease mechanism, may solve the one symptom–one target–one drug problem caused by traditional combination therapy (; ). However, these methods ignore that most of the applied PPI networks are all directional, which initializes the incorporation of this type of information in our analysis. For example, we should consider inhibitors of proviral genes rather than stimulators, even though both classes of drug-based networks show a similar pattern with the host network. Furthermore, a drug may target multiple host genes through the network. If the drug stimulates some genes and inhibits others, its effect on the virus needs to be reassessed. Importantly, current methods barely avoid the potential side effect of repurposed drugs for COVID-19 patients with chronic underlying diseases, such as diabetes mellitus, cardiovascular disease, chronic pulmonary disease, and cancers.</p><p id=\"p0065\">In this study, to identify the potential antiviral agents, we applied an end-to-end computational method to search drug-gene(s)-coronavirus (CoV) paths in a directional network and inferred the effect of drugs on CoV. We constructed the integrative network by aggregating three networks with direction information, including a virus-host interactome, a human PPI network, and a drug-target network. By leveraging the edge direction in the networks, we can determine whether a drug is reachable to the virus and the MoA between the drug and the virus protein, i.e., inhibition or stimulation. Considering that patients with specific comorbidities are more susceptible to COVID-19 infection, we inspected the side effect of the drug candidate on several common comorbidities based on the genetic information of several complex diseases. We reported 29 FDA-approved drugs (7 currently in clinical trials as COVID-19 treatment) by a scoring strategy accounting for the effects of drugs on CoVs and comorbidities and other auxiliary information. We also compared the prediction performances with the connectivity map (CMAP) method. Together, we established a precise host-based antiviral drug repositioning strategy by incorporating directional information and minimizing side effects, which will provide a new sight in therapy discovery for the pandemic.</p></sec><sec id=\"sec2\"><sec id=\"sec2.1\"><p id=\"p0070\">The analytical workflow for our drug repositioning is shown in . The whole analysis process is mainly divided into four steps. We first comprehensively collected candidate host genes from three major domains for mechanistic research of CoV infection: genetic risk genes, physical interaction genes, and biological regulation genes (A). The source of genetic risk genes was mainly collected from the GWASs of COVID-19. Physically interacting genes were obtained from the literature, PPI databases, and AP-MS experiments. The biological regulation genes were collected from the results of the CRISPR screen of CoV-infected cells. To select drugs with the correct mechanisms of action (MoA), we also determined the direction of the effect of viral infection while collecting host genes (B, ). We determined the direction of genetic risk genes by GWAS and variant annotation. We used the up- and downregulation of physically interacting genes in viral infection expression profiles as their direction. The direction of biological regulation genes was derived from their effect on the screening results. We then collected data on human protein interactions, their direction of action, and the effects of drugs and drug targets. Thus, we constructed a directed multicomponent network including CoV, host genes, human genes, drug targets, and drugs. We then searched for drug-gene(s)-CoVs paths in this network and used the direction information to screen out drugs with suitable MoAs (C). To investigate the impact of drug candidates on the underlying disease in patients with COVID-19, we also identified genetic risk genes for common complex diseases and their direction to disease from genetic data. Finally, drugs with a potential inhibitory effect on the disease were prioritized based on a combined score, and conversely, drugs that promote disease were filtered out (D, ).</p></sec><sec id=\"sec2.2\"><p id=\"p0075\">Phylogenetic analysis of human CoVs shows that SARS-CoV-2, SARS-CoV, and MERS-CoV have a high conservative genome (). Specifically, the envelope and nucleocapsid proteins of SARS-CoV-2 are two evolutionarily conserved regions, having high sequence identities between SARS-CoV and MERS-CoV (). Therefore, these three viruses may share infectious mechanisms through common host genes, and the therapeutic drugs could be overlapped. We systematically integrated candidate host genes from three different angles: (1) to identify genetic risk genes of COVID-19, we first got significant risk variants (p-value&lt;5 × 10<sup>−8</sup>) of 517 COVID-19 GWASs from GRASP () (), yielding 2,618 risk variants significantly associated with COVID-19 (). If the risk variant is an expression quantitative trait locus (eQTL) of whole blood or lung, we retrieved the eGene of the eQTL as the risk gene and inferred the direction of the risk gene to the virus by matching the effect size of the GWAS and the effect size of the eQTL. We identified 77 genetic risk genes (), including innate antiviral defenses, which are critical early in infection (IFNAR2, OAS1, and OAS3). OAS3 encodes for the oligoadenylate synthase family of proteins that degrades viral RNA and activates antiviral responses, and its expression was predicted to be significantly associated with the disease (; ). The association on chromosome 19p13.12 colocalizes an eQTL of TYK2, the target of JAK inhibitors, whose high expression is associated with COVID-19 (). (2) To collect the high confidence proteins interacting with CoVs, we gathered data from PPI databases, public literature, and affinity AP-MS analyses. We used up- and downregulation of genes in the CoV infection process as gene-to-virus direction. Expression profiling data were obtained from publicly available tissue from virus-infected and normal samples. We identified 435 genes that interact with viral proteins and their direction of action (). Among them, 16 genes interacted with both SARS-CoV and SARS-CoV-2 and were significantly differentially expressed in the expression profiles of the two viruses (). Aconitase1 (ACO1) is overexpressed in COVID-19 and underexpressed in non-COVID-19 viral infections. Interestingly, ACO1 is involved in iron metabolism, and heme appears to be interlinked with COVID-19 pathophysiology (). PCDH9, one of the downregulated genes, interacts with multiple proteins of CoVs, including M, NSP6, ORF6, ORF7a, ORF7b, and S proteins. PCDH9 protocadherin is essential in epithelial cell-cell adhesion and the integrity of endothelial barrier function (). The disruption of cadherin proteins by viral protein ORF7b could contribute to several symptoms of COVID-19 infection, including multiorgan failure (). (3) Genome-wide screens have been widely used to identify host factors for various viruses, including CoVs (; ; ). Genome-wide perturbations can identify genes in human cells that play a key role in viral infection and the body’s response. We selected the top-ranked genes from six genome-wide CRISPR screens () and determined their direction based on the perturbation pattern and the effect size. We identified 958 genes from the CRISPR screen data (). It is worth noting that seven genes were selected in the screening of all three viruses, including CTSL, which encodes the Cathepsin L protease. CTSL can functionally cleave the SARS-CoV-2 spike protein and enhance virus entry (). Studies have shown that inhibitors of CSTL can effectively prevent human CoVs infection ().</p><p id=\"p0080\">By combining the genes from the three sources, we identified 1,410 host genes (). Remarkably, no genes were collected from all three sources simultaneously, and the overlap between the three sources is also relatively small, suggesting current biological assays or genetic methods may capture complementary patterns of CoV-host crosstalk. For example, only 1 out of the 77 genetic risk genes is derived from the CRISPR screen and physical interaction, respectively (A). For SARS-CoV-2, we revealed 882 candidate host genes (B). Among them, BMPR2 is not only a genetic risk gene but also interacts with SARS-CoV-2 ORF3 protein. Heterozygous mutations in BMPR2 can cause hereditary pulmonary hypertension (). Drugs targeting BMPR2 may affect the prognosis of patients with pulmonary hypertension based on genetic evidence. SARS-CoV-2 and SARS-CoV share 67 host genes (C), including ACE2, the primary cell entry receptor for SARS-CoV-2, whereas 35 genes, including TMEM41B, were identified as host genes of both SARS-CoV-2 and MERS-CoV. TMEM41B, an essential host factor for HCoV-229E, can promote lipid localization in infected cells (). We divided these genes into antiviral and proviral genes according to their effect on the CoVs. There are 654 antiviral genes and 756 proviral genes. Antiviral genes were enriched in viral-infection-related pathways such as influenza A and hepatitis C and inflammation-related pathways such as cytokine-cytokine receptor interaction and Th17 cell differentiation (D). Proviral genes were also enriched in some infection-related pathways, such as human papillomavirus infection, Salmonella infection, and <italic>Vibrio cholerae</italic> infection (E). Both proviral and antiviral genes were also enriched in autoimmune diseases such as rheumatoid arthritis and inflammatory bowel disease.</p></sec><sec id=\"sec2.3\"><p id=\"p0085\">The catalog of host genes associated with CoVs and the directional information can provide mechanistic insight for drug target selection. As the development of a new drug takes a long time and huge cost, we believe our integrated resource could be alternatively suitable for improving drug repositioning opportunities on COVID-19. We developed an end-to-end network approach to search for potential drug repurposing. We first constructed a directional informative human PPI network by fusing STRING, KEGG, and reactome data (; ; ). The directional human interactome used in this study contains 11,126 proteins and 167,776 interactions. We also collected drug and drug-target interactions with a clear direction of action from DGIdb (). Together, 23,306 directional drug-target interactions of 9,540 drugs were used in this study. We merged drug-target interactions, human protein interactions, and virus-host interactions into a whole drug-gene-CoV network. Then we used the random walk with restart (RWR) algorithm to find drug-gene(s)-CoVs paths in the network and inferred the direction of the drug to the CoVs (see ). Totally, 6,234 drugs with inhibitory effects on CoVs were inferred. Older age and comorbidities play essential roles in influencing the severity of COVID-19. We applied a genetics-based method to predict the impact of repurposed drugs on comorbidities, including diabetes mellitus, cardiovascular disease, cerebrovascular disease, chronic liver disease, chronic kidney injury, autoimmune disease, and cancers. According to our previous strategy, we filtered and prioritized potential drugs by inspecting their impact on these complex diseases (). We collected 10,557 variants in 17 diseases and mapped these variants to 536 genes (). Finally, we identified 246 drugs with possible side effects and 269 with potential comorbidity treatments.</p><p id=\"p0090\">To select the optimal drugs, we utilized the properties of drug-gene(s)-CoVs path (including drug-target interaction, network mapping, and source of host genes) for each drug to derive a combined score and used it to prioritize drug candidates (see ). We found that 29 of the top 100 drugs are FDA approved (A, ), and seven of these are in clinical trials for the treatment of COVID-19. We collected 397 drugs in clinical trials for COVID-19 from  and overlapped with the top 100 drugs. None of the drugs in clinical trials prior to phase 3 have been found in clinical trials for COVID-19, and there are nine drugs in phase 3/4 clinical trials for COVID-19 (B); this may be because most of the COVID-19 clinical trial drugs are in clinical phase 3 or FDA approved. We identified well-studied drugs with clinical trials, such as aspirin and chlorpromazine. Chlorpromazine is an antipsychotic drug that has recently been proposed to have antiviral activity against SARS-CoV-2 (). The MoA of chlorpromazine is either via inhibition of clathrin-mediated endocytosis and/or at later stages of virus assembly and egress (). We also identified drugs with no clinical trials evidence but showed potential, such as bosutinib. Bosutinib is a treatment for chronic myeloid leukemia by inhibiting the Abelson kinase signaling pathway. In addition, experimental data suggest that bosutinib is a powerful anti-inflammatory agent (; ). Recently, bosutinib was found to strongly inhibit SARS-CoV-2 <italic>in vitro</italic> with low toxicity (). Recently, it was reported that pseudoephedrine can antagonize COVID-19 virus by blocking virus expansion (). Another example is that ponatinib is a potent inhibitor of SARS-CoV2-induced cytokine storm (), whereas epinephrine may be considered an intervention to minimize the severity of COVID-19 ().</p><p id=\"p0095\">Moreover, we compared our method with the CMAP method based on similarity analysis of expression profiles, which was frequently used in drug repositioning (). CMAP contains the cellular gene expression profiles under the perturbation of 2,428 drugs. We measured the associations of gene expression patterns between CoV-infected patients and the reference-compound-perturbed cells (). COVID-19 clinical trial drugs are used as ground truth. Our method has the highest area under the curve (AUC) of 0.61 compared with the results calculated by CMAP using two SARS-CoV-2 infection expression profiles (C).</p></sec><sec id=\"sec2.4\"><p id=\"p0100\">Network analysis methods such as deep learning, network diffusion, and network proximity can evaluate the accessibility of drugs to SARS-CoV-2 in the background network (; ; ). However, given a proviral host gene, we should choose its inhibitor/antagonist as therapeutics instead of its activator/agonist, even if the activator/agonist shows higher proximity to this host gene in the network. For example, SIGMAR1, the target of the NSP6 SARS-CoV-2 protein (), is identified as a proviral gene in our study. Although which class of drugs, i.e., agonists or antagonists of SIGMAR1, is responsible for the replication activity of SARS-CoV-2 remains to be verified by more experiments (), the latest evidence revealed that knockout or knockdown of SIGMAR1 could cause significant reductions in SARS-CoV-2 replication (). Furthermore, a clinical trial demonstrated that the antidepressant fluvoxamine, an antagonist of SIGMAR1, could prevent clinical deterioration in adult outpatients infected with SARS-CoV-2 (A) (). However, there are network-based approaches to select SIGMAR1 agonists, such as amitriptyline, as a potential treatment for COVID-19 (), implying confliction and limitation may largely exist among current methods.</p><p id=\"p0105\">Drugs with multiple targets may have implications for other diseases in the fight against viral infections. Just as we can predict the antiviral properties of existing drugs, we can also predict whether these drugs will influence preexisting conditions in patients with viral infections. For instance, cannabidiol (CBD), a member of the cannabinoid class of natural products, is an FDA-approved drug for epilepsy treatment. There are four clinical trials of CBD for the treatment of COVID-19. Studies have shown that CBD can inhibit the cell entry and replication of SARS-CoV-2 (; ). In our analysis, CBD, ranked at 23rd, directly or indirectly targeted 87 host genes. Our comorbidity treatment analysis found that CBD is also an agonist of PPARG and GPX1. Since 1997, a group of PPARG agonists, thiazolidinediones, has been used to treat type 2 diabetes by alleviating insulin resistance (). In theory, cannabis has desirable effects on hyperglycemia through its anti-inflammatory and antioxidant properties (). GPX1 has been shown to exert a protective effect against the presence of coronary artery disease (CAD) (). Its activity was a useful marker for monitoring cardiovascular disease (). There is now a growing body of evidence that CBD is also beneficial for the cardiovascular system (; ). The anti-inflammatory and antioxidant effects of CBD may make it a treatment for coronary heart disease and diabetes, together with COVID-19 (B). Estradiol cypionate (EC) is an estrogen medication used in hormone therapy for menopausal symptoms and low estrogen levels in <italic>cis</italic> women. The analysis of electronic health records of over 68,000 COVID-19 patients revealed that estrogen therapy is associated with more than 50% reduction in mortality (). EC was repositioned as COVID-19 treatment in other studies and is now in phase 2 clinical trial (). In our analysis, EC can indirectly target HK2 and PLCB4 through ESR2, identified as a risk factor for atrial fibrillation. Furthermore, estrogen may increase the risk of myocardial infarction in transgender women by promoting thrombosis (). Therefore, EC, despite its potential antiviral effect, could be excluded due to its possible cardiovascular side effects (C). Together, our new drug repositioning strategy utilizes the direction information in the network to screen the MoA of the candidate drugs and, from a genetic perspective, considers the potential unwanted side effects of many complex diseases.</p></sec></sec><sec id=\"sec3\"><p id=\"p0110\">The key step in host-based drug repurposing for the treatment of infectious diseases is identifying the true host gene. An increasing number of studies are looking for host genes on a genome-wide scale by different means, such as GWAS, CRISPR screening, and AP-MS. These large-scale and complementary screens require systematic integrative analysis to discover more accurate host genes. Moreover, current host-based drug repositioning strategies generally utilize network proximity, network diffusion, and deep learning approaches to identify potential repurposing candidates (; ; ). The main goal of these approaches is to find drugs that can directly or indirectly control as many host genes as possible. However, the networks used in these analyses are undirected graphs. In other words, some protein interactions are activating and some inhibiting. Although some host genes are proviral and some are antiviral, drugs targeting the same target sometimes contain antagonists and agonists. Thus, it is necessary to search for drugs with correct MoA in directed networks.</p><p id=\"p0115\">To this end, we comprehensively identified host genes and their directions to CoVs by systematically integrating multi-omics data from mass spectrometry, CRISPR screening, and GWAS. We identified 964 host genes for SARS-CoV-2, 477 host genes for SARS-CoV, 236 host genes for MERS-CoV, and 1,410 unique genes. These genes include genetic risk genes such as TYK2, physical interaction genes such as ACO1, and biological regulation genes such as CTSL. The host genes from the three sources overlapped little, indicating that high-throughput biological assays are complementary. There are more overlapping genes between SARS-CoV-2 and SARS-CoV than between SARS-CoV-2 and MERS-CoV; this may be because the genome and structure of SARS-CoV-2 are more similar to SARS-CoV. We also used multi-omics data to determine the direction of host genes to viruses, that is, their effects on viral infection. We identified 654 antiviral genes and 756 proviral genes. Functional enrichment analysis showed that the pathways enriched in both proviral and antiviral genes were involved in infection- and immune-related pathways. To apply the catalog of host genes to drug repositioning, we constructed a background network by fusing three directed networks, including a virus-host network, a human PPI network, and a drug-target network. We screened 6,234 drug candidates and designed a scoring system to rank them. Twenty-nine of the top 100 prioritized drugs are FDA approved, including seven in clinical trials for COVID-19. In comparison with the results of CMAP, the AUC of our method is larger. In this scoring system, besides adding drug-target interaction scoring and host gene scoring, we also considered the effects of drugs on comorbidities. We found that CBD not only inhibits SARS-CoV-2 but may also act as a therapeutic drug for coronary heart disease and diabetes. EC also has antiviral potential, but we found that it may increase the risk of cardiovascular disease. Overall, we first comprehensively characterized the host genes of CoVs and their role in viral infection. In drug relocation prediction, unlike traditional network-based methods, our method focuses more on the directionality of the MoAs and the interaction networks of drugs to screen drugs with the correct MoA. In addition, we innovatively considered the potential impact of drug candidates on comorbidities.</p><sec id=\"sec3.1\"><p id=\"p0120\">There were still several limitations of our study. We collected as many host genes as possible. However, as more research data become available, we may discover more host genes to refine the virus-host interaction network. Our analysis removed host genes whose directions could not be determined. Perhaps their direction can be identified with other experimental data, such as transcriptome-wide association studies (TWAS) and Mendelian randomization (; ; ; ). We used data such as eQTLs, expression profiles, and CRISPR effect sizes to determine the directions of host genes. However, the accuracy of directions identification in these ways is limited by statistical power and tissue specificity. The effects of these host genes we identified on CoV need to be validated in <italic>in vivo</italic> and <italic>in vitro</italic> experiments. Besides, many COVID-19 drug repositioning studies, including this study, have used clinical trial datasets as ground truth for performance evaluation. Not all drugs in clinical trials have significant antiviral effects. There is currently no good dataset to verify the robustness of our strategy and others without biases. Finally, to infer the direction of a drug to CoV in each drug-gene(s)-CoVs path, we used a random walk algorithm and then combined multiple drug paths for prioritization. We can apply some more complex algorithms and models to our network in the future.</p></sec></sec><sec id=\"sec4\"><sec id=\"sec4.1\"><p id=\"p0125\">\n",
              "  \n",
              "  </p></sec><sec id=\"sec4.2\"><sec id=\"sec4.2.1\"><p id=\"p0130\">Further information and requests for resources should be directed to and will be fulfilled by the lead contact, Mulin Jun Li ().</p></sec><sec id=\"sec4.2.2\"><p id=\"p0135\">This study did not generate new unique reagents.</p></sec></sec><sec id=\"sec4.3\"><p id=\"p0140\">This study analyses existing, publicly available data and does not contain wet lab experiments.</p></sec><sec id=\"sec4.4\"><sec id=\"sec4.4.1\"><p id=\"p0145\">Since the SARS-CoV-2 was most evolutionarily conserved with SARS-CoV and MERS-CoV, SARS-CoV-2 may share host genes or therapeutic targets with the other two CoVs. Thus, by fully utilizing the relevant information and shared mechanism of infection among these CoVs, we curated as many as host genes of these three CoVs from three categories: genetic risk genes, physical interaction genes, and biological regulatory genes.</p><sec id=\"sec4.4.1.1\"><p id=\"p0150\">We first retrieved all risk SNP (p-value &lt; 5e-8) in COVID-19 GWAS from GRASP COVID-19 data (). Then, the risk SNPs were annotated as coding variants and non-coding variants by VEP (). For coding variants, the target gene is the gene where the mutation locates. We evaluated the effect of the gene by calculating the functional impact score of the risk allele using B-SIFT (). We assigned the direction of effect as 1 for a B-SIFT score higher than 0.5 and −1 for a B-SIFT score lower than −0.95. For non-coding variants, we identified the target genes and directions by integrating significant expression quantitative trait locus (eQTL) data of whole blood and lung from GTEx (release v8) (). When a variant is both a COVID-19 risk variant and an eQTL, the associated expression gene is identified as the target gene of the variant. When the GWAS risk allele positively affects gene expression, the direction is 1; otherwise, it is −1.</p></sec><sec id=\"sec4.4.1.2\"><p id=\"p0155\">We collected host genes interacting with CoVs from four PPI databases, including BioGRID (), IntAct (), STRING.Viruses (), and APID (). We also included high-confidence proteins that interacted with CoVs from three AP-MS experiments (; ; ). In addition, we also included the host genes of CoV collected by Zhou, Y. et al. (). However, there is no information on the direction of host genes to CoV in these datasets. To determine the direction of each virus-protein interaction. We manually searched the name of the protein and matched the virus in the literature and inferred the direction according to the biological effect of protein on virus infection. The direction of a host protein on the virus is set as positive if the protein has a helpful role in the invasion and amplification of CoV. Moreover, we used gene expression profiles as complementary sources of directions. We got differentially expressed genes of all three CoVs by analyzing corresponding gene expression profiles from the NCBI Gene Expression Omnibus (; ; ). Protein-coding genes with Adjust p-value ≤ 0.05 and |log2FoldChange| &gt; 1 were included. The direction of a host protein can be regarded as the same as the sign of log2FoldChange if the protein gene is differentially expressed genes.</p></sec><sec id=\"sec4.4.1.3\"><p id=\"p0160\">We also collected results from six genome-wide CRISPR perturbation screening assays (; ; ; ; ; ). For every screen assay, the top 100 significant genes were included. We comprehensively consider the perturbation and the effect size of the gene to determine the direction of the gene to the virus. For example, a gene with a negative effector value in a CRISPR knockout screen will have a therapeutic effect on the virus.</p></sec><sec id=\"sec4.4.1.4\"><p id=\"p0165\">We construct a directed virus-host network by combining all gene-virus pairs from the three sources and their directions. Gene-virus pairs with conflicting directions between different sources were deleted. Non-protein-coding genes were removed. The network was visualized by Cytoscape ().</p></sec></sec><sec id=\"sec4.4.2\"><p id=\"p0170\">To investigate the biological functions of antiviral and proviral genes, respectively, we performed a pathway-based analysis based on KEGG annotations by R package clusterProfiler (). Gene sets in KEGG pathways with 2–500 genes were included. Genes in the network were analyzed, and significantly enriched pathways were identified with FDR &lt; 0.05.</p></sec><sec id=\"sec4.4.3\"><p id=\"p0175\">We constructed a directional human interactome by integrating three resources, including human PPIs from STRING, and human signaling pathways from KEGG and Reactome (; ; ). Specifically, for PPIs in STRING, only interactions with clear directions and scores of more than 700 were included. Interactions with explicit actions, either activation or inhibition, were selected when both source and target were protein-coding genes.</p></sec><sec id=\"sec4.4.4\"><p id=\"p0180\">Drug-target information, including ancillary data such as interacting scores, and MoAs, was retrieved from DGIdb (). The direction of the drug to the target is determined by MoAs. If the MoA is inhibitory, the direction of the drug is negative and vice versa. Drug-target interactions with unclear directions are removed. Clinical stage information for the drug was obtained from ChEMBL ().</p></sec><sec id=\"sec4.4.5\"><p id=\"p0185\">By integrating the directional networks described above, we can identify the potential repositioning of known drugs on CoVs. We searched the appropriate drug-gene(s)-CoVs paths for each compound. Since drugs can directly or indirectly affect viral infections, we searched for repurposed drug candidates by directed mapping and network mapping as described in our previous paper (). Briefly, in directed mapping, we search for the drugs that target the host genes. For network mapping, we applied the random walk with restart algorithm to seek the shortest paths from the drug target to host genes and removed the unlikely paths by measuring the relatedness scores between the start node and end nodes using Python package networkx (). Then, we inspected the accessibility of each drug to host genes and assembled directional drug-gene(s)-CoVs paths. Since all the interactions in networks were labeled with directions encoded as 1 and -1 (for positive effect and negative effect, respectively), we multiply the directions to infer the drug’s effect on CoVs. Drugs that have a negative impact on the CoV are considered a potential repositioning, and the proviral drugs were removed.</p></sec><sec id=\"sec4.4.6\"><p id=\"p0190\">To predict the impact of repurposed drugs on comorbidities, we first selected the largest GWAS for 17 complex diseases (including diabetes mellitus, cardiovascular disease, cerebrovascular disease, chronic liver disease, chronic kidney injury, autoimmune disease, and many cancers) from CAUSALdb () (). We retrieved the 95% credible sets of fine-mapping results of each GWAS from CAUSALdb. Results of FINEMAP were used (). We further selected variants in credible sets with GWAS p-value &lt; 1E-8 as credible risk variants (CRVs). Then we predicted disease-associated genes and their directions of effect on diseases by the same approach linking COVID-19 risk SNPs to genes. After identifying the associated genes and their directions of effect on comorbidities, we checked the effects of repositioned drugs on these diseases using the directed mapping strategy described above. We termed negative effects as additional treatment and positive effects as unwanted side effects. Drugs with unwanted side effects were removed.</p></sec><sec id=\"sec4.4.7\"><p id=\"p0195\">We designed a plain scoring system to prioritize repositioning candidates. A drug may have multiple drug-gene(s)-CoV paths. For each <underline>D</underline>rug-<underline>G</underline>ene(s)-Co<underline>V</underline> path, we assigned a DGV score, which equals the accumulation of drug target score, mapping score, and host gene score. For drug target scores, we retrieved drug-target interaction scores of each drug from DGIdb. One point was given to the drug when the score was higher than the median of all interacting scores of the drug; otherwise, half a point. For the mapping score, the score is set as the inverse of the number of genes. For the host gene score, it got 1 point if the direction of the interaction was identified by eQTLs and CRISPR screen or half a point if the direction was identified by gene expression profile. The score for each drug is equal to the sum of all its DGV scores. Drugs with higher scores were regarded as having higher antiviral potential.</p></sec><sec id=\"sec4.4.8\"><p id=\"p0200\">To compare our method with other methods, we used a commonly-used transcriptome analysis approach, connectivity map (CMAP) (). We used gene expression profiles of SARS-CoV-2 infected samples and a control group from alveolar type II cells and African green monkey kidney cells (; ). We input the top-50 up- and downregulated genes into the web tool () and got the drug candidates ranked by connectivity map scores. We take as ground truth all drugs currently in clinical trials for COVID-19. Then we computed the receiver operating characteristic (ROC) curve and estimated the area under the curve (AUC).</p></sec></sec><sec id=\"sec4.5\"><p id=\"p0205\">Functional enrichment analysis was conducted using clusterProfiler with significance threshold set at FDR&lt;0.05.</p></sec></sec></body>]]"
            ]
          },
          "metadata": {},
          "execution_count": 37
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Repeat for further four batches."
      ],
      "metadata": {
        "id": "w9SqwpKQogT_"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.2 Second batch of 2500 remove tags"
      ],
      "metadata": {
        "id": "tEAqPrbv9ih9"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_removed_tags_2500_5000 = list(map(remove_tags, soup_results_2))"
      ],
      "metadata": {
        "id": "yzOWE4a69mCd"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_removed_tags_2500_5000)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Gz8oV9k2-dxa",
        "outputId": "67f8bf9c-ae37-405a-b4df-59ba315ba087"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.3 Third batch of 2500 remove tags"
      ],
      "metadata": {
        "id": "bdn8Ec1x_vk8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_removed_tags_5000_7500 = list(map(remove_tags, soup_results_3))"
      ],
      "metadata": {
        "id": "i-NlzNGg_u7v"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_removed_tags_5000_7500)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "voG9kWLh_08R",
        "outputId": "b814a80c-c006-4021-8199-3858871bd6b1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.4 Fourth batch of 2500 remove tags"
      ],
      "metadata": {
        "id": "lNDyav4R_9lz"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_removed_tags_7500_10000 = list(map(remove_tags, soup_results_4))"
      ],
      "metadata": {
        "id": "z6Pn6vFb_9l0"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_removed_tags_7500_10000)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IwYi-7Gl_9l1",
        "outputId": "f240c5ac-6da4-412f-e749-fee89a1516df"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.5 Fifth batch of 1397 remove tags"
      ],
      "metadata": {
        "id": "leaIudXLAIW3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_removed_tags_10000_11397 = list(map(remove_tags, soup_results_5))"
      ],
      "metadata": {
        "id": "QKTJ_vRPAIW3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_removed_tags_10000_11397)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0_vKkGccAIW3",
        "outputId": "b642ecd0-72ce-482b-d85e-25245b3e1782"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1397"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "z7FTIUi1ZCwc"
      },
      "source": [
        "### 7.6 Strip markup and keep text\n",
        "\n",
        "Function to return only the human-readable text in a document using BeautifulSoup's get_text() method. This will return all the text in the articles as a single Unicode string without the `<body>`, `<div>` and `<p>` tags.\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "def strip_markup(articles):\n",
        "\n",
        "    for article in articles:\n",
        "\n",
        "        return article.get_text()\n",
        "\n"
      ],
      "metadata": {
        "id": "8_P5jOXKOtl7"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_2500 = list(map(strip_markup, soup_results_removed_tags_2500))"
      ],
      "metadata": {
        "id": "VUSiz8UcFdKM"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "View a couple of articles with all tags removed."
      ],
      "metadata": {
        "id": "ZTy2cKvipttY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_2500[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 174
        },
        "id": "uqJeZfRPY93p",
        "outputId": "a25b86c9-377e-48d7-e52d-042d817fa5d1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Sir James Black, a winner of the 1988 Nobel Prize, clearly recognized well before the 21st century that drug repurposing strategies would occupy an important place in the future of new drug discovery (). In 2004, Ted T. Ashburn et al. () summarized previous research and developed a general approach to drug development using drug repurposing, retrospectively looking for new indications for approved drugs and molecules that are waiting for approval for new pathways of action and targets. These molecules are usually safe in clinical trials but do not show sufficient efficacy for the treatment of the disease originally targeted (). The definition of the term “drug repurposing” has been endorsed by scholars () and used by them (; ). It should be pointed out that the synonyms of “drug repurposing” often used by academics also include drug repositioning (), drug rediscovery (), drug redirecting (), drug retasking (), and therapeutic switching (; ). After the research study by Ashburn et al., Allarakhia et al. expanded the starting materials for drug repositioning to include products that were discontinued for commercial reasons, expired patents, and candidates for laboratory testing (). In the discovery process of a completely new drug, the difficulty usually lies in its safety and efficacy, which are the main potential causes of failure of most drugs in the approval () or clinical development stage (). Using existing knowledge about a drug or known target (), the time, risk, and cost of developing a drug using drug repositioning are reduced (), thereby greatly increasing the efficiency and economics of drug development, providing a better risk–reward trade-off, and making it easier to win the favor of venture capital firms ().Since the 1990s, the repositioning of sildenafil for male erectile dysfunction () and pulmonary hypertension (), the development of a new efficacy of bupropion for smoking cessation (), new applications of thalidomide for multiple myeloma (; ), and chronic graft-versus-host disease () have generated intense interest from pharmaceutical companies and academics (). These classic success stories rely on three traditional approaches: 1) molecular biology approaches (), 2) in vivo and ex vivo experimental approaches (; ), and 3) expert knowledge-based approaches (). Due to the unknown, complex, and information-fragmented nature of drug candidates and potential new mechanisms of action (), this activity is dependent on multiple factors, and success is often fortuitous (). At the beginning of the 21st century, cheminformatics (; ), bioinformatics (; ), systems biology (; ), genomics (; ), polypharmacology (; ), precision medicine (; ), and other disciplines, combined with artificial intelligence (), have developed rapidly. These rapidly growing disciplines have promoted the generation of systematic () computer methods to make the drug repositioning process cheaper and shorter (; ). Computational drug repositioning is classified as “disease-centric” or “target/gene-centric” or “drug-centric” depending on the source of discovery (). This process relies on public biochemical databases such as DrugBank (; ), ChEMBL (), Cmap (), PDB (), OMIM (), etc., to provide the appropriate information. In fact, to make the computational drug repurposing process, including the molecular docking and virtual screening steps, more convenient, database tools specifically developed for drug repurposing, such as EK-DRD (), DREIMT (), DrugSig (), RepoDB (), Promiscuous 2.0 (), etc., have been reported in the last few years. In addition, it has been found in the literature that only 10% of the research results have been carried out in the “drug-centric” pathway, which holds great prospects for future development (). With the help of database tools, it is now possible to perform computational screening of even a staggering number of hundreds of millions of compounds (). Computer methods to carry out this screening include machine learning (), network modeling (; ), text mining, and semantic reasoning (; ; ), among others. The ultimate objective of repositioning is to transfer one or two of the most relevant results to clinical applications. Therefore, validation is quite important () and requires consideration of multiple factors, such as price, toxicity levels, bioavailability, and differences between validated and computational models (; ). Current validation methods include experimental validation (), electronic health records to aid validation (), cross-validation (; ), gold standard dataset evaluation (), literature citation validation (), and expert consultation ().Today, drug repositioning is increasingly prominent in the development of drugs for a variety of neurological diseases (; ), cancer (; ), rare diseases (; ), and infectious diseases (; ). An increasing number of pharmaceutical companies are also establishing relevant R&D programs () or funding support (). To translate relevant research results efficiently and smoothly, national departments within the United Kingdom, the United States, and the Netherlands have (; ) launched initiatives or programs to build partnerships between pharmaceutical companies and academia and to further explore scientific and commercial opportunities (). It is certain that drug repositioning currently presents several dilemmas, such as intellectual property challenges (), data platforms, and analytical techniques that need to be improved (), that financial support remains important for technology development and clinical trials (), and that some scientists deny the practical utility of the approach ().There have been systematic analyses of terminology in the drug repurposing literature (), text mining of drug–disease combinations (), and the progression of a particular drug (), but no studies have yet provided a broad overview of publications on the topic of drug repurposing research. When independent researchers or collectives (including pharmaceutical companies, academia, and government departments) seek drug repurposing partnership partners and seek to obtain a concise overview of comprehensive current research hotspots, the lack of relevant intelligence analysis to aid decision-making often makes the process convoluted and time-consuming (). The bibliometric approach can solve the aforementioned problems relatively fairly, but at present, scholars have only studied the bibliometrics of aspirin, a drug repurposing (); there has not been a panoramic study of drug repurposing, and therefore, this study is necessary. Bibliometrics is a useful tool combining multiple parameters for the quantitative analysis of scholarly publications and is currently used to assess research hotspots and trends in a wide range of disciplines and industries, such as management (; ), sociology (; ), economics (), medicine (; ), environmental engineering (; ), and agronomy (; ). Therefore, this study uses bibliometric methods () to quantitatively assess the following elements of drug repositioning publications: 1) major contributors: countries, research institutions, and authors; 2) modes of collaboration: intercountry collaborations; 3) the most productive journals; 4) the most frequently used disciplinary knowledge; and 5) research trends, judged by analyzing author keywords, Essential Science Indicators (ESI) high citations, and hot research studies.We use the Web of Science™ core database, an authoritative academic information data service platform produced by Clarivate (version © 2021 Clarivate.). Due to its rigorous selection of journals, the Web of Science (WOS) Core Collection Database is now internationally recognized as a database for evaluating the scientific output or disciplinary development of scholars and institutions. Among the subdatabases, SCI-E mainly includes global journals in basic science research, covering basic pharmacological and medical research related to the theme of this study, “drug repositioning,” while SSCI includes social science, covering ethical, nursing, psychological, and other social science research related to this study.The data were obtained on 25 October 2021 through the WOS Core Collection Database Citation Indexes SCI-E and SSCI, using the formula “drug repurposing” OR “drug repositioning” OR “drug rediscovery” OR “therapeutic switching” OR “drug redirecting” OR “drug rediscovery” OR “drug retasking” search query, searching in the “subject” field and defining the document type as “Article” and “Review”. The publication time parameters were initially limited to publications related to “drug repositioning” published between 1990 and 2020. A total of 3,009 documents were obtained, of which only 31 were published in two\\xa0decades from 1990 to 2009. Of these 31 documents, except for one document that is still frequently used by scholars as a retrospective source for drug repurposing definitions in these years (), the remaining 30 were cited by other authors during the period of 2010–2020 as shown in . The overall level of interest in these studies shows a fluctuating downward trend as opposed to the rising citation fervor for drug repurposing, entering a stage of decline even under the less-demanding evaluation criterion of a 5-year maturation window (). As the literature ages, its content becomes stale and obsolete in the perspective of intelligence sources, and the value of the metrics for judging current research trends is low. Therefore, we further narrowed the study to 2,978 publications published from 2010 to 2020.The complete records of all retrieved documents are downloaded and imported for processing into Derwent Data Analyzer (DDA) version V10, a data cleaning, multiperspective data mining, and visualization software from Clarivate that improves data analysis efficiency and reduces labor costs. After importing all records of WOS documents into DDA, they are classified and measured according to a list of fields such as keyword, country/region, institution, author, research field, journal, etc. For each item in the list fieldset, DDA has a built-in data cleaning tool for automatic data deduplication.After the machine has removed duplicates, the items in the set of fields still need to be manually verified for splitting or merging. It is to be noted that the regions of certain countries are presented separately, while they are usually considered as a single country internationally. Therefore, we need to perform merging, such as combining Wales, Scotland, England, and Northern Ireland into the United Kingdom column and combining Hong Kong and Macau regions into the China column.To address the possible problem of different authors with the same name, the following two main verification steps were performed: 1) returning to the WOS database to search for publications under that author’s name under the original search formula conditions and 2) for authors whose publications provide disputed information (this also includes three cases: first, two or more authors with the same name but not the same person; second, two or more authors with the same abbreviated name, but the full names were found to be different after a search; and third, similar signatures being different variants of the same author’s name), in addition to searching the ORCID-related information of the authors concerned for judgment, we checked different institutional websites as well as encyclopedias to look for changes in the study and work history of authors with the same or similar names from 2010 to 2020 to determine whether they were the same person. Based on the verification, we then split or merged the results.After data cleaning and matrix analysis by DDA, various types of cluster plots and bubble plots can be obtained to reveal the useful information behind the data. The bibliometric fields of publication volumes, countries, international collaborations, institutions, research areas, journals, authors, highly cited research studies, and author keywords were analyzed in this study. It should be noted that because some studies were published online ahead of time and the study publication date was a year or two behind, for statistical purposes, the year of publication of such research studies was included as the year of online publication. (e.g., a study shown in the reference as published in 2022 may have been published online in 2020).Of the 2,978 papers obtained using the search criteria mentioned previously, the main ones were research studies (2248; 75.49%) and reviews (730; 24.51%). Furthermore, individual publications are not only classified by journals in the single category of research studies or reviews but also belong to other categories. These publications were also related to proceeding studies (68; 2.28%), early access (24; 0.81%), book chapters (7; 0.24%), data studies (2; 0.07%), and retracted publications (2; 0.07%). The vast majority of research studies and reviews were published in English (2967; 99.631%), with the remainder in Japanese (3; 0.101%), Chinese (2; 0.067%), Czech (1; 0.034%), French (1; 0.034%), German (1; 0.034%), Hungarian (1; 0.034%), Korean (1; 0.034%), and Portuguese (1; 0.034%). Ninety were from SSCI, and the remaining 2888 were from SCI-E. Further, 1,996 were from Open Access. An annual analysis of published research studies is shown in . The number of publications for every year expanded from 17 in 2010 to 970 in 2020. Annual publications on the subject have increased by more than 64 times. The number of annual publications has been increasing at a relatively high rate since 2015, while in 2020, there was a spike in the number of publications and annual citations, probably due to the COVID-19 pandemic, a global public health emergency that prompted special attention from scientists. Among the four countries with the highest number of publications (the United States, China, the United Kingdom, and India), the United States has maintained a high growth volume since 2010, while China was the fastest in terms of average annual growth rate in the last three\\xa0years. In 2020, the number of publications in India surged and surpassed the production of the United Kingdom.With respect to the 2978 publications related to drug repositioning research, 89 countries contributed to the field of drug repositioning research. The number of publications and citations from the 20 most productive countries/regions is shown in . There are nine countries/regions in Europe, five in the Americas, five in Asia, and one in Oceania.The four most productive countries/regions are, in order, the United States, China, the United Kingdom, and India. The United States is the absolute leader in this field, with 918 research studies on drug repositioning published since 2010, which is already more than the next highest number of publications in China and the United Kingdom combined. This is followed by India (247), Italy (232), Germany (171), South Korea (161), and Japan (146). Other productive countries include Brazil (125), France (116), Canada (111), Spain (109), Australia (73), the Netherlands (73), and Switzerland (59). In terms of publication impact, the United States led the Total citations (TC) rankings with 27,355, twice as many as that of China (11,147), which ranked second. We also included the average citations per publication (ACPP) in the comparison, which is calculated by dividing the TC by the TP (total papers) value and is a relative number that may better reflect the individual or collective level of attention than the individual TC and TP values. Canada ranked first in ACPP at 41.81, closely followed by the United Kingdom (30.85) and Germany (30.49). In addition, the h-index was originally proposed as a simple quantification that a researcher had at least h publications cited h times, reflecting to a certain extent the research results of the researcher as an individual (). Later, the word “researcher” in the definition began to be replaced by collective words such as “academic group or institution (),\" “journal (),\" and “country (),\" becoming an indicator of the level of collective research to some extent. Undoubtedly, the h-index of the United States ranks first in this field with 74 times. Taking all parameters into account, we find that publications in the United Kingdom, the United States, and Canada perform better on average. While the number of publications in China and India is significant, they have received low levels of attention.It should be noted that DDA analysis software is nationally identified based on the location of each researcher’s institution address provided in the publication. If a publication is coauthored by institutions from more than two countries, the publication is defined as the result of an international collaboration. Whether there is some affiliation between the various institutions of the research group that produces the multicountry collaboration is not taken into account. As shown in , among the publications of the top 20 countries and regions, the proportion of international collaborations is quite high in European countries, especially in Sweden (86.21%) and Belgium (81.25%). Asian and Latin American countries are generally underrepresented. In addition, the United States, the most active country in publishing and the country with the most collaborations—with 59 countries or regions—still has over 50% of the studies published overall.\\n depicts the academic collaboration network for the top 20 countries and regions in terms of productivity. Using DDA software, the network was mapped using a co-occurrence matrix. The size of the circles is proportional to the extent of each country’s contribution, the lines between the circles represent the collaboration between countries/regions, and the thickness of the connecting lines indicates the frequency of collaboration (). The results show that the United States cooperates most frequently with China and the United Kingdom and has the closest cooperation with them. In addition, Mexico, Belgium, Argentina, Taiwan, Japan, and Korea have slightly sparser cooperation networks among the 20 most productive countries/regions, while the remaining countries have more extensive cooperation networks among themselves.A total of 3,530 institutes were involved in drug repositioning research. The top 20 productive institutes are shown in . Eight of the top 20 institutions are from the United States, again indicating the dominance of the United States in drug repositioning research; three are from the United Kingdom; two are from China; and Brazil, France, Mexico, Canada, the Netherlands, Austria, and Sweden each have one. The Chinese Academy of Science ranked first in terms of the number of research studies, followed by Case Western Reserve University and the NIH. The Icahn School of Medicine at Mt Sinai ranked first in ACPP at 77.32. The NIH had the highest h-index value of 22. The Icahn School of Medicine at Mt Sinai was the best performer in ACPP at 77.32, followed by the University of California, San Francisco (67.82) and Johns Hopkins University (65.68), both of which are US-based research institutions. Compared with US research institutions, Chinese research institutions are at the back of the pack in terms of ACPP, and their impact needs to be improved.The collaboration network between the 15 largest institutions in 2010–2020 is shown in . The collaboration network provides a more visual view of the collaboration with different institutions and thus helps in the search for more beneficial collaborations. Next to the name of each institution is its total number of publications. At the intersections of these institutions, yellow dots indicate collaborations with the other top 10 research institutions. It should be noted that the number of yellow dots can indicate the output of cooperation and the strength of interagency cooperation. The nodal data with no crossover points represent the number of publications produced by the institute, either by its independent work or in collaboration with research institutions outside the top 15 (). From , we see that the University of Cambridge established the largest collaborative network, followed by the large network established by four institutions, the NIH, the Icahn School of Medicine at Mt Sinai, Karolinska Institute, and King’s College London. In terms of the number of copublications with established institutions, the Chinese Academy of Science and Shanghai Jiao Tong University copublished as many as six, followed by the University of Cambridge and King’s College London and the NIH and the Icahn School of Medicine at Mt Sinai. Analyzing the aforementioned three pairs of institutional combinations, King’s College London has two publications that are the product of collaboration between the three research institutions. The University of Sao Paulo and Aix-Marseille University are relatively independent in this research area. Combining the ranking of multiple parameters, we found that the NIH and Icahn Sch Med Mt Sinai in the United States are the most vocal institutions in terms of academic research result perspective on the topic.An analysis of research areas gives a good indication of the scope of application of the research topic, with an unrestrained number of 74 areas covered, with the top 20 areas of research in terms of publication volume shown in . Briefly, “pharmacology and pharmacy” took first place with 962 articles, followed by “biochemistry and molecular biology”, and for ACPP, the top three were science and technology-other topics (36.1)\", “mathematics (32.79)\", and “cell biology (29.65)\".\\n shows a bubble graph of the top 20 drug repositioning research areas. The bubble plot shows three dimensions of the data, namely, research area, year of publication, and the number of publications. The horizontal change in bubble size illustrates the growing trend of research areas over time, the vertical size of the bubble shows the most popular research areas in that year, and the number in the bubble indicates the frequency of the topic in the research area and the number of publications in that year (). The number of research results in each relevant field is increasing year by year. Biophysics increased from five in 2019 to 77 in 2020, a more than 15-fold increase, suggesting that drug repositioning may have made a breakthrough or become widely used in this field. The field of virology was in a downturn from 2010 to 2014, with only one publication, with a gradual increase in relevant studies after 2015.For scholars studying drug repositioning-related topics, knowing which journals publish relevant research is important in deciding which journals to read or submit their research studies to. A total of 2,988 publications related to drug repositioning research were published in 845 journals during the period of 2010–2020. The top 25 journals in terms of a total number of studies published are shown in  Sci Rep topped the list with 75 studies published, followed by PLoS One (73; 2.52%), J. Biomol. Struct. Dyn (67; 2.45%), Bioinformatics (53; 2.25%), and BMC bioinformatics (50 articles; 1.78%). The rest of the journals had a share of less than 1.5%. In terms of total citations (TC), at present, studies in Drug Discov. have been cited a total of 2,119 times over the past 10\\xa0years, followed in rank by PLoS One (1800) and Bioinformatics (1,677). For ACPP, Drug Discov. Today still holds first place with a high frequency of 50.45 times, followed by PLoS Comput (33.14 times). The impact factor (IF) of a journal is calculated by dividing the total number of citations of all publications in the journal in the previous two\\xa0years by the number of publications (). Thus,  shows that the ACPP of drug repurposing publications included in most journals is much higher than that of IF, which roughly verifies that the number of scholars interested in drug repurposing is relatively high. In terms of the impact factor (IF) of specific journals, except for Oncotarget and BMC Syst. Biol., which have not been included in SCI since 2018 and 2020, Brief. Bioinform. has the highest value of 11.622, followed by Drug Discov. Today (7.851), Bioinformatics (6.937), Cancers (6.639), Eur. J. Med. Chem (6.514), and Expert. Opin. Drug Discov. (6.098). The bubble chart shows that J. Biomol. Struct. Dyn. featured 64 publications in 2020, compared to a combined total of only four publications in the previous ten years; the Oncotarget journal inclusion in this category peaked in 2016–2017 ().For scholars interested in the topic of drug repositioning, it is useful to know how other researchers are working on the issue to facilitate communication and collaboration between scholars. A total of 15,620 authors contributed to studies within our measurement consideration, and  shows the top 20 prolific authors by a number of publications. Of these 20 highly productive authors, seven were from the United States, three were from Argentina, and two were from Germany, indicating a relatively high concentration of drug repositioning research in certain countries. In addition, the NIH (United States), Case Western Reserve University (United States), Tech University Dresden (Germany), and the National University of La Plata (Argentina) each have two of these academics.Cheng, FX leads the list with 25 research studies, followed by Talevi, A (23) and Mucke, HAM (22). For the list of corresponding authors, the top three remain, in order, Mucke, HAM (22), Cheng, FX (17), and Talevi, A (17). In terms of ACPP ranking, Butte, AJ was ranked first with 154.33 points, followed by Cheng, FX (100.56), Tang, Y (95), and Dudley, JT (82). Cheng, FX still has the highest h-index at 21, followed by Zheng, W (17), Talevi, A (12), Xu, R (11), and Schroeder, M (11). The h-index has two drawbacks when researchers of the same topic are compared with each other (). One is that the scholar’s h-index does not decrease over time but only grows or stays the same, and it is not possible to obtain information on whether the scholar is still in an academic career. In this study, we narrow the study to the most recent publications from 2010 to 2020, taking into account the timeliness of the h-index response information. Second, older scholars usually enter academia earlier and have an advantage in their h-indexes in comparison with those of younger scholars. Therefore, this phenomenon must be targeted for analysis or illustration. Thus, by combining the authors’ educational experiences and employment relationship changes that were recorded in the WOS database and ORCID business cards, we inferred that more than half of the scholars in the top 20 in terms of the number of publications received their Ph.D. before 2008, and two scholars, Mucke, HAM and Zheng, Wei, are older. In contrast, Cheng, FX, a scholar from Case Western Reserve Univ, completed his Ph.D. without a gap in 2013 and may have a longer academic career in the future; therefore, Cheng, FX’s h-index in the field of drug repositioning is likely to grow more in the future and Cheng, FX is likely to have more academic influence.To reveal the focus of drug repositioning research and research trends, the author keywords and the highly cited and hot research topics of the ESI for each of the 2978 publications were analyzed, which were also derived from the core database of the WOS database (SCI-E/SSCI) (). Highly cited studies were defined as studies in the top 1% of the citations for all studies in the same ESI discipline within the 10-year range of inclusion of ESI inclusion (). A hot research topic of the ESI refers to a study published in two\\xa0years with a citation frequency within one of the corresponding disciplines in the world in the last two\\xa0months ().Author keywords tend to provide more information and have thus become a widespread focus (; ). The data of 6,083 author keywords in the search results were merged to make keywords with the same meaning represented by a single unified word. In the end, 5,616 author keywords were obtained. It should be specified that some publications without author keywords were excluded from the statistical analysis. Of these author keywords, 4,296 were used only once, representing 76.50% of the total. A total of 1,216 (21.65%) appeared 2–10 times, 79 (1.41%) appeared 10–20 times, 37 (0.66%) appeared 21–50 times, and the remaining eight (0.14%) were used between 51 and 1,500 times. All keywords cumulatively appear a total of 12,400 times, while the top 30 most used author keywords appear 2,967 times alone, or approximately 23.93%, as shown in . The comparison of keywords in recent years allows for tracking the frontiers of research and predicting hotspots and trends in drug repositioning research. The bubble plots show the three dimensions of the data, namely, the year of publication, the author’s keywords, and the number of corresponding publications. The horizontal change in the size of the bubble illustrates the increasing trend of author keywords over time, the vertical size of the bubble shows the most popular keywords in that year, and the numbers in the bubble indicate the frequency of author keywords and the number of publications.The top 30 keywords include five diseases: “COVID-19/SARS-CoV-2/Coronavirus” (239), “Cancer” (75), “Alzheimer’s disease” (48), “Breast cancer (36)\", and “Chagas disease” (27). Drug names appear four times, “Antiviral drugs” (56), “Metformin” (35), “Statins” (28), and “Antibiotics” (25), which reveal the diseases and applications to which drugs were often repositioned during these 11 years. There were four subject categories, “Bioinformatics” (43), “Polypharmacology” (42), “Systems biology” (42), and “Precision medicine” (28) and eight occurrences of research methods, namely, “Virtual screening” (81), “Molecular docking” (64), “Machine learning” (54, eighth), “Clinical trials” (36), “High-throughput screening” (35), “Connectivity map” (28), and “Molecular dynamics simulation” (26).In the context of the pandemic in 2020, there was a surge in research on the subject, with “COVID-19/SARS-CoV-2/Coronavirus” topping the list of keywords as soon as they appeared that year. “Virtual screening” is a research method that appeared seven times more frequently in 2020 than in the previous year. Since “Drug repurposing/repositioning” is a subject matter and a strategy for drug discovery/drug development, it would not make much sense to analyze these two keywords. Molecular docking is one of the core steps of virtual screening, and the COVID-19 pandemic generated many opportunities for the practice of drug repositioning. Therefore, high-quality studies of the keywords “COVID-19/SARS-CoV-2/Coronavirus”, “Virtual screening”, and “Molecular docking” were surveyed, as shown in the bubble chart, in the past two\\xa0years, reflecting the relevant research trend in recent years. Wang, F et al. developed a new free reverse docking server based on a consensus algorithm (combining several docking algorithm strategies) to address the original shortcomings of computational molecular docking in drug repositioning, such as a low success rate, cumbersome operational steps, and reliance on code writing (). M Lapillo et al. extensively evaluated the performance assessment of docking-based target fishing methods and developed a consensus docking-based target fishing tactic (). In a virtual screening process, Gervasoni, S. conducted a literature search for molecular binding sites for SARS-CoV-2-associated protein targets while combining pocket and docking searches to propose a new pocket mapping strategy that identifies binding cavities with significantly better performance than pocket detection alone (). Xie, L et al. screened antitoxic drugs based on the multitarget structure of the pathway center and stated that this inhibition of multiple targets in one pathway would be more effective than targeting a single protein, and the chance of drug resistance was smaller, which could be applied to other pathways (). Li, Z et al. reported a virtual screening method based on accelerated free energy perturbation absolute binding free energy (FEP-ABFE) prediction and stated that the virtual screening method based on the prediction of FEP-ABFE will play a role in many other drug repositioning studies (). After a series of drug repurposing computational screens and various validation activities by several scientists, it was agreed that raltegravir (; ), clonidine (; ), chloroquine and hydroxychloroquine () have therapeutic effects in the treatment of novel coronavirus.In addition, from the studies on the keyword “Machine learning” over the 10-year period shown in the bubble chart, it was found that the classical machine learning algorithms of support vector machines (; ; ), regularized least squares (; ), logistic regression (; ; ), and random forests (; ) have been widely used in inferring drug–target and drug–disease interactions.While the level of influence of a study is reflected by a combination of many aspects, the number of citations remains an important indicator (). Based on the definition of highly cited and hot ESI papers in  of this study, a total of 108 highly cited studies were obtained, of which 11 were hot research topics. Hot research topics are shown in . It should be noted that the first-ranked author is used here as a representative, and the corresponding institution is shown. This rule is followed in  of this study. All hot research topics were published in 2020, and with the exception of an article describing the damage caused by nonsteroidal anti-inflammatory drugs (NSAIDs) to multiple organs and new information on drug repurposing (), the remaining studies focused on drug repositioning therapeutic target studies in novel coronavirus pneumonia (; ), screening drug studies (; ; ; ; ), reviews of clinical trials (; ), and reports of other coronavirus therapeutic agents and vaccine studies (). From the perspective of cooperation, most of them were completed by a country’s independent agency. In terms of the countries and regions studied, four studies involved US scholars, five studies involved Asian scholars, and one contribution was from an African scholar. In addition, “A SARS-CoV-2 protein interaction map reveals targets for drug repurposing ()”, published in Nature by Gordon, DE with a total of 125 scholars from the United States, the United Kingdom, and France was the most cited publication with 952 citations.Eleven hot research topics were removed from the 108 highly cited ESI studies, and the top 20 most cited studies were selected from the remaining highly cited studies for analysis. In terms of year of publication, the study by Dudley, JT et al. published in NUCLEIC ACIDS RESEARCH in February 2011 was the earliest of these studies (). Five highly cited studies were published in 2013, and three studies were published as recently as 2020. Two studies were published in Nature, and one each was published in Nat. Rev. Drug Discov. and Nat. Med. subj. of Nature E; J. Med. Chem. L was next with two studies. There were 12 studies with the first author or coauthor from the United States, representing more than half of those in , followed by China (4), Canada (2), England (2), Germany (2), Japan (2), and Switzerland (2) in order of contribution of two or more studies. Nine studies were based on collaborations between different institutions in multiple countries. One of them, entitled “Alcohol-abuse drug disulfiram targets cancer via p97 segregase adapter NPL4”, published in Nature in 2017 by Skrott, Z et al. is a collaboration between scholars from six countries: Czech Republic, the United States, Denmark, Sweden, Switzerland, and China (). In TC, “DrugBank 5.0: a major update to the DrugBank database for 2018” () by Canadian University of Alberta scientists Wishart, DS et al. ranked first (1820 total citations). The most cited publication on an annual basis was “Network-based drug repurposing for novel coronavirus 2019-nCoV/SARS-CoV-2”, published in 2020, which was authored by Zhou, YD et al. and was the highest annual average cited publication with 609 citations (). The scientists Cheng, FX and Dudley, JT, contributed to two of these 20 publications and are important influencers in the field.The three studies published in 2020 focus on novel coronavirus-related drug rediscovery activities () and the use of deep learning techniques (; ).  and Pushpakom, S et al. (2019) provided systematic reviews of the methods and challenges of drug repositioning at that time (; ). Initially, Sirota, M et al. (2011) explored the role of integrating genome-wide computational approaches for predicting reusable drugs (), while from 2013 onward, Peters, JU et al., Medina-Franco et al., JL et al., Reddy, AS et al., and Anighoro, A et al. generally recognized the importance of combining multiple points of pharmacological knowledge for drug repositioning studies (; ; ; ). In the face of a worldwide health emergency caused by the Zika virus epidemic,  used drug repositioning to identify lead compounds for drug development (). Of course, techniques related to the mining of repositionable drugs through experimental high-throughput screening, a traditional experimental approach, are not without progress; for example,  () explored the challenges of transferring 3D cell culture technology to the use of high-throughput screening (HTS) ().In 1995, Mchugh et al. investigated the immunomodulatory action mechanism of thalidomide in humans, which was the first relevant publication on drug repositioning (). The publication time can be divided into three phases: the growth period of 1995–2009, the steady growth period of 2010–2018, and the rapid rise from 2019 and beyond. The 2978 publications studied between 2010 and 2020 were completed by 15,338 authors from 3,530 research institutions in 89 countries, and at the time of this study’s completion, the WOS database had surpassed more than 1,400 publications in 2021 under the same search restrictions for the topic, with more than 31,000 citations for the year, supporting further evidence that the topic is still gaining momentum worldwide.The publication countries/regions are divided into three types: first, countries with a traditionally developed medical level, mainly developed countries in Western Europe, North America, and Oceania; second, countries with a developed pharmaceutical manufacturing industry, such as India and Japan in Asia; and third, developing countries with some research potential, such as China, Brazil, Argentina, and Mexico. In terms of national cooperation, Western European countries have shown a high degree of cooperation, with the United States, China, and the United Kingdom cooperating more frequently. This may be because Western European countries have a tradition of cooperation in the field of research, and the United States, China, and the United Kingdom are the most powerful countries in terms of drug repositioning publications and therefore cooperate more with each other. The United States accounts for half of the 20 most productive institutions, which may explain why the United States still publishes more than 50% of its studies independently, despite having the largest international collaborative network base, because it already has the most active and high-quality producing institutions within the country for research institutions seeking collaboration. Furthermore, 19 of these 20 institutions are universities and research institutes, and one is a company, HM Pharma Consultancy, which was established in 2000 to focus on drug repositioning for the development of new drugs (). This evidence suggests that the topic of drug repositioning is not only widely studied in academia but also has a place in the industry.The 2978 studies are spread across 74 research areas, but pharmacology and pharmacy and biochemistry and molecular biology account for a larger proportion of the total number of studies. It is quite notable that the majority of studies reported in biophysics did not rise significantly until 2020. The reasons for this may be the following: first, there was a breakthrough in basic research in this field in 2020 and second, due to the novel coronavirus, research in this direction has increased its application for the prevention and control of the pandemic.In terms of journals, Sci Rep ranked first, followed by PLoS One and J. Biomol. Struct. Dyn. In terms of lead authors, three have the most productive and influential positions: Cheng, FX is the most prolific author, based on the number of papers and h-index; Mucke, HAM is the most frequent corresponding author; and Butte, AJ is the top author in terms of ACPP ranking. Even though Latin American countries do not have an advantage in terms of national cooperation or the total number of institutional funding units, Latin American scholars have overcome many obstacles and are actively at the forefront of scholarship, contributing significantly to the field.Through the analysis of the authors’ keywords, cancer has been the main disease addressed by this method. Metformin has been found by many scientists to have a good inhibitory effect on various tumors, mainly in gynecology (; ; ; ), and it has become a specific drug that has been most frequently mentioned in drug reuse in recent years. In terms of “antiviral drugs”, scholars not only use drug repositioning to find antiviral drugs to treat diseases, such as Ebola (; ) and HIV (), that have plagued humans for a long time but also use this method to seek treatments for infectious diseases, such as Zika virus (; ) and novel coronaviruses that have threatened several countries and even the world. For these diseases, emergency research on drug repositioning has played an important role in reducing mortality, calming patient fears, and restoring economic production when no specific drugs or vaccines were initially available during the pandemic. The combination of precision medicine and drug repositioning studies, often used to seek treatments for rare diseases () and, in particular, genetically related diseases (), is expected to be fully developed in the future. In the past 2\\xa0years, “Virtual screening”, together with “Molecular docking” and “Machine learning”, has become the most cutting-edge and important research methods in related technology fields, constantly improving the accuracy of drug reuse and screening. Currently, to develop more efficient and accurate research, there are two trends in the use of drug repositioning. One is the combination of various methods, such as the use of text mining and network analysis, and the creation of statistical models for predicting semantic link association to assess the relationship between pharmacological target pairings (); text analysis combined with machine learning () to develop drugs for Parkinson’s disease; prediction of new DTIs using data from multiple databases (); and the obtained relocated anticancer drugs were verified by cross-validation, literature, and experimental verification (). Second, the most advanced algorithms are applied and improved, such as matrix decomposition (; ; ; ; ) and matrix completion (; ) and deep learning (; ; ; ; ; ).In fact, some of the studies in the list of highly cited research topics on novel coronaviruses drug repurposing studies are currently approaching 3,000 citations on Google Scholar (). The percentage of highly cited studies and hot research topics related to novel coronaviruses is also a good indication that the method has made an indelible contribution to the study of novel coronaviruses and similar infectious diseases. Auxiliary technology for the experimental screening of traditional drugs is also developing (), which also promotes drug repositioning or other drug development processes. Furthermore, the high-quality results of Elfiky, AA, a scientist from Cairo University, Egypt (), suggest that relevant research in some economically underdeveloped countries may reach top levels worldwide due to the return or affiliation of some prominent scientists.For this research, the literature on drug repositioning research published in the SCI-E and SSCI sections of WOS core journals from 2010 to 2020 was analyzed based on bibliometrics and DDA software. This area has been of interest to scientists since the end of the 20th century and entered a period of rapid growth in 2019, with the peak far from being reached. Using bibliometrics as a tool, the United States has become a world leader in terms of the number of submissions, number of high-quality studies, funding support, strength of research institutions, and number of top scholars, followed by China and the United Kingdom, where more research is being performed in this area. As a method of drug discovery, drug repurposing is closely related to the development of various biomedical disciplines, and computer-related disciplinary methods, such as mathematical computational biology and computer science, have taken an important place in the research of this field in the last decade. The authors’ keyword analysis suggests that research in the field of the novel coronavirus will remain valuable until the associated pandemic is completely contained. Virtual screening, molecular docking, machine learning, and other related technical fields still need long-term development to achieve efficient and accurate repositioning of drugs (). Precision medicine, combined with drug repositioning, is the most promising direction for the future. In conclusion, drug repositioning can help to treat more diseases, such as drug resistance, poor drug selectivity, and limited therapeutic options.This study may help some scholars with an initial interest in drug repositioning-related research to gain a concise and rapid understanding of the current state of global research, as well as offer some relevant information to institutions or groups seeking collaboration.It is worth noting that this study has some biases and limitations. First, there are still some issues with the publications included in the study based on subject terms: 1) some relevant publications that do not use the search formula in this study may have been excluded from this study and 2) there may also be a small number of articles whose use of some of the aforementioned search terms deviates significantly from the general understanding; yet, such publications are included in this study. Second, some extraneous factors distort the credibility of the bibliometric statistics. 1) When analyzing the keywords of publications, some publications are excluded from the statistical analysis because they do not list author keywords (e.g., ()). 2) Excessive self-citation by some authors () inflates the actual level of interest in the publication. 3) for a publication, when an author submits more than one institution’s address information, this publication is counted as research results by each institution. Finally, in future work, patents from the WOS database associated with the topic of drug repositioning will be analyzed to provide another perspective on the situation of the topic in terms of applications and technological innovations.'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 41
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_2500[3]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 174
        },
        "id": "-Cx5zETBY5cz",
        "outputId": "4309efae-5e71-4ad3-88cc-b6c7f9e72e32"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'The sudden outbreak of SARS-CoV-2 in 2019 took the world by storm and despite there being vaccines, numerous other alternative treatment approaches are also being researched. Coronavirus disease-2019 (COVID-19) is a communicable disease caused by severe acute respiratory syndrome coronavirus-2 (SARS Cov-2). This disease was first detected in Wuhan, China, and has expanded its reach globally, leading to myriad deaths. According to World Health Organisation, the total number of COVID-19 confirmed cases worldwide was found to be 53,22,01,219 and total number of deaths accounted to 63,05,358 as of 10th June, 2022. Among many regions, Europe holds the top position in highest number of confirmed COVID-19 cases (22,24,17,177) then followed by America (15,89,83,746), Western Pacific (6,17,35,224), South-East Asia (5,82,17,287), Eastern Mediterranean (2,18,07,376), and Africa (90,39,645) respectively []. These statistics reveal that even today, the disease is still infecting several people, many of whom die due to complications. This pandemic requires quick and easily accessible knowledge for hastening the development of treatments and diagnostic tests []. However, the development of specific drugs for this disease takes several years, paving the way for exploring alternative approaches. One such potent strategy is repurposing drugs, for alternative utilization of already existing and approved drugs. These drugs must have already undergone all safety and clinical trials, which would further speed up the drug discovery process. The COVID-19 pandemic has ensured research to explore various alternatives, drug repurposing being one of them, as part of the disease management strategy. Several drugs such as chloroquine, hydroxychloroquine, remdesivir, tocilizumab, kaletra, and favipiravir are currently being testing in clinal trials for repurposing against COVID-19 [].SARS-CoV 2 belongs to the Nidovirales family, under the beta coronaviruses genera along with other viruses such as Middle East Respiratory Syndrome Coronavirus (MERS-CoV), HCoV-HKU1, and the HCoV-OC43 (Human Coronavirus variants) []. The replication of SARS-CoV-2 occurs with the help of mRNAs. These viruses have the largest RNA genome among other viruses that have RNA as their genetic material. The RNA genome is approximately 30 kb in size and runs in the 5’ to 3’ direction, with single-stranded positive-sense RNA (+ssRNA). A cap at 5’ end and poly-A-tail at 3’ end at the end of the genomic RNA is a feature of the genetic material. Several open-reading frames in varying numbers are present in the genomic RNA and it encodes for 4 structural proteins, 16 non-structural proteins, and a few accessory proteins [].The nucleocapsid in the virus protects the viral RNA genomes from external environmental conditions. At the relaxed position, the nucleocapsid appears to be in a helical shape, but within the virus, it appears spherical. The genome of SARS-CoV-2 codes for four main structural proteins when compared to five in other coronaviruses leaving out the hemagglutinin-esterase (HE) protein. Nucleocapsid (N), membrane (M), envelope (E), and spike (S) are the four important structural proteins expressed by the SARS-CoV-2 virus []. On the surface of the virion, three structural proteins (M, E, and S) are present, while the nucleocapsid protein resides in the core region bound to genomic RNA. The spike proteins of the virus help in the attachment, entry, pathogenesis, and tissue tropism, whereas, the infection is carried out by the envelope protein. Ribonucleoprotein is formed with nucleocapsid protein’s assistance and the membrane proteins aid in the overall viral architecture []. The SARS-CoV-2 virus enters with the help of spike proteins by binding to the human angiotensin-converting enzyme 2 (hACE2) through its receptor-binding domain (RBD) and human proteases that make them proteolytically active [].Beta, gamma, delta, and omicron are the latest variants of concern for the SARS-CoV-2 virus. South Africa was first to report the beta variant of COVID-19, gamma was noted first in Brazil, the delta in India, and omicron in western Europe. Omicron which is the most recent variant of COVID-19 carried all the mutations that were present in the previous variants along with some new improved functions such as showing partial resistance to the vaccine and enhanced transmissibility compared to others []. The major coronavirus variants can be classified into two []: previously circulating variants of concern (VoC) and currently circulating variants of concern (Table ).These five major variants of concerns are the current targets for research to find repurposed drugs for exploring treatment strategies. In addition to this, certain amino acids are conserved among the coronavirus viruses, while some have a unique mark in the amino acid indicating lineage mutations. The variants of concern belong to various lineages. B.1.1.7 has the mutations H69-V70del, P618H, and N501Y, all indicating alterations in the amino acids. Lineage B.1.351 has mutations K417N, N501Y, and E484K. Likewise, another variant of concern of lineage P.1 has mutations N501Y, K417T, and E484K. B.1.526 lineage is mutated at D614G, T95I, D253G, and S477N. B.1.427 has mutations at L452R and D614G. B.1.429 is mutated at sites D614G, S13I, W152C, and L452R. B.1.617 variant has variations P681R, E484Q, and L452R []\\xa0(Fig. ).Apart from the variants identified in the most common lineages, there are approximately 1,516 SARS-CoV-2 lineages to date [, ]. To identify the variants in all the existing lineages and those that exist not only in the spike protein of SARS-CoV-2, this study also performed a search in  and summarized the different types of variants (Tables , , , and ). Synonymous, non-synonymous, deletions, and non-coding variations were identified, and changes in the amino acid positions, and the reference and mutated codon as screened in all 1516 lineages of the coronavirus. As per the latest data as of 6th June 2022, 131 non-synonymous mutations have been noted, 51 synonymous variants, 11 deletions, and 11 non-coding variations. These mutations are in various regions of the virus, including the spike proteins, nucleocapsid proteins, different open reading frames, and membrane and envelope proteins.Additionally, some mutations are characteristic of specific regions, while some others are present in all the regions in SARS-CoV-2. For instance, in Asia, three mutations in QLA46612 (isolated from South Korea) were observed within the same sequence []. This particular study stressed the fact that binding sites of the ligands and potential drugs change if the structure of the spike protein is damaged. Moreover, the study also tried to comprehend if more than one mutation affects the binding of the drug, thereby altering the potentially identified drug binding sites. It was later revealed that drug binding sites were greatly affected in structures with multiple variations []. Thus, the SARS-CoV-2 variants may have different binding sites making them identifying a suitable drug challenging. However, this arena will pave the way for prospective vaccine trials.It has been reported that traditional Chinese medical therapies have increased the clinical manifestation of the virus among patients, along with enhanced the severity of the disease []. Major concerns of the recent vaccines which were developed during the pandemic were their safety and efficacy, the emergence of new variants, challenges that occur due to vaccine distribution among a larger population, and many more. Recently, an mRNA-based vaccine (BNT162b2) was reported to have adverse side effects such as fatigue, headaches, and immense pain at the site of injection. Variants are concerns among the present therapies due to severe mutations that occur in the spike protein as well as the increased transmission rate when compared to its predecessor. Distribution of the vaccines to the greater population becomes highly challenging to the lack of storage facilities in different geographical locations as well as the stability of the vaccines which highly depends on the storage conditions such as temperature []. These limitations warrant the use of alternative approaches, with drug repurposing being the most effective strategy that is currently being researched.The concept of drug repurposing holds promises of swift clinical trials and a reduced cost than drug development de novo. Drug repurposing, also known as drug repositioning, is the process of using approved and existing drugs for treating diseases other than the intended. Drug repurposing is pragmatic and alluring, given the considerable time and cost requirements for typical drug development []. Additionally, a great number of potential drugs do not reach clinical trials, with lesser than 15% of the compounds receiving approval []. Some examples of approved repositioned drugs include aspirin, the cyclooxygenase inhibitor for coronary heart disease, antibiotic erythromycin for gastric motility, and sildenafil, a phosphodiesterase inhibitor for erectile dysfunction []. Other drugs that have disconcerting side effects also merit repurposing as corroborated by the successful implementation of thalidomide, an antiemetic against multiple myeloma [].Currently, there is a huge interest in the repurposing of drugs to speed up the identification of drugs that can prevent or treat COVID-19. The chief importance of repositioning is to accelerate the traditional drug discovery process by the detection of novel, safe, and effective clinal-use drugs in humans. The primary rationale for this is the utilization of drugs that have the same molecular pathways but are involved in different diseases []. The mechanism of repurposing depends on the type of targets that the drugs are being used against. One of the most valuable drivers of repurposing has been the chance discovery of the pharmacological activity of new drug targets, implying a newer indication of drug use []. For instance, drugs lopinavir and ritonavir have been used for the treatment of Human Immunodeficiency Virus-1 and 2 (HIV) []. These drugs were also the first drugs to be repurposed for the treatment of COVID-19. The main reason for this is due to the resemblance of the SARS-CoV and MERS-CoV (beta coronavirus) genome sequences to SARS-CoV-2 [, ]. Thus, the drugs being utilized for the treatment of MERS-CoV and SARS-CoV-2 could be essentially used for COVID-19 treatment, implying that the mechanism of action of repositioned drugs against their targets should be similar to or resemble its family target members [, ]. Successful administration to patients with moderate COVID-19 symptoms has been previously observed, confirming the hypothesis of drug repurposing [, ]. From all this, it can be elucidated that the concept of drug repurposing relies on two major scientific bases: (1) the idea of pleiotropic drugs, (2) the discovery of drugs via human genome interpretation that certain diseases share the common targets.It begins with the extraction of primary data on the drug, disease, and target. A drug-disease network is eventually constructed and its features of it are extracted to filter the data and pre-process it thoroughly. A computational-based evaluation is carried out via methods detailed in the following sections, including advanced approaches such as machine learning that entails the structural analysis of the drug candidates. The pharmacokinetic studies ensue, which are essential for obtaining an FDA approval for use by the public. This general workflow and roadmap to drug repurposing are detailed in Fig. .Drug repurposing is known to decrease the costs needed for the development of new drugs, with additional savings in pre-clinical phases I and II []. Aside from this, the scope of drug repurposing expands to the following, with instances in COVID-19 research, as discussed in the below sections:Precision medicine: This area is an emerging strategy that studies the variability of individual genes, the lifestyle, and the environment for each person to decide upon the correct treatment approach []. Disorders with traits that are common are characterized as a single condition and are part of a spectrum of illnesses. Safer and more effective medications can be found if the drugs could be tailored to the variations in the genome of the individual, their proteome, transcriptome, or metabolome. This approach would help target the specific types of a general condition, with repositioning paving the way for precision medicine. Recent studies have implicated the use of computational techniques such as machine learning algorithms as a precision medicine tool for drug repurposing for the treatment of COVID-19 []. The decision tree algorithms used in the study identify patients with longer survival rates and better oxygen support when treatment was provided and suggest corticosteroids and remdesivir as effective COVID-19 repurposed drugs.Rapid therapeutic options: With the advent of the COVID-19 pandemic and reduced therapeutic options for the disease, repurposing offers rapid therapeutic strategies to combat the disease. El Bairi et al. [] reviewed various anticancer drugs for the management of COVID-19. Likewise, another study has elucidated and suggested the use of interleukin inhibitors, Janus-kinase pathway inhibitors, interferons, vascular endothelial growth factor (VEGF) inhibitors, immune check-point inhibitors, Bruton tyrosine kinase inhibitors (BTK), and XPO-1 (a selective inhibitor of nuclear export) as some of the alternative drugs that could be repurposed for rapid COVID-19 treatment [].Identification of multiple purposes for existing drugs: The latest applications of already approved drugs well beyond the scope of their existing medical aid in the identification of multiple purposes for such drugs in just four major steps–identification of compound, acquisition of the compound, development of the compound, and FDA approval after monitoring of its safety. This reduces the time taken for traditional drug discovery, lowers developmental cost as well as risk, and also reveals novel pathways and targets for already approved drugs, allowing them to have higher functionality [, ]. For instance, a recent study has repurposed antiviral protease inhibitors such as chloroquine, remdesivir, and lopinavir, which have already shown efficient inhibition of the SARS-CoV-2 virus in in vitro studies, using the extra-cellular vesicles, to make it more robust against the virus [].Identification of direct-acting and host-targeting antivirals: Drug repurposing also has applications in the identification of existing antivirals that are either direct-acting or host-targeting. Studies have shown that anti-hepatitis C combinations act as potential direct-acting main protease inhibitors of SARS-CoV-2 []. Similarly, another study suggested p38 MAPK (mitogen-activated protein kinase), as a vital host target for inhibiting the replication of coronaviruses, after studies proved that Ralimetiniib mesylate was found to be active in both Calu-3 and HeLa-ACE2 cells [, ].Thus, drug repositioning for COVID-19 research has now taken the front seat and has a huge scope for tackling the SARS-CoV-2 virus.Drugs are small molecules that affect the molecular pathways of important protein targets that are most likely responsible for causing a specific disease. These molecules have a lower molecular weight and are typically designed using rational drug design or structure-based drug design. With increasing incidences of emerging diseases such as COVID-19, for which immediate treatments were not available, the need for drugs to treat the disease became indispensable. With the SARS-CoV virus, mutating at a fast pace and affecting millions of people globally, there was a pressing need to address this issue by using all alternative approaches to discovering a drug [, ]. In this light, drug repurposing came into the picture. Comprehending detailed molecular mechanisms between the drugs and their targets is critical in the discovery of drugs. Even though several biochemical and biophysical methods can explicate the molecular interactions of the small molecules with their proteins, the most powerful tool is structural and computational biology which can aid in interpreting the mechanism of action of the interaction complexes []. Bearing this in mind, drug repurposing is the alternative approach meant to identify the most suitable drug candidates from among the existing libraries, to combat specific diseases.Currently, several approaches exist that pave the way for drug repurposing. Advancements in computational chemistry and chem-bioinformatics pave the way for exploring more accurate predictive repurposing.Regarding developments in multiplex assays, cell-based screening, mining of data, chem-informatics and bioinformatics approaches, and their databases, pharmaceutical industries have now shown augmented keenness in digging up molecules that had once failed due to several reasons []. Generally, there are three important steps before repositioning drugs: (a) drug candidate identification for the specific disease, (b) assessment of the selected drug candidate in pre-clinical models, (c) safety evaluation in phase II clinical trials [].The approach of repositioning itself can be subdivided into experimental and computational approaches. Using these approaches in combination yields better outcomes than using them individually. Computational strategies are driven by a huge amount of data that entails the utilization of genetic expressions, the structure of the chemical compounds, and the proteomic drug candidate data []. Several computational medicinal approaches add value to the drug repurposing process.Structure-based approaches such as virtual screening help in the identification of small molecules from an existing library, that can be used for repurposing. Previous studies have most commonly employed molecular docking for virtual screening against targets of SARS-CoV-2, such as the main protease, with possible drugs for repurposing such as digitoxigenin, beta-eudesmol, crocin, rutin, ritonavir, emetine, hesperidin, and indinavir [, ]. Likewise, other virtual screening approaches used previously for COVID-19 drug repurposing include combinations of docking with molecular dynamic (MD) simulations and calculation of free energy using MM-GBSA studies [] and two-way docking with in silico ADMET studies []. The following strategies fall under structure-based approaches for drug repurposing.Molecular docking is a chem-informatics strategy to predict the conformation of the docking site/binding site of a drug candidate to the protein target []. The prediction of the best conformation where the drug and protein form an interaction complex, that will stop the molecular pathway of the target, in turn affecting the occurrence of the disease is the idea of docking []. Novel interactions that will help in understanding the potential drug candidates for repurposing can be identified using docking studies. The advantage of this strategy is that it can be followed by other computational validation techniques such as molecular dynamics simulations and Quantitative Structure–Activity Relationship studies (QSAR), to understand the complex stability and thereby increase the drug repurposing opportunity. The authors’ previous studies have used this computational technique against various targets of SARS-CoV-2 [, ]. The disadvantage is that not all proteins of interest are available, due to a lack of macromolecular structure. Applying more than one computational medicine approach will provide further authenticity to the predicted results.The static binding energies between the drugs and targets obtained via molecular docking studies are further simulated dynamically in real-time to comprehend the stability of the interaction complex in biological environments. Molecular dynamic simulations have proved to be one of the most reliant strategies in drug repurposing. Elfiky [, ] proved via MD simulations that Setrobuvir, YAK, and ID-184 had the best antiviral activity against RNRP (RNA-dependent RNA polymerase) of SARS-CoV-2, after screening out 31 compounds that were in clinical trials or had known antiviral activity. Similarly, MD simulations identified the stability of lopinavir complexed with SARS-CoV-2 main protease as the most stable in biological conditions [, ], along with other identified antivirals such as carfilzomib, valrubicin, elbasvir, and eravacycline.Quantitative structure–activity relationship acts as a tool to identify the relationship between the structure and the activity of compounds. With regard to COVID-19, this approach combined with docking and simulations has brought immense benefits to the design and repurposing of drugs for SARS-CoV-2. A study has previously created a QSAR model founded on machine learning strategies utilizing several hundred inhibitor molecules of the SARS-CoV-2 main protease. This model was employed for virtual screening and docking, followed by shortlisting of the 20 best candidates for MD simulation validation []. Similarly, another recent study designed a binary-QSAR model for the virtual screening of several drugs that are FDA-approved and are in clinical studies against the main protease of SARS-CoV-2 []. This QSAR protocol resulted in cefuroxime pivoxetil as a potential drug repositioning molecule against SARS-CoV-2.Network-based approaches are commonly employed in repurposing due to the capability of amalgamating and integrating several sources of data [, ]. Advancements in high-throughput technology and computational approaches allow the modeling of biological systems by the network which facilitates diagnostic and pharmaceutical research that is structure-guided, with the possibility of identification of novel targets [, ]. Studies have stated the importance of drug-drug networks, drug-target networks, and drug-disease networks in drug repurposing []. The two-most common types of network-based strategies in computational biology include network-based propagation and network-based clustering approaches []. King et al. [] have previously studied PPI network-clustering technique for drug repurposing, while Vanunu et al. [] have used the disease-gene network propagation technique to accurately predict the disease-gene relationships.Recently, the use of AI technology for drug repositioning has surfaced with several studies focusing on repurposing for COVID-19 research. A deep-learning-based approach to screen a huge number of molecules having allocated datasets to seek compounds possessing SARS-CoV-2-inhibiting activities was carried out previously by Ke et al. []. To validate the designed model and verify the antiviral activity of the compound that the model predicted, an in vitro cell line for feline coronavirus replication was also set up in the study. Likewise, Beck et al. [] utilized an already trained deep-learning prediction model for drug-target interactions known as Molecule Transformer-Drug Target Interaction (MT-DTI), to find antiviral drugs that are commercially available and which could perturb SARS-CoV-2 components including RNA-dependent RNA polymerase, proteinase, and helicase. Thus, AI-based techniques are providing the scientific communities a chance to fight COVID-19 in a much faster way, owing to the accessible web servers, tools, and models that reduce the time required for a drug to come into the market.The scope and developments of computational medicinal chemistry toward COVID-19 treatment strategies are summarized in Fig. .Several computational approaches for repurposing have been used over the years to enhance the efficiency of the predictive outcomes and decrease the drawbacks. Currently, advancements in the field of computer science have led to the use of machine learning methods, network-based techniques, and matrix decomposition methods, along with other algorithms such as tensor decomposition for repurposing []. This technique has been performing better than several basic algorithms such as Network-based Random Walk with Restart (NRWRH), SNScore, and Collective Matrix Factorization (CMF). This technique solves the difficulty of negative sample training in cases where deep learning methods are used such as multilayer perceptron, auto encoders, and deep belief networks []. The tensor decomposition technique involves the construction and decomposition of 3D tensors that enable the associations among protein targets, drugs, and diseases [].Topological data analysis (TDA) is then taken into consideration to examine the drug properties, diseases, and disease targets to be able to recover the missing drug associations, diseases, and their targets, and to also predict the new repositioning candidate []. Moreover, drug repositioning can be performed on the drug candidates that arise from two clusters with robust connections, instead of assuming only the new-drug associations. To define these clusters, the diseases with similar molecular mechanisms are more inclined to appear within the same cluster. This approach, therefore, offers huge potential to explore predictive drug repurposing research.There are several applications of computational biology and chem-bioinformatic approaches for drug discovery in COVID-19 research. A recent study has developed a multi-model deep learning-based algorithm for drug repurposing to aid in COVID-19 research. This work utilized the multimodel-restricted Boltzmann machine approach (MM-RBM), which can associate data about multiple modalities, to amalgamate two data types, inclusive of the chemical structures of small drug molecules and genes expressed differentially []. The results from this study showed clusters that were employed to identify the drugs that were notably similar to the proposed and existing COVID-19 treatment medication. This work also demonstrated results in predicting drugs with minimum side effects and high efficiency.Another recent study proposed a mechanism-driven neural network technique called DeepCE, that employs a graphical network to model the gene–gene associations and the chemical structure-gene associations to predict the gene expression profile []. This work also proposed a data augmentation approach that extricates only beneficial data from a set of variable datasets, to screen novel drug candidates for COVID-19. A recent application of signature mapping on COVID-19 research demonstrated the use of Connectivity Map databases [] for generating drug perturbation profiles []. A modified version of this study was implemented by Jia et al. [] where the input used was the expression profile data from healthy and infected individuals to generate a pathway-based drug repurposing framework. Co-clusters were identified before performing reverse signature matching [].Another recent study designed and developed Deep Docking, which uses QSAR (Quantitative Structure–Activity Relationship) models to predict the docking scores of drug candidates that target the SARS-CoV-2 3CL protein. Deep Docking docked all the compounds and provided a list of potential drug hits, that could be used to repurpose for COVID-19 []. Several studies have implemented random-walk algorithms to identify drug repurposing candidates for COVID-19 [–]. The pathogenic mechanisms instigated by the spike protein using the information from all related coronaviruses were also explored and random walk algorithms were implemented on the molecular networks to recognize the most relevant COVID-19 drug targets []. Thus, these computational perspectives on drug repurposing provide a unique outlook on the alternative treatment options for the ongoing COVID-19 disease.Drugs such as lopinavir, ritonavir, boceprevir, carmofur, and doxycycline act on the 3-chymotrypsin-like proteins/SARS-CoV-2 main protease that play an important role in cleaving and releasing the 16 non-structural proteins []. Fostamatinib disodium and platycodin target the Nsp3 proteins which play a major role in the infection and pathogenesis. The RNA-dependent RNA polymerases (RdRp, Nsp 12) that are primarily involved in important activities such as transcription and replication of the viral RNA genome are targeted by drugs such as remdesivir, favipiravir, and ribavirin []. For the helicase (Nsp13) of SARS-CoV-2, a multifunctional protein that helps in the proliferation and replication of the virus, drugs such as vapreotide, atazanavir, daclatasvir, and bismuth potassium citrate commonly block this function of the target. Likewise, dinucleoside 123 and raltegravir affect the pathways of methyltransferases (guanine-N7-methyltransferase, 2’-O-methyltransferase), while gliclazide and memantine affect the SARS-CoV-2 E protein, which is the envelope protein of the virus. Arbidol and chloroquine target the spike protein that helps in the viral entry, along with ACE2 protein in the host, thereby affecting the spike protein pathway [] (Table ).\\nThe ACE receptors are blocked via losartan and captopril, transmembrane protease serine 2 (TMPRSS2) is blocked by camostat mesylate and with anyone, and cathepsin L is inhibited by hydroxychloroquine, which and teicoplanin. The viral neuraminidases and the DPP4 cell receptor pathways are hindered due to the action of oseltamivir and sitagliptin, while pemirolast, nitrofurantoin, isoniazid pyruvate, eridictoyl, hypericin, cepharanthine, and ergoloid block the human ACE2 and the S protein of SARS-CoV-2 []. The most important Janus kinase pathway is hindered by baricitinib, RdRp, helicase, exo-nuclease, and methyltransferase of SARS-CoV 2 by Efavirenz and Poly [ADP-ribose] polymerase 1 (PARP-1) by Mefuparib hydrochloride. Poly [ADP-ribose] polymerase 1 (PARP-1), glycogen synthase kinase 3 (GSK-3), and also biological pathways such as NF-kappa B signaling are all affected by the use of repurposed drugs toremifene, sirolimus, mercaptopurine, irbesartan, emodin, dactinomycin, and melatonin []. Other common drugs such as nitazoxanide, ivermectin, and azithromycin tend to symptoms of the virus, and play a role in inhibiting the role of SARS-CoV-2.Despite these drugs being implicated as a potential treatment for COVID-19, ethical issues arise regarding the use of these drugs without proper clinical phase trials and FDA (Food and Drug Administration) approval. Therefore, a list of the FDA-approved drugs for the treatment of COVID-19 is provided in Table . Mahdian et al. [] and Yuce et al. [] state that Paritaprevir is important against 3-chymotrypsin-like proteins/SARS-CoV-2 Main protease and plays an important role in cleaving as well as releasing 16 non-structural proteins (Nsps). It also affects the ACE2-SARS-CoV-2-RBD complex. Likewise, ledipasvir, vancomycin, and sirolimus target the ACE2-SARS-CoV-2-RBD complex, a spike protein that helps in the viral entry along with the help of ACE2 protein in the host organism. Similarly, FDA-approved irinotecan, abemaciclib, estradiol benzoate, capmatinib, olaparib, lumacaftor, and pazopanib target CD147, a member of the immunoglobulin superfamily (IgSF) interacts with spike protein of SARS-CoV-2 virus [, ]. Other drugs such as nilotinib, rifabutin, dactinomycin, dihydroergotamine, bromocriptine, entrectinib, selinexor, elbasvir, quinupristin, rifapentine, rutin, digitoxin, diosmin, and antrafenine target 3-chymotrypsin-like proteins/SARS-CoV-2 Main protease that plays an important role in cleaving as well as releasing 16 non-structural proteins (Nsps).According to the latest updates from , phase 3 trials are completed for remdesivir, lopinavir/ritonavir, interferon beta-1A, hydroxychloroquine, and AZD7442 ( registration number: NCT04315948). Trials for Silmitasertib are active and in phase 2, while for Pioglitazone, phase 4 trials are completed ( registration number: NCT04473274). Phase 2 clinical trials are completed for dipyridamole 100 mg (NCT04391179), and phase 2 is completed for Leronlimab 700 mg (NCT04678830). L-ascorbic acid has phases 1 and 2 completed (NCT04357782), and phase 1 for aerolized hydroxychloroquine sulfate completed (NCT04461353). Similarly, drugs duvelisib (NCT04487886), ruconest (NCT04530136), and infliximab (NCT04425538) have phase 2 clinical trials completed. This data is provided in detail in Table  and the 2D structures of some of these drugs are shown in Fig. .\\nWhen remdesivir was administered intravenously to the first patient diagnosed with SARS-CoV-2 in the USA, no adverse effects were noted with some clinical benefits in the patients []. Evidence also showed an accelerated recovery rate of 4 days when remdesivir was employed []. However, when randomized, double-blind, placebo-controlled trials were conducted by other researchers, it was found that remdesivir was not associated with significant improvement statistically [, ]. Additionally, due to some adverse effects early on, remdesivir was then discontinued, and WHO has now provided a conditional recommendation for the use of remdesivir against COVID-19 []. Likewise, real-world usage of lopinavir/ritonavir revealed no beneficial response against COVID-19 hospitalized patients in a randomized, controlled, trial []. Overall, the available data show an unfavourable response of patients to lopinavir/ritonavir, and hence the use of this drug has to be monitored and controlled vigilantly. Despite there being studies on hydroxychloroquine, an analogue of chloroquine, where improvements have been noted in patients suffering from COVID-19 in a study conducted in Wuhan, there still seems to be very limited data on the reliability of the drug []. Interferon-beta-1A has been used in hospitalized patients for treating COVID-19 at Shaid Behesti University of Medical Sciences, in combination with other drugs. However, despite it showing promising outcomes, more clinical data is required to corroborate the reliability of the drug []. Therefore, the clinical data presented provide evidence of the efficacy of the use of these drugs,however, more research is indispensable for the proper application and employability of these drugs for SARS-CoV-2 treatment.According to literature, coronavirus papain-like proteases (PLpro), main protease (3CLpro), helicase, and RNA-dependent RNA polymerase (RdRp) are the considered to be the potential molecular targets and are involved in important functional activities such as protein synthesis, RNA transcription, translation, replication, processing, and infection respectively. All these proteins come under non-structural proteins (nsps) for coronaviruses []. The below sections discuss the important protein targets from a structural perspective.The papain-like proteases have a “thumb-palm-fingers” structure architecture that houses domains and subdomains. The active site residues/binding pockets present in these domains are His272, Cys111, and Asp286 which form the catalytic triad of the same protease/protein. These papain-like proteases (PLpro) share their structure with the ubiquitin-specific proteases (USPs) which have two main domains, ubiquitin-like (Ubl) domain, an N-terminal domain along with the catalytic triad which forms thumb-palm-fingers architecture [].RNA-dependent DNA polymerase is mainly involved in the replication and transcription of the virus. Since it plays an important role in viral infection and hence it is considered a potential molecular target of the SARS-CoV-2 virus, human telomerase reverse transcriptase (hTERT) was first thought to be RNA-dependent DNA polymerase but it is DNA polymerase that shares structural similarities with RdRp. Single-stranded RNA (ssRNA) encodes for RdRp proteins whose structure has a right-handed closed polymerase-like structure with the thumb, finger, and palm domains. At N-terminal, there is the presence of beta-hairpin nidovirus RdRp-associated nucleotidyltransferase (NiRAN) domain (residues ranging from D60 to R249) and C-terminal/polymerase domain (residues ranging from S367 to F920) which houses thumb-fingers-palm subdomains. There are seven motifs present in the RdRp protein of SARS-CoV 2 ranging from A to G, where five are said to be present in the most conserved domain, i.e., the thumb domain [–].Non-structural protein 5 coding gene encodes the main protease (3CLpro) which is mainly involved in the proteolytic activity of viral polyproteins and replicates. They share structural similarities between the 3C proteases of picornaviruses in terms of core homology as well as substrate specificities. The size of the protein is approximately 30 kDa which has three domains rich in cysteine amino acids. The protease comprises chymotrypsin-like folds along with a catalytic triad and a domain that is inundated with cysteine [].The nucleoprotein (N) of SARS CoV-2 binds to the genomic RNA (gRNA) to form the ribonucleoprotein (RNP) complex, a major virion core of the coronavirus. Nucleoprotein along with other proteins plays an important role replication and transcription of the virus. The N-terminal and the C-terminal domain of the nucleocapsid protein share similarities with CoVs N proteins. The N-terminal domain is a right-handed fist-shaped structure that has the topology of five beta strands sandwiched between two alpha-helices. The C-terminal domain is a homodimer with two protomers resembling a rectangular slab-like structure. Each protomer comprises two 3–10 helices, two beta-strands, and five alpha-helices respectively. The beta hairpin-like structure connects two protomers by inserting itself into their cavities [].There are three domains in the enveloped protein of the SARS-CoV-2 virus which comprise of N-terminal domain, C-terminal domain, and transmembrane domain. The overall size of the E protein is about 8.4–12 kDa and forms the integral membrane protein of the virus comprising 76–109 amino acids. There are about 7–12 amino acids at the N-terminal which is short and makes it hydrophilic in nature. The transmembrane domain has about 25 amino acids, out of which two are non-polar and neutral making it hydrophobic in nature. The C-terminal domain forms the majority of the protein structure [].The genome of the SARS-CoV-2 encodes for the ectodomain trimeric protein which projects out of mature virions and plays an important role in the initial attachment process and entry into the host organism. The spike glycoproteins in SARS-CoV-2 are strongly glycosylated and are trimeric in structure. The protomers present in the spike glycoprotein structure contain approximately 1260 amino acids. The first and foremost part of the spike glycoprotein is the surface subunit (S1) which comprises four domains (two SD1, SD2 subdomains, N-terminal domain, and C-terminal domain which is also the receptor-binding domain of the protein) with 672 amino acids in length. The transmembrane subunit S2 is made up of 588 amino acids and contains a transmembrane domain, cytoplasmic tail, and two heptad repeats [].The non-structural proteins of the SARS-CoV-2 are encoded by the open reading frame 1a and ab (ORF1ab) which is present in the downstream region of the 5’ end of the ssRNA (single-stranded ribonucleic acid) in the SARS-CoV-2 virus. They are polyproteins pp1a and pp1ab. Nsp 1 to 11 is categorized under pp1 polyprotein, while Nsp 12 to 16 is classified under polyprotein pp1ab [].These structural details are of utmost essential when the proteins interact with the drug candidates, to understand the precise molecular mechanism of action of the small molecule.Mutations in the spike received early attention and some of the residues in spike variations influenced the polyclonal antibody recognition, thereby causing immune escape. Studies have shown that among all the receptor-binding domains (RBD) residues for which mutations in the form of substitutions affected the recognition by certain convalescent serums, E484 was identified by deep mutation scanning as having importance, with changes to amino acids K, P, or Q []. Moreover, the mutation E484K is recognized as an escape mutation, that occurs when exposure to monoclonal antibodies C144 and C121 occurs []. Moreover, E484K was the only variable that was found to reduce the neutralization capacity of MAbs REGN10934 and REGN10989 []. It was also studied that mutations K444E, L452R, G446V, and F490S also escaped some of the convalescent sera, thereby making these variants of SARS-Cov-2 stronger and more robust for survival in the human body []. Similarly, the substitution N439K enhances the SARS-CoV-2 ACE2 receptor affinity []. This mutation is also identified to decrease the potential of neutralization of the plasma. However, other deep mutation scanning studies pose contradictory results, stating that N439K did not affect the impact of the polyclonal antibodies used []. This discrepancy could be because immune escape caused by this mutation is via ACE2 affinity than by direct antibody epitope identification, making it a variant that should compel more studies.Studies have also identified that the n-terminal domain (NTD) mutations in spike also cause immune escape []. Studies have shown that mutations K150R, K150T, K150E, S151P, K150Q, and N148S were escape mutations []. Moreover, deletions in the NTD have been noted to be the cause of SARS-CoV-2 evolution repeatedly, by altering the antigenicity of the NTDs []. Another study noted that Δ140 mutant in the spike eventually acquired the E484K variation, thus directly resulting in escaping the antibody responses [].Our previous studies show that mutations in the spike glycoprotein demonstrated enhanced stability in the protein, and the residues that were undergoing mutations were also part of the hydroxychloroquine drug-binding active pockets []. The mutations that occurred in other prioritized proteins of interest evidenced a decrease in the stability of the protein itself. These mutations were also part of the drug-binding active sites of remdesivir, favipiravir, and dexamethasone. The study also concluded that the presence of mutations in various regions reduced the effect of remdesivir on the targets, thereby allowing the potential drugs to not work on the targets of interest []. This shows the presence and effect the variants have on the drug-binding sites.All these studies prove that there is distinct evidence as to the altering of the SARS-CoV-2 proteins’ structural antigenicity, causing them to evade the immune system and become more robust. As of now, evidence suggests the changes that mutations cause in the structure of spike proteins, thereby influencing the binding of small molecules. More research is required to assess the precise mechanism of how various structural and non-structural proteins are being affected by the variants. Amino acid deletions and substitutions in the spike affect the antibodies which may also lead to an increase in the global-viral resistance. A greater understanding of the viral genome sequence data and the impact of variants on the drug-binding sites will be essential to better obtain repurposed drugs.Currently, as we have seen from various evidences, there are several repurposed drugs reported for use against COVID-19, some already available on the market. The potential binding of several FDA-approved drugs has been studied previously; however, not much data is revealed on the precise molecular basis of these interactions against SARS-CoV-2 targets. In the previous section, evidence of how the variants alter the drug binding efficacy and how the virus escapes the immune system is provided. In our previous study, however, various targets of SARS-CoV-2 were analyzed at the molecular level to comprehend the changes it undergoes during the binding event of some of the important FDA-approved drugs for COVID-19. Favipiravir, lopinavir, chloroquine, hydroxychloroquine, lopinavir, ritonavir, and remdesivir were studied. These drugs have also been mentioned in other computational biological studies as well []. Though there are limited experimental data to highlight the spike glycoprotein of SARS-CoV-2 act as the target for repurposed drugs, however, several computational biology prediction revelated the scope of utilising spike glycoproteins as the putative molecular target for repurposed drugs. A recent study stated that drugs ritonavir and eriodictoyl bind to the active sites of spike protein of SARS-CoV-2 with potential binding energy and stability []. Both these drugs interacted with the binding energy of −7.6 kcal/mol, indicating stable binding. Likewise, another study showed that repurposed drugs such as pralatrexate, carumonam, aclerasteride, and granotapide interacted with spike protein of SARS-CoV-2 and acted as potential binders []. Additionally, another recent study has reported Remdesivir and ivermectin, potential repurposed drugs interacted with the spike protein of SARS-CoV-2 []. The study stated that ivermectin bound effectively to the spike protein of the virus, and inhibited the entry of SARS-CoV-2 into the host. Remdesivir was also suggested to be involved in stable interaction with the spike target, apart from the other targets. These studies demonstrate that repurposed drugs were suggested to be targeted the spike protein at various binding pockets and interacted with the spike glycoprotein. To understand further on the interaction modelling of some important repurposed drugs, molecular docking and molecular dynamic studies were performed in our previous study and the results showed ritonavir and lopinavir with better binding energy, followed by remdesivir against the prioritized targets [].At a molecular level, Fig. \\xa0showed that chloroquine interacted with amino acids Leu6820, Leu6819, Met6818, Val7021, Tyr7020, Asp7018, and Asp6942 of the spike glycoprotein. Leu331, Tyr333, Arg277, Phe274, Val270, Trp301, Thr296, Ala264, Arg262, Phe314, Gln260, and Arg259 were the major amino acids involved when spike glycoprotein interacted with Lopinavir []. Likewise, Tyr421, Asn422, Pro491, Leu492, Gln493, Ser494, Tyr495, Gly496, Phe497, Gln498, Lys444, Asp442, Tyr351, Val350, Ser349, Ala348, Phe347, Arg346, Arg509, and Val401 were involved in complex favipiravir-nucleocapsid phosphoprotein. Lys6822, Glu6821, Leu6820, Val7021, Tyr7020, Asp6942, Asn6941, Glu6940, and Thr6938 were observed for complex remdesivir-nucleocapsid phosphoprotein. NSP16 interacted with hydroxychloroquine at residues Arg6817, Met6818, Leu6819, Leu6820, Tyr7020, Val7021, Asp7018, and Asp6942 and with ritonavir at residues Phe314, Gln260, Arg259, Arg277, Phe274, Tyr333, Trp330, Trp301, Thr282, Gly284, Phe286, Leu291, and Val270 [] (Fig. ).Bearing in mind the changes that the drug binding sites undergo if variations occur as already evidenced in the previous sections, it may also be hinted that the residues at the interaction sites also change, rendering a different molecular basis in case of variants. More comprehensive studies are essential to further confirm this.The major repurposed drugs displayed in Table \\xa0dhow that research is ongoing and always improving for tackling the problem of COVID-19. Drugs such as lopinavir, ritonavir, remdesivir, ribavirin, and favipiravir are some repurposed drugs. Lopinavir/ritonavir was generally used against HIV [], remdesivir against major SARS viruses [], and favipiravir which was previously used as a general viral pro-drug []. With the advent of research techniques to identify more repurposed drugs, several other candidates have come up, evidencing the successful implementation of repositioning.The potential of in silico studies has been documented for ages, especially in the field of drug discovery. In this case, repurposing beings with computational bioinformatics work, with much exploration into chem-informatics, chemo-genomics, and computer science aspects such as machine learning, deep learning, artificial intelligence, computational biology, structural biology, and genomics/proteomics. A thorough analysis of the chemical structures of drug candidates helps in the identification of its working mechanism against specific targets. Using computational biology and computer science algorithms will help in predicting the possible and potent drug candidates in silico, a work that might otherwise take years without computational studies. The best drug candidates working against the protein targets of interest causing a specific disease will then be tested experimentally in vitro to validate the predictions. Accurate laboratory confirmation narrows down the best working drug that could be taken forward for in vivo animal model studies, before clinical trials. For example, remdesivir a broad-spectrum antiviral came into the market to be repurposed as a COVID-19 drug after in vitro studies revealed that this drug inhibited the SARS-CoV and MERS-CoV in the human respiratory epithelial cells [] initially, followed by inhibiting the epithelial cells in SARS-CoV-2 [], the pathogenic strain of COVID-19 disease. Likewise, the in vivo study of remdesivir when infected with SARS-CoV-2 infected rhesus monkey demonstrated an improvement in the pulmonary lesions and decreased the viral load after 7 days of treatment with the drug. No coronaviral disease was observed after the successful use of this drug []. Thus, repurposed drugs follow the path of in silico to in vitro to the in vivo platform before they are successfully used.There are several FDA-approved drugs/drugs in the clinical phase, their original targets, and the repurposed targets (Table ). Remdesivir was originally meant for treating the Ebola virus against the Ebola RdRp target, and now with potential repurposing capabilities against the RdRp and NSP12 of SARS-Cov-2 [, ]. Lopinavir/ritonavir is a drug for the HIV-1 protease, that now evidences potential for the main protease of SARS-CoV-2 []. Silmitasertib has its original target as casein kinase 2, which inhibits the casein kinase 2 pathway in Sonic hedgehog (SHH). This drug is also known to inhibit the casein kinase 2 pathway in SARS-CoV-2 []. Likewise, dipyridamole 100 mg is an inhibitor of phosphodiesterase originally and works against the main protease of SARS-CoV-2 and stops replication []. Leronlimab 700 mg works against the C–C Chemokine receptor type 5 in HIV and now shows potential against the same target in SARS-CoV-2 [, ]. Ruconest targets the c1 esterase and inhibits angioedema, while it targets C1, C5a, and C3a esterases in SARS-CoV-2 []. Infliximab attacks the tumor necrosis factor in SARS-CoV-2 as well as in its original target in Rheumatoid arthritis, Crohn’s disease, and ankylosing spondylitis []. This data is also available at  [] and the same is elucidated in Table .\\nWith all the important approaches and scope discussed for repurposing drugs, some important advantages can be classified into three main focuses:Translational: drug repurposing directly benefits translational research by fostering major collaborations, essential for scientists to identify important repurposed drugs. Bridging this gap is one major advantage of drug repositioning research.Target: drug repurposing research aids in understanding thoroughly the pathways associated with specific targets and how the same can be exploited to benefit patients who must be treated. The comprehension of the complete metabolic pathways for specific targets opens up avenues for remaining drug discovery research.Disease: drug repurposing has huge merits in discovering drugs for new, emerging diseases when there are no other treatment approaches yet discovered. The emergence of SARS-CoV-2 has illuminated the purpose and benefits of repositioning drugs to provide emergency treatments.Other advantages of repurposed drugs include:Lower cost of productionLesser time for identifying useful, effective, and safe drugs for quick useResearch can be carried out using the existing, easily accessible bioinformatics and chem-informatics data and resources available for specific diseases, as was the case in COVID-19 cases\\nMolecular docking: The emergence and prevalence of COVID-19 disease have paved the way for molecular docking studies. Previously, molecular docking has been used to determine the potential anti-virals for curbing the spread of the disease. Poly-ADP-ribose polymerase-1, CVL218, a potent inhibitor that binds to the N-terminal of the nucleocapsid protein of the SARS-CoV-2 was identified [].Signature matching: Drug-drug comparisons and drug-disease comparisons can be carried out using the transcriptomic signature patterns. This technique has been previously used for repositioning, however, for lymphoid malignancies []. The use of signature matching for COVID-19 remains an unexplored avenue.Pathway mapping: Identification of potential drugs using this technique has been previously carried out to analyze the network of drugs and diseases. Respiratory viral infections have been previously studied for repurposing candidates and a phosphodiesterase inhibitor was identified in the process [].Genome-wide association studies: Studying the small variations (SNPs) of the genome of interest helps in identifying traits associated with specific diseases. Reports have stated the use of GWAS in combination with other computational approaches for target prediction in drug repositioning [,\\xa0].Thus, despite there being several advantages of repurposing, there are some limitations as elucidated in the following sections.With so many repurposed drugs in the market today, and several more upcoming owing to the intensity and acceleration of COVID-19 research, there are more advantages than disadvantages to the use of such drugs. However, some limitations to the drug repositioning, include:Weak intellectual property rights protection for repurposed drugs due to reduced rate of returnThe molecular docking technique used for identifying drugs for repurposing cannot be applied to all existing drugs due to the larger computational operationsSome network-based approaches used for drug repurposing do not always yield promising candidatesThese approaches are data-dependent, due to which the results may be inaccurate due to the lower volume of existing COVID-19 dataLack of strong, solid evidence for the mechanism of action of repurposed drugs on COVID-19 19Experimental evidence is required for validating the predictive outcomesOverall low success rates in new disease settingsExtensive data to be studied before repurposingDosing and safety issues, i.e., approved range of dosage and delivery capabilities are not very strongAccurate, robust toxicological studies of repurposed drugs essential before delivery and testingRare to find unique and novel drug-protein interactions all within the restraints of the approved therapeutic window [].With drug repurposing emerging as one of the most important research strategies for treating the occurrence of sudden, new diseases, there arrives the question of its efficacy when compared to existing viral vaccines or other small molecules, both considered standard treatment options. However, using high-throughput screening techniques and various other predictive approaches for identifying drugs for repositioning will help maintain the genomic integrity, thereby resolving the issue of efficacy. These approaches when amalgamated with advanced computational techniques such as machine learning will pave the way for producing better-repurposed drugs. In the wake of the emergence of highly mutated and resistant SARS-CoV-2 strains, the question of how already repurposed drugs work comes into the picture. But with improvements in technologies that can help repurposed drugs function better against such mutations, the future for drug repurposing looks bright.Another perspective is the use of drug repurposing as a therapy for novel diseases for which specific drugs are not yet available, COVID-19 being one such important example. Repositioning played a major role in this case and is still in the research arena where different drugs are being explored using advanced computational techniques. Other advanced techniques are also being researched that will provide accurate predictive drug repositioning results that are more reliable and can be used for a variety of diseases. Although this review comprehensively lists all the current and latest drugs repurposed for SARS-CoV-2 and the existing computational approaches that in turn improve the process of repurposing, further advancements in the software and techniques such as using ensemble machine learning will improve the predictive process of repositioning.A dimension that still requires improvement is the patent protection issues that still do not provide full protection to the repurposed drugs. If these avenues are improved, then the future of drug repurposing appears bright.With high mortality rates, COVID-19 has become one of the most prevailing diseases globally. With the CDC and WHO declaring several recent variants of concern for SARS-CoV-2, there is an urgent need to explore alternative options of treatment when there are no specific drugs targeting the disease. One such alternative is drug repurposing, where the use of existing, FDA-approved drugs with similar functions and mechanisms of action against the same family of viruses can be employed. This concept relies heavily on robust preliminary in silico work using advanced and latest techniques in computational biology and chem-bioinformatics. The use of the machine and deep learning algorithms are also a way to obtain a better prediction of repurposed drugs. This review sheds light on the various existing drugs that have been repurposed for use against SARS-CoV-2 and also illuminates the important drug targets against which these drugs are used: spike glycoproteins, RDRP, main protease, papain-like protease, nucleoproteins, non-structural proteins, and enveloped proteins of SARS-CoV-2. The most important advantage of repositioning drugs against COVID-19 is the reduced cost and time of development and the drawback is the absence of safety/toxicity profiles via in vivo trials for the disease. Despite this, repurposing has found its niche with advancements in all related interdisciplinary fields of science and acts as a major research avenue in the current pandemic.\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 40
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles_2500)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "v8vIeo0CF6Ni",
        "outputId": "9d5fb4a9-bde7-43a3-b223-12608176395d"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 42
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "rUg5oh5PFprn"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-06_stripped_markup_articles_2500.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles_2500, f)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Repeat for second batch"
      ],
      "metadata": {
        "id": "sNE43Zj5qAG5"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_2500_5000 = list(map(strip_markup, soup_results_removed_tags_2500_5000))"
      ],
      "metadata": {
        "id": "doy_JJm5-mB3"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles_2500_5000)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sNNjKdqW_CPF",
        "outputId": "62d02b8b-1653-457c-a0e7-0952328043b1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check an article from second batch."
      ],
      "metadata": {
        "id": "pQF23JULqHNP"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_2500_5000[10]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 174
        },
        "id": "y4q4dXcweFZ_",
        "outputId": "e756a101-899d-489a-acbd-37f472597d2c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Corona Virus Disease 2019 (COVID-19) is an infectious disease caused by severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2). It was first identified in December 2019 in Wuhan, Hubei, China, and has since spread around the world. The WHO has declared that the COVID-19 outbreak constitutes a Public Health Emergency of International Concern (PHEIC). This disease can be clinically classified as mild, severe, or critical. Fever, dry cough, and fatigue are the main manifestations, and patients classified as severe can rapidly progress to acute respiratory distress syndrome (ARDS) and multiple organ failure (MOF), amongst other conditions. Unfortunately, at present, there is no cure officially approved for this disease, creating a formidable challenge in its treatment, prognosis, and control. Traditional Chinese medicines (TCM), that are characterized as being anti-viral and affecting multiple pathways and targets, have been proven to be significantly effective in treating COVID-19.The Xuebijing (XBJ) injection is a traditional Chinese medicine that is based on the XueFuZhuYu Decoction (XFZYD). Its main herbal components are Carthami Flos, Paeoniae Radix Rubra, Chuanxiong Rhizoma, Salviae Miltiorrhizae Radix et Rhizoma, and Angelicae Sinensis Radix, and it can have several effects, such as anti-inflammation, anti-oxidation, immune regulation, improving blood stasis, anti-endotoxin, and anti-shock. Previously, it has been used to treat patients with diseases like severe pneumonia and sepsis. In 2019, the National Drug Catalog for Basic Medical Insurance, Work-related Injury Insurance and Maternity Insurance stated that XBJ injection should be used in the first aid treatment of critically ill patients. Furthermore, the Diagnosis and Treatment Protocol for COVID-19 (Trial Versions 4, 5, 6, 7, 8) states that it should be used to treat severe and critically ill patients with COVID-19. However, its mechanisms in the treatment of COVID-19 have not yet been elucidated.Highly pathogenic viruses should be handled under strict laboratory conditions and in compliance with specific biosafety procedures, this study has attempted to investigate the possible molecular mechanisms of XBJ injection when treating COVID-19 via computer simulation based on network pharmacology and molecular docking analyses; thereby avoiding potential laboratory biosafety hazards and improving the efficiency of research into infectious disease outbreaks and modern Chinese medicines. Recently, network pharmacology-based drug repositioning has become increasingly important for the research and development of drugs. Network pharmacology is an application tool based on systems biology theory and network analysis statistics, that conducts network analysis for specific biological systems, and designs multi-target drug molecules by selecting specific signal nodes and comparing signal pathways. Zhou et al. [] identified 16 candidate drugs and 3 potential drug combinations for the treatment of human coronaviruses (HCoVs) through network predictions using over 2000 FDA-approved drugs and systems pharmacology and network pharmacology analyses. These results provide guidance for drug repositioning in the prevention and control of COVID-19. Molecular docking is an important computer-aided drug design method used to study receptor-ligand interactions such as the conformational space, binding energy, and chemical environment via stoichiometry. It has been widely used to elucidate and predict the therapeutic effects, as well as the underlying mechanisms of drugs. In recent years, molecular docking has developed into a common and preferred technique when reseraching Chinese medicines and has exhibited proven efficiency when screening for active ingredients [,].This study has improved upon the network pharmacology and molecular docking anlaysis methods, and determined how to identify the core target in the gene network using the maximum Degree value, which indicates the role of the nodes in network pharmacology. The cluster Profiler package in R language was then used to screen the latest Gene Ontology and KEGG pathways for the core target. The network pharmacology results were verified using protein-small molecule docking to simulate the binding between active ingredients and the core target. These analyses were used to predict the active ingredients, targets, and signaling pathways, provide a molecular basis to investigate the mechanisms by which the COVID-19 inflammatory stress response is inhibited by the XBJ injection, as well as enhance the available theoretical support for the wider application of this drug in the treatment of COVID-19.The main active ingredients were screened from the 5 main herbal components of the XBJ injection, including Carthami Flos, Paeoniae Radix Rubra, Chuanxiong Rhizoma, Salviae Miltiorrhizae Radix et Rhizoma and Angelicae Sinensis Radix using the Traditional Chinese Medicine Systems Pharmacology Database (TCMSP [],) according to oral bioavailability (OB) ≥ 30% and drug likeness (DL) ≥ 0.18. The drug targets of the main active ingredients were then screened through the Targets information of TCMSP. Finally, gene symbol conversion was performed for the targets screened from the TCMSP according to UniProt [] (). Additional information was added according to the literature, and 206 targets were obtained.Cytoscape 3.7.2 [] was used to import the main active ingredients and targets of the XBJ injection, and the active ingredients were numbered to build the ingredient-target network.In Genecards () [], coronavirus disease 2019, coronavirus pneumonia, coronavirus, and novel coronavirus 2019 were used as the keywords for retrieval respectively, and the results were exported in Excel. Then 277 potential targets for COVID-19 were screened for score ≥1 and the information was added according to the literature.Junction targets for the XBJ injection and COVID-19 were imported into the STRING database () [], and the relevant parameters were set as follows:\\nBasic Settings: ① Network type: full network (the edges indicating both functional and physical protein associations); ② meaning of network edges: evidence (line color indicates the type of interaction evidence); ③ active interaction sources: Textmining, Experiments, Databases, Co-expression, Neighborhood, Gene Fusion, and Co-occurrence; ④ minimum required interaction score: medium confidence (0.400); ⑤ max number of interactors to show: 1st shell: none/query proteins only, 2nd shell: none.Advanced Settings: ① network display mode: interactive svg; ② network display options: hide disconnected nodes in the network.The PPI network for the junction targets for the XBJ injection and COVID-19 was constructed using the STRING database, and the results were exported to a TSV(Tab-separated values) file for further analysis using Cytoscape 3.7.2 (① for analysis, click Tools > Network Analyzer > Network Analysis > Analyze Network > Treat the network as undirected; ② for mapping, click Tools > Network Analyzer > Network Analysis > Generate Style from Statistics) to optimize the PPI network and screen for the core target with the maximum Degree value.2.3.2 GO enrichment analysis and KEGG pathway analysis for the core target for the XBJ injection and COVID-19Gene Ontology (GO) includes three aspects: biological process (BP), cellular component (CC) and molecular function (MF). GO enrichment analysis and Kyoto Encyclopedia of Genes and Genomes (KEGG) pathway analysis were conducted for the core target for the XBJ injection and COVID-19 using R (version 3.6.1) [] and the clusterProfiler [] package, respectively. In addition, Cytoscape 3.7.2 was used to plot the results.According to the network pharmacology results, AKT1 was identified as the core target of the XBJ injection when treating COVID-19, and the active ingredients of the XBJ injection, baicalein, luteolin, and quercetin were all connected with the core target AKT1. Consequently, further molecular docking verification was carried out based on these results.RCSB PDB databases were used to download the 3D structure PDB file of the core target AKT1 (PDB ID: 1UNP), and it was then imported into the Discovery Studio 2019 Client (DS) to remove solvent molecules and ligands. Solvents were removed from the structure, while polar hydrogen atoms were added to the 1UNP by clicking ‘Add Polar’, and a forcefield was applied to the 1UNP by selecting ‘Apply Forcefield’. The PubChem database was used to obtain the SDF files of the active ingredients for the core target AKT1 as small molecules, which were imported into the DS. Apply Forcefield was used to give the force field to the small molecules. Finally, the CDOCKER (Dock Ligands) was used for protein-small molecule docking, by adjusting the Top Hits-Pose Cluster Radius to 0.5, and the other parameters were their default values. CDOKER [] is a high-accuracy molecular docking method based on the CHARMm force field. With this method, high-temperature kinetics are used to search for the flexible conformation of the ligand molecules. The lower the CDOCKER interaction energy is, the more stable the conformation of the ligand binding to the receptor is thought to be. Furthermore, ≤−5.0 kcal·mol−1 indicates that the protein and small molecule can bind, and ≤−7.0 kcal·mol−1 indicates a strong binding ability.This study aimed to investigate the molecular targets and mechanisms of the Xuebijing injection when used to treat COVID-19 using network pharmacology and molecular docking analyses. According to the ‘Drug-Ingredients-Targets-Disease’ network, AKT1 was identified as the core target, and baicalein, luteolin, and quercetin were identified as the active ingredients of the Xuebijing injection that were associated with AKT1. The results of the GO enrichment analysis showed that response to lipopolysaccharide was the most significant biological process in which AKT1 was involved, the cellular component was spindle, and its molecular function involoved nitric-oxide synthase regulator activity. The results of the KEGG pathway analysis showed that AKT1 was involved in the treatment of COVID-19, mainly by regulating TNF and other signaling pathways. Based on the results of the network pharmacology, molecular docking analysis was performed with AKT1 and the three active ingredients, and the results indicated that all three active ingredients could bind stably with AKT1.The 5 herbal components were retrieved using TCMSP. According to the screening requirements of an OB ≥ 30% and DL ≥ 0.18, we identified 44 active ingredients, including 9 from Carthami Flos, 8 from Paeoniae Radix Rubra, 6 from Chuanxiong Rhizoma, 19 from Salviae Miltiorrhizae Radix et Rhizoma, and 2 from Angelicae Sinensis Radix. MOL000449-Stigmasterol and MOL000358-beta-sitosterol were the common active ingredients mapping to Carthami Flos, Paeoniae Radix Rubra, and Angelicae Sinensis Radix. MOL002714- baicalein was the common active ingredient mapping by Carthami Flos and Paeoniae Radix Rubra, while MOL000006-luteolin was the common active ingredient mapping to Carthami Flos and Salviae Miltiorrhizae Radix et Rhizoma. Following the removal of the duplicates, a total of 38 non-repeated active ingredients were identified. In addition, 206 non-repeated drug targets were obtained from the TCMSP, according to the active ingredients mentioned above.The ingredient-target network for the XBJ injection was constructed using Cytoscape 3.7.2 and included 38 non-repeated active ingredients and 206 non-repeated potential targets (see  and ). The core targets in the ingredient-target network, including PTGS1, PTGS2, NCOA2, KCNH2, SCN5A, ADRB2, RXRA, PGR, NR3C2, NCOA1, and PDE3A, were screened according to Degree ≥ 5.\\n\\nGeneCards was used to identify 277 targets of COVID-19, and STRING and Cytoscape were used to screen out the free targets. These targets, such as TNF, IL6, GAPDH, TP53, and IL10, were then arranged in order of degree (see ).\\n\\nThe 46 junction targets of the Xuebijing injection and COVID-19 were imported into the STRING database to construct a PPI network (average node degree was 13.1, PPI enrichment p-value < 1.0\\xa0×\\xa010−16, avg. local clustering coefficient was 0.675). The PPI network was then optimized using the Cytoscape software (see ), and the core target AKT1 was identified according to Degree ≥ 34. Secondary targets were also screened according to 24 ≤\\xa0Degree < 34, and included TP53, TNF, JUN, EGFR, IL1B, IL10, and EGF.\\nGO enrichment analysis of the core target AKT1 from the junction targets mapped for the XBJ injection and COVID-19 was conducted using the clusterProfiler package and Cytoscape 3.7.2 was used to plot the results. According to the results of the GO enrichment analysis, the main biological processes in which AKT1 was involved included response to lipopolysaccharide, response to molecule of bacterial origin, regulation of neuron death, neuron death, and cellular response to oxidative stress. The cellular component was spindle. The molecular functions in which AKT1 was found to be involved included nitric-oxide synthase regulator activity, protein kinase C binding, protein phosphatase binding, protein serine/threonine kinase activity, and phosphatase binding. See .\\nThe KEGG pathway analysis of the 46 junction targets mapped to the XBJ injection and COVID-19 was conducted using the clusterProfiler package and Cytoscape 3.7.2 was used to plot the results. The results showed that the KEGG pathways in which AKT1 was involved included the TNF signaling pathway, MAPK signaling pathway, human cytomegalovirus infection, fluid shear stress and atherosclerosis, and hepatitis C (see ).\\nThe active ingredients of the XBJ injection, included baicalein (Paeoniae Radix Rubra), luteolin (Salviae Miltiorrhizae Radix et Rhizoma), and quercetin (Carthami Flos), and they were all connected with AKT1, suggesting that these three active ingredients may be effective for XBJ injection in the prevention and treatment of COVID-19 (see ).\\n3.5 Verification of molecular docking for the XBJ injection when treating COVID-19 by the Results of Network PharmacologyThe activity of the three active ingredients obtained through the network pharmacology analysis were validated using protein-small molecule docking.The results showed stable binding between the three active ingredients (baicalein, luteolin, and quercetin) and AKT1, with binding energies of −25.5854 kcal·mol−1, −31.5575 kcal·mol−1, and −31.53 kcal·mol−1, respectively. The binding models are shown in −6c.\\nAccording to the Diagnosis and Treatment Protocol for COVID-19 (Trial Version 8), critically ill patients are more likely to develop dyspnea and/or hypoxemia within a week, and severe patients, like those with SARS and MERS, may suffer from a cytokine storm (CS), which can rapidly develop into acute respiratory distress syndrome, septic shock, and multiple organ failure. The XBJ injection has been shown in previous studies to be effective at improving end-point outcomes for severe community-acquired pneumonia (CAP) and reduced the 28-day mortality in septic patients [].In this study, 38 non-repeated active ingredients and 206 non-repeated potential targets of the XBJ injection, 277 potential targets of COVID-19, and 46 junction targets were screened using network pharmacology analysis, and the core target from the junction targets was identified to be AKT1. AKT1 is a member of the AKT kinase family that regulates metabolism, proliferation, cell survival, growth, and angiogenesis through a series of downstream substrates. AKT regulates the metabolism of fat, amino acids, and other substances by activating AS160 and PFKFB2, and also induces the TSC1/TSC2 complex and mTORC signal transduction after the stimulation of inflammatory factors, so as to regulate the growth of the endothelium, restore the endothelial barrier function, and promote proliferation of the lung fibroblasts []. A previous study has shown that overexpressed constitutively active AKT1 can promote viral protein synthesis []. Furthermore, activation of the PI3K/AKT pathway is indispensable for coxsackievirus B3 infection. Dominant negative mutant of AKT1 can significantly dampen viral RNA expression and further reduce viral capsid protein expression and viral release []. The replication of another coronavirus, Middle East respiratory syndrome coronavirus, was remarkably inhibited when kinase inhibitors that target PI3K/AKT were adminstered []. Collectively, AKT1 could be an ideal target with a broad-spectrum antiviral effect. These results suggest that the XBJ injection is an effective control and treatment for COVID-19, probably due to its regulation of AKT1.According to the results of the GO enrichment analysis, the response to lipopolysaccharide (LPS) was the most significant biological process that AKT1, the core target of junction targets to the XBJ injection and COVID-19, was involved. LPS is a major component in the cell walls of Gram-negative bacteria. It has been reported that LPS can damage alveolar epithelial cells and capillary endothelium cells, resulting in changes to the intercellular space and permeability [,]. AKT could regulate the NF-κB signal transduction through phosphorylated IKKα and Tpl2 in order to ameliorate lipopolysaccharide-induced alveolar blood stasis and abnormal fibrinolysis in mice []. It has been suggested that XBJ injection may inhibit response to lipopolysaccharide by regulating AKT1 expression, thus alleviating lung injury caused by LPS. The results of this study showed that the most significant molecular function of AKT1 is the nitric-oxide synthase regulator activity. Nitric-oxide synthase (NOS) is an isozyme that is reportedly associated with pneumonia and pulmonary fibrosis []. These results suggest that XBJ injection may regulate NOS activity through AKT1, which may help to prevent and treat COVID-19.According to the results of the KEGG pathway analysis, the TNF signaling pathway is the most significant pathway in which the core target AKT1, that is the core target of junction targets to XBJ injection and COVID-19, was involved. Tumor necrosis factors (TNF) include TNF-α and TNF-β which can promote the expression of proinflammatory factors and participate in systemic inflammatory responses, while it can also produce an anti-infection effect and prevent early viral proteins from being synthesized to inhibit viral replication and kill infected cells. Therefore, the XBJ injection can regulate the TNF signal pathway through AKT1, so that it has anti-inflammation and anti-virual effects, and thus the potential to reduce the inflammation caused by COVID-19.The protein-small molecule docking analysis revealed that all three active ingredients (baicalein, luteolin, and quercetin) had perfect CDOCKER Interaction Energy with AKT1 (1UNP), with the highest found in luteolin. This inferred that luteolin was the most important active ingredient contributing to the therapeutic effects of the XBJ injection on COVID-19. The analysis of the binding model showed that there were conventional hydrogen bonds between the luteolin and Gln104, Asp108, Lys111, and Gln59 of the AKT1 and van der Waals between luteolin and Cys60, Gln61, Cys77, and Leu78 of AKT1. It has been reported that luteolin can mitigate airway inflammation by regulating the body’s antioxidative stress response and the COX-2 signaling pathway. Luteolin is also shown to relieve myocardial ischemic injury by modulating the mitogen-activated protein kinase (MAPK)/NF-κB pathway and suppressing enterovirus type 71 infections by inhibiting cell apoptosis and inflammatory factor secretion, demonstrating its antiviral properties. Additionally, luteolin and quercetin are found to induce cell apoptosis by inhibiting the phosphorylation and activation of AKT []. Despite all these findings, it has not been reported whether these three active ingredients relieve COVID-19 symptoms through the direct regulation of AKT1. In this study, molecular docking was implemented to further explore the mechanisms of action of the three active ingredients in treating COVID-19. Furthermore, considering that baicalein, luteolin, and quercetin are found in many herbal components, these active ingredients may not be unique to the XBJ injection, indicating the possibility of discovering antiviral TCM combinations in other ancient prescriptions containing baicalein, luteolin, and quercetin.In this study, potential targets were screened, a PPI network was constructed, and enrichment analysis was conducted to find potential pathways. The XBJ injection was shown to produce synergistic therapeutic effects on COVID-19 with ‘multi- active ingredients, multi-target and multi-pathway’, and molecular docking was performed to verify such results. However, the present study has some limitations: 1) The study results are based on existing data available in the given database, and differences in the results may arise if different databases were used. Therefore, future studies should focus on multidatabase analysis and databases that are updated regularly. 2) The biological network in this study was constructed using qualitative data, but TCM prescriptions usually have a complex composition, as well as intricate interactions, making it extremely difficult to perform an analysis based on the quantitative data. Consequently, the reliability of the conclusions from this study need to be further verified. Therefore, in vivo and in vitro experiments should be conducted to further examine the molecular mechanisms of the XBJ injection for the treatment of COVID-19.In conclusion, according to the network pharmacology and molecular docking analyses, the predicted active ingredients in the XBJ injection include baicalein, luteolin, and quercetin. The XBJ injection may inhibit the lipopolysaccharide-mediated inflammatory response, modulate NOS activity, and regulate the TNF signaling pathway by affecting the function of the AKT1, thereby ultimately suppressing the excessive inflammatory response associated with COVID-19. Due to the limitations of the computational methods for chemistry and biology, the results need to be verified with follow-up experiments to provide a basis for the treatment of COVID-19 with TCM.'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_stripped_markup_articles_2500_5000.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles_2500_5000, f)"
      ],
      "metadata": {
        "id": "-MvofFiX-wi8"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Repeat for third batch."
      ],
      "metadata": {
        "id": "3ONsegNPqTcY"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_5000_7500 = list(map(strip_markup, soup_results_removed_tags_5000_7500))"
      ],
      "metadata": {
        "id": "UGaIPl2mCTTi"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles_5000_7500)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mjAkj5kDCeSB",
        "outputId": "34cac17c-1d64-4003-c33f-c353b6b4b31f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_stripped_markup_articles_5000_7500.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles_5000_7500, f)"
      ],
      "metadata": {
        "id": "RkzHrBQHCrTd"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Repeat for fourth batch."
      ],
      "metadata": {
        "id": "49wZlPNBqiIb"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_7500_10000 = list(map(strip_markup, soup_results_removed_tags_7500_10000))"
      ],
      "metadata": {
        "id": "cVSMoJTlC-7n"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles_7500_10000)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "06ce7424-e2c8-4b84-f0e5-758f82f155e7",
        "id": "KF_pz6T-C-7n"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "2500"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_stripped_markup_articles_7500_10000.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles_7500_10000, f)"
      ],
      "metadata": {
        "id": "LuA9pvPRC-7n"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Repeat for fifth and final batch."
      ],
      "metadata": {
        "id": "CPP-TXtqqrtn"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles_10000_11397 = list(map(strip_markup, soup_results_removed_tags_10000_11397))"
      ],
      "metadata": {
        "id": "SSGJfdG9D_8n"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles_10000_11397)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "e3867923-5064-48b4-eb95-254b7f76b999",
        "id": "e6JF6bNjD_8o"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "1397"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_stripped_markup_articles_10000_11397.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles_10000_11397, f)"
      ],
      "metadata": {
        "id": "SRc8bVsZD_8o"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.7 Concatenate lists for five batches\n",
        "\n",
        "Using the * operator we can concatenate the five lists into one list containing all 11397 articles."
      ],
      "metadata": {
        "id": "duEJ-7Xhiqsl"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "concat_lists = [*stripped_markup_articles_2500, *stripped_markup_articles_2500_5000,\n",
        "                *stripped_markup_articles_5000_7500, *stripped_markup_articles_7500_10000,\n",
        "                *stripped_markup_articles_10000_11397]"
      ],
      "metadata": {
        "id": "HKhNgOeyI4fD"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Print first article to test."
      ],
      "metadata": {
        "id": "0qJ0UHVYr9f8"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "concat_lists[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 174
        },
        "id": "nz7loguegm9C",
        "outputId": "5e27dd3a-a1c0-4880-f55a-c1ef687be544"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'Sir James Black, a winner of the 1988 Nobel Prize, clearly recognized well before the 21st century that drug repurposing strategies would occupy an important place in the future of new drug discovery (). In 2004, Ted T. Ashburn et al. () summarized previous research and developed a general approach to drug development using drug repurposing, retrospectively looking for new indications for approved drugs and molecules that are waiting for approval for new pathways of action and targets. These molecules are usually safe in clinical trials but do not show sufficient efficacy for the treatment of the disease originally targeted (). The definition of the term “drug repurposing” has been endorsed by scholars () and used by them (; ). It should be pointed out that the synonyms of “drug repurposing” often used by academics also include drug repositioning (), drug rediscovery (), drug redirecting (), drug retasking (), and therapeutic switching (; ). After the research study by Ashburn et al., Allarakhia et al. expanded the starting materials for drug repositioning to include products that were discontinued for commercial reasons, expired patents, and candidates for laboratory testing (). In the discovery process of a completely new drug, the difficulty usually lies in its safety and efficacy, which are the main potential causes of failure of most drugs in the approval () or clinical development stage (). Using existing knowledge about a drug or known target (), the time, risk, and cost of developing a drug using drug repositioning are reduced (), thereby greatly increasing the efficiency and economics of drug development, providing a better risk–reward trade-off, and making it easier to win the favor of venture capital firms ().Since the 1990s, the repositioning of sildenafil for male erectile dysfunction () and pulmonary hypertension (), the development of a new efficacy of bupropion for smoking cessation (), new applications of thalidomide for multiple myeloma (; ), and chronic graft-versus-host disease () have generated intense interest from pharmaceutical companies and academics (). These classic success stories rely on three traditional approaches: 1) molecular biology approaches (), 2) in vivo and ex vivo experimental approaches (; ), and 3) expert knowledge-based approaches (). Due to the unknown, complex, and information-fragmented nature of drug candidates and potential new mechanisms of action (), this activity is dependent on multiple factors, and success is often fortuitous (). At the beginning of the 21st century, cheminformatics (; ), bioinformatics (; ), systems biology (; ), genomics (; ), polypharmacology (; ), precision medicine (; ), and other disciplines, combined with artificial intelligence (), have developed rapidly. These rapidly growing disciplines have promoted the generation of systematic () computer methods to make the drug repositioning process cheaper and shorter (; ). Computational drug repositioning is classified as “disease-centric” or “target/gene-centric” or “drug-centric” depending on the source of discovery (). This process relies on public biochemical databases such as DrugBank (; ), ChEMBL (), Cmap (), PDB (), OMIM (), etc., to provide the appropriate information. In fact, to make the computational drug repurposing process, including the molecular docking and virtual screening steps, more convenient, database tools specifically developed for drug repurposing, such as EK-DRD (), DREIMT (), DrugSig (), RepoDB (), Promiscuous 2.0 (), etc., have been reported in the last few years. In addition, it has been found in the literature that only 10% of the research results have been carried out in the “drug-centric” pathway, which holds great prospects for future development (). With the help of database tools, it is now possible to perform computational screening of even a staggering number of hundreds of millions of compounds (). Computer methods to carry out this screening include machine learning (), network modeling (; ), text mining, and semantic reasoning (; ; ), among others. The ultimate objective of repositioning is to transfer one or two of the most relevant results to clinical applications. Therefore, validation is quite important () and requires consideration of multiple factors, such as price, toxicity levels, bioavailability, and differences between validated and computational models (; ). Current validation methods include experimental validation (), electronic health records to aid validation (), cross-validation (; ), gold standard dataset evaluation (), literature citation validation (), and expert consultation ().Today, drug repositioning is increasingly prominent in the development of drugs for a variety of neurological diseases (; ), cancer (; ), rare diseases (; ), and infectious diseases (; ). An increasing number of pharmaceutical companies are also establishing relevant R&D programs () or funding support (). To translate relevant research results efficiently and smoothly, national departments within the United Kingdom, the United States, and the Netherlands have (; ) launched initiatives or programs to build partnerships between pharmaceutical companies and academia and to further explore scientific and commercial opportunities (). It is certain that drug repositioning currently presents several dilemmas, such as intellectual property challenges (), data platforms, and analytical techniques that need to be improved (), that financial support remains important for technology development and clinical trials (), and that some scientists deny the practical utility of the approach ().There have been systematic analyses of terminology in the drug repurposing literature (), text mining of drug–disease combinations (), and the progression of a particular drug (), but no studies have yet provided a broad overview of publications on the topic of drug repurposing research. When independent researchers or collectives (including pharmaceutical companies, academia, and government departments) seek drug repurposing partnership partners and seek to obtain a concise overview of comprehensive current research hotspots, the lack of relevant intelligence analysis to aid decision-making often makes the process convoluted and time-consuming (). The bibliometric approach can solve the aforementioned problems relatively fairly, but at present, scholars have only studied the bibliometrics of aspirin, a drug repurposing (); there has not been a panoramic study of drug repurposing, and therefore, this study is necessary. Bibliometrics is a useful tool combining multiple parameters for the quantitative analysis of scholarly publications and is currently used to assess research hotspots and trends in a wide range of disciplines and industries, such as management (; ), sociology (; ), economics (), medicine (; ), environmental engineering (; ), and agronomy (; ). Therefore, this study uses bibliometric methods () to quantitatively assess the following elements of drug repositioning publications: 1) major contributors: countries, research institutions, and authors; 2) modes of collaboration: intercountry collaborations; 3) the most productive journals; 4) the most frequently used disciplinary knowledge; and 5) research trends, judged by analyzing author keywords, Essential Science Indicators (ESI) high citations, and hot research studies.We use the Web of Science™ core database, an authoritative academic information data service platform produced by Clarivate (version © 2021 Clarivate.). Due to its rigorous selection of journals, the Web of Science (WOS) Core Collection Database is now internationally recognized as a database for evaluating the scientific output or disciplinary development of scholars and institutions. Among the subdatabases, SCI-E mainly includes global journals in basic science research, covering basic pharmacological and medical research related to the theme of this study, “drug repositioning,” while SSCI includes social science, covering ethical, nursing, psychological, and other social science research related to this study.The data were obtained on 25 October 2021 through the WOS Core Collection Database Citation Indexes SCI-E and SSCI, using the formula “drug repurposing” OR “drug repositioning” OR “drug rediscovery” OR “therapeutic switching” OR “drug redirecting” OR “drug rediscovery” OR “drug retasking” search query, searching in the “subject” field and defining the document type as “Article” and “Review”. The publication time parameters were initially limited to publications related to “drug repositioning” published between 1990 and 2020. A total of 3,009 documents were obtained, of which only 31 were published in two\\xa0decades from 1990 to 2009. Of these 31 documents, except for one document that is still frequently used by scholars as a retrospective source for drug repurposing definitions in these years (), the remaining 30 were cited by other authors during the period of 2010–2020 as shown in . The overall level of interest in these studies shows a fluctuating downward trend as opposed to the rising citation fervor for drug repurposing, entering a stage of decline even under the less-demanding evaluation criterion of a 5-year maturation window (). As the literature ages, its content becomes stale and obsolete in the perspective of intelligence sources, and the value of the metrics for judging current research trends is low. Therefore, we further narrowed the study to 2,978 publications published from 2010 to 2020.The complete records of all retrieved documents are downloaded and imported for processing into Derwent Data Analyzer (DDA) version V10, a data cleaning, multiperspective data mining, and visualization software from Clarivate that improves data analysis efficiency and reduces labor costs. After importing all records of WOS documents into DDA, they are classified and measured according to a list of fields such as keyword, country/region, institution, author, research field, journal, etc. For each item in the list fieldset, DDA has a built-in data cleaning tool for automatic data deduplication.After the machine has removed duplicates, the items in the set of fields still need to be manually verified for splitting or merging. It is to be noted that the regions of certain countries are presented separately, while they are usually considered as a single country internationally. Therefore, we need to perform merging, such as combining Wales, Scotland, England, and Northern Ireland into the United Kingdom column and combining Hong Kong and Macau regions into the China column.To address the possible problem of different authors with the same name, the following two main verification steps were performed: 1) returning to the WOS database to search for publications under that author’s name under the original search formula conditions and 2) for authors whose publications provide disputed information (this also includes three cases: first, two or more authors with the same name but not the same person; second, two or more authors with the same abbreviated name, but the full names were found to be different after a search; and third, similar signatures being different variants of the same author’s name), in addition to searching the ORCID-related information of the authors concerned for judgment, we checked different institutional websites as well as encyclopedias to look for changes in the study and work history of authors with the same or similar names from 2010 to 2020 to determine whether they were the same person. Based on the verification, we then split or merged the results.After data cleaning and matrix analysis by DDA, various types of cluster plots and bubble plots can be obtained to reveal the useful information behind the data. The bibliometric fields of publication volumes, countries, international collaborations, institutions, research areas, journals, authors, highly cited research studies, and author keywords were analyzed in this study. It should be noted that because some studies were published online ahead of time and the study publication date was a year or two behind, for statistical purposes, the year of publication of such research studies was included as the year of online publication. (e.g., a study shown in the reference as published in 2022 may have been published online in 2020).Of the 2,978 papers obtained using the search criteria mentioned previously, the main ones were research studies (2248; 75.49%) and reviews (730; 24.51%). Furthermore, individual publications are not only classified by journals in the single category of research studies or reviews but also belong to other categories. These publications were also related to proceeding studies (68; 2.28%), early access (24; 0.81%), book chapters (7; 0.24%), data studies (2; 0.07%), and retracted publications (2; 0.07%). The vast majority of research studies and reviews were published in English (2967; 99.631%), with the remainder in Japanese (3; 0.101%), Chinese (2; 0.067%), Czech (1; 0.034%), French (1; 0.034%), German (1; 0.034%), Hungarian (1; 0.034%), Korean (1; 0.034%), and Portuguese (1; 0.034%). Ninety were from SSCI, and the remaining 2888 were from SCI-E. Further, 1,996 were from Open Access. An annual analysis of published research studies is shown in . The number of publications for every year expanded from 17 in 2010 to 970 in 2020. Annual publications on the subject have increased by more than 64 times. The number of annual publications has been increasing at a relatively high rate since 2015, while in 2020, there was a spike in the number of publications and annual citations, probably due to the COVID-19 pandemic, a global public health emergency that prompted special attention from scientists. Among the four countries with the highest number of publications (the United States, China, the United Kingdom, and India), the United States has maintained a high growth volume since 2010, while China was the fastest in terms of average annual growth rate in the last three\\xa0years. In 2020, the number of publications in India surged and surpassed the production of the United Kingdom.With respect to the 2978 publications related to drug repositioning research, 89 countries contributed to the field of drug repositioning research. The number of publications and citations from the 20 most productive countries/regions is shown in . There are nine countries/regions in Europe, five in the Americas, five in Asia, and one in Oceania.The four most productive countries/regions are, in order, the United States, China, the United Kingdom, and India. The United States is the absolute leader in this field, with 918 research studies on drug repositioning published since 2010, which is already more than the next highest number of publications in China and the United Kingdom combined. This is followed by India (247), Italy (232), Germany (171), South Korea (161), and Japan (146). Other productive countries include Brazil (125), France (116), Canada (111), Spain (109), Australia (73), the Netherlands (73), and Switzerland (59). In terms of publication impact, the United States led the Total citations (TC) rankings with 27,355, twice as many as that of China (11,147), which ranked second. We also included the average citations per publication (ACPP) in the comparison, which is calculated by dividing the TC by the TP (total papers) value and is a relative number that may better reflect the individual or collective level of attention than the individual TC and TP values. Canada ranked first in ACPP at 41.81, closely followed by the United Kingdom (30.85) and Germany (30.49). In addition, the h-index was originally proposed as a simple quantification that a researcher had at least h publications cited h times, reflecting to a certain extent the research results of the researcher as an individual (). Later, the word “researcher” in the definition began to be replaced by collective words such as “academic group or institution (),\" “journal (),\" and “country (),\" becoming an indicator of the level of collective research to some extent. Undoubtedly, the h-index of the United States ranks first in this field with 74 times. Taking all parameters into account, we find that publications in the United Kingdom, the United States, and Canada perform better on average. While the number of publications in China and India is significant, they have received low levels of attention.It should be noted that DDA analysis software is nationally identified based on the location of each researcher’s institution address provided in the publication. If a publication is coauthored by institutions from more than two countries, the publication is defined as the result of an international collaboration. Whether there is some affiliation between the various institutions of the research group that produces the multicountry collaboration is not taken into account. As shown in , among the publications of the top 20 countries and regions, the proportion of international collaborations is quite high in European countries, especially in Sweden (86.21%) and Belgium (81.25%). Asian and Latin American countries are generally underrepresented. In addition, the United States, the most active country in publishing and the country with the most collaborations—with 59 countries or regions—still has over 50% of the studies published overall.\\n depicts the academic collaboration network for the top 20 countries and regions in terms of productivity. Using DDA software, the network was mapped using a co-occurrence matrix. The size of the circles is proportional to the extent of each country’s contribution, the lines between the circles represent the collaboration between countries/regions, and the thickness of the connecting lines indicates the frequency of collaboration (). The results show that the United States cooperates most frequently with China and the United Kingdom and has the closest cooperation with them. In addition, Mexico, Belgium, Argentina, Taiwan, Japan, and Korea have slightly sparser cooperation networks among the 20 most productive countries/regions, while the remaining countries have more extensive cooperation networks among themselves.A total of 3,530 institutes were involved in drug repositioning research. The top 20 productive institutes are shown in . Eight of the top 20 institutions are from the United States, again indicating the dominance of the United States in drug repositioning research; three are from the United Kingdom; two are from China; and Brazil, France, Mexico, Canada, the Netherlands, Austria, and Sweden each have one. The Chinese Academy of Science ranked first in terms of the number of research studies, followed by Case Western Reserve University and the NIH. The Icahn School of Medicine at Mt Sinai ranked first in ACPP at 77.32. The NIH had the highest h-index value of 22. The Icahn School of Medicine at Mt Sinai was the best performer in ACPP at 77.32, followed by the University of California, San Francisco (67.82) and Johns Hopkins University (65.68), both of which are US-based research institutions. Compared with US research institutions, Chinese research institutions are at the back of the pack in terms of ACPP, and their impact needs to be improved.The collaboration network between the 15 largest institutions in 2010–2020 is shown in . The collaboration network provides a more visual view of the collaboration with different institutions and thus helps in the search for more beneficial collaborations. Next to the name of each institution is its total number of publications. At the intersections of these institutions, yellow dots indicate collaborations with the other top 10 research institutions. It should be noted that the number of yellow dots can indicate the output of cooperation and the strength of interagency cooperation. The nodal data with no crossover points represent the number of publications produced by the institute, either by its independent work or in collaboration with research institutions outside the top 15 (). From , we see that the University of Cambridge established the largest collaborative network, followed by the large network established by four institutions, the NIH, the Icahn School of Medicine at Mt Sinai, Karolinska Institute, and King’s College London. In terms of the number of copublications with established institutions, the Chinese Academy of Science and Shanghai Jiao Tong University copublished as many as six, followed by the University of Cambridge and King’s College London and the NIH and the Icahn School of Medicine at Mt Sinai. Analyzing the aforementioned three pairs of institutional combinations, King’s College London has two publications that are the product of collaboration between the three research institutions. The University of Sao Paulo and Aix-Marseille University are relatively independent in this research area. Combining the ranking of multiple parameters, we found that the NIH and Icahn Sch Med Mt Sinai in the United States are the most vocal institutions in terms of academic research result perspective on the topic.An analysis of research areas gives a good indication of the scope of application of the research topic, with an unrestrained number of 74 areas covered, with the top 20 areas of research in terms of publication volume shown in . Briefly, “pharmacology and pharmacy” took first place with 962 articles, followed by “biochemistry and molecular biology”, and for ACPP, the top three were science and technology-other topics (36.1)\", “mathematics (32.79)\", and “cell biology (29.65)\".\\n shows a bubble graph of the top 20 drug repositioning research areas. The bubble plot shows three dimensions of the data, namely, research area, year of publication, and the number of publications. The horizontal change in bubble size illustrates the growing trend of research areas over time, the vertical size of the bubble shows the most popular research areas in that year, and the number in the bubble indicates the frequency of the topic in the research area and the number of publications in that year (). The number of research results in each relevant field is increasing year by year. Biophysics increased from five in 2019 to 77 in 2020, a more than 15-fold increase, suggesting that drug repositioning may have made a breakthrough or become widely used in this field. The field of virology was in a downturn from 2010 to 2014, with only one publication, with a gradual increase in relevant studies after 2015.For scholars studying drug repositioning-related topics, knowing which journals publish relevant research is important in deciding which journals to read or submit their research studies to. A total of 2,988 publications related to drug repositioning research were published in 845 journals during the period of 2010–2020. The top 25 journals in terms of a total number of studies published are shown in  Sci Rep topped the list with 75 studies published, followed by PLoS One (73; 2.52%), J. Biomol. Struct. Dyn (67; 2.45%), Bioinformatics (53; 2.25%), and BMC bioinformatics (50 articles; 1.78%). The rest of the journals had a share of less than 1.5%. In terms of total citations (TC), at present, studies in Drug Discov. have been cited a total of 2,119 times over the past 10\\xa0years, followed in rank by PLoS One (1800) and Bioinformatics (1,677). For ACPP, Drug Discov. Today still holds first place with a high frequency of 50.45 times, followed by PLoS Comput (33.14 times). The impact factor (IF) of a journal is calculated by dividing the total number of citations of all publications in the journal in the previous two\\xa0years by the number of publications (). Thus,  shows that the ACPP of drug repurposing publications included in most journals is much higher than that of IF, which roughly verifies that the number of scholars interested in drug repurposing is relatively high. In terms of the impact factor (IF) of specific journals, except for Oncotarget and BMC Syst. Biol., which have not been included in SCI since 2018 and 2020, Brief. Bioinform. has the highest value of 11.622, followed by Drug Discov. Today (7.851), Bioinformatics (6.937), Cancers (6.639), Eur. J. Med. Chem (6.514), and Expert. Opin. Drug Discov. (6.098). The bubble chart shows that J. Biomol. Struct. Dyn. featured 64 publications in 2020, compared to a combined total of only four publications in the previous ten years; the Oncotarget journal inclusion in this category peaked in 2016–2017 ().For scholars interested in the topic of drug repositioning, it is useful to know how other researchers are working on the issue to facilitate communication and collaboration between scholars. A total of 15,620 authors contributed to studies within our measurement consideration, and  shows the top 20 prolific authors by a number of publications. Of these 20 highly productive authors, seven were from the United States, three were from Argentina, and two were from Germany, indicating a relatively high concentration of drug repositioning research in certain countries. In addition, the NIH (United States), Case Western Reserve University (United States), Tech University Dresden (Germany), and the National University of La Plata (Argentina) each have two of these academics.Cheng, FX leads the list with 25 research studies, followed by Talevi, A (23) and Mucke, HAM (22). For the list of corresponding authors, the top three remain, in order, Mucke, HAM (22), Cheng, FX (17), and Talevi, A (17). In terms of ACPP ranking, Butte, AJ was ranked first with 154.33 points, followed by Cheng, FX (100.56), Tang, Y (95), and Dudley, JT (82). Cheng, FX still has the highest h-index at 21, followed by Zheng, W (17), Talevi, A (12), Xu, R (11), and Schroeder, M (11). The h-index has two drawbacks when researchers of the same topic are compared with each other (). One is that the scholar’s h-index does not decrease over time but only grows or stays the same, and it is not possible to obtain information on whether the scholar is still in an academic career. In this study, we narrow the study to the most recent publications from 2010 to 2020, taking into account the timeliness of the h-index response information. Second, older scholars usually enter academia earlier and have an advantage in their h-indexes in comparison with those of younger scholars. Therefore, this phenomenon must be targeted for analysis or illustration. Thus, by combining the authors’ educational experiences and employment relationship changes that were recorded in the WOS database and ORCID business cards, we inferred that more than half of the scholars in the top 20 in terms of the number of publications received their Ph.D. before 2008, and two scholars, Mucke, HAM and Zheng, Wei, are older. In contrast, Cheng, FX, a scholar from Case Western Reserve Univ, completed his Ph.D. without a gap in 2013 and may have a longer academic career in the future; therefore, Cheng, FX’s h-index in the field of drug repositioning is likely to grow more in the future and Cheng, FX is likely to have more academic influence.To reveal the focus of drug repositioning research and research trends, the author keywords and the highly cited and hot research topics of the ESI for each of the 2978 publications were analyzed, which were also derived from the core database of the WOS database (SCI-E/SSCI) (). Highly cited studies were defined as studies in the top 1% of the citations for all studies in the same ESI discipline within the 10-year range of inclusion of ESI inclusion (). A hot research topic of the ESI refers to a study published in two\\xa0years with a citation frequency within one of the corresponding disciplines in the world in the last two\\xa0months ().Author keywords tend to provide more information and have thus become a widespread focus (; ). The data of 6,083 author keywords in the search results were merged to make keywords with the same meaning represented by a single unified word. In the end, 5,616 author keywords were obtained. It should be specified that some publications without author keywords were excluded from the statistical analysis. Of these author keywords, 4,296 were used only once, representing 76.50% of the total. A total of 1,216 (21.65%) appeared 2–10 times, 79 (1.41%) appeared 10–20 times, 37 (0.66%) appeared 21–50 times, and the remaining eight (0.14%) were used between 51 and 1,500 times. All keywords cumulatively appear a total of 12,400 times, while the top 30 most used author keywords appear 2,967 times alone, or approximately 23.93%, as shown in . The comparison of keywords in recent years allows for tracking the frontiers of research and predicting hotspots and trends in drug repositioning research. The bubble plots show the three dimensions of the data, namely, the year of publication, the author’s keywords, and the number of corresponding publications. The horizontal change in the size of the bubble illustrates the increasing trend of author keywords over time, the vertical size of the bubble shows the most popular keywords in that year, and the numbers in the bubble indicate the frequency of author keywords and the number of publications.The top 30 keywords include five diseases: “COVID-19/SARS-CoV-2/Coronavirus” (239), “Cancer” (75), “Alzheimer’s disease” (48), “Breast cancer (36)\", and “Chagas disease” (27). Drug names appear four times, “Antiviral drugs” (56), “Metformin” (35), “Statins” (28), and “Antibiotics” (25), which reveal the diseases and applications to which drugs were often repositioned during these 11 years. There were four subject categories, “Bioinformatics” (43), “Polypharmacology” (42), “Systems biology” (42), and “Precision medicine” (28) and eight occurrences of research methods, namely, “Virtual screening” (81), “Molecular docking” (64), “Machine learning” (54, eighth), “Clinical trials” (36), “High-throughput screening” (35), “Connectivity map” (28), and “Molecular dynamics simulation” (26).In the context of the pandemic in 2020, there was a surge in research on the subject, with “COVID-19/SARS-CoV-2/Coronavirus” topping the list of keywords as soon as they appeared that year. “Virtual screening” is a research method that appeared seven times more frequently in 2020 than in the previous year. Since “Drug repurposing/repositioning” is a subject matter and a strategy for drug discovery/drug development, it would not make much sense to analyze these two keywords. Molecular docking is one of the core steps of virtual screening, and the COVID-19 pandemic generated many opportunities for the practice of drug repositioning. Therefore, high-quality studies of the keywords “COVID-19/SARS-CoV-2/Coronavirus”, “Virtual screening”, and “Molecular docking” were surveyed, as shown in the bubble chart, in the past two\\xa0years, reflecting the relevant research trend in recent years. Wang, F et al. developed a new free reverse docking server based on a consensus algorithm (combining several docking algorithm strategies) to address the original shortcomings of computational molecular docking in drug repositioning, such as a low success rate, cumbersome operational steps, and reliance on code writing (). M Lapillo et al. extensively evaluated the performance assessment of docking-based target fishing methods and developed a consensus docking-based target fishing tactic (). In a virtual screening process, Gervasoni, S. conducted a literature search for molecular binding sites for SARS-CoV-2-associated protein targets while combining pocket and docking searches to propose a new pocket mapping strategy that identifies binding cavities with significantly better performance than pocket detection alone (). Xie, L et al. screened antitoxic drugs based on the multitarget structure of the pathway center and stated that this inhibition of multiple targets in one pathway would be more effective than targeting a single protein, and the chance of drug resistance was smaller, which could be applied to other pathways (). Li, Z et al. reported a virtual screening method based on accelerated free energy perturbation absolute binding free energy (FEP-ABFE) prediction and stated that the virtual screening method based on the prediction of FEP-ABFE will play a role in many other drug repositioning studies (). After a series of drug repurposing computational screens and various validation activities by several scientists, it was agreed that raltegravir (; ), clonidine (; ), chloroquine and hydroxychloroquine () have therapeutic effects in the treatment of novel coronavirus.In addition, from the studies on the keyword “Machine learning” over the 10-year period shown in the bubble chart, it was found that the classical machine learning algorithms of support vector machines (; ; ), regularized least squares (; ), logistic regression (; ; ), and random forests (; ) have been widely used in inferring drug–target and drug–disease interactions.While the level of influence of a study is reflected by a combination of many aspects, the number of citations remains an important indicator (). Based on the definition of highly cited and hot ESI papers in  of this study, a total of 108 highly cited studies were obtained, of which 11 were hot research topics. Hot research topics are shown in . It should be noted that the first-ranked author is used here as a representative, and the corresponding institution is shown. This rule is followed in  of this study. All hot research topics were published in 2020, and with the exception of an article describing the damage caused by nonsteroidal anti-inflammatory drugs (NSAIDs) to multiple organs and new information on drug repurposing (), the remaining studies focused on drug repositioning therapeutic target studies in novel coronavirus pneumonia (; ), screening drug studies (; ; ; ; ), reviews of clinical trials (; ), and reports of other coronavirus therapeutic agents and vaccine studies (). From the perspective of cooperation, most of them were completed by a country’s independent agency. In terms of the countries and regions studied, four studies involved US scholars, five studies involved Asian scholars, and one contribution was from an African scholar. In addition, “A SARS-CoV-2 protein interaction map reveals targets for drug repurposing ()”, published in Nature by Gordon, DE with a total of 125 scholars from the United States, the United Kingdom, and France was the most cited publication with 952 citations.Eleven hot research topics were removed from the 108 highly cited ESI studies, and the top 20 most cited studies were selected from the remaining highly cited studies for analysis. In terms of year of publication, the study by Dudley, JT et al. published in NUCLEIC ACIDS RESEARCH in February 2011 was the earliest of these studies (). Five highly cited studies were published in 2013, and three studies were published as recently as 2020. Two studies were published in Nature, and one each was published in Nat. Rev. Drug Discov. and Nat. Med. subj. of Nature E; J. Med. Chem. L was next with two studies. There were 12 studies with the first author or coauthor from the United States, representing more than half of those in , followed by China (4), Canada (2), England (2), Germany (2), Japan (2), and Switzerland (2) in order of contribution of two or more studies. Nine studies were based on collaborations between different institutions in multiple countries. One of them, entitled “Alcohol-abuse drug disulfiram targets cancer via p97 segregase adapter NPL4”, published in Nature in 2017 by Skrott, Z et al. is a collaboration between scholars from six countries: Czech Republic, the United States, Denmark, Sweden, Switzerland, and China (). In TC, “DrugBank 5.0: a major update to the DrugBank database for 2018” () by Canadian University of Alberta scientists Wishart, DS et al. ranked first (1820 total citations). The most cited publication on an annual basis was “Network-based drug repurposing for novel coronavirus 2019-nCoV/SARS-CoV-2”, published in 2020, which was authored by Zhou, YD et al. and was the highest annual average cited publication with 609 citations (). The scientists Cheng, FX and Dudley, JT, contributed to two of these 20 publications and are important influencers in the field.The three studies published in 2020 focus on novel coronavirus-related drug rediscovery activities () and the use of deep learning techniques (; ).  and Pushpakom, S et al. (2019) provided systematic reviews of the methods and challenges of drug repositioning at that time (; ). Initially, Sirota, M et al. (2011) explored the role of integrating genome-wide computational approaches for predicting reusable drugs (), while from 2013 onward, Peters, JU et al., Medina-Franco et al., JL et al., Reddy, AS et al., and Anighoro, A et al. generally recognized the importance of combining multiple points of pharmacological knowledge for drug repositioning studies (; ; ; ). In the face of a worldwide health emergency caused by the Zika virus epidemic,  used drug repositioning to identify lead compounds for drug development (). Of course, techniques related to the mining of repositionable drugs through experimental high-throughput screening, a traditional experimental approach, are not without progress; for example,  () explored the challenges of transferring 3D cell culture technology to the use of high-throughput screening (HTS) ().In 1995, Mchugh et al. investigated the immunomodulatory action mechanism of thalidomide in humans, which was the first relevant publication on drug repositioning (). The publication time can be divided into three phases: the growth period of 1995–2009, the steady growth period of 2010–2018, and the rapid rise from 2019 and beyond. The 2978 publications studied between 2010 and 2020 were completed by 15,338 authors from 3,530 research institutions in 89 countries, and at the time of this study’s completion, the WOS database had surpassed more than 1,400 publications in 2021 under the same search restrictions for the topic, with more than 31,000 citations for the year, supporting further evidence that the topic is still gaining momentum worldwide.The publication countries/regions are divided into three types: first, countries with a traditionally developed medical level, mainly developed countries in Western Europe, North America, and Oceania; second, countries with a developed pharmaceutical manufacturing industry, such as India and Japan in Asia; and third, developing countries with some research potential, such as China, Brazil, Argentina, and Mexico. In terms of national cooperation, Western European countries have shown a high degree of cooperation, with the United States, China, and the United Kingdom cooperating more frequently. This may be because Western European countries have a tradition of cooperation in the field of research, and the United States, China, and the United Kingdom are the most powerful countries in terms of drug repositioning publications and therefore cooperate more with each other. The United States accounts for half of the 20 most productive institutions, which may explain why the United States still publishes more than 50% of its studies independently, despite having the largest international collaborative network base, because it already has the most active and high-quality producing institutions within the country for research institutions seeking collaboration. Furthermore, 19 of these 20 institutions are universities and research institutes, and one is a company, HM Pharma Consultancy, which was established in 2000 to focus on drug repositioning for the development of new drugs (). This evidence suggests that the topic of drug repositioning is not only widely studied in academia but also has a place in the industry.The 2978 studies are spread across 74 research areas, but pharmacology and pharmacy and biochemistry and molecular biology account for a larger proportion of the total number of studies. It is quite notable that the majority of studies reported in biophysics did not rise significantly until 2020. The reasons for this may be the following: first, there was a breakthrough in basic research in this field in 2020 and second, due to the novel coronavirus, research in this direction has increased its application for the prevention and control of the pandemic.In terms of journals, Sci Rep ranked first, followed by PLoS One and J. Biomol. Struct. Dyn. In terms of lead authors, three have the most productive and influential positions: Cheng, FX is the most prolific author, based on the number of papers and h-index; Mucke, HAM is the most frequent corresponding author; and Butte, AJ is the top author in terms of ACPP ranking. Even though Latin American countries do not have an advantage in terms of national cooperation or the total number of institutional funding units, Latin American scholars have overcome many obstacles and are actively at the forefront of scholarship, contributing significantly to the field.Through the analysis of the authors’ keywords, cancer has been the main disease addressed by this method. Metformin has been found by many scientists to have a good inhibitory effect on various tumors, mainly in gynecology (; ; ; ), and it has become a specific drug that has been most frequently mentioned in drug reuse in recent years. In terms of “antiviral drugs”, scholars not only use drug repositioning to find antiviral drugs to treat diseases, such as Ebola (; ) and HIV (), that have plagued humans for a long time but also use this method to seek treatments for infectious diseases, such as Zika virus (; ) and novel coronaviruses that have threatened several countries and even the world. For these diseases, emergency research on drug repositioning has played an important role in reducing mortality, calming patient fears, and restoring economic production when no specific drugs or vaccines were initially available during the pandemic. The combination of precision medicine and drug repositioning studies, often used to seek treatments for rare diseases () and, in particular, genetically related diseases (), is expected to be fully developed in the future. In the past 2\\xa0years, “Virtual screening”, together with “Molecular docking” and “Machine learning”, has become the most cutting-edge and important research methods in related technology fields, constantly improving the accuracy of drug reuse and screening. Currently, to develop more efficient and accurate research, there are two trends in the use of drug repositioning. One is the combination of various methods, such as the use of text mining and network analysis, and the creation of statistical models for predicting semantic link association to assess the relationship between pharmacological target pairings (); text analysis combined with machine learning () to develop drugs for Parkinson’s disease; prediction of new DTIs using data from multiple databases (); and the obtained relocated anticancer drugs were verified by cross-validation, literature, and experimental verification (). Second, the most advanced algorithms are applied and improved, such as matrix decomposition (; ; ; ; ) and matrix completion (; ) and deep learning (; ; ; ; ; ).In fact, some of the studies in the list of highly cited research topics on novel coronaviruses drug repurposing studies are currently approaching 3,000 citations on Google Scholar (). The percentage of highly cited studies and hot research topics related to novel coronaviruses is also a good indication that the method has made an indelible contribution to the study of novel coronaviruses and similar infectious diseases. Auxiliary technology for the experimental screening of traditional drugs is also developing (), which also promotes drug repositioning or other drug development processes. Furthermore, the high-quality results of Elfiky, AA, a scientist from Cairo University, Egypt (), suggest that relevant research in some economically underdeveloped countries may reach top levels worldwide due to the return or affiliation of some prominent scientists.For this research, the literature on drug repositioning research published in the SCI-E and SSCI sections of WOS core journals from 2010 to 2020 was analyzed based on bibliometrics and DDA software. This area has been of interest to scientists since the end of the 20th century and entered a period of rapid growth in 2019, with the peak far from being reached. Using bibliometrics as a tool, the United States has become a world leader in terms of the number of submissions, number of high-quality studies, funding support, strength of research institutions, and number of top scholars, followed by China and the United Kingdom, where more research is being performed in this area. As a method of drug discovery, drug repurposing is closely related to the development of various biomedical disciplines, and computer-related disciplinary methods, such as mathematical computational biology and computer science, have taken an important place in the research of this field in the last decade. The authors’ keyword analysis suggests that research in the field of the novel coronavirus will remain valuable until the associated pandemic is completely contained. Virtual screening, molecular docking, machine learning, and other related technical fields still need long-term development to achieve efficient and accurate repositioning of drugs (). Precision medicine, combined with drug repositioning, is the most promising direction for the future. In conclusion, drug repositioning can help to treat more diseases, such as drug resistance, poor drug selectivity, and limited therapeutic options.This study may help some scholars with an initial interest in drug repositioning-related research to gain a concise and rapid understanding of the current state of global research, as well as offer some relevant information to institutions or groups seeking collaboration.It is worth noting that this study has some biases and limitations. First, there are still some issues with the publications included in the study based on subject terms: 1) some relevant publications that do not use the search formula in this study may have been excluded from this study and 2) there may also be a small number of articles whose use of some of the aforementioned search terms deviates significantly from the general understanding; yet, such publications are included in this study. Second, some extraneous factors distort the credibility of the bibliometric statistics. 1) When analyzing the keywords of publications, some publications are excluded from the statistical analysis because they do not list author keywords (e.g., ()). 2) Excessive self-citation by some authors () inflates the actual level of interest in the publication. 3) for a publication, when an author submits more than one institution’s address information, this publication is counted as research results by each institution. Finally, in future work, patents from the WOS database associated with the topic of drug repositioning will be analyzed to provide another perspective on the situation of the topic in terms of applications and technological innovations.'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "len(concat_lists)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1be3eaf3-ed58-4459-87fe-930a506c4cd4",
        "id": "rqbi2LubJN0Q"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "11397"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "BgoGIaZTJN0Q"
      },
      "outputs": [],
      "source": [
        "with open('2023-01-06_concat_all_11397.pickle', 'wb') as f:\n",
        "  pickle.dump(concat_lists, f)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 7.8 Add extracted full text to DataFrame\n",
        "\n",
        "Load metadata for Europe PMC articles."
      ],
      "metadata": {
        "id": "fxR3USwo5iKQ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_europepmc_df_json_ft_urls_11397.pickle', 'rb') as f:\n",
        "    search_results_new = pickle.load(f)"
      ],
      "metadata": {
        "id": "NKRCibuUhW2L"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_full_text = search_results_new.copy()"
      ],
      "metadata": {
        "id": "8pzt3Dy8oeHs"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_full_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "pRRyGa12hbqa",
        "outputId": "0625c218-b99e-4f21-b751-85269d814392"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "0      PMC9549161  2022-09-26  2022-10-14   \n",
              "1      PMC9539342  2022-09-22  2022-11-12   \n",
              "2      PMC9357751  2022-12-01  2022-12-05   \n",
              "3      PMC9346052  2022-08-03  2022-09-05   \n",
              "4      PMC9775208  2022-12-15  2022-12-25   \n",
              "...           ...         ...         ...   \n",
              "11392  PMC6328940  2019-01-01  2020-03-09   \n",
              "11393  PMC6598402  2019-06-21  2020-09-28   \n",
              "11394  PMC6481739  2019-02-05  2020-09-28   \n",
              "11395  PMC6624471  2019-07-05  2020-09-28   \n",
              "11396  PMC6409730  2019-02-06  2020-09-30   \n",
              "\n",
              "                                                   title  \\\n",
              "0           Drug repositioning: A bibliometric analysis.   \n",
              "1      A review on computer-aided chemogenomics and d...   \n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "3      Scope of repurposed drugs against the potentia...   \n",
              "4      Drug Repurposing Using Gene Co-Expression and ...   \n",
              "...                                                  ...   \n",
              "11392  β-RA reduces DMQ/CoQ ratio and rescues the enc...   \n",
              "11393  Alzheimer Disease Pathogenesis: Insights From ...   \n",
              "11394  Modeling cardiac complexity: Advancements in m...   \n",
              "11395  Tissue Response to Neural Implants: The Use of...   \n",
              "11396  24th International Symposium on Infections in ...   \n",
              "\n",
              "                                                 journal  \\\n",
              "0                              Frontiers in pharmacology   \n",
              "1                         Chemical biology & drug design   \n",
              "2      Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                                   Structural chemistry   \n",
              "4                                                Biology   \n",
              "...                                                  ...   \n",
              "11392                            EMBO molecular medicine   \n",
              "11393                          Frontiers in neuroscience   \n",
              "11394                                 APL bioengineering   \n",
              "11395                          Frontiers in neuroscience   \n",
              "11396              Medical sciences (Basel, Switzerland)   \n",
              "\n",
              "                                                 authors  \\\n",
              "0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                      0   \n",
              "3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                    Mailem RC, Tayo LL.   \n",
              "...                                                  ...   \n",
              "11392  Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...   \n",
              "11393                                Chen XQ, Mobley WC.   \n",
              "11394  Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...   \n",
              "11395       Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.   \n",
              "11396  Artigas A, Carlet J, Garnacho J, Niederman M, ...   \n",
              "\n",
              "                              doi  \\\n",
              "0       10.3389/fphar.2022.974849   \n",
              "1              10.1111/cbdd.14136   \n",
              "2                               0   \n",
              "3      10.1007/s11224-022-02020-z   \n",
              "4         10.3390/biology11121827   \n",
              "...                           ...   \n",
              "11392     10.15252/emmm.201809466   \n",
              "11393    10.3389/fnins.2019.00659   \n",
              "11394           10.1063/1.5055873   \n",
              "11395    10.3389/fnins.2019.00689   \n",
              "11396       10.3390/medsci7020023   \n",
              "\n",
              "                                                 pdf_url  \n",
              "0      https://europepmc.org/articles/PMC9549161?pdf=...  \n",
              "1      https://europepmc.org/articles/PMC9539342?pdf=...  \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...  \n",
              "3      https://europepmc.org/articles/PMC9346052?pdf=...  \n",
              "4      https://europepmc.org/articles/PMC9775208?pdf=...  \n",
              "...                                                  ...  \n",
              "11392  https://europepmc.org/articles/PMC6328940?pdf=...  \n",
              "11393  https://europepmc.org/articles/PMC6598402?pdf=...  \n",
              "11394  https://europepmc.org/articles/PMC6481739?pdf=...  \n",
              "11395  https://europepmc.org/articles/PMC6624471?pdf=...  \n",
              "11396  https://europepmc.org/articles/PMC6409730?pdf=...  \n",
              "\n",
              "[11397 rows x 8 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-a6f031a4-f874-4a45-be66-beabe298fb9e\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11392</th>\n",
              "      <td>PMC6328940</td>\n",
              "      <td>2019-01-01</td>\n",
              "      <td>2020-03-09</td>\n",
              "      <td>β-RA reduces DMQ/CoQ ratio and rescues the enc...</td>\n",
              "      <td>EMBO molecular medicine</td>\n",
              "      <td>Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...</td>\n",
              "      <td>10.15252/emmm.201809466</td>\n",
              "      <td>https://europepmc.org/articles/PMC6328940?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11393</th>\n",
              "      <td>PMC6598402</td>\n",
              "      <td>2019-06-21</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Alzheimer Disease Pathogenesis: Insights From ...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Chen XQ, Mobley WC.</td>\n",
              "      <td>10.3389/fnins.2019.00659</td>\n",
              "      <td>https://europepmc.org/articles/PMC6598402?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11394</th>\n",
              "      <td>PMC6481739</td>\n",
              "      <td>2019-02-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Modeling cardiac complexity: Advancements in m...</td>\n",
              "      <td>APL bioengineering</td>\n",
              "      <td>Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...</td>\n",
              "      <td>10.1063/1.5055873</td>\n",
              "      <td>https://europepmc.org/articles/PMC6481739?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11395</th>\n",
              "      <td>PMC6624471</td>\n",
              "      <td>2019-07-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Tissue Response to Neural Implants: The Use of...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.</td>\n",
              "      <td>10.3389/fnins.2019.00689</td>\n",
              "      <td>https://europepmc.org/articles/PMC6624471?pdf=...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11396</th>\n",
              "      <td>PMC6409730</td>\n",
              "      <td>2019-02-06</td>\n",
              "      <td>2020-09-30</td>\n",
              "      <td>24th International Symposium on Infections in ...</td>\n",
              "      <td>Medical sciences (Basel, Switzerland)</td>\n",
              "      <td>Artigas A, Carlet J, Garnacho J, Niederman M, ...</td>\n",
              "      <td>10.3390/medsci7020023</td>\n",
              "      <td>https://europepmc.org/articles/PMC6409730?pdf=...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11397 rows × 8 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-a6f031a4-f874-4a45-be66-beabe298fb9e')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-a6f031a4-f874-4a45-be66-beabe298fb9e button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-a6f031a4-f874-4a45-be66-beabe298fb9e');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Load concatenated lists of full text and add as 'text' column to metadata DataFrame."
      ],
      "metadata": {
        "id": "44gMeAxEQN7U"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_concat_all_11397.pickle', 'rb') as f:\n",
        "    concat_lists = pickle.load(f)"
      ],
      "metadata": {
        "id": "TZOlvfMphj8a"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_full_text['text'] = concat_lists\n",
        "search_results_full_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "Rq3GmAjRhnWi",
        "outputId": "1ee06828-ddbb-4c07-e514-0c9bd8b3bafd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "0      PMC9549161  2022-09-26  2022-10-14   \n",
              "1      PMC9539342  2022-09-22  2022-11-12   \n",
              "2      PMC9357751  2022-12-01  2022-12-05   \n",
              "3      PMC9346052  2022-08-03  2022-09-05   \n",
              "4      PMC9775208  2022-12-15  2022-12-25   \n",
              "...           ...         ...         ...   \n",
              "11392  PMC6328940  2019-01-01  2020-03-09   \n",
              "11393  PMC6598402  2019-06-21  2020-09-28   \n",
              "11394  PMC6481739  2019-02-05  2020-09-28   \n",
              "11395  PMC6624471  2019-07-05  2020-09-28   \n",
              "11396  PMC6409730  2019-02-06  2020-09-30   \n",
              "\n",
              "                                                   title  \\\n",
              "0           Drug repositioning: A bibliometric analysis.   \n",
              "1      A review on computer-aided chemogenomics and d...   \n",
              "2      Repurposing Molnupiravir as a new opportunity ...   \n",
              "3      Scope of repurposed drugs against the potentia...   \n",
              "4      Drug Repurposing Using Gene Co-Expression and ...   \n",
              "...                                                  ...   \n",
              "11392  β-RA reduces DMQ/CoQ ratio and rescues the enc...   \n",
              "11393  Alzheimer Disease Pathogenesis: Insights From ...   \n",
              "11394  Modeling cardiac complexity: Advancements in m...   \n",
              "11395  Tissue Response to Neural Implants: The Use of...   \n",
              "11396  24th International Symposium on Infections in ...   \n",
              "\n",
              "                                                 journal  \\\n",
              "0                              Frontiers in pharmacology   \n",
              "1                         Chemical biology & drug design   \n",
              "2      Journal of Generic Medicines : Duplicate, mark...   \n",
              "3                                   Structural chemistry   \n",
              "4                                                Biology   \n",
              "...                                                  ...   \n",
              "11392                            EMBO molecular medicine   \n",
              "11393                          Frontiers in neuroscience   \n",
              "11394                                 APL bioengineering   \n",
              "11395                          Frontiers in neuroscience   \n",
              "11396              Medical sciences (Basel, Switzerland)   \n",
              "\n",
              "                                                 authors  \\\n",
              "0      Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...   \n",
              "1      Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...   \n",
              "2                                                      0   \n",
              "3      Niranjan V, Setlur AS, Karunakaran C, Uttarkar...   \n",
              "4                                    Mailem RC, Tayo LL.   \n",
              "...                                                  ...   \n",
              "11392  Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...   \n",
              "11393                                Chen XQ, Mobley WC.   \n",
              "11394  Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...   \n",
              "11395       Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.   \n",
              "11396  Artigas A, Carlet J, Garnacho J, Niederman M, ...   \n",
              "\n",
              "                              doi  \\\n",
              "0       10.3389/fphar.2022.974849   \n",
              "1              10.1111/cbdd.14136   \n",
              "2                               0   \n",
              "3      10.1007/s11224-022-02020-z   \n",
              "4         10.3390/biology11121827   \n",
              "...                           ...   \n",
              "11392     10.15252/emmm.201809466   \n",
              "11393    10.3389/fnins.2019.00659   \n",
              "11394           10.1063/1.5055873   \n",
              "11395    10.3389/fnins.2019.00689   \n",
              "11396       10.3390/medsci7020023   \n",
              "\n",
              "                                                 pdf_url  \\\n",
              "0      https://europepmc.org/articles/PMC9549161?pdf=...   \n",
              "1      https://europepmc.org/articles/PMC9539342?pdf=...   \n",
              "2      https://europepmc.org/articles/PMC9357751?pdf=...   \n",
              "3      https://europepmc.org/articles/PMC9346052?pdf=...   \n",
              "4      https://europepmc.org/articles/PMC9775208?pdf=...   \n",
              "...                                                  ...   \n",
              "11392  https://europepmc.org/articles/PMC6328940?pdf=...   \n",
              "11393  https://europepmc.org/articles/PMC6598402?pdf=...   \n",
              "11394  https://europepmc.org/articles/PMC6481739?pdf=...   \n",
              "11395  https://europepmc.org/articles/PMC6624471?pdf=...   \n",
              "11396  https://europepmc.org/articles/PMC6409730?pdf=...   \n",
              "\n",
              "                                                    text  \n",
              "0      Sir James Black, a winner of the 1988 Nobel Pr...  \n",
              "1      Tight and selective interaction between ligand...  \n",
              "2      The severe acute respiratory syndrome coronavi...  \n",
              "3      The sudden outbreak of SARS-CoV-2 in 2019 took...  \n",
              "4      The 2019 novel coronavirus, now dubbed SARS-Co...  \n",
              "...                                                  ...  \n",
              "11392  Mitochondria are the primary site of cellular ...  \n",
              "11393  AD is the most common cause of dementia, accou...  \n",
              "11394  Compromised contractility of the heart is a ma...  \n",
              "11395  Recent technological progress in the field of ...  \n",
              "11396  \\n\\nThis 24th International Symposium on Infec...  \n",
              "\n",
              "[11397 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-80254bd3-4073-41b3-9298-87871b2881d1\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>PMC9549161</td>\n",
              "      <td>2022-09-26</td>\n",
              "      <td>2022-10-14</td>\n",
              "      <td>Drug repositioning: A bibliometric analysis.</td>\n",
              "      <td>Frontiers in pharmacology</td>\n",
              "      <td>Sun G, Dong D, Dong Z, Zhang Q, Fang H, Wang C...</td>\n",
              "      <td>10.3389/fphar.2022.974849</td>\n",
              "      <td>https://europepmc.org/articles/PMC9549161?pdf=...</td>\n",
              "      <td>Sir James Black, a winner of the 1988 Nobel Pr...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>PMC9539342</td>\n",
              "      <td>2022-09-22</td>\n",
              "      <td>2022-11-12</td>\n",
              "      <td>A review on computer-aided chemogenomics and d...</td>\n",
              "      <td>Chemical biology &amp; drug design</td>\n",
              "      <td>Maghsoudi S, Taghavi Shahraki B, Rameh F, Naza...</td>\n",
              "      <td>10.1111/cbdd.14136</td>\n",
              "      <td>https://europepmc.org/articles/PMC9539342?pdf=...</td>\n",
              "      <td>Tight and selective interaction between ligand...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>PMC9357751</td>\n",
              "      <td>2022-12-01</td>\n",
              "      <td>2022-12-05</td>\n",
              "      <td>Repurposing Molnupiravir as a new opportunity ...</td>\n",
              "      <td>Journal of Generic Medicines : Duplicate, mark...</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC9357751?pdf=...</td>\n",
              "      <td>The severe acute respiratory syndrome coronavi...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>PMC9346052</td>\n",
              "      <td>2022-08-03</td>\n",
              "      <td>2022-09-05</td>\n",
              "      <td>Scope of repurposed drugs against the potentia...</td>\n",
              "      <td>Structural chemistry</td>\n",
              "      <td>Niranjan V, Setlur AS, Karunakaran C, Uttarkar...</td>\n",
              "      <td>10.1007/s11224-022-02020-z</td>\n",
              "      <td>https://europepmc.org/articles/PMC9346052?pdf=...</td>\n",
              "      <td>The sudden outbreak of SARS-CoV-2 in 2019 took...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>PMC9775208</td>\n",
              "      <td>2022-12-15</td>\n",
              "      <td>2022-12-25</td>\n",
              "      <td>Drug Repurposing Using Gene Co-Expression and ...</td>\n",
              "      <td>Biology</td>\n",
              "      <td>Mailem RC, Tayo LL.</td>\n",
              "      <td>10.3390/biology11121827</td>\n",
              "      <td>https://europepmc.org/articles/PMC9775208?pdf=...</td>\n",
              "      <td>The 2019 novel coronavirus, now dubbed SARS-Co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11392</th>\n",
              "      <td>PMC6328940</td>\n",
              "      <td>2019-01-01</td>\n",
              "      <td>2020-03-09</td>\n",
              "      <td>β-RA reduces DMQ/CoQ ratio and rescues the enc...</td>\n",
              "      <td>EMBO molecular medicine</td>\n",
              "      <td>Hidalgo-Gutiérrez A, Barriocanal-Casado E, Bak...</td>\n",
              "      <td>10.15252/emmm.201809466</td>\n",
              "      <td>https://europepmc.org/articles/PMC6328940?pdf=...</td>\n",
              "      <td>Mitochondria are the primary site of cellular ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11393</th>\n",
              "      <td>PMC6598402</td>\n",
              "      <td>2019-06-21</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Alzheimer Disease Pathogenesis: Insights From ...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Chen XQ, Mobley WC.</td>\n",
              "      <td>10.3389/fnins.2019.00659</td>\n",
              "      <td>https://europepmc.org/articles/PMC6598402?pdf=...</td>\n",
              "      <td>AD is the most common cause of dementia, accou...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11394</th>\n",
              "      <td>PMC6481739</td>\n",
              "      <td>2019-02-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Modeling cardiac complexity: Advancements in m...</td>\n",
              "      <td>APL bioengineering</td>\n",
              "      <td>Callaghan NI, Hadipour-Lakmehsari S, Lee SH, G...</td>\n",
              "      <td>10.1063/1.5055873</td>\n",
              "      <td>https://europepmc.org/articles/PMC6481739?pdf=...</td>\n",
              "      <td>Compromised contractility of the heart is a ma...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11395</th>\n",
              "      <td>PMC6624471</td>\n",
              "      <td>2019-07-05</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Tissue Response to Neural Implants: The Use of...</td>\n",
              "      <td>Frontiers in neuroscience</td>\n",
              "      <td>Gulino M, Kim D, Pané S, Santos SD, Pêgo AP.</td>\n",
              "      <td>10.3389/fnins.2019.00689</td>\n",
              "      <td>https://europepmc.org/articles/PMC6624471?pdf=...</td>\n",
              "      <td>Recent technological progress in the field of ...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11396</th>\n",
              "      <td>PMC6409730</td>\n",
              "      <td>2019-02-06</td>\n",
              "      <td>2020-09-30</td>\n",
              "      <td>24th International Symposium on Infections in ...</td>\n",
              "      <td>Medical sciences (Basel, Switzerland)</td>\n",
              "      <td>Artigas A, Carlet J, Garnacho J, Niederman M, ...</td>\n",
              "      <td>10.3390/medsci7020023</td>\n",
              "      <td>https://europepmc.org/articles/PMC6409730?pdf=...</td>\n",
              "      <td>\\n\\nThis 24th International Symposium on Infec...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>11397 rows × 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-80254bd3-4073-41b3-9298-87871b2881d1')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-80254bd3-4073-41b3-9298-87871b2881d1 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-80254bd3-4073-41b3-9298-87871b2881d1');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 19
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_search_results_full_text.pickle', 'wb') as f:\n",
        "  pickle.dump(search_results_full_text, f)"
      ],
      "metadata": {
        "id": "XyiOy6DTFze4"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 8. Check for missing text\n",
        "\n",
        "Concise summary of DataFrame to see if there are any articles with missing text."
      ],
      "metadata": {
        "id": "ZprdE-7iwa9-"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "search_results_full_text.info()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "z1Q3pbYYiDEy",
        "outputId": "844ced9d-e2b2-4c51-d8ee-347d5c495ffd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "RangeIndex: 11397 entries, 0 to 11396\n",
            "Data columns (total 9 columns):\n",
            " #   Column     Non-Null Count  Dtype \n",
            "---  ------     --------------  ----- \n",
            " 0   pmcid      11397 non-null  object\n",
            " 1   published  11397 non-null  object\n",
            " 2   revised    11397 non-null  object\n",
            " 3   title      11397 non-null  object\n",
            " 4   journal    11397 non-null  object\n",
            " 5   authors    11397 non-null  object\n",
            " 6   doi        11397 non-null  object\n",
            " 7   pdf_url    11397 non-null  object\n",
            " 8   text       11279 non-null  object\n",
            "dtypes: object(9)\n",
            "memory usage: 801.5+ KB\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We can see that there are 118 articles with missing values for the text column."
      ],
      "metadata": {
        "id": "t7sQmSX4RtSB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_no_full_text = search_results_full_text[search_results_full_text['text'].isna()]\n",
        "pmc_search_results_no_full_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "Hz9rLQkUijnh",
        "outputId": "0ba14cce-432c-400f-d6e6-3ff1bd2ecce5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "64     PMC9538661  2022-10-13  2022-11-22   \n",
              "188    PMC9794394  2022-12-28  2023-01-02   \n",
              "212    PMC9538837  2022-10-10  2022-11-22   \n",
              "256    PMC9788990  2022-12-24  2023-01-02   \n",
              "275    PMC9794516  2022-12-28  2023-01-02   \n",
              "...           ...         ...         ...   \n",
              "9797   PMC7162151  2020-02-28  2020-04-21   \n",
              "9809   PMC9094125  2022-05-11  2022-07-16   \n",
              "10399  PMC7492056  2020-09-15  2020-09-28   \n",
              "10767  PMC8459260  2021-02-25  2022-01-25   \n",
              "11329  PMC7163523  2019-08-01  2021-02-16   \n",
              "\n",
              "                                                   title  \\\n",
              "64     Recent Drug Development and Medicinal Chemistr...   \n",
              "188    Nonstructural protein 1 (nsp1) widespread RNA ...   \n",
              "212    The Efficacy of Traditional Medicinal Plants i...   \n",
              "256    Sleep and circadian rhythm disruption alters t...   \n",
              "275    MultiOMICs landscape of SARS-CoV-2-induced hos...   \n",
              "...                                                  ...   \n",
              "9797                                               News.   \n",
              "9809   Recent advances in metal-organic framework-bas...   \n",
              "10399                                    Full Issue PDF.   \n",
              "10767  Getting in touch with your senses: Mechanisms ...   \n",
              "11329  Literature review of baseline information on n...   \n",
              "\n",
              "                            journal  \\\n",
              "64                      ChemMedChem   \n",
              "188                        iScience   \n",
              "212        Chemistry & biodiversity   \n",
              "256                        iScience   \n",
              "275                        iScience   \n",
              "...                             ...   \n",
              "9797           Chemistry & industry   \n",
              "9809                  Nano research   \n",
              "10399          JACC. CardioOncology   \n",
              "10767   WIREs mechanisms of disease   \n",
              "11329  EFSA Supporting Publications   \n",
              "\n",
              "                                                 authors  \\\n",
              "64         Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.   \n",
              "188                       Bermudez Y, Miles J, Muller M.   \n",
              "212                      Choe J, Har Yong P, Xiang Ng Z.   \n",
              "256    Taylor L, Von Lendenfeld F, Ashton A, Sanghani...   \n",
              "275    Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...   \n",
              "...                                                  ...   \n",
              "9797                                                   0   \n",
              "9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   \n",
              "10399                                                  0   \n",
              "10767                                Gupta S, Butler SJ.   \n",
              "11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   \n",
              "\n",
              "                                 doi  \\\n",
              "64            10.1002/cmdc.202200440   \n",
              "188       10.1016/j.isci.2022.105887   \n",
              "212           10.1002/cbdv.202200655   \n",
              "256       10.1016/j.isci.2022.105877   \n",
              "275       10.1016/j.isci.2022.105895   \n",
              "...                              ...   \n",
              "9797            10.1002/cind.842_3.x   \n",
              "9809       10.1007/s12274-022-4302-x   \n",
              "10399  10.1016/s2666-0873(20)30180-0   \n",
              "10767              10.1002/wsbm.1520   \n",
              "11329                              0   \n",
              "\n",
              "                                                 pdf_url  text  \n",
              "64     https://europepmc.org/articles/PMC9538661?pdf=...  None  \n",
              "188    https://europepmc.org/articles/PMC9794394?pdf=...  None  \n",
              "212    https://europepmc.org/articles/PMC9538837?pdf=...  None  \n",
              "256    https://europepmc.org/articles/PMC9788990?pdf=...  None  \n",
              "275    https://europepmc.org/articles/PMC9794516?pdf=...  None  \n",
              "...                                                  ...   ...  \n",
              "9797   https://europepmc.org/articles/PMC7162151?pdf=...  None  \n",
              "9809   https://europepmc.org/articles/PMC9094125?pdf=...  None  \n",
              "10399  https://europepmc.org/articles/PMC7492056?pdf=...  None  \n",
              "10767  https://europepmc.org/articles/PMC8459260?pdf=...  None  \n",
              "11329  https://europepmc.org/articles/PMC7163523?pdf=...  None  \n",
              "\n",
              "[118 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-61aa63fb-e950-4540-8f6a-ab4af63e3018\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>64</th>\n",
              "      <td>PMC9538661</td>\n",
              "      <td>2022-10-13</td>\n",
              "      <td>2022-11-22</td>\n",
              "      <td>Recent Drug Development and Medicinal Chemistr...</td>\n",
              "      <td>ChemMedChem</td>\n",
              "      <td>Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.</td>\n",
              "      <td>10.1002/cmdc.202200440</td>\n",
              "      <td>https://europepmc.org/articles/PMC9538661?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>188</th>\n",
              "      <td>PMC9794394</td>\n",
              "      <td>2022-12-28</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>Nonstructural protein 1 (nsp1) widespread RNA ...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Bermudez Y, Miles J, Muller M.</td>\n",
              "      <td>10.1016/j.isci.2022.105887</td>\n",
              "      <td>https://europepmc.org/articles/PMC9794394?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>212</th>\n",
              "      <td>PMC9538837</td>\n",
              "      <td>2022-10-10</td>\n",
              "      <td>2022-11-22</td>\n",
              "      <td>The Efficacy of Traditional Medicinal Plants i...</td>\n",
              "      <td>Chemistry &amp; biodiversity</td>\n",
              "      <td>Choe J, Har Yong P, Xiang Ng Z.</td>\n",
              "      <td>10.1002/cbdv.202200655</td>\n",
              "      <td>https://europepmc.org/articles/PMC9538837?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>256</th>\n",
              "      <td>PMC9788990</td>\n",
              "      <td>2022-12-24</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>Sleep and circadian rhythm disruption alters t...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Taylor L, Von Lendenfeld F, Ashton A, Sanghani...</td>\n",
              "      <td>10.1016/j.isci.2022.105877</td>\n",
              "      <td>https://europepmc.org/articles/PMC9788990?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>275</th>\n",
              "      <td>PMC9794516</td>\n",
              "      <td>2022-12-28</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>MultiOMICs landscape of SARS-CoV-2-induced hos...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...</td>\n",
              "      <td>10.1016/j.isci.2022.105895</td>\n",
              "      <td>https://europepmc.org/articles/PMC9794516?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9797</th>\n",
              "      <td>PMC7162151</td>\n",
              "      <td>2020-02-28</td>\n",
              "      <td>2020-04-21</td>\n",
              "      <td>News.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.842_3.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7162151?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9809</th>\n",
              "      <td>PMC9094125</td>\n",
              "      <td>2022-05-11</td>\n",
              "      <td>2022-07-16</td>\n",
              "      <td>Recent advances in metal-organic framework-bas...</td>\n",
              "      <td>Nano research</td>\n",
              "      <td>Yang M, Zhang J, Wei Y, Zhang J, Tao C.</td>\n",
              "      <td>10.1007/s12274-022-4302-x</td>\n",
              "      <td>https://europepmc.org/articles/PMC9094125?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10399</th>\n",
              "      <td>PMC7492056</td>\n",
              "      <td>2020-09-15</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Full Issue PDF.</td>\n",
              "      <td>JACC. CardioOncology</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2666-0873(20)30180-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7492056?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10767</th>\n",
              "      <td>PMC8459260</td>\n",
              "      <td>2021-02-25</td>\n",
              "      <td>2022-01-25</td>\n",
              "      <td>Getting in touch with your senses: Mechanisms ...</td>\n",
              "      <td>WIREs mechanisms of disease</td>\n",
              "      <td>Gupta S, Butler SJ.</td>\n",
              "      <td>10.1002/wsbm.1520</td>\n",
              "      <td>https://europepmc.org/articles/PMC8459260?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11329</th>\n",
              "      <td>PMC7163523</td>\n",
              "      <td>2019-08-01</td>\n",
              "      <td>2021-02-16</td>\n",
              "      <td>Literature review of baseline information on n...</td>\n",
              "      <td>EFSA Supporting Publications</td>\n",
              "      <td>Dávalos A, Henriques R, Latasa M, Laparra M, C...</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7163523?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>118 rows × 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-61aa63fb-e950-4540-8f6a-ab4af63e3018')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-61aa63fb-e950-4540-8f6a-ab4af63e3018 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-61aa63fb-e950-4540-8f6a-ab4af63e3018');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_pmc_search_results_no_full_text.pickle', 'wb') as f:\n",
        "  pickle.dump(pmc_search_results_no_full_text, f)"
      ],
      "metadata": {
        "id": "XzyanKvxAvib"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check one of the articles to see 'None' in text column."
      ],
      "metadata": {
        "id": "rfuHLiSLUp9K"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_no_full_text.loc[pmc_search_results_no_full_text['pmcid']  == 'PMC8459260']"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 231
        },
        "id": "oKMKxwcsi9ZM",
        "outputId": "db58d75f-d95b-4874-82a5-eada0581f9c7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "10767  PMC8459260  2021-02-25  2022-01-25   \n",
              "\n",
              "                                                   title  \\\n",
              "10767  Getting in touch with your senses: Mechanisms ...   \n",
              "\n",
              "                           journal              authors                doi  \\\n",
              "10767  WIREs mechanisms of disease  Gupta S, Butler SJ.  10.1002/wsbm.1520   \n",
              "\n",
              "                                                 pdf_url  text  \n",
              "10767  https://europepmc.org/articles/PMC8459260?pdf=...  None  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-51067ae6-3843-4f07-b918-41850b861058\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>10767</th>\n",
              "      <td>PMC8459260</td>\n",
              "      <td>2021-02-25</td>\n",
              "      <td>2022-01-25</td>\n",
              "      <td>Getting in touch with your senses: Mechanisms ...</td>\n",
              "      <td>WIREs mechanisms of disease</td>\n",
              "      <td>Gupta S, Butler SJ.</td>\n",
              "      <td>10.1002/wsbm.1520</td>\n",
              "      <td>https://europepmc.org/articles/PMC8459260?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-51067ae6-3843-4f07-b918-41850b861058')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-51067ae6-3843-4f07-b918-41850b861058 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-51067ae6-3843-4f07-b918-41850b861058');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 24
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The above article PMC8459260 was one of the 13 exceptions raised earlier."
      ],
      "metadata": {
        "id": "TG3vs0Dkpou9"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 9. GROBID to download exceptions full text as XML\n",
        "\n",
        "We will use the parse_pdf function as for the arXiv PDFs using [GROBID](https://github.com/kermitt2/grobid/) to download parsed XML for the PDF versions of the Europe PMC articles for which there are exceptions when trying to extract the text via the Europe PMC API.\n",
        "\n",
        "The code below used the previous demo server hosted at [https://cloud.science-miner.com/grobid](https://cloud.science-miner.com/grobid) which has since been updated and redirects to a new demo server so updating the GROBID URL in the function to [https://kermitt2-grobid.hf.space/api/processFulltextDocument](https://kermitt2-grobid.hf.space/api/processFulltextDocument) will be necessary."
      ],
      "metadata": {
        "id": "G_uXseRxxBJk"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def parse_pdf(pdf_url: str):\n",
        "    \"\"\"\n",
        "    Parse PDF to XML using GROBID tool\n",
        "\n",
        "    :param pdf_url: str, URL to article PDF\n",
        "\n",
        "    :return: XML of parsed article\n",
        "    \"\"\"\n",
        "    # GROBID URL for the cloud service to parse full text of the article\n",
        "    url = \"https://cloud.science-miner.com/grobid/api/processFulltextDocument\"\n",
        "\n",
        "    if isinstance(pdf_url, str):\n",
        "            page = urllib.request.urlopen(pdf_url).read()\n",
        "            resp = requests.post(url, files={\"input\": page})\n",
        "            if resp.status_code != 200:\n",
        "              raise Exception(resp.text)\n",
        "            parsed_article = resp.text\n",
        "            time.sleep(3)\n",
        "    else:\n",
        "        raise TypeError(\"Need to supply a url\")\n",
        "\n",
        "    return parsed_article"
      ],
      "metadata": {
        "id": "bu9EnUWu637F"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Running the above function with multithreading for the 118 exceptions took 10m 27s."
      ],
      "metadata": {
        "id": "7ruobbiXZO9_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with concurrent.futures.ThreadPoolExecutor(4) as executor:\n",
        "     futures = [executor.submit(parse_pdf, pdf_url) for pdf_url in pmc_search_results_no_full_text.pdf_url]\n",
        "     concurrent.futures.wait(futures)"
      ],
      "metadata": {
        "id": "xH_Tg1vjlNVe"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create a dictionary of futures."
      ],
      "metadata": {
        "id": "RM0T_Efb86qe"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "futures_map = dict(zip(pmc_search_results_no_full_text.pdf_url, futures))"
      ],
      "metadata": {
        "id": "cQqYkULonx-n"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(futures_map)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zIVF31MMn0Kb",
        "outputId": "f7d2994b-8a55-4cb8-86aa-1cb4a8418630"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "118"
            ]
          },
          "metadata": {},
          "execution_count": 8
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Create dictionary of exceptions with PDF URL as key and exception error message as value."
      ],
      "metadata": {
        "id": "yD5LAHuZ9eUc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "exceptions = {url: f.exception() for url, f in futures_map.items() if f.exception() is not None}\n",
        "exceptions"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ITBkXsyvn4mi",
        "outputId": "6e9f67a4-8d21-4d93-c106-e54d1214690e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "{'https://europepmc.org/articles/PMC7492056?pdf=render': Exception('{\\n  \"message\":\"The upstream server is timing out\"\\n}')}"
            ]
          },
          "metadata": {},
          "execution_count": 9
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "One exception for article PMC7492056. We will amend the function to handle exceptions caused by the timeout exception, and any server errors, and run again to download the full text."
      ],
      "metadata": {
        "id": "nyGf1XE39r41"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def parse_pdf(pdf_url: str):\n",
        "    \"\"\"\n",
        "    Parse PDF to XML using GROBID tool\n",
        "\n",
        "    :param pdf_url: str, URL to article PDF\n",
        "\n",
        "    :return: XML of parsed article\n",
        "    \"\"\"\n",
        "    # GROBID URL for the cloud service to parse full text of the article\n",
        "    url = \"https://cloud.science-miner.com/grobid/api/processFulltextDocument\"\n",
        "\n",
        "    if isinstance(pdf_url, str):\n",
        "          page = urllib.request.urlopen(pdf_url).read()\n",
        "          resp = requests.post(url, files={\"input\": page})\n",
        "          if resp.status_code != 200:\n",
        "              if resp.status_code >= 500:\n",
        "                retry = 1\n",
        "              else:\n",
        "                  return \"500\"\n",
        "              if resp.text in ['{\\n  \"message\":\"The upstream server is timing out\"\\n}']:\n",
        "                  return \"0\"\n",
        "              else:\n",
        "                  raise Exception(resp.text)\n",
        "          parsed_article = resp.text\n",
        "          time.sleep(3)\n",
        "    else:\n",
        "        raise TypeError(\"Need to supply a url\")\n",
        "\n",
        "\n",
        "    return parsed_article"
      ],
      "metadata": {
        "id": "Aew0Crl9pzhU"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "with ThreadPool(4) as pool:\n",
        "  exceptions_xml = pool.map(parse_pdf, pmc_search_results_no_full_text.pdf_url)"
      ],
      "metadata": {
        "id": "ocUkgyK0oSG9"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(exceptions_xml)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9fzPHit3tp7m",
        "outputId": "370a02f6-6b14-4399-856d-1ec0e94810c0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "118"
            ]
          },
          "metadata": {},
          "execution_count": 23
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "No further exceptions this time. Print out first article to view GROBID XML output."
      ],
      "metadata": {
        "id": "TzEM4LwGdnsX"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "exceptions_xml[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 174
        },
        "id": "loAtdx1xw32h",
        "outputId": "35cbfb1b-37c5-4cba-b603-f9e7e1b0f57c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'<?xml version=\"1.0\" encoding=\"UTF-8\"?>\\n<TEI xml:space=\"preserve\" xmlns=\"http://www.tei-c.org/ns/1.0\" \\nxmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" \\nxsi:schemaLocation=\"http://www.tei-c.org/ns/1.0 https://raw.githubusercontent.com/kermitt2/grobid/master/grobid-home/schemas/xsd/Grobid.xsd\"\\n xmlns:xlink=\"http://www.w3.org/1999/xlink\">\\n\\t<teiHeader xml:lang=\"en\">\\n\\t\\t<fileDesc>\\n\\t\\t\\t<titleStmt>\\n\\t\\t\\t\\t<title level=\"a\" type=\"main\">Recent Drug Development and Medicinal Chemistry Approaches for the Treatment of SARS-CoV-2 and Covid-19</title>\\n\\t\\t\\t</titleStmt>\\n\\t\\t\\t<publicationStmt>\\n\\t\\t\\t\\t<publisher/>\\n\\t\\t\\t\\t<availability status=\"unknown\"><licence/></availability>\\n\\t\\t\\t</publicationStmt>\\n\\t\\t\\t<sourceDesc>\\n\\t\\t\\t\\t<biblStruct>\\n\\t\\t\\t\\t\\t<analytic>\\n\\t\\t\\t\\t\\t\\t<author role=\"corresp\">\\n\\t\\t\\t\\t\\t\\t\\t<persName><roleName>Professor</roleName><forename type=\"first\">Arun</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t\\t\\t\\t\\t\\t<email>akghosh@purdue.edu</email>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><forename type=\"first\">Jennifer</forename><forename type=\"middle\">L</forename><surname>Mishevich</surname></persName>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><roleName>Professor</roleName><forename type=\"first\">Andrew</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><roleName>Dr</roleName><forename type=\"first\">Hiroaki</forename><surname>Mitsuya</surname></persName>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<persName><roleName>Ms. J</roleName><forename type=\"middle\">J</forename><surname>Mishevich</surname></persName>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff0\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Department of Chemistry</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Purdue University West Lafayette</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>47907</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>IN</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff1\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Department of Medicinal Chemistry and Molecular Pharmacolgy</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Purdue University West Lafayette</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>47907</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>IN</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff2\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Department of Biological Sciences</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Purdue University West Lafayette</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>47907</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>IN</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"US\">USA</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff3\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Department of Biochemistry</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Purdue University West Lafayette</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>47907</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>IN</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff4\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Departments of Hematology and Infectious Diseases</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Kumamoto University School of Medicine Kumamoto</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>860-8556</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"JP\">Japan</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff5\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Experimental Retrovirology Section HIV and AIDS Malignancy Branch National Cancer Institute Bethesda</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>20892 [g]</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>MD</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff6\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Center for Clinical Sciences National Center for Global Health and Medicine Shinjuku</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<postCode>162-8655</postCode>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Tokyo</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"JP\">Japan</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff7\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Calcutta and Indian Institute of Technology</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Kanpur</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"IN\">India</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff8\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">University of Illinois</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Chicago</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff9\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Department of Chemistry</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Medicinal Chemistry at Purdue University</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<addrLine>West Lafayette. In 2009</addrLine>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff10\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Indiana University</orgName>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff11\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">the University of Illinois-Chicago as</orgName>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff12\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">National Kumamoto University School of Medicine</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<country key=\"JP\">Japan</country>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff13\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Experimental Retrovirology Section</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">National Cancer Institute</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<address>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<addrLine>Maryland since 1991. Since 1997</addrLine>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<settlement>Bethesda</settlement>\\n\\t\\t\\t\\t\\t\\t\\t\\t\\t<region>Dr</region>\\n\\t\\t\\t\\t\\t\\t\\t\\t</address>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff14\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep1\">Medicine and Chairman</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\" key=\"dep2\">Departments of Hematology</orgName>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<author>\\n\\t\\t\\t\\t\\t\\t\\t<affiliation key=\"aff15\">\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"department\">Clinical Immunology/Rheumatology, and Infectious Diseases</orgName>\\n\\t\\t\\t\\t\\t\\t\\t\\t<orgName type=\"institution\">Kumamoto University School</orgName>\\n\\t\\t\\t\\t\\t\\t\\t</affiliation>\\n\\t\\t\\t\\t\\t\\t</author>\\n\\t\\t\\t\\t\\t\\t<title level=\"a\" type=\"main\">Recent Drug Development and Medicinal Chemistry Approaches for the Treatment of SARS-CoV-2 and Covid-19</title>\\n\\t\\t\\t\\t\\t</analytic>\\n\\t\\t\\t\\t\\t<monogr>\\n\\t\\t\\t\\t\\t\\t<imprint>\\n\\t\\t\\t\\t\\t\\t\\t<date/>\\n\\t\\t\\t\\t\\t\\t</imprint>\\n\\t\\t\\t\\t\\t</monogr>\\n\\t\\t\\t\\t\\t<idno type=\"MD5\">EB7F531E2618E36D20DDF293BFDB2E44</idno>\\n\\t\\t\\t\\t\\t<idno type=\"DOI\">10.1002/cmdc.202200440</idno>\\n\\t\\t\\t\\t\\t<note type=\"submission\">received his B.S. in Chemistry from Purdue University in 1988 and his Ph.D. in Biochemistry in 1995 from the University of Notre Dame. He pursued post-doctoral research at UC. Berkeley (1995-1999</note>\\n\\t\\t\\t\\t</biblStruct>\\n\\t\\t\\t</sourceDesc>\\n\\t\\t</fileDesc>\\n\\t\\t<encodingDesc>\\n\\t\\t\\t<appInfo>\\n\\t\\t\\t\\t<application version=\"0.7.3-SNAPSHOT\" ident=\"GROBID\" when=\"2023-01-09T22:49+0000\">\\n\\t\\t\\t\\t\\t<desc>GROBID - A machine learning software for extracting information from scholarly documents</desc>\\n\\t\\t\\t\\t\\t<ref target=\"https://github.com/kermitt2/grobid\"/>\\n\\t\\t\\t\\t</application>\\n\\t\\t\\t</appInfo>\\n\\t\\t</encodingDesc>\\n\\t\\t<profileDesc>\\n\\t\\t\\t<abstract>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"> COVID-19<p>, caused by SARS-CoV-2 infection, continues to be a major public health crisis around the globe. Development of vaccines and the first cluster of antiviral drugs has brought promise and hope for prevention and treatment of severe coronavirus disease. However, continued development of newer, safer, and more effective antiviral drugs are critically important to combat COVID-19 and counter the looming pathogenic variants. Studies of the coronavirus life cycle revealed several important biochemical targets for drug development. In the present review, we focus on recent drug design and medicinal chemistry efforts on small molecule drug discovery including the development of nirmatrelvir that targets viral protein synthesis and remdesivir and molnupiravir that target viral RdRp. These are recent FDA approved drugs for the treatment of COVID-19.</p></div>\\n\\t\\t\\t</abstract>\\n\\t\\t</profileDesc>\\n\\t</teiHeader>\\n\\t<text xml:lang=\"en\">\\n\\t\\t<body>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"1.\">Introduction</head><p>Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) originated in Wuhan, China in late December 2019. <ref type=\"bibr\" target=\"#b0\">1,</ref><ref type=\"bibr\" target=\"#b2\">2</ref> This outbreak began spreading at an alarming rate, and unleashed a severe health crisis around the globe. Subsequently, the uncertainties caused a serious economic meltdown worldwide. On March 11, 2020, the World Health Organization (WHO) declared the novel coronavirus (COVID-19) outbreak a global pandemic. <ref type=\"bibr\" target=\"#b3\">3,</ref><ref type=\"bibr\" target=\"#b4\">4</ref> Since then, it has gone on to affect millions of lives across the globe and caused nearly 6.3 million deaths as of June 7, 2022. <ref type=\"bibr\" target=\"#b5\">5</ref> Human-human transmission for SARS-CoVs occurs primarily via respiratory droplets through sneezing, coughing, or close contact between persons. Mild symptomatic cases may include: fever, fatigue, dyspnea. <ref type=\"bibr\" target=\"#b6\">6</ref> More severe cases of SARS-CoV-2 develop pneumonia, acute respiratory distress, and hypoxia. <ref type=\"bibr\" target=\"#b6\">6</ref> Early on, many laboratories around the world got involved in the development of COVID-19 therapeutics. These include, development of therapies through drug repurposing, monoclonal antibody-based treatment, convalescent plasma therapies, vaccine therapies, and target-based drug development. <ref type=\"bibr\" target=\"#b7\">7,</ref><ref type=\"bibr\" target=\"#b8\">8</ref> We will focus our review on recent efforts and updates on new drug development since our prior review in 2020. <ref type=\"bibr\" target=\"#b9\">9</ref> Coronaviruses are the largest known single stranded RNA viruses that infect animals and humans leading to respiratory, gastrointestinal, hepatic, and neurologic diseases. <ref type=\"bibr\" target=\"#b10\">10</ref> There are currently four different genera of human coronaviruses (HCoVs): alpha-coronavirus, beta-coronavirus, gamma-coronavirus, and delta-coronavirus. SARS-CoV-2 is the seventh human coronavirus. Another six have been identified: alpha-CoVs-NL63 and HCoVs-229E, and beta-CoVs HCoVs-OC43, HCoVs-HKU1, severe acute respiratory syndrome-CoV (SARS-CoV-1), and Middle East respiratory syndrome-CoV (MERS-CoV). These have been reviewed extensively. <ref type=\"bibr\">11</ref> Recently, a newly identified alphacoronavirus was identified in patients hospitalized with pneumonia. <ref type=\"bibr\">12</ref> This new human coronavirus turns out to be a novel \"canine-feline coronavirus\" that is called CCoV-human pneumonia (HuPn)-2018, and additional studies have now turned up even more of these hybrid alpha-coronaviruses around the globe. <ref type=\"bibr\">12</ref> Sequence analysis suggests that these are novel caninefeline-porcine-like (CFPL) CoVs of Alphacoronavirus 1 species that can infect humans (hCFPL-CoVs) and are associated with acute respiratory illness.</p><p>The first SARS outbreak (SARS-CoV-1) originated in the Guandong Province, China in 2002. <ref type=\"bibr\" target=\"#b13\">13,</ref><ref type=\"bibr\" target=\"#b14\">14</ref> It spread to several Asian countries, North America, and Europe. However, it was promptly contained and affected more than 8,000 individuals with 774 deaths and a 10% mortality rate. <ref type=\"bibr\" target=\"#b15\">15,</ref><ref type=\"bibr\" target=\"#b16\">16</ref> There is a 82% genome similarity between SARS-CoV-1 and SARS-CoV-2 and a 90% resemblance in many essential enzymes. With the arrival of SARS-CoV-1, there was significant medicinal chemistry efforts geared toward developing concepts and strategies to block SARS-CoV-1 replication and to develop target-based therapeutics. Indeed, much of the ground work for blocking essential enzyme functions provided insights and a head start for the development of mechanism-based drugs against SARS-CoV-2 infection and COVID-19. <ref type=\"bibr\" target=\"#b17\">17,</ref><ref type=\"bibr\" target=\"#b18\">18,</ref><ref type=\"bibr\" target=\"#b19\">19,</ref><ref type=\"bibr\" target=\"#b20\">20</ref> We will highlight here recent developments and approaches to SARS-CoV-2 and Covid-19 therapeutics.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"2.\">Drug design Targets that block Viral Replication</head><p>SARS-CoV-2 binds to the host cell surface receptor, angiotensin-converting enzyme 2 (ACE2) to enter the cell. After cell entry, viral RNA attaches to the host cell ribosome to produce two polyproteins, pp1a and pp1ab, which are subsequently cleaved by the coronavirus main protease, or 3-chymotrypsin-like protease (3CLpro), and papain-like protease (PLpro). <ref type=\"bibr\" target=\"#b21\">21,</ref><ref type=\"bibr\" target=\"#b22\">22</ref> Cleavage of these polyproteins leads to the formation of several essential enzymes such as RNA-dependent RNA polymerase (RdRp), which is responsible for genome replication. <ref type=\"bibr\" target=\"#b23\">23</ref> As shown in Figure <ref type=\"figure\" target=\"#fig_0\">1</ref>, many of these enzymes have all been subjected to intensive research as targets for SARS-CoV antiviral therapies.</p><p>Two main targets for antiviral therapies are two cysteine proteases, PLpro and 3CLpro, also known as the main protease (Mpro). <ref type=\"bibr\" target=\"#b25\">24,</ref><ref type=\"bibr\" target=\"#b26\">25</ref> Both proteases are responsible for processing 16 nonstructural proteins (nsps), which are essential for viral replication and maturation. <ref type=\"bibr\" target=\"#b27\">26,</ref><ref type=\"bibr\" target=\"#b28\">27,</ref><ref type=\"bibr\" target=\"#b29\">28</ref> The sequence identity for 3CLpro between SARS-CoV-1 and SARS-CoV-2 is 96%. 3CLpro is responsible for cleaving the polyprotein and generating functional viral proteins such as: RdRp, RNA binding proteins, exoribonuclease, helicase, and methyltransferase. <ref type=\"bibr\" target=\"#b30\">29</ref> 3CLpro is active as a homodimer and contains a catalytic dyad of Cys-His, whereas PLpro has a catalytic triad of Cys-His-Asp. <ref type=\"bibr\" target=\"#b31\">30</ref> PLpro has an 83% sequence identity similarity between SARS-CoV-1 and SARS-CoV-2. PLpro is responsible for processing the replicase polyprotein and removing cellular substrates such as ubiquitin, termed deubiquitylation, and interferon-stimulated gene product 15 from host cell proteins. <ref type=\"bibr\" target=\"#b32\">31,</ref><ref type=\"bibr\" target=\"#b33\">32</ref> This allows the virus to escape the host innate immune system. Other important drug design targets for COVID-19 treatment include, the spike/ACE2 viral attachment, cell entry, viral helicase, replication complex, nucleocapsid, viral RdRp, and methyltransferase. Many of these areas are undergoing active research and we will highlight recent developments. </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"3.\">Currently Approved Drugs for treatment of COVID-19</head><p>Since the first SARS-CoV-1 outbreak in 2003 and later with the MERS-CoV outbreak, much effort was devoted to drug repurposing. <ref type=\"bibr\" target=\"#b34\">33,</ref><ref type=\"bibr\" target=\"#b35\">34</ref> The outbreak of COVID-19 led to significant activity in this area due to the overwhelming situation and urgency for treatment. A major advantage of drug repurposing is that the drug is already approved for use in one area, therefore drug safety and pharmacokinetic properties are well established. As a result, clinical trials can be conducted directly in patients, leading to accelerated approval if efficacy is observed. Many approved drugs have been reported to block SARS-CoV-2 replication. <ref type=\"bibr\" target=\"#b7\">7,</ref><ref type=\"bibr\" target=\"#b8\">8</ref> Several of these drugs have undergone clinical trials for potential treatment of SARS-CoV-2 infection and COVID-19. These include, remdesivir (1, Figure <ref type=\"figure\" target=\"#fig_1\">2</ref>) which was initially developed for treatment of hepatitis C infection, chloroquine and hydroxylchloroquine which were developed for treatment of malaria, favipiravir which was developed as an anti-influenza drug, and masitinib which was developed as a kinase inhibitor against mast cell tumors in animals. Among these, only remdesivir (1), developed by Gilead Sciences, was initially approved by the FDA for treatment of COVID-19 patients. <ref type=\"bibr\" target=\"#b36\">35,</ref><ref type=\"bibr\" target=\"#b37\">36</ref> Remdesivir\\'s biological mechanism of action involves interfering with genome replication by targeting RdRp. <ref type=\"bibr\" target=\"#b38\">37,</ref><ref type=\"bibr\" target=\"#b39\">38</ref> Structurally, remdesivir resembles adenosine and gets incorporated into nascent viral RNA, which causes premature termination of the viral RNA chain. Remdesivir was approved as an intravenous drug that must be administered in a health care setting. However, clinical efficacy of remdesivir has remained inconclusive. <ref type=\"bibr\" target=\"#b40\">39,</ref><ref type=\"bibr\" target=\"#b41\">40</ref> One of the reasons for this controversy is that the clinical trial was conducted on severe COVID-19 patients and antiviral drugs may not have been useful at that stage of infection. A recent clinical trial of remdesivir for treatment of mild COVID-19 patients showed higher efficacy. <ref type=\"bibr\" target=\"#b42\">41,</ref><ref type=\"bibr\" target=\"#b43\">42</ref> It is now generally accepted that antiviral drugs for COVID-19 treatment should be administered in the earliest viral replication phase. Molnupiravir (FIDD-2801), a nucleoside analog, developed by Merck Research Laboratories, was recently given emergency use authorization (EUA) by the US FDA in 2021 as an oral drug. <ref type=\"bibr\" target=\"#b44\">43,</ref><ref type=\"bibr\" target=\"#b45\">44</ref> Molnupiravir has the same mechanism as remdesivir. It is an RdRp inhibitor, originally developed for treatment of viral infection caused by RNA viruses. One of the main advantages of oral drugs is that the patients can be prescribed antiviral drugs at the earliest indication without hospitalization. Early phase 3 clinical trials with COVID-19 patients showed about 50% reduction in hospitalization compared to the placebo group with no reported deaths. <ref type=\"bibr\" target=\"#b45\">44,</ref><ref type=\"bibr\" target=\"#b46\">45</ref> Paxlovid (PF-07321332), developed by Pfizer, was recently granted EUA by the FDA in 2021 as an oral drug for treatment of COVID-19 patients. <ref type=\"bibr\" target=\"#b47\">46,</ref><ref type=\"bibr\" target=\"#b48\">47</ref> The biological mechanism of action involves inhibition of viral protease (3CLpro or Mpro) which plays a critical role in the viral replication cycle by cleaving viral proteins and producing essential individual, mature proteins necessary for viral replication. <ref type=\"bibr\" target=\"#b25\">24,</ref><ref type=\"bibr\" target=\"#b26\">25</ref> Many inhibitors of 3CLpro were initially developed for treatment of SARS-CoV-1 during the outbreak in 2003-2004. <ref type=\"bibr\" target=\"#b9\">9,</ref><ref type=\"bibr\" target=\"#b26\">25,</ref><ref type=\"bibr\" target=\"#b49\">48</ref> The Phase II-III data of Paxlovid clinical trials showed that treatment of patients within three days of COVID-19 symptoms reduced COVID related hospitalization by 89% compared to the placebo group. <ref type=\"bibr\" target=\"#b47\">46,</ref><ref type=\"bibr\" target=\"#b50\">49</ref> There are many other drug discovery targets being actively investigated for development of effective COVID-19 therapeutics. The major targets and progress in these areas will be discussed in the following section.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"4.\">SARS-CoV-2 Protease Inhibitors for COVID-19 treatment</head><p>critical roles in the treatment of chronic viral infections, such as HIV/AIDS, hepatitis C virus, hepatitis B virus, herpesvirus and influenza virus. <ref type=\"bibr\" target=\"#b51\">50,</ref><ref type=\"bibr\" target=\"#b52\">51,</ref><ref type=\"bibr\" target=\"#b53\">52</ref> Since the SARS-CoV-1 outbreak in 2003 and MERS-CoV outbreak in 2012, SARS-CoV-1 3CLpro and SARS-CoV-1 PLpro became very attractive targets for drug discovery and early development of peptidomimetic and small molecule SARS-CoV-1 protease inhibitors. <ref type=\"bibr\" target=\"#b9\">9,</ref><ref type=\"bibr\" target=\"#b26\">25,</ref><ref type=\"bibr\" target=\"#b49\">48</ref> The in vivo efficacy of 3CLpro inhibitors in mice and feline species has been documented prior to the development of paxlovid. <ref type=\"bibr\" target=\"#b54\">53,</ref><ref type=\"bibr\" target=\"#b55\">54</ref> As the activity of 3CLpro and PLpro are essential for coronavirus replication, an intense effort towards protease inhibitor design and development has commenced with the COVID-19 pandemic. We will discuss recent updates of SARS-CoV-2 3CLpro and PLpro inhibition design in this section.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"4.1\">SARS-CoV-2 3CLpro Inhibitors</head><p>Of the two main proteases, SARS-CoV-2 3CLpro is the most-well characterized. <ref type=\"bibr\" target=\"#b56\">55,</ref><ref type=\"bibr\" target=\"#b57\">56</ref> Several different types of inhibitors have been investigated including peptidomimetic covalent inhibitors, small molecule covalent inhibitors, and noncovalent inhibitors. <ref type=\"bibr\" target=\"#b9\">9,</ref><ref type=\"bibr\" target=\"#b58\">57</ref> Historically, covalent inhibitors were thought to be generally cytotoxic due to off target effects; however, recent developments have shown that this is not always the case. Development of covalent cysteine proteases has shown significant potential for SARS-CoV-2 drug development. Several different classes of protease inhibitors will be discussed in this section including peptidomimetic covalent inhibitors, small molecule covalent inhibitors, and noncovalent inhibitors.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"4.1.1\">Peptidomimetic Covalent 3CLpro Inhibitors</head><p>The development of early peptidomimetic SARS-CoV-1 3CLpro inhibitors has been reviewed previously. <ref type=\"bibr\" target=\"#b9\">9,</ref><ref type=\"bibr\" target=\"#b49\">48,</ref><ref type=\"bibr\" target=\"#b58\">57</ref> Structural evolution of 3CLpro inhibitors started with the design of covalent inhibitor 4 (Figure <ref type=\"figure\" target=\"#fig_2\">3</ref>) with an α,β-unsaturated ester, Michael acceptor warhead. <ref type=\"bibr\" target=\"#b59\">58,</ref><ref type=\"bibr\" target=\"#b60\">59</ref> Inhibitor design also evolved to reversible inhibitors like the phthalhydrazido-methyl ketone warhead in inhibitor 5, and benzothiazolyl ketone-derived inhibitors 6 and 7. <ref type=\"bibr\" target=\"#b26\">25,</ref><ref type=\"bibr\" target=\"#b61\">60</ref> Inhibitor 7 with a P3 indole methoxy group was synthesized and evaluated against SARS-CoV-2 3CLpro. <ref type=\"bibr\" target=\"#b62\">61</ref> Compound 7 is a tight binding reversible covalent inhibitor and showed a Ki value of 17.6 nM against SARS-CoV-2 3CLpro. The compound displayed an antiviral EC50 value of 4.2 µM in VeroE6 cell-based assay with RNA-qPCR. Its apparent cytotoxic CC50 value was &gt;100 µM. Interestingly, inhibitor 7 when combined with remdesivir, an RdRp inhibitor, exerted synergistic activity against SARS-CoV-2 and viral breakthrough did not occur. At 2 µM remdesivir and 2 µM inhibitor 7 viral replication was suppressed by 0.67-fold and 1.3-fold, respectively. When both compounds were combined, the suppression was 1.8-fold. At 10 µM remdesivir and 10 µM inhibitor 7 results showed viral suppression to be 20-fold and 210-fold, respectively. However, when combined, that suppression was 590,000-fold. At 20 µM remdesivir and 20 µM inhibitor 7 viral suppression was 1,600,000-fold. <ref type=\"bibr\" target=\"#b62\">61</ref> The X-ray structure of SARS-CoV-2 3CLpro and inhibitor 7 complex was determined and the structure gave molecular insight into the inhibitor-protease interactions. The active site inhibitorbound structure is shown in Figure <ref type=\"figure\" target=\"#fig_3\">4</ref>. As can be seen, the sulfur atom of Cys145 forms a covalent bond with the carbonyl carbon next to the benzothiazole of compound 7, resulting in the formation of a tetrahedral intermediate. The hemiacetal alcohol forms one direct hydrogen bond and water-mediated hydrogen bond interactions around the oxyanion hole residues, Cys145 and Gly143. The P3 indole amide carbonyl  Bisulfite derivative, GC-376 (8, Figure <ref type=\"figure\" target=\"#fig_4\">5</ref>) has been shown to potently inhibit SARS-CoV-2 3CLpro. <ref type=\"bibr\" target=\"#b63\">62,</ref><ref type=\"bibr\" target=\"#b64\">63</ref> GC-373 (9) was initially developed to inhibit feline infectious peritonitis virus (FIVP), a fatal infection in cats caused by mutations of feline enteric coronavirus (FECV). <ref type=\"bibr\" target=\"#b54\">53</ref> GC-376 is the prodrug of aldehyde inhibitor GC-373, and it was shown to exhibit in vivo efficacy in treating certain forms of FIVP. <ref type=\"bibr\" target=\"#b54\">53</ref> With the outbreak of COVID-19, there has been much research into GC-373 and its derivatives. Vuong et. al. and others have shown that GC-373 inhibits SARS-CoV-2 3CLpro with a Ki value of 150 nM and antiviral activity of 700 nM. <ref type=\"bibr\" target=\"#b65\">64,</ref><ref type=\"bibr\" target=\"#b66\">65</ref> Structural modifications were carried out to further improve activity. Derivatives 10 and 11 were shown to inhibit SARS-CoV-2 3CLpro more potently, however, their antiviral activity was not reported. 66 The X-ray structure of GC-376-bound SARS-CoV-2 3CLpro has been determined. As shown in Figure <ref type=\"figure\" target=\"#fig_5\">6</ref>, the structure revealed that Cys145 forms a covalent bond with the aldehyde functionality and forms a hemithioacetal. <ref type=\"bibr\" target=\"#b66\">65,</ref><ref type=\"bibr\" target=\"#b67\">66</ref> The P1 lactam carbonyl forms a hydrogen bond with His163 and the lactam NH forms a hydrogen bond with the side chain carboxylic acid of Glu166. The P2 isobutyl group fits in the S2 pocket surrounding Arg40, His41, and Asp187.</p><p>Repurposing of previously approved protease inhibitors for the treatment of COVID-19 has been conducted in many laboratories. <ref type=\"bibr\" target=\"#b7\">7,</ref><ref type=\"bibr\" target=\"#b8\">8</ref> This led to the identification of many drug candidates. However, there was wide variation in assay results. Among the most promising outcome is the identification of HCV NSP3 protease inhibitor drugs boceprevir 12 (Figure <ref type=\"figure\" target=\"#fig_6\">7</ref>) and telaprevir 13. <ref type=\"bibr\" target=\"#b66\">65,</ref><ref type=\"bibr\" target=\"#b68\">67</ref> Both drugs are potent peptidyl inhibitors of HCV NSP3, which is a serine protease. <ref type=\"bibr\" target=\"#b69\">68,</ref><ref type=\"bibr\" target=\"#b70\">69</ref> Both compounds have been shown to inhibit SARS-CoV-2 Mpro from several labs. <ref type=\"bibr\" target=\"#b66\">65,</ref><ref type=\"bibr\" target=\"#b68\">67</ref> However, inhibition data show wide variability. Like other peptidomimetic inhibitors of 3CLpro, these serine protease inhibitor drugs possess the same α-ketoamide warhead. Like the ketobenzothiazole warhead in compound 7, the ketoamide functionality forms a covalent bond with Cys145 in the active site. Boceprevir, 12, has inhibited SARS-CoV-2 3CLpro with an IC50 range of 1.6 µM to 8 µM. In contrast, teleprevir showed weak inhibition with an IC50 value of 56 µM. Boceprevir antiviral activity EC50 value was determined to be 15.6 µM with remdesivir as a positive control and an EC50 value of 0.58 µM. Teleprevir showed marginal antiviral activity. <ref type=\"bibr\" target=\"#b66\">65,</ref><ref type=\"bibr\" target=\"#b68\">67,</ref><ref type=\"bibr\">70</ref> Structurally, boceprevir possesses a P1β-cyclobutylalanyl group and teleprevir contains a norleucine moiety. Other key features include a P2 dimethylcyclopropyl proline for boceprevir and a bicyclic cyclopentyl fused proline derivative for teleprevir. Both drugs contain a P3-tert-butyl glycine ligand. The first X-ray crystal structure of SARS-CoV-2 3CLpro bound to boceprevir was determined by our group early in the COVID-19 pandemic and was released via the PDB (6WNP). <ref type=\"bibr\">70</ref> The X-ray crystal structures of boceprevir and teleprevir bound to SARS-CoV-2 3CLpro were also determined by others. <ref type=\"bibr\" target=\"#b66\">65,</ref><ref type=\"bibr\">70</ref> Structural analysis of boceprevir-bound 3CLpro is shown in Figure <ref type=\"figure\" target=\"#fig_8\">8</ref>. Cys145 forms a covalent bond with the ketoamide warhead and subsequently forms a thiohemiacetal moiety. The resulting hydroxyl group forms a hydrogen bond with the His41 side chain and stabilizes the oxyanion hole. The P1 cyclobutylmethyl group occupies the shallow S1 pocket. The backbone of His164 and Glu166 forms important hydrogen bonds with the main chain of boceprevir. The X-ray structure of boceprevir-bound and teleprevir-bound 3CLpro are being utilized for improving drug-like properties, structures and antiviral activity against SARS-CoV-2 3CLpro. These are highlighted in the following section. Based upon boceprevir or telaprevir-bound X-ray structures, a number of bicycloproline derived inhibitors have been prepared and evaluated by Yang and collaborators. <ref type=\"bibr\" target=\"#b73\">71</ref> As shown in Figure <ref type=\"figure\" target=\"#fig_9\">9</ref>, previously optimized, the P1 lactam was maintained as a fixed substituent in compound 14. The P2 proline derivative of boceprevir and the bicyclic telaprevir ligand have been introduced in combination with a small electrophilic aldehyde warhead. This P1 moiety is the active pharmacophore, which forms a covalent bond with Cys145. Unfortunately, the extremely reactive nature of the aldehyde raises some clinical safety concerns for these inhibitors. The 𝛾-lactam mimics the native P1 glutamine structure and fills the S1 subsite, and hydrophobic, medium sized subgroups were utilized in the P3 position to enhance the potency and pharmacokinetic properties. <ref type=\"bibr\" target=\"#b73\">71</ref> All compounds were tested for their biological activity against SARS-CoV-2 3CLpro in a FRET assay. IC50 values were found to be in the low nanomolar range (from 7.6 to 748 nM), for compounds 14-16.</p><p>The crystal structure of 16-bound to SARS-CoV-2 3CLpro was determined. As shown in Figure <ref type=\"figure\" target=\"#fig_10\">10</ref>, a covalent bond is formed between the aldehyde carbonyl carbon and the sulfur on Cys145. Two hydrogen bonds between the aldehyde carbonyl oxygen and the backbone amides of Cys145 and Gly143 can be observed. The bicycloproline extends into the S2 pocket where it forms hydrophobic interactions with the pocket residues. The P3 portion interacts with the amide backbone of Glu166 and the extended conformation of ethyl-3,5-difluorobenzene side chain extends into the S4 subsite.</p><p>Several compounds showed nanomolar or low micromolar EC50 values (range between 0.53 and 30.5 𝜇M). <ref type=\"bibr\" target=\"#b73\">71</ref> Interestingly, some compounds with high potency in the enzymatic assay exhibited only marginal activity in the antiviral assay. This was  proposed to be caused by low lipophilic properties of certain P3 ligands, resulting in poor cell membrane permeability. This class of compounds displayed no cytotoxicity in any of the cell lines tested including, Vero E6, HPAEpiC, LO2, BEAS-2B, A549, and Huh7 cells. Preliminary pharmacokinetic property experiments were performed to determine which compounds would be good for in vivo antiviral studies. Compounds 14 and 15 showed oral bioavailabilities of 11.2% and 14.6%, respectively. <ref type=\"bibr\" target=\"#b73\">71</ref> In vivo tests were performed on mice in which both 14 and 15 were utilized, and it was found that the treatment group contained lower mean viral RNA loads in lung tissues than those of the control group. Recent investigations by Liu and co-workers into boceprevir based derivatives determined that a P4 N-terminal carbamate improves activity and that an O-tert-butyl-threonine P3 ligand allows compounds to achieve higher cellular and antiviral activity. <ref type=\"bibr\">72</ref> Other recent investigations into 𝛼-ketoamide warhead containing 3CLpro inhibitors have culminated in lead compounds with good activity and oral bioavailability against several SARS-CoV-2 variants. <ref type=\"bibr\" target=\"#b79\">73</ref> Based upon similar studies mentioned above, Xia and coworkers reported peptidomimetic covalent inhibitors 17 and 18 shown in Figure <ref type=\"figure\" target=\"#fig_12\">11</ref>. <ref type=\"bibr\" target=\"#b84\">74</ref> Inhibitor 17 is a hybrid of GC-376 ( <ref type=\"formula\">8</ref>) and telaprevir and compound 18 is a hybrid of 8 and boceprevir. These inhibitors displayed potent enzymatic inhibition against SARS-CoV-2 3CLpro as well as several other coronaviruses. In a FRET assay both inhibitors had similar potency to 8, although 18 was the more potent compound based on their in house designed Flip-GFP Mpro assay. <ref type=\"bibr\" target=\"#b84\">74</ref> Antiviral activities were determined using immunofluorescence assay in VeroE6 and Caco2-ACE2 cell lines. Compound 18 showed an EC50 value of 0.37 µM in Vero E6 cells. Inhibitor 18 also had antiviral activity in Caco2-ACE2 cells with an EC50 of 1.06 µM compared to inhibitor 17 at 5.24 µM, and lead compound 8 at 2.9 µM. Both derivatives inhibited 3CLpro in other coronaviruses such as, SARS-CoV, MERS-CoV, HCoV-OC43, HCoV-NL63, and HCoV-229E in the FRET-based enzymatic assay. Inhibitor 17 had improved selectivity compared to the control compound against host cysteine proteases calpain I and cathepsin L. <ref type=\"bibr\" target=\"#b84\">74</ref> Selectivity was tested for each compound against cathepsin L, calpain I, cathepsin K, caspase-3, and serine protease trypsin. It was found that 8 was a potent inhibitor of calpain I and cathepsin L. Inhibitors 17 and 18 displayed significantly reduced inhibition of the off-target enzyme. <ref type=\"bibr\" target=\"#b84\">74</ref> Lead compound 8 was also a potent inhibitor for cathepsin K, and while both designed inhibitors had reduced inhibitory activity, they also were quite potent against the off-target enzyme. A crystal structure of inhibitor 18 bound SARS-CoV-2 3CLpro determined the active site interactions, including formation of covalent bond with Cys145 is shown in Figure <ref type=\"figure\" target=\"#fig_1\">12</ref>.</p><p>Bai and co-workers reported a series of peptidomimetic SARS-CoV-2 3CLpro inhibitors with 𝛼 -acyloxymethylketone lactam as the glutamine mimetic. <ref type=\"bibr\">75</ref> Representative inhibitors 19-21 are shown in Figure <ref type=\"figure\" target=\"#fig_13\">13</ref>. The design work stemmed from studies of Krantz and co-workers, who reported related work which involved design of inactivators of cysteine proteases like cathepsin B. <ref type=\"bibr\" target=\"#b91\">76,</ref><ref type=\"bibr\" target=\"#b92\">77</ref> In particular, this earlier work examined leaving groups at the α-position of an α-substituted methyl ketone. It was observed that the pKa of the leaving group correlated with the irreversible, covalent interaction. The choice of the α-leaving group is important as chloromethyl ketones are too reactive and cytotoxic. Fluoromethyl ketone is more selective, but it may also generate toxic metabolites. <ref type=\"bibr\" target=\"#b93\">78,</ref><ref type=\"bibr\" target=\"#b94\">79</ref> Krantz and co-workers previously utilized benzoate esters and a few aliphatic esters of hydroxymethyl ketones as inactivators of cysteine proteases. Results indicated 2,6-bistrifluoromethyl and 2,6-dichloro benzoates as potent irreversible inactivators. <ref type=\"bibr\" target=\"#b91\">76,</ref><ref type=\"bibr\" target=\"#b92\">77</ref> Based upon the results of diphenylphosphinyl, tretronoyl, and peptidyl esters, an α-acyloxy group was examined. This allows for pKa adjustments to balance the rapid irreversible adduct formation and excessive reactivity that sequester glutathione or cause cytotoxicity.</p><p>Heteroaromatic rings were incorporated to allow more polar surfaces to increase solubility and reduce molecular planarity based upon previous studies. Steric hinderance around the ester was introduced to slow hydrolysis and increase stability. The P2 4-methoxyindole and leucine were left unchanged throughout the SAR study. <ref type=\"bibr\">75</ref> The six-membered lactam was also explored compared to the standard 5-membered lactam for further investigation in the S1 pocket. This was found to not have a significant effect on inhibitory activity. Variation of the ester moiety to contain an electron-withdrawing group was also explored in order to reduce the pKa of the corresponding acid leaving group. This phenomenon was exemplified with inhibitor 19, which displayed an enzyme IC50 value of 86 nM. <ref type=\"bibr\">75</ref> Inhibitor 20 displayed very potent enzyme inhibitory activity with IC50 value of 1 nM. Substituted pyridinyl derivatives such as in compound 21 were less potent than pyridine alone, but more potent than 8.</p><p>Antiviral activity was observed at low micromolar range, including compounds 20 and 21 and no toxicity was observed up to 200 𝜇M. Inhibitor 19 had good potency and plasma stability. <ref type=\"bibr\">75</ref> These compounds show antiviral activity against other coronavirus strains. The X-ray crystal structure of the inhibitor 21bound to SARS-CoV-2 3CLpro was determined. As shown in Figure <ref type=\"figure\" target=\"#fig_14\">14</ref>, Cys145 formed a covalent linkage with the inhibitor and ester carbonyl turned into a thioether functionality. Hoffman and co-workers also investigated peptidomimetic covalent inhibitors with hydroxymethyl ketone and derivatives as warheads. <ref type=\"bibr\" target=\"#b95\">80</ref> This class of compounds were initially developed during the first SARS outbreak in 2003. Following the COVID-19 outbreak, investigators examined hydroxymethyl ketone 22 (Figure <ref type=\"figure\" target=\"#fig_4\">15</ref>) and its derivatives as potential therapeutics for COVID-19 treatment. The antiviral activity of compound 22 was evaluated against a panel of human viruses. Compound 22 does not inhibit human rhinovirus strains HRv-14 and HRv-16, HIV-1, HCMV in cell culture. Also, it does not inhibit HCV replication. The X-ray structural studies of 22-bound to the 3CLpro of SARS-CoV-1 and CoV-2 were determined. The ligand binding sites are nearly identical. Also, as expected, the hydroxymethyl ketone carbonyl carbon formed a covalent bond to the 3CLpro active site Cys145 and generated a tetrahedral carbinol complex as shown in Figure <ref type=\"figure\" target=\"#fig_5\">16</ref>  benzothiazol-2-yl ketone as the warhead resulted in compound 24, which showed reduction of both enzyme and antiviral activity.</p><p>Replacement of the methoxyindole P3 ligand with a (S)-valsulfonamide derivative resulted in compound 25 (Figure <ref type=\"figure\" target=\"#fig_17\">17</ref>) with improved activity and slight improvement in oral bioavailability (F=10%). Inhibitor 26 resulted from varying the sulfonamide P3capping group with a trifluoroacetamide group. This compound exhibited comparable enzymatic inhibitory activity to 25; however, its antiviral activity in VeroE6 cells improved nearly 10-fold. Furthermore, this replacement resulted in a significant improvement in the oral bioavailability in rats (10 mpk, F=33%). Introduction of a P1\\' nitrile and replacement of the (S)-val with a (S)-tert-leucine provided compound 2 with further improvement in activity as well as oral bioavailability in rats (10 mpk, F=50%). This compound became known as PF-07321332 (2). It exhibited low oral bioavailability in monkeys (10 mpk, F=8.5%) due to first-pass metabolism along the gastrointestinal tract by cytochrome P450 (CYP) enzymes. Subsequent experiments involving coadministration with CYP34A inactivator, ritonavir (RTU) significantly improved the plasma concentration of PF-07321332. 47 The X-ray co-crystal structure of inhibitor 2 with SARS-CoV-2 3CLpro was determined. As shown in Figure <ref type=\"figure\" target=\"#fig_18\">18</ref>, the P1\\' nitrile functionality of compound 2 forms a reversible covalent thioimidate adduct with the catalytic Cys145. The P1 lactam carbonyl forms a hydrogen bond with His163 and the lactam NH forms a hydrogen bond with the side chain carboxylic acid of Glu166 similar to other structures. A recent report by Kovalevsky and co-workers extensively explored the active site interactions of three in house compounds and inhibitor 2. <ref type=\"bibr\" target=\"#b96\">81</ref> Interestingly, it was reported that the active site of Mpro can, and does, undergo a significant amount of distortion to accommodate bulkier substituents, particularly in the S4 and S5 subsites. Nitrile warhead compounds, including 2, were examined utilizing room temperature X-ray crystallography. Results displayed the presence of the thioimidate adduct formation with Cys145 and subsequent N insertion into the oxyanion hole. Other important interactions are the unconventional interactions with the CF3 group. The small electronegative CF3 is capable of forming favorable F…O interactions, as shown in this study, or F…N interactions, as reported in another study <ref type=\"bibr\" target=\"#b97\">82</ref> , with the protease. These types of unconventional interactions have shown to be of significant importance in inhibitor activity, metabolic stability, and pharmacokinetics.</p><p>This oral SARS-CoV-2 3CLpro inhibitor (Figure <ref type=\"figure\" target=\"#fig_17\">17</ref>, inhibitor 2) and its entrance into clinical trials were unveiled by Pfizer in April 2021. <ref type=\"bibr\" target=\"#b98\">83</ref> As such, it is the first orally administered compound to enter clinical trials that targets the main protease of SARS-CoV-2. It can be taken orally as a pill or capsule, allowing for the freedom to be given outside of hospitals. Interestingly, the first 7 mg of the compound were synthesized in late July 2020. <ref type=\"bibr\" target=\"#b98\">83</ref> A massive scale up effort was undertaken and by late October 100 g of the compound was synthesized, and two weeks later the chemists were able to scale up the synthesis to even more than 1 kg. <ref type=\"bibr\" target=\"#b98\">83</ref> According to phase II and III clinical trial data, inhibitor 2 or nirmatrelvir, is extremely effective in reducing the risk of COVID-19 related hospitalization and death. Nirmatrelvir is marketed and sold under the name paxlovid, which is a combination of nirmatrelvir and ritonavir. As nirmatrelvir is metabolized by CYP3A4, ritonavir is necessary as a pharmacokinetic booster to increase the oral bioavailability of nirmatrelvir. <ref type=\"bibr\" target=\"#b99\">84</ref> In a randomized clinical trial with 1:1 paxlovid:placebo received orally every 12 hours for five days, it was found that those receiving paxlovid had significantly reduced hospital admissions and deaths among those affected by COVID-19. <ref type=\"bibr\" target=\"#b100\">85</ref> It was found that among participants who received treatment within three days of beginning COVID-19 symptoms, the risk of hospitalization or death was 89% lower than that for the placebo group. <ref type=\"bibr\" target=\"#b100\">85,</ref><ref type=\"bibr\" target=\"#b101\">86</ref> Phase II/III clinical trials for administration of paxlovid in pediatric patients between the ages of 6 and 12 who test positive for COVID-19 and are at risk for severe disease began in March of 2022. <ref type=\"bibr\" target=\"#b102\">87</ref> A recent report by Pfizer researchers and collaborators reports the activity of nirmatrelvir against several SARS-CoV-2 variants of concern and variants of interest. <ref type=\"bibr\" target=\"#b103\">88</ref> Nirmatrelvir results are compared to the FDA approved remdesivir as the control.  Results in Table <ref type=\"table\" target=\"#tab_0\">1</ref> indicate that nirmatrelvir does in fact potently inhibit the many variants. EUA was granted to Pfizer for paxlovid on December 22, 2021 for the treatment of mild-tomoderate COVID-19 in adults and pediatric patients 12 years of age or older weighing at least 40 kg who have tested positive for COVID-19 and who are at high risk for progression to severe disease including hospitalization or death. A recent study by Pan and co-workers reported the treatment of omicron infected calu-3 cells with varying doses of nirmatrelvir and observed potent inhibition of viral RNA. <ref type=\"bibr\" target=\"#b101\">86</ref> Calculated IC50 values against the wild type variant and omicron variant were reportedly 0.176 𝜇M and 0.0246 𝜇M, respectively. 86</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"4.1.2\">Small Molecule Covalent SARS-CoV-2 3CLpro Inhibitors</head><p>Indole-5-chloropyridinyl esters were designed and synthesized to inhibit SARS-CoV-1 3CLpro. <ref type=\"bibr\" target=\"#b62\">61,</ref><ref type=\"bibr\" target=\"#b104\">89</ref> Several derivatives have been shown to potently inhibit the 3CLpro enzyme by formation of a covalent bond with the catalytic Cys145 residue in the active site. Ghosh and co-workers further investigated the potential of this class of compounds as SARS-CoV-2 3CLpro inhibitors with the outbreak of COVID-19. <ref type=\"bibr\" target=\"#b104\">89,</ref><ref type=\"bibr\" target=\"#b105\">90</ref> A number of compounds with varying substituents and functionalities have been synthesized and evaluated. Structure-activity studies show that the position of the carboxylic acid on the indole ring is important for activity. Selected representative derivatives 27-31 are shown in Figure <ref type=\"figure\" target=\"#fig_9\">19</ref>. Among them, inhibitor 27 displayed a SARS-CoV-2 3CLpro inhibitory IC50 value of 250 nM. The compound also exhibited potent antiviral activity with an EC50 value of 2.8 µM in VeroE6 cells. Remdesivir, an RdRp inhibitor, showed an antiviral EC50 value of 1.2 µM in the same assay. Furthermore, compound 27 and remdesivir exhibited comparable antiviral activity in immunocytochemistry assays. <ref type=\"bibr\" target=\"#b62\">61</ref> The X-ray structures of compound 27-bound to SARS-CoV 3CLpro and SARS-CoV-2 3CLpro were determined. Both structures revealed that catalytic Cys145 formed a covalent bond with the indole ester carbonyl group of compound 27. The X-ray crystal structure of 27-bound to SARS-CoV-2 3CLpro is shown in Figure <ref type=\"figure\" target=\"#fig_1\">20</ref>. Interestingly, the indole ring forms π-π stacking interactions with the imidazole ring of the His41. Methyl substitution on the indole ring was also investigated to lock the bioactive conformation of the pyridyl ester. Incorporation of the methyl at position 3 resulted in significantly reduced enzyme and antiviral activity. However, incorporation of the methyl at position 6 resulted in compound 28. This compound exhibited comparable 3CLpro inhibitory and antiviral activity to compound 27. An inhibitor with the methyl at the 5 position of the indole displayed no antiviral activity. Varying the position of the chloropyridinyl ring to the 2 or 5 position of the indole led to a reduction in activity (compounds 29 and 30) compared to 27. The original study displayed this loss directly for the 5-indole chloropyridinyl ester, but inhibitor 29 was noted to be even less potent. This further loss of activity may be attributed to the large sulfonamide, which could cause unfavorable steric interactions in the active site. Substitution of the chloropyridinyl ester at the 4 position of the indole ring also showed a reduction in enzyme inhibition and a decrease in antiviral activity.</p><p>Substitutions on the indole ring and the chloropyridine ring were also examined. Addition of a 3-nitro sulfonamide on the compared to 27. However, addition of an N-allyl substituent at this position led to potent derivative 31 with a 3CLpro inhibitory IC50 of 73 nM. Although, antiviral EC50 was reduced to 15 𝜇M. An X-ray crystal structure of sulfonamide 29 bound to SARS-CoV-2 3CLpro was determined and the structure is shown in Figure <ref type=\"figure\" target=\"#fig_21\">21</ref>.</p><p>Confirmation of the covalent nature can be observed in the thioester formation between Cys145 and the carbonyl of the inhibitor. Another important interaction is shown in the π-π stacking of the inhibitor indole ring with the imidazole ring of His41. <ref type=\"bibr\" target=\"#b104\">89</ref> Subsequent studies in other laboratories have expanded upon these studies utilizing compound 27 as a lead. Pillaiyar and co-workers investigated a series of small molecule thioesters, and Müller\\'s group used an in house protocol to identify lead compounds and performed a small subsequent optimization study. <ref type=\"bibr\" target=\"#b106\">91,</ref><ref type=\"bibr\" target=\"#b109\">92</ref> The design and synthesis of a series of 5-chloropyridinyl esters of common nonsteroidal anti-inflammatory agents (NSAID) has been investigated by Ghosh and co-workers. <ref type=\"bibr\" target=\"#b105\">90</ref> The chloropyridinyl esters of salicylic acid, ibuprofen, indomethacin, and related aromatic carboxylic acids exhibited potent moderate nanomolar enzyme inhibitory activity. As highlighted in Figure <ref type=\"figure\" target=\"#fig_22\">22</ref>, acetoxysalicylic acid derivative (aspirin-derived) 32 showed SARS-CoV-2 3CLpro inhibitory activity of 360 nM. Antiviral activity was determined using a quantitative VeroE6 cell-based assay with RNA-qPCR as described recently. <ref type=\"bibr\" target=\"#b62\">61,</ref><ref type=\"bibr\" target=\"#b112\">93</ref> However, these compounds did not show appreciable antiviral activity. Salicylic acid and methyl substituted derivatives were prepared and evaluated. Methyl group substitution on the aromatic ring resulted in the synthesis of monomeric, dimeric, and trimeric ester derivatives. Monomeric derivatives 33-35 showed moderate nanomolar 3CLpro inhibitory activity; however, only the 5-methyl derivative 35 showed antiviral activity (EC50 value 46 µM). Representative dimeric compound 36 showed slightly higher 3CLpro inhibitory activity. These compounds did not show cytotoxicity up to 100 µM. Racemic ibuprofen derivative 37 showed moderate enzyme inhibition and essentially no antiviral activity. (R)-Naproxenderived ester 38 exhibited significantly better 3CLpro inhibitory activity (IC50 of 160 nM) over (S)-naproxen-derived 39. However, both compounds did not show much antiviral activity. The mode of inhibition presumably involves the formation of a covalent bond with catalytic Cys145 and the ester carbonyl carbon in the active site. An active model of more potent (R)-naproxen derivative 38bound 3CL protease has been presented. The model in Figure <ref type=\"figure\" target=\"#fig_23\">23</ref> shows formation of a covalent bond with Cys145 and a strong hydrogen bond with Gln189. Also, His41 forms a nice π-π stacking interaction with the napthyl ring of (R)-naproxen and similar π-π stacking interactions to those observed in other X-ray crystallographic studies of indole-derived active ester derivatives as shown above. Zhang and collaborators reported a series of SARS-CoV-2 3CLpro inhibitors that incorporated pyrogallol warheads. <ref type=\"bibr\" target=\"#b113\">94</ref> Initial screening of flavonoid natural products led to the discovery that myricetin 40 and dihydromyricetin 41, shown in Figure <ref type=\"figure\" target=\"#fig_24\">24</ref>, exhibited potent 3CLpro inhibitory activity (&gt;90% inhibition). <ref type=\"bibr\" target=\"#b113\">94</ref> The 10.1002/cmdc.202200440</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Accepted Manuscript</head></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>ChemMedChem</head><p>This article is protected by copyright. All rights reserved.</p><p>X-ray structure of myricetin-bound SARS-CoV-2 3CLpro was determined. As depicted in Figure <ref type=\"figure\" target=\"#fig_25\">25</ref>, structural analysis revealed that myricetin forms a covalent bond with Cys145.  Yarovaya and co-workers reported the design, synthesis, and evaluation of a series of bispidine based SARS-CoV-2 3CLpro inhibitors. <ref type=\"bibr\" target=\"#b114\">95</ref> All derivatives were examined against SARS-CoV-2 3CLpro. Many compounds were found to have inhibitory activity ranging from 1-10 𝜇 M, and several compounds exhibited submicromolar activity. Unsubstituted bispidinone, 44 in Figure <ref type=\"figure\" target=\"#fig_26\">26</ref>, exhibited an IC50 of 2.6 µM. Other substituted bispidine derivatives, including amide derivatives 45 and derivatives without a carbonyl at C9 (46) were prepared and evaluated. In general, amide derivatives 45a-c were active; however, derivatives 46 without the C-9 ketone carbonyl were mostly inactive. Bis-amide derivatives 45a and 45b in Figure <ref type=\"figure\" target=\"#fig_28\">27</ref> displayed a 3CLpro IC50 value of 1.4 µM. Inhibitor 45c, containing the dihydrobenzo indazole, exhibited the best activity. Also, Nalkyl derivatives 47a and 47b displayed low micromolar inhibitory activity. In general, compounds with the scaffold of inhibitors 47 showed promising results with many of them in the low micromolar range. All compounds were also tested for their cytotoxicity in HEK293T cells and were found to be nontoxic. The mechanism of action was speculated to involve the formation of a covalent bond with Cys145 via the carbonyl at C9. <ref type=\"bibr\" target=\"#b114\">95</ref> ADMET analysis was performed on these compounds, and of those that displayed good drug-like properties. Compound 47a also displayed good in vitro activity.  </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"4.1.3\">Noncovalent Reversible Inhibitors</head><p>In 2013, Jacobs and co-workers discovered a novel class of non-covalent SARS-CoV-1 3CLpro inhibitors following a screening of NIH molecular libraries. As shown in Figure <ref type=\"figure\" target=\"#fig_29\">28</ref>, an initial hit provided lead compound 48 as an inhibitor of both SARS-CoV-1 3CLpro and PLpro enzymes. <ref type=\"bibr\" target=\"#b115\">96</ref> Subsequent synthesis of derivatives and SAR exploration led to the identification of active racemic compound 49 and the stereochemically defined, more active isomer 50. This R-isomer showed a 3CLpro IC50 value of 1.5 µM, and it is more potent than the S-isomer (IC50 28 µM). The X-ray structure of compound 50-bound SARS-CoV-1 3CLpro was determined and the structure provided detailed molecular interactions in the SARS-CoV-1 3CLpro active site. Detailed structural analysis revealed that compound 49 inhibits 3CLpro enzyme without forming a covalent bond with catalytic Cys145. Compound 50 fills in the S3-S1\\' subpockets of SARS-CoV-1 3CLpro. This discovery triggered further search of potent compounds through the synthesis of novel derivatives and this has been reviewed recently. With the outbreak of SARS-CoV-2, many laboratories further explored the potential of these lead structures by synthesizing other structural variants. Kitamura and collaborators examined a series of derivatives as potential SARS-CoV-2 3CLpro inhibitors. <ref type=\"bibr\" target=\"#b116\">97</ref> Representative compounds 51-53 in Figure <ref type=\"figure\" target=\"#fig_30\">29</ref> showed SARS-CoV-2 3CLpro inhibitory activity, with inhibitors 51 and 52 showing submicromolar activity. Compound 51 showed a SARS-CoV-2 3CLpro Ki value of 0.2 µM. It was tested against SARS-CoV-2 in VeroE6 cells using the immunofluorescence assay, and it was found to have an EC50 of 1.27 𝜇M. <ref type=\"bibr\" target=\"#b116\">97</ref> This compound did not show any activity against SARS-CoV-2 PLpro. Also, compound 51 did not exhibit any cytotoxicity against VeroE6 cells up to 100 𝜇M. <ref type=\"bibr\" target=\"#b116\">97</ref> Compound 52 with a biphenyl group as the P2 ligand, benzylmethyl as the P3 ligand, and imidazole as the P1\\' ligand  showed a 3CLpro inhibitory activity of 0.67 𝜇M. In general it was found that the biphenyl P2 ligand and the benzyl methyl P3 were the most active substitutions. An X-ray crystal structure of 51 bound to SARS-CoV-2 3CLpro was determined. As shown in Figure <ref type=\"figure\" target=\"#fig_31\">30</ref>, there is no covalent interaction with Cys145, as was previously shown with the 50-bound SARS-CoV-1 3CLpro structure. Ma and co-workers have reported the synthesis of Ugiderivatives with a P1\\'-dichloroacetamide, dibromoacetamide, tribromoacetamide, 2-bromo-2,2-dichloroacetamide, and 2chloro-2,2-dibromoacetamide functionalities. <ref type=\"bibr\" target=\"#b117\">98</ref> As highlighted in Figure <ref type=\"figure\" target=\"#fig_33\">31</ref>, compounds 54 and 55 were found to be the most potent with SARS-CoV-2 3CLpro IC50 values of 0.43 𝜇M and 0.08 𝜇 M, respectively. Compound 55 showed antiviral activity in VeroE6 and Caco2-hACE2 cells with EC50 values in the low micromolar range. The compound also displayed good selectivity (greater than 20 𝜇M) against calpain I, cathepsin B, cathepsin K, caspase-3 and trypsin. <ref type=\"bibr\" target=\"#b117\">98</ref> Structural analysis showed that the chloroacetamide or bromoacetamide warheads form a covalent linkage with Cys145 and the rest of the molecule interacts similar to other noncovalent inhibitors described previously. Jorgensen and collaborators reported an interesting series of potent noncovalent and nonpeptidic inhibitors. <ref type=\"bibr\" target=\"#b118\">99</ref> The investigators carried out virtual screenings of 2,000 known, approved drugs, and found 14 hits as inhibitors of SARS-CoV-2 3CLpro. <ref type=\"bibr\" target=\"#b118\">99</ref> As shown in Figure <ref type=\"figure\" target=\"#fig_34\">32</ref>, perampanel (56) was chosen as a lead compound. Further design and optimization based upon docking studies of compound 56 with 3CLpro were carried out. As highlighted, several potent derivatives, such as 57-59, were identified. Propoxy derivative 57 showed an enzyme IC50 value of 140 nM and an antiviral activity of 2.5 µM. Benzyloxy derivative 58 also showed potent 3CLpro inhibitory activity with an IC50 value of 128 nM. Compound 59 exhibited the most potent activity with an IC50 value of 18 nM. However, this compound did not show antiviral activity.</p><p>A crystal structure with synthesized derivative 57 with 3CLpro was determined. As shown in Figure <ref type=\"figure\" target=\"#fig_35\">33</ref>, catalytic Cys145 does not make any covalent bond with the inhibitor. The side pyridine nitrogen forms a hydrogen bond with His163 in the active site. The pyridine carbonyl group forms a hydrogen bond with the backbone NH of Glu166. A hydrogen bonding interaction between the nitrile and amide nitrogen of Gly143 can also be observed. Further structural modifications are highlighted in Figure <ref type=\"figure\" target=\"#fig_36\">34</ref>. Monofluoro derivative 60 was also found to be very potent. Overall results showed that this series of inhibitors are quite potent and show promise for further development. Inhibitors 61 and 62 also exhibited good activity. Inhibitor 59 shows synergistic effects with the FDA approved polymerase inhibitor remdesivir. <ref type=\"bibr\" target=\"#b118\">99</ref> Inhibitor 62 turned out to be the most potent with an EC50 value of 1.1 𝜇M.   Stauffer and collaborators reported a series of noncovalent SARS-CoV-2 3CLpro inhibitors. <ref type=\"bibr\" target=\"#b119\">100</ref> Lead compound ML300 ( <ref type=\"formula\">63</ref>) was a MLPCN probe compound that was chosen for optimization studies. ML300 was initially developed in the wake of the SARS-CoV-1 outbreak, and was chosen as the lead compound based on several factors including the need for further optimization to improve ADME properties like metabolic stability. <ref type=\"bibr\" target=\"#b115\">96</ref> A crystal structure with 63 (Figure <ref type=\"figure\" target=\"#fig_38\">35</ref>) bound SARS-CoV-2 3CLpro was obtained depicting key interactions in the enzyme-inhibitor complex shown in Figure <ref type=\"figure\" target=\"#fig_5\">36</ref>. Several key hydrogen bonding interactions are observed. The benzotriazole nitrogen forms a hydrogen bond with the side chain of His163. The amide carbonyl of 63 forms a hydrogen bonding interaction with the backbone nitrogen of Glu166, and the terminal amide associated with the cyclopropane forms a hydrogen bond with the alcohol of Ser46. The benzotriazole and Cys145 are involved in an interesting interaction with a slightly longer hydrogen bond.  Lead optimization efforts of ML300 originally led to inhibitor 64 with an IC50 of 0.95 µM against SARS-CoV-2 3CLpro, which was comparable to that of SARS-CoV-1 3CLpro. Several sub sites were varied, as shown in the lead compound. Imidazole derivative 65 exhibited very good nanomolar potency. Nmethylated derivatives of either pyrazole or imidazole inhibitors lost considerable potency, likely due to the loss of a hydrogen bond donor. The replacement of the thiophene moiety on 65 with a 3-chlorophenyl group resulted compound 66, exhibiting an IC50 of 0.27 µM. Further optimization keeping the benzotriazole and the 3chlorophenyl as fixed ligands led to potent compounds 67-70 shown in Figure <ref type=\"figure\" target=\"#fig_6\">37</ref>. Antiviral activities of the most promising compounds were tested in VeroE6 cells using the cytopathic effect inhibition and in a plaque reduction assay. <ref type=\"bibr\" target=\"#b119\">100</ref> Lead compound ML300 showed an EC50 of 19.9 µM in the live virus cytopathic effect (CPE) assay, whereas derivative 66 had an improved EC50 value of 1.7 µM in the CPE assay. It was observed that derivatives that substituted the thienyl moiety for 3chlorophenyl displayed much better antiviral activity. Compound 70 displayed submicromolar activity in both assays and had a good selectivity index.</p><p>Unoh and co-workers reported a novel series of noncovalent SARS-CoV-2 3CLpro inhibitors, one of which is undergoing clinical trials in Japan. <ref type=\"bibr\" target=\"#b120\">101,</ref><ref type=\"bibr\" target=\"#b121\">102</ref> A structure-based virtual screening was utilized to identify lead compounds from their inhouse compound library. One of the lead compounds, 71 (Figure <ref type=\"figure\" target=\"#fig_41\">38</ref>), showed a SARS-CoV-2 3CLpro IC50 value of 8.6 µM. <ref type=\"bibr\" target=\"#b120\">101</ref> The X-ray structure of compound 71-bound SARS-CoV-2 3CLpro was determined. Subsequent structure-based optimization led to potent derivatives, one of these potent compounds was 72 (S-217622), which was selected for clinical development. Inhibitor 72 exhibited potent in vitro activity against all tested SARS-CoV-2 variants including 𝛼, 𝛽, 𝛾, 𝛿, and 𝜊 strains. Inhibitor 72 also showed no inhibition of host cell proteases including caspase-2, chymotrypsin, cathepsin B, cathepsin D, cathepsin G, cathepsin L and thrombin. <ref type=\"bibr\" target=\"#b120\">101</ref> In vivo efficacy was also tested with SARS-CoV-2 infected mice, yielding favorable results. Treatment groups displayed considerably lower viral loads than nontreatment groups. <ref type=\"bibr\" target=\"#b120\">101</ref> These favorable results have allowed this orally active drug to progress to clinical trials in Japan, where it has completed Phase IIa. <ref type=\"bibr\" target=\"#b121\">102</ref> The efficacy and safety of a once daily dose over 5 days were evaluated, and showed a decrease in viral loads of 60-80% compared to the placebo group. <ref type=\"bibr\" target=\"#b121\">102</ref> Since then, approval for manufacture and sales in Japan were filed and an NIH funded global Phase III clinical trial has been announced. This clinical candidate does not require co-administration of another drug and can be given once daily. <ref type=\"bibr\" target=\"#b121\">102</ref> The X-ray co-crystal structure of 72-bound 3CLpro was determined. As shown in Figure <ref type=\"figure\" target=\"#fig_42\">39</ref>, the 1-methyl-1H-1,24-triazole moiety occupies the S1 pocket and forms a hydrogen bond with the side chain NH of His163. The 2,4,5 trifluorobenzyl group fits in the S2 site and is involved in π-π stacking interactions with the imidazole side chain of His41. The P1\\' ligand 6-chloro-2-methyl-2H-indazole forms a hydrogen bond with the backbone NH of Thr26. It also shows hydrophobic contacts with Met49. </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"4.2\">PLpro Inhibitors</head><p>SARS-CoV-2 PLpro is also an important drug design target since it plays a critical role in the coronavirus replication cycle. <ref type=\"bibr\" target=\"#b10\">10,</ref><ref type=\"bibr\" target=\"#b122\">103</ref> In particular, PLpro is involved in processing and maturation of viral polyproteins, assembly of the replicasetranscriptase complex, and disruption of host immune response. <ref type=\"bibr\" target=\"#b124\">104,</ref><ref type=\"bibr\" target=\"#b125\">105</ref> The X-ray crystallographic studies of SARS-CoV-1 PLpro, noncovalent inhibitor design, and a number of inhibitorbound SARS-CoV-1 PLpro were previously reported and reviewed. <ref type=\"bibr\" target=\"#b9\">9,</ref><ref type=\"bibr\" target=\"#b126\">106,</ref><ref type=\"bibr\" target=\"#b127\">107</ref> Recently, X-ray structural studies of GRL-0617bound SARS-CoV-2 PLpro have been reported. <ref type=\"bibr\" target=\"#b128\">108,</ref><ref type=\"bibr\" target=\"#b129\">109,</ref><ref type=\"bibr\" target=\"#b130\">110</ref> Osipiuk and co-workers reported a high resolution X-ray structure of inhibitor 73 (GRL-0617)-bound SARS-CoV-2 PLpro. <ref type=\"bibr\" target=\"#b128\">108</ref> As can be seen in the co-crystal structure in Figure <ref type=\"figure\" target=\"#fig_43\">40</ref>, the benzamide moiety forms hydrogen-bonding interactions with the main chain nitrogen of Gln270 and side chain of Asp165. Replacing this moiety with a benzylamine or benzyl sulfonamide isostere led to a significant reduction in potency and this portion was conserved in subsequent optimization studies. <ref type=\"bibr\" target=\"#b128\">108,</ref><ref type=\"bibr\" target=\"#b131\">111</ref> Subsequently, they designed a number of inhibitors (74-76) highlighted in Figure <ref type=\"figure\" target=\"#fig_44\">41</ref>.  All compounds have shown comparable SARS-CoV-2 PLpro activity and a few of them displayed antiviral acitivity, but are less potent than compound 73. An investigation by Shan et al. expanded upon the naphthyl derivatives. <ref type=\"bibr\" target=\"#b132\">112</ref> Two of their most potent compounds were found to have low micromolar activity in their novel assay.</p><p>Shen and co-workers reported a series of SARS-CoV-2 PLpro inhibitors via high-throughput screening. <ref type=\"bibr\" target=\"#b131\">111</ref> They utilized the naphthalenyl benzamide core structure of 73 as reported previously. <ref type=\"bibr\" target=\"#b126\">106</ref> As highlighted in Figure <ref type=\"figure\" target=\"#fig_45\">42</ref>, naphthalenyl derivative 77 is less potent than inhibitor 73, and as such it was noted that any variation in the benzylmethyl led to a loss in activity. Replacement of the naphthalene of 77 was then investigated. This was done in an attempt to improve metabolic stability. Fused heteroaryls such as benzothiophene, indole, and carbazole with differing linkages were investigated. Unfortunately, most modifications led to a loss in activity. The 3-benzothiphene and carabazole based analogues had reasonable potency. The biaryl derivatives such as 2-phenylthiphene and 3-phenylthiophene, however, showed a slight improvement in potency in compound 78.</p><p>Further modifications are shown in Figure <ref type=\"figure\" target=\"#fig_46\">43</ref>. Modification of the aryl amine to an azetidine ring resulted in a dramatic increase in potency due to enabling the electrostatic interactions with Glu167. Two of their most potent inhibitors, 78 and 79, were tested for their efficacy and bioavailability in human lung epithelial A549 cells. Inhibitor 81 was not effective in preliminary antiviral studies compared to the other two inhibitors, although it had a high binding affinity and low dissociation rate. Both 78 and 79 displayed more cytotoxicity than 77 at 100 µM and no cytotoxicity up to 30 µM. </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"5.\">RdRp Polymerase Inhibitors</head><p>RdRp is a nonstructural protein responsible for synthesizing viral RNA, which is then transcribed into viral proteins. Inhibition of this essential enzyme prevents protein synthesis, a vital step in the viral life cycle. This is also the only protein that is mostly conserved among RNA viruses, making it an attractive target for broad-spectrum antivirals. <ref type=\"bibr\" target=\"#b133\">113</ref> </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"5.1\">RdRp Nucleoside Inhibitors</head><p>Nucleoside inhibitors mimic substrates for the viral RdRp and lead to either chain termination or lethal mutagenesis. Unfortunately, nucleoside derivatives, particularly those that cause chain termination, can be less effective against coronaviruses. Coronaviruses contain an exonucleolytic proofreading mechanism that can remove misincorporated nucleotides. <ref type=\"bibr\" target=\"#b135\">114</ref> Therefore, these drug therapies need to be effective at circumventing the natural proofreading ability of RdRp. Remdesivir (1, Figures <ref type=\"figure\" target=\"#fig_48\">2 and 44</ref>) is an antiviral nucleotide phosphoramidate prodrug initially developed for treatment of Hepatitis C and later utilized as a therapy for Ebola and Marburg viral infections. <ref type=\"bibr\" target=\"#b136\">115</ref> Remdesivir triphosphate is the active form of the drug and a natural drug metabolite that competes with the native adenosine triphosphate for chain inclusion. <ref type=\"bibr\" target=\"#b136\">115,</ref><ref type=\"bibr\" target=\"#b137\">116</ref> Inclusion of remdesivir into the RNA chain prevents further RNA synthesis and leads to chain termination. This termination has been shown to be the result of a translocation barrier, which causes the coronavirus RdRp to stall after the addition of three nucleotides after incorporation of remdesivir. <ref type=\"bibr\" target=\"#b138\">117,</ref><ref type=\"bibr\" target=\"#b139\">118</ref> This is expected to occur due to the C1 nitrile in remdesivir, as it has been shown to be crucial for antiviral potency in the Ebola virus. <ref type=\"bibr\" target=\"#b138\">117</ref> Molecular modeling has also shown a steric clash between the nitrile of remdesivir and side chain of Ser861 in the Nsp12 subunit of RdRp. <ref type=\"bibr\" target=\"#b138\">117</ref> To combat this, viral proofreading can occur, which renders remdesivir less efficient. Figure <ref type=\"figure\" target=\"#fig_49\">45</ref> illustrates remdesivir and SARS-CoV-2-RdRp active site interactions based upon X-ray structural studies. <ref type=\"bibr\" target=\"#b139\">118</ref> Several studies have been done to show remdesivir efficacy; however mixed results have been observed. Clinical trials with remdesivir have been performed to determine its effect on patients with moderate COVID-19 versus standard of care, as well as the effect of a 5 day remdesivir treatment compared to 10 day treatment regimen. <ref type=\"bibr\" target=\"#b142\">119,</ref><ref type=\"bibr\" target=\"#b143\">120</ref> In the randomized, open-labeled trial of hospitalized patients with confirmed COVID-19 pneumonia across 105 hospitals in the US, Europe, and Asia it was found that these patients with moderate COVID-19 no statistical significance was observed for those that were treated with remdesivir for 10 days. However, a statistically significant difference was observed for those that received a 5-day treatment of remdesivir compared to standard of care treatment. This difference was of uncertain clinical importance though. <ref type=\"bibr\" target=\"#b142\">119</ref> In another randomized, open-label phase III clinical trial involving hospitalized patients with confirmed severe COVID-19 it was found that there was no significant difference between a 5-day and 10-day treatment regimen of remdesivir. A large, double-blind, randomized, placebo-controlled clinical trial of remdesivir treatment of COVID-19 was published on November 5, 2020. <ref type=\"bibr\" target=\"#b37\">36</ref> This was one of the larger studies and included patients of varying levels of COVID-19 infection. It was observed that those that received treatment with remdesivir had a median recovery time of 10 days as opposed to 15 days in the placebo control group. <ref type=\"bibr\" target=\"#b37\">36</ref> Patients who received remdesivir were also found to be more likely than those who received a placebo to have clinical improvement by day 15 and those in the remdesivir group had lower mortality estimates by day 15 and day 29. <ref type=\"bibr\" target=\"#b37\">36</ref> Based on the evidence from many clinical trials, remdesivir was FDA approved for use in hospitalized individuals infected with SARS-CoV-2 on October 22, 2020. <ref type=\"bibr\" target=\"#b144\">121</ref> However, remdesivir is currently only available as an intravenous injection and must be administered in a health care setting making it less available to the general public. Research into an orally bioavailable analogue is underway in many laboratories. One study found compound 82 (Figure <ref type=\"figure\" target=\"#fig_48\">44</ref>), which is an oral prodrug of the remdesivir triol 83. <ref type=\"bibr\" target=\"#b145\">122</ref> This analogue displayed high oral bioavailability in two animal species including non-human primates. <ref type=\"bibr\" target=\"#b145\">122</ref> Tested in VeroE6 cells against different strains of SARS-CoV-2 led to a range of EC50 values between 0.11 and 0.73 𝜇 M for GS-621763 and no appreciable cytotoxicity. <ref type=\"bibr\" target=\"#b146\">123</ref> Interestingly, it was also observed that treatment blocked viral transmission between untreated direct-contact animals.</p><p>Additionally, favipiravir (84, Figure <ref type=\"figure\" target=\"#fig_51\">46</ref>) has also been tested for its activity against SARS-CoV-2 RdRp and is currently undergoing clinical trials. <ref type=\"bibr\" target=\"#b147\">124,</ref><ref type=\"bibr\" target=\"#b148\">125,</ref><ref type=\"bibr\" target=\"#b149\">126</ref> Favipiravir is a potent influenza RdRp inhibitor that is intracellularly converted to its active triphosphate metabolite. <ref type=\"bibr\" target=\"#b136\">115</ref> It has been shown to have activity against several RNA viruses, and has displayed efficacy in VeroE6 cells infected with SARS-CoV-2 with an EC50 of 61.88 𝜇M and no appreciable cytotoxicity observed. <ref type=\"bibr\" target=\"#b136\">115,</ref><ref type=\"bibr\" target=\"#b150\">127</ref> Recently the structure of SARS-CoV-2 RdRp in the presence of favipiravir-RTP was published. <ref type=\"bibr\" target=\"#b151\">128</ref> Molnupiravir, (3) an orally bioavailable RdRp acting drug developed in conjunction with Emory University, Ridgeback Biotherapeutics, and Merck is also currently undergoing phase III clinical trials and received an approval for EUA. <ref type=\"bibr\" target=\"#b152\">129,</ref><ref type=\"bibr\" target=\"#b153\">130</ref> It was originally developed for use against influenza, but has demonstrated broad-spectrum antiviral activity, a good safety profile, tolerability, and oral bioavailability in humans. <ref type=\"bibr\" target=\"#b153\">130</ref> The mechanism of action for molnupiravir is slightly different than that for remdesivir. Like remdesivir, molnupiravir is converted to its active form, 𝛽-D-N 4 -hydroxycytidine triphosphate, in cells. <ref type=\"bibr\" target=\"#b135\">114,</ref><ref type=\"bibr\" target=\"#b153\">130</ref> This active form is a substrate for viral RNA polymerase, which subsequently allows incorporation into the viral RNA. Molnupiravir is incorporated into the RNA strand in place of cytidine triphosphate or uridine triphosphate, which forms stable base pairs with either guanine or adenine. <ref type=\"bibr\" target=\"#b135\">114</ref> Utilizing this erroneous RNA as a template results in molnupiravir directing the incorporation of guanine or adenine, and thus mutated RNA  products. <ref type=\"bibr\" target=\"#b135\">114</ref> As these mutations accumulate, lethal mutagenesis occurs. Molnupiravir escapes detection from the viral proofreading exonuclease via a couple of ways. First, it does not cause chain termination, and it is expected that its incorporation is not recognized by the viral proof-reading exonuclease as a misincorporation. <ref type=\"bibr\" target=\"#b135\">114</ref> Additionally, this evasion could be due to the stability of M-G and M-A base pairing, which also does not promote RdRp backtracking. <ref type=\"bibr\" target=\"#b135\">114</ref> Studies utilizing molnupiravir and favipiravir in tandem have been reported and found this to be particularly effective in hamsters, and it was also observed that it largely reduces the transmission of the virus to uninfected individuals. <ref type=\"bibr\" target=\"#b154\">131</ref> Due to the mechanism of action there has been cause for concern, among them possibilities of molnupiravir aiding in the emergence of variant coronaviruses. Despite the possible risks, a panel on the FDA voted 13-10 in favor of granting its emergency use authorization on November 30, 2021.</p><p>Since its EUA clinical data has shown that it is considerably less effective at reducing risk than was originally expected. Initial reports led researchers to believe molnupiravir would be closer to 50% effective, however, with more data becoming available only a 30% reduction in COVID-19 related hospitalization or death has been observed and reported. <ref type=\"bibr\" target=\"#b100\">85</ref> It has been reported to effectively reduce omicron viral RNA in cultured lung epithelial cells by 70% with an estimated IC50 against wild type SARS-CoV-2 of 1.965 𝜇M and 0.7556 𝜇M against the omicron variant. <ref type=\"bibr\" target=\"#b101\">86</ref> The most successful RdRp inhibitors have been mentioned above; however, there is no shortage of weapons in the drugrepurposing arsenal. Galidesivir, ribavirin, sofosbuvir, and tenofovir have been reported to be under consideration. <ref type=\"bibr\" target=\"#b136\">115,</ref><ref type=\"bibr\" target=\"#b146\">123</ref> Independently, an in silico drug repurposing study utilizing molecular docking and dynamics studies was conducted on over 30 compounds in which the ones to most tightly bind the active site of SARS-CoV-2 RdRp were found to be: remdesivir, favipiravir sofosbuvir, ribavirin, galidesivir, cefuroxime, tenofovir, and hydroxychloroquin. <ref type=\"bibr\" target=\"#b155\">132</ref> </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"5.2\">RdRp Non-Nucleoside Inhibitors</head><p>Unlike nucleoside inhibitors, non-nucleoside inhibitors are unencumbered by the proofreading activity of the coronavirus. Cen and coworkers reported a drug repurposing study and found a non-nucleoside RdRp inhibitor, corilagin 85 (Figure <ref type=\"figure\" target=\"#fig_52\">47</ref>), which binds directly to RdRp and inhibits the polymerase activity in both cell-free and cell-based assays. <ref type=\"bibr\" target=\"#b156\">133</ref> It fully resists the proofreading activity and potently inhibits SARS-CoV-2 infection with a low EC50. Their work began with a virtual screening of over 15,000 compounds that culminated in the top 50 hits with strong binding energies being selected. These hits binding affinities were further validated through bio-layer interferometry binding assay. Of the initial 50 compounds, 6 showed direct SARS-CoV-2 RdRp binding.</p><p>Corilagin showed the strongest RdRp binding affinity and was shown to effectively inhibit SARS-CoV-2 RdRp in Vero cells and showed moderate synergistic effects with remdesivir on inhibiting HCoV-OC43. <ref type=\"bibr\" target=\"#b156\">133</ref> Zandi and co-workers reported, two natural products, baicalein (86) and baicalin (87), displayed significant antiviral activity against SARS-CoV-2. <ref type=\"bibr\" target=\"#b157\">134</ref> Through their cell-based and biochemical studies, both compounds were determined to act as RdRp inhibitors. Antiviral activity was measured utilizing their previously established and optimized in vitro SARS-CoV-2 cell-based virus yield assay method and cytotoxicity was measured against Vero cells using a cell viability MTS assay. <ref type=\"bibr\" target=\"#b158\">135</ref> Assays were completed utilizing remdesivir as a control, and none of the tested compounds exhibited any cytotoxicity. Baicalein displayed more potent activity in contrast with baicalin and was found to inhibit intracellular replication even when added up to 8 hours post infection. Both compounds did show significant effects on the early stages of SARS-CoV-2 replication through 6-8 hours post infection. The hypothesized mechanism of action is that these inhibitors act by binding to SARS-CoV-2 RdRp in a place other than the active site and that the mechanism of action for these natural products differ from that of remdesivir. <ref type=\"bibr\" target=\"#b157\">134</ref> </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"6.\">Methyltransferase Inhibitors</head><p>Methyltransferase is an emerging target for drug design for antivirals of SARS-CoV-2, which contains two methyltransferases (MTase): Nsp14 and Nsp16. These enzymes are responsible for capping viral mRNAs, which is essential for transcription and avoiding the host\\'s innate immune system. Capping ensures integrity of the viral RNA and consists of an N-methylated guanosine triphosphate and C2\\'-O-methyl-ribosyladenine. <ref type=\"bibr\" target=\"#b160\">136</ref> It resembles the native mRNA of host cells, stabilizes the RNA, and ensures effective translation. <ref type=\"bibr\" target=\"#b160\">136</ref> This also allows the viral RNA to escape detection of the host\\'s innate immune system by mimicking the natural mRNA in host cells.</p><p>Both enzymes bind Nsp10, which is a stable monomeric protein with no currently identified individual purpose. It is mainly known to stabilize the S-adenosylmethionine (SAM)-binding pockets of both Nsp14 and Nsp16. <ref type=\"bibr\" target=\"#b160\">136</ref> Both Nsps are SAMdependent MTases. Nsp14 is responsible for methylation of the cap on the guanine of the GTP, N 7 -MTase, and exonuclease activities. <ref type=\"bibr\" target=\"#b161\">137</ref> Association with Nsp14 stimulates exonuclease activity, but does not affect the MTase activity. <ref type=\"bibr\" target=\"#b160\">136</ref> There is currently no crystal structure for SARS-CoV-2 Nsp14 MTase, although there is a solved crystal structure for the Nsp14 methyltransferase of SARS-CoV, which is expected to have a 94.9% homogeny with the SARS-CoV-2 Nsp14 methyltransferase. <ref type=\"bibr\" target=\"#b162\">138</ref> Nsp16 is a 7-methylguanine-triphosphate-adenosine specific, 2\\'-O-MTase, whose activation occurs via binding cofactor Nsp10. <ref type=\"bibr\" target=\"#b161\">137</ref> Nsp16 is responsible for the methylation of the C2\\' hydroxyl group of the following nucleotide. The crystal structure of Nsp10-Nsp16 complex bound to the inhibitor sinefungin has been reported and solved by Boura and coworkers as well as the high-resolution structures for Nsp16-Nsp10 heterodimers bound to SAM or sinefungin reported by Rosas-Lemus and collaborators. <ref type=\"bibr\" target=\"#b160\">136,</ref><ref type=\"bibr\" target=\"#b161\">137</ref> Inhibitors that prevent methyltransferase activity either by competitively binding the native substrate, SAM, or by blocking interactions with Nsp10 can be developed, and in fact, several small molecules have been reported in the last year. Jaudzems and co-workers reported a series of Nsp14 and Nsp16 inhibitors designed utilizing bioisosteric substitution of the sulfonium and amino acid side chains of the cosubstrate SAM. <ref type=\"bibr\" target=\"#b163\">139</ref> Three of their most potent compounds (88-91) are shown in Figure <ref type=\"figure\" target=\"#fig_54\">48</ref>, and sinefungin (91) was utilized as a control. It was found that replacing the sulfonium with a thioether resulted in an increase in potency. This is said to indicate that conformational flexibility and bulkiness is more essential for activity rather than the positive charge. <ref type=\"bibr\" target=\"#b163\">139</ref> An aromatic or heteroaromatic group in place of the aliphatic amino acid side chain also improved potency. Interestingly, it was noted that the improvements were more dramatic for Nsp16 over Nsp14, and this was hypothesized to be due to the ability for the inhibitor\\'s adenine fragment to bind Nsp16 in a conformation more similar to SAM than in Nsp14.</p><p>In order to determine inhibitor selectivity, their derivatives were also tested for their inhibition against human glycine Nmethyltransferase (GNMT). It was found that the inhibitors are also potent inhibitors of GNMT and are in fact not selective for SARS-CoV-2 methyltransferase. Despite this, the compounds exhibited no cytotoxicity in mouse embryo fibroblast, human liver cancer, and adenocarcinomic human alveolar basal epithelial cell lines. From there, it was suspected there was an issue with compound permeability. This was then tested by determining cell lysates and culture media by mass spectrometry and it was indeed found that the lysates were low and often below the detection limit. <ref type=\"bibr\" target=\"#b163\">139</ref> Thus, further optimization of this class of compounds is required to improve selectivity and lipophilicity. Nencka and collaborators reported a series of Nsp14 methyltransferase inhibitors created on structure based design of SARS-CoV-2 Nsp14 ligands. <ref type=\"bibr\" target=\"#b162\">138</ref> The structure and activity of the most active compounds, 92-94, are shown in Figure <ref type=\"figure\" target=\"#fig_55\">49</ref>. Inhibitor design was derived from a model of SARS-CoV-2 bound SAM based on crystallized SARS-CoV Nsp14. It was hypothesized that aromatic systems would be able to interact significantly with the aromatic amino acid residues and Arg289 through cation-𝜋 interactions. <ref type=\"bibr\" target=\"#b162\">138</ref> The mechanism of action of 92 and 93 were investigated and it was shown that both derivatives compete with SAM, but are RNA noncompetitive Nsp14 inhibitors. Inhibitor selectivity was also investigated against 33 human RNA-, DNA-, and protein-MTases. No significant inhibition was observed from 20 of the protein lysine-MTases, but both compounds did exhibit some inhibitory activity against other MTases.</p><p>Debart and co-workers reported their synthesis and evaluation of nucleoside-derived inhibitors against SARS-CoV-2 Nsp14 MTase, which catalyzes the transfer of the SAM methyl to the N7-guanosine cap. <ref type=\"bibr\" target=\"#b164\">140</ref> They tested 39 SAM analogues, of which 7 were found to exhibit double-digit nanomolar activity against the methyltransferase. 140 Three of the most potent discovered inhibitors stabilized Nsp14, and the best inhibitor showed high selectivity over human RNA N7-methyltransferase. They set out to synthesize smaller, efficient molecules with the Three regions focused on included the linker between the deoxyadenosine and the phenyl ring, substituents on the phenyl ring, and functionalization of the 5\\'-nitrogen on the nucleoside. The sulfonamide linker proved crucial, as it was replaced with an amide and lost inhibitory activity. Substitution on the sulfonamide nitrogen with an ethyl (R 4 ) was tested as it was expected to improve hydrophobicity and cellular penetration. Further investigations involved attaching a butyl chain with various terminal groups, such as an ethyl ester, acetate, phthalate, or phthalimide. An increase in potency was experienced when R 1 was varied to a nitro, which was then changed to a nitrile (96) as the nitro is expected to have increased mutagenic potential and the nitrile is expected to interact similarly with Arg310. <ref type=\"bibr\" target=\"#b164\">140</ref> Several compounds were also synthesized incorporating hydrophobic substituents at R 2 in order to fill the hydrophobic pocket. The most active compounds, 96-98, are shown in Figure <ref type=\"figure\" target=\"#fig_57\">50</ref>.</p><p>Biological activity was determined for all synthesized compounds, and only 8 were found to be not active against SARS-CoV-2 Nsp14. At least 65% inhibition was displayed for all of the active compounds, 11 of which displayed better activity than the control compound, sinefungin. <ref type=\"bibr\" target=\"#b164\">140</ref> Further investigation into the mechanism of action showed that the most active compound, 96, is a SAM-competitive inhibitor for Nsp14. Specificity of this inhibitor was also tested in dose response assays containing N7-MTase from vaccinia virus, 2\\'-O-MTases from Dengue virus, vaccinia virus, SARS-CoV-2, and human RNA N7-MTase (hRNMT). No inhibition was found for any of the viral MTases up to 50 𝜇M, and against hRNMT at 50 𝜇M compound 96 had an IC50 of 52.8 𝜇M. <ref type=\"bibr\" target=\"#b164\">140</ref> Thus, a high selectivity for SARS-CoV-2 Nsp14 was observed.</p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"7.\">Other Therapies for SARS-Cov-2</head><p>Prevention of viral entry into the cell has also proven to be an active area of research. There are several ways to prevent this: ACE-2 blockers, host cell protease inhibitors such as transmembrane serine protease 2 (TMPRSS2), inhibitors of viral fusion, and monoclonal antibodies directed towards the spike protein. Entry inhibitors targeting any of the steps of the viral entry process into cells are an emerging area of interest. It is known that the viral spike protein must bind to the angiotensin-converting enzyme 2 (ACE2) cell receptor in order to initiate cell entry. The spike protein has been most heavily investigated for vaccine production, but viral entry inhibitors have also been investigated to prevent this process.</p><p>Monoclonal antibodies have been investigated and given emergency use authorization for the treatment of COVID-19. Monoclonal antibodies developed center around the spike protein and utilize this to prevent viral entry into the cell. <ref type=\"bibr\" target=\"#b166\">141</ref> A risk reduction of 70-85% against mild to moderate COVID-19 infection in outpatient settings has been reported. <ref type=\"bibr\" target=\"#b166\">141</ref> Unfortunately, as treatment effectiveness revolves around the spike protein, new variants with extensive mutations in the spike protein may be less susceptible to current antibody treatment methods. Treatment is also more intensive as antibodies are administered as an intravenous injection, which must occur in a health care setting. In addition to antibodies, some small molecules have also been shown to inhibit the spike protein and ACE2 interaction. <ref type=\"bibr\" target=\"#b167\">142</ref> Buchwald and co-workers reported a series of smallmolecule inhibitors of the COVID Spike ACE2 protein-protein interaction (PPI) to prevent viral cell entry. <ref type=\"bibr\" target=\"#b168\">143</ref> Interestingly they chose to investigate several dyes and novel drug-like compounds derived from these dyes, stemming from a screening of their compound library. The depicted compounds 99-103 (Figure <ref type=\"figure\" target=\"#fig_58\">51</ref>) showed the most promising results for preventing PPI of ACE2 and the spike protein of SARS-CoV-2. They also investigated the binding partner for these compounds utilizing a thermal shift assay, and it was determined that the compounds were targeting the coronavirus spike protein. <ref type=\"bibr\" target=\"#b168\">143</ref> These promising results can be further expanded upon and optimized in the future.</p><p>Coronaviruses require calcium ions to coordinate amino acid residues within the conserved fusion peptide of the spike protein to enter cells. Daniel and co-workers investigated calciumchannel-blocker drugs to inhibit SARS-CoV-2 entry into cells. <ref type=\"bibr\" target=\"#b169\">144</ref> Cell lines tested included the most common VeroE6 cell line as well as epithelial lung cells. A total of five drugs from the different classes were chosen and tested for their activity. The most active and least cytotoxic compounds were found to be nifedipine (104, Figure <ref type=\"figure\" target=\"#fig_60\">52</ref>) and felodipine (105), with nifedipine exhibiting no noticeable cytotoxicity. These compounds are expected to inhibit viral entry into the cell via calcium ion chelation; however, this hypothesis needs to undergo further testing for confirmation as it is possible they could inhibit viral spread at other stages of the life cycle.  Hall and collaborators reported a virtual screen of 120 different compounds and found two novel chemotypes of entry inhibitors that target the fusion peptide of SARS-CoV-2.  Kinase inhibitors that interfere with the phosphorylation of proteins essential for cell growth, signaling, and survival are another proposed area of treatment for COVID-19. These inhibitors can prevent phosphorylation of proteins vital to the immune activation and inflammation, causing immunosuppression and less severe symptoms. <ref type=\"bibr\" target=\"#b171\">146,</ref><ref type=\"bibr\" target=\"#b172\">147</ref> Janus  <ref type=\"bibr\" target=\"#b173\">148</ref> Baricitinib received FDA approval on May 11, 2022 for the treatment of COVID-19 in hospitalized adults requiring supplemental oxygen, non-invasive or invasive mechanical ventilation, or extracorporeal membrane oxygen. <ref type=\"bibr\" target=\"#b173\">148,</ref><ref type=\"bibr\" target=\"#b175\">150</ref> Baricitinib in particular has a possibility of having direct antiviral activity via viral endocytosis interference, which would prevent SARS-CoV-2 from entering and infecting cells. <ref type=\"bibr\" target=\"#b172\">147</ref> Clinical trials involving baricitinib and remdesivir are currently underway. <ref type=\"bibr\" target=\"#b176\">151</ref> </p></div>\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head n=\"8.\">Conclusion</head><p>The emergence of the COVID-19 pandemic at the end of 2019 shocked the world and demanded the most serious attention for the development of efficacious treatment of COVID-19. As a consequence, scientists across the globe plunged into developing vaccines and antiviral drugs in an unprecedented manner. The development of COVID-19 vaccines is a remarkable achievement. Never before have such safe and effective vaccines been developed with such rapid speed. Certainly, the COVID-19 vaccine\\'s success will change the future of vaccine science. Vaccines will play a critical role in terminating the current pandemic.</p><p>However, development of effective antiviral drugs with proven efficacy is critically important to save lives and reduce morbidity and hospitalization. The  ABBREVIATIONS USED: 3CLpro, 3-chymotrypsin-like protease; 3CLpro, 3-chymotrypsin-like protease (3CLpro); ACE2, angiotensin-converting enzyme 2; ADMET, absorption, distribution, metabolism, excretion, toxicity; AIDS, acquired immunodeficiency syndrome; CPE, cytopathic effect; CYP, cytochrome P450 enzymes; DNA, deoxyribonucleic acid; FRET, Fluorescence Resonance Energy Transfer; GTP, guanosine triphosphate; GNMT, glycine N-methyltransferase; HCoVs, human coronaviruses; HCV, hepatitis C virus; NSP3, non-structural protein 3; HIV, human immunodeficiency virus; MERS-CoV, Middle east respiratory syndrome coronavirus; MHV coronavirus, Mouse Hepatitis Virus coronavirus; Mpro, main protease; MTase, methyltransferases; NSAID, nonsteroidal antiinflammatory agents; PDB, protein data base; PLpro, papain-like protease; RdRp, RNA-dependent RNA polymerase; RNA, ribonucleic acid; SAM, S-adenosylmethionine; SARS-CoV-2, severe acute respiratory syndrome coronavirus 2.</p></div><figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_0\"><head>Figure 1 .</head><label>1</label><figDesc>Figure 1. Important biochemical targets for COVID-19 drug discoveries.</figDesc><graphic coords=\"6,68.05,135.21,243.75,178.85\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_1\"><head>Figure 2 .</head><label>2</label><figDesc>Figure 2. Currently available COVID-19 Drug Therapies.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_2\"><head>Figure 3 .</head><label>3</label><figDesc>Figure 3. Structure of SARS-CoV 3CLpro inhibitors 4-7.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_3\"><head>Figure 4 .</head><label>4</label><figDesc>Figure 4. X-ray structure of SARS-CoV-2 3CLpro and inhibitor 7 complex (PDB: 7JKV)</figDesc><graphic coords=\"7,326.85,597.76,196.00,141.85\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_4\"><head>Figure 5 .</head><label>5</label><figDesc>Figure 5. Structure of SARS-CoV 3CLpro inhibitors 8-11.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_5\"><head>Figure 6 .</head><label>6</label><figDesc>Figure 6. X-ray structure of SARS-CoV-2 3CLpro and GC-376 complex (PDB: 6WTK)</figDesc><graphic coords=\"8,336.75,100.11,183.30,142.65\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_6\"><head>Figure 7 .</head><label>7</label><figDesc>Figure 7. Structure of HCV NSP3 and SARS-CoV 3CLpro inhibitors, boceprevir (12) and telaprevir (13).</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_7\"><head></head><label></label><figDesc>protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_8\"><head>Figure 8 .</head><label>8</label><figDesc>Figure 8. X-ray structure of the SARS-CoV-2 3CLpro-boceprevir (12) complex determined to 1.45 Å (PDB: 6WNP)</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_9\"><head>Figure 9 .</head><label>9</label><figDesc>Figure 9. Bicycloproline Inhibitors.</figDesc><graphic coords=\"9,67.10,272.46,206.60,155.50\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_10\"><head>Figure 10 .</head><label>10</label><figDesc>Figure 10. X-ray Structure of 16 with 3CLpro (PDB 7D3I).</figDesc><graphic coords=\"9,331.20,434.58,176.60,168.45\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_11\"><head></head><label></label><figDesc>protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_12\"><head>Figure 11 .</head><label>11</label><figDesc>Figure 11. Peptidomimetic covalent inhibitors</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_13\"><head>Figure 13 .</head><label>13</label><figDesc>Figure 13. Peptidomimetic α-acyloxymethylketone Inhibitors</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_14\"><head>Figure 14 .</head><label>14</label><figDesc>Figure 14. Peptidomimetic Inhibitor 21 bound 3CLpro PDB 7MBI</figDesc><graphic coords=\"11,82.00,475.91,167.30,153.90\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_15\"><head></head><label></label><figDesc>. Compound 22 was tested for its SARS-CoV-2 3CLpro and antiviral activity. It was found that 22 has acceptable solubility, stability in plasma, and low in vitro and in vivo clearances suitable for further development. The discovery of compound 22 paved the way for the development of nirmatrelvir 2 as the first SARS-CoV-2 3CLpro inhibitor drug for the treatment of COVID-19. Rat oral bioavailability of compound 22 was limited (F=1.4%). To improve oral absorption, other warheads were designed to reduce the hydrogen bonding ability of the α-hydroxymethyl ketone warhead in 22. Introduction of nitrile functionality resulted in compound 23 with good enzyme activity, and an improvement in oral bioavailability (F=7.8%), however, antiviral activity was reduced compared to 22. Incorporation of a boceprevir P2 ligand, 6,6dimethyl-3-azabicyclo[3.1.0] hexane in place of P2 leucine and</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_16\"><head>Figure 15 . 24 Figure 16 .</head><label>152416</label><figDesc>Figure 15. SARS-CoV-2 3CLpro Inhibitor 22-24</figDesc><graphic coords=\"11,345.25,617.86,153.00,133.80\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_17\"><head>Figure 17 .</head><label>17</label><figDesc>Figure 17. Development of nirmatrelvir 2 (PF-07321332)</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_18\"><head>Figure 18 .</head><label>18</label><figDesc>Figure 18. PF-07321332 bound 3CLpro (PDB 7VH8)</figDesc><graphic coords=\"12,337.35,309.56,186.80,152.20\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_19\"><head></head><label></label><figDesc>10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_20\"><head>Figure 19 .Figure 20 .</head><label>1920</label><figDesc>Figure 19. 5-chloropyridinyl ester SARS-CoV-2 covalent inhibitors</figDesc><graphic coords=\"13,328.20,354.45,210.55,132.75\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_21\"><head>Figure 21 .</head><label>21</label><figDesc>Figure 21. Inhibitor 29 (green) bound SARS-CoV-3CLpro (PDB 7RC1).</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_22\"><head>Figure 22 .</head><label>22</label><figDesc>Figure 22. NSAID derived SARS-CoV-2 3CLpro inhibitors.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_23\"><head>Figure 23 .</head><label>23</label><figDesc>Figure 23. An active site model of inhibitor 38 (violet) with SARS-CoV-2 3CLprotease. (PDB 7RBZ).</figDesc><graphic coords=\"14,321.00,536.16,207.40,149.00\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_24\"><head>Figure 24 .</head><label>24</label><figDesc>Figure 24. Pyrogallol 3CLpro Inhibitors 40-43.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_25\"><head>Figure 25 .</head><label>25</label><figDesc>Figure 25. Myricetin 40 (green) bound SARS-CoV-2 3CLpro (P DB 7DPP)</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_26\"><head>Figure 26 .</head><label>26</label><figDesc>Figure 26. Bis-pidinone-derived 3CLpro inhibitors 44-47.</figDesc><graphic coords=\"15,47.55,493.66,222.95,163.30\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_27\"><head>95</head><label>95</label><figDesc></figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_28\"><head>Figure 27 .</head><label>27</label><figDesc>Figure 27. Potent bispidinone-derived inhibitors 45a-c and 46a-b.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_29\"><head>Figure 28 .</head><label>28</label><figDesc>Figure 28. Noncovalent SARS-CoV-2 3CLpro Inhibitors</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_30\"><head>Figure 29 .</head><label>29</label><figDesc>Figure 29. Noncovalent Inhibitors 51-53.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_31\"><head>Figure 30 .</head><label>30</label><figDesc>Figure 30. Noncovalent Inhibitor 51 (magenta) bound 3CLpro (PDB 7KX5).</figDesc><graphic coords=\"16,313.85,528.76,199.70,146.35\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_32\"><head></head><label></label><figDesc>protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_33\"><head>Figure 31 .</head><label>31</label><figDesc>Figure 31. Dichloroacetamide and tribromoacetamide 3CLpro Inhibitors</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_34\"><head>Figure 32 .</head><label>32</label><figDesc>Figure 32. Noncovalent Perampanel Derived Inhibitors 56-59.</figDesc><graphic coords=\"17,305.40,525.40,210.95,172.80\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_35\"><head>Figure 33 .</head><label>33</label><figDesc>Figure 33. Inhibitor 57 (green) bound 3CLpro (PDB 7L11)</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_36\"><head>Figure 34 .</head><label>34</label><figDesc>Figure 34. Perampanel Derived Inhibitors 60-62.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_37\"><head>100</head><label>100</label><figDesc></figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_38\"><head>Figure 35 .</head><label>35</label><figDesc>Figure 35. Noncovalent 3CLpro Inhibitors 63-66</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_39\"><head>Figure 36 .Figure 37 .</head><label>3637</label><figDesc>Figure 36. Inhibitor 63 bound 3CLpro PDB 7LME</figDesc><graphic coords=\"18,322.50,215.25,200.00,122.25\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_40\"><head></head><label></label><figDesc>10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_41\"><head>Figure 38 .</head><label>38</label><figDesc>Figure 38. SARS-CoV-2 3CLpro noncovalent inhibitors 71 and 72</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_42\"><head>Figure 39 .</head><label>39</label><figDesc>Figure 39. Inhibitor 72-bound 3CLpro (PDB code: 7VU6).</figDesc><graphic coords=\"19,308.30,117.91,233.25,197.10\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_43\"><head>Figure 40 .</head><label>40</label><figDesc>Figure 40. Inhibitor 73-bound PLpro PDB 3E9S</figDesc><graphic coords=\"19,336.55,603.16,133.17,132.45\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_44\"><head>Figure 41 .</head><label>41</label><figDesc>Figure 41. PLpro Inhibitors 74-76.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_45\"><head>Figure 42 .</head><label>42</label><figDesc>Figure 42. PLpro Inhibitors 77 and 78.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_46\"><head>Figure 43 .</head><label>43</label><figDesc>Figure 43. PLpro Inhibitors 72 -74.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_47\"><head></head><label></label><figDesc>10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_48\"><head>Figure 44 .</head><label>44</label><figDesc>Figure 44. RdRp Nucleoside Inhibitors</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_49\"><head>Figure 45 .</head><label>45</label><figDesc>Figure 45. Remdesivir (green) bound RdRp (PDB 7BV2).</figDesc><graphic coords=\"21,304.75,200.71,238.95,155.15\" type=\"bitmap\" /></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_50\"><head></head><label></label><figDesc>10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_51\"><head>Figure 46 .</head><label>46</label><figDesc>Figure 46. Inhibitors 3 and 84.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_52\"><head>Figure 47 .</head><label>47</label><figDesc>Figure 47. Non-Nucleoside RdRp Inhibitors 85-87.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_53\"><head></head><label></label><figDesc>10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_54\"><head>Figure 48 .</head><label>48</label><figDesc>Figure 48. MTase Nsp14 and Nsp16 Inhibitors 88-91.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_55\"><head>Figure 49 .</head><label>49</label><figDesc>Figure 49. MTase Nsp 14 Inhibitors 92-94</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_56\"><head></head><label></label><figDesc>10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved. ability to enter cells. Several regions were varied as part of their SAR study, shown in red in compound 95 in Figure .</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_57\"><head>Figure 50 .</head><label>50</label><figDesc>Figure 50. Nsp14 Inhibitors 95-98.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_58\"><head>Figure 51 .</head><label>51</label><figDesc>Figure 51. Dye based PPI Inhibitors, 99-103</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_59\"><head>144</head><label>144</label><figDesc></figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_60\"><head>Figure 52 .</head><label>52</label><figDesc>Figure 52. Calcium ion blockers, 104-105</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_61\"><head>145</head><label>145</label><figDesc></figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_62\"><head>Figure 53 .</head><label>53</label><figDesc>Figure 53. Entry Inhibitors 106-108</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_63\"><head>Figure 54 .</head><label>54</label><figDesc>Figure 54. Kinase inhibitors baricitinib and tofacitinib</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_64\"><head></head><label></label><figDesc>groundwork for the development of small molecule antivirals for COVID-19 treatment has been nicely laid in the context of drug treatment options for both SARS-CoV infections in 2003 and MERS-CoV infections in 2012. Past and present research on coronavirus replication 10.1002/cmdc.202200440 Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" xml:id=\"fig_65\"><head></head><label></label><figDesc>unveiled numerous biochemical targets for antiviral development. Many different strategies and approaches are being pursued, which include de novo drug development, drug repurposing, and monoclonal antibody therapy development. In this review, we have outlined recent approaches to small molecule drug development targeting different pathways to block viral replication. Drug repurposing efforts led to the development and approval of remdesivir and molnupiravir, both drug targets viral RdRp. Remdesivir is an intravenous drug and molnupiravir is an oral therapy. Development of protease inhibitors have had much success for the treatment of HIV/AIDS and HCV infections. Early work on the coronavirus protease provided a much needed head start with lead structures and medicinal chemistry development efforts on SARS-CoV 3CLpro and SARS-CoV-PLpro targets. The development of the first FDA approved orally active SARS-CoV-2 3CLpro inhibitor drug is a very important achievement. Other new and effective treatments are needed to deal with side effects, resistance, and emerging variants. We hope the present review will further stimulate the development of effective broad-spectrum antivirals against COVID and emerging variants of SARS-CoV-2.</figDesc></figure>\\n<figure xmlns=\"http://www.tei-c.org/ns/1.0\" type=\"table\" xml:id=\"tab_0\"><head>Table 1 :</head><label>1</label><figDesc>Pfizer nirmatrelvir variant antiviral activity studies</figDesc><table><row><cell>Variant</cell><cell>Drug</cell><cell>EC50 (nM)</cell></row><row><cell>USA-WA1</cell><cell>Remdesivir Nirmatrelvir</cell><cell>15.4 32.2</cell></row><row><cell>𝛼</cell><cell>Remdesivir Nirmatrelvir</cell><cell>7.0 41.0</cell></row><row><cell>𝛽</cell><cell>Remdesivir Nirmatrelvir</cell><cell>14.8 127.2</cell></row><row><cell>𝛾</cell><cell>Remdesivir Nirmatrelvir</cell><cell>3.9 24.9</cell></row><row><cell>𝜆</cell><cell>Remdesivir Nirmatrelvir</cell><cell>3.3 21.2</cell></row><row><cell>𝛿</cell><cell>Remdesivir Nirmatrelvir</cell><cell>1.9 15.9</cell></row><row><cell>𝜇</cell><cell>Remdesivir Nirmatrelvir</cell><cell>5.8 25.7</cell></row><row><cell>𝜊</cell><cell>Remdesivir Nirmatrelvir</cell><cell>3.2 16.2</cell></row></table></figure>\\n\\t\\t</body>\\n\\t\\t<back>\\n\\n\\t\\t\\t<div type=\"acknowledgement\">\\n<div xmlns=\"http://www.tei-c.org/ns/1.0\"><head>Acknowledgements</head><p>The research was supported in part by grants from the National Institute of Allergy and Infectious Diseases, National Institutes of Health (A.K.G., AI158649 and A.D.M., contract No. HHSN272201700060C). The present work was also supported by a grant for Development of Novel Drugs for Treating COVID-19 from the Intramural Research Program of National Center for Global Health and Medicine (H.M., 19A3001 and S.H., 20A2001D), in part by Japan Agency for Medical Research and Development (AMED) (H.M., 20fk0108257). The authors acknowledge support from the Purdue Center for Cancer Research, NIH grant P30 CA023168 for use of the shared NMR and mass spectrometry facilities. This research used resources of the Advanced Photon Source, a U.S. Department of Energy (DOE) Office of Science User Facility operated for the DOE Office of Science by Argonne National Laboratory under Contract No. DE-AC02-06CH11357. Use of the LS-CAT Sector 21 was supported by the Michigan Economic Development Corporation and the Michigan Technology Tri-Corridor (Grant 085P1000817). Keywords: antiviral drugs • COVID-19 • coronavirus • drug design • inhibitors • protein X-ray structures • SARS-CoV-2</p></div>\\n\\t\\t\\t</div>\\n\\n\\t\\t\\t<div type=\"references\">\\n\\n\\t\\t\\t\\t<listBibl>\\n\\n<biblStruct xml:id=\"b0\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Yu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-M</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z.-G</forename><surname>Song</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b1\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J.-H</forename><surname>Tao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-Y</forename><surname>Tian</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M.-L</forename><surname>Pei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-L</forename><surname>Yuan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F.-H</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Dai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q.-M</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J.-J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">C</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-Z</forename><surname>Holmes</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">579</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"265\" to=\"269\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b2\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X.-L</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X.-G</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H.-R</forename><surname>Si</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C.-L</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H.-D</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Luo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R.-D</forename><surname>Jiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M.-Q</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X.-R</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X.-S</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q.-J</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Deng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L.-L</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Yan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F.-X</forename><surname>Zhan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-Y</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G.-F</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z.-L</forename><surname>Shi</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">579</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"270\" to=\"273\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b3\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Pandemic as a Scientific and Social Challenge in the 21st Century</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Zoumpourlis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Goulielmaki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Rizos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Baliou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Spandidos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno>COVID-19</idno>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Mol. Med. Rep</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b4\">\\n\\t<monogr>\\n\\t\\t<ptr target=\"https://www.who.int/publications-detail-redirect/who-convened-global-study-of-origins-of-sars-cov-2-china-part\" />\\n\\t\\t<title level=\"m\">WHO-convened global study of origins of SARS-CoV-2: China Part</title>\\n\\t\\t\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b5\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Coronavirus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno>COVID-19</idno>\\n\\t\\t<ptr target=\"https://gisanddata.maps.arcgis.com/apps/dashboards/bda7594740fd40299423467b48e9ecf6\" />\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019-nCoV</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b6\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">The Lancet</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Ren</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Fan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Gu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Cheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Yu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Xie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Yin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Jiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Jin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t\\t<biblScope unit=\"volume\">395</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"497\" to=\"506\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b7\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Microbiol. Biotechnol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">32</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1\" to=\"5\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b8\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Rodrigues</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Bento</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Cunha</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Vassilevskaia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Viveiros</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Cunha</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Molecules</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">27</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">2723</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b9\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Brindisi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shahabi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Chapman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ChemMedChem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">15</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"907\" to=\"932\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b10\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">M</forename><surname>Báez-Santos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">E</forename><surname>St</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>John</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antiviral Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">115</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"21\" to=\"38\" />\\n\\t\\t\\t<date type=\"published\" when=\"2015\">2015</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b11\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Cui</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z.-L</forename><surname>Shi</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Rev. Microbiol</title>\\n\\t\\t<editor>Zumla, D. S. Hui, S. Perlman</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">17</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"995\" to=\"1007\" />\\n\\t\\t\\t<date type=\"published\" when=\"2015\">2019. 2015</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>The Lancet</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b12\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">N</forename><surname>Vlasova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Diaz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Damtie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Xiu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">H</forename><surname>Toh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">S</forename><surname>Lee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">J</forename><surname>Saif</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">C</forename><surname>Gray</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Emerg. Microbes Infect</title>\\n\\t\\t<editor>N. Vlasova, T. H. Toh, J. S. Lee, Y. Poovorawan, P. Davis, M. S. P. Azevedo, J. A. Lednicky, L. J. Saif, G. C. Gray</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">74</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"699\" to=\"702\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022. 2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>Clin. Infect. Dis.</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b13\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">S M</forename><surname>Peiris</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Guan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">Y</forename><surname>Yuen</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature Medicine</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"S88\" to=\"S97\" />\\n\\t\\t\\t<date type=\"published\" when=\"2004\">2004</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b14\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Pyrc</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Berkhout</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Van Der Hoek</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Expert Rev. Anti Infect. Ther</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">5</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"245\" to=\"253\" />\\n\\t\\t\\t<date type=\"published\" when=\"2007\">2007</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b15\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Drosten</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Günther</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Preiser</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Van Der Werf</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H.-R</forename><surname>Brodt</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Becker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Rabenau</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Panning</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Kolesnikova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">A M</forename><surname>Fouchier</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Berger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A.-M</forename><surname>Burguière</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Cinatl</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Eickmann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Escriou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Grywna</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kramme</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J.-C</forename><surname>Manuguerra</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Müller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Rickerts</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Stürmer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Vieth</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H.-D</forename><surname>Klenk</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D M E</forename><surname>Osterhaus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Schmitz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">W</forename><surname>Doerr</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">348</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1967\" to=\"1976\" />\\n\\t\\t\\t<date type=\"published\" when=\"2003\">2003</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b16\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Christian</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">M</forename><surname>Poutanen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Loutfy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">P</forename><surname>Muller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">E</forename><surname>Low</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Clin. Infect. Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">38</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1420\" to=\"1427\" />\\n\\t\\t\\t<date type=\"published\" when=\"2004\">2004</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b17\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Hilgenfeld</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Peiris</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antiviral Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">100</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"286\" to=\"295\" />\\n\\t\\t\\t<date type=\"published\" when=\"2013\">2013</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b18\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Eckerle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Müller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kallies</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">N</forename><surname>Gotthardt</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Drosten</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"middle\">J</forename><surname>Virol</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2013\">2013</date>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">359</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b19\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Butler</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">522</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"139\" to=\"140\" />\\n\\t\\t\\t<date type=\"published\" when=\"2015\">2015</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b20\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Learning from the Past: Possible Urgent Prevention and Treatment Options for Severe Acute Respiratory Infections Caused by</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">S</forename><surname>Morse</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Lalonde</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">NCoV</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019. 2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>preprint; Chemistry</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b21\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">V</forename><surname>Baranov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M</forename><surname>Henderson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">B</forename><surname>Anderson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">F</forename><surname>Gesteland</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">F</forename><surname>Atkins</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">T</forename><surname>Howard</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Virology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">332</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"498\" to=\"510\" />\\n\\t\\t\\t<date type=\"published\" when=\"2005\">2005</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b22\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">E</forename><surname>Gorbalenya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">J</forename><surname>Snijder</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Ziebuhr</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Gen. Virol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">81</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"853\" to=\"879\" />\\n\\t\\t\\t<date type=\"published\" when=\"2000\">2000</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b23\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno type=\"DOI\">7117-7130.10.1002/cmdc.202200440</idno>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nucleic Acids Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">31</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2003\">2003</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b24\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved</title>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b25\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Dömling</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t\\t<biblScope unit=\"volume\">6</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1283\" to=\"1295\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b26\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Pillaiyar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Manickam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Namasivayam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hayashi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-H</forename><surname>Jung</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">59</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"6595\" to=\"6628\" />\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b27\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Forni</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Cagliani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Clerici</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Sironi</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Trends Microbiol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">25</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"35\" to=\"48\" />\\n\\t\\t\\t<date type=\"published\" when=\"2017\">2017</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b28\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Graham</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">S</forename><surname>Sparks</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">D</forename><surname>Eckerle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Sims</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Denison</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Virus Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">133</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"88\" to=\"100\" />\\n\\t\\t\\t<date type=\"published\" when=\"2008\">2008</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b29\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Kanjanahaluethai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Jukneliene</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Virology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">361</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"391\" to=\"401\" />\\n\\t\\t\\t<date type=\"published\" when=\"2007\">2007</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b30\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Thiel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">A</forename><surname>Ivanov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Á</forename><surname>Putics</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Hertzig</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Schelle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Bayer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Weißbrich</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">J</forename><surname>Snijder</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Rabenau</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">W</forename><surname>Doerr</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">E</forename><surname>Gorbalenya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Ziebuhr</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Gen. Virol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">84</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2305\" to=\"2315\" />\\n\\t\\t\\t<date type=\"published\" when=\"2003\">2003</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b31\">\\n\\t<monogr>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Ding</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Lou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Sun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Mo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Ye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Pang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">F</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Anand</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Bartlam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Hilgenfeld</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Rao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<title level=\"m\">Proc. Natl. Acad. Sci</title>\\n\\t\\t\\t\\t<meeting>Natl. Acad. Sci</meeting>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2003\">2003</date>\\n\\t\\t\\t<biblScope unit=\"volume\">100</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"13190\" to=\"13195\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b32\">\\n\\t<monogr>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">S</forename><surname>Saikatendu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">D</forename><surname>Santarsiero</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Barretto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">C</forename><surname>Stevens</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<title level=\"m\">Proc. Natl. Acad. Sci</title>\\n\\t\\t\\t\\t<meeting>Natl. Acad. Sci</meeting>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2006\">2006</date>\\n\\t\\t\\t<biblScope unit=\"volume\">103</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"5717\" to=\"5722\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b33\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">M</forename><surname>Baez-Santos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">M</forename><surname>Mielech</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Deng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Virol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">88</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"12511\" to=\"12527\" />\\n\\t\\t\\t<date type=\"published\" when=\"2014\">2014</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b34\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">B</forename><surname>Serafin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Bottega</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">S</forename><surname>Foletto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">F</forename><surname>Da Rosa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Hörner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Hörner</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Int. J. Antimicrob. Agents</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">55</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">105969</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b35\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Sun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Simeonov</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Br. J. Pharmacol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">175</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"181\" to=\"191\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b36\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">J</forename><surname>Brown</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">J</forename><surname>Won</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Graham</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">H</forename><surname>Dinnon</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Sims</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">Y</forename><surname>Feng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Cihlar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Denison</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">S</forename><surname>Baric</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">P</forename><surname>Sheahan</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antiviral Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">104541</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2019\">2019</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b37\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">H</forename><surname>Beigel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>Tomashek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">E</forename><surname>Dodd</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Mehta</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">S</forename><surname>Zingman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Kalil</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Hohmann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">Y</forename><surname>Chu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Luetkemeyer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kline</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Lopez De Castilla</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">W</forename><surname>Finberg</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Dierberg</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Tapson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Hsieh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">F</forename><surname>Patterson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Paredes</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">A</forename><surname>Sweeney</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">R</forename><surname>Short</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Touloumi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">C</forename><surname>Lye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Ohmagari</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Oh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">M</forename><surname>Ruiz-Palacios</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Benfield</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Fätkenheuer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">G</forename><surname>Kortepeter</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Atmar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<editor>C. B. Creech, J. Lundgren, A. G. Babiker, S. Pett, J. D. Neaton, T. H. Burgess, T. Bonnett, M. Green, M. Makowski, A. Osinusi, S. Nayak, H. C. Lane</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">383</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1813\" to=\"1826\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b38\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">T</forename><surname>Eastman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">S</forename><surname>Roth</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">R</forename><surname>Brimacombe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Simeonov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Patnaik</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Hall</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Cent. Sci</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">6</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"672\" to=\"683\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b39\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Remdesivir for the Treatment of Covid-19 -Preliminary Report</title>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">383</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"992\" to=\"994\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b40\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">C</forename><surname>Yan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">L</forename><surname>Muller</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antimicrob. Agents Chemother</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"e01117\" to=\"1121\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b41\">\\n\\t<monogr>\\n\\t\\t<ptr target=\"https://www.who.int/news-room/feature-stories/detail/who-recommends-against-the-use-of-remdesivir-in-covid-19-patients\" />\\n\\t\\t<title level=\"m\">WHO recommends against the use of remdesivir in COVID-19 patients</title>\\n\\t\\t\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b42\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Gottlieb</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">E</forename><surname>Vaca</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Paredes</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Mera</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">J</forename><surname>Webb</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Perez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Oguchi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Ryan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">U</forename><surname>Nielsen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Brown</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Hidalgo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Sachdeva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Mittal</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Osiyemi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Skarbinski</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Juneja</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">H</forename><surname>Hyland</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Osinusi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Camus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Abdelghany</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Davies</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Behenna-Renton</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<editor>F. Duff, F. M. Marty, M. J. Katz, A. A. Ginde, S. M. Brown, J. T. Schiffer, J. A. Hill</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">386</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"305\" to=\"315\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b43\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Outpatient Remdesivir to Prevent Progression to Severe Covid-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P.-V</forename><surname>Ennezat</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">386</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1094\" to=\"1094\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b44\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Molnupiravir, an Oral Antiviral Treatment for COVID-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Fischer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">J</forename><surname>Eron</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Holman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">S</forename><surname>Cohen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Fang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">J</forename><surname>Szewczyk</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">P</forename><surname>Sheahan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Baric</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">R</forename><surname>Mollan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">R</forename><surname>Wolfe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">R</forename><surname>Duke</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Azizad</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Borroto-Esoda</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">A</forename><surname>Wohl</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">J</forename><surname>Loftis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Alabanza</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Lipansky</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">P</forename><surname>Painter</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">2021</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>preprint; Infectious Diseases (except HIV/AIDS</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b45\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">Jayk</forename><surname>Bernal</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Gomes Da Silva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">B</forename><surname>Musungaie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Kovalchuk</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Gonzalez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Reyes</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Martín-Quirós</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Caraco</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Williams-Diaz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">L</forename><surname>Brown</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Du</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Pedley</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Assaid</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Strizki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Grobler</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">H</forename><surname>Shamsuddin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Tipping</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Wan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Paschke</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">R</forename><surname>Butterton</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">G</forename><surname>Johnson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">De</forename><surname>Anda</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">386</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"509\" to=\"520\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b46\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Update: FDA Authorizes Additional Oral Antiviral for Treatment of COVID-19 in Certain Adults</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Commissioner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Coronavirus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno>COVID-19</idno>\\n\\t\\t<ptr target=\"https://www.fda.gov/news-events/press-announcements/coronavirus-covid-19-update-fda-authorizes-additional-oral-antiviral-treatment-covid-19-certain\" />\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b47\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\">EUA 105 Pfizer Paxlovid LOA</title>\\n\\t\\t\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b48\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">R</forename><surname>Owen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M N</forename><surname>Allerton</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">S</forename><surname>Anderson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Aschenbrenner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Avery</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Berritt</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Boras</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">D</forename><surname>Cardin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Carlo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">J</forename><surname>Coffman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Dantonio</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Di</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Eng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Ferre</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">S</forename><surname>Gajiwala</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Gibson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">E</forename><surname>Greasley</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">L</forename><surname>Hurst</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">P</forename><surname>Kadar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">S</forename><surname>Kalgutkar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">C</forename><surname>Lee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">W</forename><surname>Mason</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Noell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">J</forename><surname>Novak</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">S</forename><surname>Obach</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ogilvie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">C</forename><surname>Patel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Pettersson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">K</forename><surname>Rai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Reese</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">F</forename><surname>Sammons</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">G</forename><surname>Sathish</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">S P</forename><surname>Singh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M</forename><surname>Steppan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">E</forename><surname>Stewart</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">B</forename><surname>Tuttle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Updyke</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">R</forename><surname>Verhoest</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Wei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Science</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">374</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1586\" to=\"1593\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b49\">\\n\\t<analytic>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Xi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Johnson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"m\">Progress in Anti-SARS Coronavirus Chemistry</title>\\n\\t\\t\\t\\t<imprint>\\n\\t\\t\\t<publisher>Elsevier</publisher>\\n\\t\\t\\t<date type=\"published\" when=\"2006\">2006</date>\\n\\t\\t\\t<biblScope unit=\"volume\">41</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"183\" to=\"196\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b50\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Pfizer&apos;s Novel COVID-19 Oral Antiviral Treatment Candidate Reduced Risk of Hospitalization or Death by 89% in Interim Analysis of Phase 2/3 EPIC-HR Study | Pfizer</title>\\n\\t\\t<ptr target=\"https://www.pfizer.com/news/press-release/press-release-detail/pfizers-novel-covid-19-oral-antiviral-treatment-candidate\" />\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b51\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Hsu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H.-C</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G.-W</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-R</forename><surname>Shih</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Curr. Pharm. Des</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1301\" to=\"1314\" />\\n\\t\\t\\t<date type=\"published\" when=\"2006\">2006</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b52\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Shyr</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">C</forename><surname>Lo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Pharmacol. Exp. Ther</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">378</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"166\" to=\"172\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b53\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Prokofjeva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">N</forename><surname>Kochetkov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">S</forename><surname>Prassolov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2016\">Acta Naturae 2016</date>\\n\\t\\t\\t<biblScope unit=\"volume\">8</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"23\" to=\"32\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b54\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">C</forename><surname>Pedersen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Galasiti Kankanamalage</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Eckstrand</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">C</forename><surname>Groutas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Bannasch</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">M</forename><surname>Meadows</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K.-O</forename><surname>Chang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Feline Med. Surg</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">20</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"378\" to=\"392\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b55\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sharun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Tiwari</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Dhama</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Ann. Med. Surg</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">2021</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"122\" to=\"125\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b56\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Mody</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Ho</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Wills</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Mawri</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Lawson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">C C J C</forename><surname>Ebert</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">M</forename><surname>Fortin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Rayalam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Commun. Biol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"1993\">2021, 4, 93</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b57\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>He</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Deng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Sun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Yin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Shang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t\\t<biblScope unit=\"volume\">10</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"5871\" to=\"5890\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b58\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Macip</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Garcia-Segura</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Mestres-Truyol</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Saldivar-Espinoza</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Pujadas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Garcia-Vallvé</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Int. J. Mol. Sci</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">23</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">259</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b59\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Xi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">D</forename><surname>Santarsiero</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Fu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">H</forename><surname>Harcourt</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">A</forename><surname>Rota</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Johnson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">48</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"6767\" to=\"6771\" />\\n\\t\\t\\t<date type=\"published\" when=\"2005\">2005</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b60\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Xi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Grum-Tokars</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Fu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">V</forename><surname>Houser</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Johnson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Bioorg. Med. Chem. Lett</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">17</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"5876\" to=\"5880\" />\\n\\t\\t\\t<date type=\"published\" when=\"2007\">2007</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b61\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">P</forename><surname>Jain</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">I</forename><surname>Pettersson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">D</forename><surname>Aull</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">D</forename><surname>Fortin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Huitema</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">D</forename><surname>Eltis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">C</forename><surname>Parrish</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">N G</forename><surname>James</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">S</forename><surname>Wishart</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">C</forename><surname>Vederas</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">47</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"6113\" to=\"6116\" />\\n\\t\\t\\t<date type=\"published\" when=\"2004\">2004</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b62\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Hattori</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Higashi-Kuwata</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Hayashi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">R</forename><surname>Allu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Raghavaiah</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Bulut</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Das</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">J</forename><surname>Anson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">K</forename><surname>Lendy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Takamatsu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Takamune</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Kishimoto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Murayama</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Hasegawa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">A</forename><surname>Davis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">N</forename><surname>Kodama</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Yarchoan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Wlodawer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Misumi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Mitsuya</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">668</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b63\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Lovell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K.-C</forename><surname>Tiew</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">R</forename><surname>Mandadapu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">R</forename><surname>Alliston</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">P</forename><surname>Battaile</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">C</forename><surname>Groutas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K.-O</forename><surname>Chang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Virol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">86</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"11754\" to=\"11762\" />\\n\\t\\t\\t<date type=\"published\" when=\"2012\">2012</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b64\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Shivanna</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Narayanan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">M</forename><surname>Prior</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Weerasekara</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">H</forename><surname>Hua</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C G</forename><surname>Kankanamalage</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">C</forename><surname>Groutas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K.-O</forename><surname>Chang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Virol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">89</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"4942\" to=\"4950\" />\\n\\t\\t\\t<date type=\"published\" when=\"2015\">2015</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b65\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Vuong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Fischer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">B</forename><surname>Khan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Van Belkum</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Lamer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">D</forename><surname>Willoughby</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Arutyunova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Joyce</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">A</forename><surname>Saffran</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Shields</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">S</forename><surname>Young</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Nieman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">L</forename><surname>Tyrrell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Lemieux</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">C</forename><surname>Vederas</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Eur. J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">113584</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b66\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Fu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Ye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Feng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Yu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Sun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Niu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Song</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Shi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Tan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Qi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">F</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">4417</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b67\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">B</forename><surname>Khan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Brassard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Arutyunova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Lamer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Vuong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Fischer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">S</forename><surname>Young</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">C</forename><surname>Vederas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Lemieux</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Front. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">852210</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b68\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Sacco</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Hurst</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Townsend</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Szeto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Tarbet</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">T</forename><surname>Marty</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">30</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"678\" to=\"692\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b69\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">J</forename><surname>Prongay</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Yao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Pichardo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Fischmann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Strickland</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Myers</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">C</forename><surname>Weber</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">M</forename><surname>Beyer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Ingram</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Hong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">W</forename><surname>Prosise</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Ramanathan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">S</forename><surname>Taremi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Yarosh-Tomaine</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Senior</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R.-S</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Malcolm</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Arasappan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Bennett</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">L</forename><surname>Bogen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Jao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-T</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">G</forename><surname>Lovey</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Saksena</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Venkatraman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Girijavallabhan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">G</forename><surname>Njoroge</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Madison</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">50</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2310\" to=\"2318\" />\\n\\t\\t\\t<date type=\"published\" when=\"2007\">2007</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b70\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">A</forename><surname>Malcolm</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Lahser</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Agrawal</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Belanger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Butkiewicz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Chase</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Gheyas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Hart</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Hesk</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Ingravallo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Jiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Kong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Pichardo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Prongay</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Skelton</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Tong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Venkatraman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Xia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Girijavallabhan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">G</forename><surname>Njoroge</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antimicrob. Agents Chemother</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">50</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1013\" to=\"1020\" />\\n\\t\\t\\t<date type=\"published\" when=\"2006\">2006</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b71\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">J</forename><surname>Anson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b72\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Broad-Spectrum inhibition of coronavirus main and papain-like proteases by HCV drugs</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Chapman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">K</forename><surname>Lendy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Pshenychnyi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">T</forename><surname>D\\'auila</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">K J F</forename><surname>Satchell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno type=\"DOI\">10.21203/rs.3.rs-26344/v1</idno>\\n\\t\\t<ptr target=\"https://doi.org.10.21203/rs.3.rs-26344/v1\" />\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Res. Square, PREPRINT</title>\\n\\t\\t<editor>b) R. Oerlemans, A. J. Ruiz-Moreno, Y. Cong, N. Dinesh Kumar, M. A. Velasco-Velazquez, C. G. Neochoritis, J. Smith, F. Reggiori, M. R. Groves, A. Dömling</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"370\" to=\"379\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>RSC Med. Chem.</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b73\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Qiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-S</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Zeng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F.-L</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R.-H</forename><surname>Luo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-F</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Quan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Mao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Sun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Ni</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z.-L</forename><surname>Duan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q.-C</forename><surname>Zou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H.-L</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Qu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-H.-P</forename><surname>Long</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M.-H</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R.-C</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>You</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Yao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W.-P</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J.-M</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G.-F</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G.-W</forename><surname>Lu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W.-M</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-Q</forename><surname>Wei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-T</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Science</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">371</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1374\" to=\"1378\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b74\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">S</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><forename type=\"middle\">Z</forename><surname>Geng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">R</forename><surname>Alugubelli</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Shaabani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">C</forename><surname>Vatansever</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><forename type=\"middle\">R</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">C</forename><surname>Cho</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Khatua</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">J</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">R</forename><surname>Blankenship</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b75\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Yu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Sankaran</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Allen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Ji</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">R E</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Z. Z. Geng; K. S</title>\\n\\t\\t<editor>b) Y. R. Alugubelli</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022, 240, 114570</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>J. Med. Chem.</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b76\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Shaabani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Khatua</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><forename type=\"middle\">R</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">C</forename><surname>Vatansever</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">C</forename><surname>Cho</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b77\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">R</forename><surname>Blankenship</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Yu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Sankaran</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Allen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Ji</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b78\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">R E</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">114596</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b79\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">X</forename><surname>Quan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Shuai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b80\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">J</forename><surname>Xia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Zeng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><forename type=\"middle\">L</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">F</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">X</forename><surname>Qiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b81\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">P</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">L</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">J</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">T</forename><surname>Yuen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">X</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b82\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Yoon</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">Y</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">F</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">W</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Tian</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b83\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">M</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">Q</forename><surname>Wei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">Y</forename><surname>Yuen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Chan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Chu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature Microbiology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">7</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"716\" to=\"725\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b84\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Xia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Sacco</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Meng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Szeto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Pharmacol. Transl. Sci</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">2021</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1408\" to=\"1421\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b85\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Joyce</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">A</forename><surname>Saffran</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Shields</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b86\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">B</forename><surname>Khan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Arutyunova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">K</forename><surname>Bajwa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Hockman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Fischer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Lamer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">W</forename><surname>Vuong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b87\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Van Belkum</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">Z</forename><surname>Gu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b88\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Du</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b89\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Rahim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">S</forename><surname>Young</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">C</forename><surname>Vederas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">L</forename><surname>Tyrrell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b90\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Lemieux</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Nieman</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2905\" to=\"2925\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b91\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Krantz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">J</forename><surname>Copp</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">J</forename><surname>Coles</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">A</forename><surname>Smith</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">B</forename><surname>Heard</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Biochemistry</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">30</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"4678\" to=\"4687\" />\\n\\t\\t\\t<date type=\"published\" when=\"1991\">1991</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b92\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">A</forename><surname>Smith</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">J</forename><surname>Copp</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">J</forename><surname>Coles</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">W</forename><surname>Pauls</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">J</forename><surname>Robinson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">W</forename><surname>Spencer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">B</forename><surname>Heard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Krantz</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Am. Chem. Soc</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">110</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"4429\" to=\"4431\" />\\n\\t\\t\\t<date type=\"published\" when=\"1988\">1988</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b93\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">H</forename><surname>Eichhold</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">B</forename><surname>Hookfin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">O</forename><surname>Taiwo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>De</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">R</forename><surname>Wehmeyer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Pharm. Biomed. Anal</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">16</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"459\" to=\"467\" />\\n\\t\\t\\t<date type=\"published\" when=\"1997\">1997</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b94\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">H</forename><surname>Yeo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">L</forename><surname>Goh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Chow</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Toxicol. Mech. Methods</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">28</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"157\" to=\"166\" />\\n\\t\\t\\t<date type=\"published\" when=\"2018\">2018</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b95\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Hoffman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">S</forename><surname>Kania</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Brothers</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">F</forename><surname>Davies</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">A</forename><surname>Ferre</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">S</forename><surname>Gajiwala</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>He</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">J</forename><surname>Hogan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Kozminski</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">Y</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">W</forename><surname>Lockner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Lou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">T</forename><surname>Marra</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">J</forename><surname>Mitchell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">W</forename><surname>Murray</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Nieman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Noell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">P</forename><surname>Planken</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Rowe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ryan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">J</forename><surname>Smith</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">E</forename><surname>Solowiej</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M</forename><surname>Steppan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Taggart</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">63</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"12725\" to=\"12747\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b96\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">W</forename><surname>Kneller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Phillips</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">L</forename><surname>Weiss</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">A</forename><surname>Arnould</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">B</forename><surname>Jonsson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Surendranathan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Parvathareddy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">P</forename><surname>Blakeley</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Coates</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">M</forename><surname>Louis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">V</forename><surname>Bonnesen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Kovalevsky</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Nat</forename></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">2268</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b97\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Fang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Duan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Feng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Shao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Peng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Rao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Yang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Protein Cell</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">13</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"689\" to=\"693\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b98\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Pfizer unveils its oral SARS-CoV-2 inhibitor</title>\\n\\t\\t<ptr target=\"https://cen.acs.org/acs-news/acs-meeting-news/Pfizer-unveils-oral-SARS-CoV/99/i13\" />\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Chemical &amp; Engineering News</title>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b99\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Hammond</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Leister-Tebbe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Gardner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Abreu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Bao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Wisemandle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Baniecki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">M</forename><surname>Hendrick</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Damle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Simón-Campos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Pypstra</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">M</forename><surname>Rusnak</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">386</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1397\" to=\"1408\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b100\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Mahase</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">BMJ</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">2021</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">2697</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b101\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Lavrijsen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Lamers</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>De Vries</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">J</forename><surname>Rottier</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Bruno</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">P</forename><surname>Peppelenbosch</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">L</forename><surname>Haagmans</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Pan</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">32</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"322\" to=\"324\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b102\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">EPIC-Peds: Study of Oral PF-07321332 (Nirmatrelvir)/Ritonavir in Nonhospitalized COVID-19 Pediatric Patients at Risk for Severe Disease -Full Text View -ClinicalTrials</title>\\n\\t\\t<ptr target=\"https://clinicaltrials.gov/ct2/show/NCT05261139\" />\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b103\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Nirmatrelvir, an Orally Active Mpro Inhibitor, Is a Potent Inhibitor of SARS-CoV-2 Variants of Concern</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">K D</forename><surname>Rai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Yurgelonis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Mcmonagle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">A</forename><surname>Rothan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Hao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Gribenko</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">E</forename><surname>Titova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Kreiswirth</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>White</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">S</forename><surname>Anderson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">D</forename><surname>Cardin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>preprint; Microbiology</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b104\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Raghavaiah</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shahabi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Yadav</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">J</forename><surname>Anson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">K</forename><surname>Lendy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Hattori</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Higashi-Kuwata</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Mitsuya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">64</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"14702\" to=\"14714\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b105\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shahabi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Yadav</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kovela</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">J</forename><surname>Anson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">K</forename><surname>Lendy</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Bonham</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Sirohi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">A</forename><surname>Brito-Sierra</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Hattori</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Kuhn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Mitsuya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Molecules</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">26</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">5782</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b106\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Pillaiyar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Flury</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Krüger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Su</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Schäkel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">B</forename><surname>Da Silva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">O</forename><surname>Eppler</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b107\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Kronenberger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Jie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Luedtke</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Rocha</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sylvester</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Petry</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">H</forename><surname>Mckerrow</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Poso</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Pöhlmann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b108\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Gütschow</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">J</forename><surname>O\\'donoghue</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">Y</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">E</forename><surname>Müller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Laufer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"9376\" to=\"9395\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b109\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Breidenbach</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Lemke</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Pillaiyar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Schäkel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">A</forename><surname>Hamwi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Diett</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b110\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Gedschold</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Geiger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Lopez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Mirza</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Namasivayam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Schiedel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sylvester</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Thimm</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">C</forename><surname>Vielmuth</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">P</forename><surname>Vu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Zyulina</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Bodem</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b111\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Gütschow</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">E</forename><surname>Müller</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Angew. Chem. Int. Ed</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">60</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"10423\" to=\"10429\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b112\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Ren</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Fan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Gu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Cheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Yu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Xie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Yin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Xie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Jiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Jin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Lancet</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">395</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"497\" to=\"506\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b113\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Su</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Yao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Shao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Xie</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Shang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Ke</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Feng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Jiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Jiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Ye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">3623</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b114\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shcherbakov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Baev</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">M</forename><surname>Kalinin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">A</forename><surname>Dalinger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Chirkova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Belenkaya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">A</forename><surname>Khvostov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Krut\\'ko</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">A</forename><surname>Medved\\'ko</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">E</forename><surname>Volosnikova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">E</forename><surname>Sharlaeva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shanshin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">T</forename><surname>Tolstikova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Yarovaya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Maksyutov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Salakhutdinov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">S</forename><surname>Vatsadze</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Med. Chem. Lett</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">13</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"140\" to=\"147\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b115\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Jacobs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Grum-Tokars</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Turlington</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Saldanha</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Chase</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Eggler</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">S</forename><surname>Dawson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">M</forename><surname>Baez-Santos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Tomar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">M</forename><surname>Mielech</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">W</forename><surname>Lindsley</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Hodder</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">R</forename><surname>Stauffer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">56</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"534\" to=\"546\" />\\n\\t\\t\\t<date type=\"published\" when=\"2013\">2013</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b116\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Kitamura</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Sacco</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Townsend</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Meng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Ba</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Szeto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Kukuljac</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">T</forename><surname>Marty</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Schultz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Cherry</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2848\" to=\"2865\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b117\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Xia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Sacco</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Townsend</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Meng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Choza</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Tan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Jang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">V</forename><surname>Gongora</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">T</forename><surname>Marty</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Am. Chem. Soc</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">143</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"20697\" to=\"20709\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b118\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C.-H</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">A</forename><surname>Stone</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Deshmukh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Ippolito</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Ghahremanpour</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Tirado-Rives</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">A</forename><surname>Spasov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Takeo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">N</forename><surname>Kudalkar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Isaacs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Lindenbach</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">J</forename><surname>Miller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">S</forename><surname>Anderson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">L</forename><surname>Jorgensen</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Cent. Sci</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">7</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"467\" to=\"475\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b119\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">H</forename><surname>Han</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M</forename><surname>Goins</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Arya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W.-J</forename><surname>Shin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Maw</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Hooper</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">P</forename><surname>Sonawane</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Porter</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">E</forename><surname>Bannister</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">D</forename><surname>Crouch</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">A</forename><surname>Lindsey</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Lakatos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">R</forename><surname>Martinez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Alvarado</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">S</forename><surname>Akers</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">S</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">U</forename><surname>Jung</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">D</forename><surname>Macdonald</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">R</forename><surname>Stauffer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2880\" to=\"2904\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b120\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Unoh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Uehara</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Nakahara</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Nobori</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Yamatsu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Yamamoto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Maruyama</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Taoda</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Kasamatsu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Suto</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Kouki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Nakahashi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kawashima</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Sanaki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Toba</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Uemura</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Mizutare</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Ando</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Sasaki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Orba</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Sawa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Sato</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Sato</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Kato</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Tachibana</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"6499\" to=\"6512\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b121\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">D A</forename><surname>Tyndall</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"6496\" to=\"6498\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b122\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Mukherjee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Grewe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Bojkova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Baek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Bhattacharya</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Schulz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Widera</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">R</forename><surname>Mehdipour</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Tascher</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">P</forename><surname>Geurink</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Wilhelm</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">J</forename><surname>Van Der Heden Van Noort</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Ovaa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Muller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K.-P</forename></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b123\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Knobeloch</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">A</forename><surname>Rajalingam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Schulman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Cinatl</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Hummer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Ciesek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Dikic</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nature</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">587</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"657\" to=\"662\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b124\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Kilianski</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">M</forename><surname>Baez-Santos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">PLoS Pathog</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2014\">2014, 10, e1004113</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b125\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-W</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C.-Y</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-J</forename><surname>Jou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-H</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L.-H</forename><surname>Hsiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Wan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-J</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-H</forename><surname>Kung</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C.-W</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Int. J. Mol. Sci</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">17</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">678</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2016\">2016</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b126\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Pegan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Takayama</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sleeman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Coughlin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Baliji</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Chaudhuri</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Fu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">S</forename><surname>Prabhakar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Johnson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Proc. Natl., Acad. Sci. U S A</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">105</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"16119\" to=\"16124\" />\\n\\t\\t\\t<date type=\"published\" when=\"2008\">2008</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b127\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Ghosh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Takayama</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Aubin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Chaudhuri</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Baez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sleeman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Coughlin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">B</forename><surname>Nichols</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">C</forename><surname>Mulhearn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">S</forename><surname>Prabhakar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">C</forename><surname>Baker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">E</forename><surname>Johnson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Mesecar</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">52</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"5228\" to=\"5240\" />\\n\\t\\t\\t<date type=\"published\" when=\"2009\">2009</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b128\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Osipiuk</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-A</forename><surname>Azizi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Dvorkin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Endres</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Jedrzejczak</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">A</forename><surname>Jones</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">S</forename><surname>Kathayat</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">G</forename><surname>Lisnyak</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">L</forename><surname>Maki</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Nicolaescu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">A</forename><surname>Taylor</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Tesar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-A</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Randall</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Michalska</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Snyder</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">C</forename><surname>Dickinson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Joachimiak</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">743</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b129\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Fu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Tang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Ye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Xiong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Niu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Zhou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">J</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">488</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b130\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Gao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Qin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Zhu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Hou</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Wojdyla</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Cui</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Acta Pharm Sin B</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">11</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"237\" to=\"245\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b131\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ratia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Cooper</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Kong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Lee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Kwon</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Alqarni</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Huang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Dubrovskyi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Rong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">R</forename><surname>Thatcher</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Xiong</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2940\" to=\"2955\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b132\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Shan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Dai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Lu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Han</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Tong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Xiang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">Z</forename><surname>Ai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Zhuang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Pan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">L</forename><surname>Tan</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Cell Chemical Biology</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">28</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"855\" to=\"865\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b133\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b134\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Allosteric Activation of SARS-CoV-2 RNA-Dependent RNA Polymerase by Remdesivir Triphosphate and Other Phosphorylated Nucleotides</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Svetlov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><forename type=\"middle\">I</forename><surname>Wolf</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">V</forename><surname>Koonin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">;</forename><forename type=\"middle\">E</forename><surname>Nudler</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Artsimovitch</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e0142321</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b135\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Kabinger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Stiller</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Schmitzová</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Dienemann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Kokic</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">S</forename><surname>Hillen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Höbartner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Cramer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Struct. Mol. Biol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">28</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"740\" to=\"746\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b136\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Eroglu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Toprak</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Int. J. Pharm. Sci. Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">2021</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1950\" to=\"1957\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b137\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Olender</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">L</forename><surname>Walunas</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Martinez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">K</forename><surname>Perez</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Castagna</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Kurbegov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Goyal</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Ripamonti</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Balani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">G</forename><surname>De Rosa</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>De Wit</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-W</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Diaz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Bruno</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>Mullane</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">C</forename><surname>Lye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Gottlieb</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">H</forename><surname>Haubrich</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">P</forename><surname>Chokkalingam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Diaz-Cuervo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">M</forename><surname>Brainard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I.-H</forename><surname>Lee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">O</forename><surname>Osinusi</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. I. Bernardino, M. Boffito, Open Forum Infect. Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">8</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">278</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b138\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Kokic</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">S</forename><surname>Hillen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Tegunov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Dienemann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Seitz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Schmitzova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Farnung</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Siewert</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Höbartner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Cramer</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">279</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b139\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">Y</forename><surname>Ghazwani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b140\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">H</forename><surname>Bakheit</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b141\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">R</forename><surname>Hakami</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">M</forename><surname>Alkahtani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">A</forename></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Almehizia. Crystals</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">471</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b142\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">D</forename><surname>Spinner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Gottlieb</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">J</forename><surname>Criner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">R</forename><surname>Arribas López</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">M</forename><surname>Cattelan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Soriano</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Viladomiu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Ogbuagu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Malhotra</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Mullane</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">Y A</forename><surname>Castagna</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Chai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><forename type=\"middle\">T Y</forename><surname>Roestenberg</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Tsang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Bernasconi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-C</forename><surname>Le Turnier</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Chang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">H</forename><surname>Sengupta</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">O</forename><surname>Hyland</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Osinusi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Blair</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">M</forename><surname>Gaggar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Brainard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Mcphail</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">Y</forename><surname>Bhagani</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">J</forename><surname>Ahn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Sanyal</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">M</forename><surname>Huhn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Marty</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">JAMA</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">324</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1048\" to=\"1057\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>for the GS-US-540-5774 Investigators</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b143\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">D</forename><surname>Goldman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">C B</forename><surname>Lye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">S</forename><surname>Hui</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>Marks</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Bruno</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Montejano</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">D</forename><surname>Spinner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Galli</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M.-Y</forename><surname>Ahn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">G</forename><surname>Nahass</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y.-S</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Sengupta</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">H</forename><surname>Hyland</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">O</forename><surname>Osinusi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Blair</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Wei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Gaggar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">M</forename><surname>Brainard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><forename type=\"middle\">J</forename><surname>Towner</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Muñoz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>Mullane</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><forename type=\"middle\">M</forename><surname>Marty</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">T</forename><surname>Tashima</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Diaz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Subramanian</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">383</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1827\" to=\"1837\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b144\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Grein</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Ohmagari</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Shin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Diaz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Asperges</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Castagna</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Feldt</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Green</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">L</forename><surname>Green</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F.-X</forename><surname>Lescure</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Nicastri</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Oda</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Yo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Quiros-Roldan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Studemeister</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Redinski</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Ahmed</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Bernett</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Chelliah</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Chihara</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">H</forename><surname>Cohen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Cunningham</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>D\\'arminio Monforte</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Ismail</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Kato</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Lapadula</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Her</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Maeno</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Majumder</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Massari</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Mora-Rillo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Mutoh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Nguyen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Verweij</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Zoufaly</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">O</forename><surname>Osinusi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Dezure</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Zhong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Chokkalingam</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Elboudwarej</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Telep</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Timbs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Henne</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Sellers</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">K</forename><surname>Tan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Winterbourne</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Desai</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Mera</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Gaggar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">P</forename><surname>Myers</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">M</forename><surname>Brainard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Childs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Flanigan</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">382</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2327\" to=\"2336\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b145\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">M</forename><surname>Cox</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">D</forename><surname>Wolf</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">M</forename><surname>Lieber</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Sourimant</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Babusis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Dupont</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Chan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">T</forename><surname>Barrett</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Lye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Kalla</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Chun</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Mackman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Ye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Cihlar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Martinez-Sobrido</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">L</forename><surname>Greninger</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">P</forename><surname>Bilello</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">K</forename><surname>Plemper</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">6415</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b146\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Yi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Ma</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Lin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Cen</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antiviral Res</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021, 190, 105078</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b147\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">A Phase 2 Randomized, Double Blinded, Placebo Controlled Study of Oral Favipiravir Compared to Standard Supportive Care in Subjects With Mild or Asymptomatic COVID-19</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t\\t<respStmt>\\n\\t\\t\\t<orgName>Stanford University</orgName>\\n\\t\\t</respStmt>\\n\\t</monogr>\\n\\t<note>Clinical trial registration NCT04346628; clinicaltrials.gov</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b148\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Open Label, Randomized, Controlled Phase 2 Proof-of-Concept Study of the Use of Favipiravir Compared to Standard of Care in Hospitalized Subjects With COVID-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Fujifilm</forename><surname>Pharmaceuticals</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">U</forename><forename type=\"middle\">S A</forename><surname>Inc</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>Clinical trial registration NCT04358549; clinicaltrials.gov</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b149\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Favipiravir for Patients With Mild to Moderate Disease From Novel Coronavirus (COVID-19); Clinical trial registration NCT04600895; clinicaltrials.gov</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t\\t<publisher>Appili Therapeutics Inc</publisher>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b150\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Shannon</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Selisko</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N.-T.-T</forename><surname>Le</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Huchting</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Touret</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Piorkowski</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Fattorini</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Ferron</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Decroly</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Meier</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Coutard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Peersen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Canard</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">4682</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b151\">\\n\\t<monogr>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Naydenova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">W</forename><surname>Muir</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L.-F</forename><surname>Wu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Coscia</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">J</forename><surname>Peet</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Castro-Hartmann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Qian</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Sader</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Dent</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Kimanius</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">D</forename><surname>Sutherland</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Löwe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Barford</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">J</forename><surname>Russo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<title level=\"m\">Proc. Natl. Acad. Sci. 2021</title>\\n\\t\\t\\t\\t<meeting>Natl. Acad. Sci. 2021</meeting>\\n\\t\\t<imprint>\\n\\t\\t\\t<date>e2021946118</date>\\n\\t\\t\\t<biblScope unit=\"volume\">118</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b152\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">A Phase 2/3, Randomized, Placebo-Controlled, Double-Blind Clinical Study to Evaluate the Efficacy, Safety, and Pharmacokinetics of MK-4482 in Non-Hospitalized Adults With COVID-19</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Merck</forename><surname>Sharp</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Dohme</forename><surname>Corp</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n\\t<note>Clinical trial registration NCT04575597; clinicaltrials.gov</note>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b153\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Imran</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">K</forename><surname>Arora</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">M B</forename><surname>Asdaq</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">A</forename><surname>Khan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">I</forename><surname>Alaqel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">K</forename><surname>Alshammari</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Alshehri</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Molecules</title>\\n\\t\\t<editor>A. S. Alshrari, A. Mateq Ali, A. M. Alshammeri, B. D. Alhazmi, A. A. Harshan, Md. T. Alam, A. Abida</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">26</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">5795</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b154\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Abdelnabi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">S</forename><surname>Foo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">J F</forename><surname>Kaptein</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">N D</forename><surname>Do</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Langendries</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Vangeel</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Breuer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Pang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Williams</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Vergote</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Heylen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Leyssen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Dallmeier</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Coelmont</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Chatterjee</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Mols</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Augustijns</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>De Jonghe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Jochmans</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Weynand</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Neyts</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t\\t<biblScope unit=\"volume\">72</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">103595</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b155\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">A</forename><surname>Elfiky</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Biomol. Struct. Dyn</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">2020</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1\" to=\"9\" />\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b156\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Yi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Lei</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Cui</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Xiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Jiao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Dong</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Zeng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Ren</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Cen</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Acta Pharm. Sin. B</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">11</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1555\" to=\"1567\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b157\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Zandi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Musall</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Oo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Hassandarvish</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Lan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Slack</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">A</forename><surname>Kirby</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Bassit</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Amblard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Abubakar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">G</forename><surname>Sarafianos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">F</forename><surname>Schinazi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint>\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t\\t<biblScope unit=\"volume\">9</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">893</biblScope>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b158\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Zandi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Amblard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Musall</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Downs-Bowen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b159\">\\n\\t<analytic>\\n\\t\\t<title level=\"a\" type=\"main\">Repurposing Nucleoside Analogs for Human Coronaviruses</title>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Kleinbard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">D</forename><surname>Oo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Cao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><forename type=\"middle\">O</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Russell</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Mcbrayer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Bassit</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">F</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><surname>Schinazi</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Antimicrob. Agents Chemother</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"e01652\" to=\"1672\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b160\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Krafcikova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Silhan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Nencka</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Boura</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Commun</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"page\">3717</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b161\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Rosas-Lemus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Minasov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Shuvalova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">L</forename><surname>Inniss</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Kiryukhina</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Brunzelle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">J F</forename><surname>Satchell</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Sci. Signal</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">13</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">e1202</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b162\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Otava</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Šála</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Fanfrlík</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Devkota</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Perveen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Chau</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Pakarian</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Hobza</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Vedadi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Boura</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Nencka</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Infect. Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">7</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2214\" to=\"2220\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b163\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Bobiļeva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Bobrovs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Kaņepe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Patetko</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Kalniņš</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Šišovs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">L</forename><surname>Bula</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Gri ̅ Nberga</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Borodušķis</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Ramata-Stunda</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Rostoks</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Jirgensons</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Ta ̅ Rs</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Jaudzems</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Med. Chem. Lett</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1102\" to=\"1107\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b164\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><surname>Ahmed-Belkacem</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Hausdorff</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Delpal</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Sutto-Ortiz</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">M G</forename><surname>Colmant</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Touret</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">S</forename><surname>Ogando</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">J</forename><surname>Snijder</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Canard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Coutard</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J.-J</forename><surname>Vasseur</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Decroly</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Debart</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno type=\"DOI\">10.1002/cmdc.202200440</idno>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">J. Med. Chem</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">65</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"6231\" to=\"6249\" />\\n\\t\\t\\t<date type=\"published\" when=\"2022\">2022</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b165\">\\n\\t<monogr>\\n\\t\\t<title level=\"m\" type=\"main\">Accepted Manuscript ChemMedChem This article is protected by copyright. All rights reserved</title>\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b166\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">C</forename><surname>Taylor</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Adams</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">M</forename><surname>Hufford</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>De La Torre</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Winthrop</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">L</forename><surname>Gottlieb</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Nat. Rev. Immunol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">21</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"382\" to=\"393\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b167\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">Z</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Itkin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Shinn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Ivin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Leek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">J</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Hall</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Med. Chem. Lett</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1267\" to=\"1274\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b168\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Bojadzic</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Alcazar</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S.-T</forename><surname>Chuang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">M</forename><surname>Condor Capcha</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><forename type=\"middle\">A</forename><surname>Shehadeh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Buchwald</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Infect. Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">7</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1519\" to=\"1534\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b169\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">R</forename><surname>Straus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">K</forename><surname>Bidon</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Tang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">A</forename><surname>Jaimes</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">R</forename><surname>Whittaker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Daniel</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Infect. Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">7</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"2807\" to=\"2815\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b170\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">Z</forename><surname>Chen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Xu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Hu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Guo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Itkin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Shinn</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Ivin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Leek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">J</forename><surname>Liang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Shen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zheng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">D</forename><surname>Hall</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">ACS Med. Chem. Lett</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">12</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1267\" to=\"1274\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b171\">\\n\\t<monogr>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Kinase</forename><surname>Inhibitors</surname></persName>\\n\\t\\t</author>\\n\\t\\t<ptr target=\"https://www.covid19treatmentguidelines.nih.gov/therapies/immunomodulators/kinase-inhibitors/\" />\\n\\t\\t<title level=\"m\">COVID-19 Treatment Guidelines</title>\\n\\t\\t\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b172\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">W</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Zhao</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">F</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Q</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Li</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Z</forename><surname>Liu</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Wang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Qin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Yan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">X</forename><surname>Zeng</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Zhang</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Clin. Immunol</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">214</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\">108393</biblScope>\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b173\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Melikhov</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><surname>Kruglova</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><surname>Lytkina</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Melkonyan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Prokhorovich</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Putsman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><surname>Rodoman</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Vertkin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Zagrebneva</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Stebbing</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Ann. Rheum. Dis</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">80</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"1245\" to=\"1246\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b174\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><surname>Richardson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">I</forename><surname>Griffin</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Tucker</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><surname>Smith</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">O</forename><surname>Oechsle</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><surname>Phelan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Rawling</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><surname>Savory</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><surname>Stebbing</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">Lancet</title>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">395</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"e30\" to=\"e31\" />\\n\\t\\t\\t<date type=\"published\" when=\"2020\">2020</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b175\">\\n\\t<monogr>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><surname>Research</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">E</forename><surname>Coronavirus</surname></persName>\\n\\t\\t</author>\\n\\t\\t<idno>COVID-19) | Drugs. FDA</idno>\\n\\t\\t<ptr target=\"https://www.fda.gov/drugs/emergency-preparedness-drugs/coronavirus-covid-19-drugs\" />\\n\\t\\t<imprint/>\\n\\t</monogr>\\n</biblStruct>\\n\\n<biblStruct xml:id=\"b176\">\\n\\t<analytic>\\n\\t\\t<title/>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">C</forename><surname>Kalil</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">T</forename><forename type=\"middle\">F</forename><surname>Patterson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">K</forename><surname>Mehta</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">K</forename><forename type=\"middle\">M</forename><surname>Tomashek</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">R</forename><surname>Wolfe</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Ghazaryan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">C</forename><surname>Marconi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">G</forename><forename type=\"middle\">M</forename><surname>Ruiz-Palacios</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Hsieh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><surname>Kline</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><surname>Tapson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">M</forename><surname>Iovine</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><forename type=\"middle\">K</forename><surname>Jain</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">A</forename><surname>Sweeney</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><forename type=\"middle\">M</forename><surname>El Sahly</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">R</forename><surname>Branche</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">J</forename><forename type=\"middle\">Regalado</forename><surname>Pineda</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">D</forename><forename type=\"middle\">C</forename><surname>Lye</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">U</forename><surname>Sandkovsky</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">A</forename><forename type=\"middle\">F</forename><surname>Luetkemeyer</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">H</forename><surname>Cohen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">W</forename><surname>Finberg</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">E H</forename><surname>Jackson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><surname>Taiwo</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">C</forename><forename type=\"middle\">I</forename><surname>Paules</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Arguinchona</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Erdmann</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><surname>Ahuja</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Frank</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">M</forename><surname>Oh</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E.-S</forename><surname>Kim</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">S</forename><forename type=\"middle\">Y</forename><surname>Tan</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">R</forename><forename type=\"middle\">A</forename><surname>Mularski</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">H</forename><surname>Nielsen</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">P</forename><forename type=\"middle\">O</forename><surname>Ponce</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">B</forename><forename type=\"middle\">S</forename><surname>Taylor</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">L</forename><surname>Larson</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">N</forename><forename type=\"middle\">G</forename><surname>Rouphael</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">Y</forename><surname>Saklawi</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">V</forename><forename type=\"middle\">D</forename><surname>Cantos</surname></persName>\\n\\t\\t</author>\\n\\t\\t<author>\\n\\t\\t\\t<persName><forename type=\"first\">E</forename><forename type=\"middle\">R</forename><surname>Ko</surname></persName>\\n\\t\\t</author>\\n\\t</analytic>\\n\\t<monogr>\\n\\t\\t<title level=\"j\">N. Engl. J. Med</title>\\n\\t\\t<editor>J. J. Engemann, A. N. Amin, M. Watanabe, J. Billings, M.-C. Elie, R. T. Davey, T. H. Burgess, J. Ferreira, M. Green, M. Makowski, A. Cardoso, S. de Bono, T. Bonnett, M. Proschan, G. A. Deye, W. Dempsey, S. U. Nayak, L. E. Dodd, J. H. Beigel</editor>\\n\\t\\t<imprint>\\n\\t\\t\\t<biblScope unit=\"volume\">384</biblScope>\\n\\t\\t\\t<biblScope unit=\"page\" from=\"795\" to=\"807\" />\\n\\t\\t\\t<date type=\"published\" when=\"2021\">2021</date>\\n\\t\\t</imprint>\\n\\t</monogr>\\n</biblStruct>\\n\\n\\t\\t\\t\\t</listBibl>\\n\\t\\t\\t</div>\\n\\t\\t</back>\\n\\t</text>\\n</TEI>\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 24
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_pmc_search_results_no_full_text_exceptions_xml_v2.pickle', 'wb') as f:\n",
        "  pickle.dump(exceptions_xml, f)"
      ],
      "metadata": {
        "id": "v_sADyrbov6u"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "##  10. Clean and return XML as Beautiful Soup object\n",
        "Function to remove xmlns attribute which specifies the XML namespace and return the cleaned, parsed text for the body tag as a Beautiful Soup object. This object represents the document as a nested data structure so that we can navigate and extract text using XML tags."
      ],
      "metadata": {
        "id": "2UjTGY4FgECB"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "# remove xmlns attributes from XML and convert to soup object\n",
        "\n",
        "def soupify(text):\n",
        "    cleaned_text = re.sub('\\s*xmlns(:\\w+)?=\\\"[^\\\"]*\\\"', '', text)\n",
        "    return  BeautifulSoup(cleaned_text, 'lxml-xml').find(\"body\")"
      ],
      "metadata": {
        "id": "VpI9fTo6kIbZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_exceptions_xml = list(map(soupify, exceptions_xml))"
      ],
      "metadata": {
        "id": "s8PL6WXXkRKM"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_exceptions_xml)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "j6h_9WB3ka3z",
        "outputId": "0edeeafc-6018-4a68-f3ae-1d176587b11c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "118"
            ]
          },
          "metadata": {},
          "execution_count": 60
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Check the first article has been converted into a Tag object, which corresponds to the `<body>` tag in the original document."
      ],
      "metadata": {
        "id": "cru1XSyyhs5F"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "type(soup_results_exceptions_xml[0])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4QRdPUonkrUl",
        "outputId": "6d11ceaf-3a48-4040-cfd5-de7ffe7e124e"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "bs4.element.Tag"
            ]
          },
          "metadata": {},
          "execution_count": 61
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "We can navigate the parse tree using tag names e.g. head."
      ],
      "metadata": {
        "id": "jgvgKKj7nQXZ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_exceptions_xml[0].head"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "R4ED6buykwT6",
        "outputId": "9d0cd470-ca4f-48f3-ffc5-0a06cde6c09b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<head n=\"1.\">Introduction</head>"
            ]
          },
          "metadata": {},
          "execution_count": 62
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 10.1 Find all tags including descendants\n",
        "\n",
        "\n",
        "The function below uses the find_all() method to return all of the direct child and descendant tags of the `<body>` tag for each article."
      ],
      "metadata": {
        "id": "ydx3F-Gqk8sz"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_all_tags(article):\n",
        "\n",
        "    tags_list = []\n",
        "\n",
        "    try:\n",
        "        for tag in article.find_all(True):\n",
        "            if tag.name != None:\n",
        "              tags_list.append(tag.name)\n",
        "    except Exception as e:\n",
        "      print(e)\n",
        "\n",
        "    return sorted(list(set(tags_list)))"
      ],
      "metadata": {
        "id": "nT2AGG3-lD8l"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "all_tags = list(map(find_all_tags, soup_results_exceptions_xml))"
      ],
      "metadata": {
        "id": "3-T5vKO1m5nA"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "There should be no exceptions this time and we can print out all the child and descendant tags."
      ],
      "metadata": {
        "id": "aZ_tcBR1A_NR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "all_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "V5Tsdfy3nbfu",
        "outputId": "57eea845-f94c-4a38-d243-3376e68cc618"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'head', 'p'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " [],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'formula', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref'],\n",
              " ['div', 'head', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'note', 'p', 'ref', 'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['div', 'head', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'head', 'note', 'p'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'note'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table'],\n",
              " ['div', 'figDesc', 'figure', 'graphic', 'head', 'label', 'p', 'ref'],\n",
              " ['cell',\n",
              "  'div',\n",
              "  'figDesc',\n",
              "  'figure',\n",
              "  'formula',\n",
              "  'graphic',\n",
              "  'head',\n",
              "  'label',\n",
              "  'note',\n",
              "  'p',\n",
              "  'ref',\n",
              "  'row',\n",
              "  'table']]"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in all_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "Z7rGKFKbB3iV"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_all_tags = sum(all_tags, [])\n",
        "sum_all_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jAk3qVk8nhBh",
        "outputId": "16961b70-3522-43c3-949c-6e9ea4454814"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'head',\n",
              " 'p',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'head',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'formula',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'note',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'cell',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'row',\n",
              " 'table',\n",
              " 'div',\n",
              " 'figDesc',\n",
              " 'figure',\n",
              " 'graphic',\n",
              " 'head',\n",
              " 'label',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'cell',\n",
              " 'div',\n",
              " ...]"
            ]
          },
          "metadata": {},
          "execution_count": 13
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Counter class with additional most_common() method to return a list of the n most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "GGz8iKrhB8R3"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_all_unique_tags = Counter(sum_all_tags).most_common()\n",
        "num_all_unique_tags = num_all_unique_tags[::]\n",
        "\n",
        "for tag, count in num_all_unique_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3M7G6vyJn1kL",
        "outputId": "12bd5fed-f421-4767-d028-9012d44a98de"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "div 117\n",
            "p 116\n",
            "head 115\n",
            "ref 113\n",
            "figDesc 111\n",
            "figure 111\n",
            "label 111\n",
            "graphic 89\n",
            "table 83\n",
            "cell 79\n",
            "row 79\n",
            "note 78\n",
            "formula 32\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 10.2 Find direct child tags of `<body>` tag\n",
        "\n",
        "Every tag has a name which can be accessed using the `.name` attribute.\n",
        "\n",
        "We will use this to find the direct child tags of the `<body>` tag.\n",
        "\n",
        "\n"
      ],
      "metadata": {
        "id": "Hxi7ZzkroQrC"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_body_child_tags(article):\n",
        "\n",
        "    body_child_list = []\n",
        "\n",
        "    try:\n",
        "        for tag in article:\n",
        "            if tag.name != None:\n",
        "                  body_child_list.append(tag.name)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "    return sorted(list(set(body_child_list)))"
      ],
      "metadata": {
        "id": "3aV1H13HoQrC"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "body_child_tags = list(map(find_body_child_tags, soup_results_exceptions_xml))"
      ],
      "metadata": {
        "id": "biMP9PfwoQrD"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "We can see below that the `<body>` tag has direct child tags for `<div>`, `<figure>` and `<note>`."
      ],
      "metadata": {
        "id": "kN9hmsi6CgdH"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "body_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "bd66d20e-06c6-44ce-bd4f-b1526bbb47b9",
        "id": "cZnsOxbGoQrD"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " [],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure', 'note'],\n",
              " ['div', 'figure'],\n",
              " ['div', 'figure', 'note']]"
            ]
          },
          "metadata": {},
          "execution_count": 17
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in body_child_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "xn3TVpOSCoHw"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_body_child_tags = sum(body_child_tags, [])\n",
        "sum_body_child_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9c60211e-e71d-4877-811b-fa00b7200919",
        "id": "_P6pe6vjoQrD"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'div',\n",
              " 'figure',\n",
              " 'note']"
            ]
          },
          "metadata": {},
          "execution_count": 18
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again use the Counter class with most_common() method to return a list of the n most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "FdvtBFBECsmW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_body_child_tags = Counter(sum_body_child_tags).most_common()\n",
        "num_unique_body_child_tags = num_unique_body_child_tags[::]\n",
        "\n",
        "for tag, count in num_unique_body_child_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "05b21b9b-a204-41d8-dca1-47d6c8206150",
        "id": "LKZJBTZnoQrE"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "div 117\n",
            "figure 111\n",
            "note 51\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 10.3 Find direct child tags of `<div>` tags\n",
        "\n",
        "Function to find all `<div>` tags and append all direct child tags to a list."
      ],
      "metadata": {
        "id": "I47ftk-JpFvd"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def find_div_tags(article):\n",
        "\n",
        "    div_tags = []\n",
        "\n",
        "    try:\n",
        "        divs = article.find_all(\"div\")\n",
        "        for div in divs:\n",
        "          for tag in div:\n",
        "              if tag.name != None:\n",
        "                  div_tags.append(tag.name)\n",
        "    except Exception as e:\n",
        "        print(e)\n",
        "\n",
        "    return sorted(list(set(div_tags)))"
      ],
      "metadata": {
        "id": "UQo9gdnfpFvd"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "div_tags = list(map(find_div_tags, soup_results_exceptions_xml))"
      ],
      "metadata": {
        "id": "0nv_eqJlpFvd"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "div_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4ea28187-8d2f-4d4b-fb51-13bb6332b8d5",
        "id": "YcOoTts2pFve"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " [],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p', 'ref'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'note', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p', 'ref'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p'],\n",
              " ['head', 'p'],\n",
              " ['head', 'p'],\n",
              " [],\n",
              " ['formula', 'head', 'note', 'p'],\n",
              " ['head', 'p'],\n",
              " ['formula', 'head', 'p']]"
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Sum function to take in div_tags nested list and return the sum of all elements as one list."
      ],
      "metadata": {
        "id": "KSNzPJuxDAMz"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "sum_div_tags = sum(div_tags, [])\n",
        "sum_div_tags"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "93d758cc-8d63-4b52-9f7c-dd4cc352e5fe",
        "id": "1KKonpZ3pFve"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "['head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'ref',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'note',\n",
              " 'p',\n",
              " 'head',\n",
              " 'p',\n",
              " 'formula',\n",
              " 'head',\n",
              " 'p']"
            ]
          },
          "metadata": {},
          "execution_count": 23
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Again use the Counter class with most_common() method to return a list of the n most common elements and their counts from the most common to the least."
      ],
      "metadata": {
        "id": "u29prrr9DExp"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "num_unique_div_tags = Counter(sum_div_tags).most_common()\n",
        "num_unique_div_tags = num_unique_div_tags[::]\n",
        "\n",
        "for tag, count in num_unique_div_tags:\n",
        "  print(tag, count)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4045eedb-3e1d-459e-b257-5e8b64f77a09",
        "id": "XIN_aAmPpFvg"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "p 116\n",
            "head 112\n",
            "formula 32\n",
            "note 10\n",
            "ref 2\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 11. Remove unwanted tags and content\n",
        "\n",
        "Define functions to remove unwanted tags and their contents using decompose() method which removes a tag from the tree, then completely destroys it and its contents."
      ],
      "metadata": {
        "id": "prqF1C40p33K"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def remove_headings(article):\n",
        "\n",
        "    for head in article(\"head\"):\n",
        "        head.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_figures(article):\n",
        "\n",
        "    for figure in article(\"figure\"):\n",
        "        figure.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_tables(article):\n",
        "\n",
        "    for table in article(\"table\"):\n",
        "        table.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "def remove_formulas(article):\n",
        "\n",
        "    for formula in article(\"formula\"):\n",
        "        formula.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_labels(article):\n",
        "\n",
        "    for label in article(\"label\"):\n",
        "        label.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_refs(article):\n",
        "\n",
        "    for ref in article(\"ref\"):\n",
        "        ref.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_graphics(article):\n",
        "\n",
        "    for graphic in article(\"graphic\"):\n",
        "        graphic.decompose()\n",
        "\n",
        "    return article\n",
        "\n",
        "\n",
        "def remove_notes(article):\n",
        "\n",
        "    for note in article(\"note\"):\n",
        "        note.decompose()\n",
        "\n",
        "    return article"
      ],
      "metadata": {
        "id": "Q0mwiy3Jp33L"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Function to call the functions above on each article and return new article list."
      ],
      "metadata": {
        "id": "ROP0eJ2yDjKT"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def remove_tags(article):\n",
        "\n",
        "    new_article_list = []\n",
        "\n",
        "    if article != None:\n",
        "        try:\n",
        "              new_article = remove_headings(article)\n",
        "              new_article = remove_figures(new_article)\n",
        "              new_article = remove_tables(new_article)\n",
        "              new_article = remove_formulas(new_article)\n",
        "              new_article = remove_labels(new_article)\n",
        "              new_article = remove_refs(new_article)\n",
        "              new_article = remove_graphics(article)\n",
        "              new_article = remove_notes(article)\n",
        "              new_article_list.append(new_article)\n",
        "        except Exception as e:\n",
        "            print(e)\n",
        "\n",
        "    return new_article_list"
      ],
      "metadata": {
        "id": "Y6mdMUIAp33L"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_exceptions_removed_tags = list(map(remove_tags, soup_results_exceptions_xml))"
      ],
      "metadata": {
        "id": "aiqT2Jyxp33M"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(soup_results_exceptions_removed_tags)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4cfaea00-ed95-44ad-e8c1-52338a85242e",
        "id": "PTEk6wEqp33M"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "118"
            ]
          },
          "metadata": {},
          "execution_count": 28
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View article with unwanted tags and contents removed."
      ],
      "metadata": {
        "id": "87sXqm6GDqG_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "soup_results_exceptions_removed_tags[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "01aca3b5-0f9e-473c-d5db-1e1184547202",
        "id": "eUM5XPJvp33M"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[<body>\n",
              " <div><p>Severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) originated in Wuhan, China in late December 2019.  This outbreak began spreading at an alarming rate, and unleashed a severe health crisis around the globe. Subsequently, the uncertainties caused a serious economic meltdown worldwide. On March 11, 2020, the World Health Organization (WHO) declared the novel coronavirus (COVID-19) outbreak a global pandemic.  Since then, it has gone on to affect millions of lives across the globe and caused nearly 6.3 million deaths as of June 7, 2022.  Human-human transmission for SARS-CoVs occurs primarily via respiratory droplets through sneezing, coughing, or close contact between persons. Mild symptomatic cases may include: fever, fatigue, dyspnea.  More severe cases of SARS-CoV-2 develop pneumonia, acute respiratory distress, and hypoxia.  Early on, many laboratories around the world got involved in the development of COVID-19 therapeutics. These include, development of therapies through drug repurposing, monoclonal antibody-based treatment, convalescent plasma therapies, vaccine therapies, and target-based drug development.  We will focus our review on recent efforts and updates on new drug development since our prior review in 2020.  Coronaviruses are the largest known single stranded RNA viruses that infect animals and humans leading to respiratory, gastrointestinal, hepatic, and neurologic diseases.  There are currently four different genera of human coronaviruses (HCoVs): alpha-coronavirus, beta-coronavirus, gamma-coronavirus, and delta-coronavirus. SARS-CoV-2 is the seventh human coronavirus. Another six have been identified: alpha-CoVs-NL63 and HCoVs-229E, and beta-CoVs HCoVs-OC43, HCoVs-HKU1, severe acute respiratory syndrome-CoV (SARS-CoV-1), and Middle East respiratory syndrome-CoV (MERS-CoV). These have been reviewed extensively.  Recently, a newly identified alphacoronavirus was identified in patients hospitalized with pneumonia.  This new human coronavirus turns out to be a novel \"canine-feline coronavirus\" that is called CCoV-human pneumonia (HuPn)-2018, and additional studies have now turned up even more of these hybrid alpha-coronaviruses around the globe.  Sequence analysis suggests that these are novel caninefeline-porcine-like (CFPL) CoVs of Alphacoronavirus 1 species that can infect humans (hCFPL-CoVs) and are associated with acute respiratory illness.</p><p>The first SARS outbreak (SARS-CoV-1) originated in the Guandong Province, China in 2002.  It spread to several Asian countries, North America, and Europe. However, it was promptly contained and affected more than 8,000 individuals with 774 deaths and a 10% mortality rate.  There is a 82% genome similarity between SARS-CoV-1 and SARS-CoV-2 and a 90% resemblance in many essential enzymes. With the arrival of SARS-CoV-1, there was significant medicinal chemistry efforts geared toward developing concepts and strategies to block SARS-CoV-1 replication and to develop target-based therapeutics. Indeed, much of the ground work for blocking essential enzyme functions provided insights and a head start for the development of mechanism-based drugs against SARS-CoV-2 infection and COVID-19.  We will highlight here recent developments and approaches to SARS-CoV-2 and Covid-19 therapeutics.</p></div>\n",
              " <div><p>SARS-CoV-2 binds to the host cell surface receptor, angiotensin-converting enzyme 2 (ACE2) to enter the cell. After cell entry, viral RNA attaches to the host cell ribosome to produce two polyproteins, pp1a and pp1ab, which are subsequently cleaved by the coronavirus main protease, or 3-chymotrypsin-like protease (3CLpro), and papain-like protease (PLpro).  Cleavage of these polyproteins leads to the formation of several essential enzymes such as RNA-dependent RNA polymerase (RdRp), which is responsible for genome replication.  As shown in Figure , many of these enzymes have all been subjected to intensive research as targets for SARS-CoV antiviral therapies.</p><p>Two main targets for antiviral therapies are two cysteine proteases, PLpro and 3CLpro, also known as the main protease (Mpro).  Both proteases are responsible for processing 16 nonstructural proteins (nsps), which are essential for viral replication and maturation.  The sequence identity for 3CLpro between SARS-CoV-1 and SARS-CoV-2 is 96%. 3CLpro is responsible for cleaving the polyprotein and generating functional viral proteins such as: RdRp, RNA binding proteins, exoribonuclease, helicase, and methyltransferase.  3CLpro is active as a homodimer and contains a catalytic dyad of Cys-His, whereas PLpro has a catalytic triad of Cys-His-Asp.  PLpro has an 83% sequence identity similarity between SARS-CoV-1 and SARS-CoV-2. PLpro is responsible for processing the replicase polyprotein and removing cellular substrates such as ubiquitin, termed deubiquitylation, and interferon-stimulated gene product 15 from host cell proteins.  This allows the virus to escape the host innate immune system. Other important drug design targets for COVID-19 treatment include, the spike/ACE2 viral attachment, cell entry, viral helicase, replication complex, nucleocapsid, viral RdRp, and methyltransferase. Many of these areas are undergoing active research and we will highlight recent developments. </p></div>\n",
              " <div><p>Since the first SARS-CoV-1 outbreak in 2003 and later with the MERS-CoV outbreak, much effort was devoted to drug repurposing.  The outbreak of COVID-19 led to significant activity in this area due to the overwhelming situation and urgency for treatment. A major advantage of drug repurposing is that the drug is already approved for use in one area, therefore drug safety and pharmacokinetic properties are well established. As a result, clinical trials can be conducted directly in patients, leading to accelerated approval if efficacy is observed. Many approved drugs have been reported to block SARS-CoV-2 replication.  Several of these drugs have undergone clinical trials for potential treatment of SARS-CoV-2 infection and COVID-19. These include, remdesivir (1, Figure ) which was initially developed for treatment of hepatitis C infection, chloroquine and hydroxylchloroquine which were developed for treatment of malaria, favipiravir which was developed as an anti-influenza drug, and masitinib which was developed as a kinase inhibitor against mast cell tumors in animals. Among these, only remdesivir (1), developed by Gilead Sciences, was initially approved by the FDA for treatment of COVID-19 patients.  Remdesivir's biological mechanism of action involves interfering with genome replication by targeting RdRp.  Structurally, remdesivir resembles adenosine and gets incorporated into nascent viral RNA, which causes premature termination of the viral RNA chain. Remdesivir was approved as an intravenous drug that must be administered in a health care setting. However, clinical efficacy of remdesivir has remained inconclusive.  One of the reasons for this controversy is that the clinical trial was conducted on severe COVID-19 patients and antiviral drugs may not have been useful at that stage of infection. A recent clinical trial of remdesivir for treatment of mild COVID-19 patients showed higher efficacy.  It is now generally accepted that antiviral drugs for COVID-19 treatment should be administered in the earliest viral replication phase. Molnupiravir (FIDD-2801), a nucleoside analog, developed by Merck Research Laboratories, was recently given emergency use authorization (EUA) by the US FDA in 2021 as an oral drug.  Molnupiravir has the same mechanism as remdesivir. It is an RdRp inhibitor, originally developed for treatment of viral infection caused by RNA viruses. One of the main advantages of oral drugs is that the patients can be prescribed antiviral drugs at the earliest indication without hospitalization. Early phase 3 clinical trials with COVID-19 patients showed about 50% reduction in hospitalization compared to the placebo group with no reported deaths.  Paxlovid (PF-07321332), developed by Pfizer, was recently granted EUA by the FDA in 2021 as an oral drug for treatment of COVID-19 patients.  The biological mechanism of action involves inhibition of viral protease (3CLpro or Mpro) which plays a critical role in the viral replication cycle by cleaving viral proteins and producing essential individual, mature proteins necessary for viral replication.  Many inhibitors of 3CLpro were initially developed for treatment of SARS-CoV-1 during the outbreak in 2003-2004.  The Phase II-III data of Paxlovid clinical trials showed that treatment of patients within three days of COVID-19 symptoms reduced COVID related hospitalization by 89% compared to the placebo group.  There are many other drug discovery targets being actively investigated for development of effective COVID-19 therapeutics. The major targets and progress in these areas will be discussed in the following section.</p></div>\n",
              " <div><p>critical roles in the treatment of chronic viral infections, such as HIV/AIDS, hepatitis C virus, hepatitis B virus, herpesvirus and influenza virus.  Since the SARS-CoV-1 outbreak in 2003 and MERS-CoV outbreak in 2012, SARS-CoV-1 3CLpro and SARS-CoV-1 PLpro became very attractive targets for drug discovery and early development of peptidomimetic and small molecule SARS-CoV-1 protease inhibitors.  The in vivo efficacy of 3CLpro inhibitors in mice and feline species has been documented prior to the development of paxlovid.  As the activity of 3CLpro and PLpro are essential for coronavirus replication, an intense effort towards protease inhibitor design and development has commenced with the COVID-19 pandemic. We will discuss recent updates of SARS-CoV-2 3CLpro and PLpro inhibition design in this section.</p></div>\n",
              " <div><p>Of the two main proteases, SARS-CoV-2 3CLpro is the most-well characterized.  Several different types of inhibitors have been investigated including peptidomimetic covalent inhibitors, small molecule covalent inhibitors, and noncovalent inhibitors.  Historically, covalent inhibitors were thought to be generally cytotoxic due to off target effects; however, recent developments have shown that this is not always the case. Development of covalent cysteine proteases has shown significant potential for SARS-CoV-2 drug development. Several different classes of protease inhibitors will be discussed in this section including peptidomimetic covalent inhibitors, small molecule covalent inhibitors, and noncovalent inhibitors.</p></div>\n",
              " <div><p>The development of early peptidomimetic SARS-CoV-1 3CLpro inhibitors has been reviewed previously.  Structural evolution of 3CLpro inhibitors started with the design of covalent inhibitor 4 (Figure ) with an α,β-unsaturated ester, Michael acceptor warhead.  Inhibitor design also evolved to reversible inhibitors like the phthalhydrazido-methyl ketone warhead in inhibitor 5, and benzothiazolyl ketone-derived inhibitors 6 and 7.  Inhibitor 7 with a P3 indole methoxy group was synthesized and evaluated against SARS-CoV-2 3CLpro.  Compound 7 is a tight binding reversible covalent inhibitor and showed a Ki value of 17.6 nM against SARS-CoV-2 3CLpro. The compound displayed an antiviral EC50 value of 4.2 µM in VeroE6 cell-based assay with RNA-qPCR. Its apparent cytotoxic CC50 value was &gt;100 µM. Interestingly, inhibitor 7 when combined with remdesivir, an RdRp inhibitor, exerted synergistic activity against SARS-CoV-2 and viral breakthrough did not occur. At 2 µM remdesivir and 2 µM inhibitor 7 viral replication was suppressed by 0.67-fold and 1.3-fold, respectively. When both compounds were combined, the suppression was 1.8-fold. At 10 µM remdesivir and 10 µM inhibitor 7 results showed viral suppression to be 20-fold and 210-fold, respectively. However, when combined, that suppression was 590,000-fold. At 20 µM remdesivir and 20 µM inhibitor 7 viral suppression was 1,600,000-fold.  The X-ray structure of SARS-CoV-2 3CLpro and inhibitor 7 complex was determined and the structure gave molecular insight into the inhibitor-protease interactions. The active site inhibitorbound structure is shown in Figure . As can be seen, the sulfur atom of Cys145 forms a covalent bond with the carbonyl carbon next to the benzothiazole of compound 7, resulting in the formation of a tetrahedral intermediate. The hemiacetal alcohol forms one direct hydrogen bond and water-mediated hydrogen bond interactions around the oxyanion hole residues, Cys145 and Gly143. The P3 indole amide carbonyl  Bisulfite derivative, GC-376 (8, Figure ) has been shown to potently inhibit SARS-CoV-2 3CLpro.  GC-373 (9) was initially developed to inhibit feline infectious peritonitis virus (FIVP), a fatal infection in cats caused by mutations of feline enteric coronavirus (FECV).  GC-376 is the prodrug of aldehyde inhibitor GC-373, and it was shown to exhibit in vivo efficacy in treating certain forms of FIVP.  With the outbreak of COVID-19, there has been much research into GC-373 and its derivatives. Vuong et. al. and others have shown that GC-373 inhibits SARS-CoV-2 3CLpro with a Ki value of 150 nM and antiviral activity of 700 nM.  Structural modifications were carried out to further improve activity. Derivatives 10 and 11 were shown to inhibit SARS-CoV-2 3CLpro more potently, however, their antiviral activity was not reported. 66 The X-ray structure of GC-376-bound SARS-CoV-2 3CLpro has been determined. As shown in Figure , the structure revealed that Cys145 forms a covalent bond with the aldehyde functionality and forms a hemithioacetal.  The P1 lactam carbonyl forms a hydrogen bond with His163 and the lactam NH forms a hydrogen bond with the side chain carboxylic acid of Glu166. The P2 isobutyl group fits in the S2 pocket surrounding Arg40, His41, and Asp187.</p><p>Repurposing of previously approved protease inhibitors for the treatment of COVID-19 has been conducted in many laboratories.  This led to the identification of many drug candidates. However, there was wide variation in assay results. Among the most promising outcome is the identification of HCV NSP3 protease inhibitor drugs boceprevir 12 (Figure ) and telaprevir 13.  Both drugs are potent peptidyl inhibitors of HCV NSP3, which is a serine protease.  Both compounds have been shown to inhibit SARS-CoV-2 Mpro from several labs.  However, inhibition data show wide variability. Like other peptidomimetic inhibitors of 3CLpro, these serine protease inhibitor drugs possess the same α-ketoamide warhead. Like the ketobenzothiazole warhead in compound 7, the ketoamide functionality forms a covalent bond with Cys145 in the active site. Boceprevir, 12, has inhibited SARS-CoV-2 3CLpro with an IC50 range of 1.6 µM to 8 µM. In contrast, teleprevir showed weak inhibition with an IC50 value of 56 µM. Boceprevir antiviral activity EC50 value was determined to be 15.6 µM with remdesivir as a positive control and an EC50 value of 0.58 µM. Teleprevir showed marginal antiviral activity.  Structurally, boceprevir possesses a P1β-cyclobutylalanyl group and teleprevir contains a norleucine moiety. Other key features include a P2 dimethylcyclopropyl proline for boceprevir and a bicyclic cyclopentyl fused proline derivative for teleprevir. Both drugs contain a P3-tert-butyl glycine ligand. The first X-ray crystal structure of SARS-CoV-2 3CLpro bound to boceprevir was determined by our group early in the COVID-19 pandemic and was released via the PDB (6WNP).  The X-ray crystal structures of boceprevir and teleprevir bound to SARS-CoV-2 3CLpro were also determined by others.  Structural analysis of boceprevir-bound 3CLpro is shown in Figure . Cys145 forms a covalent bond with the ketoamide warhead and subsequently forms a thiohemiacetal moiety. The resulting hydroxyl group forms a hydrogen bond with the His41 side chain and stabilizes the oxyanion hole. The P1 cyclobutylmethyl group occupies the shallow S1 pocket. The backbone of His164 and Glu166 forms important hydrogen bonds with the main chain of boceprevir. The X-ray structure of boceprevir-bound and teleprevir-bound 3CLpro are being utilized for improving drug-like properties, structures and antiviral activity against SARS-CoV-2 3CLpro. These are highlighted in the following section. Based upon boceprevir or telaprevir-bound X-ray structures, a number of bicycloproline derived inhibitors have been prepared and evaluated by Yang and collaborators.  As shown in Figure , previously optimized, the P1 lactam was maintained as a fixed substituent in compound 14. The P2 proline derivative of boceprevir and the bicyclic telaprevir ligand have been introduced in combination with a small electrophilic aldehyde warhead. This P1 moiety is the active pharmacophore, which forms a covalent bond with Cys145. Unfortunately, the extremely reactive nature of the aldehyde raises some clinical safety concerns for these inhibitors. The 𝛾-lactam mimics the native P1 glutamine structure and fills the S1 subsite, and hydrophobic, medium sized subgroups were utilized in the P3 position to enhance the potency and pharmacokinetic properties.  All compounds were tested for their biological activity against SARS-CoV-2 3CLpro in a FRET assay. IC50 values were found to be in the low nanomolar range (from 7.6 to 748 nM), for compounds 14-16.</p><p>The crystal structure of 16-bound to SARS-CoV-2 3CLpro was determined. As shown in Figure , a covalent bond is formed between the aldehyde carbonyl carbon and the sulfur on Cys145. Two hydrogen bonds between the aldehyde carbonyl oxygen and the backbone amides of Cys145 and Gly143 can be observed. The bicycloproline extends into the S2 pocket where it forms hydrophobic interactions with the pocket residues. The P3 portion interacts with the amide backbone of Glu166 and the extended conformation of ethyl-3,5-difluorobenzene side chain extends into the S4 subsite.</p><p>Several compounds showed nanomolar or low micromolar EC50 values (range between 0.53 and 30.5 𝜇M).  Interestingly, some compounds with high potency in the enzymatic assay exhibited only marginal activity in the antiviral assay. This was  proposed to be caused by low lipophilic properties of certain P3 ligands, resulting in poor cell membrane permeability. This class of compounds displayed no cytotoxicity in any of the cell lines tested including, Vero E6, HPAEpiC, LO2, BEAS-2B, A549, and Huh7 cells. Preliminary pharmacokinetic property experiments were performed to determine which compounds would be good for in vivo antiviral studies. Compounds 14 and 15 showed oral bioavailabilities of 11.2% and 14.6%, respectively.  In vivo tests were performed on mice in which both 14 and 15 were utilized, and it was found that the treatment group contained lower mean viral RNA loads in lung tissues than those of the control group. Recent investigations by Liu and co-workers into boceprevir based derivatives determined that a P4 N-terminal carbamate improves activity and that an O-tert-butyl-threonine P3 ligand allows compounds to achieve higher cellular and antiviral activity.  Other recent investigations into 𝛼-ketoamide warhead containing 3CLpro inhibitors have culminated in lead compounds with good activity and oral bioavailability against several SARS-CoV-2 variants.  Based upon similar studies mentioned above, Xia and coworkers reported peptidomimetic covalent inhibitors 17 and 18 shown in Figure .  Inhibitor 17 is a hybrid of GC-376 ( ) and telaprevir and compound 18 is a hybrid of 8 and boceprevir. These inhibitors displayed potent enzymatic inhibition against SARS-CoV-2 3CLpro as well as several other coronaviruses. In a FRET assay both inhibitors had similar potency to 8, although 18 was the more potent compound based on their in house designed Flip-GFP Mpro assay.  Antiviral activities were determined using immunofluorescence assay in VeroE6 and Caco2-ACE2 cell lines. Compound 18 showed an EC50 value of 0.37 µM in Vero E6 cells. Inhibitor 18 also had antiviral activity in Caco2-ACE2 cells with an EC50 of 1.06 µM compared to inhibitor 17 at 5.24 µM, and lead compound 8 at 2.9 µM. Both derivatives inhibited 3CLpro in other coronaviruses such as, SARS-CoV, MERS-CoV, HCoV-OC43, HCoV-NL63, and HCoV-229E in the FRET-based enzymatic assay. Inhibitor 17 had improved selectivity compared to the control compound against host cysteine proteases calpain I and cathepsin L.  Selectivity was tested for each compound against cathepsin L, calpain I, cathepsin K, caspase-3, and serine protease trypsin. It was found that 8 was a potent inhibitor of calpain I and cathepsin L. Inhibitors 17 and 18 displayed significantly reduced inhibition of the off-target enzyme.  Lead compound 8 was also a potent inhibitor for cathepsin K, and while both designed inhibitors had reduced inhibitory activity, they also were quite potent against the off-target enzyme. A crystal structure of inhibitor 18 bound SARS-CoV-2 3CLpro determined the active site interactions, including formation of covalent bond with Cys145 is shown in Figure .</p><p>Bai and co-workers reported a series of peptidomimetic SARS-CoV-2 3CLpro inhibitors with 𝛼 -acyloxymethylketone lactam as the glutamine mimetic.  Representative inhibitors 19-21 are shown in Figure . The design work stemmed from studies of Krantz and co-workers, who reported related work which involved design of inactivators of cysteine proteases like cathepsin B.  In particular, this earlier work examined leaving groups at the α-position of an α-substituted methyl ketone. It was observed that the pKa of the leaving group correlated with the irreversible, covalent interaction. The choice of the α-leaving group is important as chloromethyl ketones are too reactive and cytotoxic. Fluoromethyl ketone is more selective, but it may also generate toxic metabolites.  Krantz and co-workers previously utilized benzoate esters and a few aliphatic esters of hydroxymethyl ketones as inactivators of cysteine proteases. Results indicated 2,6-bistrifluoromethyl and 2,6-dichloro benzoates as potent irreversible inactivators.  Based upon the results of diphenylphosphinyl, tretronoyl, and peptidyl esters, an α-acyloxy group was examined. This allows for pKa adjustments to balance the rapid irreversible adduct formation and excessive reactivity that sequester glutathione or cause cytotoxicity.</p><p>Heteroaromatic rings were incorporated to allow more polar surfaces to increase solubility and reduce molecular planarity based upon previous studies. Steric hinderance around the ester was introduced to slow hydrolysis and increase stability. The P2 4-methoxyindole and leucine were left unchanged throughout the SAR study.  The six-membered lactam was also explored compared to the standard 5-membered lactam for further investigation in the S1 pocket. This was found to not have a significant effect on inhibitory activity. Variation of the ester moiety to contain an electron-withdrawing group was also explored in order to reduce the pKa of the corresponding acid leaving group. This phenomenon was exemplified with inhibitor 19, which displayed an enzyme IC50 value of 86 nM.  Inhibitor 20 displayed very potent enzyme inhibitory activity with IC50 value of 1 nM. Substituted pyridinyl derivatives such as in compound 21 were less potent than pyridine alone, but more potent than 8.</p><p>Antiviral activity was observed at low micromolar range, including compounds 20 and 21 and no toxicity was observed up to 200 𝜇M. Inhibitor 19 had good potency and plasma stability.  These compounds show antiviral activity against other coronavirus strains. The X-ray crystal structure of the inhibitor 21bound to SARS-CoV-2 3CLpro was determined. As shown in Figure , Cys145 formed a covalent linkage with the inhibitor and ester carbonyl turned into a thioether functionality. Hoffman and co-workers also investigated peptidomimetic covalent inhibitors with hydroxymethyl ketone and derivatives as warheads.  This class of compounds were initially developed during the first SARS outbreak in 2003. Following the COVID-19 outbreak, investigators examined hydroxymethyl ketone 22 (Figure ) and its derivatives as potential therapeutics for COVID-19 treatment. The antiviral activity of compound 22 was evaluated against a panel of human viruses. Compound 22 does not inhibit human rhinovirus strains HRv-14 and HRv-16, HIV-1, HCMV in cell culture. Also, it does not inhibit HCV replication. The X-ray structural studies of 22-bound to the 3CLpro of SARS-CoV-1 and CoV-2 were determined. The ligand binding sites are nearly identical. Also, as expected, the hydroxymethyl ketone carbonyl carbon formed a covalent bond to the 3CLpro active site Cys145 and generated a tetrahedral carbinol complex as shown in Figure   benzothiazol-2-yl ketone as the warhead resulted in compound 24, which showed reduction of both enzyme and antiviral activity.</p><p>Replacement of the methoxyindole P3 ligand with a (S)-valsulfonamide derivative resulted in compound 25 (Figure ) with improved activity and slight improvement in oral bioavailability (F=10%). Inhibitor 26 resulted from varying the sulfonamide P3capping group with a trifluoroacetamide group. This compound exhibited comparable enzymatic inhibitory activity to 25; however, its antiviral activity in VeroE6 cells improved nearly 10-fold. Furthermore, this replacement resulted in a significant improvement in the oral bioavailability in rats (10 mpk, F=33%). Introduction of a P1' nitrile and replacement of the (S)-val with a (S)-tert-leucine provided compound 2 with further improvement in activity as well as oral bioavailability in rats (10 mpk, F=50%). This compound became known as PF-07321332 (2). It exhibited low oral bioavailability in monkeys (10 mpk, F=8.5%) due to first-pass metabolism along the gastrointestinal tract by cytochrome P450 (CYP) enzymes. Subsequent experiments involving coadministration with CYP34A inactivator, ritonavir (RTU) significantly improved the plasma concentration of PF-07321332. 47 The X-ray co-crystal structure of inhibitor 2 with SARS-CoV-2 3CLpro was determined. As shown in Figure , the P1' nitrile functionality of compound 2 forms a reversible covalent thioimidate adduct with the catalytic Cys145. The P1 lactam carbonyl forms a hydrogen bond with His163 and the lactam NH forms a hydrogen bond with the side chain carboxylic acid of Glu166 similar to other structures. A recent report by Kovalevsky and co-workers extensively explored the active site interactions of three in house compounds and inhibitor 2.  Interestingly, it was reported that the active site of Mpro can, and does, undergo a significant amount of distortion to accommodate bulkier substituents, particularly in the S4 and S5 subsites. Nitrile warhead compounds, including 2, were examined utilizing room temperature X-ray crystallography. Results displayed the presence of the thioimidate adduct formation with Cys145 and subsequent N insertion into the oxyanion hole. Other important interactions are the unconventional interactions with the CF3 group. The small electronegative CF3 is capable of forming favorable F…O interactions, as shown in this study, or F…N interactions, as reported in another study  , with the protease. These types of unconventional interactions have shown to be of significant importance in inhibitor activity, metabolic stability, and pharmacokinetics.</p><p>This oral SARS-CoV-2 3CLpro inhibitor (Figure , inhibitor 2) and its entrance into clinical trials were unveiled by Pfizer in April 2021.  As such, it is the first orally administered compound to enter clinical trials that targets the main protease of SARS-CoV-2. It can be taken orally as a pill or capsule, allowing for the freedom to be given outside of hospitals. Interestingly, the first 7 mg of the compound were synthesized in late July 2020.  A massive scale up effort was undertaken and by late October 100 g of the compound was synthesized, and two weeks later the chemists were able to scale up the synthesis to even more than 1 kg.  According to phase II and III clinical trial data, inhibitor 2 or nirmatrelvir, is extremely effective in reducing the risk of COVID-19 related hospitalization and death. Nirmatrelvir is marketed and sold under the name paxlovid, which is a combination of nirmatrelvir and ritonavir. As nirmatrelvir is metabolized by CYP3A4, ritonavir is necessary as a pharmacokinetic booster to increase the oral bioavailability of nirmatrelvir.  In a randomized clinical trial with 1:1 paxlovid:placebo received orally every 12 hours for five days, it was found that those receiving paxlovid had significantly reduced hospital admissions and deaths among those affected by COVID-19.  It was found that among participants who received treatment within three days of beginning COVID-19 symptoms, the risk of hospitalization or death was 89% lower than that for the placebo group.  Phase II/III clinical trials for administration of paxlovid in pediatric patients between the ages of 6 and 12 who test positive for COVID-19 and are at risk for severe disease began in March of 2022.  A recent report by Pfizer researchers and collaborators reports the activity of nirmatrelvir against several SARS-CoV-2 variants of concern and variants of interest.  Nirmatrelvir results are compared to the FDA approved remdesivir as the control.  Results in Table  indicate that nirmatrelvir does in fact potently inhibit the many variants. EUA was granted to Pfizer for paxlovid on December 22, 2021 for the treatment of mild-tomoderate COVID-19 in adults and pediatric patients 12 years of age or older weighing at least 40 kg who have tested positive for COVID-19 and who are at high risk for progression to severe disease including hospitalization or death. A recent study by Pan and co-workers reported the treatment of omicron infected calu-3 cells with varying doses of nirmatrelvir and observed potent inhibition of viral RNA.  Calculated IC50 values against the wild type variant and omicron variant were reportedly 0.176 𝜇M and 0.0246 𝜇M, respectively. 86</p></div>\n",
              " <div><p>Indole-5-chloropyridinyl esters were designed and synthesized to inhibit SARS-CoV-1 3CLpro.  Several derivatives have been shown to potently inhibit the 3CLpro enzyme by formation of a covalent bond with the catalytic Cys145 residue in the active site. Ghosh and co-workers further investigated the potential of this class of compounds as SARS-CoV-2 3CLpro inhibitors with the outbreak of COVID-19.  A number of compounds with varying substituents and functionalities have been synthesized and evaluated. Structure-activity studies show that the position of the carboxylic acid on the indole ring is important for activity. Selected representative derivatives 27-31 are shown in Figure . Among them, inhibitor 27 displayed a SARS-CoV-2 3CLpro inhibitory IC50 value of 250 nM. The compound also exhibited potent antiviral activity with an EC50 value of 2.8 µM in VeroE6 cells. Remdesivir, an RdRp inhibitor, showed an antiviral EC50 value of 1.2 µM in the same assay. Furthermore, compound 27 and remdesivir exhibited comparable antiviral activity in immunocytochemistry assays.  The X-ray structures of compound 27-bound to SARS-CoV 3CLpro and SARS-CoV-2 3CLpro were determined. Both structures revealed that catalytic Cys145 formed a covalent bond with the indole ester carbonyl group of compound 27. The X-ray crystal structure of 27-bound to SARS-CoV-2 3CLpro is shown in Figure . Interestingly, the indole ring forms π-π stacking interactions with the imidazole ring of the His41. Methyl substitution on the indole ring was also investigated to lock the bioactive conformation of the pyridyl ester. Incorporation of the methyl at position 3 resulted in significantly reduced enzyme and antiviral activity. However, incorporation of the methyl at position 6 resulted in compound 28. This compound exhibited comparable 3CLpro inhibitory and antiviral activity to compound 27. An inhibitor with the methyl at the 5 position of the indole displayed no antiviral activity. Varying the position of the chloropyridinyl ring to the 2 or 5 position of the indole led to a reduction in activity (compounds 29 and 30) compared to 27. The original study displayed this loss directly for the 5-indole chloropyridinyl ester, but inhibitor 29 was noted to be even less potent. This further loss of activity may be attributed to the large sulfonamide, which could cause unfavorable steric interactions in the active site. Substitution of the chloropyridinyl ester at the 4 position of the indole ring also showed a reduction in enzyme inhibition and a decrease in antiviral activity.</p><p>Substitutions on the indole ring and the chloropyridine ring were also examined. Addition of a 3-nitro sulfonamide on the compared to 27. However, addition of an N-allyl substituent at this position led to potent derivative 31 with a 3CLpro inhibitory IC50 of 73 nM. Although, antiviral EC50 was reduced to 15 𝜇M. An X-ray crystal structure of sulfonamide 29 bound to SARS-CoV-2 3CLpro was determined and the structure is shown in Figure .</p><p>Confirmation of the covalent nature can be observed in the thioester formation between Cys145 and the carbonyl of the inhibitor. Another important interaction is shown in the π-π stacking of the inhibitor indole ring with the imidazole ring of His41.  Subsequent studies in other laboratories have expanded upon these studies utilizing compound 27 as a lead. Pillaiyar and co-workers investigated a series of small molecule thioesters, and Müller's group used an in house protocol to identify lead compounds and performed a small subsequent optimization study.  The design and synthesis of a series of 5-chloropyridinyl esters of common nonsteroidal anti-inflammatory agents (NSAID) has been investigated by Ghosh and co-workers.  The chloropyridinyl esters of salicylic acid, ibuprofen, indomethacin, and related aromatic carboxylic acids exhibited potent moderate nanomolar enzyme inhibitory activity. As highlighted in Figure , acetoxysalicylic acid derivative (aspirin-derived) 32 showed SARS-CoV-2 3CLpro inhibitory activity of 360 nM. Antiviral activity was determined using a quantitative VeroE6 cell-based assay with RNA-qPCR as described recently.  However, these compounds did not show appreciable antiviral activity. Salicylic acid and methyl substituted derivatives were prepared and evaluated. Methyl group substitution on the aromatic ring resulted in the synthesis of monomeric, dimeric, and trimeric ester derivatives. Monomeric derivatives 33-35 showed moderate nanomolar 3CLpro inhibitory activity; however, only the 5-methyl derivative 35 showed antiviral activity (EC50 value 46 µM). Representative dimeric compound 36 showed slightly higher 3CLpro inhibitory activity. These compounds did not show cytotoxicity up to 100 µM. Racemic ibuprofen derivative 37 showed moderate enzyme inhibition and essentially no antiviral activity. (R)-Naproxenderived ester 38 exhibited significantly better 3CLpro inhibitory activity (IC50 of 160 nM) over (S)-naproxen-derived 39. However, both compounds did not show much antiviral activity. The mode of inhibition presumably involves the formation of a covalent bond with catalytic Cys145 and the ester carbonyl carbon in the active site. An active model of more potent (R)-naproxen derivative 38bound 3CL protease has been presented. The model in Figure  shows formation of a covalent bond with Cys145 and a strong hydrogen bond with Gln189. Also, His41 forms a nice π-π stacking interaction with the napthyl ring of (R)-naproxen and similar π-π stacking interactions to those observed in other X-ray crystallographic studies of indole-derived active ester derivatives as shown above. Zhang and collaborators reported a series of SARS-CoV-2 3CLpro inhibitors that incorporated pyrogallol warheads.  Initial screening of flavonoid natural products led to the discovery that myricetin 40 and dihydromyricetin 41, shown in Figure , exhibited potent 3CLpro inhibitory activity (&gt;90% inhibition).  The 10.1002/cmdc.202200440</p></div>\n",
              " <div/>\n",
              " <div><p>This article is protected by copyright. All rights reserved.</p><p>X-ray structure of myricetin-bound SARS-CoV-2 3CLpro was determined. As depicted in Figure , structural analysis revealed that myricetin forms a covalent bond with Cys145.  Yarovaya and co-workers reported the design, synthesis, and evaluation of a series of bispidine based SARS-CoV-2 3CLpro inhibitors.  All derivatives were examined against SARS-CoV-2 3CLpro. Many compounds were found to have inhibitory activity ranging from 1-10 𝜇 M, and several compounds exhibited submicromolar activity. Unsubstituted bispidinone, 44 in Figure , exhibited an IC50 of 2.6 µM. Other substituted bispidine derivatives, including amide derivatives 45 and derivatives without a carbonyl at C9 (46) were prepared and evaluated. In general, amide derivatives 45a-c were active; however, derivatives 46 without the C-9 ketone carbonyl were mostly inactive. Bis-amide derivatives 45a and 45b in Figure  displayed a 3CLpro IC50 value of 1.4 µM. Inhibitor 45c, containing the dihydrobenzo indazole, exhibited the best activity. Also, Nalkyl derivatives 47a and 47b displayed low micromolar inhibitory activity. In general, compounds with the scaffold of inhibitors 47 showed promising results with many of them in the low micromolar range. All compounds were also tested for their cytotoxicity in HEK293T cells and were found to be nontoxic. The mechanism of action was speculated to involve the formation of a covalent bond with Cys145 via the carbonyl at C9.  ADMET analysis was performed on these compounds, and of those that displayed good drug-like properties. Compound 47a also displayed good in vitro activity.  </p></div>\n",
              " <div><p>In 2013, Jacobs and co-workers discovered a novel class of non-covalent SARS-CoV-1 3CLpro inhibitors following a screening of NIH molecular libraries. As shown in Figure , an initial hit provided lead compound 48 as an inhibitor of both SARS-CoV-1 3CLpro and PLpro enzymes.  Subsequent synthesis of derivatives and SAR exploration led to the identification of active racemic compound 49 and the stereochemically defined, more active isomer 50. This R-isomer showed a 3CLpro IC50 value of 1.5 µM, and it is more potent than the S-isomer (IC50 28 µM). The X-ray structure of compound 50-bound SARS-CoV-1 3CLpro was determined and the structure provided detailed molecular interactions in the SARS-CoV-1 3CLpro active site. Detailed structural analysis revealed that compound 49 inhibits 3CLpro enzyme without forming a covalent bond with catalytic Cys145. Compound 50 fills in the S3-S1' subpockets of SARS-CoV-1 3CLpro. This discovery triggered further search of potent compounds through the synthesis of novel derivatives and this has been reviewed recently. With the outbreak of SARS-CoV-2, many laboratories further explored the potential of these lead structures by synthesizing other structural variants. Kitamura and collaborators examined a series of derivatives as potential SARS-CoV-2 3CLpro inhibitors.  Representative compounds 51-53 in Figure  showed SARS-CoV-2 3CLpro inhibitory activity, with inhibitors 51 and 52 showing submicromolar activity. Compound 51 showed a SARS-CoV-2 3CLpro Ki value of 0.2 µM. It was tested against SARS-CoV-2 in VeroE6 cells using the immunofluorescence assay, and it was found to have an EC50 of 1.27 𝜇M.  This compound did not show any activity against SARS-CoV-2 PLpro. Also, compound 51 did not exhibit any cytotoxicity against VeroE6 cells up to 100 𝜇M.  Compound 52 with a biphenyl group as the P2 ligand, benzylmethyl as the P3 ligand, and imidazole as the P1' ligand  showed a 3CLpro inhibitory activity of 0.67 𝜇M. In general it was found that the biphenyl P2 ligand and the benzyl methyl P3 were the most active substitutions. An X-ray crystal structure of 51 bound to SARS-CoV-2 3CLpro was determined. As shown in Figure , there is no covalent interaction with Cys145, as was previously shown with the 50-bound SARS-CoV-1 3CLpro structure. Ma and co-workers have reported the synthesis of Ugiderivatives with a P1'-dichloroacetamide, dibromoacetamide, tribromoacetamide, 2-bromo-2,2-dichloroacetamide, and 2chloro-2,2-dibromoacetamide functionalities.  As highlighted in Figure , compounds 54 and 55 were found to be the most potent with SARS-CoV-2 3CLpro IC50 values of 0.43 𝜇M and 0.08 𝜇 M, respectively. Compound 55 showed antiviral activity in VeroE6 and Caco2-hACE2 cells with EC50 values in the low micromolar range. The compound also displayed good selectivity (greater than 20 𝜇M) against calpain I, cathepsin B, cathepsin K, caspase-3 and trypsin.  Structural analysis showed that the chloroacetamide or bromoacetamide warheads form a covalent linkage with Cys145 and the rest of the molecule interacts similar to other noncovalent inhibitors described previously. Jorgensen and collaborators reported an interesting series of potent noncovalent and nonpeptidic inhibitors.  The investigators carried out virtual screenings of 2,000 known, approved drugs, and found 14 hits as inhibitors of SARS-CoV-2 3CLpro.  As shown in Figure , perampanel (56) was chosen as a lead compound. Further design and optimization based upon docking studies of compound 56 with 3CLpro were carried out. As highlighted, several potent derivatives, such as 57-59, were identified. Propoxy derivative 57 showed an enzyme IC50 value of 140 nM and an antiviral activity of 2.5 µM. Benzyloxy derivative 58 also showed potent 3CLpro inhibitory activity with an IC50 value of 128 nM. Compound 59 exhibited the most potent activity with an IC50 value of 18 nM. However, this compound did not show antiviral activity.</p><p>A crystal structure with synthesized derivative 57 with 3CLpro was determined. As shown in Figure , catalytic Cys145 does not make any covalent bond with the inhibitor. The side pyridine nitrogen forms a hydrogen bond with His163 in the active site. The pyridine carbonyl group forms a hydrogen bond with the backbone NH of Glu166. A hydrogen bonding interaction between the nitrile and amide nitrogen of Gly143 can also be observed. Further structural modifications are highlighted in Figure . Monofluoro derivative 60 was also found to be very potent. Overall results showed that this series of inhibitors are quite potent and show promise for further development. Inhibitors 61 and 62 also exhibited good activity. Inhibitor 59 shows synergistic effects with the FDA approved polymerase inhibitor remdesivir.  Inhibitor 62 turned out to be the most potent with an EC50 value of 1.1 𝜇M.   Stauffer and collaborators reported a series of noncovalent SARS-CoV-2 3CLpro inhibitors.  Lead compound ML300 ( ) was a MLPCN probe compound that was chosen for optimization studies. ML300 was initially developed in the wake of the SARS-CoV-1 outbreak, and was chosen as the lead compound based on several factors including the need for further optimization to improve ADME properties like metabolic stability.  A crystal structure with 63 (Figure ) bound SARS-CoV-2 3CLpro was obtained depicting key interactions in the enzyme-inhibitor complex shown in Figure . Several key hydrogen bonding interactions are observed. The benzotriazole nitrogen forms a hydrogen bond with the side chain of His163. The amide carbonyl of 63 forms a hydrogen bonding interaction with the backbone nitrogen of Glu166, and the terminal amide associated with the cyclopropane forms a hydrogen bond with the alcohol of Ser46. The benzotriazole and Cys145 are involved in an interesting interaction with a slightly longer hydrogen bond.  Lead optimization efforts of ML300 originally led to inhibitor 64 with an IC50 of 0.95 µM against SARS-CoV-2 3CLpro, which was comparable to that of SARS-CoV-1 3CLpro. Several sub sites were varied, as shown in the lead compound. Imidazole derivative 65 exhibited very good nanomolar potency. Nmethylated derivatives of either pyrazole or imidazole inhibitors lost considerable potency, likely due to the loss of a hydrogen bond donor. The replacement of the thiophene moiety on 65 with a 3-chlorophenyl group resulted compound 66, exhibiting an IC50 of 0.27 µM. Further optimization keeping the benzotriazole and the 3chlorophenyl as fixed ligands led to potent compounds 67-70 shown in Figure . Antiviral activities of the most promising compounds were tested in VeroE6 cells using the cytopathic effect inhibition and in a plaque reduction assay.  Lead compound ML300 showed an EC50 of 19.9 µM in the live virus cytopathic effect (CPE) assay, whereas derivative 66 had an improved EC50 value of 1.7 µM in the CPE assay. It was observed that derivatives that substituted the thienyl moiety for 3chlorophenyl displayed much better antiviral activity. Compound 70 displayed submicromolar activity in both assays and had a good selectivity index.</p><p>Unoh and co-workers reported a novel series of noncovalent SARS-CoV-2 3CLpro inhibitors, one of which is undergoing clinical trials in Japan.  A structure-based virtual screening was utilized to identify lead compounds from their inhouse compound library. One of the lead compounds, 71 (Figure ), showed a SARS-CoV-2 3CLpro IC50 value of 8.6 µM.  The X-ray structure of compound 71-bound SARS-CoV-2 3CLpro was determined. Subsequent structure-based optimization led to potent derivatives, one of these potent compounds was 72 (S-217622), which was selected for clinical development. Inhibitor 72 exhibited potent in vitro activity against all tested SARS-CoV-2 variants including 𝛼, 𝛽, 𝛾, 𝛿, and 𝜊 strains. Inhibitor 72 also showed no inhibition of host cell proteases including caspase-2, chymotrypsin, cathepsin B, cathepsin D, cathepsin G, cathepsin L and thrombin.  In vivo efficacy was also tested with SARS-CoV-2 infected mice, yielding favorable results. Treatment groups displayed considerably lower viral loads than nontreatment groups.  These favorable results have allowed this orally active drug to progress to clinical trials in Japan, where it has completed Phase IIa.  The efficacy and safety of a once daily dose over 5 days were evaluated, and showed a decrease in viral loads of 60-80% compared to the placebo group.  Since then, approval for manufacture and sales in Japan were filed and an NIH funded global Phase III clinical trial has been announced. This clinical candidate does not require co-administration of another drug and can be given once daily.  The X-ray co-crystal structure of 72-bound 3CLpro was determined. As shown in Figure , the 1-methyl-1H-1,24-triazole moiety occupies the S1 pocket and forms a hydrogen bond with the side chain NH of His163. The 2,4,5 trifluorobenzyl group fits in the S2 site and is involved in π-π stacking interactions with the imidazole side chain of His41. The P1' ligand 6-chloro-2-methyl-2H-indazole forms a hydrogen bond with the backbone NH of Thr26. It also shows hydrophobic contacts with Met49. </p></div>\n",
              " <div><p>SARS-CoV-2 PLpro is also an important drug design target since it plays a critical role in the coronavirus replication cycle.  In particular, PLpro is involved in processing and maturation of viral polyproteins, assembly of the replicasetranscriptase complex, and disruption of host immune response.  The X-ray crystallographic studies of SARS-CoV-1 PLpro, noncovalent inhibitor design, and a number of inhibitorbound SARS-CoV-1 PLpro were previously reported and reviewed.  Recently, X-ray structural studies of GRL-0617bound SARS-CoV-2 PLpro have been reported.  Osipiuk and co-workers reported a high resolution X-ray structure of inhibitor 73 (GRL-0617)-bound SARS-CoV-2 PLpro.  As can be seen in the co-crystal structure in Figure , the benzamide moiety forms hydrogen-bonding interactions with the main chain nitrogen of Gln270 and side chain of Asp165. Replacing this moiety with a benzylamine or benzyl sulfonamide isostere led to a significant reduction in potency and this portion was conserved in subsequent optimization studies.  Subsequently, they designed a number of inhibitors (74-76) highlighted in Figure .  All compounds have shown comparable SARS-CoV-2 PLpro activity and a few of them displayed antiviral acitivity, but are less potent than compound 73. An investigation by Shan et al. expanded upon the naphthyl derivatives.  Two of their most potent compounds were found to have low micromolar activity in their novel assay.</p><p>Shen and co-workers reported a series of SARS-CoV-2 PLpro inhibitors via high-throughput screening.  They utilized the naphthalenyl benzamide core structure of 73 as reported previously.  As highlighted in Figure , naphthalenyl derivative 77 is less potent than inhibitor 73, and as such it was noted that any variation in the benzylmethyl led to a loss in activity. Replacement of the naphthalene of 77 was then investigated. This was done in an attempt to improve metabolic stability. Fused heteroaryls such as benzothiophene, indole, and carbazole with differing linkages were investigated. Unfortunately, most modifications led to a loss in activity. The 3-benzothiphene and carabazole based analogues had reasonable potency. The biaryl derivatives such as 2-phenylthiphene and 3-phenylthiophene, however, showed a slight improvement in potency in compound 78.</p><p>Further modifications are shown in Figure . Modification of the aryl amine to an azetidine ring resulted in a dramatic increase in potency due to enabling the electrostatic interactions with Glu167. Two of their most potent inhibitors, 78 and 79, were tested for their efficacy and bioavailability in human lung epithelial A549 cells. Inhibitor 81 was not effective in preliminary antiviral studies compared to the other two inhibitors, although it had a high binding affinity and low dissociation rate. Both 78 and 79 displayed more cytotoxicity than 77 at 100 µM and no cytotoxicity up to 30 µM. </p></div>\n",
              " <div><p>RdRp is a nonstructural protein responsible for synthesizing viral RNA, which is then transcribed into viral proteins. Inhibition of this essential enzyme prevents protein synthesis, a vital step in the viral life cycle. This is also the only protein that is mostly conserved among RNA viruses, making it an attractive target for broad-spectrum antivirals.  </p></div>\n",
              " <div><p>Nucleoside inhibitors mimic substrates for the viral RdRp and lead to either chain termination or lethal mutagenesis. Unfortunately, nucleoside derivatives, particularly those that cause chain termination, can be less effective against coronaviruses. Coronaviruses contain an exonucleolytic proofreading mechanism that can remove misincorporated nucleotides.  Therefore, these drug therapies need to be effective at circumventing the natural proofreading ability of RdRp. Remdesivir (1, Figures ) is an antiviral nucleotide phosphoramidate prodrug initially developed for treatment of Hepatitis C and later utilized as a therapy for Ebola and Marburg viral infections.  Remdesivir triphosphate is the active form of the drug and a natural drug metabolite that competes with the native adenosine triphosphate for chain inclusion.  Inclusion of remdesivir into the RNA chain prevents further RNA synthesis and leads to chain termination. This termination has been shown to be the result of a translocation barrier, which causes the coronavirus RdRp to stall after the addition of three nucleotides after incorporation of remdesivir.  This is expected to occur due to the C1 nitrile in remdesivir, as it has been shown to be crucial for antiviral potency in the Ebola virus.  Molecular modeling has also shown a steric clash between the nitrile of remdesivir and side chain of Ser861 in the Nsp12 subunit of RdRp.  To combat this, viral proofreading can occur, which renders remdesivir less efficient. Figure  illustrates remdesivir and SARS-CoV-2-RdRp active site interactions based upon X-ray structural studies.  Several studies have been done to show remdesivir efficacy; however mixed results have been observed. Clinical trials with remdesivir have been performed to determine its effect on patients with moderate COVID-19 versus standard of care, as well as the effect of a 5 day remdesivir treatment compared to 10 day treatment regimen.  In the randomized, open-labeled trial of hospitalized patients with confirmed COVID-19 pneumonia across 105 hospitals in the US, Europe, and Asia it was found that these patients with moderate COVID-19 no statistical significance was observed for those that were treated with remdesivir for 10 days. However, a statistically significant difference was observed for those that received a 5-day treatment of remdesivir compared to standard of care treatment. This difference was of uncertain clinical importance though.  In another randomized, open-label phase III clinical trial involving hospitalized patients with confirmed severe COVID-19 it was found that there was no significant difference between a 5-day and 10-day treatment regimen of remdesivir. A large, double-blind, randomized, placebo-controlled clinical trial of remdesivir treatment of COVID-19 was published on November 5, 2020.  This was one of the larger studies and included patients of varying levels of COVID-19 infection. It was observed that those that received treatment with remdesivir had a median recovery time of 10 days as opposed to 15 days in the placebo control group.  Patients who received remdesivir were also found to be more likely than those who received a placebo to have clinical improvement by day 15 and those in the remdesivir group had lower mortality estimates by day 15 and day 29.  Based on the evidence from many clinical trials, remdesivir was FDA approved for use in hospitalized individuals infected with SARS-CoV-2 on October 22, 2020.  However, remdesivir is currently only available as an intravenous injection and must be administered in a health care setting making it less available to the general public. Research into an orally bioavailable analogue is underway in many laboratories. One study found compound 82 (Figure ), which is an oral prodrug of the remdesivir triol 83.  This analogue displayed high oral bioavailability in two animal species including non-human primates.  Tested in VeroE6 cells against different strains of SARS-CoV-2 led to a range of EC50 values between 0.11 and 0.73 𝜇 M for GS-621763 and no appreciable cytotoxicity.  Interestingly, it was also observed that treatment blocked viral transmission between untreated direct-contact animals.</p><p>Additionally, favipiravir (84, Figure ) has also been tested for its activity against SARS-CoV-2 RdRp and is currently undergoing clinical trials.  Favipiravir is a potent influenza RdRp inhibitor that is intracellularly converted to its active triphosphate metabolite.  It has been shown to have activity against several RNA viruses, and has displayed efficacy in VeroE6 cells infected with SARS-CoV-2 with an EC50 of 61.88 𝜇M and no appreciable cytotoxicity observed.  Recently the structure of SARS-CoV-2 RdRp in the presence of favipiravir-RTP was published.  Molnupiravir, (3) an orally bioavailable RdRp acting drug developed in conjunction with Emory University, Ridgeback Biotherapeutics, and Merck is also currently undergoing phase III clinical trials and received an approval for EUA.  It was originally developed for use against influenza, but has demonstrated broad-spectrum antiviral activity, a good safety profile, tolerability, and oral bioavailability in humans.  The mechanism of action for molnupiravir is slightly different than that for remdesivir. Like remdesivir, molnupiravir is converted to its active form, 𝛽-D-N 4 -hydroxycytidine triphosphate, in cells.  This active form is a substrate for viral RNA polymerase, which subsequently allows incorporation into the viral RNA. Molnupiravir is incorporated into the RNA strand in place of cytidine triphosphate or uridine triphosphate, which forms stable base pairs with either guanine or adenine.  Utilizing this erroneous RNA as a template results in molnupiravir directing the incorporation of guanine or adenine, and thus mutated RNA  products.  As these mutations accumulate, lethal mutagenesis occurs. Molnupiravir escapes detection from the viral proofreading exonuclease via a couple of ways. First, it does not cause chain termination, and it is expected that its incorporation is not recognized by the viral proof-reading exonuclease as a misincorporation.  Additionally, this evasion could be due to the stability of M-G and M-A base pairing, which also does not promote RdRp backtracking.  Studies utilizing molnupiravir and favipiravir in tandem have been reported and found this to be particularly effective in hamsters, and it was also observed that it largely reduces the transmission of the virus to uninfected individuals.  Due to the mechanism of action there has been cause for concern, among them possibilities of molnupiravir aiding in the emergence of variant coronaviruses. Despite the possible risks, a panel on the FDA voted 13-10 in favor of granting its emergency use authorization on November 30, 2021.</p><p>Since its EUA clinical data has shown that it is considerably less effective at reducing risk than was originally expected. Initial reports led researchers to believe molnupiravir would be closer to 50% effective, however, with more data becoming available only a 30% reduction in COVID-19 related hospitalization or death has been observed and reported.  It has been reported to effectively reduce omicron viral RNA in cultured lung epithelial cells by 70% with an estimated IC50 against wild type SARS-CoV-2 of 1.965 𝜇M and 0.7556 𝜇M against the omicron variant.  The most successful RdRp inhibitors have been mentioned above; however, there is no shortage of weapons in the drugrepurposing arsenal. Galidesivir, ribavirin, sofosbuvir, and tenofovir have been reported to be under consideration.  Independently, an in silico drug repurposing study utilizing molecular docking and dynamics studies was conducted on over 30 compounds in which the ones to most tightly bind the active site of SARS-CoV-2 RdRp were found to be: remdesivir, favipiravir sofosbuvir, ribavirin, galidesivir, cefuroxime, tenofovir, and hydroxychloroquin.  </p></div>\n",
              " <div><p>Unlike nucleoside inhibitors, non-nucleoside inhibitors are unencumbered by the proofreading activity of the coronavirus. Cen and coworkers reported a drug repurposing study and found a non-nucleoside RdRp inhibitor, corilagin 85 (Figure ), which binds directly to RdRp and inhibits the polymerase activity in both cell-free and cell-based assays.  It fully resists the proofreading activity and potently inhibits SARS-CoV-2 infection with a low EC50. Their work began with a virtual screening of over 15,000 compounds that culminated in the top 50 hits with strong binding energies being selected. These hits binding affinities were further validated through bio-layer interferometry binding assay. Of the initial 50 compounds, 6 showed direct SARS-CoV-2 RdRp binding.</p><p>Corilagin showed the strongest RdRp binding affinity and was shown to effectively inhibit SARS-CoV-2 RdRp in Vero cells and showed moderate synergistic effects with remdesivir on inhibiting HCoV-OC43.  Zandi and co-workers reported, two natural products, baicalein (86) and baicalin (87), displayed significant antiviral activity against SARS-CoV-2.  Through their cell-based and biochemical studies, both compounds were determined to act as RdRp inhibitors. Antiviral activity was measured utilizing their previously established and optimized in vitro SARS-CoV-2 cell-based virus yield assay method and cytotoxicity was measured against Vero cells using a cell viability MTS assay.  Assays were completed utilizing remdesivir as a control, and none of the tested compounds exhibited any cytotoxicity. Baicalein displayed more potent activity in contrast with baicalin and was found to inhibit intracellular replication even when added up to 8 hours post infection. Both compounds did show significant effects on the early stages of SARS-CoV-2 replication through 6-8 hours post infection. The hypothesized mechanism of action is that these inhibitors act by binding to SARS-CoV-2 RdRp in a place other than the active site and that the mechanism of action for these natural products differ from that of remdesivir.  </p></div>\n",
              " <div><p>Methyltransferase is an emerging target for drug design for antivirals of SARS-CoV-2, which contains two methyltransferases (MTase): Nsp14 and Nsp16. These enzymes are responsible for capping viral mRNAs, which is essential for transcription and avoiding the host's innate immune system. Capping ensures integrity of the viral RNA and consists of an N-methylated guanosine triphosphate and C2'-O-methyl-ribosyladenine.  It resembles the native mRNA of host cells, stabilizes the RNA, and ensures effective translation.  This also allows the viral RNA to escape detection of the host's innate immune system by mimicking the natural mRNA in host cells.</p><p>Both enzymes bind Nsp10, which is a stable monomeric protein with no currently identified individual purpose. It is mainly known to stabilize the S-adenosylmethionine (SAM)-binding pockets of both Nsp14 and Nsp16.  Both Nsps are SAMdependent MTases. Nsp14 is responsible for methylation of the cap on the guanine of the GTP, N 7 -MTase, and exonuclease activities.  Association with Nsp14 stimulates exonuclease activity, but does not affect the MTase activity.  There is currently no crystal structure for SARS-CoV-2 Nsp14 MTase, although there is a solved crystal structure for the Nsp14 methyltransferase of SARS-CoV, which is expected to have a 94.9% homogeny with the SARS-CoV-2 Nsp14 methyltransferase.  Nsp16 is a 7-methylguanine-triphosphate-adenosine specific, 2'-O-MTase, whose activation occurs via binding cofactor Nsp10.  Nsp16 is responsible for the methylation of the C2' hydroxyl group of the following nucleotide. The crystal structure of Nsp10-Nsp16 complex bound to the inhibitor sinefungin has been reported and solved by Boura and coworkers as well as the high-resolution structures for Nsp16-Nsp10 heterodimers bound to SAM or sinefungin reported by Rosas-Lemus and collaborators.  Inhibitors that prevent methyltransferase activity either by competitively binding the native substrate, SAM, or by blocking interactions with Nsp10 can be developed, and in fact, several small molecules have been reported in the last year. Jaudzems and co-workers reported a series of Nsp14 and Nsp16 inhibitors designed utilizing bioisosteric substitution of the sulfonium and amino acid side chains of the cosubstrate SAM.  Three of their most potent compounds (88-91) are shown in Figure , and sinefungin (91) was utilized as a control. It was found that replacing the sulfonium with a thioether resulted in an increase in potency. This is said to indicate that conformational flexibility and bulkiness is more essential for activity rather than the positive charge.  An aromatic or heteroaromatic group in place of the aliphatic amino acid side chain also improved potency. Interestingly, it was noted that the improvements were more dramatic for Nsp16 over Nsp14, and this was hypothesized to be due to the ability for the inhibitor's adenine fragment to bind Nsp16 in a conformation more similar to SAM than in Nsp14.</p><p>In order to determine inhibitor selectivity, their derivatives were also tested for their inhibition against human glycine Nmethyltransferase (GNMT). It was found that the inhibitors are also potent inhibitors of GNMT and are in fact not selective for SARS-CoV-2 methyltransferase. Despite this, the compounds exhibited no cytotoxicity in mouse embryo fibroblast, human liver cancer, and adenocarcinomic human alveolar basal epithelial cell lines. From there, it was suspected there was an issue with compound permeability. This was then tested by determining cell lysates and culture media by mass spectrometry and it was indeed found that the lysates were low and often below the detection limit.  Thus, further optimization of this class of compounds is required to improve selectivity and lipophilicity. Nencka and collaborators reported a series of Nsp14 methyltransferase inhibitors created on structure based design of SARS-CoV-2 Nsp14 ligands.  The structure and activity of the most active compounds, 92-94, are shown in Figure . Inhibitor design was derived from a model of SARS-CoV-2 bound SAM based on crystallized SARS-CoV Nsp14. It was hypothesized that aromatic systems would be able to interact significantly with the aromatic amino acid residues and Arg289 through cation-𝜋 interactions.  The mechanism of action of 92 and 93 were investigated and it was shown that both derivatives compete with SAM, but are RNA noncompetitive Nsp14 inhibitors. Inhibitor selectivity was also investigated against 33 human RNA-, DNA-, and protein-MTases. No significant inhibition was observed from 20 of the protein lysine-MTases, but both compounds did exhibit some inhibitory activity against other MTases.</p><p>Debart and co-workers reported their synthesis and evaluation of nucleoside-derived inhibitors against SARS-CoV-2 Nsp14 MTase, which catalyzes the transfer of the SAM methyl to the N7-guanosine cap.  They tested 39 SAM analogues, of which 7 were found to exhibit double-digit nanomolar activity against the methyltransferase. 140 Three of the most potent discovered inhibitors stabilized Nsp14, and the best inhibitor showed high selectivity over human RNA N7-methyltransferase. They set out to synthesize smaller, efficient molecules with the Three regions focused on included the linker between the deoxyadenosine and the phenyl ring, substituents on the phenyl ring, and functionalization of the 5'-nitrogen on the nucleoside. The sulfonamide linker proved crucial, as it was replaced with an amide and lost inhibitory activity. Substitution on the sulfonamide nitrogen with an ethyl (R 4 ) was tested as it was expected to improve hydrophobicity and cellular penetration. Further investigations involved attaching a butyl chain with various terminal groups, such as an ethyl ester, acetate, phthalate, or phthalimide. An increase in potency was experienced when R 1 was varied to a nitro, which was then changed to a nitrile (96) as the nitro is expected to have increased mutagenic potential and the nitrile is expected to interact similarly with Arg310.  Several compounds were also synthesized incorporating hydrophobic substituents at R 2 in order to fill the hydrophobic pocket. The most active compounds, 96-98, are shown in Figure .</p><p>Biological activity was determined for all synthesized compounds, and only 8 were found to be not active against SARS-CoV-2 Nsp14. At least 65% inhibition was displayed for all of the active compounds, 11 of which displayed better activity than the control compound, sinefungin.  Further investigation into the mechanism of action showed that the most active compound, 96, is a SAM-competitive inhibitor for Nsp14. Specificity of this inhibitor was also tested in dose response assays containing N7-MTase from vaccinia virus, 2'-O-MTases from Dengue virus, vaccinia virus, SARS-CoV-2, and human RNA N7-MTase (hRNMT). No inhibition was found for any of the viral MTases up to 50 𝜇M, and against hRNMT at 50 𝜇M compound 96 had an IC50 of 52.8 𝜇M.  Thus, a high selectivity for SARS-CoV-2 Nsp14 was observed.</p></div>\n",
              " <div><p>Prevention of viral entry into the cell has also proven to be an active area of research. There are several ways to prevent this: ACE-2 blockers, host cell protease inhibitors such as transmembrane serine protease 2 (TMPRSS2), inhibitors of viral fusion, and monoclonal antibodies directed towards the spike protein. Entry inhibitors targeting any of the steps of the viral entry process into cells are an emerging area of interest. It is known that the viral spike protein must bind to the angiotensin-converting enzyme 2 (ACE2) cell receptor in order to initiate cell entry. The spike protein has been most heavily investigated for vaccine production, but viral entry inhibitors have also been investigated to prevent this process.</p><p>Monoclonal antibodies have been investigated and given emergency use authorization for the treatment of COVID-19. Monoclonal antibodies developed center around the spike protein and utilize this to prevent viral entry into the cell.  A risk reduction of 70-85% against mild to moderate COVID-19 infection in outpatient settings has been reported.  Unfortunately, as treatment effectiveness revolves around the spike protein, new variants with extensive mutations in the spike protein may be less susceptible to current antibody treatment methods. Treatment is also more intensive as antibodies are administered as an intravenous injection, which must occur in a health care setting. In addition to antibodies, some small molecules have also been shown to inhibit the spike protein and ACE2 interaction.  Buchwald and co-workers reported a series of smallmolecule inhibitors of the COVID Spike ACE2 protein-protein interaction (PPI) to prevent viral cell entry.  Interestingly they chose to investigate several dyes and novel drug-like compounds derived from these dyes, stemming from a screening of their compound library. The depicted compounds 99-103 (Figure ) showed the most promising results for preventing PPI of ACE2 and the spike protein of SARS-CoV-2. They also investigated the binding partner for these compounds utilizing a thermal shift assay, and it was determined that the compounds were targeting the coronavirus spike protein.  These promising results can be further expanded upon and optimized in the future.</p><p>Coronaviruses require calcium ions to coordinate amino acid residues within the conserved fusion peptide of the spike protein to enter cells. Daniel and co-workers investigated calciumchannel-blocker drugs to inhibit SARS-CoV-2 entry into cells.  Cell lines tested included the most common VeroE6 cell line as well as epithelial lung cells. A total of five drugs from the different classes were chosen and tested for their activity. The most active and least cytotoxic compounds were found to be nifedipine (104, Figure ) and felodipine (105), with nifedipine exhibiting no noticeable cytotoxicity. These compounds are expected to inhibit viral entry into the cell via calcium ion chelation; however, this hypothesis needs to undergo further testing for confirmation as it is possible they could inhibit viral spread at other stages of the life cycle.  Hall and collaborators reported a virtual screen of 120 different compounds and found two novel chemotypes of entry inhibitors that target the fusion peptide of SARS-CoV-2.  Kinase inhibitors that interfere with the phosphorylation of proteins essential for cell growth, signaling, and survival are another proposed area of treatment for COVID-19. These inhibitors can prevent phosphorylation of proteins vital to the immune activation and inflammation, causing immunosuppression and less severe symptoms.  Janus   Baricitinib received FDA approval on May 11, 2022 for the treatment of COVID-19 in hospitalized adults requiring supplemental oxygen, non-invasive or invasive mechanical ventilation, or extracorporeal membrane oxygen.  Baricitinib in particular has a possibility of having direct antiviral activity via viral endocytosis interference, which would prevent SARS-CoV-2 from entering and infecting cells.  Clinical trials involving baricitinib and remdesivir are currently underway.  </p></div>\n",
              " <div><p>The emergence of the COVID-19 pandemic at the end of 2019 shocked the world and demanded the most serious attention for the development of efficacious treatment of COVID-19. As a consequence, scientists across the globe plunged into developing vaccines and antiviral drugs in an unprecedented manner. The development of COVID-19 vaccines is a remarkable achievement. Never before have such safe and effective vaccines been developed with such rapid speed. Certainly, the COVID-19 vaccine's success will change the future of vaccine science. Vaccines will play a critical role in terminating the current pandemic.</p><p>However, development of effective antiviral drugs with proven efficacy is critically important to save lives and reduce morbidity and hospitalization. The  ABBREVIATIONS USED: 3CLpro, 3-chymotrypsin-like protease; 3CLpro, 3-chymotrypsin-like protease (3CLpro); ACE2, angiotensin-converting enzyme 2; ADMET, absorption, distribution, metabolism, excretion, toxicity; AIDS, acquired immunodeficiency syndrome; CPE, cytopathic effect; CYP, cytochrome P450 enzymes; DNA, deoxyribonucleic acid; FRET, Fluorescence Resonance Energy Transfer; GTP, guanosine triphosphate; GNMT, glycine N-methyltransferase; HCoVs, human coronaviruses; HCV, hepatitis C virus; NSP3, non-structural protein 3; HIV, human immunodeficiency virus; MERS-CoV, Middle east respiratory syndrome coronavirus; MHV coronavirus, Mouse Hepatitis Virus coronavirus; Mpro, main protease; MTase, methyltransferases; NSAID, nonsteroidal antiinflammatory agents; PDB, protein data base; PLpro, papain-like protease; RdRp, RNA-dependent RNA polymerase; RNA, ribonucleic acid; SAM, S-adenosylmethionine; SARS-CoV-2, severe acute respiratory syndrome coronavirus 2.</p></div>\n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " \n",
              " </body>]"
            ]
          },
          "metadata": {},
          "execution_count": 31
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "###  11.1 Strip markup and keep text\n",
        "\n",
        "We only want to keep the human-readable text so we will use the get_text() method to return all the text in the articles as a single Unicode string without the `<body>`, `<div>` and `<p>` tags."
      ],
      "metadata": {
        "id": "dGYunfixp33M"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "def strip_markup(articles):\n",
        "\n",
        "    for article in articles:\n",
        "\n",
        "        return article.get_text()"
      ],
      "metadata": {
        "id": "g062Bv2Fp33N"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles = list(map(strip_markup, soup_results_exceptions_removed_tags))"
      ],
      "metadata": {
        "id": "ELO3P42vp33N"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "9e9e0f0b-b1e4-4e4d-a28e-b1cc3500e75b",
        "id": "kazp3mn_p33O"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "118"
            ]
          },
          "metadata": {},
          "execution_count": 34
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "View article with all tags removed."
      ],
      "metadata": {
        "id": "gAskqBo8I6rh"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "stripped_markup_articles[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 174
        },
        "outputId": "3c2e7971-ee4b-485e-e642-0f0140ad1c27",
        "id": "0GHwNjQHp33O"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "'\\nSevere acute respiratory syndrome coronavirus 2 (SARS-CoV-2) originated in Wuhan, China in late December 2019.  This outbreak began spreading at an alarming rate, and unleashed a severe health crisis around the globe. Subsequently, the uncertainties caused a serious economic meltdown worldwide. On March 11, 2020, the World Health Organization (WHO) declared the novel coronavirus (COVID-19) outbreak a global pandemic.  Since then, it has gone on to affect millions of lives across the globe and caused nearly 6.3 million deaths as of June 7, 2022.  Human-human transmission for SARS-CoVs occurs primarily via respiratory droplets through sneezing, coughing, or close contact between persons. Mild symptomatic cases may include: fever, fatigue, dyspnea.  More severe cases of SARS-CoV-2 develop pneumonia, acute respiratory distress, and hypoxia.  Early on, many laboratories around the world got involved in the development of COVID-19 therapeutics. These include, development of therapies through drug repurposing, monoclonal antibody-based treatment, convalescent plasma therapies, vaccine therapies, and target-based drug development.  We will focus our review on recent efforts and updates on new drug development since our prior review in 2020.  Coronaviruses are the largest known single stranded RNA viruses that infect animals and humans leading to respiratory, gastrointestinal, hepatic, and neurologic diseases.  There are currently four different genera of human coronaviruses (HCoVs): alpha-coronavirus, beta-coronavirus, gamma-coronavirus, and delta-coronavirus. SARS-CoV-2 is the seventh human coronavirus. Another six have been identified: alpha-CoVs-NL63 and HCoVs-229E, and beta-CoVs HCoVs-OC43, HCoVs-HKU1, severe acute respiratory syndrome-CoV (SARS-CoV-1), and Middle East respiratory syndrome-CoV (MERS-CoV). These have been reviewed extensively.  Recently, a newly identified alphacoronavirus was identified in patients hospitalized with pneumonia.  This new human coronavirus turns out to be a novel \"canine-feline coronavirus\" that is called CCoV-human pneumonia (HuPn)-2018, and additional studies have now turned up even more of these hybrid alpha-coronaviruses around the globe.  Sequence analysis suggests that these are novel caninefeline-porcine-like (CFPL) CoVs of Alphacoronavirus 1 species that can infect humans (hCFPL-CoVs) and are associated with acute respiratory illness.The first SARS outbreak (SARS-CoV-1) originated in the Guandong Province, China in 2002.  It spread to several Asian countries, North America, and Europe. However, it was promptly contained and affected more than 8,000 individuals with 774 deaths and a 10% mortality rate.  There is a 82% genome similarity between SARS-CoV-1 and SARS-CoV-2 and a 90% resemblance in many essential enzymes. With the arrival of SARS-CoV-1, there was significant medicinal chemistry efforts geared toward developing concepts and strategies to block SARS-CoV-1 replication and to develop target-based therapeutics. Indeed, much of the ground work for blocking essential enzyme functions provided insights and a head start for the development of mechanism-based drugs against SARS-CoV-2 infection and COVID-19.  We will highlight here recent developments and approaches to SARS-CoV-2 and Covid-19 therapeutics.\\nSARS-CoV-2 binds to the host cell surface receptor, angiotensin-converting enzyme 2 (ACE2) to enter the cell. After cell entry, viral RNA attaches to the host cell ribosome to produce two polyproteins, pp1a and pp1ab, which are subsequently cleaved by the coronavirus main protease, or 3-chymotrypsin-like protease (3CLpro), and papain-like protease (PLpro).  Cleavage of these polyproteins leads to the formation of several essential enzymes such as RNA-dependent RNA polymerase (RdRp), which is responsible for genome replication.  As shown in Figure , many of these enzymes have all been subjected to intensive research as targets for SARS-CoV antiviral therapies.Two main targets for antiviral therapies are two cysteine proteases, PLpro and 3CLpro, also known as the main protease (Mpro).  Both proteases are responsible for processing 16 nonstructural proteins (nsps), which are essential for viral replication and maturation.  The sequence identity for 3CLpro between SARS-CoV-1 and SARS-CoV-2 is 96%. 3CLpro is responsible for cleaving the polyprotein and generating functional viral proteins such as: RdRp, RNA binding proteins, exoribonuclease, helicase, and methyltransferase.  3CLpro is active as a homodimer and contains a catalytic dyad of Cys-His, whereas PLpro has a catalytic triad of Cys-His-Asp.  PLpro has an 83% sequence identity similarity between SARS-CoV-1 and SARS-CoV-2. PLpro is responsible for processing the replicase polyprotein and removing cellular substrates such as ubiquitin, termed deubiquitylation, and interferon-stimulated gene product 15 from host cell proteins.  This allows the virus to escape the host innate immune system. Other important drug design targets for COVID-19 treatment include, the spike/ACE2 viral attachment, cell entry, viral helicase, replication complex, nucleocapsid, viral RdRp, and methyltransferase. Many of these areas are undergoing active research and we will highlight recent developments. \\nSince the first SARS-CoV-1 outbreak in 2003 and later with the MERS-CoV outbreak, much effort was devoted to drug repurposing.  The outbreak of COVID-19 led to significant activity in this area due to the overwhelming situation and urgency for treatment. A major advantage of drug repurposing is that the drug is already approved for use in one area, therefore drug safety and pharmacokinetic properties are well established. As a result, clinical trials can be conducted directly in patients, leading to accelerated approval if efficacy is observed. Many approved drugs have been reported to block SARS-CoV-2 replication.  Several of these drugs have undergone clinical trials for potential treatment of SARS-CoV-2 infection and COVID-19. These include, remdesivir (1, Figure ) which was initially developed for treatment of hepatitis C infection, chloroquine and hydroxylchloroquine which were developed for treatment of malaria, favipiravir which was developed as an anti-influenza drug, and masitinib which was developed as a kinase inhibitor against mast cell tumors in animals. Among these, only remdesivir (1), developed by Gilead Sciences, was initially approved by the FDA for treatment of COVID-19 patients.  Remdesivir\\'s biological mechanism of action involves interfering with genome replication by targeting RdRp.  Structurally, remdesivir resembles adenosine and gets incorporated into nascent viral RNA, which causes premature termination of the viral RNA chain. Remdesivir was approved as an intravenous drug that must be administered in a health care setting. However, clinical efficacy of remdesivir has remained inconclusive.  One of the reasons for this controversy is that the clinical trial was conducted on severe COVID-19 patients and antiviral drugs may not have been useful at that stage of infection. A recent clinical trial of remdesivir for treatment of mild COVID-19 patients showed higher efficacy.  It is now generally accepted that antiviral drugs for COVID-19 treatment should be administered in the earliest viral replication phase. Molnupiravir (FIDD-2801), a nucleoside analog, developed by Merck Research Laboratories, was recently given emergency use authorization (EUA) by the US FDA in 2021 as an oral drug.  Molnupiravir has the same mechanism as remdesivir. It is an RdRp inhibitor, originally developed for treatment of viral infection caused by RNA viruses. One of the main advantages of oral drugs is that the patients can be prescribed antiviral drugs at the earliest indication without hospitalization. Early phase 3 clinical trials with COVID-19 patients showed about 50% reduction in hospitalization compared to the placebo group with no reported deaths.  Paxlovid (PF-07321332), developed by Pfizer, was recently granted EUA by the FDA in 2021 as an oral drug for treatment of COVID-19 patients.  The biological mechanism of action involves inhibition of viral protease (3CLpro or Mpro) which plays a critical role in the viral replication cycle by cleaving viral proteins and producing essential individual, mature proteins necessary for viral replication.  Many inhibitors of 3CLpro were initially developed for treatment of SARS-CoV-1 during the outbreak in 2003-2004.  The Phase II-III data of Paxlovid clinical trials showed that treatment of patients within three days of COVID-19 symptoms reduced COVID related hospitalization by 89% compared to the placebo group.  There are many other drug discovery targets being actively investigated for development of effective COVID-19 therapeutics. The major targets and progress in these areas will be discussed in the following section.\\ncritical roles in the treatment of chronic viral infections, such as HIV/AIDS, hepatitis C virus, hepatitis B virus, herpesvirus and influenza virus.  Since the SARS-CoV-1 outbreak in 2003 and MERS-CoV outbreak in 2012, SARS-CoV-1 3CLpro and SARS-CoV-1 PLpro became very attractive targets for drug discovery and early development of peptidomimetic and small molecule SARS-CoV-1 protease inhibitors.  The in vivo efficacy of 3CLpro inhibitors in mice and feline species has been documented prior to the development of paxlovid.  As the activity of 3CLpro and PLpro are essential for coronavirus replication, an intense effort towards protease inhibitor design and development has commenced with the COVID-19 pandemic. We will discuss recent updates of SARS-CoV-2 3CLpro and PLpro inhibition design in this section.\\nOf the two main proteases, SARS-CoV-2 3CLpro is the most-well characterized.  Several different types of inhibitors have been investigated including peptidomimetic covalent inhibitors, small molecule covalent inhibitors, and noncovalent inhibitors.  Historically, covalent inhibitors were thought to be generally cytotoxic due to off target effects; however, recent developments have shown that this is not always the case. Development of covalent cysteine proteases has shown significant potential for SARS-CoV-2 drug development. Several different classes of protease inhibitors will be discussed in this section including peptidomimetic covalent inhibitors, small molecule covalent inhibitors, and noncovalent inhibitors.\\nThe development of early peptidomimetic SARS-CoV-1 3CLpro inhibitors has been reviewed previously.  Structural evolution of 3CLpro inhibitors started with the design of covalent inhibitor 4 (Figure ) with an α,β-unsaturated ester, Michael acceptor warhead.  Inhibitor design also evolved to reversible inhibitors like the phthalhydrazido-methyl ketone warhead in inhibitor 5, and benzothiazolyl ketone-derived inhibitors 6 and 7.  Inhibitor 7 with a P3 indole methoxy group was synthesized and evaluated against SARS-CoV-2 3CLpro.  Compound 7 is a tight binding reversible covalent inhibitor and showed a Ki value of 17.6 nM against SARS-CoV-2 3CLpro. The compound displayed an antiviral EC50 value of 4.2 µM in VeroE6 cell-based assay with RNA-qPCR. Its apparent cytotoxic CC50 value was >100 µM. Interestingly, inhibitor 7 when combined with remdesivir, an RdRp inhibitor, exerted synergistic activity against SARS-CoV-2 and viral breakthrough did not occur. At 2 µM remdesivir and 2 µM inhibitor 7 viral replication was suppressed by 0.67-fold and 1.3-fold, respectively. When both compounds were combined, the suppression was 1.8-fold. At 10 µM remdesivir and 10 µM inhibitor 7 results showed viral suppression to be 20-fold and 210-fold, respectively. However, when combined, that suppression was 590,000-fold. At 20 µM remdesivir and 20 µM inhibitor 7 viral suppression was 1,600,000-fold.  The X-ray structure of SARS-CoV-2 3CLpro and inhibitor 7 complex was determined and the structure gave molecular insight into the inhibitor-protease interactions. The active site inhibitorbound structure is shown in Figure . As can be seen, the sulfur atom of Cys145 forms a covalent bond with the carbonyl carbon next to the benzothiazole of compound 7, resulting in the formation of a tetrahedral intermediate. The hemiacetal alcohol forms one direct hydrogen bond and water-mediated hydrogen bond interactions around the oxyanion hole residues, Cys145 and Gly143. The P3 indole amide carbonyl  Bisulfite derivative, GC-376 (8, Figure ) has been shown to potently inhibit SARS-CoV-2 3CLpro.  GC-373 (9) was initially developed to inhibit feline infectious peritonitis virus (FIVP), a fatal infection in cats caused by mutations of feline enteric coronavirus (FECV).  GC-376 is the prodrug of aldehyde inhibitor GC-373, and it was shown to exhibit in vivo efficacy in treating certain forms of FIVP.  With the outbreak of COVID-19, there has been much research into GC-373 and its derivatives. Vuong et. al. and others have shown that GC-373 inhibits SARS-CoV-2 3CLpro with a Ki value of 150 nM and antiviral activity of 700 nM.  Structural modifications were carried out to further improve activity. Derivatives 10 and 11 were shown to inhibit SARS-CoV-2 3CLpro more potently, however, their antiviral activity was not reported. 66 The X-ray structure of GC-376-bound SARS-CoV-2 3CLpro has been determined. As shown in Figure , the structure revealed that Cys145 forms a covalent bond with the aldehyde functionality and forms a hemithioacetal.  The P1 lactam carbonyl forms a hydrogen bond with His163 and the lactam NH forms a hydrogen bond with the side chain carboxylic acid of Glu166. The P2 isobutyl group fits in the S2 pocket surrounding Arg40, His41, and Asp187.Repurposing of previously approved protease inhibitors for the treatment of COVID-19 has been conducted in many laboratories.  This led to the identification of many drug candidates. However, there was wide variation in assay results. Among the most promising outcome is the identification of HCV NSP3 protease inhibitor drugs boceprevir 12 (Figure ) and telaprevir 13.  Both drugs are potent peptidyl inhibitors of HCV NSP3, which is a serine protease.  Both compounds have been shown to inhibit SARS-CoV-2 Mpro from several labs.  However, inhibition data show wide variability. Like other peptidomimetic inhibitors of 3CLpro, these serine protease inhibitor drugs possess the same α-ketoamide warhead. Like the ketobenzothiazole warhead in compound 7, the ketoamide functionality forms a covalent bond with Cys145 in the active site. Boceprevir, 12, has inhibited SARS-CoV-2 3CLpro with an IC50 range of 1.6 µM to 8 µM. In contrast, teleprevir showed weak inhibition with an IC50 value of 56 µM. Boceprevir antiviral activity EC50 value was determined to be 15.6 µM with remdesivir as a positive control and an EC50 value of 0.58 µM. Teleprevir showed marginal antiviral activity.  Structurally, boceprevir possesses a P1β-cyclobutylalanyl group and teleprevir contains a norleucine moiety. Other key features include a P2 dimethylcyclopropyl proline for boceprevir and a bicyclic cyclopentyl fused proline derivative for teleprevir. Both drugs contain a P3-tert-butyl glycine ligand. The first X-ray crystal structure of SARS-CoV-2 3CLpro bound to boceprevir was determined by our group early in the COVID-19 pandemic and was released via the PDB (6WNP).  The X-ray crystal structures of boceprevir and teleprevir bound to SARS-CoV-2 3CLpro were also determined by others.  Structural analysis of boceprevir-bound 3CLpro is shown in Figure . Cys145 forms a covalent bond with the ketoamide warhead and subsequently forms a thiohemiacetal moiety. The resulting hydroxyl group forms a hydrogen bond with the His41 side chain and stabilizes the oxyanion hole. The P1 cyclobutylmethyl group occupies the shallow S1 pocket. The backbone of His164 and Glu166 forms important hydrogen bonds with the main chain of boceprevir. The X-ray structure of boceprevir-bound and teleprevir-bound 3CLpro are being utilized for improving drug-like properties, structures and antiviral activity against SARS-CoV-2 3CLpro. These are highlighted in the following section. Based upon boceprevir or telaprevir-bound X-ray structures, a number of bicycloproline derived inhibitors have been prepared and evaluated by Yang and collaborators.  As shown in Figure , previously optimized, the P1 lactam was maintained as a fixed substituent in compound 14. The P2 proline derivative of boceprevir and the bicyclic telaprevir ligand have been introduced in combination with a small electrophilic aldehyde warhead. This P1 moiety is the active pharmacophore, which forms a covalent bond with Cys145. Unfortunately, the extremely reactive nature of the aldehyde raises some clinical safety concerns for these inhibitors. The 𝛾-lactam mimics the native P1 glutamine structure and fills the S1 subsite, and hydrophobic, medium sized subgroups were utilized in the P3 position to enhance the potency and pharmacokinetic properties.  All compounds were tested for their biological activity against SARS-CoV-2 3CLpro in a FRET assay. IC50 values were found to be in the low nanomolar range (from 7.6 to 748 nM), for compounds 14-16.The crystal structure of 16-bound to SARS-CoV-2 3CLpro was determined. As shown in Figure , a covalent bond is formed between the aldehyde carbonyl carbon and the sulfur on Cys145. Two hydrogen bonds between the aldehyde carbonyl oxygen and the backbone amides of Cys145 and Gly143 can be observed. The bicycloproline extends into the S2 pocket where it forms hydrophobic interactions with the pocket residues. The P3 portion interacts with the amide backbone of Glu166 and the extended conformation of ethyl-3,5-difluorobenzene side chain extends into the S4 subsite.Several compounds showed nanomolar or low micromolar EC50 values (range between 0.53 and 30.5 𝜇M).  Interestingly, some compounds with high potency in the enzymatic assay exhibited only marginal activity in the antiviral assay. This was  proposed to be caused by low lipophilic properties of certain P3 ligands, resulting in poor cell membrane permeability. This class of compounds displayed no cytotoxicity in any of the cell lines tested including, Vero E6, HPAEpiC, LO2, BEAS-2B, A549, and Huh7 cells. Preliminary pharmacokinetic property experiments were performed to determine which compounds would be good for in vivo antiviral studies. Compounds 14 and 15 showed oral bioavailabilities of 11.2% and 14.6%, respectively.  In vivo tests were performed on mice in which both 14 and 15 were utilized, and it was found that the treatment group contained lower mean viral RNA loads in lung tissues than those of the control group. Recent investigations by Liu and co-workers into boceprevir based derivatives determined that a P4 N-terminal carbamate improves activity and that an O-tert-butyl-threonine P3 ligand allows compounds to achieve higher cellular and antiviral activity.  Other recent investigations into 𝛼-ketoamide warhead containing 3CLpro inhibitors have culminated in lead compounds with good activity and oral bioavailability against several SARS-CoV-2 variants.  Based upon similar studies mentioned above, Xia and coworkers reported peptidomimetic covalent inhibitors 17 and 18 shown in Figure .  Inhibitor 17 is a hybrid of GC-376 ( ) and telaprevir and compound 18 is a hybrid of 8 and boceprevir. These inhibitors displayed potent enzymatic inhibition against SARS-CoV-2 3CLpro as well as several other coronaviruses. In a FRET assay both inhibitors had similar potency to 8, although 18 was the more potent compound based on their in house designed Flip-GFP Mpro assay.  Antiviral activities were determined using immunofluorescence assay in VeroE6 and Caco2-ACE2 cell lines. Compound 18 showed an EC50 value of 0.37 µM in Vero E6 cells. Inhibitor 18 also had antiviral activity in Caco2-ACE2 cells with an EC50 of 1.06 µM compared to inhibitor 17 at 5.24 µM, and lead compound 8 at 2.9 µM. Both derivatives inhibited 3CLpro in other coronaviruses such as, SARS-CoV, MERS-CoV, HCoV-OC43, HCoV-NL63, and HCoV-229E in the FRET-based enzymatic assay. Inhibitor 17 had improved selectivity compared to the control compound against host cysteine proteases calpain I and cathepsin L.  Selectivity was tested for each compound against cathepsin L, calpain I, cathepsin K, caspase-3, and serine protease trypsin. It was found that 8 was a potent inhibitor of calpain I and cathepsin L. Inhibitors 17 and 18 displayed significantly reduced inhibition of the off-target enzyme.  Lead compound 8 was also a potent inhibitor for cathepsin K, and while both designed inhibitors had reduced inhibitory activity, they also were quite potent against the off-target enzyme. A crystal structure of inhibitor 18 bound SARS-CoV-2 3CLpro determined the active site interactions, including formation of covalent bond with Cys145 is shown in Figure .Bai and co-workers reported a series of peptidomimetic SARS-CoV-2 3CLpro inhibitors with 𝛼 -acyloxymethylketone lactam as the glutamine mimetic.  Representative inhibitors 19-21 are shown in Figure . The design work stemmed from studies of Krantz and co-workers, who reported related work which involved design of inactivators of cysteine proteases like cathepsin B.  In particular, this earlier work examined leaving groups at the α-position of an α-substituted methyl ketone. It was observed that the pKa of the leaving group correlated with the irreversible, covalent interaction. The choice of the α-leaving group is important as chloromethyl ketones are too reactive and cytotoxic. Fluoromethyl ketone is more selective, but it may also generate toxic metabolites.  Krantz and co-workers previously utilized benzoate esters and a few aliphatic esters of hydroxymethyl ketones as inactivators of cysteine proteases. Results indicated 2,6-bistrifluoromethyl and 2,6-dichloro benzoates as potent irreversible inactivators.  Based upon the results of diphenylphosphinyl, tretronoyl, and peptidyl esters, an α-acyloxy group was examined. This allows for pKa adjustments to balance the rapid irreversible adduct formation and excessive reactivity that sequester glutathione or cause cytotoxicity.Heteroaromatic rings were incorporated to allow more polar surfaces to increase solubility and reduce molecular planarity based upon previous studies. Steric hinderance around the ester was introduced to slow hydrolysis and increase stability. The P2 4-methoxyindole and leucine were left unchanged throughout the SAR study.  The six-membered lactam was also explored compared to the standard 5-membered lactam for further investigation in the S1 pocket. This was found to not have a significant effect on inhibitory activity. Variation of the ester moiety to contain an electron-withdrawing group was also explored in order to reduce the pKa of the corresponding acid leaving group. This phenomenon was exemplified with inhibitor 19, which displayed an enzyme IC50 value of 86 nM.  Inhibitor 20 displayed very potent enzyme inhibitory activity with IC50 value of 1 nM. Substituted pyridinyl derivatives such as in compound 21 were less potent than pyridine alone, but more potent than 8.Antiviral activity was observed at low micromolar range, including compounds 20 and 21 and no toxicity was observed up to 200 𝜇M. Inhibitor 19 had good potency and plasma stability.  These compounds show antiviral activity against other coronavirus strains. The X-ray crystal structure of the inhibitor 21bound to SARS-CoV-2 3CLpro was determined. As shown in Figure , Cys145 formed a covalent linkage with the inhibitor and ester carbonyl turned into a thioether functionality. Hoffman and co-workers also investigated peptidomimetic covalent inhibitors with hydroxymethyl ketone and derivatives as warheads.  This class of compounds were initially developed during the first SARS outbreak in 2003. Following the COVID-19 outbreak, investigators examined hydroxymethyl ketone 22 (Figure ) and its derivatives as potential therapeutics for COVID-19 treatment. The antiviral activity of compound 22 was evaluated against a panel of human viruses. Compound 22 does not inhibit human rhinovirus strains HRv-14 and HRv-16, HIV-1, HCMV in cell culture. Also, it does not inhibit HCV replication. The X-ray structural studies of 22-bound to the 3CLpro of SARS-CoV-1 and CoV-2 were determined. The ligand binding sites are nearly identical. Also, as expected, the hydroxymethyl ketone carbonyl carbon formed a covalent bond to the 3CLpro active site Cys145 and generated a tetrahedral carbinol complex as shown in Figure   benzothiazol-2-yl ketone as the warhead resulted in compound 24, which showed reduction of both enzyme and antiviral activity.Replacement of the methoxyindole P3 ligand with a (S)-valsulfonamide derivative resulted in compound 25 (Figure ) with improved activity and slight improvement in oral bioavailability (F=10%). Inhibitor 26 resulted from varying the sulfonamide P3capping group with a trifluoroacetamide group. This compound exhibited comparable enzymatic inhibitory activity to 25; however, its antiviral activity in VeroE6 cells improved nearly 10-fold. Furthermore, this replacement resulted in a significant improvement in the oral bioavailability in rats (10 mpk, F=33%). Introduction of a P1\\' nitrile and replacement of the (S)-val with a (S)-tert-leucine provided compound 2 with further improvement in activity as well as oral bioavailability in rats (10 mpk, F=50%). This compound became known as PF-07321332 (2). It exhibited low oral bioavailability in monkeys (10 mpk, F=8.5%) due to first-pass metabolism along the gastrointestinal tract by cytochrome P450 (CYP) enzymes. Subsequent experiments involving coadministration with CYP34A inactivator, ritonavir (RTU) significantly improved the plasma concentration of PF-07321332. 47 The X-ray co-crystal structure of inhibitor 2 with SARS-CoV-2 3CLpro was determined. As shown in Figure , the P1\\' nitrile functionality of compound 2 forms a reversible covalent thioimidate adduct with the catalytic Cys145. The P1 lactam carbonyl forms a hydrogen bond with His163 and the lactam NH forms a hydrogen bond with the side chain carboxylic acid of Glu166 similar to other structures. A recent report by Kovalevsky and co-workers extensively explored the active site interactions of three in house compounds and inhibitor 2.  Interestingly, it was reported that the active site of Mpro can, and does, undergo a significant amount of distortion to accommodate bulkier substituents, particularly in the S4 and S5 subsites. Nitrile warhead compounds, including 2, were examined utilizing room temperature X-ray crystallography. Results displayed the presence of the thioimidate adduct formation with Cys145 and subsequent N insertion into the oxyanion hole. Other important interactions are the unconventional interactions with the CF3 group. The small electronegative CF3 is capable of forming favorable F…O interactions, as shown in this study, or F…N interactions, as reported in another study  , with the protease. These types of unconventional interactions have shown to be of significant importance in inhibitor activity, metabolic stability, and pharmacokinetics.This oral SARS-CoV-2 3CLpro inhibitor (Figure , inhibitor 2) and its entrance into clinical trials were unveiled by Pfizer in April 2021.  As such, it is the first orally administered compound to enter clinical trials that targets the main protease of SARS-CoV-2. It can be taken orally as a pill or capsule, allowing for the freedom to be given outside of hospitals. Interestingly, the first 7 mg of the compound were synthesized in late July 2020.  A massive scale up effort was undertaken and by late October 100 g of the compound was synthesized, and two weeks later the chemists were able to scale up the synthesis to even more than 1 kg.  According to phase II and III clinical trial data, inhibitor 2 or nirmatrelvir, is extremely effective in reducing the risk of COVID-19 related hospitalization and death. Nirmatrelvir is marketed and sold under the name paxlovid, which is a combination of nirmatrelvir and ritonavir. As nirmatrelvir is metabolized by CYP3A4, ritonavir is necessary as a pharmacokinetic booster to increase the oral bioavailability of nirmatrelvir.  In a randomized clinical trial with 1:1 paxlovid:placebo received orally every 12 hours for five days, it was found that those receiving paxlovid had significantly reduced hospital admissions and deaths among those affected by COVID-19.  It was found that among participants who received treatment within three days of beginning COVID-19 symptoms, the risk of hospitalization or death was 89% lower than that for the placebo group.  Phase II/III clinical trials for administration of paxlovid in pediatric patients between the ages of 6 and 12 who test positive for COVID-19 and are at risk for severe disease began in March of 2022.  A recent report by Pfizer researchers and collaborators reports the activity of nirmatrelvir against several SARS-CoV-2 variants of concern and variants of interest.  Nirmatrelvir results are compared to the FDA approved remdesivir as the control.  Results in Table  indicate that nirmatrelvir does in fact potently inhibit the many variants. EUA was granted to Pfizer for paxlovid on December 22, 2021 for the treatment of mild-tomoderate COVID-19 in adults and pediatric patients 12 years of age or older weighing at least 40 kg who have tested positive for COVID-19 and who are at high risk for progression to severe disease including hospitalization or death. A recent study by Pan and co-workers reported the treatment of omicron infected calu-3 cells with varying doses of nirmatrelvir and observed potent inhibition of viral RNA.  Calculated IC50 values against the wild type variant and omicron variant were reportedly 0.176 𝜇M and 0.0246 𝜇M, respectively. 86\\nIndole-5-chloropyridinyl esters were designed and synthesized to inhibit SARS-CoV-1 3CLpro.  Several derivatives have been shown to potently inhibit the 3CLpro enzyme by formation of a covalent bond with the catalytic Cys145 residue in the active site. Ghosh and co-workers further investigated the potential of this class of compounds as SARS-CoV-2 3CLpro inhibitors with the outbreak of COVID-19.  A number of compounds with varying substituents and functionalities have been synthesized and evaluated. Structure-activity studies show that the position of the carboxylic acid on the indole ring is important for activity. Selected representative derivatives 27-31 are shown in Figure . Among them, inhibitor 27 displayed a SARS-CoV-2 3CLpro inhibitory IC50 value of 250 nM. The compound also exhibited potent antiviral activity with an EC50 value of 2.8 µM in VeroE6 cells. Remdesivir, an RdRp inhibitor, showed an antiviral EC50 value of 1.2 µM in the same assay. Furthermore, compound 27 and remdesivir exhibited comparable antiviral activity in immunocytochemistry assays.  The X-ray structures of compound 27-bound to SARS-CoV 3CLpro and SARS-CoV-2 3CLpro were determined. Both structures revealed that catalytic Cys145 formed a covalent bond with the indole ester carbonyl group of compound 27. The X-ray crystal structure of 27-bound to SARS-CoV-2 3CLpro is shown in Figure . Interestingly, the indole ring forms π-π stacking interactions with the imidazole ring of the His41. Methyl substitution on the indole ring was also investigated to lock the bioactive conformation of the pyridyl ester. Incorporation of the methyl at position 3 resulted in significantly reduced enzyme and antiviral activity. However, incorporation of the methyl at position 6 resulted in compound 28. This compound exhibited comparable 3CLpro inhibitory and antiviral activity to compound 27. An inhibitor with the methyl at the 5 position of the indole displayed no antiviral activity. Varying the position of the chloropyridinyl ring to the 2 or 5 position of the indole led to a reduction in activity (compounds 29 and 30) compared to 27. The original study displayed this loss directly for the 5-indole chloropyridinyl ester, but inhibitor 29 was noted to be even less potent. This further loss of activity may be attributed to the large sulfonamide, which could cause unfavorable steric interactions in the active site. Substitution of the chloropyridinyl ester at the 4 position of the indole ring also showed a reduction in enzyme inhibition and a decrease in antiviral activity.Substitutions on the indole ring and the chloropyridine ring were also examined. Addition of a 3-nitro sulfonamide on the compared to 27. However, addition of an N-allyl substituent at this position led to potent derivative 31 with a 3CLpro inhibitory IC50 of 73 nM. Although, antiviral EC50 was reduced to 15 𝜇M. An X-ray crystal structure of sulfonamide 29 bound to SARS-CoV-2 3CLpro was determined and the structure is shown in Figure .Confirmation of the covalent nature can be observed in the thioester formation between Cys145 and the carbonyl of the inhibitor. Another important interaction is shown in the π-π stacking of the inhibitor indole ring with the imidazole ring of His41.  Subsequent studies in other laboratories have expanded upon these studies utilizing compound 27 as a lead. Pillaiyar and co-workers investigated a series of small molecule thioesters, and Müller\\'s group used an in house protocol to identify lead compounds and performed a small subsequent optimization study.  The design and synthesis of a series of 5-chloropyridinyl esters of common nonsteroidal anti-inflammatory agents (NSAID) has been investigated by Ghosh and co-workers.  The chloropyridinyl esters of salicylic acid, ibuprofen, indomethacin, and related aromatic carboxylic acids exhibited potent moderate nanomolar enzyme inhibitory activity. As highlighted in Figure , acetoxysalicylic acid derivative (aspirin-derived) 32 showed SARS-CoV-2 3CLpro inhibitory activity of 360 nM. Antiviral activity was determined using a quantitative VeroE6 cell-based assay with RNA-qPCR as described recently.  However, these compounds did not show appreciable antiviral activity. Salicylic acid and methyl substituted derivatives were prepared and evaluated. Methyl group substitution on the aromatic ring resulted in the synthesis of monomeric, dimeric, and trimeric ester derivatives. Monomeric derivatives 33-35 showed moderate nanomolar 3CLpro inhibitory activity; however, only the 5-methyl derivative 35 showed antiviral activity (EC50 value 46 µM). Representative dimeric compound 36 showed slightly higher 3CLpro inhibitory activity. These compounds did not show cytotoxicity up to 100 µM. Racemic ibuprofen derivative 37 showed moderate enzyme inhibition and essentially no antiviral activity. (R)-Naproxenderived ester 38 exhibited significantly better 3CLpro inhibitory activity (IC50 of 160 nM) over (S)-naproxen-derived 39. However, both compounds did not show much antiviral activity. The mode of inhibition presumably involves the formation of a covalent bond with catalytic Cys145 and the ester carbonyl carbon in the active site. An active model of more potent (R)-naproxen derivative 38bound 3CL protease has been presented. The model in Figure  shows formation of a covalent bond with Cys145 and a strong hydrogen bond with Gln189. Also, His41 forms a nice π-π stacking interaction with the napthyl ring of (R)-naproxen and similar π-π stacking interactions to those observed in other X-ray crystallographic studies of indole-derived active ester derivatives as shown above. Zhang and collaborators reported a series of SARS-CoV-2 3CLpro inhibitors that incorporated pyrogallol warheads.  Initial screening of flavonoid natural products led to the discovery that myricetin 40 and dihydromyricetin 41, shown in Figure , exhibited potent 3CLpro inhibitory activity (>90% inhibition).  The 10.1002/cmdc.202200440\\n\\nThis article is protected by copyright. All rights reserved.X-ray structure of myricetin-bound SARS-CoV-2 3CLpro was determined. As depicted in Figure , structural analysis revealed that myricetin forms a covalent bond with Cys145.  Yarovaya and co-workers reported the design, synthesis, and evaluation of a series of bispidine based SARS-CoV-2 3CLpro inhibitors.  All derivatives were examined against SARS-CoV-2 3CLpro. Many compounds were found to have inhibitory activity ranging from 1-10 𝜇 M, and several compounds exhibited submicromolar activity. Unsubstituted bispidinone, 44 in Figure , exhibited an IC50 of 2.6 µM. Other substituted bispidine derivatives, including amide derivatives 45 and derivatives without a carbonyl at C9 (46) were prepared and evaluated. In general, amide derivatives 45a-c were active; however, derivatives 46 without the C-9 ketone carbonyl were mostly inactive. Bis-amide derivatives 45a and 45b in Figure  displayed a 3CLpro IC50 value of 1.4 µM. Inhibitor 45c, containing the dihydrobenzo indazole, exhibited the best activity. Also, Nalkyl derivatives 47a and 47b displayed low micromolar inhibitory activity. In general, compounds with the scaffold of inhibitors 47 showed promising results with many of them in the low micromolar range. All compounds were also tested for their cytotoxicity in HEK293T cells and were found to be nontoxic. The mechanism of action was speculated to involve the formation of a covalent bond with Cys145 via the carbonyl at C9.  ADMET analysis was performed on these compounds, and of those that displayed good drug-like properties. Compound 47a also displayed good in vitro activity.  \\nIn 2013, Jacobs and co-workers discovered a novel class of non-covalent SARS-CoV-1 3CLpro inhibitors following a screening of NIH molecular libraries. As shown in Figure , an initial hit provided lead compound 48 as an inhibitor of both SARS-CoV-1 3CLpro and PLpro enzymes.  Subsequent synthesis of derivatives and SAR exploration led to the identification of active racemic compound 49 and the stereochemically defined, more active isomer 50. This R-isomer showed a 3CLpro IC50 value of 1.5 µM, and it is more potent than the S-isomer (IC50 28 µM). The X-ray structure of compound 50-bound SARS-CoV-1 3CLpro was determined and the structure provided detailed molecular interactions in the SARS-CoV-1 3CLpro active site. Detailed structural analysis revealed that compound 49 inhibits 3CLpro enzyme without forming a covalent bond with catalytic Cys145. Compound 50 fills in the S3-S1\\' subpockets of SARS-CoV-1 3CLpro. This discovery triggered further search of potent compounds through the synthesis of novel derivatives and this has been reviewed recently. With the outbreak of SARS-CoV-2, many laboratories further explored the potential of these lead structures by synthesizing other structural variants. Kitamura and collaborators examined a series of derivatives as potential SARS-CoV-2 3CLpro inhibitors.  Representative compounds 51-53 in Figure  showed SARS-CoV-2 3CLpro inhibitory activity, with inhibitors 51 and 52 showing submicromolar activity. Compound 51 showed a SARS-CoV-2 3CLpro Ki value of 0.2 µM. It was tested against SARS-CoV-2 in VeroE6 cells using the immunofluorescence assay, and it was found to have an EC50 of 1.27 𝜇M.  This compound did not show any activity against SARS-CoV-2 PLpro. Also, compound 51 did not exhibit any cytotoxicity against VeroE6 cells up to 100 𝜇M.  Compound 52 with a biphenyl group as the P2 ligand, benzylmethyl as the P3 ligand, and imidazole as the P1\\' ligand  showed a 3CLpro inhibitory activity of 0.67 𝜇M. In general it was found that the biphenyl P2 ligand and the benzyl methyl P3 were the most active substitutions. An X-ray crystal structure of 51 bound to SARS-CoV-2 3CLpro was determined. As shown in Figure , there is no covalent interaction with Cys145, as was previously shown with the 50-bound SARS-CoV-1 3CLpro structure. Ma and co-workers have reported the synthesis of Ugiderivatives with a P1\\'-dichloroacetamide, dibromoacetamide, tribromoacetamide, 2-bromo-2,2-dichloroacetamide, and 2chloro-2,2-dibromoacetamide functionalities.  As highlighted in Figure , compounds 54 and 55 were found to be the most potent with SARS-CoV-2 3CLpro IC50 values of 0.43 𝜇M and 0.08 𝜇 M, respectively. Compound 55 showed antiviral activity in VeroE6 and Caco2-hACE2 cells with EC50 values in the low micromolar range. The compound also displayed good selectivity (greater than 20 𝜇M) against calpain I, cathepsin B, cathepsin K, caspase-3 and trypsin.  Structural analysis showed that the chloroacetamide or bromoacetamide warheads form a covalent linkage with Cys145 and the rest of the molecule interacts similar to other noncovalent inhibitors described previously. Jorgensen and collaborators reported an interesting series of potent noncovalent and nonpeptidic inhibitors.  The investigators carried out virtual screenings of 2,000 known, approved drugs, and found 14 hits as inhibitors of SARS-CoV-2 3CLpro.  As shown in Figure , perampanel (56) was chosen as a lead compound. Further design and optimization based upon docking studies of compound 56 with 3CLpro were carried out. As highlighted, several potent derivatives, such as 57-59, were identified. Propoxy derivative 57 showed an enzyme IC50 value of 140 nM and an antiviral activity of 2.5 µM. Benzyloxy derivative 58 also showed potent 3CLpro inhibitory activity with an IC50 value of 128 nM. Compound 59 exhibited the most potent activity with an IC50 value of 18 nM. However, this compound did not show antiviral activity.A crystal structure with synthesized derivative 57 with 3CLpro was determined. As shown in Figure , catalytic Cys145 does not make any covalent bond with the inhibitor. The side pyridine nitrogen forms a hydrogen bond with His163 in the active site. The pyridine carbonyl group forms a hydrogen bond with the backbone NH of Glu166. A hydrogen bonding interaction between the nitrile and amide nitrogen of Gly143 can also be observed. Further structural modifications are highlighted in Figure . Monofluoro derivative 60 was also found to be very potent. Overall results showed that this series of inhibitors are quite potent and show promise for further development. Inhibitors 61 and 62 also exhibited good activity. Inhibitor 59 shows synergistic effects with the FDA approved polymerase inhibitor remdesivir.  Inhibitor 62 turned out to be the most potent with an EC50 value of 1.1 𝜇M.   Stauffer and collaborators reported a series of noncovalent SARS-CoV-2 3CLpro inhibitors.  Lead compound ML300 ( ) was a MLPCN probe compound that was chosen for optimization studies. ML300 was initially developed in the wake of the SARS-CoV-1 outbreak, and was chosen as the lead compound based on several factors including the need for further optimization to improve ADME properties like metabolic stability.  A crystal structure with 63 (Figure ) bound SARS-CoV-2 3CLpro was obtained depicting key interactions in the enzyme-inhibitor complex shown in Figure . Several key hydrogen bonding interactions are observed. The benzotriazole nitrogen forms a hydrogen bond with the side chain of His163. The amide carbonyl of 63 forms a hydrogen bonding interaction with the backbone nitrogen of Glu166, and the terminal amide associated with the cyclopropane forms a hydrogen bond with the alcohol of Ser46. The benzotriazole and Cys145 are involved in an interesting interaction with a slightly longer hydrogen bond.  Lead optimization efforts of ML300 originally led to inhibitor 64 with an IC50 of 0.95 µM against SARS-CoV-2 3CLpro, which was comparable to that of SARS-CoV-1 3CLpro. Several sub sites were varied, as shown in the lead compound. Imidazole derivative 65 exhibited very good nanomolar potency. Nmethylated derivatives of either pyrazole or imidazole inhibitors lost considerable potency, likely due to the loss of a hydrogen bond donor. The replacement of the thiophene moiety on 65 with a 3-chlorophenyl group resulted compound 66, exhibiting an IC50 of 0.27 µM. Further optimization keeping the benzotriazole and the 3chlorophenyl as fixed ligands led to potent compounds 67-70 shown in Figure . Antiviral activities of the most promising compounds were tested in VeroE6 cells using the cytopathic effect inhibition and in a plaque reduction assay.  Lead compound ML300 showed an EC50 of 19.9 µM in the live virus cytopathic effect (CPE) assay, whereas derivative 66 had an improved EC50 value of 1.7 µM in the CPE assay. It was observed that derivatives that substituted the thienyl moiety for 3chlorophenyl displayed much better antiviral activity. Compound 70 displayed submicromolar activity in both assays and had a good selectivity index.Unoh and co-workers reported a novel series of noncovalent SARS-CoV-2 3CLpro inhibitors, one of which is undergoing clinical trials in Japan.  A structure-based virtual screening was utilized to identify lead compounds from their inhouse compound library. One of the lead compounds, 71 (Figure ), showed a SARS-CoV-2 3CLpro IC50 value of 8.6 µM.  The X-ray structure of compound 71-bound SARS-CoV-2 3CLpro was determined. Subsequent structure-based optimization led to potent derivatives, one of these potent compounds was 72 (S-217622), which was selected for clinical development. Inhibitor 72 exhibited potent in vitro activity against all tested SARS-CoV-2 variants including 𝛼, 𝛽, 𝛾, 𝛿, and 𝜊 strains. Inhibitor 72 also showed no inhibition of host cell proteases including caspase-2, chymotrypsin, cathepsin B, cathepsin D, cathepsin G, cathepsin L and thrombin.  In vivo efficacy was also tested with SARS-CoV-2 infected mice, yielding favorable results. Treatment groups displayed considerably lower viral loads than nontreatment groups.  These favorable results have allowed this orally active drug to progress to clinical trials in Japan, where it has completed Phase IIa.  The efficacy and safety of a once daily dose over 5 days were evaluated, and showed a decrease in viral loads of 60-80% compared to the placebo group.  Since then, approval for manufacture and sales in Japan were filed and an NIH funded global Phase III clinical trial has been announced. This clinical candidate does not require co-administration of another drug and can be given once daily.  The X-ray co-crystal structure of 72-bound 3CLpro was determined. As shown in Figure , the 1-methyl-1H-1,24-triazole moiety occupies the S1 pocket and forms a hydrogen bond with the side chain NH of His163. The 2,4,5 trifluorobenzyl group fits in the S2 site and is involved in π-π stacking interactions with the imidazole side chain of His41. The P1\\' ligand 6-chloro-2-methyl-2H-indazole forms a hydrogen bond with the backbone NH of Thr26. It also shows hydrophobic contacts with Met49. \\nSARS-CoV-2 PLpro is also an important drug design target since it plays a critical role in the coronavirus replication cycle.  In particular, PLpro is involved in processing and maturation of viral polyproteins, assembly of the replicasetranscriptase complex, and disruption of host immune response.  The X-ray crystallographic studies of SARS-CoV-1 PLpro, noncovalent inhibitor design, and a number of inhibitorbound SARS-CoV-1 PLpro were previously reported and reviewed.  Recently, X-ray structural studies of GRL-0617bound SARS-CoV-2 PLpro have been reported.  Osipiuk and co-workers reported a high resolution X-ray structure of inhibitor 73 (GRL-0617)-bound SARS-CoV-2 PLpro.  As can be seen in the co-crystal structure in Figure , the benzamide moiety forms hydrogen-bonding interactions with the main chain nitrogen of Gln270 and side chain of Asp165. Replacing this moiety with a benzylamine or benzyl sulfonamide isostere led to a significant reduction in potency and this portion was conserved in subsequent optimization studies.  Subsequently, they designed a number of inhibitors (74-76) highlighted in Figure .  All compounds have shown comparable SARS-CoV-2 PLpro activity and a few of them displayed antiviral acitivity, but are less potent than compound 73. An investigation by Shan et al. expanded upon the naphthyl derivatives.  Two of their most potent compounds were found to have low micromolar activity in their novel assay.Shen and co-workers reported a series of SARS-CoV-2 PLpro inhibitors via high-throughput screening.  They utilized the naphthalenyl benzamide core structure of 73 as reported previously.  As highlighted in Figure , naphthalenyl derivative 77 is less potent than inhibitor 73, and as such it was noted that any variation in the benzylmethyl led to a loss in activity. Replacement of the naphthalene of 77 was then investigated. This was done in an attempt to improve metabolic stability. Fused heteroaryls such as benzothiophene, indole, and carbazole with differing linkages were investigated. Unfortunately, most modifications led to a loss in activity. The 3-benzothiphene and carabazole based analogues had reasonable potency. The biaryl derivatives such as 2-phenylthiphene and 3-phenylthiophene, however, showed a slight improvement in potency in compound 78.Further modifications are shown in Figure . Modification of the aryl amine to an azetidine ring resulted in a dramatic increase in potency due to enabling the electrostatic interactions with Glu167. Two of their most potent inhibitors, 78 and 79, were tested for their efficacy and bioavailability in human lung epithelial A549 cells. Inhibitor 81 was not effective in preliminary antiviral studies compared to the other two inhibitors, although it had a high binding affinity and low dissociation rate. Both 78 and 79 displayed more cytotoxicity than 77 at 100 µM and no cytotoxicity up to 30 µM. \\nRdRp is a nonstructural protein responsible for synthesizing viral RNA, which is then transcribed into viral proteins. Inhibition of this essential enzyme prevents protein synthesis, a vital step in the viral life cycle. This is also the only protein that is mostly conserved among RNA viruses, making it an attractive target for broad-spectrum antivirals.  \\nNucleoside inhibitors mimic substrates for the viral RdRp and lead to either chain termination or lethal mutagenesis. Unfortunately, nucleoside derivatives, particularly those that cause chain termination, can be less effective against coronaviruses. Coronaviruses contain an exonucleolytic proofreading mechanism that can remove misincorporated nucleotides.  Therefore, these drug therapies need to be effective at circumventing the natural proofreading ability of RdRp. Remdesivir (1, Figures ) is an antiviral nucleotide phosphoramidate prodrug initially developed for treatment of Hepatitis C and later utilized as a therapy for Ebola and Marburg viral infections.  Remdesivir triphosphate is the active form of the drug and a natural drug metabolite that competes with the native adenosine triphosphate for chain inclusion.  Inclusion of remdesivir into the RNA chain prevents further RNA synthesis and leads to chain termination. This termination has been shown to be the result of a translocation barrier, which causes the coronavirus RdRp to stall after the addition of three nucleotides after incorporation of remdesivir.  This is expected to occur due to the C1 nitrile in remdesivir, as it has been shown to be crucial for antiviral potency in the Ebola virus.  Molecular modeling has also shown a steric clash between the nitrile of remdesivir and side chain of Ser861 in the Nsp12 subunit of RdRp.  To combat this, viral proofreading can occur, which renders remdesivir less efficient. Figure  illustrates remdesivir and SARS-CoV-2-RdRp active site interactions based upon X-ray structural studies.  Several studies have been done to show remdesivir efficacy; however mixed results have been observed. Clinical trials with remdesivir have been performed to determine its effect on patients with moderate COVID-19 versus standard of care, as well as the effect of a 5 day remdesivir treatment compared to 10 day treatment regimen.  In the randomized, open-labeled trial of hospitalized patients with confirmed COVID-19 pneumonia across 105 hospitals in the US, Europe, and Asia it was found that these patients with moderate COVID-19 no statistical significance was observed for those that were treated with remdesivir for 10 days. However, a statistically significant difference was observed for those that received a 5-day treatment of remdesivir compared to standard of care treatment. This difference was of uncertain clinical importance though.  In another randomized, open-label phase III clinical trial involving hospitalized patients with confirmed severe COVID-19 it was found that there was no significant difference between a 5-day and 10-day treatment regimen of remdesivir. A large, double-blind, randomized, placebo-controlled clinical trial of remdesivir treatment of COVID-19 was published on November 5, 2020.  This was one of the larger studies and included patients of varying levels of COVID-19 infection. It was observed that those that received treatment with remdesivir had a median recovery time of 10 days as opposed to 15 days in the placebo control group.  Patients who received remdesivir were also found to be more likely than those who received a placebo to have clinical improvement by day 15 and those in the remdesivir group had lower mortality estimates by day 15 and day 29.  Based on the evidence from many clinical trials, remdesivir was FDA approved for use in hospitalized individuals infected with SARS-CoV-2 on October 22, 2020.  However, remdesivir is currently only available as an intravenous injection and must be administered in a health care setting making it less available to the general public. Research into an orally bioavailable analogue is underway in many laboratories. One study found compound 82 (Figure ), which is an oral prodrug of the remdesivir triol 83.  This analogue displayed high oral bioavailability in two animal species including non-human primates.  Tested in VeroE6 cells against different strains of SARS-CoV-2 led to a range of EC50 values between 0.11 and 0.73 𝜇 M for GS-621763 and no appreciable cytotoxicity.  Interestingly, it was also observed that treatment blocked viral transmission between untreated direct-contact animals.Additionally, favipiravir (84, Figure ) has also been tested for its activity against SARS-CoV-2 RdRp and is currently undergoing clinical trials.  Favipiravir is a potent influenza RdRp inhibitor that is intracellularly converted to its active triphosphate metabolite.  It has been shown to have activity against several RNA viruses, and has displayed efficacy in VeroE6 cells infected with SARS-CoV-2 with an EC50 of 61.88 𝜇M and no appreciable cytotoxicity observed.  Recently the structure of SARS-CoV-2 RdRp in the presence of favipiravir-RTP was published.  Molnupiravir, (3) an orally bioavailable RdRp acting drug developed in conjunction with Emory University, Ridgeback Biotherapeutics, and Merck is also currently undergoing phase III clinical trials and received an approval for EUA.  It was originally developed for use against influenza, but has demonstrated broad-spectrum antiviral activity, a good safety profile, tolerability, and oral bioavailability in humans.  The mechanism of action for molnupiravir is slightly different than that for remdesivir. Like remdesivir, molnupiravir is converted to its active form, 𝛽-D-N 4 -hydroxycytidine triphosphate, in cells.  This active form is a substrate for viral RNA polymerase, which subsequently allows incorporation into the viral RNA. Molnupiravir is incorporated into the RNA strand in place of cytidine triphosphate or uridine triphosphate, which forms stable base pairs with either guanine or adenine.  Utilizing this erroneous RNA as a template results in molnupiravir directing the incorporation of guanine or adenine, and thus mutated RNA  products.  As these mutations accumulate, lethal mutagenesis occurs. Molnupiravir escapes detection from the viral proofreading exonuclease via a couple of ways. First, it does not cause chain termination, and it is expected that its incorporation is not recognized by the viral proof-reading exonuclease as a misincorporation.  Additionally, this evasion could be due to the stability of M-G and M-A base pairing, which also does not promote RdRp backtracking.  Studies utilizing molnupiravir and favipiravir in tandem have been reported and found this to be particularly effective in hamsters, and it was also observed that it largely reduces the transmission of the virus to uninfected individuals.  Due to the mechanism of action there has been cause for concern, among them possibilities of molnupiravir aiding in the emergence of variant coronaviruses. Despite the possible risks, a panel on the FDA voted 13-10 in favor of granting its emergency use authorization on November 30, 2021.Since its EUA clinical data has shown that it is considerably less effective at reducing risk than was originally expected. Initial reports led researchers to believe molnupiravir would be closer to 50% effective, however, with more data becoming available only a 30% reduction in COVID-19 related hospitalization or death has been observed and reported.  It has been reported to effectively reduce omicron viral RNA in cultured lung epithelial cells by 70% with an estimated IC50 against wild type SARS-CoV-2 of 1.965 𝜇M and 0.7556 𝜇M against the omicron variant.  The most successful RdRp inhibitors have been mentioned above; however, there is no shortage of weapons in the drugrepurposing arsenal. Galidesivir, ribavirin, sofosbuvir, and tenofovir have been reported to be under consideration.  Independently, an in silico drug repurposing study utilizing molecular docking and dynamics studies was conducted on over 30 compounds in which the ones to most tightly bind the active site of SARS-CoV-2 RdRp were found to be: remdesivir, favipiravir sofosbuvir, ribavirin, galidesivir, cefuroxime, tenofovir, and hydroxychloroquin.  \\nUnlike nucleoside inhibitors, non-nucleoside inhibitors are unencumbered by the proofreading activity of the coronavirus. Cen and coworkers reported a drug repurposing study and found a non-nucleoside RdRp inhibitor, corilagin 85 (Figure ), which binds directly to RdRp and inhibits the polymerase activity in both cell-free and cell-based assays.  It fully resists the proofreading activity and potently inhibits SARS-CoV-2 infection with a low EC50. Their work began with a virtual screening of over 15,000 compounds that culminated in the top 50 hits with strong binding energies being selected. These hits binding affinities were further validated through bio-layer interferometry binding assay. Of the initial 50 compounds, 6 showed direct SARS-CoV-2 RdRp binding.Corilagin showed the strongest RdRp binding affinity and was shown to effectively inhibit SARS-CoV-2 RdRp in Vero cells and showed moderate synergistic effects with remdesivir on inhibiting HCoV-OC43.  Zandi and co-workers reported, two natural products, baicalein (86) and baicalin (87), displayed significant antiviral activity against SARS-CoV-2.  Through their cell-based and biochemical studies, both compounds were determined to act as RdRp inhibitors. Antiviral activity was measured utilizing their previously established and optimized in vitro SARS-CoV-2 cell-based virus yield assay method and cytotoxicity was measured against Vero cells using a cell viability MTS assay.  Assays were completed utilizing remdesivir as a control, and none of the tested compounds exhibited any cytotoxicity. Baicalein displayed more potent activity in contrast with baicalin and was found to inhibit intracellular replication even when added up to 8 hours post infection. Both compounds did show significant effects on the early stages of SARS-CoV-2 replication through 6-8 hours post infection. The hypothesized mechanism of action is that these inhibitors act by binding to SARS-CoV-2 RdRp in a place other than the active site and that the mechanism of action for these natural products differ from that of remdesivir.  \\nMethyltransferase is an emerging target for drug design for antivirals of SARS-CoV-2, which contains two methyltransferases (MTase): Nsp14 and Nsp16. These enzymes are responsible for capping viral mRNAs, which is essential for transcription and avoiding the host\\'s innate immune system. Capping ensures integrity of the viral RNA and consists of an N-methylated guanosine triphosphate and C2\\'-O-methyl-ribosyladenine.  It resembles the native mRNA of host cells, stabilizes the RNA, and ensures effective translation.  This also allows the viral RNA to escape detection of the host\\'s innate immune system by mimicking the natural mRNA in host cells.Both enzymes bind Nsp10, which is a stable monomeric protein with no currently identified individual purpose. It is mainly known to stabilize the S-adenosylmethionine (SAM)-binding pockets of both Nsp14 and Nsp16.  Both Nsps are SAMdependent MTases. Nsp14 is responsible for methylation of the cap on the guanine of the GTP, N 7 -MTase, and exonuclease activities.  Association with Nsp14 stimulates exonuclease activity, but does not affect the MTase activity.  There is currently no crystal structure for SARS-CoV-2 Nsp14 MTase, although there is a solved crystal structure for the Nsp14 methyltransferase of SARS-CoV, which is expected to have a 94.9% homogeny with the SARS-CoV-2 Nsp14 methyltransferase.  Nsp16 is a 7-methylguanine-triphosphate-adenosine specific, 2\\'-O-MTase, whose activation occurs via binding cofactor Nsp10.  Nsp16 is responsible for the methylation of the C2\\' hydroxyl group of the following nucleotide. The crystal structure of Nsp10-Nsp16 complex bound to the inhibitor sinefungin has been reported and solved by Boura and coworkers as well as the high-resolution structures for Nsp16-Nsp10 heterodimers bound to SAM or sinefungin reported by Rosas-Lemus and collaborators.  Inhibitors that prevent methyltransferase activity either by competitively binding the native substrate, SAM, or by blocking interactions with Nsp10 can be developed, and in fact, several small molecules have been reported in the last year. Jaudzems and co-workers reported a series of Nsp14 and Nsp16 inhibitors designed utilizing bioisosteric substitution of the sulfonium and amino acid side chains of the cosubstrate SAM.  Three of their most potent compounds (88-91) are shown in Figure , and sinefungin (91) was utilized as a control. It was found that replacing the sulfonium with a thioether resulted in an increase in potency. This is said to indicate that conformational flexibility and bulkiness is more essential for activity rather than the positive charge.  An aromatic or heteroaromatic group in place of the aliphatic amino acid side chain also improved potency. Interestingly, it was noted that the improvements were more dramatic for Nsp16 over Nsp14, and this was hypothesized to be due to the ability for the inhibitor\\'s adenine fragment to bind Nsp16 in a conformation more similar to SAM than in Nsp14.In order to determine inhibitor selectivity, their derivatives were also tested for their inhibition against human glycine Nmethyltransferase (GNMT). It was found that the inhibitors are also potent inhibitors of GNMT and are in fact not selective for SARS-CoV-2 methyltransferase. Despite this, the compounds exhibited no cytotoxicity in mouse embryo fibroblast, human liver cancer, and adenocarcinomic human alveolar basal epithelial cell lines. From there, it was suspected there was an issue with compound permeability. This was then tested by determining cell lysates and culture media by mass spectrometry and it was indeed found that the lysates were low and often below the detection limit.  Thus, further optimization of this class of compounds is required to improve selectivity and lipophilicity. Nencka and collaborators reported a series of Nsp14 methyltransferase inhibitors created on structure based design of SARS-CoV-2 Nsp14 ligands.  The structure and activity of the most active compounds, 92-94, are shown in Figure . Inhibitor design was derived from a model of SARS-CoV-2 bound SAM based on crystallized SARS-CoV Nsp14. It was hypothesized that aromatic systems would be able to interact significantly with the aromatic amino acid residues and Arg289 through cation-𝜋 interactions.  The mechanism of action of 92 and 93 were investigated and it was shown that both derivatives compete with SAM, but are RNA noncompetitive Nsp14 inhibitors. Inhibitor selectivity was also investigated against 33 human RNA-, DNA-, and protein-MTases. No significant inhibition was observed from 20 of the protein lysine-MTases, but both compounds did exhibit some inhibitory activity against other MTases.Debart and co-workers reported their synthesis and evaluation of nucleoside-derived inhibitors against SARS-CoV-2 Nsp14 MTase, which catalyzes the transfer of the SAM methyl to the N7-guanosine cap.  They tested 39 SAM analogues, of which 7 were found to exhibit double-digit nanomolar activity against the methyltransferase. 140 Three of the most potent discovered inhibitors stabilized Nsp14, and the best inhibitor showed high selectivity over human RNA N7-methyltransferase. They set out to synthesize smaller, efficient molecules with the Three regions focused on included the linker between the deoxyadenosine and the phenyl ring, substituents on the phenyl ring, and functionalization of the 5\\'-nitrogen on the nucleoside. The sulfonamide linker proved crucial, as it was replaced with an amide and lost inhibitory activity. Substitution on the sulfonamide nitrogen with an ethyl (R 4 ) was tested as it was expected to improve hydrophobicity and cellular penetration. Further investigations involved attaching a butyl chain with various terminal groups, such as an ethyl ester, acetate, phthalate, or phthalimide. An increase in potency was experienced when R 1 was varied to a nitro, which was then changed to a nitrile (96) as the nitro is expected to have increased mutagenic potential and the nitrile is expected to interact similarly with Arg310.  Several compounds were also synthesized incorporating hydrophobic substituents at R 2 in order to fill the hydrophobic pocket. The most active compounds, 96-98, are shown in Figure .Biological activity was determined for all synthesized compounds, and only 8 were found to be not active against SARS-CoV-2 Nsp14. At least 65% inhibition was displayed for all of the active compounds, 11 of which displayed better activity than the control compound, sinefungin.  Further investigation into the mechanism of action showed that the most active compound, 96, is a SAM-competitive inhibitor for Nsp14. Specificity of this inhibitor was also tested in dose response assays containing N7-MTase from vaccinia virus, 2\\'-O-MTases from Dengue virus, vaccinia virus, SARS-CoV-2, and human RNA N7-MTase (hRNMT). No inhibition was found for any of the viral MTases up to 50 𝜇M, and against hRNMT at 50 𝜇M compound 96 had an IC50 of 52.8 𝜇M.  Thus, a high selectivity for SARS-CoV-2 Nsp14 was observed.\\nPrevention of viral entry into the cell has also proven to be an active area of research. There are several ways to prevent this: ACE-2 blockers, host cell protease inhibitors such as transmembrane serine protease 2 (TMPRSS2), inhibitors of viral fusion, and monoclonal antibodies directed towards the spike protein. Entry inhibitors targeting any of the steps of the viral entry process into cells are an emerging area of interest. It is known that the viral spike protein must bind to the angiotensin-converting enzyme 2 (ACE2) cell receptor in order to initiate cell entry. The spike protein has been most heavily investigated for vaccine production, but viral entry inhibitors have also been investigated to prevent this process.Monoclonal antibodies have been investigated and given emergency use authorization for the treatment of COVID-19. Monoclonal antibodies developed center around the spike protein and utilize this to prevent viral entry into the cell.  A risk reduction of 70-85% against mild to moderate COVID-19 infection in outpatient settings has been reported.  Unfortunately, as treatment effectiveness revolves around the spike protein, new variants with extensive mutations in the spike protein may be less susceptible to current antibody treatment methods. Treatment is also more intensive as antibodies are administered as an intravenous injection, which must occur in a health care setting. In addition to antibodies, some small molecules have also been shown to inhibit the spike protein and ACE2 interaction.  Buchwald and co-workers reported a series of smallmolecule inhibitors of the COVID Spike ACE2 protein-protein interaction (PPI) to prevent viral cell entry.  Interestingly they chose to investigate several dyes and novel drug-like compounds derived from these dyes, stemming from a screening of their compound library. The depicted compounds 99-103 (Figure ) showed the most promising results for preventing PPI of ACE2 and the spike protein of SARS-CoV-2. They also investigated the binding partner for these compounds utilizing a thermal shift assay, and it was determined that the compounds were targeting the coronavirus spike protein.  These promising results can be further expanded upon and optimized in the future.Coronaviruses require calcium ions to coordinate amino acid residues within the conserved fusion peptide of the spike protein to enter cells. Daniel and co-workers investigated calciumchannel-blocker drugs to inhibit SARS-CoV-2 entry into cells.  Cell lines tested included the most common VeroE6 cell line as well as epithelial lung cells. A total of five drugs from the different classes were chosen and tested for their activity. The most active and least cytotoxic compounds were found to be nifedipine (104, Figure ) and felodipine (105), with nifedipine exhibiting no noticeable cytotoxicity. These compounds are expected to inhibit viral entry into the cell via calcium ion chelation; however, this hypothesis needs to undergo further testing for confirmation as it is possible they could inhibit viral spread at other stages of the life cycle.  Hall and collaborators reported a virtual screen of 120 different compounds and found two novel chemotypes of entry inhibitors that target the fusion peptide of SARS-CoV-2.  Kinase inhibitors that interfere with the phosphorylation of proteins essential for cell growth, signaling, and survival are another proposed area of treatment for COVID-19. These inhibitors can prevent phosphorylation of proteins vital to the immune activation and inflammation, causing immunosuppression and less severe symptoms.  Janus   Baricitinib received FDA approval on May 11, 2022 for the treatment of COVID-19 in hospitalized adults requiring supplemental oxygen, non-invasive or invasive mechanical ventilation, or extracorporeal membrane oxygen.  Baricitinib in particular has a possibility of having direct antiviral activity via viral endocytosis interference, which would prevent SARS-CoV-2 from entering and infecting cells.  Clinical trials involving baricitinib and remdesivir are currently underway.  \\nThe emergence of the COVID-19 pandemic at the end of 2019 shocked the world and demanded the most serious attention for the development of efficacious treatment of COVID-19. As a consequence, scientists across the globe plunged into developing vaccines and antiviral drugs in an unprecedented manner. The development of COVID-19 vaccines is a remarkable achievement. Never before have such safe and effective vaccines been developed with such rapid speed. Certainly, the COVID-19 vaccine\\'s success will change the future of vaccine science. Vaccines will play a critical role in terminating the current pandemic.However, development of effective antiviral drugs with proven efficacy is critically important to save lives and reduce morbidity and hospitalization. The  ABBREVIATIONS USED: 3CLpro, 3-chymotrypsin-like protease; 3CLpro, 3-chymotrypsin-like protease (3CLpro); ACE2, angiotensin-converting enzyme 2; ADMET, absorption, distribution, metabolism, excretion, toxicity; AIDS, acquired immunodeficiency syndrome; CPE, cytopathic effect; CYP, cytochrome P450 enzymes; DNA, deoxyribonucleic acid; FRET, Fluorescence Resonance Energy Transfer; GTP, guanosine triphosphate; GNMT, glycine N-methyltransferase; HCoVs, human coronaviruses; HCV, hepatitis C virus; NSP3, non-structural protein 3; HIV, human immunodeficiency virus; MERS-CoV, Middle east respiratory syndrome coronavirus; MHV coronavirus, Mouse Hepatitis Virus coronavirus; Mpro, main protease; MTase, methyltransferases; NSAID, nonsteroidal antiinflammatory agents; PDB, protein data base; PLpro, papain-like protease; RdRp, RNA-dependent RNA polymerase; RNA, ribonucleic acid; SAM, S-adenosylmethionine; SARS-CoV-2, severe acute respiratory syndrome coronavirus 2.\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n'"
            ],
            "application/vnd.google.colaboratory.intrinsic+json": {
              "type": "string"
            }
          },
          "metadata": {},
          "execution_count": 35
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_stripped_markup_articles_exceptions_118_v2.pickle', 'wb') as f:\n",
        "  pickle.dump(stripped_markup_articles, f)"
      ],
      "metadata": {
        "id": "l8OANXO3s2H0"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "### 11.2 Add extracted full text to DataFrame\n",
        "\n",
        "Add extracted full text to DataFrame for 118 exceptions."
      ],
      "metadata": {
        "id": "9POPN5y8JdHO"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_pmc_search_results_no_full_text.pickle', 'rb') as f:\n",
        "    pmc_search_results_no_full_text = pickle.load(f)"
      ],
      "metadata": {
        "id": "-I_6KYh4s2H0"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_no_full_text"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "19884a2e-023a-42af-fd52-f2e5b60dac1b",
        "id": "10GtjQm6s2H1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "64     PMC9538661  2022-10-13  2022-11-22   \n",
              "188    PMC9794394  2022-12-28  2023-01-02   \n",
              "212    PMC9538837  2022-10-10  2022-11-22   \n",
              "256    PMC9788990  2022-12-24  2023-01-02   \n",
              "275    PMC9794516  2022-12-28  2023-01-02   \n",
              "...           ...         ...         ...   \n",
              "9797   PMC7162151  2020-02-28  2020-04-21   \n",
              "9809   PMC9094125  2022-05-11  2022-07-16   \n",
              "10399  PMC7492056  2020-09-15  2020-09-28   \n",
              "10767  PMC8459260  2021-02-25  2022-01-25   \n",
              "11329  PMC7163523  2019-08-01  2021-02-16   \n",
              "\n",
              "                                                   title  \\\n",
              "64     Recent Drug Development and Medicinal Chemistr...   \n",
              "188    Nonstructural protein 1 (nsp1) widespread RNA ...   \n",
              "212    The Efficacy of Traditional Medicinal Plants i...   \n",
              "256    Sleep and circadian rhythm disruption alters t...   \n",
              "275    MultiOMICs landscape of SARS-CoV-2-induced hos...   \n",
              "...                                                  ...   \n",
              "9797                                               News.   \n",
              "9809   Recent advances in metal-organic framework-bas...   \n",
              "10399                                    Full Issue PDF.   \n",
              "10767  Getting in touch with your senses: Mechanisms ...   \n",
              "11329  Literature review of baseline information on n...   \n",
              "\n",
              "                            journal  \\\n",
              "64                      ChemMedChem   \n",
              "188                        iScience   \n",
              "212        Chemistry & biodiversity   \n",
              "256                        iScience   \n",
              "275                        iScience   \n",
              "...                             ...   \n",
              "9797           Chemistry & industry   \n",
              "9809                  Nano research   \n",
              "10399          JACC. CardioOncology   \n",
              "10767   WIREs mechanisms of disease   \n",
              "11329  EFSA Supporting Publications   \n",
              "\n",
              "                                                 authors  \\\n",
              "64         Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.   \n",
              "188                       Bermudez Y, Miles J, Muller M.   \n",
              "212                      Choe J, Har Yong P, Xiang Ng Z.   \n",
              "256    Taylor L, Von Lendenfeld F, Ashton A, Sanghani...   \n",
              "275    Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...   \n",
              "...                                                  ...   \n",
              "9797                                                   0   \n",
              "9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   \n",
              "10399                                                  0   \n",
              "10767                                Gupta S, Butler SJ.   \n",
              "11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   \n",
              "\n",
              "                                 doi  \\\n",
              "64            10.1002/cmdc.202200440   \n",
              "188       10.1016/j.isci.2022.105887   \n",
              "212           10.1002/cbdv.202200655   \n",
              "256       10.1016/j.isci.2022.105877   \n",
              "275       10.1016/j.isci.2022.105895   \n",
              "...                              ...   \n",
              "9797            10.1002/cind.842_3.x   \n",
              "9809       10.1007/s12274-022-4302-x   \n",
              "10399  10.1016/s2666-0873(20)30180-0   \n",
              "10767              10.1002/wsbm.1520   \n",
              "11329                              0   \n",
              "\n",
              "                                                 pdf_url  text  \n",
              "64     https://europepmc.org/articles/PMC9538661?pdf=...  None  \n",
              "188    https://europepmc.org/articles/PMC9794394?pdf=...  None  \n",
              "212    https://europepmc.org/articles/PMC9538837?pdf=...  None  \n",
              "256    https://europepmc.org/articles/PMC9788990?pdf=...  None  \n",
              "275    https://europepmc.org/articles/PMC9794516?pdf=...  None  \n",
              "...                                                  ...   ...  \n",
              "9797   https://europepmc.org/articles/PMC7162151?pdf=...  None  \n",
              "9809   https://europepmc.org/articles/PMC9094125?pdf=...  None  \n",
              "10399  https://europepmc.org/articles/PMC7492056?pdf=...  None  \n",
              "10767  https://europepmc.org/articles/PMC8459260?pdf=...  None  \n",
              "11329  https://europepmc.org/articles/PMC7163523?pdf=...  None  \n",
              "\n",
              "[118 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-20f13e20-5920-455a-88ad-2782d08978fa\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>64</th>\n",
              "      <td>PMC9538661</td>\n",
              "      <td>2022-10-13</td>\n",
              "      <td>2022-11-22</td>\n",
              "      <td>Recent Drug Development and Medicinal Chemistr...</td>\n",
              "      <td>ChemMedChem</td>\n",
              "      <td>Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.</td>\n",
              "      <td>10.1002/cmdc.202200440</td>\n",
              "      <td>https://europepmc.org/articles/PMC9538661?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>188</th>\n",
              "      <td>PMC9794394</td>\n",
              "      <td>2022-12-28</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>Nonstructural protein 1 (nsp1) widespread RNA ...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Bermudez Y, Miles J, Muller M.</td>\n",
              "      <td>10.1016/j.isci.2022.105887</td>\n",
              "      <td>https://europepmc.org/articles/PMC9794394?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>212</th>\n",
              "      <td>PMC9538837</td>\n",
              "      <td>2022-10-10</td>\n",
              "      <td>2022-11-22</td>\n",
              "      <td>The Efficacy of Traditional Medicinal Plants i...</td>\n",
              "      <td>Chemistry &amp; biodiversity</td>\n",
              "      <td>Choe J, Har Yong P, Xiang Ng Z.</td>\n",
              "      <td>10.1002/cbdv.202200655</td>\n",
              "      <td>https://europepmc.org/articles/PMC9538837?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>256</th>\n",
              "      <td>PMC9788990</td>\n",
              "      <td>2022-12-24</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>Sleep and circadian rhythm disruption alters t...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Taylor L, Von Lendenfeld F, Ashton A, Sanghani...</td>\n",
              "      <td>10.1016/j.isci.2022.105877</td>\n",
              "      <td>https://europepmc.org/articles/PMC9788990?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>275</th>\n",
              "      <td>PMC9794516</td>\n",
              "      <td>2022-12-28</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>MultiOMICs landscape of SARS-CoV-2-induced hos...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...</td>\n",
              "      <td>10.1016/j.isci.2022.105895</td>\n",
              "      <td>https://europepmc.org/articles/PMC9794516?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9797</th>\n",
              "      <td>PMC7162151</td>\n",
              "      <td>2020-02-28</td>\n",
              "      <td>2020-04-21</td>\n",
              "      <td>News.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.842_3.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7162151?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9809</th>\n",
              "      <td>PMC9094125</td>\n",
              "      <td>2022-05-11</td>\n",
              "      <td>2022-07-16</td>\n",
              "      <td>Recent advances in metal-organic framework-bas...</td>\n",
              "      <td>Nano research</td>\n",
              "      <td>Yang M, Zhang J, Wei Y, Zhang J, Tao C.</td>\n",
              "      <td>10.1007/s12274-022-4302-x</td>\n",
              "      <td>https://europepmc.org/articles/PMC9094125?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10399</th>\n",
              "      <td>PMC7492056</td>\n",
              "      <td>2020-09-15</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Full Issue PDF.</td>\n",
              "      <td>JACC. CardioOncology</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2666-0873(20)30180-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7492056?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10767</th>\n",
              "      <td>PMC8459260</td>\n",
              "      <td>2021-02-25</td>\n",
              "      <td>2022-01-25</td>\n",
              "      <td>Getting in touch with your senses: Mechanisms ...</td>\n",
              "      <td>WIREs mechanisms of disease</td>\n",
              "      <td>Gupta S, Butler SJ.</td>\n",
              "      <td>10.1002/wsbm.1520</td>\n",
              "      <td>https://europepmc.org/articles/PMC8459260?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11329</th>\n",
              "      <td>PMC7163523</td>\n",
              "      <td>2019-08-01</td>\n",
              "      <td>2021-02-16</td>\n",
              "      <td>Literature review of baseline information on n...</td>\n",
              "      <td>EFSA Supporting Publications</td>\n",
              "      <td>Dávalos A, Henriques R, Latasa M, Laparra M, C...</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7163523?pdf=...</td>\n",
              "      <td>None</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>118 rows × 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-20f13e20-5920-455a-88ad-2782d08978fa')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-20f13e20-5920-455a-88ad-2782d08978fa button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-20f13e20-5920-455a-88ad-2782d08978fa');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 38
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_full_text_grobid = pmc_search_results_no_full_text.copy()"
      ],
      "metadata": {
        "id": "4TPUR7Svs2H1"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_stripped_markup_articles_exceptions_118_v2.pickle', 'rb') as f:\n",
        "    stripped_markup_articles = pickle.load(f)"
      ],
      "metadata": {
        "id": "fstAme-Fs2H1"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "len(stripped_markup_articles)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "a063ea95-a478-4afd-bb66-fb55ffefd947",
        "id": "3bhZw0qAs2H1"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "118"
            ]
          },
          "metadata": {},
          "execution_count": 41
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Add full text with stripped markup as 'text' column to DataFrame."
      ],
      "metadata": {
        "id": "0wtTuPRFKHcc"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_full_text_grobid['text'] = stripped_markup_articles"
      ],
      "metadata": {
        "id": "4CM-d_-Rs2H2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_full_text_grobid"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "outputId": "69c3ba54-f6a4-4c35-cf8e-70df8abc594a",
        "id": "ctXpeH2Us2H2"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "            pmcid   published     revised  \\\n",
              "64     PMC9538661  2022-10-13  2022-11-22   \n",
              "188    PMC9794394  2022-12-28  2023-01-02   \n",
              "212    PMC9538837  2022-10-10  2022-11-22   \n",
              "256    PMC9788990  2022-12-24  2023-01-02   \n",
              "275    PMC9794516  2022-12-28  2023-01-02   \n",
              "...           ...         ...         ...   \n",
              "9797   PMC7162151  2020-02-28  2020-04-21   \n",
              "9809   PMC9094125  2022-05-11  2022-07-16   \n",
              "10399  PMC7492056  2020-09-15  2020-09-28   \n",
              "10767  PMC8459260  2021-02-25  2022-01-25   \n",
              "11329  PMC7163523  2019-08-01  2021-02-16   \n",
              "\n",
              "                                                   title  \\\n",
              "64     Recent Drug Development and Medicinal Chemistr...   \n",
              "188    Nonstructural protein 1 (nsp1) widespread RNA ...   \n",
              "212    The Efficacy of Traditional Medicinal Plants i...   \n",
              "256    Sleep and circadian rhythm disruption alters t...   \n",
              "275    MultiOMICs landscape of SARS-CoV-2-induced hos...   \n",
              "...                                                  ...   \n",
              "9797                                               News.   \n",
              "9809   Recent advances in metal-organic framework-bas...   \n",
              "10399                                    Full Issue PDF.   \n",
              "10767  Getting in touch with your senses: Mechanisms ...   \n",
              "11329  Literature review of baseline information on n...   \n",
              "\n",
              "                            journal  \\\n",
              "64                      ChemMedChem   \n",
              "188                        iScience   \n",
              "212        Chemistry & biodiversity   \n",
              "256                        iScience   \n",
              "275                        iScience   \n",
              "...                             ...   \n",
              "9797           Chemistry & industry   \n",
              "9809                  Nano research   \n",
              "10399          JACC. CardioOncology   \n",
              "10767   WIREs mechanisms of disease   \n",
              "11329  EFSA Supporting Publications   \n",
              "\n",
              "                                                 authors  \\\n",
              "64         Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.   \n",
              "188                       Bermudez Y, Miles J, Muller M.   \n",
              "212                      Choe J, Har Yong P, Xiang Ng Z.   \n",
              "256    Taylor L, Von Lendenfeld F, Ashton A, Sanghani...   \n",
              "275    Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...   \n",
              "...                                                  ...   \n",
              "9797                                                   0   \n",
              "9809             Yang M, Zhang J, Wei Y, Zhang J, Tao C.   \n",
              "10399                                                  0   \n",
              "10767                                Gupta S, Butler SJ.   \n",
              "11329  Dávalos A, Henriques R, Latasa M, Laparra M, C...   \n",
              "\n",
              "                                 doi  \\\n",
              "64            10.1002/cmdc.202200440   \n",
              "188       10.1016/j.isci.2022.105887   \n",
              "212           10.1002/cbdv.202200655   \n",
              "256       10.1016/j.isci.2022.105877   \n",
              "275       10.1016/j.isci.2022.105895   \n",
              "...                              ...   \n",
              "9797            10.1002/cind.842_3.x   \n",
              "9809       10.1007/s12274-022-4302-x   \n",
              "10399  10.1016/s2666-0873(20)30180-0   \n",
              "10767              10.1002/wsbm.1520   \n",
              "11329                              0   \n",
              "\n",
              "                                                 pdf_url  \\\n",
              "64     https://europepmc.org/articles/PMC9538661?pdf=...   \n",
              "188    https://europepmc.org/articles/PMC9794394?pdf=...   \n",
              "212    https://europepmc.org/articles/PMC9538837?pdf=...   \n",
              "256    https://europepmc.org/articles/PMC9788990?pdf=...   \n",
              "275    https://europepmc.org/articles/PMC9794516?pdf=...   \n",
              "...                                                  ...   \n",
              "9797   https://europepmc.org/articles/PMC7162151?pdf=...   \n",
              "9809   https://europepmc.org/articles/PMC9094125?pdf=...   \n",
              "10399  https://europepmc.org/articles/PMC7492056?pdf=...   \n",
              "10767  https://europepmc.org/articles/PMC8459260?pdf=...   \n",
              "11329  https://europepmc.org/articles/PMC7163523?pdf=...   \n",
              "\n",
              "                                                    text  \n",
              "64     \\nSevere acute respiratory syndrome coronaviru...  \n",
              "188    \\nThe past 20 years have seen the emergence of...  \n",
              "212    \\nCoronavirus disease (Covid- 19) is a human r...  \n",
              "256    \\n\\n\\nJ o u r n a l P r e -p r o o f\\nRespirat...  \n",
              "275    \\nThe rapid emergence of the COVID-19 pandemic...  \n",
              "...                                                  ...  \n",
              "9797   \\n\\n\\nRestricting the amount of the amino acid...  \n",
              "9809                            \\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n  \n",
              "10399  \\nT he survival of children with cancer has co...  \n",
              "10767  \\nSomatosensation is essential for survival. I...  \n",
              "11329  \\nThis part provides baseline information on t...  \n",
              "\n",
              "[118 rows x 9 columns]"
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-5e108476-e1d7-4d0f-9661-d11909037791\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>64</th>\n",
              "      <td>PMC9538661</td>\n",
              "      <td>2022-10-13</td>\n",
              "      <td>2022-11-22</td>\n",
              "      <td>Recent Drug Development and Medicinal Chemistr...</td>\n",
              "      <td>ChemMedChem</td>\n",
              "      <td>Ghosh AK, Mishevich JL, Mesecar A, Mitsuya H.</td>\n",
              "      <td>10.1002/cmdc.202200440</td>\n",
              "      <td>https://europepmc.org/articles/PMC9538661?pdf=...</td>\n",
              "      <td>\\nSevere acute respiratory syndrome coronaviru...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>188</th>\n",
              "      <td>PMC9794394</td>\n",
              "      <td>2022-12-28</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>Nonstructural protein 1 (nsp1) widespread RNA ...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Bermudez Y, Miles J, Muller M.</td>\n",
              "      <td>10.1016/j.isci.2022.105887</td>\n",
              "      <td>https://europepmc.org/articles/PMC9794394?pdf=...</td>\n",
              "      <td>\\nThe past 20 years have seen the emergence of...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>212</th>\n",
              "      <td>PMC9538837</td>\n",
              "      <td>2022-10-10</td>\n",
              "      <td>2022-11-22</td>\n",
              "      <td>The Efficacy of Traditional Medicinal Plants i...</td>\n",
              "      <td>Chemistry &amp; biodiversity</td>\n",
              "      <td>Choe J, Har Yong P, Xiang Ng Z.</td>\n",
              "      <td>10.1002/cbdv.202200655</td>\n",
              "      <td>https://europepmc.org/articles/PMC9538837?pdf=...</td>\n",
              "      <td>\\nCoronavirus disease (Covid- 19) is a human r...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>256</th>\n",
              "      <td>PMC9788990</td>\n",
              "      <td>2022-12-24</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>Sleep and circadian rhythm disruption alters t...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Taylor L, Von Lendenfeld F, Ashton A, Sanghani...</td>\n",
              "      <td>10.1016/j.isci.2022.105877</td>\n",
              "      <td>https://europepmc.org/articles/PMC9788990?pdf=...</td>\n",
              "      <td>\\n\\n\\nJ o u r n a l P r e -p r o o f\\nRespirat...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>275</th>\n",
              "      <td>PMC9794516</td>\n",
              "      <td>2022-12-28</td>\n",
              "      <td>2023-01-02</td>\n",
              "      <td>MultiOMICs landscape of SARS-CoV-2-induced hos...</td>\n",
              "      <td>iScience</td>\n",
              "      <td>Pinto SM, Subbannayya Y, Kim H, Hagen L, Górna...</td>\n",
              "      <td>10.1016/j.isci.2022.105895</td>\n",
              "      <td>https://europepmc.org/articles/PMC9794516?pdf=...</td>\n",
              "      <td>\\nThe rapid emergence of the COVID-19 pandemic...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>...</th>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "      <td>...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9797</th>\n",
              "      <td>PMC7162151</td>\n",
              "      <td>2020-02-28</td>\n",
              "      <td>2020-04-21</td>\n",
              "      <td>News.</td>\n",
              "      <td>Chemistry &amp; industry</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1002/cind.842_3.x</td>\n",
              "      <td>https://europepmc.org/articles/PMC7162151?pdf=...</td>\n",
              "      <td>\\n\\n\\nRestricting the amount of the amino acid...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9809</th>\n",
              "      <td>PMC9094125</td>\n",
              "      <td>2022-05-11</td>\n",
              "      <td>2022-07-16</td>\n",
              "      <td>Recent advances in metal-organic framework-bas...</td>\n",
              "      <td>Nano research</td>\n",
              "      <td>Yang M, Zhang J, Wei Y, Zhang J, Tao C.</td>\n",
              "      <td>10.1007/s12274-022-4302-x</td>\n",
              "      <td>https://europepmc.org/articles/PMC9094125?pdf=...</td>\n",
              "      <td>\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n\\n</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10399</th>\n",
              "      <td>PMC7492056</td>\n",
              "      <td>2020-09-15</td>\n",
              "      <td>2020-09-28</td>\n",
              "      <td>Full Issue PDF.</td>\n",
              "      <td>JACC. CardioOncology</td>\n",
              "      <td>0</td>\n",
              "      <td>10.1016/s2666-0873(20)30180-0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7492056?pdf=...</td>\n",
              "      <td>\\nT he survival of children with cancer has co...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10767</th>\n",
              "      <td>PMC8459260</td>\n",
              "      <td>2021-02-25</td>\n",
              "      <td>2022-01-25</td>\n",
              "      <td>Getting in touch with your senses: Mechanisms ...</td>\n",
              "      <td>WIREs mechanisms of disease</td>\n",
              "      <td>Gupta S, Butler SJ.</td>\n",
              "      <td>10.1002/wsbm.1520</td>\n",
              "      <td>https://europepmc.org/articles/PMC8459260?pdf=...</td>\n",
              "      <td>\\nSomatosensation is essential for survival. I...</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11329</th>\n",
              "      <td>PMC7163523</td>\n",
              "      <td>2019-08-01</td>\n",
              "      <td>2021-02-16</td>\n",
              "      <td>Literature review of baseline information on n...</td>\n",
              "      <td>EFSA Supporting Publications</td>\n",
              "      <td>Dávalos A, Henriques R, Latasa M, Laparra M, C...</td>\n",
              "      <td>0</td>\n",
              "      <td>https://europepmc.org/articles/PMC7163523?pdf=...</td>\n",
              "      <td>\\nThis part provides baseline information on t...</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "<p>118 rows × 9 columns</p>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-5e108476-e1d7-4d0f-9661-d11909037791')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-5e108476-e1d7-4d0f-9661-d11909037791 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-5e108476-e1d7-4d0f-9661-d11909037791');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 43
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open('2023-01-06_pmc_search_results_full_text_grobid_v2.pickle', 'wb') as f:\n",
        "  pickle.dump(pmc_search_results_full_text_grobid, f)"
      ],
      "metadata": {
        "id": "w9T5BSpns2H2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 12. Check for missing text\n",
        "\n",
        "Concise summary of DataFrame to see if there are any articles with missing text."
      ],
      "metadata": {
        "id": "YxKSErJBK2Qz"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_full_text_grobid.info()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DbRSN87DxAJi",
        "outputId": "f4eb54ed-3b39-41ae-c9df-96cf81116a25"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "<class 'pandas.core.frame.DataFrame'>\n",
            "Int64Index: 118 entries, 64 to 11329\n",
            "Data columns (total 9 columns):\n",
            " #   Column     Non-Null Count  Dtype \n",
            "---  ------     --------------  ----- \n",
            " 0   pmcid      118 non-null    object\n",
            " 1   published  118 non-null    object\n",
            " 2   revised    118 non-null    object\n",
            " 3   title      118 non-null    object\n",
            " 4   journal    118 non-null    object\n",
            " 5   authors    118 non-null    object\n",
            " 6   doi        118 non-null    object\n",
            " 7   pdf_url    118 non-null    object\n",
            " 8   text       118 non-null    object\n",
            "dtypes: object(9)\n",
            "memory usage: 9.2+ KB\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "No null values for any of the 118 exceptions but we can try checking for an empty string in the text column."
      ],
      "metadata": {
        "id": "xd1AR4B1La5B"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_full_text_grobid[pmc_search_results_full_text_grobid['text']=='']"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 214
        },
        "id": "zXSAkkbtw5Gf",
        "outputId": "81bf0560-f165-4a15-eb35-11dba2d22d54"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "           pmcid   published     revised  \\\n",
              "1305  PMC8010379  2021-03-31  2022-11-08   \n",
              "\n",
              "                                                  title  \\\n",
              "1305  Repurposing antiviral drugs on recently emerge...   \n",
              "\n",
              "                           journal  \\\n",
              "1305  Materials today. Proceedings   \n",
              "\n",
              "                                                authors  \\\n",
              "1305  Swathi K, Nikitha B, Chandrakala B, Lakshmanad...   \n",
              "\n",
              "                              doi  \\\n",
              "1305  10.1016/j.matpr.2021.03.143   \n",
              "\n",
              "                                                pdf_url text  \n",
              "1305  https://europepmc.org/articles/PMC8010379?pdf=...       "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-3da42e83-c39d-4d9b-bf47-ec82b2d989e5\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>pmcid</th>\n",
              "      <th>published</th>\n",
              "      <th>revised</th>\n",
              "      <th>title</th>\n",
              "      <th>journal</th>\n",
              "      <th>authors</th>\n",
              "      <th>doi</th>\n",
              "      <th>pdf_url</th>\n",
              "      <th>text</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>1305</th>\n",
              "      <td>PMC8010379</td>\n",
              "      <td>2021-03-31</td>\n",
              "      <td>2022-11-08</td>\n",
              "      <td>Repurposing antiviral drugs on recently emerge...</td>\n",
              "      <td>Materials today. Proceedings</td>\n",
              "      <td>Swathi K, Nikitha B, Chandrakala B, Lakshmanad...</td>\n",
              "      <td>10.1016/j.matpr.2021.03.143</td>\n",
              "      <td>https://europepmc.org/articles/PMC8010379?pdf=...</td>\n",
              "      <td></td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-3da42e83-c39d-4d9b-bf47-ec82b2d989e5')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-3da42e83-c39d-4d9b-bf47-ec82b2d989e5 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-3da42e83-c39d-4d9b-bf47-ec82b2d989e5');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 56
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "pmc_search_results_full_text_grobid.loc[pmc_search_results_full_text_grobid['pmcid'] == 'PMC8010379'].iloc[0]"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "76b4CQ12_oF4",
        "outputId": "086c3405-b542-42ef-ec7d-95d60a5e238b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "pmcid                                               PMC8010379\n",
              "published                                           2021-03-31\n",
              "revised                                             2022-11-08\n",
              "title        Repurposing antiviral drugs on recently emerge...\n",
              "journal                           Materials today. Proceedings\n",
              "authors      Swathi K, Nikitha B, Chandrakala B, Lakshmanad...\n",
              "doi                                10.1016/j.matpr.2021.03.143\n",
              "pdf_url      https://europepmc.org/articles/PMC8010379?pdf=...\n",
              "text                                                          \n",
              "Name: 1305, dtype: object"
            ]
          },
          "metadata": {},
          "execution_count": 10
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "One article was withdrawn hence the empty string for the text column."
      ],
      "metadata": {
        "id": "juCQno73OW5l"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "### References\n",
        "\n",
        "* Europe PMC https://europepmc.org/\n",
        "\n",
        "* Articles RESTful API https://europepmc.org/RestfulWebService\n",
        "\n",
        "* Full list of publication types https://europepmc.org/advancesearch\n",
        "\n",
        "* Search syntax reference https://europepmc.org/searchsyntax\n",
        "\n",
        "* Web Service Reference https://europepmc.org/docs/EBI_Europe_PMC_Web_Service_Reference.pdf\n",
        "\n",
        "* University of Virginia Claude Moore Health Sciences Library project https://github.com/carrlucy/HSL_OA/blob/main/streamlit_app.py\n",
        "\n",
        "\n",
        "* Beautiful Soup documentation https://www.crummy.com/software/BeautifulSoup/bs4/doc/\n",
        "\n",
        "* GROBID https://github.com/kermitt2/grobid/\n"
      ],
      "metadata": {
        "id": "hndnmRmi7cXB"
      }
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "toc_visible": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}